{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "d001e5b1",
   "metadata": {},
   "outputs": [],
   "source": [
    "import chromatin_algos\n",
    "import evaluator"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "4328129e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0\n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "BINS  IntervalIndex([(3000016, 3000016], (3000016, 3000116], (3000116, 3000216],\n",
      "               (3000216, 3000316], (3000316, 3000416], (3000416, 3000516],\n",
      "               (3000516, 3000616], (3000616, 3000716], (3000716, 3000816],\n",
      "               (3000816, 3000916],\n",
      "               ...\n",
      "               (9716616, 9716716], (9716716, 9716816], (9716816, 9716916],\n",
      "               (9716916, 9717016], (9717016, 9717116], (9717116, 9717216],\n",
      "               (9717216, 9717316], (9717316, 9717416], (9717416, 9717516],\n",
      "               (9717516, 9717593]],\n",
      "              dtype='interval[int64, right]', length=67177)\n",
      "HEAD with BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0   \n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "\n",
      "         bin_offset_0  \n",
      "0  (3000016, 3000116]  \n",
      "1  (3000016, 3000116]  \n",
      "2  (3000016, 3000116]  \n",
      "3  (3000016, 3000116]  \n",
      "4  (3000016, 3000116]  \n",
      "TAIL with BINS\n",
      "       seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  9717555  9717555  18_9717555    1.0    0.0    0.0   \n",
      "699996       19  9717571  9717571  19_9717571    0.0    0.0    0.0   \n",
      "699997       19  9717581  9717581  19_9717581    0.0    0.0    0.0   \n",
      "699998       18  9717582  9717582  18_9717582    0.0    0.0    0.0   \n",
      "699999       18  9717593  9717593  18_9717593    0.0    0.0    0.0   \n",
      "\n",
      "              bin_offset_0  \n",
      "699995  (9717516, 9717593]  \n",
      "699996  (9717516, 9717593]  \n",
      "699997  (9717516, 9717593]  \n",
      "699998  (9717516, 9717593]  \n",
      "699999  (9717516, 9717593]  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:136: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "          bin_offset_0  TT_S0  starting_nt\n",
      "0   (3000016, 3000016]    0.0      3000016\n",
      "1   (3000016, 3000116]    7.0      3000016\n",
      "2   (3000116, 3000216]    1.0      3000116\n",
      "3   (3000216, 3000316]    2.0      3000216\n",
      "4   (3000316, 3000416]    0.0      3000316\n",
      "5   (3000416, 3000516]    0.0      3000416\n",
      "6   (3000516, 3000616]    0.0      3000516\n",
      "7   (3000616, 3000716]    0.0      3000616\n",
      "8   (3000716, 3000816]    0.0      3000716\n",
      "9   (3000816, 3000916]    0.0      3000816\n",
      "10  (3000916, 3001016]    0.0      3000916\n",
      "11  (3001016, 3001116]    2.0      3001016\n",
      "12  (3001116, 3001216]    0.0      3001116\n",
      "13  (3001216, 3001316]    0.0      3001216\n",
      "14  (3001316, 3001416]    0.0      3001316\n",
      "15  (3001416, 3001516]    1.0      3001416\n",
      "16  (3001516, 3001616]    3.0      3001516\n",
      "17  (3001616, 3001716]    3.0      3001616\n",
      "18  (3001716, 3001816]    0.0      3001716\n",
      "19  (3001816, 3001916]    0.0      3001816\n",
      "20  (3001916, 3002016]    1.0      3001916\n",
      "21  (3002016, 3002116]    0.0      3002016\n",
      "22  (3002116, 3002216]    1.0      3002116\n",
      "23  (3002216, 3002316]   10.0      3002216\n",
      "24  (3002316, 3002416]    2.0      3002316\n",
      "25  (3002416, 3002516]   19.0      3002416\n",
      "26  (3002516, 3002616]    4.0      3002516\n",
      "27  (3002616, 3002716]    0.0      3002616\n",
      "28  (3002716, 3002816]    0.0      3002716\n",
      "29  (3002816, 3002916]    0.0      3002816\n",
      "30  (3002916, 3003016]    1.0      3002916\n",
      "31  (3003016, 3003116]    2.0      3003016\n",
      "32  (3003116, 3003216]    0.0      3003116\n",
      "33  (3003216, 3003316]    0.0      3003216\n",
      "34  (3003316, 3003416]    1.0      3003316\n",
      "35  (3003416, 3003516]    0.0      3003416\n",
      "36  (3003516, 3003616]    1.0      3003516\n",
      "37  (3003616, 3003716]    1.0      3003616\n",
      "38  (3003716, 3003816]    3.0      3003716\n",
      "39  (3003816, 3003916]    0.0      3003816\n",
      "40  (3003916, 3004016]    8.0      3003916\n",
      "41  (3004016, 3004116]    0.0      3004016\n",
      "42  (3004116, 3004216]    0.0      3004116\n",
      "43  (3004216, 3004316]    5.0      3004216\n",
      "44  (3004316, 3004416]    0.0      3004316\n",
      "45  (3004416, 3004516]    0.0      3004416\n",
      "46  (3004516, 3004616]    1.0      3004516\n",
      "47  (3004616, 3004716]    0.0      3004616\n",
      "48  (3004716, 3004816]    0.0      3004716\n",
      "49  (3004816, 3004916]   13.0      3004816\n",
      "(67177, 3)\n",
      "bin_offset_0    category\n",
      "TT_S0            float64\n",
      "starting_nt        int64\n",
      "dtype: object\n",
      "sumos ilgis: 67177\n",
      "ranges ilgis: 67177\n",
      "Model input - NO SCALE\n",
      "       TT_S0  genes  transcripts  exons  CDSs\n",
      "0        0.0      0            0      0     0\n",
      "1        7.0      0            0      0     0\n",
      "2        1.0      0            0      0     0\n",
      "3        2.0      0            0      0     0\n",
      "4        0.0      0            0      0     0\n",
      "...      ...    ...          ...    ...   ...\n",
      "67172    0.0      0            0      0     0\n",
      "67173    0.0      0            0      0     0\n",
      "67174    0.0      0            0      0     0\n",
      "67175    1.0      0            0      0     0\n",
      "67176    1.0      0            0      0     0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "       TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0        0.0             0                   0             0            0\n",
      "1        7.0             0                   0             0            0\n",
      "2        1.0             0                   0             0            0\n",
      "3        2.0             0                   0             0            0\n",
      "4        0.0             0                   0             0            0\n",
      "...      ...           ...                 ...           ...          ...\n",
      "67172    0.0             0                   0             0            0\n",
      "67173    0.0             0                   0             0            0\n",
      "67174    0.0             0                   0             0            0\n",
      "67175    1.0             0                   0             0            0\n",
      "67176    1.0             0                   0             0            0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "0.052309570239814224\n",
      "[0 2 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 2 0 7 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 2 0 0 2 0 0 0 0 0 6 0 2 9 9 2 7 6 3 3 8 8 0 6 6 0 0 0 2 0 0 2 0 0 2\n",
      " 0 0 0 0 0 0 0 2 0 0 0 0 0 0 2 0 0 6 6 0 0 0 0 0 0 2 6 6 1 7 6 2 2 2 1 0 6\n",
      " 0 6 0 0 0 0 0 0 0 0 0 2 0 0 6 0 6 6 6 6 0 2 2 0 0 2 2 0 0 0 0 0 0 0 0 0 0\n",
      " 6 0 2 0 0 0 2 0 6 0 0 0 6 0 6 0 0 2 6 0 0 0 0 0 0 0 0 0 0 0 0 0 0 2 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 6 7 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 2 0 0 0 0 0 0 0 0 0 0 0 0 0 0 2 2 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 2 0 0 0 6 0 0 0 0 0 0 0 0 0 6 6 0 6 2 6 2 0 0 0 2 2\n",
      " 0 0 2 0 0 0 6 2 6 0 2 2 0 0 0 0 0 0 2 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 2 0 0 0 0 0 0 0 0 0 0 0 0 0 0 2 0 0 0 2 2 0 0 0 2 2 0 2 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 6 2 0 0 0 2 0 0 2 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 2 0 0 0 0 0 2 0 0 0 0 2\n",
      " 0 0 0 0 0 0 0 0 0 0 2 0 0 0 2 0 0 0 0 0 0 2 0 0 0 0 0 0 0 0 2 0 0 2 0 0 2\n",
      " 0 2 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 2 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 6 0 0 2 0 0 0 0 2 2 2 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 2 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 2 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 2 0 0 0 0 2 0 0 0 2\n",
      " 2 0 2 2 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 6 0 0 0 0 0 2 0 0 0 0 0 0 0\n",
      " 0]\n",
      "threshold density 0.15152264201500534\n",
      "cluster signal 40053.0\n",
      "cluster length 2751100\n",
      "!!cluster density 0.014558903711242776\n",
      "cluster signal 117505.0\n",
      "cluster length 1623600\n",
      "!!cluster density 0.07237312145848732\n",
      "cluster signal 119719.0\n",
      "cluster length 580500\n",
      "!!cluster density 0.20623428079242032\n",
      "cluster signal 148606.0\n",
      "cluster length 1088200\n",
      "!!cluster density 0.1365612938798015\n",
      "cluster signal 76216.0\n",
      "cluster length 197600\n",
      "!!cluster density 0.3857085020242915\n",
      "cluster signal 9968.0\n",
      "cluster length 10200\n",
      "!!cluster density 0.9772549019607844\n",
      "cluster signal 678.0\n",
      "cluster length 200\n",
      "!!cluster density 3.39\n",
      "cluster signal 99791.0\n",
      "cluster length 351400\n",
      "!!cluster density 0.28398121798520204\n",
      "cluster signal 42357.0\n",
      "cluster length 81200\n",
      "!!cluster density 0.5216379310344827\n",
      "cluster signal 23686.0\n",
      "cluster length 33700\n",
      "!!cluster density 0.7028486646884273\n",
      "\n",
      "Labels mapping\n",
      " {0: 0, 2: 0, 7: 1, 6: 0, 9: 1, 3: 1, 8: 1, 1: 1, 4: 1, 5: 1}\n",
      "DEBUG: 54629 12548 \n",
      "\n",
      " +--------------------+-------------+---------------+--------------+-----------+-----------+-----------+---------------+-----------+-----------+-----------+\n",
      "| bin_offset_0       | TT_S0       | starting_nt   | Chromosome   | Start     | End       | genes     | transcripts   | exons     | CDSs      | labels    |\n",
      "| (category)         | (float64)   | (int64)       | (category)   | (int64)   | (int64)   | (int64)   | (int64)       | (int64)   | (int64)   | (int64)   |\n",
      "|--------------------+-------------+---------------+--------------+-----------+-----------+-----------+---------------+-----------+-----------+-----------|\n",
      "| (3000016, 3000016] | 0.0         | 3000016       | 18           | 3000016   | 3000016   | 0         | 0             | 0         | 0         | 0         |\n",
      "| (3000016, 3000116] | 7.0         | 3000016       | 18           | 3000016   | 3000116   | 0         | 0             | 0         | 0         | 0         |\n",
      "| (3000116, 3000216] | 1.0         | 3000116       | 18           | 3000116   | 3000216   | 0         | 0             | 0         | 0         | 0         |\n",
      "| (3000216, 3000316] | 2.0         | 3000216       | 18           | 3000216   | 3000316   | 0         | 0             | 0         | 0         | 0         |\n",
      "| ...                | ...         | ...           | ...          | ...       | ...       | ...       | ...           | ...       | ...       | ...       |\n",
      "| (9717216, 9717316] | 0.0         | 9717216       | 18           | 9717216   | 9717316   | 0         | 0             | 0         | 0         | 0         |\n",
      "| (9717316, 9717416] | 0.0         | 9717316       | 18           | 9717316   | 9717416   | 0         | 0             | 0         | 0         | 0         |\n",
      "| (9717416, 9717516] | 1.0         | 9717416       | 18           | 9717416   | 9717516   | 0         | 0             | 0         | 0         | 0         |\n",
      "| (9717516, 9717593] | 1.0         | 9717516       | 18           | 9717516   | 9717593   | 0         | 0             | 0         | 0         | 0         |\n",
      "+--------------------+-------------+---------------+--------------+-----------+-----------+-----------+---------------+-----------+-----------+-----------+\n",
      "Unstranded PyRanges object has 54,629 rows and 11 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "os.environ['COLUMNS']=\"300\"\n",
    "\n",
    "result = evaluator.evaluate(merged_ranges = chromatin_algos.algo5d_aggregated(1, n_clusters=10))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "6201148c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0\n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "BINS  IntervalIndex([(3000016, 3000016], (3000016, 3000116], (3000116, 3000216],\n",
      "               (3000216, 3000316], (3000316, 3000416], (3000416, 3000516],\n",
      "               (3000516, 3000616], (3000616, 3000716], (3000716, 3000816],\n",
      "               (3000816, 3000916],\n",
      "               ...\n",
      "               (9716616, 9716716], (9716716, 9716816], (9716816, 9716916],\n",
      "               (9716916, 9717016], (9717016, 9717116], (9717116, 9717216],\n",
      "               (9717216, 9717316], (9717316, 9717416], (9717416, 9717516],\n",
      "               (9717516, 9717593]],\n",
      "              dtype='interval[int64, right]', length=67177)\n",
      "HEAD with BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0   \n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "\n",
      "         bin_offset_0  \n",
      "0  (3000016, 3000116]  \n",
      "1  (3000016, 3000116]  \n",
      "2  (3000016, 3000116]  \n",
      "3  (3000016, 3000116]  \n",
      "4  (3000016, 3000116]  \n",
      "TAIL with BINS\n",
      "       seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  9717555  9717555  18_9717555    1.0    0.0    0.0   \n",
      "699996       19  9717571  9717571  19_9717571    0.0    0.0    0.0   \n",
      "699997       19  9717581  9717581  19_9717581    0.0    0.0    0.0   \n",
      "699998       18  9717582  9717582  18_9717582    0.0    0.0    0.0   \n",
      "699999       18  9717593  9717593  18_9717593    0.0    0.0    0.0   \n",
      "\n",
      "              bin_offset_0  \n",
      "699995  (9717516, 9717593]  \n",
      "699996  (9717516, 9717593]  \n",
      "699997  (9717516, 9717593]  \n",
      "699998  (9717516, 9717593]  \n",
      "699999  (9717516, 9717593]  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:136: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "          bin_offset_0  TT_S0  starting_nt\n",
      "0   (3000016, 3000016]    0.0      3000016\n",
      "1   (3000016, 3000116]    7.0      3000016\n",
      "2   (3000116, 3000216]    1.0      3000116\n",
      "3   (3000216, 3000316]    2.0      3000216\n",
      "4   (3000316, 3000416]    0.0      3000316\n",
      "5   (3000416, 3000516]    0.0      3000416\n",
      "6   (3000516, 3000616]    0.0      3000516\n",
      "7   (3000616, 3000716]    0.0      3000616\n",
      "8   (3000716, 3000816]    0.0      3000716\n",
      "9   (3000816, 3000916]    0.0      3000816\n",
      "10  (3000916, 3001016]    0.0      3000916\n",
      "11  (3001016, 3001116]    2.0      3001016\n",
      "12  (3001116, 3001216]    0.0      3001116\n",
      "13  (3001216, 3001316]    0.0      3001216\n",
      "14  (3001316, 3001416]    0.0      3001316\n",
      "15  (3001416, 3001516]    1.0      3001416\n",
      "16  (3001516, 3001616]    3.0      3001516\n",
      "17  (3001616, 3001716]    3.0      3001616\n",
      "18  (3001716, 3001816]    0.0      3001716\n",
      "19  (3001816, 3001916]    0.0      3001816\n",
      "20  (3001916, 3002016]    1.0      3001916\n",
      "21  (3002016, 3002116]    0.0      3002016\n",
      "22  (3002116, 3002216]    1.0      3002116\n",
      "23  (3002216, 3002316]   10.0      3002216\n",
      "24  (3002316, 3002416]    2.0      3002316\n",
      "25  (3002416, 3002516]   19.0      3002416\n",
      "26  (3002516, 3002616]    4.0      3002516\n",
      "27  (3002616, 3002716]    0.0      3002616\n",
      "28  (3002716, 3002816]    0.0      3002716\n",
      "29  (3002816, 3002916]    0.0      3002816\n",
      "30  (3002916, 3003016]    1.0      3002916\n",
      "31  (3003016, 3003116]    2.0      3003016\n",
      "32  (3003116, 3003216]    0.0      3003116\n",
      "33  (3003216, 3003316]    0.0      3003216\n",
      "34  (3003316, 3003416]    1.0      3003316\n",
      "35  (3003416, 3003516]    0.0      3003416\n",
      "36  (3003516, 3003616]    1.0      3003516\n",
      "37  (3003616, 3003716]    1.0      3003616\n",
      "38  (3003716, 3003816]    3.0      3003716\n",
      "39  (3003816, 3003916]    0.0      3003816\n",
      "40  (3003916, 3004016]    8.0      3003916\n",
      "41  (3004016, 3004116]    0.0      3004016\n",
      "42  (3004116, 3004216]    0.0      3004116\n",
      "43  (3004216, 3004316]    5.0      3004216\n",
      "44  (3004316, 3004416]    0.0      3004316\n",
      "45  (3004416, 3004516]    0.0      3004416\n",
      "46  (3004516, 3004616]    1.0      3004516\n",
      "47  (3004616, 3004716]    0.0      3004616\n",
      "48  (3004716, 3004816]    0.0      3004716\n",
      "49  (3004816, 3004916]   13.0      3004816\n",
      "(67177, 3)\n",
      "bin_offset_0    category\n",
      "TT_S0            float64\n",
      "starting_nt        int64\n",
      "dtype: object\n",
      "sumos ilgis: 67177\n",
      "ranges ilgis: 67177\n",
      "Model input - NO SCALE\n",
      "       TT_S0  genes  transcripts  exons  CDSs\n",
      "0        0.0      0            0      0     0\n",
      "1        7.0      0            0      0     0\n",
      "2        1.0      0            0      0     0\n",
      "3        2.0      0            0      0     0\n",
      "4        0.0      0            0      0     0\n",
      "...      ...    ...          ...    ...   ...\n",
      "67172    0.0      0            0      0     0\n",
      "67173    0.0      0            0      0     0\n",
      "67174    0.0      0            0      0     0\n",
      "67175    1.0      0            0      0     0\n",
      "67176    1.0      0            0      0     0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "       TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0        0.0             0                   0             0            0\n",
      "1        7.0             0                   0             0            0\n",
      "2        1.0             0                   0             0            0\n",
      "3        2.0             0                   0             0            0\n",
      "4        0.0             0                   0             0            0\n",
      "...      ...           ...                 ...           ...          ...\n",
      "67172    0.0             0                   0             0            0\n",
      "67173    0.0             0                   0             0            0\n",
      "67174    0.0             0                   0             0            0\n",
      "67175    1.0             0                   0             0            0\n",
      "67176    1.0             0                   0             0            0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "0.012534051833216726\n",
      "[6 2 6 6 6 6 6 6 6 6 6 6 6 6 6 6 2 2 6 6 6 6 6 0 6 5 2 6 6 6 6 6 6 6 6 6 6\n",
      " 6 2 6 0 6 6 2 6 6 6 6 6 0 6 2 1 7 2 5 0 8 8 9 9 6 5 5 2 6 6 2 6 6 2 2 6 2\n",
      " 6 6 6 6 6 6 6 0 6 6 2 2 6 2 2 6 6 0 0 6 6 6 2 6 2 0 0 0 3 5 0 0 2 0 3 2 0\n",
      " 6 0 2 6 6 6 6 6 6 6 6 2 6 6 0 6 5 5 0 5 6 2 0 2 6 2 0 2 6 6 6 6 6 6 6 6 2\n",
      " 0 6 0 2 6 6 2 6 0 2 6 6 5 6 0 6 6 0 0 2 6 6 6 6 6 6 6 6 6 6 6 6 2 0 6 2 6\n",
      " 6 2 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 0 5 6 6 6 6 6 6 6 6 6 6 6 6 6\n",
      " 6 6 6 6 6 6 6 6 6 6 6 6 6 6 2 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6\n",
      " 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6\n",
      " 2 6 6 6 6 6 2 6 6 6 6 6 6 2 6 2 2 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6\n",
      " 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6\n",
      " 6 6 6 6 6 6 6 6 6 6 6 2 6 6 6 0 6 6 6 6 6 6 6 6 6 0 5 6 0 0 0 2 2 6 6 2 0\n",
      " 2 6 2 6 6 6 0 2 0 2 0 2 6 2 6 6 6 6 2 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6\n",
      " 2 6 6 6 2 6 6 6 6 6 2 6 6 6 6 6 6 6 2 0 6 6 6 2 2 2 6 2 2 2 2 0 6 6 2 6 6\n",
      " 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6\n",
      " 6 0 2 6 6 6 2 6 6 0 6 2 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6\n",
      " 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6\n",
      " 6 6 6 6 2 6 6 2 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 2 6 6 6 6 6 0 6 6 6 6 2\n",
      " 6 6 6 2 6 6 6 6 6 2 0 6 6 2 2 6 6 6 6 6 6 2 6 6 6 6 6 6 6 2 2 6 6 2 2 6 2\n",
      " 6 2 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 2 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6\n",
      " 6 6 6 6 6 6 6 6 6 6 6 6 6 2 6 0 2 6 2 6 6 6 6 6 6 6 6 6 6 2 6 6 6 6 6 6 6\n",
      " 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 2 6 6 6 6 6 2 6 6 6\n",
      " 6 6 6 6 6 6 6 6 6 6 6 6 6 6 2 6 6 6 6 6 0 6 6 2 6 6 6 6 0 2 0 6 2 6 6 2 6\n",
      " 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 2 2 2 6 6 2 6 6 6 6 6 6 6 6\n",
      " 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 2 6 6 6 6 2 6 2 6 6 6 6 6 6 6 6 6 6\n",
      " 6 6 2 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 2 2 0 6 6 6 6 0 2 6 6 2\n",
      " 2 6 2 2 6 2 6 6 6 6 6 6 6 2 2 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 2 2 6\n",
      " 6 6 6 6 6 6 2 6 2 6 6 6 6 6 6 6 6 6 6 6 6 6 6 0 2 6 6 2 6 2 2 6 6 6 6 6 6\n",
      " 6]\n",
      "threshold density 0.15152264201500534\n",
      "cluster signal 13441.0\n",
      "cluster length 1985500\n",
      "!!cluster density 0.006769579451019895\n",
      "cluster signal 81365.0\n",
      "cluster length 1687200\n",
      "!!cluster density 0.048224869606448555\n",
      "cluster signal 170537.0\n",
      "cluster length 1542700\n",
      "!!cluster density 0.11054449990276788\n",
      "cluster signal 145876.0\n",
      "cluster length 766900\n",
      "!!cluster density 0.19021515191028818\n",
      "cluster signal 42683.0\n",
      "cluster length 84200\n",
      "!!cluster density 0.5069239904988123\n",
      "cluster signal 80542.0\n",
      "cluster length 214200\n",
      "!!cluster density 0.3760130718954248\n",
      "cluster signal 11221.0\n",
      "cluster length 11700\n",
      "!!cluster density 0.959059829059829\n",
      "cluster signal 678.0\n",
      "cluster length 200\n",
      "!!cluster density 3.39\n",
      "cluster signal 105809.0\n",
      "cluster length 386300\n",
      "!!cluster density 0.27390370178617657\n",
      "cluster signal 26427.0\n",
      "cluster length 38800\n",
      "!!cluster density 0.6811082474226804\n",
      "\n",
      "Labels mapping\n",
      " {6: 0, 2: 0, 0: 0, 5: 1, 1: 1, 7: 1, 8: 1, 9: 1, 3: 1, 4: 1}\n",
      "DEBUG: 52154 15023 \n",
      "\n",
      " +--------------------+-------------+---------------+--------------+-------+\n",
      "| bin_offset_0       | TT_S0       | starting_nt   | Chromosome   | +7    |\n",
      "| (category)         | (float64)   | (int64)       | (category)   | ...   |\n",
      "|--------------------+-------------+---------------+--------------+-------|\n",
      "| (3000016, 3000016] | 0.0         | 3000016       | 18           | ...   |\n",
      "| (3000016, 3000116] | 7.0         | 3000016       | 18           | ...   |\n",
      "| (3000116, 3000216] | 1.0         | 3000116       | 18           | ...   |\n",
      "| (3000216, 3000316] | 2.0         | 3000216       | 18           | ...   |\n",
      "| ...                | ...         | ...           | ...          | ...   |\n",
      "| (9717216, 9717316] | 0.0         | 9717216       | 18           | ...   |\n",
      "| (9717316, 9717416] | 0.0         | 9717316       | 18           | ...   |\n",
      "| (9717416, 9717516] | 1.0         | 9717416       | 18           | ...   |\n",
      "| (9717516, 9717593] | 1.0         | 9717516       | 18           | ...   |\n",
      "+--------------------+-------------+---------------+--------------+-------+\n",
      "Unstranded PyRanges object has 52,154 rows and 11 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "7 hidden columns: Start, End, genes, transcripts, exons, CDSs, labels\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0\n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "BINS  IntervalIndex([(3000016, 3000021], (3000021, 3000121], (3000121, 3000221],\n",
      "               (3000221, 3000321], (3000321, 3000421], (3000421, 3000521],\n",
      "               (3000521, 3000621], (3000621, 3000721], (3000721, 3000821],\n",
      "               (3000821, 3000921],\n",
      "               ...\n",
      "               (9716621, 9716721], (9716721, 9716821], (9716821, 9716921],\n",
      "               (9716921, 9717021], (9717021, 9717121], (9717121, 9717221],\n",
      "               (9717221, 9717321], (9717321, 9717421], (9717421, 9717521],\n",
      "               (9717521, 9717593]],\n",
      "              dtype='interval[int64, right]', length=67177)\n",
      "HEAD with BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0   \n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "\n",
      "         bin_offset_5  \n",
      "0  (3000016, 3000021]  \n",
      "1  (3000016, 3000021]  \n",
      "2  (3000021, 3000121]  \n",
      "3  (3000021, 3000121]  \n",
      "4  (3000021, 3000121]  \n",
      "TAIL with BINS\n",
      "       seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  9717555  9717555  18_9717555    1.0    0.0    0.0   \n",
      "699996       19  9717571  9717571  19_9717571    0.0    0.0    0.0   \n",
      "699997       19  9717581  9717581  19_9717581    0.0    0.0    0.0   \n",
      "699998       18  9717582  9717582  18_9717582    0.0    0.0    0.0   \n",
      "699999       18  9717593  9717593  18_9717593    0.0    0.0    0.0   \n",
      "\n",
      "              bin_offset_5  \n",
      "699995  (9717521, 9717593]  \n",
      "699996  (9717521, 9717593]  \n",
      "699997  (9717521, 9717593]  \n",
      "699998  (9717521, 9717593]  \n",
      "699999  (9717521, 9717593]  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:136: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "          bin_offset_5  TT_S0 starting_nt\n",
      "0   (3000016, 3000021]    0.0     3000016\n",
      "1   (3000021, 3000121]    7.0     3000021\n",
      "2   (3000121, 3000221]    1.0     3000121\n",
      "3   (3000221, 3000321]    2.0     3000221\n",
      "4   (3000321, 3000421]    0.0     3000321\n",
      "5   (3000421, 3000521]    0.0     3000421\n",
      "6   (3000521, 3000621]    0.0     3000521\n",
      "7   (3000621, 3000721]    0.0     3000621\n",
      "8   (3000721, 3000821]    0.0     3000721\n",
      "9   (3000821, 3000921]    0.0     3000821\n",
      "10  (3000921, 3001021]    0.0     3000921\n",
      "11  (3001021, 3001121]    2.0     3001021\n",
      "12  (3001121, 3001221]    0.0     3001121\n",
      "13  (3001221, 3001321]    0.0     3001221\n",
      "14  (3001321, 3001421]    0.0     3001321\n",
      "15  (3001421, 3001521]    1.0     3001421\n",
      "16  (3001521, 3001621]    3.0     3001521\n",
      "17  (3001621, 3001721]    3.0     3001621\n",
      "18  (3001721, 3001821]    0.0     3001721\n",
      "19  (3001821, 3001921]    0.0     3001821\n",
      "20  (3001921, 3002021]    1.0     3001921\n",
      "21  (3002021, 3002121]    0.0     3002021\n",
      "22  (3002121, 3002221]    1.0     3002121\n",
      "23  (3002221, 3002321]   10.0     3002221\n",
      "24  (3002321, 3002421]   12.0     3002321\n",
      "25  (3002421, 3002521]    9.0     3002421\n",
      "26  (3002521, 3002621]    4.0     3002521\n",
      "27  (3002621, 3002721]    0.0     3002621\n",
      "28  (3002721, 3002821]    0.0     3002721\n",
      "29  (3002821, 3002921]    0.0     3002821\n",
      "30  (3002921, 3003021]    1.0     3002921\n",
      "31  (3003021, 3003121]    2.0     3003021\n",
      "32  (3003121, 3003221]    0.0     3003121\n",
      "33  (3003221, 3003321]    0.0     3003221\n",
      "34  (3003321, 3003421]    1.0     3003321\n",
      "35  (3003421, 3003521]    0.0     3003421\n",
      "36  (3003521, 3003621]    1.0     3003521\n",
      "37  (3003621, 3003721]    3.0     3003621\n",
      "38  (3003721, 3003821]    1.0     3003721\n",
      "39  (3003821, 3003921]    0.0     3003821\n",
      "40  (3003921, 3004021]    8.0     3003921\n",
      "41  (3004021, 3004121]    0.0     3004021\n",
      "42  (3004121, 3004221]    0.0     3004121\n",
      "43  (3004221, 3004321]    5.0     3004221\n",
      "44  (3004321, 3004421]    0.0     3004321\n",
      "45  (3004421, 3004521]    0.0     3004421\n",
      "46  (3004521, 3004621]    1.0     3004521\n",
      "47  (3004621, 3004721]    0.0     3004621\n",
      "48  (3004721, 3004821]    0.0     3004721\n",
      "49  (3004821, 3004921]   13.0     3004821\n",
      "(67177, 3)\n",
      "bin_offset_5    category\n",
      "TT_S0            float64\n",
      "starting_nt     category\n",
      "dtype: object\n",
      "sumos ilgis: 67177\n",
      "ranges ilgis: 67177\n",
      "Model input - NO SCALE\n",
      "       TT_S0  genes  transcripts  exons  CDSs\n",
      "0        0.0      0            0      0     0\n",
      "1        7.0      0            0      0     0\n",
      "2        1.0      0            0      0     0\n",
      "3        2.0      0            0      0     0\n",
      "4        0.0      0            0      0     0\n",
      "...      ...    ...          ...    ...   ...\n",
      "67172    0.0      0            0      0     0\n",
      "67173    0.0      0            0      0     0\n",
      "67174    0.0      0            0      0     0\n",
      "67175    1.0      0            0      0     0\n",
      "67176    1.0      0            0      0     0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "       TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0        0.0             0                   0             0            0\n",
      "1        7.0             0                   0             0            0\n",
      "2        1.0             0                   0             0            0\n",
      "3        2.0             0                   0             0            0\n",
      "4        0.0             0                   0             0            0\n",
      "...      ...           ...                 ...           ...          ...\n",
      "67172    0.0             0                   0             0            0\n",
      "67173    0.0             0                   0             0            0\n",
      "67174    0.0             0                   0             0            0\n",
      "67175    1.0             0                   0             0            0\n",
      "67176    1.0             0                   0             0            0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "0.11094571058546825\n",
      "[0 6 0 0 0 0 0 0 0 0 0 0 0 0 0 0 6 6 0 0 0 0 0 3 3 3 6 0 0 0 0 0 0 0 0 0 0\n",
      " 6 0 0 3 0 0 6 0 0 0 0 0 3 0 5 1 2 0 9 3 4 4 7 7 0 1 1 6 0 0 6 0 0 6 0 0 6\n",
      " 0 0 0 0 0 0 0 3 0 0 6 6 0 6 6 0 0 3 1 0 0 0 6 0 6 3 3 3 5 9 1 3 6 3 9 6 3\n",
      " 0 1 6 0 0 0 0 0 0 0 0 6 0 0 1 0 1 1 3 1 0 6 3 6 0 6 3 6 0 0 0 0 0 0 0 0 3\n",
      " 3 0 3 6 0 0 6 6 3 6 0 0 1 0 3 0 0 3 3 6 0 0 0 0 0 0 0 0 0 0 0 0 6 6 0 6 0\n",
      " 0 6 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 3 9 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 6 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 6 0 0 0 0 0 6 0 0 0 0 0 0 6 0 6 6 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 6 0 0 0 3 0 0 0 0 0 0 0 0 0 3 1 0 3 1 3 6 6 0 0 6 3\n",
      " 6 0 6 0 0 0 1 6 1 6 3 6 0 6 0 0 0 0 6 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 6 0 0 0 6 0 0 0 0 0 6 0 0 0 0 0 0 0 6 3 0 0 0 6 6 6 0 6 6 6 6 3 0 0 6 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 1 6 0 0 0 6 0 0 3 6 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 6 0 0 6 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 6 0 0 0 0 0 6 0 0 0 0 6\n",
      " 0 0 6 0 0 0 0 0 0 6 3 0 0 6 6 0 0 0 0 0 0 6 0 0 0 0 0 0 0 6 6 0 0 6 6 0 6\n",
      " 0 6 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 6 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 6 6 6 6 0 6 0 0 0 0 0 0 0 0 0 0 6 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 6 0 0 0 0 0 6 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 6 0 0 0 0 0 1 0 0 6 0 0 0 0 3 6 3 0 6 0 0 6 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 6 6 6 0 0 6 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 6 0 0 0 0 6 0 6 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 6 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 6 6 6 0 0 0 0 3 6 0 0 6\n",
      " 6 0 6 6 0 6 0 0 0 0 0 0 0 6 6 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 6 0 0\n",
      " 0 0 0 0 0 0 6 0 6 0 0 0 0 0 0 0 0 0 0 0 0 0 0 3 6 0 0 6 0 6 6 0 0 0 0 0 0\n",
      " 0]\n",
      "threshold density 0.1515225292347496\n",
      "cluster signal 13374.0\n",
      "cluster length 1983300\n",
      "!!cluster density 0.006743306610195129\n",
      "cluster signal 80560.0\n",
      "cluster length 1675300\n",
      "!!cluster density 0.04808690980719871\n",
      "cluster signal 129069.0\n",
      "cluster length 1260900\n",
      "!!cluster density 0.10236259814418272\n",
      "cluster signal 92223.0\n",
      "cluster length 289600\n",
      "!!cluster density 0.31844958563535913\n",
      "cluster signal 120949.0\n",
      "cluster length 745300\n",
      "!!cluster density 0.16228230242855227\n",
      "cluster signal 68772.0\n",
      "cluster length 157100\n",
      "!!cluster density 0.4377593889242521\n",
      "cluster signal 122450.0\n",
      "cluster length 532800\n",
      "!!cluster density 0.22982357357357358\n",
      "cluster signal 14749.0\n",
      "cluster length 16100\n",
      "!!cluster density 0.9160869565217391\n",
      "cluster signal 664.0\n",
      "cluster length 200\n",
      "!!cluster density 3.32\n",
      "cluster signal 35769.0\n",
      "cluster length 57100\n",
      "!!cluster density 0.6264273204903678\n",
      "\n",
      "Labels mapping\n",
      " {0: 0, 6: 0, 3: 0, 5: 1, 1: 1, 2: 1, 9: 1, 4: 1, 7: 1, 8: 1}\n",
      "DEBUG: 49195 17982 \n",
      "\n",
      " +--------------------+-------------+---------------+--------------+-------+\n",
      "| bin_offset_5       | TT_S0       | starting_nt   | Chromosome   | +7    |\n",
      "| (category)         | (float64)   | (category)    | (category)   | ...   |\n",
      "|--------------------+-------------+---------------+--------------+-------|\n",
      "| (3000016, 3000021] | 0.0         | 3000016       | 18           | ...   |\n",
      "| (3000021, 3000121] | 7.0         | 3000021       | 18           | ...   |\n",
      "| (3000121, 3000221] | 1.0         | 3000121       | 18           | ...   |\n",
      "| (3000221, 3000321] | 2.0         | 3000221       | 18           | ...   |\n",
      "| ...                | ...         | ...           | ...          | ...   |\n",
      "| (9717221, 9717321] | 0.0         | 9717221       | 18           | ...   |\n",
      "| (9717321, 9717421] | 0.0         | 9717321       | 18           | ...   |\n",
      "| (9717421, 9717521] | 1.0         | 9717421       | 18           | ...   |\n",
      "| (9717521, 9717593] | 1.0         | 9717521       | 18           | ...   |\n",
      "+--------------------+-------------+---------------+--------------+-------+\n",
      "Unstranded PyRanges object has 49,195 rows and 11 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "7 hidden columns: Start, End, genes, transcripts, exons, CDSs, labels\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0\n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "BINS  IntervalIndex([(3000016, 3000026], (3000026, 3000126], (3000126, 3000226],\n",
      "               (3000226, 3000326], (3000326, 3000426], (3000426, 3000526],\n",
      "               (3000526, 3000626], (3000626, 3000726], (3000726, 3000826],\n",
      "               (3000826, 3000926],\n",
      "               ...\n",
      "               (9716626, 9716726], (9716726, 9716826], (9716826, 9716926],\n",
      "               (9716926, 9717026], (9717026, 9717126], (9717126, 9717226],\n",
      "               (9717226, 9717326], (9717326, 9717426], (9717426, 9717526],\n",
      "               (9717526, 9717593]],\n",
      "              dtype='interval[int64, right]', length=67177)\n",
      "HEAD with BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0   \n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "\n",
      "        bin_offset_10  \n",
      "0  (3000016, 3000026]  \n",
      "1  (3000016, 3000026]  \n",
      "2  (3000016, 3000026]  \n",
      "3  (3000016, 3000026]  \n",
      "4  (3000026, 3000126]  \n",
      "TAIL with BINS\n",
      "       seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  9717555  9717555  18_9717555    1.0    0.0    0.0   \n",
      "699996       19  9717571  9717571  19_9717571    0.0    0.0    0.0   \n",
      "699997       19  9717581  9717581  19_9717581    0.0    0.0    0.0   \n",
      "699998       18  9717582  9717582  18_9717582    0.0    0.0    0.0   \n",
      "699999       18  9717593  9717593  18_9717593    0.0    0.0    0.0   \n",
      "\n",
      "             bin_offset_10  \n",
      "699995  (9717526, 9717593]  \n",
      "699996  (9717526, 9717593]  \n",
      "699997  (9717526, 9717593]  \n",
      "699998  (9717526, 9717593]  \n",
      "699999  (9717526, 9717593]  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:136: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "         bin_offset_10  TT_S0 starting_nt\n",
      "0   (3000016, 3000026]    0.0     3000016\n",
      "1   (3000026, 3000126]    7.0     3000026\n",
      "2   (3000126, 3000226]    1.0     3000126\n",
      "3   (3000226, 3000326]    2.0     3000226\n",
      "4   (3000326, 3000426]    0.0     3000326\n",
      "5   (3000426, 3000526]    0.0     3000426\n",
      "6   (3000526, 3000626]    0.0     3000526\n",
      "7   (3000626, 3000726]    0.0     3000626\n",
      "8   (3000726, 3000826]    0.0     3000726\n",
      "9   (3000826, 3000926]    0.0     3000826\n",
      "10  (3000926, 3001026]    0.0     3000926\n",
      "11  (3001026, 3001126]    2.0     3001026\n",
      "12  (3001126, 3001226]    0.0     3001126\n",
      "13  (3001226, 3001326]    0.0     3001226\n",
      "14  (3001326, 3001426]    0.0     3001326\n",
      "15  (3001426, 3001526]    1.0     3001426\n",
      "16  (3001526, 3001626]    3.0     3001526\n",
      "17  (3001626, 3001726]    3.0     3001626\n",
      "18  (3001726, 3001826]    0.0     3001726\n",
      "19  (3001826, 3001926]    0.0     3001826\n",
      "20  (3001926, 3002026]    1.0     3001926\n",
      "21  (3002026, 3002126]    0.0     3002026\n",
      "22  (3002126, 3002226]    1.0     3002126\n",
      "23  (3002226, 3002326]   10.0     3002226\n",
      "24  (3002326, 3002426]   12.0     3002326\n",
      "25  (3002426, 3002526]   13.0     3002426\n",
      "26  (3002526, 3002626]    0.0     3002526\n",
      "27  (3002626, 3002726]    0.0     3002626\n",
      "28  (3002726, 3002826]    0.0     3002726\n",
      "29  (3002826, 3002926]    0.0     3002826\n",
      "30  (3002926, 3003026]    1.0     3002926\n",
      "31  (3003026, 3003126]    2.0     3003026\n",
      "32  (3003126, 3003226]    0.0     3003126\n",
      "33  (3003226, 3003326]    0.0     3003226\n",
      "34  (3003326, 3003426]    1.0     3003326\n",
      "35  (3003426, 3003526]    0.0     3003426\n",
      "36  (3003526, 3003626]    1.0     3003526\n",
      "37  (3003626, 3003726]    3.0     3003626\n",
      "38  (3003726, 3003826]    1.0     3003726\n",
      "39  (3003826, 3003926]    1.0     3003826\n",
      "40  (3003926, 3004026]    7.0     3003926\n",
      "41  (3004026, 3004126]    0.0     3004026\n",
      "42  (3004126, 3004226]    0.0     3004126\n",
      "43  (3004226, 3004326]    5.0     3004226\n",
      "44  (3004326, 3004426]    0.0     3004326\n",
      "45  (3004426, 3004526]    0.0     3004426\n",
      "46  (3004526, 3004626]    1.0     3004526\n",
      "47  (3004626, 3004726]    0.0     3004626\n",
      "48  (3004726, 3004826]    0.0     3004726\n",
      "49  (3004826, 3004926]   13.0     3004826\n",
      "(67177, 3)\n",
      "bin_offset_10    category\n",
      "TT_S0             float64\n",
      "starting_nt      category\n",
      "dtype: object\n",
      "sumos ilgis: 67177\n",
      "ranges ilgis: 67177\n",
      "Model input - NO SCALE\n",
      "       TT_S0  genes  transcripts  exons  CDSs\n",
      "0        0.0      0            0      0     0\n",
      "1        7.0      0            0      0     0\n",
      "2        1.0      0            0      0     0\n",
      "3        2.0      0            0      0     0\n",
      "4        0.0      0            0      0     0\n",
      "...      ...    ...          ...    ...   ...\n",
      "67172    0.0      0            0      0     0\n",
      "67173    0.0      0            0      0     0\n",
      "67174    0.0      0            0      0     0\n",
      "67175    1.0      0            0      0     0\n",
      "67176    1.0      0            0      0     0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "       TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0        0.0             0                   0             0            0\n",
      "1        7.0             0                   0             0            0\n",
      "2        1.0             0                   0             0            0\n",
      "3        2.0             0                   0             0            0\n",
      "4        0.0             0                   0             0            0\n",
      "...      ...           ...                 ...           ...          ...\n",
      "67172    0.0             0                   0             0            0\n",
      "67173    0.0             0                   0             0            0\n",
      "67174    0.0             0                   0             0            0\n",
      "67175    1.0             0                   0             0            0\n",
      "67176    1.0             0                   0             0            0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "0.00790449112047278\n",
      "[3 7 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 0 0 8 3 3 3 3 3 3 3 3 3 3 3\n",
      " 3 3 3 7 3 3 7 3 3 3 3 3 8 3 2 5 4 3 5 8 6 6 9 9 3 8 8 7 3 3 7 3 3 7 3 3 7\n",
      " 3 3 3 3 3 3 3 0 3 3 3 7 3 7 7 3 3 0 8 3 3 3 3 3 3 0 0 8 2 5 8 7 0 7 2 7 0\n",
      " 3 8 3 3 3 3 3 3 3 3 3 7 3 3 8 3 8 8 0 8 3 7 7 7 3 7 0 3 3 3 3 3 3 3 3 3 0\n",
      " 7 7 7 3 3 3 7 3 0 3 3 3 8 3 8 3 3 0 0 3 3 3 3 3 3 3 3 3 3 3 3 3 7 7 3 3 3\n",
      " 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 0 5 3 3 3 3 3 3 3 3 3 3 3 3 3\n",
      " 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3\n",
      " 3 3 3 3 3 3 3 3 3 3 3 3 3 3 7 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3\n",
      " 7 3 3 3 3 3 3 3 3 3 3 3 3 7 3 7 7 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3\n",
      " 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3\n",
      " 3 3 3 3 3 3 3 3 3 3 3 7 3 3 3 0 3 3 3 3 3 3 3 3 3 8 8 3 0 8 0 7 3 3 3 7 7\n",
      " 3 3 7 3 3 3 8 7 8 3 8 3 3 3 3 3 3 3 7 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3\n",
      " 3 3 3 3 7 3 3 3 3 3 3 3 3 3 3 3 3 3 3 0 3 3 3 7 7 7 3 7 7 7 7 7 3 3 3 3 3\n",
      " 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3\n",
      " 7 7 7 3 3 3 7 3 3 7 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3\n",
      " 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3\n",
      " 3 3 3 3 3 3 3 7 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 7 3 3 3 3 3 7 3 3 3 3 7\n",
      " 3 3 7 3 3 3 3 3 3 3 0 3 3 3 7 3 3 3 3 3 3 7 3 3 3 3 3 3 3 0 3 3 7 3 3 3 7\n",
      " 3 7 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 7 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3\n",
      " 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 7 7 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3\n",
      " 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3\n",
      " 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 0 3 3 3 3 3 3 3 0 7 0 3 3 3 3 3 3\n",
      " 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 7 3 3 3 7 3 3 3 3 3 3 3 3\n",
      " 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 7 3 3 3 3 3 3 7 3 3 3 3 3 3 3 3 3 3\n",
      " 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 7 7 7 3 3 3 3 7 7 3 7 3\n",
      " 7 3 0 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3\n",
      " 3 3 3 3 3 3 7 3 7 3 3 3 3 3 3 3 3 3 3 3 3 3 3 0 3 3 3 3 3 7 3 3 3 3 3 3 3\n",
      " 3]\n",
      "threshold density 0.1515224164546617\n",
      "cluster signal 25656.0\n",
      "cluster length 2394200\n",
      "!!cluster density 0.010715896750480328\n",
      "cluster signal 88943.0\n",
      "cluster length 1525400\n",
      "!!cluster density 0.058307984790874526\n",
      "cluster signal 86487.0\n",
      "cluster length 831700\n",
      "!!cluster density 0.10398821690513406\n",
      "cluster signal 106143.0\n",
      "cluster length 715600\n",
      "!!cluster density 0.14832727780883176\n",
      "cluster signal 118301.0\n",
      "cluster length 387500\n",
      "!!cluster density 0.30529290322580643\n",
      "cluster signal 133253.0\n",
      "cluster length 632900\n",
      "!!cluster density 0.2105435297835361\n",
      "cluster signal 71847.0\n",
      "cluster length 162700\n",
      "!!cluster density 0.4415918869084204\n",
      "cluster signal 13287.0\n",
      "cluster length 14400\n",
      "!!cluster density 0.9227083333333334\n",
      "cluster signal 664.0\n",
      "cluster length 200\n",
      "!!cluster density 3.32\n",
      "cluster signal 33998.0\n",
      "cluster length 53100\n",
      "!!cluster density 0.6402636534839925\n",
      "\n",
      "Labels mapping\n",
      " {3: 0, 7: 0, 0: 0, 8: 0, 2: 1, 5: 1, 4: 1, 6: 1, 9: 1, 1: 1}\n",
      "DEBUG: 54669 12508 \n",
      "\n",
      " +--------------------+-------------+---------------+--------------+-------+\n",
      "| bin_offset_10      | TT_S0       | starting_nt   | Chromosome   | +7    |\n",
      "| (category)         | (float64)   | (category)    | (category)   | ...   |\n",
      "|--------------------+-------------+---------------+--------------+-------|\n",
      "| (3000016, 3000026] | 0.0         | 3000016       | 18           | ...   |\n",
      "| (3000026, 3000126] | 7.0         | 3000026       | 18           | ...   |\n",
      "| (3000126, 3000226] | 1.0         | 3000126       | 18           | ...   |\n",
      "| (3000226, 3000326] | 2.0         | 3000226       | 18           | ...   |\n",
      "| ...                | ...         | ...           | ...          | ...   |\n",
      "| (9717226, 9717326] | 0.0         | 9717226       | 18           | ...   |\n",
      "| (9717326, 9717426] | 0.0         | 9717326       | 18           | ...   |\n",
      "| (9717426, 9717526] | 1.0         | 9717426       | 18           | ...   |\n",
      "| (9717526, 9717593] | 1.0         | 9717526       | 18           | ...   |\n",
      "+--------------------+-------------+---------------+--------------+-------+\n",
      "Unstranded PyRanges object has 54,669 rows and 11 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "7 hidden columns: Start, End, genes, transcripts, exons, CDSs, labels\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0\n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "BINS  IntervalIndex([(3000016, 3000031], (3000031, 3000131], (3000131, 3000231],\n",
      "               (3000231, 3000331], (3000331, 3000431], (3000431, 3000531],\n",
      "               (3000531, 3000631], (3000631, 3000731], (3000731, 3000831],\n",
      "               (3000831, 3000931],\n",
      "               ...\n",
      "               (9716631, 9716731], (9716731, 9716831], (9716831, 9716931],\n",
      "               (9716931, 9717031], (9717031, 9717131], (9717131, 9717231],\n",
      "               (9717231, 9717331], (9717331, 9717431], (9717431, 9717531],\n",
      "               (9717531, 9717593]],\n",
      "              dtype='interval[int64, right]', length=67177)\n",
      "HEAD with BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0   \n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "\n",
      "        bin_offset_15  \n",
      "0  (3000016, 3000031]  \n",
      "1  (3000016, 3000031]  \n",
      "2  (3000016, 3000031]  \n",
      "3  (3000016, 3000031]  \n",
      "4  (3000031, 3000131]  \n",
      "TAIL with BINS\n",
      "       seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  9717555  9717555  18_9717555    1.0    0.0    0.0   \n",
      "699996       19  9717571  9717571  19_9717571    0.0    0.0    0.0   \n",
      "699997       19  9717581  9717581  19_9717581    0.0    0.0    0.0   \n",
      "699998       18  9717582  9717582  18_9717582    0.0    0.0    0.0   \n",
      "699999       18  9717593  9717593  18_9717593    0.0    0.0    0.0   \n",
      "\n",
      "             bin_offset_15  \n",
      "699995  (9717531, 9717593]  \n",
      "699996  (9717531, 9717593]  \n",
      "699997  (9717531, 9717593]  \n",
      "699998  (9717531, 9717593]  \n",
      "699999  (9717531, 9717593]  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:136: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "         bin_offset_15  TT_S0 starting_nt\n",
      "0   (3000016, 3000031]    0.0     3000016\n",
      "1   (3000031, 3000131]    7.0     3000031\n",
      "2   (3000131, 3000231]    1.0     3000131\n",
      "3   (3000231, 3000331]    2.0     3000231\n",
      "4   (3000331, 3000431]    0.0     3000331\n",
      "5   (3000431, 3000531]    0.0     3000431\n",
      "6   (3000531, 3000631]    0.0     3000531\n",
      "7   (3000631, 3000731]    0.0     3000631\n",
      "8   (3000731, 3000831]    0.0     3000731\n",
      "9   (3000831, 3000931]    0.0     3000831\n",
      "10  (3000931, 3001031]    0.0     3000931\n",
      "11  (3001031, 3001131]    2.0     3001031\n",
      "12  (3001131, 3001231]    0.0     3001131\n",
      "13  (3001231, 3001331]    0.0     3001231\n",
      "14  (3001331, 3001431]    0.0     3001331\n",
      "15  (3001431, 3001531]    1.0     3001431\n",
      "16  (3001531, 3001631]    3.0     3001531\n",
      "17  (3001631, 3001731]    3.0     3001631\n",
      "18  (3001731, 3001831]    0.0     3001731\n",
      "19  (3001831, 3001931]    0.0     3001831\n",
      "20  (3001931, 3002031]    1.0     3001931\n",
      "21  (3002031, 3002131]    0.0     3002031\n",
      "22  (3002131, 3002231]    1.0     3002131\n",
      "23  (3002231, 3002331]   10.0     3002231\n",
      "24  (3002331, 3002431]   12.0     3002331\n",
      "25  (3002431, 3002531]   13.0     3002431\n",
      "26  (3002531, 3002631]    0.0     3002531\n",
      "27  (3002631, 3002731]    0.0     3002631\n",
      "28  (3002731, 3002831]    0.0     3002731\n",
      "29  (3002831, 3002931]    0.0     3002831\n",
      "30  (3002931, 3003031]    1.0     3002931\n",
      "31  (3003031, 3003131]    2.0     3003031\n",
      "32  (3003131, 3003231]    0.0     3003131\n",
      "33  (3003231, 3003331]    0.0     3003231\n",
      "34  (3003331, 3003431]    1.0     3003331\n",
      "35  (3003431, 3003531]    0.0     3003431\n",
      "36  (3003531, 3003631]    1.0     3003531\n",
      "37  (3003631, 3003731]    3.0     3003631\n",
      "38  (3003731, 3003831]    1.0     3003731\n",
      "39  (3003831, 3003931]    1.0     3003831\n",
      "40  (3003931, 3004031]    7.0     3003931\n",
      "41  (3004031, 3004131]    0.0     3004031\n",
      "42  (3004131, 3004231]    0.0     3004131\n",
      "43  (3004231, 3004331]    5.0     3004231\n",
      "44  (3004331, 3004431]    0.0     3004331\n",
      "45  (3004431, 3004531]    0.0     3004431\n",
      "46  (3004531, 3004631]    1.0     3004531\n",
      "47  (3004631, 3004731]    0.0     3004631\n",
      "48  (3004731, 3004831]    0.0     3004731\n",
      "49  (3004831, 3004931]   13.0     3004831\n",
      "(67177, 3)\n",
      "bin_offset_15    category\n",
      "TT_S0             float64\n",
      "starting_nt      category\n",
      "dtype: object\n",
      "sumos ilgis: 67177\n",
      "ranges ilgis: 67177\n",
      "Model input - NO SCALE\n",
      "       TT_S0  genes  transcripts  exons  CDSs\n",
      "0        0.0      0            0      0     0\n",
      "1        7.0      0            0      0     0\n",
      "2        1.0      0            0      0     0\n",
      "3        2.0      0            0      0     0\n",
      "4        0.0      0            0      0     0\n",
      "...      ...    ...          ...    ...   ...\n",
      "67172    0.0      0            0      0     0\n",
      "67173    0.0      0            0      0     0\n",
      "67174    0.0      0            0      0     0\n",
      "67175    1.0      0            0      0     0\n",
      "67176    1.0      0            0      0     0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "       TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0        0.0             0                   0             0            0\n",
      "1        7.0             0                   0             0            0\n",
      "2        1.0             0                   0             0            0\n",
      "3        2.0             0                   0             0            0\n",
      "4        0.0             0                   0             0            0\n",
      "...      ...           ...                 ...           ...          ...\n",
      "67172    0.0             0                   0             0            0\n",
      "67173    0.0             0                   0             0            0\n",
      "67174    0.0             0                   0             0            0\n",
      "67175    1.0             0                   0             0            0\n",
      "67176    1.0             0                   0             0            0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "0.07008351072539709\n",
      "[4 0 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 5 5 5 4 4 4 4 4 4 4 4 4 4 4\n",
      " 4 4 4 0 4 4 0 4 4 4 4 4 5 4 3 9 6 4 1 5 7 8 7 8 4 9 9 0 4 4 0 4 4 0 4 4 4\n",
      " 4 4 4 4 4 4 4 5 4 4 4 0 4 0 0 4 4 9 5 4 4 4 4 4 4 5 5 9 3 9 5 0 5 0 3 4 5\n",
      " 4 9 4 4 4 4 4 4 4 4 4 0 4 4 5 4 9 9 5 9 4 0 0 0 4 0 5 4 4 4 4 4 4 4 4 4 5\n",
      " 0 0 0 4 4 4 0 4 5 4 4 4 9 4 5 4 4 5 5 4 4 4 4 4 4 4 4 4 4 4 4 4 0 0 4 4 4\n",
      " 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 0 5 9 4 4 4 4 4 4 4 4 4 4 4 4 4\n",
      " 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4\n",
      " 4 4 4 4 4 4 4 4 4 4 4 4 4 4 0 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4\n",
      " 0 4 4 4 4 4 4 4 4 4 4 4 4 0 4 0 0 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4\n",
      " 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4\n",
      " 4 4 4 4 4 4 4 4 4 4 4 0 4 4 4 5 4 4 4 4 4 4 4 4 4 5 9 4 5 9 5 0 4 4 4 0 0\n",
      " 4 4 0 4 4 4 5 0 5 4 5 4 4 4 4 4 4 4 0 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4\n",
      " 4 4 4 4 0 4 4 4 4 4 4 4 4 4 4 4 4 4 0 0 4 4 4 0 0 0 4 0 0 0 0 0 4 4 4 4 4\n",
      " 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4\n",
      " 0 5 4 4 4 4 0 4 4 0 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4\n",
      " 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4\n",
      " 4 4 4 4 4 4 4 0 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 0 4 4 4 4 4 0 4 4 4 4 0\n",
      " 4 4 0 4 4 4 4 4 4 4 5 4 4 4 0 4 4 4 4 4 4 0 4 4 4 4 4 4 4 5 4 4 0 0 4 4 0\n",
      " 4 0 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 0 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4\n",
      " 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 0 0 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4\n",
      " 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4\n",
      " 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 5 4 4 4 4 4 4 4 5 0 5 4 4 4 4 4 4\n",
      " 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 0 0 4 4 4 0 4 4 4 4 4 4 4 4\n",
      " 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 0 0 4 4 4 4 4 4 0 4 4 4 4 4 4 4 4 4 4\n",
      " 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 0 0 0 4 4 4 4 0 0 4 0 4\n",
      " 0 4 5 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4\n",
      " 4 4 4 4 4 4 4 4 0 4 4 4 4 4 4 4 4 4 4 4 4 4 4 5 4 4 4 4 4 0 4 4 4 4 4 4 4\n",
      " 4]\n",
      "threshold density 0.1515223036747417\n",
      "cluster signal 25584.0\n",
      "cluster length 2390200\n",
      "!!cluster density 0.010703706802778011\n",
      "cluster signal 89377.0\n",
      "cluster length 1532500\n",
      "!!cluster density 0.058321044045677\n",
      "cluster signal 129717.0\n",
      "cluster length 1151900\n",
      "!!cluster density 0.1126113377897387\n",
      "cluster signal 98877.0\n",
      "cluster length 320800\n",
      "!!cluster density 0.3082200748129676\n",
      "cluster signal 101164.0\n",
      "cluster length 601500\n",
      "!!cluster density 0.16818620116375727\n",
      "cluster signal 70885.0\n",
      "cluster length 166500\n",
      "!!cluster density 0.42573573573573575\n",
      "cluster signal 106801.0\n",
      "cluster length 470800\n",
      "!!cluster density 0.2268500424808836\n",
      "cluster signal 16773.0\n",
      "cluster length 18900\n",
      "!!cluster density 0.8874603174603175\n",
      "cluster signal 637.0\n",
      "cluster length 200\n",
      "!!cluster density 3.185\n",
      "cluster signal 38764.0\n",
      "cluster length 64400\n",
      "!!cluster density 0.6019254658385094\n",
      "\n",
      "Labels mapping\n",
      " {4: 0, 0: 0, 5: 0, 3: 1, 9: 1, 6: 1, 1: 1, 7: 1, 8: 1, 2: 1}\n",
      "DEBUG: 50746 16431 \n",
      "\n",
      " +--------------------+-------------+---------------+--------------+-------+\n",
      "| bin_offset_15      | TT_S0       | starting_nt   | Chromosome   | +7    |\n",
      "| (category)         | (float64)   | (category)    | (category)   | ...   |\n",
      "|--------------------+-------------+---------------+--------------+-------|\n",
      "| (3000016, 3000031] | 0.0         | 3000016       | 18           | ...   |\n",
      "| (3000031, 3000131] | 7.0         | 3000031       | 18           | ...   |\n",
      "| (3000131, 3000231] | 1.0         | 3000131       | 18           | ...   |\n",
      "| (3000231, 3000331] | 2.0         | 3000231       | 18           | ...   |\n",
      "| ...                | ...         | ...           | ...          | ...   |\n",
      "| (9717231, 9717331] | 0.0         | 9717231       | 18           | ...   |\n",
      "| (9717331, 9717431] | 0.0         | 9717331       | 18           | ...   |\n",
      "| (9717431, 9717531] | 1.0         | 9717431       | 18           | ...   |\n",
      "| (9717531, 9717593] | 1.0         | 9717531       | 18           | ...   |\n",
      "+--------------------+-------------+---------------+--------------+-------+\n",
      "Unstranded PyRanges object has 50,746 rows and 11 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "7 hidden columns: Start, End, genes, transcripts, exons, CDSs, labels\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0\n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "BINS  IntervalIndex([(3000016, 3000036], (3000036, 3000136], (3000136, 3000236],\n",
      "               (3000236, 3000336], (3000336, 3000436], (3000436, 3000536],\n",
      "               (3000536, 3000636], (3000636, 3000736], (3000736, 3000836],\n",
      "               (3000836, 3000936],\n",
      "               ...\n",
      "               (9716636, 9716736], (9716736, 9716836], (9716836, 9716936],\n",
      "               (9716936, 9717036], (9717036, 9717136], (9717136, 9717236],\n",
      "               (9717236, 9717336], (9717336, 9717436], (9717436, 9717536],\n",
      "               (9717536, 9717593]],\n",
      "              dtype='interval[int64, right]', length=67177)\n",
      "HEAD with BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0   \n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "\n",
      "        bin_offset_20  \n",
      "0  (3000016, 3000036]  \n",
      "1  (3000016, 3000036]  \n",
      "2  (3000016, 3000036]  \n",
      "3  (3000016, 3000036]  \n",
      "4  (3000036, 3000136]  \n",
      "TAIL with BINS\n",
      "       seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  9717555  9717555  18_9717555    1.0    0.0    0.0   \n",
      "699996       19  9717571  9717571  19_9717571    0.0    0.0    0.0   \n",
      "699997       19  9717581  9717581  19_9717581    0.0    0.0    0.0   \n",
      "699998       18  9717582  9717582  18_9717582    0.0    0.0    0.0   \n",
      "699999       18  9717593  9717593  18_9717593    0.0    0.0    0.0   \n",
      "\n",
      "             bin_offset_20  \n",
      "699995  (9717536, 9717593]  \n",
      "699996  (9717536, 9717593]  \n",
      "699997  (9717536, 9717593]  \n",
      "699998  (9717536, 9717593]  \n",
      "699999  (9717536, 9717593]  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:136: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "         bin_offset_20  TT_S0 starting_nt\n",
      "0   (3000016, 3000036]    0.0     3000016\n",
      "1   (3000036, 3000136]    8.0     3000036\n",
      "2   (3000136, 3000236]    0.0     3000136\n",
      "3   (3000236, 3000336]    2.0     3000236\n",
      "4   (3000336, 3000436]    0.0     3000336\n",
      "5   (3000436, 3000536]    0.0     3000436\n",
      "6   (3000536, 3000636]    0.0     3000536\n",
      "7   (3000636, 3000736]    0.0     3000636\n",
      "8   (3000736, 3000836]    0.0     3000736\n",
      "9   (3000836, 3000936]    0.0     3000836\n",
      "10  (3000936, 3001036]    0.0     3000936\n",
      "11  (3001036, 3001136]    2.0     3001036\n",
      "12  (3001136, 3001236]    0.0     3001136\n",
      "13  (3001236, 3001336]    0.0     3001236\n",
      "14  (3001336, 3001436]    0.0     3001336\n",
      "15  (3001436, 3001536]    1.0     3001436\n",
      "16  (3001536, 3001636]    3.0     3001536\n",
      "17  (3001636, 3001736]    3.0     3001636\n",
      "18  (3001736, 3001836]    0.0     3001736\n",
      "19  (3001836, 3001936]    0.0     3001836\n",
      "20  (3001936, 3002036]    1.0     3001936\n",
      "21  (3002036, 3002136]    0.0     3002036\n",
      "22  (3002136, 3002236]    4.0     3002136\n",
      "23  (3002236, 3002336]    7.0     3002236\n",
      "24  (3002336, 3002436]   12.0     3002336\n",
      "25  (3002436, 3002536]   13.0     3002436\n",
      "26  (3002536, 3002636]    0.0     3002536\n",
      "27  (3002636, 3002736]    0.0     3002636\n",
      "28  (3002736, 3002836]    0.0     3002736\n",
      "29  (3002836, 3002936]    0.0     3002836\n",
      "30  (3002936, 3003036]    1.0     3002936\n",
      "31  (3003036, 3003136]    2.0     3003036\n",
      "32  (3003136, 3003236]    0.0     3003136\n",
      "33  (3003236, 3003336]    0.0     3003236\n",
      "34  (3003336, 3003436]    1.0     3003336\n",
      "35  (3003436, 3003536]    0.0     3003436\n",
      "36  (3003536, 3003636]    1.0     3003536\n",
      "37  (3003636, 3003736]    3.0     3003636\n",
      "38  (3003736, 3003836]    1.0     3003736\n",
      "39  (3003836, 3003936]    1.0     3003836\n",
      "40  (3003936, 3004036]    7.0     3003936\n",
      "41  (3004036, 3004136]    0.0     3004036\n",
      "42  (3004136, 3004236]    0.0     3004136\n",
      "43  (3004236, 3004336]    5.0     3004236\n",
      "44  (3004336, 3004436]    0.0     3004336\n",
      "45  (3004436, 3004536]    0.0     3004436\n",
      "46  (3004536, 3004636]    1.0     3004536\n",
      "47  (3004636, 3004736]    0.0     3004636\n",
      "48  (3004736, 3004836]    1.0     3004736\n",
      "49  (3004836, 3004936]   12.0     3004836\n",
      "(67177, 3)\n",
      "bin_offset_20    category\n",
      "TT_S0             float64\n",
      "starting_nt      category\n",
      "dtype: object\n",
      "sumos ilgis: 67177\n",
      "ranges ilgis: 67177\n",
      "Model input - NO SCALE\n",
      "       TT_S0  genes  transcripts  exons  CDSs\n",
      "0        0.0      0            0      0     0\n",
      "1        8.0      0            0      0     0\n",
      "2        0.0      0            0      0     0\n",
      "3        2.0      0            0      0     0\n",
      "4        0.0      0            0      0     0\n",
      "...      ...    ...          ...    ...   ...\n",
      "67172    0.0      0            0      0     0\n",
      "67173    0.0      0            0      0     0\n",
      "67174    1.0      0            0      0     0\n",
      "67175    0.0      0            0      0     0\n",
      "67176    1.0      0            0      0     0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "       TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0        0.0             0                   0             0            0\n",
      "1        8.0             0                   0             0            0\n",
      "2        0.0             0                   0             0            0\n",
      "3        2.0             0                   0             0            0\n",
      "4        0.0             0                   0             0            0\n",
      "...      ...           ...                 ...           ...          ...\n",
      "67172    0.0             0                   0             0            0\n",
      "67173    0.0             0                   0             0            0\n",
      "67174    1.0             0                   0             0            0\n",
      "67175    0.0             0                   0             0            0\n",
      "67176    1.0             0                   0             0            0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "0.2086279530196347\n",
      "[9 4 9 9 9 9 9 9 9 9 9 9 9 9 9 9 1 1 9 9 9 9 1 4 7 7 9 9 9 9 9 9 9 9 9 9 9\n",
      " 1 9 9 4 9 9 1 9 9 9 9 9 7 9 5 0 2 9 0 7 3 3 3 3 9 5 9 1 9 9 4 9 9 1 9 9 1\n",
      " 9 9 9 9 9 9 9 4 9 9 1 1 9 1 1 9 1 7 4 9 9 9 1 1 9 4 7 0 5 5 1 1 4 4 5 9 4\n",
      " 9 7 1 9 9 9 9 9 9 9 9 1 9 9 7 9 7 0 4 7 9 4 1 1 9 1 4 1 9 9 9 9 9 9 9 9 4\n",
      " 4 4 1 9 9 9 1 1 4 1 9 9 0 9 7 9 9 7 1 1 9 9 9 9 9 9 9 9 9 9 9 9 4 4 9 1 9\n",
      " 9 1 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 1 4 7 9 9 9 9 9 9 9 9 9 9 9 9 9\n",
      " 9 9 9 9 9 9 9 9 9 9 9 9 9 1 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9\n",
      " 9 9 9 9 9 9 9 9 9 1 9 9 9 9 1 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9\n",
      " 1 9 9 9 9 9 1 9 9 9 9 9 9 1 1 1 1 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9\n",
      " 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9\n",
      " 9 9 9 9 9 9 9 9 9 9 9 1 9 9 9 7 9 9 9 9 9 9 9 9 1 4 7 7 9 7 4 1 1 9 9 4 1\n",
      " 1 9 1 9 9 9 7 4 7 9 7 9 9 1 9 9 9 9 1 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9\n",
      " 1 9 9 9 1 9 9 9 9 9 9 9 9 9 9 9 9 9 1 4 9 9 9 1 1 1 9 1 4 1 4 1 9 9 1 9 9\n",
      " 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9\n",
      " 1 7 9 9 9 9 1 9 9 4 1 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9\n",
      " 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9\n",
      " 9 9 9 9 1 9 9 1 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 4 9 9 9 9 9 1 9 9 9 9 1\n",
      " 9 9 1 9 9 9 9 9 9 1 4 9 9 1 4 9 9 9 9 9 9 1 9 9 9 9 9 9 9 4 9 9 1 1 9 9 1\n",
      " 9 1 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 1 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9\n",
      " 9 9 9 9 9 9 9 9 9 9 9 9 9 1 1 4 1 9 1 9 9 9 9 9 9 9 9 9 1 9 9 9 9 9 9 9 9\n",
      " 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 1 9 9 9 9 9 1 9 9 9\n",
      " 9 9 9 9 9 9 9 9 9 9 9 9 9 9 1 9 9 9 9 9 7 9 9 1 9 9 9 9 4 1 4 9 1 9 9 1 9\n",
      " 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 1 1 9 9 9 1 1 9 9 9 9 9 9 9\n",
      " 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 1 1 9 9 9 9 9 9 1 9 9 9 9 9 9 9 9 9 9\n",
      " 9 9 1 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 1 1 1 9 9 9 1 4 1 9 1 9\n",
      " 1 1 4 9 9 1 9 9 9 9 9 9 9 1 1 9 9 9 9 1 9 9 9 9 9 9 9 9 9 9 9 9 9 9 1 9 9\n",
      " 9 9 9 9 9 1 9 9 1 9 9 9 9 9 9 9 9 9 9 9 9 9 9 4 1 9 9 1 9 4 9 9 9 9 9 9 9\n",
      " 9]\n",
      "threshold density 0.15152219089498958\n",
      "cluster signal 13431.0\n",
      "cluster length 1984100\n",
      "!!cluster density 0.006769316062698452\n",
      "cluster signal 104360.0\n",
      "cluster length 1183100\n",
      "!!cluster density 0.08820894260840166\n",
      "cluster signal 61353.0\n",
      "cluster length 1401500\n",
      "!!cluster density 0.043776667855868714\n",
      "cluster signal 107461.0\n",
      "cluster length 777300\n",
      "!!cluster density 0.13824906728418887\n",
      "cluster signal 89622.0\n",
      "cluster length 274100\n",
      "!!cluster density 0.32696825975921195\n",
      "cluster signal 93061.0\n",
      "cluster length 495000\n",
      "!!cluster density 0.1880020202020202\n",
      "cluster signal 65791.0\n",
      "cluster length 146600\n",
      "!!cluster density 0.44877899045020464\n",
      "cluster signal 12415.0\n",
      "cluster length 12600\n",
      "!!cluster density 0.9853174603174604\n",
      "cluster signal 95437.0\n",
      "cluster length 388000\n",
      "!!cluster density 0.2459716494845361\n",
      "cluster signal 35648.0\n",
      "cluster length 55400\n",
      "!!cluster density 0.6434657039711191\n",
      "\n",
      "Labels mapping\n",
      " {9: 0, 4: 0, 1: 0, 7: 0, 5: 1, 0: 1, 2: 1, 3: 1, 8: 1, 6: 1}\n",
      "DEBUG: 53460 13717 \n",
      "\n",
      " +--------------------+-------------+---------------+--------------+-------+\n",
      "| bin_offset_20      | TT_S0       | starting_nt   | Chromosome   | +7    |\n",
      "| (category)         | (float64)   | (category)    | (category)   | ...   |\n",
      "|--------------------+-------------+---------------+--------------+-------|\n",
      "| (3000016, 3000036] | 0.0         | 3000016       | 18           | ...   |\n",
      "| (3000036, 3000136] | 8.0         | 3000036       | 18           | ...   |\n",
      "| (3000136, 3000236] | 0.0         | 3000136       | 18           | ...   |\n",
      "| (3000236, 3000336] | 2.0         | 3000236       | 18           | ...   |\n",
      "| ...                | ...         | ...           | ...          | ...   |\n",
      "| (9717236, 9717336] | 0.0         | 9717236       | 18           | ...   |\n",
      "| (9717336, 9717436] | 1.0         | 9717336       | 18           | ...   |\n",
      "| (9717436, 9717536] | 0.0         | 9717436       | 18           | ...   |\n",
      "| (9717536, 9717593] | 1.0         | 9717536       | 18           | ...   |\n",
      "+--------------------+-------------+---------------+--------------+-------+\n",
      "Unstranded PyRanges object has 53,460 rows and 11 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "7 hidden columns: Start, End, genes, transcripts, exons, CDSs, labels\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0\n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "BINS  IntervalIndex([(3000016, 3000041], (3000041, 3000141], (3000141, 3000241],\n",
      "               (3000241, 3000341], (3000341, 3000441], (3000441, 3000541],\n",
      "               (3000541, 3000641], (3000641, 3000741], (3000741, 3000841],\n",
      "               (3000841, 3000941],\n",
      "               ...\n",
      "               (9716641, 9716741], (9716741, 9716841], (9716841, 9716941],\n",
      "               (9716941, 9717041], (9717041, 9717141], (9717141, 9717241],\n",
      "               (9717241, 9717341], (9717341, 9717441], (9717441, 9717541],\n",
      "               (9717541, 9717593]],\n",
      "              dtype='interval[int64, right]', length=67177)\n",
      "HEAD with BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0   \n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "\n",
      "        bin_offset_25  \n",
      "0  (3000016, 3000041]  \n",
      "1  (3000016, 3000041]  \n",
      "2  (3000016, 3000041]  \n",
      "3  (3000016, 3000041]  \n",
      "4  (3000016, 3000041]  \n",
      "TAIL with BINS\n",
      "       seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  9717555  9717555  18_9717555    1.0    0.0    0.0   \n",
      "699996       19  9717571  9717571  19_9717571    0.0    0.0    0.0   \n",
      "699997       19  9717581  9717581  19_9717581    0.0    0.0    0.0   \n",
      "699998       18  9717582  9717582  18_9717582    0.0    0.0    0.0   \n",
      "699999       18  9717593  9717593  18_9717593    0.0    0.0    0.0   \n",
      "\n",
      "             bin_offset_25  \n",
      "699995  (9717541, 9717593]  \n",
      "699996  (9717541, 9717593]  \n",
      "699997  (9717541, 9717593]  \n",
      "699998  (9717541, 9717593]  \n",
      "699999  (9717541, 9717593]  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:136: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "         bin_offset_25  TT_S0 starting_nt\n",
      "0   (3000016, 3000041]    4.0     3000016\n",
      "1   (3000041, 3000141]    4.0     3000041\n",
      "2   (3000141, 3000241]    0.0     3000141\n",
      "3   (3000241, 3000341]    2.0     3000241\n",
      "4   (3000341, 3000441]    0.0     3000341\n",
      "5   (3000441, 3000541]    0.0     3000441\n",
      "6   (3000541, 3000641]    0.0     3000541\n",
      "7   (3000641, 3000741]    0.0     3000641\n",
      "8   (3000741, 3000841]    0.0     3000741\n",
      "9   (3000841, 3000941]    0.0     3000841\n",
      "10  (3000941, 3001041]    2.0     3000941\n",
      "11  (3001041, 3001141]    0.0     3001041\n",
      "12  (3001141, 3001241]    0.0     3001141\n",
      "13  (3001241, 3001341]    0.0     3001241\n",
      "14  (3001341, 3001441]    0.0     3001341\n",
      "15  (3001441, 3001541]    1.0     3001441\n",
      "16  (3001541, 3001641]    3.0     3001541\n",
      "17  (3001641, 3001741]    3.0     3001641\n",
      "18  (3001741, 3001841]    0.0     3001741\n",
      "19  (3001841, 3001941]    0.0     3001841\n",
      "20  (3001941, 3002041]    1.0     3001941\n",
      "21  (3002041, 3002141]    0.0     3002041\n",
      "22  (3002141, 3002241]    4.0     3002141\n",
      "23  (3002241, 3002341]    7.0     3002241\n",
      "24  (3002341, 3002441]   12.0     3002341\n",
      "25  (3002441, 3002541]   13.0     3002441\n",
      "26  (3002541, 3002641]    0.0     3002541\n",
      "27  (3002641, 3002741]    0.0     3002641\n",
      "28  (3002741, 3002841]    0.0     3002741\n",
      "29  (3002841, 3002941]    0.0     3002841\n",
      "30  (3002941, 3003041]    1.0     3002941\n",
      "31  (3003041, 3003141]    2.0     3003041\n",
      "32  (3003141, 3003241]    0.0     3003141\n",
      "33  (3003241, 3003341]    1.0     3003241\n",
      "34  (3003341, 3003441]    0.0     3003341\n",
      "35  (3003441, 3003541]    0.0     3003441\n",
      "36  (3003541, 3003641]    1.0     3003541\n",
      "37  (3003641, 3003741]    3.0     3003641\n",
      "38  (3003741, 3003841]    1.0     3003741\n",
      "39  (3003841, 3003941]    1.0     3003841\n",
      "40  (3003941, 3004041]    7.0     3003941\n",
      "41  (3004041, 3004141]    0.0     3004041\n",
      "42  (3004141, 3004241]    1.0     3004141\n",
      "43  (3004241, 3004341]    4.0     3004241\n",
      "44  (3004341, 3004441]    0.0     3004341\n",
      "45  (3004441, 3004541]    0.0     3004441\n",
      "46  (3004541, 3004641]    1.0     3004541\n",
      "47  (3004641, 3004741]    0.0     3004641\n",
      "48  (3004741, 3004841]    2.0     3004741\n",
      "49  (3004841, 3004941]   11.0     3004841\n",
      "(67177, 3)\n",
      "bin_offset_25    category\n",
      "TT_S0             float64\n",
      "starting_nt      category\n",
      "dtype: object\n",
      "sumos ilgis: 67177\n",
      "ranges ilgis: 67177\n",
      "Model input - NO SCALE\n",
      "       TT_S0  genes  transcripts  exons  CDSs\n",
      "0        4.0      0            0      0     0\n",
      "1        4.0      0            0      0     0\n",
      "2        0.0      0            0      0     0\n",
      "3        2.0      0            0      0     0\n",
      "4        0.0      0            0      0     0\n",
      "...      ...    ...          ...    ...   ...\n",
      "67172    0.0      0            0      0     0\n",
      "67173    0.0      0            0      0     0\n",
      "67174    1.0      0            0      0     0\n",
      "67175    0.0      0            0      0     0\n",
      "67176    1.0      0            0      0     0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "       TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0        4.0             0                   0             0            0\n",
      "1        4.0             0                   0             0            0\n",
      "2        0.0             0                   0             0            0\n",
      "3        2.0             0                   0             0            0\n",
      "4        0.0             0                   0             0            0\n",
      "...      ...           ...                 ...           ...          ...\n",
      "67172    0.0             0                   0             0            0\n",
      "67173    0.0             0                   0             0            0\n",
      "67174    1.0             0                   0             0            0\n",
      "67175    0.0             0                   0             0            0\n",
      "67176    1.0             0                   0             0            0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "0.2085237506884797\n",
      "[1 1 4 4 4 4 4 4 4 4 4 4 4 4 4 4 1 1 4 4 4 4 1 6 2 2 4 4 4 4 4 4 4 4 4 4 4\n",
      " 1 4 4 6 4 4 1 4 4 4 4 4 6 4 5 2 5 4 8 2 7 7 7 3 6 0 4 1 4 4 6 4 4 1 4 1 4\n",
      " 4 4 4 4 4 4 1 6 4 4 1 1 4 1 1 4 1 2 6 4 4 4 1 1 4 6 2 8 0 0 1 1 6 6 0 4 6\n",
      " 4 2 1 4 4 4 4 4 4 4 4 1 4 4 2 4 2 8 6 2 4 6 1 1 4 1 6 1 4 4 4 4 4 4 4 4 6\n",
      " 6 6 1 4 4 4 1 6 1 1 4 4 2 4 2 4 1 6 1 1 4 4 4 4 4 4 4 4 4 4 4 4 6 6 4 1 4\n",
      " 4 1 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 6 6 2 4 4 4 4 4 4 4 4 4 4 4 4 4\n",
      " 4 4 4 4 4 4 4 4 4 4 4 4 4 1 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4\n",
      " 4 4 4 4 4 4 4 4 4 1 4 4 4 4 1 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4\n",
      " 1 4 4 4 4 4 1 4 4 4 4 4 4 1 4 1 1 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4\n",
      " 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4\n",
      " 4 4 4 4 4 4 4 4 4 4 4 1 4 4 1 6 4 4 4 4 4 4 4 4 6 6 6 2 4 2 6 1 1 4 4 6 1\n",
      " 1 4 1 4 4 4 2 6 2 4 2 4 4 1 4 4 4 4 1 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4\n",
      " 1 4 4 4 1 4 4 4 4 4 4 4 4 4 4 4 4 4 1 6 4 4 4 1 1 1 4 1 6 1 6 1 1 4 1 4 4\n",
      " 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4\n",
      " 1 2 4 4 4 4 1 4 4 6 1 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4\n",
      " 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4\n",
      " 4 4 4 4 1 4 4 1 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 6 4 4 4 4 4 1 4 4 4 4 1\n",
      " 4 4 1 4 4 4 4 4 4 1 6 4 4 1 6 4 4 4 4 4 4 1 4 4 4 4 4 4 4 6 4 4 1 1 4 4 1\n",
      " 1 1 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 1 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4\n",
      " 4 4 4 4 4 4 4 4 4 4 4 4 4 1 1 6 4 4 1 4 4 4 4 4 4 4 4 4 1 4 4 4 4 4 4 4 4\n",
      " 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 1 4 4 4 4 4 4 1 4 4 4\n",
      " 4 4 4 4 4 4 4 4 4 4 4 4 4 4 1 4 4 4 4 6 6 4 4 1 4 4 4 6 4 1 6 4 1 4 4 1 4\n",
      " 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 1 1 4 4 4 1 1 4 4 4 4 4 4 4\n",
      " 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 1 1 4 4 4 4 4 4 1 4 4 4 4 4 4 4 4 4 4\n",
      " 4 4 1 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 1 4 1 1 4 4 4 1 6 1 4 1 4\n",
      " 1 1 6 4 1 4 4 4 4 4 4 4 4 1 1 4 4 4 4 1 4 4 4 4 4 4 4 4 4 4 4 4 4 4 1 4 4\n",
      " 4 4 4 4 4 1 4 4 1 4 4 4 4 4 4 4 4 4 4 4 4 4 4 6 1 4 4 1 4 6 4 4 4 4 4 4 4\n",
      " 4]\n",
      "threshold density 0.15152207811540538\n",
      "cluster signal 61490.0\n",
      "cluster length 1400800\n",
      "!!cluster density 0.04389634494574529\n",
      "cluster signal 13425.0\n",
      "cluster length 1985700\n",
      "!!cluster density 0.006760840006043209\n",
      "cluster signal 103845.0\n",
      "cluster length 1177300\n",
      "!!cluster density 0.08820606472436932\n",
      "cluster signal 128132.0\n",
      "cluster length 900700\n",
      "!!cluster density 0.1422582435883202\n",
      "cluster signal 68023.0\n",
      "cluster length 178400\n",
      "!!cluster density 0.38129484304932737\n",
      "cluster signal 119182.0\n",
      "cluster length 577900\n",
      "!!cluster density 0.20623291226855858\n",
      "cluster signal 5961.0\n",
      "cluster length 5100\n",
      "!!cluster density 1.1688235294117648\n",
      "cluster signal 26914.0\n",
      "cluster length 37100\n",
      "!!cluster density 0.72544474393531\n",
      "cluster signal 102329.0\n",
      "cluster length 358500\n",
      "!!cluster density 0.28543654114365413\n",
      "cluster signal 49278.0\n",
      "cluster length 96200\n",
      "!!cluster density 0.5122453222453223\n",
      "\n",
      "Labels mapping\n",
      " {1: 0, 4: 0, 6: 0, 2: 0, 5: 1, 8: 1, 7: 1, 3: 1, 0: 1, 9: 1}\n",
      "DEBUG: 54645 12532 \n",
      "\n",
      " +--------------------+-------------+---------------+--------------+-------+\n",
      "| bin_offset_25      | TT_S0       | starting_nt   | Chromosome   | +7    |\n",
      "| (category)         | (float64)   | (category)    | (category)   | ...   |\n",
      "|--------------------+-------------+---------------+--------------+-------|\n",
      "| (3000016, 3000041] | 4.0         | 3000016       | 18           | ...   |\n",
      "| (3000041, 3000141] | 4.0         | 3000041       | 18           | ...   |\n",
      "| (3000141, 3000241] | 0.0         | 3000141       | 18           | ...   |\n",
      "| (3000241, 3000341] | 2.0         | 3000241       | 18           | ...   |\n",
      "| ...                | ...         | ...           | ...          | ...   |\n",
      "| (9717241, 9717341] | 0.0         | 9717241       | 18           | ...   |\n",
      "| (9717341, 9717441] | 1.0         | 9717341       | 18           | ...   |\n",
      "| (9717441, 9717541] | 0.0         | 9717441       | 18           | ...   |\n",
      "| (9717541, 9717593] | 1.0         | 9717541       | 18           | ...   |\n",
      "+--------------------+-------------+---------------+--------------+-------+\n",
      "Unstranded PyRanges object has 54,645 rows and 11 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "7 hidden columns: Start, End, genes, transcripts, exons, CDSs, labels\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0\n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "BINS  IntervalIndex([(3000016, 3000046], (3000046, 3000146], (3000146, 3000246],\n",
      "               (3000246, 3000346], (3000346, 3000446], (3000446, 3000546],\n",
      "               (3000546, 3000646], (3000646, 3000746], (3000746, 3000846],\n",
      "               (3000846, 3000946],\n",
      "               ...\n",
      "               (9716646, 9716746], (9716746, 9716846], (9716846, 9716946],\n",
      "               (9716946, 9717046], (9717046, 9717146], (9717146, 9717246],\n",
      "               (9717246, 9717346], (9717346, 9717446], (9717446, 9717546],\n",
      "               (9717546, 9717593]],\n",
      "              dtype='interval[int64, right]', length=67177)\n",
      "HEAD with BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0   \n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "\n",
      "        bin_offset_30  \n",
      "0  (3000016, 3000046]  \n",
      "1  (3000016, 3000046]  \n",
      "2  (3000016, 3000046]  \n",
      "3  (3000016, 3000046]  \n",
      "4  (3000016, 3000046]  \n",
      "TAIL with BINS\n",
      "       seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  9717555  9717555  18_9717555    1.0    0.0    0.0   \n",
      "699996       19  9717571  9717571  19_9717571    0.0    0.0    0.0   \n",
      "699997       19  9717581  9717581  19_9717581    0.0    0.0    0.0   \n",
      "699998       18  9717582  9717582  18_9717582    0.0    0.0    0.0   \n",
      "699999       18  9717593  9717593  18_9717593    0.0    0.0    0.0   \n",
      "\n",
      "             bin_offset_30  \n",
      "699995  (9717546, 9717593]  \n",
      "699996  (9717546, 9717593]  \n",
      "699997  (9717546, 9717593]  \n",
      "699998  (9717546, 9717593]  \n",
      "699999  (9717546, 9717593]  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:136: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "         bin_offset_30  TT_S0 starting_nt\n",
      "0   (3000016, 3000046]    4.0     3000016\n",
      "1   (3000046, 3000146]    4.0     3000046\n",
      "2   (3000146, 3000246]    0.0     3000146\n",
      "3   (3000246, 3000346]    2.0     3000246\n",
      "4   (3000346, 3000446]    0.0     3000346\n",
      "5   (3000446, 3000546]    0.0     3000446\n",
      "6   (3000546, 3000646]    0.0     3000546\n",
      "7   (3000646, 3000746]    0.0     3000646\n",
      "8   (3000746, 3000846]    0.0     3000746\n",
      "9   (3000846, 3000946]    0.0     3000846\n",
      "10  (3000946, 3001046]    2.0     3000946\n",
      "11  (3001046, 3001146]    0.0     3001046\n",
      "12  (3001146, 3001246]    0.0     3001146\n",
      "13  (3001246, 3001346]    0.0     3001246\n",
      "14  (3001346, 3001446]    0.0     3001346\n",
      "15  (3001446, 3001546]    1.0     3001446\n",
      "16  (3001546, 3001646]    5.0     3001546\n",
      "17  (3001646, 3001746]    1.0     3001646\n",
      "18  (3001746, 3001846]    0.0     3001746\n",
      "19  (3001846, 3001946]    0.0     3001846\n",
      "20  (3001946, 3002046]    1.0     3001946\n",
      "21  (3002046, 3002146]    0.0     3002046\n",
      "22  (3002146, 3002246]    4.0     3002146\n",
      "23  (3002246, 3002346]    7.0     3002246\n",
      "24  (3002346, 3002446]   14.0     3002346\n",
      "25  (3002446, 3002546]   11.0     3002446\n",
      "26  (3002546, 3002646]    0.0     3002546\n",
      "27  (3002646, 3002746]    0.0     3002646\n",
      "28  (3002746, 3002846]    0.0     3002746\n",
      "29  (3002846, 3002946]    0.0     3002846\n",
      "30  (3002946, 3003046]    1.0     3002946\n",
      "31  (3003046, 3003146]    2.0     3003046\n",
      "32  (3003146, 3003246]    0.0     3003146\n",
      "33  (3003246, 3003346]    1.0     3003246\n",
      "34  (3003346, 3003446]    0.0     3003346\n",
      "35  (3003446, 3003546]    0.0     3003446\n",
      "36  (3003546, 3003646]    1.0     3003546\n",
      "37  (3003646, 3003746]    3.0     3003646\n",
      "38  (3003746, 3003846]    1.0     3003746\n",
      "39  (3003846, 3003946]    1.0     3003846\n",
      "40  (3003946, 3004046]    7.0     3003946\n",
      "41  (3004046, 3004146]    0.0     3004046\n",
      "42  (3004146, 3004246]    1.0     3004146\n",
      "43  (3004246, 3004346]    4.0     3004246\n",
      "44  (3004346, 3004446]    0.0     3004346\n",
      "45  (3004446, 3004546]    0.0     3004446\n",
      "46  (3004546, 3004646]    1.0     3004546\n",
      "47  (3004646, 3004746]    0.0     3004646\n",
      "48  (3004746, 3004846]    2.0     3004746\n",
      "49  (3004846, 3004946]   11.0     3004846\n",
      "(67177, 3)\n",
      "bin_offset_30    category\n",
      "TT_S0             float64\n",
      "starting_nt      category\n",
      "dtype: object\n",
      "sumos ilgis: 67177\n",
      "ranges ilgis: 67177\n",
      "Model input - NO SCALE\n",
      "       TT_S0  genes  transcripts  exons  CDSs\n",
      "0        4.0      0            0      0     0\n",
      "1        4.0      0            0      0     0\n",
      "2        0.0      0            0      0     0\n",
      "3        2.0      0            0      0     0\n",
      "4        0.0      0            0      0     0\n",
      "...      ...    ...          ...    ...   ...\n",
      "67172    0.0      0            0      0     0\n",
      "67173    0.0      0            0      0     0\n",
      "67174    1.0      0            0      0     0\n",
      "67175    0.0      0            0      0     0\n",
      "67176    1.0      0            0      0     0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "       TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0        4.0             0                   0             0            0\n",
      "1        4.0             0                   0             0            0\n",
      "2        0.0             0                   0             0            0\n",
      "3        2.0             0                   0             0            0\n",
      "4        0.0             0                   0             0            0\n",
      "...      ...           ...                 ...           ...          ...\n",
      "67172    0.0             0                   0             0            0\n",
      "67173    0.0             0                   0             0            0\n",
      "67174    1.0             0                   0             0            0\n",
      "67175    0.0             0                   0             0            0\n",
      "67176    1.0             0                   0             0            0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "0.02156988254908674\n",
      "[7 7 2 2 2 2 2 2 2 2 2 2 2 2 2 2 7 2 2 2 2 2 7 7 8 0 2 2 2 2 2 2 2 2 2 2 2\n",
      " 2 2 2 7 2 2 7 2 2 2 2 2 0 2 1 0 1 2 3 0 4 4 9 4 7 3 2 7 2 2 7 2 2 7 2 7 2\n",
      " 2 2 2 2 2 2 7 7 2 2 2 7 2 7 2 2 2 0 0 2 2 2 2 2 2 0 0 8 6 6 7 7 7 7 6 2 0\n",
      " 2 8 2 2 2 2 2 2 2 2 2 7 2 2 8 2 8 8 0 8 2 0 7 7 2 7 0 2 2 2 2 2 2 2 2 2 0\n",
      " 7 7 7 2 2 2 7 0 7 2 2 2 8 2 0 2 7 8 2 2 2 2 2 2 2 2 2 2 2 2 2 2 7 7 2 2 2\n",
      " 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 7 0 8 2 2 2 2 2 2 2 2 2 2 2 2 2\n",
      " 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2\n",
      " 2 2 2 2 2 2 2 2 2 2 2 2 2 2 7 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2\n",
      " 7 2 2 2 2 2 2 2 2 2 2 2 2 7 2 2 7 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2\n",
      " 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2\n",
      " 2 2 2 2 2 2 2 2 2 2 2 7 2 2 2 0 2 2 2 2 2 2 2 2 7 0 7 0 2 3 7 7 7 2 2 7 7\n",
      " 2 2 7 2 2 2 0 7 0 2 0 2 2 2 2 2 2 2 7 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2\n",
      " 2 2 2 2 7 2 2 2 2 2 2 2 2 2 2 2 2 2 0 2 2 2 2 7 0 2 2 7 7 7 7 2 7 2 2 2 2\n",
      " 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2\n",
      " 7 0 2 2 2 2 7 2 2 7 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2\n",
      " 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2\n",
      " 2 2 2 2 2 2 2 7 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 7 2 2 2 2 2 7 2 2 2 2 7\n",
      " 2 2 7 2 2 2 2 2 2 7 7 2 2 0 2 2 2 2 2 2 2 7 2 2 2 2 2 2 2 0 2 2 7 7 2 2 7\n",
      " 7 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 7 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2\n",
      " 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 0 7 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2\n",
      " 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2\n",
      " 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 7 7 2 2 2 2 2 2 0 2 7 0 2 2 2 2 2 2\n",
      " 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 7 7 2 2 7 2 2 2 2 2 2 2 2 2\n",
      " 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 7 2 2 2 2 2 2 2 7 2 2 2 2 2 2 2 2 2 2\n",
      " 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 7 2 7 7 2 2 2 2 7 7 2 7 2\n",
      " 7 7 0 2 2 2 2 2 2 2 2 2 2 7 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2\n",
      " 2 2 2 2 2 2 2 2 7 2 2 2 2 2 2 2 2 2 2 2 2 2 2 0 2 2 2 2 2 7 2 2 2 2 2 2 2\n",
      " 2]\n",
      "threshold density 0.15152196533598902\n",
      "cluster signal 89619.0\n",
      "cluster length 1535800\n",
      "!!cluster density 0.05835330121109519\n",
      "cluster signal 25439.0\n",
      "cluster length 2385200\n",
      "!!cluster density 0.01066535301022975\n",
      "cluster signal 121172.0\n",
      "cluster length 747200\n",
      "!!cluster density 0.1621680942184154\n",
      "cluster signal 108676.0\n",
      "cluster length 1002700\n",
      "!!cluster density 0.10838336491473023\n",
      "cluster signal 65188.0\n",
      "cluster length 144900\n",
      "!!cluster density 0.44988267770876467\n",
      "cluster signal 118094.0\n",
      "cluster length 514600\n",
      "!!cluster density 0.22948698017877964\n",
      "cluster signal 11813.0\n",
      "cluster length 12200\n",
      "!!cluster density 0.9682786885245902\n",
      "cluster signal 398.0\n",
      "cluster length 100\n",
      "!!cluster density 3.98\n",
      "cluster signal 102700.0\n",
      "cluster length 319800\n",
      "!!cluster density 0.32113821138211385\n",
      "cluster signal 35480.0\n",
      "cluster length 55200\n",
      "!!cluster density 0.6427536231884058\n",
      "\n",
      "Labels mapping\n",
      " {7: 0, 2: 0, 8: 1, 0: 0, 1: 1, 3: 1, 4: 1, 9: 1, 6: 1, 5: 1}\n",
      "DEBUG: 49237 17940 \n",
      "\n",
      " +--------------------+-------------+---------------+--------------+-------+\n",
      "| bin_offset_30      | TT_S0       | starting_nt   | Chromosome   | +7    |\n",
      "| (category)         | (float64)   | (category)    | (category)   | ...   |\n",
      "|--------------------+-------------+---------------+--------------+-------|\n",
      "| (3000016, 3000046] | 4.0         | 3000016       | 18           | ...   |\n",
      "| (3000046, 3000146] | 4.0         | 3000046       | 18           | ...   |\n",
      "| (3000146, 3000246] | 0.0         | 3000146       | 18           | ...   |\n",
      "| (3000246, 3000346] | 2.0         | 3000246       | 18           | ...   |\n",
      "| ...                | ...         | ...           | ...          | ...   |\n",
      "| (9717246, 9717346] | 0.0         | 9717246       | 18           | ...   |\n",
      "| (9717346, 9717446] | 1.0         | 9717346       | 18           | ...   |\n",
      "| (9717446, 9717546] | 0.0         | 9717446       | 18           | ...   |\n",
      "| (9717546, 9717593] | 1.0         | 9717546       | 18           | ...   |\n",
      "+--------------------+-------------+---------------+--------------+-------+\n",
      "Unstranded PyRanges object has 49,237 rows and 11 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "7 hidden columns: Start, End, genes, transcripts, exons, CDSs, labels\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0\n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "BINS  IntervalIndex([(3000016, 3000051], (3000051, 3000151], (3000151, 3000251],\n",
      "               (3000251, 3000351], (3000351, 3000451], (3000451, 3000551],\n",
      "               (3000551, 3000651], (3000651, 3000751], (3000751, 3000851],\n",
      "               (3000851, 3000951],\n",
      "               ...\n",
      "               (9716651, 9716751], (9716751, 9716851], (9716851, 9716951],\n",
      "               (9716951, 9717051], (9717051, 9717151], (9717151, 9717251],\n",
      "               (9717251, 9717351], (9717351, 9717451], (9717451, 9717551],\n",
      "               (9717551, 9717593]],\n",
      "              dtype='interval[int64, right]', length=67177)\n",
      "HEAD with BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0   \n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "\n",
      "        bin_offset_35  \n",
      "0  (3000016, 3000051]  \n",
      "1  (3000016, 3000051]  \n",
      "2  (3000016, 3000051]  \n",
      "3  (3000016, 3000051]  \n",
      "4  (3000016, 3000051]  \n",
      "TAIL with BINS\n",
      "       seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  9717555  9717555  18_9717555    1.0    0.0    0.0   \n",
      "699996       19  9717571  9717571  19_9717571    0.0    0.0    0.0   \n",
      "699997       19  9717581  9717581  19_9717581    0.0    0.0    0.0   \n",
      "699998       18  9717582  9717582  18_9717582    0.0    0.0    0.0   \n",
      "699999       18  9717593  9717593  18_9717593    0.0    0.0    0.0   \n",
      "\n",
      "             bin_offset_35  \n",
      "699995  (9717551, 9717593]  \n",
      "699996  (9717551, 9717593]  \n",
      "699997  (9717551, 9717593]  \n",
      "699998  (9717551, 9717593]  \n",
      "699999  (9717551, 9717593]  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:136: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "         bin_offset_35  TT_S0 starting_nt\n",
      "0   (3000016, 3000051]    4.0     3000016\n",
      "1   (3000051, 3000151]    4.0     3000051\n",
      "2   (3000151, 3000251]    0.0     3000151\n",
      "3   (3000251, 3000351]    2.0     3000251\n",
      "4   (3000351, 3000451]    0.0     3000351\n",
      "5   (3000451, 3000551]    0.0     3000451\n",
      "6   (3000551, 3000651]    0.0     3000551\n",
      "7   (3000651, 3000751]    0.0     3000651\n",
      "8   (3000751, 3000851]    0.0     3000751\n",
      "9   (3000851, 3000951]    0.0     3000851\n",
      "10  (3000951, 3001051]    2.0     3000951\n",
      "11  (3001051, 3001151]    0.0     3001051\n",
      "12  (3001151, 3001251]    0.0     3001151\n",
      "13  (3001251, 3001351]    0.0     3001251\n",
      "14  (3001351, 3001451]    0.0     3001351\n",
      "15  (3001451, 3001551]    1.0     3001451\n",
      "16  (3001551, 3001651]    5.0     3001551\n",
      "17  (3001651, 3001751]    1.0     3001651\n",
      "18  (3001751, 3001851]    0.0     3001751\n",
      "19  (3001851, 3001951]    0.0     3001851\n",
      "20  (3001951, 3002051]    1.0     3001951\n",
      "21  (3002051, 3002151]    0.0     3002051\n",
      "22  (3002151, 3002251]    4.0     3002151\n",
      "23  (3002251, 3002351]    7.0     3002251\n",
      "24  (3002351, 3002451]   14.0     3002351\n",
      "25  (3002451, 3002551]   11.0     3002451\n",
      "26  (3002551, 3002651]    0.0     3002551\n",
      "27  (3002651, 3002751]    0.0     3002651\n",
      "28  (3002751, 3002851]    0.0     3002751\n",
      "29  (3002851, 3002951]    0.0     3002851\n",
      "30  (3002951, 3003051]    1.0     3002951\n",
      "31  (3003051, 3003151]    2.0     3003051\n",
      "32  (3003151, 3003251]    0.0     3003151\n",
      "33  (3003251, 3003351]    1.0     3003251\n",
      "34  (3003351, 3003451]    0.0     3003351\n",
      "35  (3003451, 3003551]    0.0     3003451\n",
      "36  (3003551, 3003651]    1.0     3003551\n",
      "37  (3003651, 3003751]    3.0     3003651\n",
      "38  (3003751, 3003851]    1.0     3003751\n",
      "39  (3003851, 3003951]    1.0     3003851\n",
      "40  (3003951, 3004051]    7.0     3003951\n",
      "41  (3004051, 3004151]    0.0     3004051\n",
      "42  (3004151, 3004251]    1.0     3004151\n",
      "43  (3004251, 3004351]    4.0     3004251\n",
      "44  (3004351, 3004451]    0.0     3004351\n",
      "45  (3004451, 3004551]    0.0     3004451\n",
      "46  (3004551, 3004651]    1.0     3004551\n",
      "47  (3004651, 3004751]    0.0     3004651\n",
      "48  (3004751, 3004851]    2.0     3004751\n",
      "49  (3004851, 3004951]   11.0     3004851\n",
      "(67177, 3)\n",
      "bin_offset_35    category\n",
      "TT_S0             float64\n",
      "starting_nt      category\n",
      "dtype: object\n",
      "sumos ilgis: 67177\n",
      "ranges ilgis: 67177\n",
      "Model input - NO SCALE\n",
      "       TT_S0  genes  transcripts  exons  CDSs\n",
      "0        4.0      0            0      0     0\n",
      "1        4.0      0            0      0     0\n",
      "2        0.0      0            0      0     0\n",
      "3        2.0      0            0      0     0\n",
      "4        0.0      0            0      0     0\n",
      "...      ...    ...          ...    ...   ...\n",
      "67172    0.0      0            0      0     0\n",
      "67173    0.0      0            0      0     0\n",
      "67174    1.0      0            0      0     0\n",
      "67175    0.0      0            0      0     0\n",
      "67176    1.0      0            0      0     0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "       TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0        4.0             0                   0             0            0\n",
      "1        4.0             0                   0             0            0\n",
      "2        0.0             0                   0             0            0\n",
      "3        2.0             0                   0             0            0\n",
      "4        0.0             0                   0             0            0\n",
      "...      ...           ...                 ...           ...          ...\n",
      "67172    0.0             0                   0             0            0\n",
      "67173    0.0             0                   0             0            0\n",
      "67174    1.0             0                   0             0            0\n",
      "67175    0.0             0                   0             0            0\n",
      "67176    1.0             0                   0             0            0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "0.03687273918156512\n",
      "[0 0 5 5 5 5 5 5 5 5 5 5 5 5 5 5 0 5 5 5 5 5 0 0 2 8 5 5 5 5 5 5 5 5 5 5 5\n",
      " 5 5 5 0 5 5 0 5 5 5 5 5 8 5 1 8 7 5 3 1 4 4 6 1 0 3 5 0 5 5 0 5 5 0 5 0 5\n",
      " 5 5 5 5 5 5 0 0 5 5 5 0 5 0 5 5 0 8 8 5 5 5 5 5 5 8 2 2 1 3 0 0 8 0 1 0 0\n",
      " 5 2 5 5 5 5 5 5 5 5 5 0 5 5 2 5 2 2 8 2 5 8 5 0 5 0 8 5 5 5 5 5 5 5 5 5 8\n",
      " 0 0 0 5 5 5 0 8 0 5 5 8 0 5 8 5 0 2 5 5 5 5 5 5 5 5 5 5 5 5 5 5 0 0 5 5 5\n",
      " 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 0 8 2 5 5 5 5 5 5 5 5 5 5 5 5 5\n",
      " 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5\n",
      " 5 5 5 5 5 5 5 5 5 5 5 5 5 5 0 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5\n",
      " 0 5 5 5 5 5 5 5 5 5 5 5 5 0 5 0 0 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5\n",
      " 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5\n",
      " 5 5 5 5 5 5 5 5 5 5 5 0 5 5 5 8 5 5 5 5 5 5 5 5 8 8 0 8 5 2 0 0 0 5 5 0 0\n",
      " 5 5 5 5 5 5 8 2 0 5 8 5 5 5 5 5 5 5 0 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5\n",
      " 5 5 5 5 0 5 5 5 5 5 5 5 5 5 5 5 5 5 8 5 5 5 5 0 8 5 5 0 0 0 0 0 5 5 5 5 5\n",
      " 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5\n",
      " 0 8 5 5 5 5 0 5 5 0 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5\n",
      " 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5\n",
      " 5 5 5 5 5 5 5 0 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 0 5 5 5 5 5 5 0 5 5 5 5 0\n",
      " 5 5 0 5 5 5 5 5 5 8 5 5 5 8 5 5 5 5 5 5 5 0 5 5 5 5 5 5 5 8 5 5 0 0 5 5 0\n",
      " 0 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 0 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5\n",
      " 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 8 0 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5\n",
      " 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5\n",
      " 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 0 0 5 0 5 5 5 5 8 5 0 8 5 5 5 5 5 5\n",
      " 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 0 0 5 5 0 5 5 5 5 5 5 5 5 5\n",
      " 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 0 5 5 5 5 5 5 5 0 5 5 5 5 5 5 5 5 5 5\n",
      " 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 0 5 0 0 5 5 5 5 0 0 5 0 0\n",
      " 5 0 8 5 5 5 5 5 5 5 5 5 5 0 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5\n",
      " 5 5 5 5 5 5 5 5 0 5 5 5 5 5 5 5 5 5 5 5 5 5 5 8 5 5 5 5 5 0 5 5 5 5 5 5 5\n",
      " 5]\n",
      "threshold density 0.15152185255674055\n",
      "cluster signal 89657.0\n",
      "cluster length 1534400\n",
      "!!cluster density 0.0584313086548488\n",
      "cluster signal 25478.0\n",
      "cluster length 2387100\n",
      "!!cluster density 0.01067320179297055\n",
      "cluster signal 139153.0\n",
      "cluster length 836900\n",
      "!!cluster density 0.1662719560281993\n",
      "cluster signal 107887.0\n",
      "cluster length 996200\n",
      "!!cluster density 0.10829853443083719\n",
      "cluster signal 83767.0\n",
      "cluster length 247700\n",
      "!!cluster density 0.3381792490916431\n",
      "cluster signal 55107.0\n",
      "cluster length 121700\n",
      "!!cluster density 0.452810188989318\n",
      "cluster signal 126800.0\n",
      "cluster length 520500\n",
      "!!cluster density 0.243611911623439\n",
      "cluster signal 15375.0\n",
      "cluster length 16800\n",
      "!!cluster density 0.9151785714285714\n",
      "cluster signal 461.0\n",
      "cluster length 100\n",
      "!!cluster density 4.61\n",
      "cluster signal 34894.0\n",
      "cluster length 56300\n",
      "!!cluster density 0.6197868561278863\n",
      "\n",
      "Labels mapping\n",
      " {0: 0, 5: 0, 2: 1, 8: 0, 1: 1, 7: 1, 3: 1, 4: 1, 6: 1, 9: 1}\n",
      "DEBUG: 49177 18000 \n",
      "\n",
      " +--------------------+-------------+---------------+--------------+-------+\n",
      "| bin_offset_35      | TT_S0       | starting_nt   | Chromosome   | +7    |\n",
      "| (category)         | (float64)   | (category)    | (category)   | ...   |\n",
      "|--------------------+-------------+---------------+--------------+-------|\n",
      "| (3000016, 3000051] | 4.0         | 3000016       | 18           | ...   |\n",
      "| (3000051, 3000151] | 4.0         | 3000051       | 18           | ...   |\n",
      "| (3000151, 3000251] | 0.0         | 3000151       | 18           | ...   |\n",
      "| (3000251, 3000351] | 2.0         | 3000251       | 18           | ...   |\n",
      "| ...                | ...         | ...           | ...          | ...   |\n",
      "| (9717251, 9717351] | 0.0         | 9717251       | 18           | ...   |\n",
      "| (9717351, 9717451] | 1.0         | 9717351       | 18           | ...   |\n",
      "| (9717451, 9717551] | 0.0         | 9717451       | 18           | ...   |\n",
      "| (9717551, 9717593] | 1.0         | 9717551       | 18           | ...   |\n",
      "+--------------------+-------------+---------------+--------------+-------+\n",
      "Unstranded PyRanges object has 49,177 rows and 11 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "7 hidden columns: Start, End, genes, transcripts, exons, CDSs, labels\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0\n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "BINS  IntervalIndex([(3000016, 3000056], (3000056, 3000156], (3000156, 3000256],\n",
      "               (3000256, 3000356], (3000356, 3000456], (3000456, 3000556],\n",
      "               (3000556, 3000656], (3000656, 3000756], (3000756, 3000856],\n",
      "               (3000856, 3000956],\n",
      "               ...\n",
      "               (9716656, 9716756], (9716756, 9716856], (9716856, 9716956],\n",
      "               (9716956, 9717056], (9717056, 9717156], (9717156, 9717256],\n",
      "               (9717256, 9717356], (9717356, 9717456], (9717456, 9717556],\n",
      "               (9717556, 9717593]],\n",
      "              dtype='interval[int64, right]', length=67177)\n",
      "HEAD with BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0   \n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "\n",
      "        bin_offset_40  \n",
      "0  (3000016, 3000056]  \n",
      "1  (3000016, 3000056]  \n",
      "2  (3000016, 3000056]  \n",
      "3  (3000016, 3000056]  \n",
      "4  (3000016, 3000056]  \n",
      "TAIL with BINS\n",
      "       seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  9717555  9717555  18_9717555    1.0    0.0    0.0   \n",
      "699996       19  9717571  9717571  19_9717571    0.0    0.0    0.0   \n",
      "699997       19  9717581  9717581  19_9717581    0.0    0.0    0.0   \n",
      "699998       18  9717582  9717582  18_9717582    0.0    0.0    0.0   \n",
      "699999       18  9717593  9717593  18_9717593    0.0    0.0    0.0   \n",
      "\n",
      "             bin_offset_40  \n",
      "699995  (9717456, 9717556]  \n",
      "699996  (9717556, 9717593]  \n",
      "699997  (9717556, 9717593]  \n",
      "699998  (9717556, 9717593]  \n",
      "699999  (9717556, 9717593]  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:136: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "         bin_offset_40  TT_S0 starting_nt\n",
      "0   (3000016, 3000056]    4.0     3000016\n",
      "1   (3000056, 3000156]    4.0     3000056\n",
      "2   (3000156, 3000256]    0.0     3000156\n",
      "3   (3000256, 3000356]    2.0     3000256\n",
      "4   (3000356, 3000456]    0.0     3000356\n",
      "5   (3000456, 3000556]    0.0     3000456\n",
      "6   (3000556, 3000656]    0.0     3000556\n",
      "7   (3000656, 3000756]    0.0     3000656\n",
      "8   (3000756, 3000856]    0.0     3000756\n",
      "9   (3000856, 3000956]    0.0     3000856\n",
      "10  (3000956, 3001056]    2.0     3000956\n",
      "11  (3001056, 3001156]    0.0     3001056\n",
      "12  (3001156, 3001256]    0.0     3001156\n",
      "13  (3001256, 3001356]    0.0     3001256\n",
      "14  (3001356, 3001456]    0.0     3001356\n",
      "15  (3001456, 3001556]    1.0     3001456\n",
      "16  (3001556, 3001656]    5.0     3001556\n",
      "17  (3001656, 3001756]    1.0     3001656\n",
      "18  (3001756, 3001856]    0.0     3001756\n",
      "19  (3001856, 3001956]    0.0     3001856\n",
      "20  (3001956, 3002056]    1.0     3001956\n",
      "21  (3002056, 3002156]    0.0     3002056\n",
      "22  (3002156, 3002256]    4.0     3002156\n",
      "23  (3002256, 3002356]    7.0     3002256\n",
      "24  (3002356, 3002456]   14.0     3002356\n",
      "25  (3002456, 3002556]   11.0     3002456\n",
      "26  (3002556, 3002656]    0.0     3002556\n",
      "27  (3002656, 3002756]    0.0     3002656\n",
      "28  (3002756, 3002856]    0.0     3002756\n",
      "29  (3002856, 3002956]    0.0     3002856\n",
      "30  (3002956, 3003056]    1.0     3002956\n",
      "31  (3003056, 3003156]    2.0     3003056\n",
      "32  (3003156, 3003256]    0.0     3003156\n",
      "33  (3003256, 3003356]    1.0     3003256\n",
      "34  (3003356, 3003456]    0.0     3003356\n",
      "35  (3003456, 3003556]    0.0     3003456\n",
      "36  (3003556, 3003656]    1.0     3003556\n",
      "37  (3003656, 3003756]    4.0     3003656\n",
      "38  (3003756, 3003856]    0.0     3003756\n",
      "39  (3003856, 3003956]    1.0     3003856\n",
      "40  (3003956, 3004056]    7.0     3003956\n",
      "41  (3004056, 3004156]    0.0     3004056\n",
      "42  (3004156, 3004256]    1.0     3004156\n",
      "43  (3004256, 3004356]    4.0     3004256\n",
      "44  (3004356, 3004456]    0.0     3004356\n",
      "45  (3004456, 3004556]    0.0     3004456\n",
      "46  (3004556, 3004656]    1.0     3004556\n",
      "47  (3004656, 3004756]    0.0     3004656\n",
      "48  (3004756, 3004856]   10.0     3004756\n",
      "49  (3004856, 3004956]    3.0     3004856\n",
      "(67177, 3)\n",
      "bin_offset_40    category\n",
      "TT_S0             float64\n",
      "starting_nt      category\n",
      "dtype: object\n",
      "sumos ilgis: 67177\n",
      "ranges ilgis: 67177\n",
      "Model input - NO SCALE\n",
      "       TT_S0  genes  transcripts  exons  CDSs\n",
      "0        4.0      0            0      0     0\n",
      "1        4.0      0            0      0     0\n",
      "2        0.0      0            0      0     0\n",
      "3        2.0      0            0      0     0\n",
      "4        0.0      0            0      0     0\n",
      "...      ...    ...          ...    ...   ...\n",
      "67172    0.0      0            0      0     0\n",
      "67173    0.0      0            0      0     0\n",
      "67174    1.0      0            0      0     0\n",
      "67175    1.0      0            0      0     0\n",
      "67176    0.0      0            0      0     0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "       TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0        4.0             0                   0             0            0\n",
      "1        4.0             0                   0             0            0\n",
      "2        0.0             0                   0             0            0\n",
      "3        2.0             0                   0             0            0\n",
      "4        0.0             0                   0             0            0\n",
      "...      ...           ...                 ...           ...          ...\n",
      "67172    0.0             0                   0             0            0\n",
      "67173    0.0             0                   0             0            0\n",
      "67174    1.0             0                   0             0            0\n",
      "67175    1.0             0                   0             0            0\n",
      "67176    0.0             0                   0             0            0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "0.3554341515697337\n",
      "[4 4 1 1 1 1 1 1 1 1 1 1 1 1 1 1 4 1 1 1 1 1 4 4 0 0 1 1 1 1 1 1 1 1 1 1 1\n",
      " 4 1 1 4 1 1 4 1 1 1 1 0 1 1 7 4 7 1 2 2 5 5 6 1 4 8 1 4 1 1 4 1 1 4 1 4 1\n",
      " 1 1 1 1 1 1 1 4 1 1 1 4 1 4 1 1 4 0 4 1 1 1 1 1 1 0 9 0 2 2 4 4 0 4 2 4 4\n",
      " 4 0 1 1 1 1 1 1 1 1 1 4 1 1 0 1 9 8 0 0 1 0 1 4 1 4 4 1 1 1 1 1 1 1 1 1 9\n",
      " 1 4 4 1 1 1 4 4 4 1 1 0 4 1 0 1 4 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 4 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 4 0 9 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 4 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 4 1 1 1 1 4 1 1 1 1 1 1 1 4 1 4 4 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 4 1 1 1 4 1 1 1 1 1 1 1 1 4 0 4 0 1 8 4 4 4 1 1 0 1\n",
      " 1 1 1 1 1 1 0 9 4 4 4 1 1 1 1 1 1 1 4 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 4 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 4 0 1 1 4 4 4 4 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 4 0 1 1 1 4 1 1 1 4 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 4 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 4 1 1 1 1 1 4 1 1 1 1 1 4\n",
      " 1 1 4 1 1 1 1 1 1 0 1 1 1 4 1 1 1 1 1 1 1 4 1 1 1 1 1 1 1 4 1 1 4 4 1 1 4\n",
      " 4 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 4 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 4 4 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 4 4 1 4 1 1 1 1 0 1 4 0 1 1 1 4 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 4 4 1 1 4 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 4 1 1 1 1 1 1 1 4 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 4 1 4 4 1 1 1 1 4 4 1 4 4\n",
      " 1 1 0 1 1 1 1 1 1 1 1 1 1 4 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 4 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 1 4 1 1 1 1 1 1 1\n",
      " 1]\n",
      "threshold density 0.15152173977766\n",
      "cluster signal 110444.0\n",
      "cluster length 1766800\n",
      "!!cluster density 0.06251075390536563\n",
      "cluster signal 25430.0\n",
      "cluster length 2387700\n",
      "!!cluster density 0.010650416719018303\n",
      "cluster signal 108740.0\n",
      "cluster length 918300\n",
      "!!cluster density 0.11841446150495481\n",
      "cluster signal 73119.0\n",
      "cluster length 184300\n",
      "!!cluster density 0.39673901247965276\n",
      "cluster signal 96136.0\n",
      "cluster length 326900\n",
      "!!cluster density 0.2940838176812481\n",
      "cluster signal 19920.0\n",
      "cluster length 22800\n",
      "!!cluster density 0.8736842105263158\n",
      "cluster signal 490.0\n",
      "cluster length 100\n",
      "!!cluster density 4.9\n",
      "cluster signal 95273.0\n",
      "cluster length 429500\n",
      "!!cluster density 0.2218230500582072\n",
      "cluster signal 99984.0\n",
      "cluster length 593900\n",
      "!!cluster density 0.16835157433911432\n",
      "cluster signal 49043.0\n",
      "cluster length 87400\n",
      "!!cluster density 0.5611327231121281\n",
      "\n",
      "Labels mapping\n",
      " {4: 0, 1: 0, 0: 0, 7: 1, 2: 1, 5: 1, 6: 1, 8: 1, 9: 1, 3: 1}\n",
      "DEBUG: 50728 16449 \n",
      "\n",
      " +--------------------+-------------+---------------+--------------+-------+\n",
      "| bin_offset_40      | TT_S0       | starting_nt   | Chromosome   | +7    |\n",
      "| (category)         | (float64)   | (category)    | (category)   | ...   |\n",
      "|--------------------+-------------+---------------+--------------+-------|\n",
      "| (3000016, 3000056] | 4.0         | 3000016       | 18           | ...   |\n",
      "| (3000056, 3000156] | 4.0         | 3000056       | 18           | ...   |\n",
      "| (3000156, 3000256] | 0.0         | 3000156       | 18           | ...   |\n",
      "| (3000256, 3000356] | 2.0         | 3000256       | 18           | ...   |\n",
      "| ...                | ...         | ...           | ...          | ...   |\n",
      "| (9717256, 9717356] | 0.0         | 9717256       | 18           | ...   |\n",
      "| (9717356, 9717456] | 1.0         | 9717356       | 18           | ...   |\n",
      "| (9717456, 9717556] | 1.0         | 9717456       | 18           | ...   |\n",
      "| (9717556, 9717593] | 0.0         | 9717556       | 18           | ...   |\n",
      "+--------------------+-------------+---------------+--------------+-------+\n",
      "Unstranded PyRanges object has 50,728 rows and 11 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "7 hidden columns: Start, End, genes, transcripts, exons, CDSs, labels\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0\n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "BINS  IntervalIndex([(3000016, 3000061], (3000061, 3000161], (3000161, 3000261],\n",
      "               (3000261, 3000361], (3000361, 3000461], (3000461, 3000561],\n",
      "               (3000561, 3000661], (3000661, 3000761], (3000761, 3000861],\n",
      "               (3000861, 3000961],\n",
      "               ...\n",
      "               (9716661, 9716761], (9716761, 9716861], (9716861, 9716961],\n",
      "               (9716961, 9717061], (9717061, 9717161], (9717161, 9717261],\n",
      "               (9717261, 9717361], (9717361, 9717461], (9717461, 9717561],\n",
      "               (9717561, 9717593]],\n",
      "              dtype='interval[int64, right]', length=67177)\n",
      "HEAD with BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0   \n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "\n",
      "        bin_offset_45  \n",
      "0  (3000016, 3000061]  \n",
      "1  (3000016, 3000061]  \n",
      "2  (3000016, 3000061]  \n",
      "3  (3000016, 3000061]  \n",
      "4  (3000016, 3000061]  \n",
      "TAIL with BINS\n",
      "       seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  9717555  9717555  18_9717555    1.0    0.0    0.0   \n",
      "699996       19  9717571  9717571  19_9717571    0.0    0.0    0.0   \n",
      "699997       19  9717581  9717581  19_9717581    0.0    0.0    0.0   \n",
      "699998       18  9717582  9717582  18_9717582    0.0    0.0    0.0   \n",
      "699999       18  9717593  9717593  18_9717593    0.0    0.0    0.0   \n",
      "\n",
      "             bin_offset_45  \n",
      "699995  (9717461, 9717561]  \n",
      "699996  (9717561, 9717593]  \n",
      "699997  (9717561, 9717593]  \n",
      "699998  (9717561, 9717593]  \n",
      "699999  (9717561, 9717593]  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:136: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "         bin_offset_45  TT_S0 starting_nt\n",
      "0   (3000016, 3000061]    4.0     3000016\n",
      "1   (3000061, 3000161]    4.0     3000061\n",
      "2   (3000161, 3000261]    0.0     3000161\n",
      "3   (3000261, 3000361]    2.0     3000261\n",
      "4   (3000361, 3000461]    0.0     3000361\n",
      "5   (3000461, 3000561]    0.0     3000461\n",
      "6   (3000561, 3000661]    0.0     3000561\n",
      "7   (3000661, 3000761]    0.0     3000661\n",
      "8   (3000761, 3000861]    0.0     3000761\n",
      "9   (3000861, 3000961]    0.0     3000861\n",
      "10  (3000961, 3001061]    2.0     3000961\n",
      "11  (3001061, 3001161]    0.0     3001061\n",
      "12  (3001161, 3001261]    0.0     3001161\n",
      "13  (3001261, 3001361]    0.0     3001261\n",
      "14  (3001361, 3001461]    0.0     3001361\n",
      "15  (3001461, 3001561]    1.0     3001461\n",
      "16  (3001561, 3001661]    6.0     3001561\n",
      "17  (3001661, 3001761]    0.0     3001661\n",
      "18  (3001761, 3001861]    0.0     3001761\n",
      "19  (3001861, 3001961]    0.0     3001861\n",
      "20  (3001961, 3002061]    1.0     3001961\n",
      "21  (3002061, 3002161]    0.0     3002061\n",
      "22  (3002161, 3002261]    4.0     3002161\n",
      "23  (3002261, 3002361]    9.0     3002261\n",
      "24  (3002361, 3002461]   19.0     3002361\n",
      "25  (3002461, 3002561]    4.0     3002461\n",
      "26  (3002561, 3002661]    0.0     3002561\n",
      "27  (3002661, 3002761]    0.0     3002661\n",
      "28  (3002761, 3002861]    0.0     3002761\n",
      "29  (3002861, 3002961]    0.0     3002861\n",
      "30  (3002961, 3003061]    3.0     3002961\n",
      "31  (3003061, 3003161]    0.0     3003061\n",
      "32  (3003161, 3003261]    0.0     3003161\n",
      "33  (3003261, 3003361]    1.0     3003261\n",
      "34  (3003361, 3003461]    0.0     3003361\n",
      "35  (3003461, 3003561]    0.0     3003461\n",
      "36  (3003561, 3003661]    1.0     3003561\n",
      "37  (3003661, 3003761]    4.0     3003661\n",
      "38  (3003761, 3003861]    0.0     3003761\n",
      "39  (3003861, 3003961]    1.0     3003861\n",
      "40  (3003961, 3004061]    7.0     3003961\n",
      "41  (3004061, 3004161]    0.0     3004061\n",
      "42  (3004161, 3004261]    1.0     3004161\n",
      "43  (3004261, 3004361]    4.0     3004261\n",
      "44  (3004361, 3004461]    0.0     3004361\n",
      "45  (3004461, 3004561]    0.0     3004461\n",
      "46  (3004561, 3004661]    1.0     3004561\n",
      "47  (3004661, 3004761]    0.0     3004661\n",
      "48  (3004761, 3004861]   10.0     3004761\n",
      "49  (3004861, 3004961]    3.0     3004861\n",
      "(67177, 3)\n",
      "bin_offset_45    category\n",
      "TT_S0             float64\n",
      "starting_nt      category\n",
      "dtype: object\n",
      "sumos ilgis: 67177\n",
      "ranges ilgis: 67177\n",
      "Model input - NO SCALE\n",
      "       TT_S0  genes  transcripts  exons  CDSs\n",
      "0        4.0      0            0      0     0\n",
      "1        4.0      0            0      0     0\n",
      "2        0.0      0            0      0     0\n",
      "3        2.0      0            0      0     0\n",
      "4        0.0      0            0      0     0\n",
      "...      ...    ...          ...    ...   ...\n",
      "67172    0.0      0            0      0     0\n",
      "67173    0.0      0            0      0     0\n",
      "67174    1.0      0            0      0     0\n",
      "67175    1.0      0            0      0     0\n",
      "67176    0.0      0            0      0     0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "       TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0        4.0             0                   0             0            0\n",
      "1        4.0             0                   0             0            0\n",
      "2        0.0             0                   0             0            0\n",
      "3        2.0             0                   0             0            0\n",
      "4        0.0             0                   0             0            0\n",
      "...      ...           ...                 ...           ...          ...\n",
      "67172    0.0             0                   0             0            0\n",
      "67173    0.0             0                   0             0            0\n",
      "67174    1.0             0                   0             0            0\n",
      "67175    1.0             0                   0             0            0\n",
      "67176    0.0             0                   0             0            0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "0.044256218646262856\n",
      "[0 0 6 6 6 6 6 6 6 6 6 6 6 6 6 6 0 6 6 6 6 6 0 7 2 0 6 6 6 6 0 6 6 6 6 6 6\n",
      " 0 6 6 0 6 6 0 6 6 6 6 7 0 6 9 0 9 6 1 4 4 4 8 6 0 5 6 0 6 6 0 6 6 7 6 0 6\n",
      " 6 6 6 6 6 6 6 0 6 0 6 0 6 7 6 6 0 7 7 6 6 6 0 0 6 7 2 2 1 5 0 0 7 1 0 0 0\n",
      " 0 7 0 6 6 6 6 6 6 6 6 0 6 6 2 6 2 5 0 2 6 2 6 0 0 6 7 0 6 6 6 6 6 6 6 6 2\n",
      " 6 0 0 6 6 6 0 7 0 0 6 7 7 0 7 6 0 2 6 6 6 6 6 6 6 6 6 6 6 6 6 6 7 0 6 0 6\n",
      " 6 0 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 7 2 7 6 6 6 6 6 6 6 6 6 6 6 6 6\n",
      " 6 6 6 6 6 6 6 6 6 6 6 6 6 0 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6\n",
      " 6 6 6 6 6 6 6 6 6 0 6 6 6 0 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6\n",
      " 0 6 6 6 6 0 6 6 6 6 6 6 6 0 0 0 0 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6\n",
      " 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6\n",
      " 6 6 6 6 6 6 6 6 6 6 6 0 6 6 0 7 6 6 6 6 6 6 6 6 7 2 0 7 6 5 7 0 0 6 0 7 6\n",
      " 0 0 6 6 6 6 2 2 0 0 7 6 6 0 6 6 6 6 0 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6\n",
      " 0 6 6 6 0 6 6 6 6 6 6 6 6 6 6 6 6 0 7 6 6 6 0 6 7 6 6 0 7 0 0 0 6 6 0 6 6\n",
      " 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6\n",
      " 0 7 6 6 6 0 6 6 6 7 0 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6\n",
      " 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6\n",
      " 6 6 6 0 6 6 6 0 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 0 6 6 6 6 6 0 6 6 6 6 6 0\n",
      " 6 6 0 6 6 6 6 6 6 7 6 6 6 7 0 6 6 6 6 6 6 0 6 6 6 6 6 6 6 7 6 6 0 0 6 6 0\n",
      " 0 0 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 0 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6\n",
      " 6 6 6 6 6 6 6 6 6 6 6 6 6 0 0 7 0 6 0 6 6 6 6 6 6 6 6 6 0 6 6 6 6 6 6 6 6\n",
      " 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 0 6 6 6 6 6 0 6 6 6 6\n",
      " 6 6 6 6 6 6 6 6 6 6 6 6 6 6 0 6 6 6 6 0 0 6 0 6 6 6 6 7 6 0 7 6 0 6 0 6 6\n",
      " 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 0 0 6 6 0 6 0 6 6 6 6 6 6 6\n",
      " 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 0 6 6 6 6 0 6 6 0 6 6 6 6 6 6 6 6 6 6\n",
      " 6 6 0 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 0 0 0 0 6 6 6 0 0 0 6 0 0\n",
      " 0 0 7 6 0 0 6 6 6 6 6 6 6 0 6 6 6 6 6 0 6 6 6 6 6 6 6 6 6 6 6 6 6 6 0 6 6\n",
      " 6 6 6 6 6 0 6 6 0 6 6 6 6 6 6 6 6 6 6 6 6 6 6 7 0 6 6 0 6 7 6 6 6 6 6 6 6\n",
      " 6]\n",
      "threshold density 0.15152162699874733\n",
      "cluster signal 81134.0\n",
      "cluster length 1683700\n",
      "!!cluster density 0.04818791946308725\n",
      "cluster signal 13370.0\n",
      "cluster length 1981900\n",
      "!!cluster density 0.00674605176850497\n",
      "cluster signal 106855.0\n",
      "cluster length 1086600\n",
      "!!cluster density 0.0983388551444874\n",
      "cluster signal 143370.0\n",
      "cluster length 915900\n",
      "!!cluster density 0.15653455617425482\n",
      "cluster signal 59159.0\n",
      "cluster length 138100\n",
      "!!cluster density 0.4283779869659667\n",
      "cluster signal 94623.0\n",
      "cluster length 297300\n",
      "!!cluster density 0.3182744702320888\n",
      "cluster signal 18279.0\n",
      "cluster length 20600\n",
      "!!cluster density 0.8873300970873786\n",
      "cluster signal 490.0\n",
      "cluster length 100\n",
      "!!cluster density 4.9\n",
      "cluster signal 120133.0\n",
      "cluster length 523500\n",
      "!!cluster density 0.22948042024832857\n",
      "cluster signal 41166.0\n",
      "cluster length 70000\n",
      "!!cluster density 0.5880857142857143\n",
      "\n",
      "Labels mapping\n",
      " {0: 0, 6: 0, 7: 0, 2: 1, 9: 1, 1: 1, 4: 1, 8: 1, 5: 1, 3: 1}\n",
      "DEBUG: 47522 19655 \n",
      "\n",
      " +--------------------+-------------+---------------+--------------+-------+\n",
      "| bin_offset_45      | TT_S0       | starting_nt   | Chromosome   | +7    |\n",
      "| (category)         | (float64)   | (category)    | (category)   | ...   |\n",
      "|--------------------+-------------+---------------+--------------+-------|\n",
      "| (3000016, 3000061] | 4.0         | 3000016       | 18           | ...   |\n",
      "| (3000061, 3000161] | 4.0         | 3000061       | 18           | ...   |\n",
      "| (3000161, 3000261] | 0.0         | 3000161       | 18           | ...   |\n",
      "| (3000261, 3000361] | 2.0         | 3000261       | 18           | ...   |\n",
      "| ...                | ...         | ...           | ...          | ...   |\n",
      "| (9717261, 9717361] | 0.0         | 9717261       | 18           | ...   |\n",
      "| (9717361, 9717461] | 1.0         | 9717361       | 18           | ...   |\n",
      "| (9717461, 9717561] | 1.0         | 9717461       | 18           | ...   |\n",
      "| (9717561, 9717593] | 0.0         | 9717561       | 18           | ...   |\n",
      "+--------------------+-------------+---------------+--------------+-------+\n",
      "Unstranded PyRanges object has 47,522 rows and 11 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "7 hidden columns: Start, End, genes, transcripts, exons, CDSs, labels\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0\n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "BINS  IntervalIndex([(3000016, 3000066], (3000066, 3000166], (3000166, 3000266],\n",
      "               (3000266, 3000366], (3000366, 3000466], (3000466, 3000566],\n",
      "               (3000566, 3000666], (3000666, 3000766], (3000766, 3000866],\n",
      "               (3000866, 3000966],\n",
      "               ...\n",
      "               (9716666, 9716766], (9716766, 9716866], (9716866, 9716966],\n",
      "               (9716966, 9717066], (9717066, 9717166], (9717166, 9717266],\n",
      "               (9717266, 9717366], (9717366, 9717466], (9717466, 9717566],\n",
      "               (9717566, 9717593]],\n",
      "              dtype='interval[int64, right]', length=67177)\n",
      "HEAD with BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0   \n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "\n",
      "        bin_offset_50  \n",
      "0  (3000016, 3000066]  \n",
      "1  (3000016, 3000066]  \n",
      "2  (3000016, 3000066]  \n",
      "3  (3000016, 3000066]  \n",
      "4  (3000016, 3000066]  \n",
      "TAIL with BINS\n",
      "       seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  9717555  9717555  18_9717555    1.0    0.0    0.0   \n",
      "699996       19  9717571  9717571  19_9717571    0.0    0.0    0.0   \n",
      "699997       19  9717581  9717581  19_9717581    0.0    0.0    0.0   \n",
      "699998       18  9717582  9717582  18_9717582    0.0    0.0    0.0   \n",
      "699999       18  9717593  9717593  18_9717593    0.0    0.0    0.0   \n",
      "\n",
      "             bin_offset_50  \n",
      "699995  (9717466, 9717566]  \n",
      "699996  (9717566, 9717593]  \n",
      "699997  (9717566, 9717593]  \n",
      "699998  (9717566, 9717593]  \n",
      "699999  (9717566, 9717593]  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:136: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "         bin_offset_50  TT_S0 starting_nt\n",
      "0   (3000016, 3000066]    4.0     3000016\n",
      "1   (3000066, 3000166]    4.0     3000066\n",
      "2   (3000166, 3000266]    0.0     3000166\n",
      "3   (3000266, 3000366]    2.0     3000266\n",
      "4   (3000366, 3000466]    0.0     3000366\n",
      "5   (3000466, 3000566]    0.0     3000466\n",
      "6   (3000566, 3000666]    0.0     3000566\n",
      "7   (3000666, 3000766]    0.0     3000666\n",
      "8   (3000766, 3000866]    0.0     3000766\n",
      "9   (3000866, 3000966]    0.0     3000866\n",
      "10  (3000966, 3001066]    2.0     3000966\n",
      "11  (3001066, 3001166]    0.0     3001066\n",
      "12  (3001166, 3001266]    0.0     3001166\n",
      "13  (3001266, 3001366]    0.0     3001266\n",
      "14  (3001366, 3001466]    0.0     3001366\n",
      "15  (3001466, 3001566]    1.0     3001466\n",
      "16  (3001566, 3001666]    6.0     3001566\n",
      "17  (3001666, 3001766]    0.0     3001666\n",
      "18  (3001766, 3001866]    0.0     3001766\n",
      "19  (3001866, 3001966]    0.0     3001866\n",
      "20  (3001966, 3002066]    1.0     3001966\n",
      "21  (3002066, 3002166]    0.0     3002066\n",
      "22  (3002166, 3002266]    4.0     3002166\n",
      "23  (3002266, 3002366]    9.0     3002266\n",
      "24  (3002366, 3002466]   19.0     3002366\n",
      "25  (3002466, 3002566]    4.0     3002466\n",
      "26  (3002566, 3002666]    0.0     3002566\n",
      "27  (3002666, 3002766]    0.0     3002666\n",
      "28  (3002766, 3002866]    0.0     3002766\n",
      "29  (3002866, 3002966]    0.0     3002866\n",
      "30  (3002966, 3003066]    3.0     3002966\n",
      "31  (3003066, 3003166]    0.0     3003066\n",
      "32  (3003166, 3003266]    0.0     3003166\n",
      "33  (3003266, 3003366]    1.0     3003266\n",
      "34  (3003366, 3003466]    0.0     3003366\n",
      "35  (3003466, 3003566]    0.0     3003466\n",
      "36  (3003566, 3003666]    1.0     3003566\n",
      "37  (3003666, 3003766]    4.0     3003666\n",
      "38  (3003766, 3003866]    0.0     3003766\n",
      "39  (3003866, 3003966]    6.0     3003866\n",
      "40  (3003966, 3004066]    2.0     3003966\n",
      "41  (3004066, 3004166]    0.0     3004066\n",
      "42  (3004166, 3004266]    3.0     3004166\n",
      "43  (3004266, 3004366]    2.0     3004266\n",
      "44  (3004366, 3004466]    0.0     3004366\n",
      "45  (3004466, 3004566]    0.0     3004466\n",
      "46  (3004566, 3004666]    1.0     3004566\n",
      "47  (3004666, 3004766]    0.0     3004666\n",
      "48  (3004766, 3004866]   10.0     3004766\n",
      "49  (3004866, 3004966]    3.0     3004866\n",
      "(67177, 3)\n",
      "bin_offset_50    category\n",
      "TT_S0             float64\n",
      "starting_nt      category\n",
      "dtype: object\n",
      "sumos ilgis: 67177\n",
      "ranges ilgis: 67177\n",
      "Model input - NO SCALE\n",
      "       TT_S0  genes  transcripts  exons  CDSs\n",
      "0        4.0      0            0      0     0\n",
      "1        4.0      0            0      0     0\n",
      "2        0.0      0            0      0     0\n",
      "3        2.0      0            0      0     0\n",
      "4        0.0      0            0      0     0\n",
      "...      ...    ...          ...    ...   ...\n",
      "67172    0.0      0            0      0     0\n",
      "67173    0.0      0            0      0     0\n",
      "67174    1.0      0            0      0     0\n",
      "67175    1.0      0            0      0     0\n",
      "67176    0.0      0            0      0     0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "       TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0        4.0             0                   0             0            0\n",
      "1        4.0             0                   0             0            0\n",
      "2        0.0             0                   0             0            0\n",
      "3        2.0             0                   0             0            0\n",
      "4        0.0             0                   0             0            0\n",
      "...      ...           ...                 ...           ...          ...\n",
      "67172    0.0             0                   0             0            0\n",
      "67173    0.0             0                   0             0            0\n",
      "67174    1.0             0                   0             0            0\n",
      "67175    1.0             0                   0             0            0\n",
      "67176    0.0             0                   0             0            0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "0.06298286615954865\n",
      "[7 7 0 0 0 0 0 0 0 0 0 0 0 0 0 0 7 0 0 0 0 0 7 2 6 7 0 0 0 0 0 0 0 0 0 0 0\n",
      " 7 0 7 0 0 0 0 0 0 0 0 2 0 0 3 6 6 0 1 4 8 8 5 0 7 1 0 7 0 0 7 0 0 7 0 7 0\n",
      " 0 0 0 0 0 0 0 7 0 0 0 0 0 2 0 0 7 2 2 0 0 0 0 0 7 7 6 6 1 1 7 7 2 1 7 2 0\n",
      " 7 2 0 0 0 0 0 0 0 0 7 0 0 0 2 0 1 6 2 2 0 2 0 7 7 0 2 0 0 0 0 0 0 0 0 0 6\n",
      " 0 7 7 0 0 0 7 2 0 0 0 2 7 7 2 0 7 2 0 0 0 0 0 0 0 0 0 0 0 0 0 0 2 7 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 2 2 2 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 7 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 7 0 0 0 0 0 0 0 0 0 0 0 0 7 0 7 7 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 7 0 0 0 2 0 0 0 0 0 0 0 0 2 2 7 2 0 6 7 7 0 0 7 2 0\n",
      " 0 0 0 0 0 0 6 6 0 7 2 0 0 0 0 0 0 0 7 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 7 0 0 0 0 0 0 0 0 0 0 0 0 0 7 0 0 0 0 0 2 0 0 7 7 7 7 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 7 2 0 0 0 7 0 0 0 7 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 7 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 7 0 0 0 0 0 7 0 0 0 0 0 7\n",
      " 0 0 7 0 0 0 0 0 0 2 0 0 0 2 0 0 0 0 0 0 0 7 0 0 0 0 0 0 0 2 0 0 7 7 0 0 7\n",
      " 7 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 7 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 2 7 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 7 7 0 7 0 0 0 0 2 0 2 7 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 7 0 7 0 0 7 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 7 0 0 0 0 0 0 0 7 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 7 0 7 7 0 0 0 0 7 7 0 7 7\n",
      " 0 0 2 0 0 0 0 0 0 0 0 0 0 7 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 7 0 0 0 0 0 0 0 0 0 0 0 0 0 0 2 0 0 0 0 0 7 0 0 0 0 0 0 0\n",
      " 0]\n",
      "threshold density 0.15152151422000254\n",
      "cluster signal 89824.0\n",
      "cluster length 1539300\n",
      "!!cluster density 0.058353797180536605\n",
      "cluster signal 25471.0\n",
      "cluster length 2381800\n",
      "!!cluster density 0.010694012931396423\n",
      "cluster signal 129688.0\n",
      "cluster length 1154500\n",
      "!!cluster density 0.11233261152013858\n",
      "cluster signal 165968.0\n",
      "cluster length 902500\n",
      "!!cluster density 0.18389806094182826\n",
      "cluster signal 41532.0\n",
      "cluster length 80600\n",
      "!!cluster density 0.5152853598014888\n",
      "cluster signal 117688.0\n",
      "cluster length 423100\n",
      "!!cluster density 0.2781564641928622\n",
      "cluster signal 23454.0\n",
      "cluster length 33900\n",
      "!!cluster density 0.691858407079646\n",
      "cluster signal 11174.0\n",
      "cluster length 11400\n",
      "!!cluster density 0.9801754385964913\n",
      "cluster signal 490.0\n",
      "cluster length 100\n",
      "!!cluster density 4.9\n",
      "cluster signal 73290.0\n",
      "cluster length 190500\n",
      "!!cluster density 0.3847244094488189\n",
      "\n",
      "Labels mapping\n",
      " {7: 0, 0: 0, 2: 0, 6: 1, 3: 1, 1: 1, 4: 1, 8: 1, 5: 1, 9: 1}\n",
      "DEBUG: 50756 16421 \n",
      "\n",
      " +--------------------+-------------+---------------+--------------+-------+\n",
      "| bin_offset_50      | TT_S0       | starting_nt   | Chromosome   | +7    |\n",
      "| (category)         | (float64)   | (category)    | (category)   | ...   |\n",
      "|--------------------+-------------+---------------+--------------+-------|\n",
      "| (3000016, 3000066] | 4.0         | 3000016       | 18           | ...   |\n",
      "| (3000066, 3000166] | 4.0         | 3000066       | 18           | ...   |\n",
      "| (3000166, 3000266] | 0.0         | 3000166       | 18           | ...   |\n",
      "| (3000266, 3000366] | 2.0         | 3000266       | 18           | ...   |\n",
      "| ...                | ...         | ...           | ...          | ...   |\n",
      "| (9717266, 9717366] | 0.0         | 9717266       | 18           | ...   |\n",
      "| (9717366, 9717466] | 1.0         | 9717366       | 18           | ...   |\n",
      "| (9717466, 9717566] | 1.0         | 9717466       | 18           | ...   |\n",
      "| (9717566, 9717593] | 0.0         | 9717566       | 18           | ...   |\n",
      "+--------------------+-------------+---------------+--------------+-------+\n",
      "Unstranded PyRanges object has 50,756 rows and 11 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "7 hidden columns: Start, End, genes, transcripts, exons, CDSs, labels\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0\n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "BINS  IntervalIndex([(3000016, 3000071], (3000071, 3000171], (3000171, 3000271],\n",
      "               (3000271, 3000371], (3000371, 3000471], (3000471, 3000571],\n",
      "               (3000571, 3000671], (3000671, 3000771], (3000771, 3000871],\n",
      "               (3000871, 3000971],\n",
      "               ...\n",
      "               (9716671, 9716771], (9716771, 9716871], (9716871, 9716971],\n",
      "               (9716971, 9717071], (9717071, 9717171], (9717171, 9717271],\n",
      "               (9717271, 9717371], (9717371, 9717471], (9717471, 9717571],\n",
      "               (9717571, 9717593]],\n",
      "              dtype='interval[int64, right]', length=67177)\n",
      "HEAD with BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0   \n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "\n",
      "        bin_offset_55  \n",
      "0  (3000016, 3000071]  \n",
      "1  (3000016, 3000071]  \n",
      "2  (3000016, 3000071]  \n",
      "3  (3000016, 3000071]  \n",
      "4  (3000016, 3000071]  \n",
      "TAIL with BINS\n",
      "       seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  9717555  9717555  18_9717555    1.0    0.0    0.0   \n",
      "699996       19  9717571  9717571  19_9717571    0.0    0.0    0.0   \n",
      "699997       19  9717581  9717581  19_9717581    0.0    0.0    0.0   \n",
      "699998       18  9717582  9717582  18_9717582    0.0    0.0    0.0   \n",
      "699999       18  9717593  9717593  18_9717593    0.0    0.0    0.0   \n",
      "\n",
      "             bin_offset_55  \n",
      "699995  (9717471, 9717571]  \n",
      "699996  (9717471, 9717571]  \n",
      "699997  (9717571, 9717593]  \n",
      "699998  (9717571, 9717593]  \n",
      "699999  (9717571, 9717593]  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:136: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "         bin_offset_55  TT_S0 starting_nt\n",
      "0   (3000016, 3000071]    4.0     3000016\n",
      "1   (3000071, 3000171]    4.0     3000071\n",
      "2   (3000171, 3000271]    0.0     3000171\n",
      "3   (3000271, 3000371]    2.0     3000271\n",
      "4   (3000371, 3000471]    0.0     3000371\n",
      "5   (3000471, 3000571]    0.0     3000471\n",
      "6   (3000571, 3000671]    0.0     3000571\n",
      "7   (3000671, 3000771]    0.0     3000671\n",
      "8   (3000771, 3000871]    0.0     3000771\n",
      "9   (3000871, 3000971]    0.0     3000871\n",
      "10  (3000971, 3001071]    2.0     3000971\n",
      "11  (3001071, 3001171]    0.0     3001071\n",
      "12  (3001171, 3001271]    0.0     3001171\n",
      "13  (3001271, 3001371]    0.0     3001271\n",
      "14  (3001371, 3001471]    0.0     3001371\n",
      "15  (3001471, 3001571]    4.0     3001471\n",
      "16  (3001571, 3001671]    3.0     3001571\n",
      "17  (3001671, 3001771]    0.0     3001671\n",
      "18  (3001771, 3001871]    0.0     3001771\n",
      "19  (3001871, 3001971]    0.0     3001871\n",
      "20  (3001971, 3002071]    1.0     3001971\n",
      "21  (3002071, 3002171]    0.0     3002071\n",
      "22  (3002171, 3002271]    4.0     3002171\n",
      "23  (3002271, 3002371]    9.0     3002271\n",
      "24  (3002371, 3002471]   19.0     3002371\n",
      "25  (3002471, 3002571]    4.0     3002471\n",
      "26  (3002571, 3002671]    0.0     3002571\n",
      "27  (3002671, 3002771]    0.0     3002671\n",
      "28  (3002771, 3002871]    0.0     3002771\n",
      "29  (3002871, 3002971]    1.0     3002871\n",
      "30  (3002971, 3003071]    2.0     3002971\n",
      "31  (3003071, 3003171]    0.0     3003071\n",
      "32  (3003171, 3003271]    0.0     3003171\n",
      "33  (3003271, 3003371]    1.0     3003271\n",
      "34  (3003371, 3003471]    0.0     3003371\n",
      "35  (3003471, 3003571]    0.0     3003471\n",
      "36  (3003571, 3003671]    1.0     3003571\n",
      "37  (3003671, 3003771]    4.0     3003671\n",
      "38  (3003771, 3003871]    0.0     3003771\n",
      "39  (3003871, 3003971]    7.0     3003871\n",
      "40  (3003971, 3004071]    1.0     3003971\n",
      "41  (3004071, 3004171]    0.0     3004071\n",
      "42  (3004171, 3004271]    3.0     3004171\n",
      "43  (3004271, 3004371]    2.0     3004271\n",
      "44  (3004371, 3004471]    0.0     3004371\n",
      "45  (3004471, 3004571]    0.0     3004471\n",
      "46  (3004571, 3004671]    1.0     3004571\n",
      "47  (3004671, 3004771]    0.0     3004671\n",
      "48  (3004771, 3004871]   10.0     3004771\n",
      "49  (3004871, 3004971]    3.0     3004871\n",
      "(67177, 3)\n",
      "bin_offset_55    category\n",
      "TT_S0             float64\n",
      "starting_nt      category\n",
      "dtype: object\n",
      "sumos ilgis: 67177\n",
      "ranges ilgis: 67177\n",
      "Model input - NO SCALE\n",
      "       TT_S0  genes  transcripts  exons  CDSs\n",
      "0        4.0      0            0      0     0\n",
      "1        4.0      0            0      0     0\n",
      "2        0.0      0            0      0     0\n",
      "3        2.0      0            0      0     0\n",
      "4        0.0      0            0      0     0\n",
      "...      ...    ...          ...    ...   ...\n",
      "67172    0.0      0            0      0     0\n",
      "67173    0.0      0            0      0     0\n",
      "67174    1.0      0            0      0     0\n",
      "67175    1.0      0            0      0     0\n",
      "67176    0.0      0            0      0     0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "       TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0        4.0             0                   0             0            0\n",
      "1        4.0             0                   0             0            0\n",
      "2        0.0             0                   0             0            0\n",
      "3        2.0             0                   0             0            0\n",
      "4        0.0             0                   0             0            0\n",
      "...      ...           ...                 ...           ...          ...\n",
      "67172    0.0             0                   0             0            0\n",
      "67173    0.0             0                   0             0            0\n",
      "67174    1.0             0                   0             0            0\n",
      "67175    1.0             0                   0             0            0\n",
      "67176    0.0             0                   0             0            0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "0.3539753189335636\n",
      "[8 8 1 1 1 1 1 1 1 1 1 1 1 1 1 8 1 1 1 1 1 1 8 3 0 8 1 1 1 1 1 1 1 1 1 1 1\n",
      " 8 1 8 1 1 1 1 1 1 1 1 3 1 1 2 4 3 1 4 6 5 5 7 1 8 0 8 1 1 1 8 1 1 3 1 8 1\n",
      " 1 1 1 1 1 1 1 8 1 1 1 1 1 3 1 1 8 9 3 1 1 1 1 1 8 8 9 0 4 0 8 8 3 4 8 3 1\n",
      " 3 8 1 1 1 1 1 1 1 1 8 1 1 1 9 8 0 9 3 9 1 9 1 8 8 1 3 1 1 1 1 1 1 1 1 1 0\n",
      " 1 8 8 1 1 8 1 3 1 1 1 9 8 8 3 1 8 9 1 1 1 1 1 1 1 1 1 1 1 1 1 1 3 8 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 3 9 3 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 8 1 8 8 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 8 1 1 1 1 3 1 1 1 1 1 1 1 1 3 9 8 9 1 0 3 8 1 1 8 3 1\n",
      " 1 1 1 1 1 1 9 9 1 8 3 1 1 1 1 1 1 1 8 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 8 1 1 1 1 1 1 1 1 1 1 1 1 1 3 1 1 1 1 1 3 1 1 8 3 8 8 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 8 9 1 1 1 8 1 1 1 3 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 8 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 8 1 1 1 1 1 8 1 1 1 1 1 8\n",
      " 1 1 8 1 1 1 1 1 1 9 1 1 1 3 1 1 1 1 1 1 8 1 1 1 1 1 1 1 8 8 1 1 8 8 1 8 1\n",
      " 8 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 8 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 3 8 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 8 1 1 1 1 1 8 8 1 8 1 1 1 1 3 1 3 8 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 8 1 8 1 1 8 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 8 1 1 1 1 1 1 8 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 8 1 8 8 1 1 1 1 8 8 1 8 8\n",
      " 1 8 3 1 1 1 1 1 1 1 1 1 1 8 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 8 1 1 1 1 1 1 1 1 1 1 1 1 1 1 3 1 1 1 1 1 3 1 1 1 1 1 1 1\n",
      " 1]\n",
      "threshold density 0.15152140144142562\n",
      "cluster signal 69894.0\n",
      "cluster length 1292800\n",
      "!!cluster density 0.05406404702970297\n",
      "cluster signal 25252.0\n",
      "cluster length 2377900\n",
      "!!cluster density 0.010619454140207747\n",
      "cluster signal 85167.0\n",
      "cluster length 907100\n",
      "!!cluster density 0.09388931760555617\n",
      "cluster signal 144702.0\n",
      "cluster length 676800\n",
      "!!cluster density 0.2138031914893617\n",
      "cluster signal 66969.0\n",
      "cluster length 149000\n",
      "!!cluster density 0.44945637583892617\n",
      "cluster signal 111840.0\n",
      "cluster length 356600\n",
      "!!cluster density 0.3136287156477846\n",
      "cluster signal 33816.0\n",
      "cluster length 53000\n",
      "!!cluster density 0.6380377358490567\n",
      "cluster signal 13210.0\n",
      "cluster length 14000\n",
      "!!cluster density 0.9435714285714286\n",
      "cluster signal 490.0\n",
      "cluster length 100\n",
      "!!cluster density 4.9\n",
      "cluster signal 127239.0\n",
      "cluster length 890400\n",
      "!!cluster density 0.14290094339622642\n",
      "\n",
      "Labels mapping\n",
      " {8: 0, 1: 0, 3: 0, 0: 1, 2: 1, 4: 1, 6: 1, 5: 1, 7: 1, 9: 0}\n",
      "DEBUG: 54682 12495 \n",
      "\n",
      " +--------------------+-------------+---------------+--------------+-------+\n",
      "| bin_offset_55      | TT_S0       | starting_nt   | Chromosome   | +7    |\n",
      "| (category)         | (float64)   | (category)    | (category)   | ...   |\n",
      "|--------------------+-------------+---------------+--------------+-------|\n",
      "| (3000016, 3000071] | 4.0         | 3000016       | 18           | ...   |\n",
      "| (3000071, 3000171] | 4.0         | 3000071       | 18           | ...   |\n",
      "| (3000171, 3000271] | 0.0         | 3000171       | 18           | ...   |\n",
      "| (3000271, 3000371] | 2.0         | 3000271       | 18           | ...   |\n",
      "| ...                | ...         | ...           | ...          | ...   |\n",
      "| (9717271, 9717371] | 0.0         | 9717271       | 18           | ...   |\n",
      "| (9717371, 9717471] | 1.0         | 9717371       | 18           | ...   |\n",
      "| (9717471, 9717571] | 1.0         | 9717471       | 18           | ...   |\n",
      "| (9717571, 9717593] | 0.0         | 9717571       | 18           | ...   |\n",
      "+--------------------+-------------+---------------+--------------+-------+\n",
      "Unstranded PyRanges object has 54,682 rows and 11 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "7 hidden columns: Start, End, genes, transcripts, exons, CDSs, labels\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0\n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "BINS  IntervalIndex([(3000016, 3000076], (3000076, 3000176], (3000176, 3000276],\n",
      "               (3000276, 3000376], (3000376, 3000476], (3000476, 3000576],\n",
      "               (3000576, 3000676], (3000676, 3000776], (3000776, 3000876],\n",
      "               (3000876, 3000976],\n",
      "               ...\n",
      "               (9716676, 9716776], (9716776, 9716876], (9716876, 9716976],\n",
      "               (9716976, 9717076], (9717076, 9717176], (9717176, 9717276],\n",
      "               (9717276, 9717376], (9717376, 9717476], (9717476, 9717576],\n",
      "               (9717576, 9717593]],\n",
      "              dtype='interval[int64, right]', length=67177)\n",
      "HEAD with BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0   \n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "\n",
      "        bin_offset_60  \n",
      "0  (3000016, 3000076]  \n",
      "1  (3000016, 3000076]  \n",
      "2  (3000016, 3000076]  \n",
      "3  (3000016, 3000076]  \n",
      "4  (3000016, 3000076]  \n",
      "TAIL with BINS\n",
      "       seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  9717555  9717555  18_9717555    1.0    0.0    0.0   \n",
      "699996       19  9717571  9717571  19_9717571    0.0    0.0    0.0   \n",
      "699997       19  9717581  9717581  19_9717581    0.0    0.0    0.0   \n",
      "699998       18  9717582  9717582  18_9717582    0.0    0.0    0.0   \n",
      "699999       18  9717593  9717593  18_9717593    0.0    0.0    0.0   \n",
      "\n",
      "             bin_offset_60  \n",
      "699995  (9717476, 9717576]  \n",
      "699996  (9717476, 9717576]  \n",
      "699997  (9717576, 9717593]  \n",
      "699998  (9717576, 9717593]  \n",
      "699999  (9717576, 9717593]  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:136: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "         bin_offset_60  TT_S0 starting_nt\n",
      "0   (3000016, 3000076]    4.0     3000016\n",
      "1   (3000076, 3000176]    4.0     3000076\n",
      "2   (3000176, 3000276]    0.0     3000176\n",
      "3   (3000276, 3000376]    2.0     3000276\n",
      "4   (3000376, 3000476]    0.0     3000376\n",
      "5   (3000476, 3000576]    0.0     3000476\n",
      "6   (3000576, 3000676]    0.0     3000576\n",
      "7   (3000676, 3000776]    0.0     3000676\n",
      "8   (3000776, 3000876]    0.0     3000776\n",
      "9   (3000876, 3000976]    0.0     3000876\n",
      "10  (3000976, 3001076]    2.0     3000976\n",
      "11  (3001076, 3001176]    0.0     3001076\n",
      "12  (3001176, 3001276]    0.0     3001176\n",
      "13  (3001276, 3001376]    0.0     3001276\n",
      "14  (3001376, 3001476]    0.0     3001376\n",
      "15  (3001476, 3001576]    4.0     3001476\n",
      "16  (3001576, 3001676]    3.0     3001576\n",
      "17  (3001676, 3001776]    0.0     3001676\n",
      "18  (3001776, 3001876]    0.0     3001776\n",
      "19  (3001876, 3001976]    0.0     3001876\n",
      "20  (3001976, 3002076]    1.0     3001976\n",
      "21  (3002076, 3002176]    0.0     3002076\n",
      "22  (3002176, 3002276]    4.0     3002176\n",
      "23  (3002276, 3002376]    9.0     3002276\n",
      "24  (3002376, 3002476]   19.0     3002376\n",
      "25  (3002476, 3002576]    4.0     3002476\n",
      "26  (3002576, 3002676]    0.0     3002576\n",
      "27  (3002676, 3002776]    0.0     3002676\n",
      "28  (3002776, 3002876]    0.0     3002776\n",
      "29  (3002876, 3002976]    1.0     3002876\n",
      "30  (3002976, 3003076]    2.0     3002976\n",
      "31  (3003076, 3003176]    0.0     3003076\n",
      "32  (3003176, 3003276]    0.0     3003176\n",
      "33  (3003276, 3003376]    1.0     3003276\n",
      "34  (3003376, 3003476]    0.0     3003376\n",
      "35  (3003476, 3003576]    0.0     3003476\n",
      "36  (3003576, 3003676]    1.0     3003576\n",
      "37  (3003676, 3003776]    4.0     3003676\n",
      "38  (3003776, 3003876]    0.0     3003776\n",
      "39  (3003876, 3003976]    7.0     3003876\n",
      "40  (3003976, 3004076]    1.0     3003976\n",
      "41  (3004076, 3004176]    0.0     3004076\n",
      "42  (3004176, 3004276]    3.0     3004176\n",
      "43  (3004276, 3004376]    2.0     3004276\n",
      "44  (3004376, 3004476]    0.0     3004376\n",
      "45  (3004476, 3004576]    0.0     3004476\n",
      "46  (3004576, 3004676]    1.0     3004576\n",
      "47  (3004676, 3004776]    0.0     3004676\n",
      "48  (3004776, 3004876]   10.0     3004776\n",
      "49  (3004876, 3004976]    3.0     3004876\n",
      "(67177, 3)\n",
      "bin_offset_60    category\n",
      "TT_S0             float64\n",
      "starting_nt      category\n",
      "dtype: object\n",
      "sumos ilgis: 67177\n",
      "ranges ilgis: 67177\n",
      "Model input - NO SCALE\n",
      "       TT_S0  genes  transcripts  exons  CDSs\n",
      "0        4.0      0            0      0     0\n",
      "1        4.0      0            0      0     0\n",
      "2        0.0      0            0      0     0\n",
      "3        2.0      0            0      0     0\n",
      "4        0.0      0            0      0     0\n",
      "...      ...    ...          ...    ...   ...\n",
      "67172    0.0      0            0      0     0\n",
      "67173    0.0      0            0      0     0\n",
      "67174    1.0      0            0      0     0\n",
      "67175    1.0      0            0      0     0\n",
      "67176    0.0      0            0      0     0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "       TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0        4.0             0                   0             0            0\n",
      "1        4.0             0                   0             0            0\n",
      "2        0.0             0                   0             0            0\n",
      "3        2.0             0                   0             0            0\n",
      "4        0.0             0                   0             0            0\n",
      "...      ...           ...                 ...           ...          ...\n",
      "67172    0.0             0                   0             0            0\n",
      "67173    0.0             0                   0             0            0\n",
      "67174    1.0             0                   0             0            0\n",
      "67175    1.0             0                   0             0            0\n",
      "67176    0.0             0                   0             0            0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "0.05296455632135999\n",
      "[0 0 4 4 4 4 4 4 4 4 4 4 4 4 4 0 4 4 4 4 4 4 0 9 5 0 4 4 4 4 4 4 4 4 4 4 4\n",
      " 0 4 0 4 4 4 4 4 4 4 4 9 4 4 3 1 9 4 1 6 6 6 8 4 0 5 0 4 4 4 0 4 4 0 4 0 4\n",
      " 4 4 4 4 4 4 9 4 4 4 4 4 4 9 4 4 0 9 9 4 4 4 4 4 0 0 7 5 1 5 0 0 9 1 0 9 4\n",
      " 9 0 4 4 4 4 4 4 4 4 0 4 4 4 7 0 5 7 9 9 4 7 4 0 0 4 9 4 4 4 4 4 4 4 4 4 7\n",
      " 4 0 0 4 4 0 4 9 4 4 4 9 0 0 9 4 0 7 4 4 4 4 4 4 4 4 4 4 4 4 4 4 9 0 4 4 4\n",
      " 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 9 7 0 4 4 4 4 4 4 4 4 4 4 4 4 4\n",
      " 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4\n",
      " 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4\n",
      " 4 4 4 4 4 4 4 4 4 4 4 4 0 4 4 0 0 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4\n",
      " 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4\n",
      " 4 4 4 4 4 4 4 4 4 4 0 4 4 4 0 9 4 4 4 4 4 4 4 4 9 7 0 9 4 5 0 0 4 4 0 9 4\n",
      " 4 4 4 4 4 4 7 7 4 0 9 4 4 4 4 4 4 4 0 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4\n",
      " 4 4 4 4 0 4 4 4 4 4 4 4 4 4 4 4 4 4 0 4 4 4 4 4 9 4 4 0 0 0 0 4 4 4 4 4 4\n",
      " 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4\n",
      " 0 9 4 4 4 0 4 4 4 0 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4\n",
      " 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4\n",
      " 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 0 4 4 4 4 4 0 4 4 4 4 4 0\n",
      " 4 4 0 4 4 4 4 4 4 9 4 4 4 9 4 4 4 4 4 4 0 4 4 4 4 4 4 4 0 0 4 4 0 0 4 0 4\n",
      " 0 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 0 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4\n",
      " 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 9 0 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4\n",
      " 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4\n",
      " 4 4 4 4 4 4 4 4 4 4 4 4 4 0 4 4 4 4 4 0 0 4 0 4 4 4 4 9 4 9 0 4 4 4 4 4 4\n",
      " 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 0 4 0 4 4 0 4 4 4 4 4 4 4 4 4\n",
      " 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 0 4 4 4 4 4 4 0 4 4 4 4 4 4 4 4 4 4 4\n",
      " 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 0 4 0 0 4 4 4 4 9 4 4 0 0\n",
      " 4 0 9 4 4 4 4 4 4 4 4 4 4 0 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4\n",
      " 4 4 4 4 4 4 4 4 0 4 4 4 4 4 4 4 4 4 4 4 4 4 4 9 4 4 4 4 4 0 4 4 4 4 4 4 4\n",
      " 4]\n",
      "threshold density 0.1515212886630166\n",
      "cluster signal 90425.0\n",
      "cluster length 1549200\n",
      "!!cluster density 0.05836883552801446\n",
      "cluster signal 25152.0\n",
      "cluster length 2377900\n",
      "!!cluster density 0.010577400227091131\n",
      "cluster signal 107371.0\n",
      "cluster length 992800\n",
      "!!cluster density 0.1081496776792909\n",
      "cluster signal 111599.0\n",
      "cluster length 515500\n",
      "!!cluster density 0.21648690591658584\n",
      "cluster signal 76915.0\n",
      "cluster length 183300\n",
      "!!cluster density 0.4196126568466994\n",
      "cluster signal 106012.0\n",
      "cluster length 355800\n",
      "!!cluster density 0.2979539066891512\n",
      "cluster signal 16213.0\n",
      "cluster length 17900\n",
      "!!cluster density 0.905754189944134\n",
      "cluster signal 490.0\n",
      "cluster length 100\n",
      "!!cluster density 4.9\n",
      "cluster signal 104128.0\n",
      "cluster length 658300\n",
      "!!cluster density 0.15817712289229835\n",
      "cluster signal 40274.0\n",
      "cluster length 66900\n",
      "!!cluster density 0.6020029895366218\n",
      "\n",
      "Labels mapping\n",
      " {0: 0, 4: 0, 9: 0, 5: 1, 3: 1, 1: 1, 6: 1, 8: 1, 7: 1, 2: 1}\n",
      "DEBUG: 49199 17978 \n",
      "\n",
      " +--------------------+-------------+---------------+--------------+-------+\n",
      "| bin_offset_60      | TT_S0       | starting_nt   | Chromosome   | +7    |\n",
      "| (category)         | (float64)   | (category)    | (category)   | ...   |\n",
      "|--------------------+-------------+---------------+--------------+-------|\n",
      "| (3000016, 3000076] | 4.0         | 3000016       | 18           | ...   |\n",
      "| (3000076, 3000176] | 4.0         | 3000076       | 18           | ...   |\n",
      "| (3000176, 3000276] | 0.0         | 3000176       | 18           | ...   |\n",
      "| (3000276, 3000376] | 2.0         | 3000276       | 18           | ...   |\n",
      "| ...                | ...         | ...           | ...          | ...   |\n",
      "| (9717276, 9717376] | 0.0         | 9717276       | 18           | ...   |\n",
      "| (9717376, 9717476] | 1.0         | 9717376       | 18           | ...   |\n",
      "| (9717476, 9717576] | 1.0         | 9717476       | 18           | ...   |\n",
      "| (9717576, 9717593] | 0.0         | 9717576       | 18           | ...   |\n",
      "+--------------------+-------------+---------------+--------------+-------+\n",
      "Unstranded PyRanges object has 49,199 rows and 11 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "7 hidden columns: Start, End, genes, transcripts, exons, CDSs, labels\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0\n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "BINS  IntervalIndex([(3000016, 3000081], (3000081, 3000181], (3000181, 3000281],\n",
      "               (3000281, 3000381], (3000381, 3000481], (3000481, 3000581],\n",
      "               (3000581, 3000681], (3000681, 3000781], (3000781, 3000881],\n",
      "               (3000881, 3000981],\n",
      "               ...\n",
      "               (9716681, 9716781], (9716781, 9716881], (9716881, 9716981],\n",
      "               (9716981, 9717081], (9717081, 9717181], (9717181, 9717281],\n",
      "               (9717281, 9717381], (9717381, 9717481], (9717481, 9717581],\n",
      "               (9717581, 9717593]],\n",
      "              dtype='interval[int64, right]', length=67177)\n",
      "HEAD with BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0   \n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "\n",
      "        bin_offset_65  \n",
      "0  (3000016, 3000081]  \n",
      "1  (3000016, 3000081]  \n",
      "2  (3000016, 3000081]  \n",
      "3  (3000016, 3000081]  \n",
      "4  (3000016, 3000081]  \n",
      "TAIL with BINS\n",
      "       seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  9717555  9717555  18_9717555    1.0    0.0    0.0   \n",
      "699996       19  9717571  9717571  19_9717571    0.0    0.0    0.0   \n",
      "699997       19  9717581  9717581  19_9717581    0.0    0.0    0.0   \n",
      "699998       18  9717582  9717582  18_9717582    0.0    0.0    0.0   \n",
      "699999       18  9717593  9717593  18_9717593    0.0    0.0    0.0   \n",
      "\n",
      "             bin_offset_65  \n",
      "699995  (9717481, 9717581]  \n",
      "699996  (9717481, 9717581]  \n",
      "699997  (9717481, 9717581]  \n",
      "699998  (9717581, 9717593]  \n",
      "699999  (9717581, 9717593]  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:136: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "         bin_offset_65  TT_S0 starting_nt\n",
      "0   (3000016, 3000081]    4.0     3000016\n",
      "1   (3000081, 3000181]    4.0     3000081\n",
      "2   (3000181, 3000281]    0.0     3000181\n",
      "3   (3000281, 3000381]    2.0     3000281\n",
      "4   (3000381, 3000481]    0.0     3000381\n",
      "5   (3000481, 3000581]    0.0     3000481\n",
      "6   (3000581, 3000681]    0.0     3000581\n",
      "7   (3000681, 3000781]    0.0     3000681\n",
      "8   (3000781, 3000881]    0.0     3000781\n",
      "9   (3000881, 3000981]    0.0     3000881\n",
      "10  (3000981, 3001081]    2.0     3000981\n",
      "11  (3001081, 3001181]    0.0     3001081\n",
      "12  (3001181, 3001281]    0.0     3001181\n",
      "13  (3001281, 3001381]    0.0     3001281\n",
      "14  (3001381, 3001481]    0.0     3001381\n",
      "15  (3001481, 3001581]    4.0     3001481\n",
      "16  (3001581, 3001681]    3.0     3001581\n",
      "17  (3001681, 3001781]    0.0     3001681\n",
      "18  (3001781, 3001881]    0.0     3001781\n",
      "19  (3001881, 3001981]    0.0     3001881\n",
      "20  (3001981, 3002081]    1.0     3001981\n",
      "21  (3002081, 3002181]    0.0     3002081\n",
      "22  (3002181, 3002281]    4.0     3002181\n",
      "23  (3002281, 3002381]    9.0     3002281\n",
      "24  (3002381, 3002481]   19.0     3002381\n",
      "25  (3002481, 3002581]    4.0     3002481\n",
      "26  (3002581, 3002681]    0.0     3002581\n",
      "27  (3002681, 3002781]    0.0     3002681\n",
      "28  (3002781, 3002881]    0.0     3002781\n",
      "29  (3002881, 3002981]    1.0     3002881\n",
      "30  (3002981, 3003081]    2.0     3002981\n",
      "31  (3003081, 3003181]    0.0     3003081\n",
      "32  (3003181, 3003281]    0.0     3003181\n",
      "33  (3003281, 3003381]    1.0     3003281\n",
      "34  (3003381, 3003481]    0.0     3003381\n",
      "35  (3003481, 3003581]    0.0     3003481\n",
      "36  (3003581, 3003681]    1.0     3003581\n",
      "37  (3003681, 3003781]    4.0     3003681\n",
      "38  (3003781, 3003881]    0.0     3003781\n",
      "39  (3003881, 3003981]    8.0     3003881\n",
      "40  (3003981, 3004081]    0.0     3003981\n",
      "41  (3004081, 3004181]    0.0     3004081\n",
      "42  (3004181, 3004281]    3.0     3004181\n",
      "43  (3004281, 3004381]    2.0     3004281\n",
      "44  (3004381, 3004481]    0.0     3004381\n",
      "45  (3004481, 3004581]    0.0     3004481\n",
      "46  (3004581, 3004681]    1.0     3004581\n",
      "47  (3004681, 3004781]    0.0     3004681\n",
      "48  (3004781, 3004881]   10.0     3004781\n",
      "49  (3004881, 3004981]    3.0     3004881\n",
      "(67177, 3)\n",
      "bin_offset_65    category\n",
      "TT_S0             float64\n",
      "starting_nt      category\n",
      "dtype: object\n",
      "sumos ilgis: 67177\n",
      "ranges ilgis: 67177\n",
      "Model input - NO SCALE\n",
      "       TT_S0  genes  transcripts  exons  CDSs\n",
      "0        4.0      0            0      0     0\n",
      "1        4.0      0            0      0     0\n",
      "2        0.0      0            0      0     0\n",
      "3        2.0      0            0      0     0\n",
      "4        0.0      0            0      0     0\n",
      "...      ...    ...          ...    ...   ...\n",
      "67172    0.0      0            0      0     0\n",
      "67173    0.0      0            0      0     0\n",
      "67174    1.0      0            0      0     0\n",
      "67175    1.0      0            0      0     0\n",
      "67176    0.0      0            0      0     0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "       TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0        4.0             0                   0             0            0\n",
      "1        4.0             0                   0             0            0\n",
      "2        0.0             0                   0             0            0\n",
      "3        2.0             0                   0             0            0\n",
      "4        0.0             0                   0             0            0\n",
      "...      ...           ...                 ...           ...          ...\n",
      "67172    0.0             0                   0             0            0\n",
      "67173    0.0             0                   0             0            0\n",
      "67174    1.0             0                   0             0            0\n",
      "67175    1.0             0                   0             0            0\n",
      "67176    0.0             0                   0             0            0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "0.029816752757640265\n",
      "[2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 6 5 2 2 2 2 2 2 2 2 2 2 2 2\n",
      " 2 2 6 2 2 2 2 2 2 2 2 0 2 2 9 3 0 2 1 7 4 4 8 2 0 5 6 2 2 6 2 2 2 6 2 2 2\n",
      " 2 2 2 2 2 2 6 2 2 2 2 2 2 6 2 2 6 0 0 2 2 2 2 2 6 6 5 5 3 5 6 2 0 3 6 0 2\n",
      " 6 6 2 2 2 2 2 2 2 2 2 2 2 2 0 6 5 0 6 0 2 0 2 2 2 2 6 2 2 2 2 2 2 2 2 2 5\n",
      " 2 6 6 2 2 6 2 0 2 2 2 5 2 6 6 2 6 0 2 2 2 2 2 2 2 2 2 2 2 2 2 2 0 6 2 2 2\n",
      " 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 0 5 2 2 2 2 2 2 2 2 2 2 2 2 2 2\n",
      " 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2\n",
      " 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2\n",
      " 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 6 6 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2\n",
      " 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2\n",
      " 2 2 2 2 2 2 2 2 2 2 6 2 2 2 2 6 2 2 2 2 2 2 2 2 0 0 2 0 2 5 6 6 2 2 6 0 2\n",
      " 2 2 2 2 2 2 0 5 2 6 0 2 2 2 2 2 2 2 6 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2\n",
      " 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 6 2 2 2 2 2 0 2 2 2 6 6 6 2 2 2 2 2 2\n",
      " 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2\n",
      " 6 0 2 2 2 6 2 2 2 6 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2\n",
      " 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2\n",
      " 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 6 2 2 2 2 2 6 2 2 2 2 2 6\n",
      " 2 2 2 2 2 2 2 2 2 0 2 2 2 6 2 2 2 2 2 2 6 2 2 2 2 2 2 2 2 6 2 2 6 2 2 6 2\n",
      " 6 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2\n",
      " 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 6 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2\n",
      " 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2\n",
      " 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 6 6 2 2 2 2 2 2 0 2 0 6 2 2 2 2 2 2\n",
      " 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 6 2 2 2 2 2 2 2 2 2\n",
      " 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 6 2 2 2 2 2 2 6 2 2 2 2 2 2 2 2 2 2 2\n",
      " 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 6 2 2 2 2 2 2 6 6 2 2 6 2\n",
      " 2 6 6 2 2 2 2 2 2 2 2 2 2 6 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2\n",
      " 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 0 2 2 2 2 2 6 2 2 2 2 2 2 2\n",
      " 2]\n",
      "threshold density 0.15152117588477543\n",
      "cluster signal 39990.0\n",
      "cluster length 2750300\n",
      "!!cluster density 0.01454023197469367\n",
      "cluster signal 95538.0\n",
      "cluster length 1399500\n",
      "!!cluster density 0.0682658092175777\n",
      "cluster signal 144292.0\n",
      "cluster length 759600\n",
      "!!cluster density 0.18995787256450763\n",
      "cluster signal 130927.0\n",
      "cluster length 1069300\n",
      "!!cluster density 0.12244178434489854\n",
      "cluster signal 47047.0\n",
      "cluster length 94800\n",
      "!!cluster density 0.4962763713080169\n",
      "cluster signal 107261.0\n",
      "cluster length 391600\n",
      "!!cluster density 0.27390449438202247\n",
      "cluster signal 74290.0\n",
      "cluster length 200300\n",
      "!!cluster density 0.3708936595107339\n",
      "cluster signal 27072.0\n",
      "cluster length 40100\n",
      "!!cluster density 0.6751122194513716\n",
      "cluster signal 11670.0\n",
      "cluster length 12100\n",
      "!!cluster density 0.9644628099173553\n",
      "cluster signal 492.0\n",
      "cluster length 100\n",
      "!!cluster density 4.92\n",
      "\n",
      "Labels mapping\n",
      " {2: 0, 6: 0, 5: 1, 0: 0, 9: 1, 3: 1, 1: 1, 7: 1, 4: 1, 8: 1}\n",
      "DEBUG: 52191 14986 \n",
      "\n",
      " +--------------------+-------------+---------------+--------------+-------+\n",
      "| bin_offset_65      | TT_S0       | starting_nt   | Chromosome   | +7    |\n",
      "| (category)         | (float64)   | (category)    | (category)   | ...   |\n",
      "|--------------------+-------------+---------------+--------------+-------|\n",
      "| (3000016, 3000081] | 4.0         | 3000016       | 18           | ...   |\n",
      "| (3000081, 3000181] | 4.0         | 3000081       | 18           | ...   |\n",
      "| (3000181, 3000281] | 0.0         | 3000181       | 18           | ...   |\n",
      "| (3000281, 3000381] | 2.0         | 3000281       | 18           | ...   |\n",
      "| ...                | ...         | ...           | ...          | ...   |\n",
      "| (9717281, 9717381] | 0.0         | 9717281       | 18           | ...   |\n",
      "| (9717381, 9717481] | 1.0         | 9717381       | 18           | ...   |\n",
      "| (9717481, 9717581] | 1.0         | 9717481       | 18           | ...   |\n",
      "| (9717581, 9717593] | 0.0         | 9717581       | 18           | ...   |\n",
      "+--------------------+-------------+---------------+--------------+-------+\n",
      "Unstranded PyRanges object has 52,191 rows and 11 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "7 hidden columns: Start, End, genes, transcripts, exons, CDSs, labels\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0\n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "BINS  IntervalIndex([(3000016, 3000086], (3000086, 3000186], (3000186, 3000286],\n",
      "               (3000286, 3000386], (3000386, 3000486], (3000486, 3000586],\n",
      "               (3000586, 3000686], (3000686, 3000786], (3000786, 3000886],\n",
      "               (3000886, 3000986],\n",
      "               ...\n",
      "               (9716686, 9716786], (9716786, 9716886], (9716886, 9716986],\n",
      "               (9716986, 9717086], (9717086, 9717186], (9717186, 9717286],\n",
      "               (9717286, 9717386], (9717386, 9717486], (9717486, 9717586],\n",
      "               (9717586, 9717593]],\n",
      "              dtype='interval[int64, right]', length=67177)\n",
      "HEAD with BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0   \n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "\n",
      "        bin_offset_70  \n",
      "0  (3000016, 3000086]  \n",
      "1  (3000016, 3000086]  \n",
      "2  (3000016, 3000086]  \n",
      "3  (3000016, 3000086]  \n",
      "4  (3000016, 3000086]  \n",
      "TAIL with BINS\n",
      "       seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  9717555  9717555  18_9717555    1.0    0.0    0.0   \n",
      "699996       19  9717571  9717571  19_9717571    0.0    0.0    0.0   \n",
      "699997       19  9717581  9717581  19_9717581    0.0    0.0    0.0   \n",
      "699998       18  9717582  9717582  18_9717582    0.0    0.0    0.0   \n",
      "699999       18  9717593  9717593  18_9717593    0.0    0.0    0.0   \n",
      "\n",
      "             bin_offset_70  \n",
      "699995  (9717486, 9717586]  \n",
      "699996  (9717486, 9717586]  \n",
      "699997  (9717486, 9717586]  \n",
      "699998  (9717486, 9717586]  \n",
      "699999  (9717586, 9717593]  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:136: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "         bin_offset_70  TT_S0 starting_nt\n",
      "0   (3000016, 3000086]    7.0     3000016\n",
      "1   (3000086, 3000186]    1.0     3000086\n",
      "2   (3000186, 3000286]    0.0     3000186\n",
      "3   (3000286, 3000386]    2.0     3000286\n",
      "4   (3000386, 3000486]    0.0     3000386\n",
      "5   (3000486, 3000586]    0.0     3000486\n",
      "6   (3000586, 3000686]    0.0     3000586\n",
      "7   (3000686, 3000786]    0.0     3000686\n",
      "8   (3000786, 3000886]    0.0     3000786\n",
      "9   (3000886, 3000986]    0.0     3000886\n",
      "10  (3000986, 3001086]    2.0     3000986\n",
      "11  (3001086, 3001186]    0.0     3001086\n",
      "12  (3001186, 3001286]    0.0     3001186\n",
      "13  (3001286, 3001386]    0.0     3001286\n",
      "14  (3001386, 3001486]    0.0     3001386\n",
      "15  (3001486, 3001586]    4.0     3001486\n",
      "16  (3001586, 3001686]    3.0     3001586\n",
      "17  (3001686, 3001786]    0.0     3001686\n",
      "18  (3001786, 3001886]    0.0     3001786\n",
      "19  (3001886, 3001986]    0.0     3001886\n",
      "20  (3001986, 3002086]    1.0     3001986\n",
      "21  (3002086, 3002186]    0.0     3002086\n",
      "22  (3002186, 3002286]    4.0     3002186\n",
      "23  (3002286, 3002386]    9.0     3002286\n",
      "24  (3002386, 3002486]   19.0     3002386\n",
      "25  (3002486, 3002586]    4.0     3002486\n",
      "26  (3002586, 3002686]    0.0     3002586\n",
      "27  (3002686, 3002786]    0.0     3002686\n",
      "28  (3002786, 3002886]    0.0     3002786\n",
      "29  (3002886, 3002986]    1.0     3002886\n",
      "30  (3002986, 3003086]    2.0     3002986\n",
      "31  (3003086, 3003186]    0.0     3003086\n",
      "32  (3003186, 3003286]    0.0     3003186\n",
      "33  (3003286, 3003386]    1.0     3003286\n",
      "34  (3003386, 3003486]    0.0     3003386\n",
      "35  (3003486, 3003586]    1.0     3003486\n",
      "36  (3003586, 3003686]    1.0     3003586\n",
      "37  (3003686, 3003786]    3.0     3003686\n",
      "38  (3003786, 3003886]    0.0     3003786\n",
      "39  (3003886, 3003986]    8.0     3003886\n",
      "40  (3003986, 3004086]    0.0     3003986\n",
      "41  (3004086, 3004186]    0.0     3004086\n",
      "42  (3004186, 3004286]    3.0     3004186\n",
      "43  (3004286, 3004386]    2.0     3004286\n",
      "44  (3004386, 3004486]    0.0     3004386\n",
      "45  (3004486, 3004586]    0.0     3004486\n",
      "46  (3004586, 3004686]    1.0     3004586\n",
      "47  (3004686, 3004786]    0.0     3004686\n",
      "48  (3004786, 3004886]   10.0     3004786\n",
      "49  (3004886, 3004986]    3.0     3004886\n",
      "(67177, 3)\n",
      "bin_offset_70    category\n",
      "TT_S0             float64\n",
      "starting_nt      category\n",
      "dtype: object\n",
      "sumos ilgis: 67177\n",
      "ranges ilgis: 67177\n",
      "Model input - NO SCALE\n",
      "       TT_S0  genes  transcripts  exons  CDSs\n",
      "0        7.0      0            0      0     0\n",
      "1        1.0      0            0      0     0\n",
      "2        0.0      0            0      0     0\n",
      "3        2.0      0            0      0     0\n",
      "4        0.0      0            0      0     0\n",
      "...      ...    ...          ...    ...   ...\n",
      "67172    0.0      0            0      0     0\n",
      "67173    0.0      0            0      0     0\n",
      "67174    1.0      0            0      0     0\n",
      "67175    1.0      0            0      0     0\n",
      "67176    0.0      0            0      0     0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "       TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0        7.0             0                   0             0            0\n",
      "1        1.0             0                   0             0            0\n",
      "2        0.0             0                   0             0            0\n",
      "3        2.0             0                   0             0            0\n",
      "4        0.0             0                   0             0            0\n",
      "...      ...           ...                 ...           ...          ...\n",
      "67172    0.0             0                   0             0            0\n",
      "67173    0.0             0                   0             0            0\n",
      "67174    1.0             0                   0             0            0\n",
      "67175    1.0             0                   0             0            0\n",
      "67176    0.0             0                   0             0            0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "0.07155722940887506\n",
      "[9 7 7 0 7 7 7 7 7 7 0 7 7 7 7 0 0 7 7 7 7 7 0 9 3 0 7 7 7 7 0 7 7 7 7 7 7\n",
      " 0 7 9 7 7 0 0 7 7 7 7 4 0 7 2 1 4 3 3 8 8 8 6 7 4 3 0 7 7 9 7 7 7 9 7 0 7\n",
      " 7 7 0 7 7 0 9 7 0 0 0 7 0 9 7 7 9 4 4 7 0 0 7 0 9 0 3 3 1 3 9 0 4 1 0 4 7\n",
      " 9 9 7 7 7 7 7 7 7 7 0 0 7 7 3 9 1 4 4 4 9 9 0 0 0 0 9 7 7 7 7 7 0 7 7 0 3\n",
      " 7 9 0 0 7 0 0 4 0 0 7 3 0 9 0 7 9 4 0 7 7 7 7 7 7 7 0 7 7 7 7 7 4 0 7 0 7\n",
      " 0 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 4 3 7 7 7 7 0 7 7 7 7 0 7 7 7 7\n",
      " 7 7 7 7 7 7 7 7 7 7 7 7 7 0 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 0 7 7\n",
      " 7 7 7 7 7 0 7 7 7 0 0 7 0 0 0 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 0\n",
      " 0 7 7 7 0 0 7 0 7 7 7 7 0 7 0 9 0 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7\n",
      " 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7\n",
      " 7 7 7 7 7 7 7 7 7 7 0 7 0 7 0 9 7 0 7 7 7 7 7 7 4 3 0 4 7 3 9 0 0 7 0 4 0\n",
      " 7 0 7 7 7 7 3 3 0 0 4 7 7 0 7 7 0 0 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 0\n",
      " 7 7 7 0 0 7 7 7 7 0 7 7 7 7 7 7 7 0 9 7 7 7 0 0 4 7 7 0 9 0 9 0 0 0 7 7 7\n",
      " 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7\n",
      " 9 4 7 7 7 0 7 7 0 9 0 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7\n",
      " 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7\n",
      " 7 7 7 0 7 0 0 7 7 7 7 7 7 7 7 7 7 7 7 7 7 0 7 7 9 7 7 7 7 7 9 7 7 7 7 7 9\n",
      " 7 7 0 7 7 7 7 7 7 4 7 7 0 9 0 7 7 7 7 7 9 7 0 7 7 7 7 7 0 0 7 7 0 0 7 0 0\n",
      " 9 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 0 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7\n",
      " 7 7 7 7 7 7 7 7 7 7 7 7 0 7 0 4 0 0 7 7 0 7 7 7 7 7 7 7 0 7 7 7 7 7 7 7 7\n",
      " 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 0 7 7 7 7 0 0 7 7 7 7\n",
      " 7 7 7 7 7 7 7 0 7 7 7 7 7 0 7 7 7 7 7 9 9 7 0 7 7 7 7 4 7 4 0 0 7 0 0 7 7\n",
      " 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 0 7 0 0 0 7 7 9 0 7 7 0 7 7 7 7 7\n",
      " 7 7 7 7 7 7 7 7 7 7 0 7 7 7 7 7 0 7 9 7 7 7 7 0 7 0 7 7 7 7 7 7 7 7 7 7 7\n",
      " 7 7 0 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 0 7 7 7 0 0 0 0 7 7 7 9 0 7 7 0 0\n",
      " 0 9 9 7 0 0 7 7 7 7 7 7 0 0 0 7 7 7 7 0 7 7 7 7 7 7 7 0 7 7 7 7 7 0 7 0 7\n",
      " 7 7 7 7 7 0 7 7 0 0 7 7 7 7 7 7 7 7 7 7 7 7 7 4 7 7 7 0 7 9 7 7 7 7 7 7 7\n",
      " 7]\n",
      "threshold density 0.15152106310670216\n",
      "cluster signal 78705.0\n",
      "cluster length 1065400\n",
      "!!cluster density 0.0738736624741881\n",
      "cluster signal 4611.0\n",
      "cluster length 1543600\n",
      "!!cluster density 0.002987172842705364\n",
      "cluster signal 52040.0\n",
      "cluster length 1538300\n",
      "!!cluster density 0.03382955210297081\n",
      "cluster signal 150905.0\n",
      "cluster length 838400\n",
      "!!cluster density 0.17999165076335877\n",
      "cluster signal 109860.0\n",
      "cluster length 929000\n",
      "!!cluster density 0.1182561894510226\n",
      "cluster signal 45667.0\n",
      "cluster length 77600\n",
      "!!cluster density 0.5884922680412371\n",
      "cluster signal 130431.0\n",
      "cluster length 480700\n",
      "!!cluster density 0.271335552319534\n",
      "cluster signal 16355.0\n",
      "cluster length 18200\n",
      "!!cluster density 0.8986263736263737\n",
      "cluster signal 492.0\n",
      "cluster length 100\n",
      "!!cluster density 4.92\n",
      "cluster signal 89513.0\n",
      "cluster length 226400\n",
      "!!cluster density 0.39537544169611305\n",
      "\n",
      "Labels mapping\n",
      " {9: 0, 7: 0, 0: 0, 3: 1, 4: 0, 2: 1, 1: 1, 8: 1, 6: 1, 5: 1}\n",
      "DEBUG: 50763 16414 \n",
      "\n",
      " +--------------------+-------------+---------------+--------------+-------+\n",
      "| bin_offset_70      | TT_S0       | starting_nt   | Chromosome   | +7    |\n",
      "| (category)         | (float64)   | (category)    | (category)   | ...   |\n",
      "|--------------------+-------------+---------------+--------------+-------|\n",
      "| (3000016, 3000086] | 7.0         | 3000016       | 18           | ...   |\n",
      "| (3000086, 3000186] | 1.0         | 3000086       | 18           | ...   |\n",
      "| (3000186, 3000286] | 0.0         | 3000186       | 18           | ...   |\n",
      "| (3000286, 3000386] | 2.0         | 3000286       | 18           | ...   |\n",
      "| ...                | ...         | ...           | ...          | ...   |\n",
      "| (9717286, 9717386] | 0.0         | 9717286       | 18           | ...   |\n",
      "| (9717386, 9717486] | 1.0         | 9717386       | 18           | ...   |\n",
      "| (9717486, 9717586] | 1.0         | 9717486       | 18           | ...   |\n",
      "| (9717586, 9717593] | 0.0         | 9717586       | 18           | ...   |\n",
      "+--------------------+-------------+---------------+--------------+-------+\n",
      "Unstranded PyRanges object has 50,763 rows and 11 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "7 hidden columns: Start, End, genes, transcripts, exons, CDSs, labels\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0\n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "BINS  IntervalIndex([(3000016, 3000091], (3000091, 3000191], (3000191, 3000291],\n",
      "               (3000291, 3000391], (3000391, 3000491], (3000491, 3000591],\n",
      "               (3000591, 3000691], (3000691, 3000791], (3000791, 3000891],\n",
      "               (3000891, 3000991],\n",
      "               ...\n",
      "               (9716691, 9716791], (9716791, 9716891], (9716891, 9716991],\n",
      "               (9716991, 9717091], (9717091, 9717191], (9717191, 9717291],\n",
      "               (9717291, 9717391], (9717391, 9717491], (9717491, 9717591],\n",
      "               (9717591, 9717593]],\n",
      "              dtype='interval[int64, right]', length=67177)\n",
      "HEAD with BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0   \n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "\n",
      "        bin_offset_75  \n",
      "0  (3000016, 3000091]  \n",
      "1  (3000016, 3000091]  \n",
      "2  (3000016, 3000091]  \n",
      "3  (3000016, 3000091]  \n",
      "4  (3000016, 3000091]  \n",
      "TAIL with BINS\n",
      "       seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  9717555  9717555  18_9717555    1.0    0.0    0.0   \n",
      "699996       19  9717571  9717571  19_9717571    0.0    0.0    0.0   \n",
      "699997       19  9717581  9717581  19_9717581    0.0    0.0    0.0   \n",
      "699998       18  9717582  9717582  18_9717582    0.0    0.0    0.0   \n",
      "699999       18  9717593  9717593  18_9717593    0.0    0.0    0.0   \n",
      "\n",
      "             bin_offset_75  \n",
      "699995  (9717491, 9717591]  \n",
      "699996  (9717491, 9717591]  \n",
      "699997  (9717491, 9717591]  \n",
      "699998  (9717491, 9717591]  \n",
      "699999  (9717591, 9717593]  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:136: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "         bin_offset_75  TT_S0 starting_nt\n",
      "0   (3000016, 3000091]    7.0     3000016\n",
      "1   (3000091, 3000191]    1.0     3000091\n",
      "2   (3000191, 3000291]    0.0     3000191\n",
      "3   (3000291, 3000391]    2.0     3000291\n",
      "4   (3000391, 3000491]    0.0     3000391\n",
      "5   (3000491, 3000591]    0.0     3000491\n",
      "6   (3000591, 3000691]    0.0     3000591\n",
      "7   (3000691, 3000791]    0.0     3000691\n",
      "8   (3000791, 3000891]    0.0     3000791\n",
      "9   (3000891, 3000991]    0.0     3000891\n",
      "10  (3000991, 3001091]    2.0     3000991\n",
      "11  (3001091, 3001191]    0.0     3001091\n",
      "12  (3001191, 3001291]    0.0     3001191\n",
      "13  (3001291, 3001391]    0.0     3001291\n",
      "14  (3001391, 3001491]    0.0     3001391\n",
      "15  (3001491, 3001591]    4.0     3001491\n",
      "16  (3001591, 3001691]    3.0     3001591\n",
      "17  (3001691, 3001791]    0.0     3001691\n",
      "18  (3001791, 3001891]    0.0     3001791\n",
      "19  (3001891, 3001991]    0.0     3001891\n",
      "20  (3001991, 3002091]    1.0     3001991\n",
      "21  (3002091, 3002191]    0.0     3002091\n",
      "22  (3002191, 3002291]    4.0     3002191\n",
      "23  (3002291, 3002391]    9.0     3002291\n",
      "24  (3002391, 3002491]   19.0     3002391\n",
      "25  (3002491, 3002591]    4.0     3002491\n",
      "26  (3002591, 3002691]    0.0     3002591\n",
      "27  (3002691, 3002791]    0.0     3002691\n",
      "28  (3002791, 3002891]    0.0     3002791\n",
      "29  (3002891, 3002991]    1.0     3002891\n",
      "30  (3002991, 3003091]    2.0     3002991\n",
      "31  (3003091, 3003191]    0.0     3003091\n",
      "32  (3003191, 3003291]    0.0     3003191\n",
      "33  (3003291, 3003391]    1.0     3003291\n",
      "34  (3003391, 3003491]    0.0     3003391\n",
      "35  (3003491, 3003591]    1.0     3003491\n",
      "36  (3003591, 3003691]    1.0     3003591\n",
      "37  (3003691, 3003791]    3.0     3003691\n",
      "38  (3003791, 3003891]    0.0     3003791\n",
      "39  (3003891, 3003991]    8.0     3003891\n",
      "40  (3003991, 3004091]    0.0     3003991\n",
      "41  (3004091, 3004191]    0.0     3004091\n",
      "42  (3004191, 3004291]    3.0     3004191\n",
      "43  (3004291, 3004391]    2.0     3004291\n",
      "44  (3004391, 3004491]    0.0     3004391\n",
      "45  (3004491, 3004591]    0.0     3004491\n",
      "46  (3004591, 3004691]    1.0     3004591\n",
      "47  (3004691, 3004791]    0.0     3004691\n",
      "48  (3004791, 3004891]   10.0     3004791\n",
      "49  (3004891, 3004991]    3.0     3004891\n",
      "(67177, 3)\n",
      "bin_offset_75    category\n",
      "TT_S0             float64\n",
      "starting_nt      category\n",
      "dtype: object\n",
      "sumos ilgis: 67177\n",
      "ranges ilgis: 67177\n",
      "Model input - NO SCALE\n",
      "       TT_S0  genes  transcripts  exons  CDSs\n",
      "0        7.0      0            0      0     0\n",
      "1        1.0      0            0      0     0\n",
      "2        0.0      0            0      0     0\n",
      "3        2.0      0            0      0     0\n",
      "4        0.0      0            0      0     0\n",
      "...      ...    ...          ...    ...   ...\n",
      "67172    0.0      0            0      0     0\n",
      "67173    0.0      0            0      0     0\n",
      "67174    1.0      0            0      0     0\n",
      "67175    1.0      0            0      0     0\n",
      "67176    0.0      0            0      0     0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "       TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0        7.0             0                   0             0            0\n",
      "1        1.0             0                   0             0            0\n",
      "2        0.0             0                   0             0            0\n",
      "3        2.0             0                   0             0            0\n",
      "4        0.0             0                   0             0            0\n",
      "...      ...           ...                 ...           ...          ...\n",
      "67172    0.0             0                   0             0            0\n",
      "67173    0.0             0                   0             0            0\n",
      "67174    1.0             0                   0             0            0\n",
      "67175    1.0             0                   0             0            0\n",
      "67176    0.0             0                   0             0            0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "0.06334013129493725\n",
      "[9 0 0 0 0 0 0 0 0 0 0 0 0 0 0 9 9 0 0 0 0 0 9 2 5 9 0 0 0 0 0 0 0 0 0 0 0\n",
      " 9 0 2 0 0 9 0 0 0 0 0 2 9 0 6 1 9 5 2 3 7 7 8 0 2 5 9 0 0 9 0 0 0 2 0 9 0\n",
      " 0 0 0 0 0 0 2 0 0 9 9 0 9 2 0 0 9 5 9 0 0 9 0 9 9 9 5 5 1 5 9 9 2 1 9 2 0\n",
      " 2 9 0 0 0 0 0 0 0 0 9 9 0 0 5 2 5 2 2 2 9 2 9 0 9 9 2 0 0 0 0 0 0 0 0 0 5\n",
      " 0 9 9 0 0 9 0 2 9 0 0 5 0 2 9 0 2 2 0 0 0 0 0 0 0 0 0 0 0 0 0 0 2 0 0 9 0\n",
      " 9 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 2 5 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 9 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 9 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 9 0 0 0 0 9 0 0 0 0 0 0 9 0 9 9 9 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 9 0 0 0 9 2 0 0 0 0 0 0 0 0 2 2 9 2 0 1 9 9 0 0 9 2 0\n",
      " 0 9 0 0 0 0 5 2 9 9 2 0 0 9 0 0 0 9 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 9\n",
      " 0 0 0 9 0 0 0 0 0 0 0 0 0 0 0 0 0 9 2 0 0 0 9 0 2 0 9 0 2 9 9 9 0 9 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 9 2 0 0 0 9 0 0 2 0 9 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 9 0 0 9 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 9 0 0 0 0 0 2 0 0 0 0 0 9\n",
      " 0 0 9 0 0 0 0 0 0 2 0 0 9 9 9 0 0 0 0 0 9 0 9 0 0 0 0 0 9 9 0 0 9 9 0 9 0\n",
      " 9 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 9 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 9 0 9 2 0 9 0 0 0 0 0 0 0 0 0 0 9 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 9 0 0 0 0 0 9 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 9 0 0 0 0 0 9 9 0 9 0 0 0 0 2 0 2 9 9 0 0 9 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 9 0 9 0 0 9 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 9 0 0 0 0 9 0 9 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 9 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 9 0 0 0 9 9 9 9 0 0 0 2 9 0 0 9 9\n",
      " 9 9 9 0 9 0 0 0 0 0 0 0 0 9 0 0 0 0 0 9 0 0 0 0 0 0 0 0 0 0 0 0 0 9 0 0 0\n",
      " 0 0 0 0 0 9 0 0 9 0 0 0 0 0 0 0 0 0 0 0 0 0 0 2 0 0 0 9 9 9 0 0 0 0 0 0 0\n",
      " 0]\n",
      "threshold density 0.1515209503287968\n",
      "cluster signal 80849.0\n",
      "cluster length 1678300\n",
      "!!cluster density 0.048173151403205626\n",
      "cluster signal 13421.0\n",
      "cluster length 1984600\n",
      "!!cluster density 0.006762571802882193\n",
      "cluster signal 150828.0\n",
      "cluster length 1415500\n",
      "!!cluster density 0.10655457435535147\n",
      "cluster signal 164682.0\n",
      "cluster length 895300\n",
      "!!cluster density 0.1839405785770133\n",
      "cluster signal 43322.0\n",
      "cluster length 83400\n",
      "!!cluster density 0.5194484412470024\n",
      "cluster signal 118092.0\n",
      "cluster length 425500\n",
      "!!cluster density 0.2775370152761457\n",
      "cluster signal 23789.0\n",
      "cluster length 33800\n",
      "!!cluster density 0.7038165680473373\n",
      "cluster signal 9479.0\n",
      "cluster length 9500\n",
      "!!cluster density 0.9977894736842106\n",
      "cluster signal 492.0\n",
      "cluster length 100\n",
      "!!cluster density 4.92\n",
      "cluster signal 73625.0\n",
      "cluster length 191700\n",
      "!!cluster density 0.38406364110589464\n",
      "\n",
      "Labels mapping\n",
      " {9: 0, 0: 0, 2: 0, 5: 1, 6: 1, 1: 1, 3: 1, 7: 1, 8: 1, 4: 1}\n",
      "DEBUG: 50784 16393 \n",
      "\n",
      " +--------------------+-------------+---------------+--------------+-------+\n",
      "| bin_offset_75      | TT_S0       | starting_nt   | Chromosome   | +7    |\n",
      "| (category)         | (float64)   | (category)    | (category)   | ...   |\n",
      "|--------------------+-------------+---------------+--------------+-------|\n",
      "| (3000016, 3000091] | 7.0         | 3000016       | 18           | ...   |\n",
      "| (3000091, 3000191] | 1.0         | 3000091       | 18           | ...   |\n",
      "| (3000191, 3000291] | 0.0         | 3000191       | 18           | ...   |\n",
      "| (3000291, 3000391] | 2.0         | 3000291       | 18           | ...   |\n",
      "| ...                | ...         | ...           | ...          | ...   |\n",
      "| (9717291, 9717391] | 0.0         | 9717291       | 18           | ...   |\n",
      "| (9717391, 9717491] | 1.0         | 9717391       | 18           | ...   |\n",
      "| (9717491, 9717591] | 1.0         | 9717491       | 18           | ...   |\n",
      "| (9717591, 9717593] | 0.0         | 9717591       | 18           | ...   |\n",
      "+--------------------+-------------+---------------+--------------+-------+\n",
      "Unstranded PyRanges object has 50,784 rows and 11 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "7 hidden columns: Start, End, genes, transcripts, exons, CDSs, labels\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0\n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "BINS  IntervalIndex([(3000016, 3000096], (3000096, 3000196], (3000196, 3000296],\n",
      "               (3000296, 3000396], (3000396, 3000496], (3000496, 3000596],\n",
      "               (3000596, 3000696], (3000696, 3000796], (3000796, 3000896],\n",
      "               (3000896, 3000996],\n",
      "               ...\n",
      "               (9716596, 9716696], (9716696, 9716796], (9716796, 9716896],\n",
      "               (9716896, 9716996], (9716996, 9717096], (9717096, 9717196],\n",
      "               (9717196, 9717296], (9717296, 9717396], (9717396, 9717496],\n",
      "               (9717496, 9717593]],\n",
      "              dtype='interval[int64, right]', length=67176)\n",
      "HEAD with BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0   \n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "\n",
      "        bin_offset_80  \n",
      "0  (3000016, 3000096]  \n",
      "1  (3000016, 3000096]  \n",
      "2  (3000016, 3000096]  \n",
      "3  (3000016, 3000096]  \n",
      "4  (3000016, 3000096]  \n",
      "TAIL with BINS\n",
      "       seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  9717555  9717555  18_9717555    1.0    0.0    0.0   \n",
      "699996       19  9717571  9717571  19_9717571    0.0    0.0    0.0   \n",
      "699997       19  9717581  9717581  19_9717581    0.0    0.0    0.0   \n",
      "699998       18  9717582  9717582  18_9717582    0.0    0.0    0.0   \n",
      "699999       18  9717593  9717593  18_9717593    0.0    0.0    0.0   \n",
      "\n",
      "             bin_offset_80  \n",
      "699995  (9717496, 9717593]  \n",
      "699996  (9717496, 9717593]  \n",
      "699997  (9717496, 9717593]  \n",
      "699998  (9717496, 9717593]  \n",
      "699999  (9717496, 9717593]  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:136: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "         bin_offset_80  TT_S0 starting_nt\n",
      "0   (3000016, 3000096]    7.0     3000016\n",
      "1   (3000096, 3000196]    1.0     3000096\n",
      "2   (3000196, 3000296]    0.0     3000196\n",
      "3   (3000296, 3000396]    2.0     3000296\n",
      "4   (3000396, 3000496]    0.0     3000396\n",
      "5   (3000496, 3000596]    0.0     3000496\n",
      "6   (3000596, 3000696]    0.0     3000596\n",
      "7   (3000696, 3000796]    0.0     3000696\n",
      "8   (3000796, 3000896]    0.0     3000796\n",
      "9   (3000896, 3000996]    0.0     3000896\n",
      "10  (3000996, 3001096]    2.0     3000996\n",
      "11  (3001096, 3001196]    0.0     3001096\n",
      "12  (3001196, 3001296]    0.0     3001196\n",
      "13  (3001296, 3001396]    0.0     3001296\n",
      "14  (3001396, 3001496]    0.0     3001396\n",
      "15  (3001496, 3001596]    4.0     3001496\n",
      "16  (3001596, 3001696]    3.0     3001596\n",
      "17  (3001696, 3001796]    0.0     3001696\n",
      "18  (3001796, 3001896]    0.0     3001796\n",
      "19  (3001896, 3001996]    0.0     3001896\n",
      "20  (3001996, 3002096]    1.0     3001996\n",
      "21  (3002096, 3002196]    0.0     3002096\n",
      "22  (3002196, 3002296]   11.0     3002196\n",
      "23  (3002296, 3002396]    2.0     3002296\n",
      "24  (3002396, 3002496]   19.0     3002396\n",
      "25  (3002496, 3002596]    4.0     3002496\n",
      "26  (3002596, 3002696]    0.0     3002596\n",
      "27  (3002696, 3002796]    0.0     3002696\n",
      "28  (3002796, 3002896]    0.0     3002796\n",
      "29  (3002896, 3002996]    1.0     3002896\n",
      "30  (3002996, 3003096]    2.0     3002996\n",
      "31  (3003096, 3003196]    0.0     3003096\n",
      "32  (3003196, 3003296]    0.0     3003196\n",
      "33  (3003296, 3003396]    1.0     3003296\n",
      "34  (3003396, 3003496]    0.0     3003396\n",
      "35  (3003496, 3003596]    1.0     3003496\n",
      "36  (3003596, 3003696]    1.0     3003596\n",
      "37  (3003696, 3003796]    3.0     3003696\n",
      "38  (3003796, 3003896]    0.0     3003796\n",
      "39  (3003896, 3003996]    8.0     3003896\n",
      "40  (3003996, 3004096]    0.0     3003996\n",
      "41  (3004096, 3004196]    0.0     3004096\n",
      "42  (3004196, 3004296]    3.0     3004196\n",
      "43  (3004296, 3004396]    2.0     3004296\n",
      "44  (3004396, 3004496]    0.0     3004396\n",
      "45  (3004496, 3004596]    0.0     3004496\n",
      "46  (3004596, 3004696]    1.0     3004596\n",
      "47  (3004696, 3004796]    0.0     3004696\n",
      "48  (3004796, 3004896]   10.0     3004796\n",
      "49  (3004896, 3004996]    3.0     3004896\n",
      "(67176, 3)\n",
      "bin_offset_80    category\n",
      "TT_S0             float64\n",
      "starting_nt      category\n",
      "dtype: object\n",
      "sumos ilgis: 67176\n",
      "ranges ilgis: 67176\n",
      "Model input - NO SCALE\n",
      "       TT_S0  genes  transcripts  exons  CDSs\n",
      "0        7.0      0            0      0     0\n",
      "1        1.0      0            0      0     0\n",
      "2        0.0      0            0      0     0\n",
      "3        2.0      0            0      0     0\n",
      "4        0.0      0            0      0     0\n",
      "...      ...    ...          ...    ...   ...\n",
      "67171    0.0      0            0      0     0\n",
      "67172    0.0      0            0      0     0\n",
      "67173    0.0      0            0      0     0\n",
      "67174    1.0      0            0      0     0\n",
      "67175    1.0      0            0      0     0\n",
      "\n",
      "[67176 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "       TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0        7.0             0                   0             0            0\n",
      "1        1.0             0                   0             0            0\n",
      "2        0.0             0                   0             0            0\n",
      "3        2.0             0                   0             0            0\n",
      "4        0.0             0                   0             0            0\n",
      "...      ...           ...                 ...           ...          ...\n",
      "67171    0.0             0                   0             0            0\n",
      "67172    0.0             0                   0             0            0\n",
      "67173    0.0             0                   0             0            0\n",
      "67174    1.0             0                   0             0            0\n",
      "67175    1.0             0                   0             0            0\n",
      "\n",
      "[67176 rows x 5 columns]\n",
      "0.08513457187090627\n",
      "[3 0 0 0 0 0 0 0 0 0 0 0 0 0 0 7 7 0 0 0 0 0 3 0 8 7 0 0 0 0 0 0 0 0 0 0 0\n",
      " 7 0 3 0 0 7 0 0 0 0 0 3 7 0 2 4 3 8 8 9 5 5 6 0 3 1 7 0 0 3 0 0 7 7 0 7 0\n",
      " 0 0 0 0 0 0 3 0 0 7 7 0 0 3 0 0 3 8 0 0 0 7 0 7 3 7 8 1 4 8 7 7 8 1 7 3 0\n",
      " 8 7 0 0 0 0 0 0 0 0 7 7 0 8 0 3 8 3 3 3 7 3 7 0 7 7 3 0 0 0 0 0 0 0 0 0 8\n",
      " 0 7 7 0 0 7 0 8 7 0 0 8 0 3 7 0 3 8 0 0 0 0 0 0 0 0 0 0 0 0 0 0 3 0 0 7 0\n",
      " 7 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 3 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 7 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 7 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 7 0 0 0 0 7 0 0 0 0 0 0 7 0 7 7 7 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 7 0 0 0 7 3 0 0 0 0 0 0 0 0 3 8 7 3 0 1 7 7 0 0 7 3 0\n",
      " 0 7 0 0 0 0 8 8 7 3 3 0 0 7 0 0 0 7 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 7\n",
      " 0 0 0 7 0 0 0 0 0 0 0 0 0 0 0 0 0 7 3 0 0 0 7 0 3 0 7 0 3 7 3 7 0 7 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 3 3 0 0 0 7 0 0 3 0 7 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 7 0 0 7 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 7 0 0 0 0 0 3 0 0 0 0 0 7\n",
      " 0 0 7 0 0 0 0 0 0 3 0 0 7 7 7 0 0 0 0 0 7 0 7 0 0 0 0 0 7 7 0 0 7 7 0 3 0\n",
      " 3 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 7 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 7 0 7 3 0 7 0 0 0 0 0 0 0 0 0 0 7 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 7 0 0 0 0 0 7 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 7 0 0 0 0 0 3 3 0 7 0 0 0 0 3 7 3 0 7 0 0 7 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 7 0 7 0 0 7 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 3 0 0 0 0 7 0 7 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 7 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 7 0 0 0 7 7 7 7 0 0 0 3 7 0 0 7 7\n",
      " 7 7 3 0 7 0 0 0 0 0 0 0 0 7 0 0 0 0 0 7 0 0 0 0 0 0 0 0 0 0 0 0 0 7 0 0 0\n",
      " 0 0 0 0 0 7 0 7 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 8 0 0 7 0 7 7 0 0 0 0 0 0 0\n",
      " 0]\n",
      "threshold density 0.15152309313770734\n",
      "cluster signal 127115.0\n",
      "cluster length 1372300\n",
      "!!cluster density 0.09262916271952197\n",
      "cluster signal 13448.0\n",
      "cluster length 1985700\n",
      "!!cluster density 0.006772422823185778\n",
      "cluster signal 61706.0\n",
      "cluster length 1403400\n",
      "!!cluster density 0.043968932592275904\n",
      "cluster signal 141111.0\n",
      "cluster length 900700\n",
      "!!cluster density 0.15666814699678028\n",
      "cluster signal 57190.0\n",
      "cluster length 122000\n",
      "!!cluster density 0.4687704918032787\n",
      "cluster signal 100556.0\n",
      "cluster length 301200\n",
      "!!cluster density 0.3338512616201859\n",
      "cluster signal 31919.0\n",
      "cluster length 48600\n",
      "!!cluster density 0.6567695473251028\n",
      "cluster signal 11311.0\n",
      "cluster length 11700\n",
      "!!cluster density 0.9667521367521368\n",
      "cluster signal 492.0\n",
      "cluster length 100\n",
      "!!cluster density 4.92\n",
      "cluster signal 133731.0\n",
      "cluster length 571900\n",
      "!!cluster density 0.2338363350236055\n",
      "\n",
      "Labels mapping\n",
      " {3: 0, 0: 0, 7: 0, 8: 1, 2: 1, 4: 1, 9: 1, 5: 1, 6: 1, 1: 1}\n",
      "DEBUG: 47614 19562 \n",
      "\n",
      " +--------------------+-------------+---------------+--------------+-------+\n",
      "| bin_offset_80      | TT_S0       | starting_nt   | Chromosome   | +7    |\n",
      "| (category)         | (float64)   | (category)    | (category)   | ...   |\n",
      "|--------------------+-------------+---------------+--------------+-------|\n",
      "| (3000016, 3000096] | 7.0         | 3000016       | 18           | ...   |\n",
      "| (3000096, 3000196] | 1.0         | 3000096       | 18           | ...   |\n",
      "| (3000196, 3000296] | 0.0         | 3000196       | 18           | ...   |\n",
      "| (3000296, 3000396] | 2.0         | 3000296       | 18           | ...   |\n",
      "| ...                | ...         | ...           | ...          | ...   |\n",
      "| (9717196, 9717296] | 0.0         | 9717196       | 18           | ...   |\n",
      "| (9717296, 9717396] | 0.0         | 9717296       | 18           | ...   |\n",
      "| (9717396, 9717496] | 1.0         | 9717396       | 18           | ...   |\n",
      "| (9717496, 9717593] | 1.0         | 9717496       | 18           | ...   |\n",
      "+--------------------+-------------+---------------+--------------+-------+\n",
      "Unstranded PyRanges object has 47,614 rows and 11 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "7 hidden columns: Start, End, genes, transcripts, exons, CDSs, labels\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0\n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "BINS  IntervalIndex([(3000016, 3000101], (3000101, 3000201], (3000201, 3000301],\n",
      "               (3000301, 3000401], (3000401, 3000501], (3000501, 3000601],\n",
      "               (3000601, 3000701], (3000701, 3000801], (3000801, 3000901],\n",
      "               (3000901, 3001001],\n",
      "               ...\n",
      "               (9716601, 9716701], (9716701, 9716801], (9716801, 9716901],\n",
      "               (9716901, 9717001], (9717001, 9717101], (9717101, 9717201],\n",
      "               (9717201, 9717301], (9717301, 9717401], (9717401, 9717501],\n",
      "               (9717501, 9717593]],\n",
      "              dtype='interval[int64, right]', length=67176)\n",
      "HEAD with BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0   \n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "\n",
      "        bin_offset_85  \n",
      "0  (3000016, 3000101]  \n",
      "1  (3000016, 3000101]  \n",
      "2  (3000016, 3000101]  \n",
      "3  (3000016, 3000101]  \n",
      "4  (3000016, 3000101]  \n",
      "TAIL with BINS\n",
      "       seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  9717555  9717555  18_9717555    1.0    0.0    0.0   \n",
      "699996       19  9717571  9717571  19_9717571    0.0    0.0    0.0   \n",
      "699997       19  9717581  9717581  19_9717581    0.0    0.0    0.0   \n",
      "699998       18  9717582  9717582  18_9717582    0.0    0.0    0.0   \n",
      "699999       18  9717593  9717593  18_9717593    0.0    0.0    0.0   \n",
      "\n",
      "             bin_offset_85  \n",
      "699995  (9717501, 9717593]  \n",
      "699996  (9717501, 9717593]  \n",
      "699997  (9717501, 9717593]  \n",
      "699998  (9717501, 9717593]  \n",
      "699999  (9717501, 9717593]  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:136: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "         bin_offset_85  TT_S0 starting_nt\n",
      "0   (3000016, 3000101]    7.0     3000016\n",
      "1   (3000101, 3000201]    1.0     3000101\n",
      "2   (3000201, 3000301]    0.0     3000201\n",
      "3   (3000301, 3000401]    2.0     3000301\n",
      "4   (3000401, 3000501]    0.0     3000401\n",
      "5   (3000501, 3000601]    0.0     3000501\n",
      "6   (3000601, 3000701]    0.0     3000601\n",
      "7   (3000701, 3000801]    0.0     3000701\n",
      "8   (3000801, 3000901]    0.0     3000801\n",
      "9   (3000901, 3001001]    0.0     3000901\n",
      "10  (3001001, 3001101]    2.0     3001001\n",
      "11  (3001101, 3001201]    0.0     3001101\n",
      "12  (3001201, 3001301]    0.0     3001201\n",
      "13  (3001301, 3001401]    0.0     3001301\n",
      "14  (3001401, 3001501]    1.0     3001401\n",
      "15  (3001501, 3001601]    3.0     3001501\n",
      "16  (3001601, 3001701]    3.0     3001601\n",
      "17  (3001701, 3001801]    0.0     3001701\n",
      "18  (3001801, 3001901]    0.0     3001801\n",
      "19  (3001901, 3002001]    0.0     3001901\n",
      "20  (3002001, 3002101]    1.0     3002001\n",
      "21  (3002101, 3002201]    0.0     3002101\n",
      "22  (3002201, 3002301]   11.0     3002201\n",
      "23  (3002301, 3002401]    2.0     3002301\n",
      "24  (3002401, 3002501]   19.0     3002401\n",
      "25  (3002501, 3002601]    4.0     3002501\n",
      "26  (3002601, 3002701]    0.0     3002601\n",
      "27  (3002701, 3002801]    0.0     3002701\n",
      "28  (3002801, 3002901]    0.0     3002801\n",
      "29  (3002901, 3003001]    1.0     3002901\n",
      "30  (3003001, 3003101]    2.0     3003001\n",
      "31  (3003101, 3003201]    0.0     3003101\n",
      "32  (3003201, 3003301]    0.0     3003201\n",
      "33  (3003301, 3003401]    1.0     3003301\n",
      "34  (3003401, 3003501]    0.0     3003401\n",
      "35  (3003501, 3003601]    1.0     3003501\n",
      "36  (3003601, 3003701]    1.0     3003601\n",
      "37  (3003701, 3003801]    3.0     3003701\n",
      "38  (3003801, 3003901]    0.0     3003801\n",
      "39  (3003901, 3004001]    8.0     3003901\n",
      "40  (3004001, 3004101]    0.0     3004001\n",
      "41  (3004101, 3004201]    0.0     3004101\n",
      "42  (3004201, 3004301]    5.0     3004201\n",
      "43  (3004301, 3004401]    0.0     3004301\n",
      "44  (3004401, 3004501]    0.0     3004401\n",
      "45  (3004501, 3004601]    0.0     3004501\n",
      "46  (3004601, 3004701]    1.0     3004601\n",
      "47  (3004701, 3004801]    0.0     3004701\n",
      "48  (3004801, 3004901]   10.0     3004801\n",
      "49  (3004901, 3005001]    3.0     3004901\n",
      "(67176, 3)\n",
      "bin_offset_85    category\n",
      "TT_S0             float64\n",
      "starting_nt      category\n",
      "dtype: object\n",
      "sumos ilgis: 67176\n",
      "ranges ilgis: 67176\n",
      "Model input - NO SCALE\n",
      "       TT_S0  genes  transcripts  exons  CDSs\n",
      "0        7.0      0            0      0     0\n",
      "1        1.0      0            0      0     0\n",
      "2        0.0      0            0      0     0\n",
      "3        2.0      0            0      0     0\n",
      "4        0.0      0            0      0     0\n",
      "...      ...    ...          ...    ...   ...\n",
      "67171    0.0      0            0      0     0\n",
      "67172    0.0      0            0      0     0\n",
      "67173    0.0      0            0      0     0\n",
      "67174    1.0      0            0      0     0\n",
      "67175    1.0      0            0      0     0\n",
      "\n",
      "[67176 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "       TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0        7.0             0                   0             0            0\n",
      "1        1.0             0                   0             0            0\n",
      "2        0.0             0                   0             0            0\n",
      "3        2.0             0                   0             0            0\n",
      "4        0.0             0                   0             0            0\n",
      "...      ...           ...                 ...           ...          ...\n",
      "67171    0.0             0                   0             0            0\n",
      "67172    0.0             0                   0             0            0\n",
      "67173    0.0             0                   0             0            0\n",
      "67174    1.0             0                   0             0            0\n",
      "67175    1.0             0                   0             0            0\n",
      "\n",
      "[67176 rows x 5 columns]\n",
      "0.2959092533047517\n",
      "[4 1 1 1 1 1 1 1 1 1 1 1 1 1 1 9 9 1 1 1 1 1 4 1 5 9 1 1 1 1 1 1 1 1 1 1 1\n",
      " 9 1 4 1 1 9 1 1 1 1 1 4 9 1 8 3 4 5 2 7 7 7 6 1 4 5 9 1 1 4 1 1 9 9 1 9 1\n",
      " 1 1 1 1 1 1 4 1 1 9 9 1 9 9 1 1 4 2 1 1 1 9 1 9 4 9 5 5 5 2 9 4 4 5 9 4 1\n",
      " 2 9 1 1 1 1 1 1 1 1 9 9 1 2 1 2 2 4 2 4 9 4 9 1 9 9 4 1 1 1 1 1 1 1 1 1 2\n",
      " 1 4 9 1 1 9 1 2 9 1 1 2 1 4 9 1 4 4 9 1 1 1 1 1 1 1 1 1 1 1 1 1 2 1 9 1 1\n",
      " 9 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 4 5 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 9 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 9 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 9 1 1 1 1 9 1 1 1 1 1 1 9 1 9 4 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 9 1 1 1 4 9 1 1 1 1 1 1 1 1 4 2 9 2 1 5 9 9 1 1 9 4 1\n",
      " 1 9 1 1 1 1 5 2 9 4 4 1 1 9 1 1 1 9 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 9\n",
      " 1 1 1 9 1 1 1 1 1 1 1 1 1 1 1 1 1 9 4 1 1 1 9 1 4 1 9 1 4 9 4 1 1 9 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 4 2 1 1 1 9 1 1 4 1 9 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 9 1 1 9 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 4 1 1 1 1 1 4 1 1 1 1 1 9\n",
      " 1 1 9 1 1 1 1 1 1 2 1 1 9 4 1 1 1 1 1 1 9 1 1 1 1 1 1 1 9 9 1 1 9 9 1 4 1\n",
      " 4 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 9 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 9 1 4 9 1 9 1 1 1 1 1 1 1 1 1 1 9 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 9 1 1 1 1 1 9 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 9 1 1 1 1 1 4 9 1 9 1 1 1 1 4 9 4 1 9 1 1 9 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 9 1 9 1 1 9 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 9 1 1 1 1 9 1 9 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 9 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 9 1 1 1 9 9 4 1 1 1 1 4 9 1 1 9 9\n",
      " 1 9 4 1 9 1 1 1 1 1 1 1 9 9 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 9 1 1 1\n",
      " 1 1 1 1 1 9 1 9 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 2 1 1 9 1 9 9 1 1 1 1 1 1 1\n",
      " 1]\n",
      "threshold density 0.15152298035678002\n",
      "cluster signal 104344.0\n",
      "cluster length 1180900\n",
      "!!cluster density 0.08835972563299178\n",
      "cluster signal 13430.0\n",
      "cluster length 1987800\n",
      "!!cluster density 0.00675621289868196\n",
      "cluster signal 61834.0\n",
      "cluster length 1404800\n",
      "!!cluster density 0.04401623006833713\n",
      "cluster signal 161334.0\n",
      "cluster length 741000\n",
      "!!cluster density 0.2177246963562753\n",
      "cluster signal 58689.0\n",
      "cluster length 128700\n",
      "!!cluster density 0.456013986013986\n",
      "cluster signal 105388.0\n",
      "cluster length 324300\n",
      "!!cluster density 0.32497070613629353\n",
      "cluster signal 125422.0\n",
      "cluster length 882200\n",
      "!!cluster density 0.14216957605985037\n",
      "cluster signal 13831.0\n",
      "cluster length 14800\n",
      "!!cluster density 0.9345270270270271\n",
      "cluster signal 492.0\n",
      "cluster length 100\n",
      "!!cluster density 4.92\n",
      "cluster signal 33815.0\n",
      "cluster length 53000\n",
      "!!cluster density 0.6380188679245283\n",
      "\n",
      "Labels mapping\n",
      " {4: 0, 1: 0, 9: 0, 5: 1, 8: 1, 3: 1, 2: 0, 7: 1, 6: 1, 0: 1}\n",
      "DEBUG: 54557 12619 \n",
      "\n",
      " +--------------------+-------------+---------------+--------------+-------+\n",
      "| bin_offset_85      | TT_S0       | starting_nt   | Chromosome   | +7    |\n",
      "| (category)         | (float64)   | (category)    | (category)   | ...   |\n",
      "|--------------------+-------------+---------------+--------------+-------|\n",
      "| (3000016, 3000101] | 7.0         | 3000016       | 18           | ...   |\n",
      "| (3000101, 3000201] | 1.0         | 3000101       | 18           | ...   |\n",
      "| (3000201, 3000301] | 0.0         | 3000201       | 18           | ...   |\n",
      "| (3000301, 3000401] | 2.0         | 3000301       | 18           | ...   |\n",
      "| ...                | ...         | ...           | ...          | ...   |\n",
      "| (9717201, 9717301] | 0.0         | 9717201       | 18           | ...   |\n",
      "| (9717301, 9717401] | 0.0         | 9717301       | 18           | ...   |\n",
      "| (9717401, 9717501] | 1.0         | 9717401       | 18           | ...   |\n",
      "| (9717501, 9717593] | 1.0         | 9717501       | 18           | ...   |\n",
      "+--------------------+-------------+---------------+--------------+-------+\n",
      "Unstranded PyRanges object has 54,557 rows and 11 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "7 hidden columns: Start, End, genes, transcripts, exons, CDSs, labels\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0\n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "BINS  IntervalIndex([(3000016, 3000106], (3000106, 3000206], (3000206, 3000306],\n",
      "               (3000306, 3000406], (3000406, 3000506], (3000506, 3000606],\n",
      "               (3000606, 3000706], (3000706, 3000806], (3000806, 3000906],\n",
      "               (3000906, 3001006],\n",
      "               ...\n",
      "               (9716606, 9716706], (9716706, 9716806], (9716806, 9716906],\n",
      "               (9716906, 9717006], (9717006, 9717106], (9717106, 9717206],\n",
      "               (9717206, 9717306], (9717306, 9717406], (9717406, 9717506],\n",
      "               (9717506, 9717593]],\n",
      "              dtype='interval[int64, right]', length=67176)\n",
      "HEAD with BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0   \n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "\n",
      "        bin_offset_90  \n",
      "0  (3000016, 3000106]  \n",
      "1  (3000016, 3000106]  \n",
      "2  (3000016, 3000106]  \n",
      "3  (3000016, 3000106]  \n",
      "4  (3000016, 3000106]  \n",
      "TAIL with BINS\n",
      "       seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  9717555  9717555  18_9717555    1.0    0.0    0.0   \n",
      "699996       19  9717571  9717571  19_9717571    0.0    0.0    0.0   \n",
      "699997       19  9717581  9717581  19_9717581    0.0    0.0    0.0   \n",
      "699998       18  9717582  9717582  18_9717582    0.0    0.0    0.0   \n",
      "699999       18  9717593  9717593  18_9717593    0.0    0.0    0.0   \n",
      "\n",
      "             bin_offset_90  \n",
      "699995  (9717506, 9717593]  \n",
      "699996  (9717506, 9717593]  \n",
      "699997  (9717506, 9717593]  \n",
      "699998  (9717506, 9717593]  \n",
      "699999  (9717506, 9717593]  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:136: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "         bin_offset_90  TT_S0 starting_nt\n",
      "0   (3000016, 3000106]    7.0     3000016\n",
      "1   (3000106, 3000206]    1.0     3000106\n",
      "2   (3000206, 3000306]    0.0     3000206\n",
      "3   (3000306, 3000406]    2.0     3000306\n",
      "4   (3000406, 3000506]    0.0     3000406\n",
      "5   (3000506, 3000606]    0.0     3000506\n",
      "6   (3000606, 3000706]    0.0     3000606\n",
      "7   (3000706, 3000806]    0.0     3000706\n",
      "8   (3000806, 3000906]    0.0     3000806\n",
      "9   (3000906, 3001006]    0.0     3000906\n",
      "10  (3001006, 3001106]    2.0     3001006\n",
      "11  (3001106, 3001206]    0.0     3001106\n",
      "12  (3001206, 3001306]    0.0     3001206\n",
      "13  (3001306, 3001406]    0.0     3001306\n",
      "14  (3001406, 3001506]    1.0     3001406\n",
      "15  (3001506, 3001606]    3.0     3001506\n",
      "16  (3001606, 3001706]    3.0     3001606\n",
      "17  (3001706, 3001806]    0.0     3001706\n",
      "18  (3001806, 3001906]    0.0     3001806\n",
      "19  (3001906, 3002006]    1.0     3001906\n",
      "20  (3002006, 3002106]    0.0     3002006\n",
      "21  (3002106, 3002206]    0.0     3002106\n",
      "22  (3002206, 3002306]   11.0     3002206\n",
      "23  (3002306, 3002406]    2.0     3002306\n",
      "24  (3002406, 3002506]   19.0     3002406\n",
      "25  (3002506, 3002606]    4.0     3002506\n",
      "26  (3002606, 3002706]    0.0     3002606\n",
      "27  (3002706, 3002806]    0.0     3002706\n",
      "28  (3002806, 3002906]    0.0     3002806\n",
      "29  (3002906, 3003006]    1.0     3002906\n",
      "30  (3003006, 3003106]    2.0     3003006\n",
      "31  (3003106, 3003206]    0.0     3003106\n",
      "32  (3003206, 3003306]    0.0     3003206\n",
      "33  (3003306, 3003406]    1.0     3003306\n",
      "34  (3003406, 3003506]    0.0     3003406\n",
      "35  (3003506, 3003606]    1.0     3003506\n",
      "36  (3003606, 3003706]    1.0     3003606\n",
      "37  (3003706, 3003806]    3.0     3003706\n",
      "38  (3003806, 3003906]    0.0     3003806\n",
      "39  (3003906, 3004006]    8.0     3003906\n",
      "40  (3004006, 3004106]    0.0     3004006\n",
      "41  (3004106, 3004206]    0.0     3004106\n",
      "42  (3004206, 3004306]    5.0     3004206\n",
      "43  (3004306, 3004406]    0.0     3004306\n",
      "44  (3004406, 3004506]    0.0     3004406\n",
      "45  (3004506, 3004606]    1.0     3004506\n",
      "46  (3004606, 3004706]    0.0     3004606\n",
      "47  (3004706, 3004806]    0.0     3004706\n",
      "48  (3004806, 3004906]   10.0     3004806\n",
      "49  (3004906, 3005006]    3.0     3004906\n",
      "(67176, 3)\n",
      "bin_offset_90    category\n",
      "TT_S0             float64\n",
      "starting_nt      category\n",
      "dtype: object\n",
      "sumos ilgis: 67176\n",
      "ranges ilgis: 67176\n",
      "Model input - NO SCALE\n",
      "       TT_S0  genes  transcripts  exons  CDSs\n",
      "0        7.0      0            0      0     0\n",
      "1        1.0      0            0      0     0\n",
      "2        0.0      0            0      0     0\n",
      "3        2.0      0            0      0     0\n",
      "4        0.0      0            0      0     0\n",
      "...      ...    ...          ...    ...   ...\n",
      "67171    0.0      0            0      0     0\n",
      "67172    0.0      0            0      0     0\n",
      "67173    0.0      0            0      0     0\n",
      "67174    1.0      0            0      0     0\n",
      "67175    1.0      0            0      0     0\n",
      "\n",
      "[67176 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "       TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0        7.0             0                   0             0            0\n",
      "1        1.0             0                   0             0            0\n",
      "2        0.0             0                   0             0            0\n",
      "3        2.0             0                   0             0            0\n",
      "4        0.0             0                   0             0            0\n",
      "...      ...           ...                 ...           ...          ...\n",
      "67171    0.0             0                   0             0            0\n",
      "67172    0.0             0                   0             0            0\n",
      "67173    0.0             0                   0             0            0\n",
      "67174    1.0             0                   0             0            0\n",
      "67175    1.0             0                   0             0            0\n",
      "\n",
      "[67176 rows x 5 columns]\n",
      "0.1047695605573419\n",
      "[3 0 0 0 0 0 0 0 0 0 0 0 0 0 0 8 8 0 0 0 0 0 7 0 1 8 0 0 0 0 0 0 0 0 0 0 0\n",
      " 8 0 3 0 0 8 0 0 0 0 0 3 8 0 2 4 3 1 7 6 6 6 5 0 3 1 8 0 0 3 0 0 8 8 0 8 0\n",
      " 0 0 0 0 0 0 3 0 0 8 8 8 8 8 0 0 3 7 0 0 0 8 0 8 3 8 1 4 4 7 8 3 7 4 8 7 0\n",
      " 7 8 0 0 0 0 0 0 0 0 3 0 0 7 0 1 7 7 7 3 8 3 8 0 8 8 3 0 0 0 0 0 0 0 0 8 7\n",
      " 0 3 8 0 0 8 0 7 8 0 0 1 0 3 8 0 3 7 8 0 0 0 0 0 0 0 0 0 0 0 0 0 7 0 8 0 0\n",
      " 8 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 7 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 8 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 8 0 0 0 0 8 0 0 0 0 0 0 8 0 8 3 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 8 0 0 0 3 8 0 0 0 0 0 0 0 0 7 7 8 7 0 1 3 8 0 0 3 3 0\n",
      " 0 8 0 0 0 8 7 7 8 3 8 0 8 0 0 0 0 8 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 8\n",
      " 0 0 0 8 0 0 0 0 0 0 0 0 0 0 0 0 0 8 3 0 0 0 8 0 3 0 8 0 3 8 3 0 0 8 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 7 3 0 0 0 8 0 0 3 0 8 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 8 0 0 8 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 3 0 0 0 0 0 3 0 0 0 0 0 8\n",
      " 0 0 8 0 0 0 0 0 0 7 0 0 8 3 0 0 0 0 0 0 8 0 0 0 0 0 0 0 8 8 0 0 8 8 0 3 0\n",
      " 3 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 8 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 8 0 3 8 0 8 0 0 0 0 0 0 0 0 0 0 8 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 8 0 0 0 0 0 8 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 8 0 0 0 0 0 7 0 0 8 0 0 0 0 7 8 3 0 8 0 0 8 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 8 8 8 0 0 8 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 3 0 0 0 0 8 0 8 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 8 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 8 0 0 0 8 8 3 0 0 0 0 3 8 0 0 8 8\n",
      " 0 8 3 0 8 0 0 0 0 0 0 0 8 8 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 8 0 0 0\n",
      " 0 0 0 0 0 8 0 8 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 7 0 0 8 0 8 8 0 0 0 0 0 0 0\n",
      " 0]\n",
      "threshold density 0.15152286757602057\n",
      "cluster signal 82572.0\n",
      "cluster length 982400\n",
      "!!cluster density 0.08405130293159609\n",
      "cluster signal 13391.0\n",
      "cluster length 1984100\n",
      "!!cluster density 0.006749155788518724\n",
      "cluster signal 62166.0\n",
      "cluster length 1410700\n",
      "!!cluster density 0.04406748422768838\n",
      "cluster signal 126345.0\n",
      "cluster length 956500\n",
      "!!cluster density 0.1320909566126503\n",
      "cluster signal 140605.0\n",
      "cluster length 703800\n",
      "!!cluster density 0.19977976697925548\n",
      "cluster signal 51558.0\n",
      "cluster length 96400\n",
      "!!cluster density 0.5348340248962655\n",
      "cluster signal 103552.0\n",
      "cluster length 364500\n",
      "!!cluster density 0.28409327846364885\n",
      "cluster signal 25384.0\n",
      "cluster length 30800\n",
      "!!cluster density 0.8241558441558442\n",
      "cluster signal 492.0\n",
      "cluster length 100\n",
      "!!cluster density 4.92\n",
      "cluster signal 72514.0\n",
      "cluster length 188300\n",
      "!!cluster density 0.38509824747742966\n",
      "\n",
      "Labels mapping\n",
      " {3: 0, 0: 0, 8: 0, 7: 0, 1: 1, 2: 1, 4: 1, 6: 1, 5: 1, 9: 1}\n",
      "DEBUG: 53337 13839 \n",
      "\n",
      " +--------------------+-------------+---------------+--------------+-------+\n",
      "| bin_offset_90      | TT_S0       | starting_nt   | Chromosome   | +7    |\n",
      "| (category)         | (float64)   | (category)    | (category)   | ...   |\n",
      "|--------------------+-------------+---------------+--------------+-------|\n",
      "| (3000016, 3000106] | 7.0         | 3000016       | 18           | ...   |\n",
      "| (3000106, 3000206] | 1.0         | 3000106       | 18           | ...   |\n",
      "| (3000206, 3000306] | 0.0         | 3000206       | 18           | ...   |\n",
      "| (3000306, 3000406] | 2.0         | 3000306       | 18           | ...   |\n",
      "| ...                | ...         | ...           | ...          | ...   |\n",
      "| (9717206, 9717306] | 0.0         | 9717206       | 18           | ...   |\n",
      "| (9717306, 9717406] | 0.0         | 9717306       | 18           | ...   |\n",
      "| (9717406, 9717506] | 1.0         | 9717406       | 18           | ...   |\n",
      "| (9717506, 9717593] | 1.0         | 9717506       | 18           | ...   |\n",
      "+--------------------+-------------+---------------+--------------+-------+\n",
      "Unstranded PyRanges object has 53,337 rows and 11 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "7 hidden columns: Start, End, genes, transcripts, exons, CDSs, labels\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0\n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "BINS  IntervalIndex([(3000016, 3000111], (3000111, 3000211], (3000211, 3000311],\n",
      "               (3000311, 3000411], (3000411, 3000511], (3000511, 3000611],\n",
      "               (3000611, 3000711], (3000711, 3000811], (3000811, 3000911],\n",
      "               (3000911, 3001011],\n",
      "               ...\n",
      "               (9716611, 9716711], (9716711, 9716811], (9716811, 9716911],\n",
      "               (9716911, 9717011], (9717011, 9717111], (9717111, 9717211],\n",
      "               (9717211, 9717311], (9717311, 9717411], (9717411, 9717511],\n",
      "               (9717511, 9717593]],\n",
      "              dtype='interval[int64, right]', length=67176)\n",
      "HEAD with BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0   \n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "\n",
      "        bin_offset_95  \n",
      "0  (3000016, 3000111]  \n",
      "1  (3000016, 3000111]  \n",
      "2  (3000016, 3000111]  \n",
      "3  (3000016, 3000111]  \n",
      "4  (3000016, 3000111]  \n",
      "TAIL with BINS\n",
      "       seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  9717555  9717555  18_9717555    1.0    0.0    0.0   \n",
      "699996       19  9717571  9717571  19_9717571    0.0    0.0    0.0   \n",
      "699997       19  9717581  9717581  19_9717581    0.0    0.0    0.0   \n",
      "699998       18  9717582  9717582  18_9717582    0.0    0.0    0.0   \n",
      "699999       18  9717593  9717593  18_9717593    0.0    0.0    0.0   \n",
      "\n",
      "             bin_offset_95  \n",
      "699995  (9717511, 9717593]  \n",
      "699996  (9717511, 9717593]  \n",
      "699997  (9717511, 9717593]  \n",
      "699998  (9717511, 9717593]  \n",
      "699999  (9717511, 9717593]  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:136: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "         bin_offset_95  TT_S0 starting_nt\n",
      "0   (3000016, 3000111]    7.0     3000016\n",
      "1   (3000111, 3000211]    1.0     3000111\n",
      "2   (3000211, 3000311]    2.0     3000211\n",
      "3   (3000311, 3000411]    0.0     3000311\n",
      "4   (3000411, 3000511]    0.0     3000411\n",
      "5   (3000511, 3000611]    0.0     3000511\n",
      "6   (3000611, 3000711]    0.0     3000611\n",
      "7   (3000711, 3000811]    0.0     3000711\n",
      "8   (3000811, 3000911]    0.0     3000811\n",
      "9   (3000911, 3001011]    0.0     3000911\n",
      "10  (3001011, 3001111]    2.0     3001011\n",
      "11  (3001111, 3001211]    0.0     3001111\n",
      "12  (3001211, 3001311]    0.0     3001211\n",
      "13  (3001311, 3001411]    0.0     3001311\n",
      "14  (3001411, 3001511]    1.0     3001411\n",
      "15  (3001511, 3001611]    3.0     3001511\n",
      "16  (3001611, 3001711]    3.0     3001611\n",
      "17  (3001711, 3001811]    0.0     3001711\n",
      "18  (3001811, 3001911]    0.0     3001811\n",
      "19  (3001911, 3002011]    1.0     3001911\n",
      "20  (3002011, 3002111]    0.0     3002011\n",
      "21  (3002111, 3002211]    1.0     3002111\n",
      "22  (3002211, 3002311]   10.0     3002211\n",
      "23  (3002311, 3002411]    2.0     3002311\n",
      "24  (3002411, 3002511]   19.0     3002411\n",
      "25  (3002511, 3002611]    4.0     3002511\n",
      "26  (3002611, 3002711]    0.0     3002611\n",
      "27  (3002711, 3002811]    0.0     3002711\n",
      "28  (3002811, 3002911]    0.0     3002811\n",
      "29  (3002911, 3003011]    1.0     3002911\n",
      "30  (3003011, 3003111]    2.0     3003011\n",
      "31  (3003111, 3003211]    0.0     3003111\n",
      "32  (3003211, 3003311]    0.0     3003211\n",
      "33  (3003311, 3003411]    1.0     3003311\n",
      "34  (3003411, 3003511]    0.0     3003411\n",
      "35  (3003511, 3003611]    1.0     3003511\n",
      "36  (3003611, 3003711]    1.0     3003611\n",
      "37  (3003711, 3003811]    3.0     3003711\n",
      "38  (3003811, 3003911]    0.0     3003811\n",
      "39  (3003911, 3004011]    8.0     3003911\n",
      "40  (3004011, 3004111]    0.0     3004011\n",
      "41  (3004111, 3004211]    0.0     3004111\n",
      "42  (3004211, 3004311]    5.0     3004211\n",
      "43  (3004311, 3004411]    0.0     3004311\n",
      "44  (3004411, 3004511]    0.0     3004411\n",
      "45  (3004511, 3004611]    1.0     3004511\n",
      "46  (3004611, 3004711]    0.0     3004611\n",
      "47  (3004711, 3004811]    0.0     3004711\n",
      "48  (3004811, 3004911]   10.0     3004811\n",
      "49  (3004911, 3005011]    3.0     3004911\n",
      "(67176, 3)\n",
      "bin_offset_95    category\n",
      "TT_S0             float64\n",
      "starting_nt      category\n",
      "dtype: object\n",
      "sumos ilgis: 67176\n",
      "ranges ilgis: 67176\n",
      "Model input - NO SCALE\n",
      "       TT_S0  genes  transcripts  exons  CDSs\n",
      "0        7.0      0            0      0     0\n",
      "1        1.0      0            0      0     0\n",
      "2        2.0      0            0      0     0\n",
      "3        0.0      0            0      0     0\n",
      "4        0.0      0            0      0     0\n",
      "...      ...    ...          ...    ...   ...\n",
      "67171    0.0      0            0      0     0\n",
      "67172    0.0      0            0      0     0\n",
      "67173    0.0      0            0      0     0\n",
      "67174    1.0      0            0      0     0\n",
      "67175    1.0      0            0      0     0\n",
      "\n",
      "[67176 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "       TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0        7.0             0                   0             0            0\n",
      "1        1.0             0                   0             0            0\n",
      "2        2.0             0                   0             0            0\n",
      "3        0.0             0                   0             0            0\n",
      "4        0.0             0                   0             0            0\n",
      "...      ...           ...                 ...           ...          ...\n",
      "67171    0.0             0                   0             0            0\n",
      "67172    0.0             0                   0             0            0\n",
      "67173    0.0             0                   0             0            0\n",
      "67174    1.0             0                   0             0            0\n",
      "67175    1.0             0                   0             0            0\n",
      "\n",
      "[67176 rows x 5 columns]\n",
      "0.2635167321662499\n",
      "[1 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 7 4 2 1 4 4 4 4 4 4 4 4 4 4 4\n",
      " 4 4 1 4 4 1 4 4 4 4 4 7 4 1 3 0 1 2 7 9 9 6 6 4 7 8 1 4 4 1 4 4 4 1 4 1 4\n",
      " 4 4 4 4 4 4 1 4 4 4 1 4 1 1 4 4 7 2 4 4 4 4 4 4 7 1 2 8 8 7 1 1 7 8 1 7 4\n",
      " 7 1 4 4 4 4 4 4 4 4 1 4 4 7 4 2 7 7 7 1 1 1 1 4 1 1 4 4 4 4 4 4 4 4 4 4 2\n",
      " 4 1 4 4 4 1 4 7 4 4 4 2 4 7 4 4 7 7 4 4 4 4 4 4 4 4 4 4 4 4 4 4 7 4 4 4 4\n",
      " 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 7 2 4 4 4 4 4 4 4 4 4 4 4 4 4 4\n",
      " 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4\n",
      " 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 1\n",
      " 4 4 4 4 4 4 4 4 4 4 4 4 1 4 1 1 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4\n",
      " 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4\n",
      " 4 4 4 4 4 4 4 4 4 4 1 4 4 4 1 1 4 4 4 4 4 4 4 4 7 7 1 7 4 8 1 4 4 4 1 1 4\n",
      " 4 1 4 4 4 1 7 7 4 7 1 4 4 4 4 4 4 1 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4\n",
      " 4 4 4 1 4 4 4 4 4 4 4 4 4 4 4 4 4 4 1 4 4 4 1 4 7 4 1 1 1 4 1 4 4 4 4 4 4\n",
      " 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4\n",
      " 7 1 4 4 4 1 4 4 1 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4\n",
      " 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4\n",
      " 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 1 4 4 4 4 4 1 4 4 4 4 4 1\n",
      " 4 4 1 4 4 4 4 4 4 7 4 4 1 1 4 4 4 4 4 4 1 4 4 4 4 4 4 4 1 1 4 4 1 1 4 1 4\n",
      " 1 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 1 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4\n",
      " 4 4 4 4 4 4 4 4 4 4 4 4 4 4 1 1 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4\n",
      " 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4\n",
      " 4 4 4 4 4 4 4 4 4 4 4 4 4 1 4 4 4 4 4 7 4 4 1 4 4 4 4 7 1 7 4 4 4 4 4 4 4\n",
      " 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 1 4 4 4 4 4 4 4 4 4\n",
      " 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 1 4 4 4 4 4 4 1 4 4 4 4 4 4 4 4 4 4 4\n",
      " 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 1 4 1 4 4 4 4 1 1 4 4 1 1\n",
      " 4 1 1 4 1 4 4 4 4 4 4 4 1 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4\n",
      " 4 4 4 4 4 4 4 1 4 4 4 4 4 4 4 4 4 4 4 4 4 4 7 4 4 4 4 4 1 4 4 4 4 4 4 4 4\n",
      " 4]\n",
      "threshold density 0.151522754795429\n",
      "cluster signal 110488.0\n",
      "cluster length 1770200\n",
      "!!cluster density 0.062415546265958646\n",
      "cluster signal 25504.0\n",
      "cluster length 2387400\n",
      "!!cluster density 0.010682751109994136\n",
      "cluster signal 109115.0\n",
      "cluster length 922900\n",
      "!!cluster density 0.11823057752735941\n",
      "cluster signal 148826.0\n",
      "cluster length 825500\n",
      "!!cluster density 0.18028588734100545\n",
      "cluster signal 52204.0\n",
      "cluster length 110200\n",
      "!!cluster density 0.47372050816696915\n",
      "cluster signal 82510.0\n",
      "cluster length 234400\n",
      "!!cluster density 0.35200511945392493\n",
      "cluster signal 14082.0\n",
      "cluster length 15200\n",
      "!!cluster density 0.9264473684210527\n",
      "cluster signal 678.0\n",
      "cluster length 200\n",
      "!!cluster density 3.39\n",
      "cluster signal 105496.0\n",
      "cluster length 405800\n",
      "!!cluster density 0.25997042878265153\n",
      "cluster signal 29676.0\n",
      "cluster length 45800\n",
      "!!cluster density 0.6479475982532751\n",
      "\n",
      "Labels mapping\n",
      " {1: 0, 4: 0, 7: 0, 2: 1, 3: 1, 0: 1, 9: 1, 6: 1, 8: 1, 5: 1}\n",
      "DEBUG: 50805 16371 \n",
      "\n",
      " +--------------------+-------------+---------------+--------------+-------+\n",
      "| bin_offset_95      | TT_S0       | starting_nt   | Chromosome   | +7    |\n",
      "| (category)         | (float64)   | (category)    | (category)   | ...   |\n",
      "|--------------------+-------------+---------------+--------------+-------|\n",
      "| (3000016, 3000111] | 7.0         | 3000016       | 18           | ...   |\n",
      "| (3000111, 3000211] | 1.0         | 3000111       | 18           | ...   |\n",
      "| (3000211, 3000311] | 2.0         | 3000211       | 18           | ...   |\n",
      "| (3000311, 3000411] | 0.0         | 3000311       | 18           | ...   |\n",
      "| ...                | ...         | ...           | ...          | ...   |\n",
      "| (9717211, 9717311] | 0.0         | 9717211       | 18           | ...   |\n",
      "| (9717311, 9717411] | 0.0         | 9717311       | 18           | ...   |\n",
      "| (9717411, 9717511] | 1.0         | 9717411       | 18           | ...   |\n",
      "| (9717511, 9717593] | 1.0         | 9717511       | 18           | ...   |\n",
      "+--------------------+-------------+---------------+--------------+-------+\n",
      "Unstranded PyRanges object has 50,805 rows and 11 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "7 hidden columns: Start, End, genes, transcripts, exons, CDSs, labels\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/anaconda3/lib/python3.12/site-packages/pyranges/methods/intersection.py:95: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '[1 1 1 ... 1 1 1]' has dtype incompatible with int32, please explicitly cast to a compatible dtype first.\n",
      "  sx.loc[vc.index, 0] = vc.values\n",
      "/opt/anaconda3/lib/python3.12/site-packages/pyranges/methods/intersection.py:95: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '[1 1 1 ... 1 1 1]' has dtype incompatible with int32, please explicitly cast to a compatible dtype first.\n",
      "  sx.loc[vc.index, 0] = vc.values\n",
      "/opt/anaconda3/lib/python3.12/site-packages/pyranges/methods/intersection.py:95: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '[1 1 1 ... 1 1 1]' has dtype incompatible with int32, please explicitly cast to a compatible dtype first.\n",
      "  sx.loc[vc.index, 0] = vc.values\n",
      "/opt/anaconda3/lib/python3.12/site-packages/pyranges/methods/intersection.py:95: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '[1 1 1 ... 1 1 1]' has dtype incompatible with int32, please explicitly cast to a compatible dtype first.\n",
      "  sx.loc[vc.index, 0] = vc.values\n",
      "/opt/anaconda3/lib/python3.12/site-packages/pyranges/methods/intersection.py:95: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '[1 1 1 ... 1 1 1]' has dtype incompatible with int32, please explicitly cast to a compatible dtype first.\n",
      "  sx.loc[vc.index, 0] = vc.values\n",
      "/opt/anaconda3/lib/python3.12/site-packages/pyranges/methods/intersection.py:95: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '[1 1 1 ... 1 1 1]' has dtype incompatible with int32, please explicitly cast to a compatible dtype first.\n",
      "  sx.loc[vc.index, 0] = vc.values\n",
      "/opt/anaconda3/lib/python3.12/site-packages/pyranges/methods/intersection.py:95: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '[1 1 1 ... 1 1 1]' has dtype incompatible with int32, please explicitly cast to a compatible dtype first.\n",
      "  sx.loc[vc.index, 0] = vc.values\n",
      "/opt/anaconda3/lib/python3.12/site-packages/pyranges/methods/intersection.py:95: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '[1 1 1 ... 1 1 1]' has dtype incompatible with int32, please explicitly cast to a compatible dtype first.\n",
      "  sx.loc[vc.index, 0] = vc.values\n",
      "/opt/anaconda3/lib/python3.12/site-packages/pyranges/methods/intersection.py:95: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '[1 1 1 ... 1 1 1]' has dtype incompatible with int32, please explicitly cast to a compatible dtype first.\n",
      "  sx.loc[vc.index, 0] = vc.values\n",
      "/opt/anaconda3/lib/python3.12/site-packages/pyranges/methods/intersection.py:95: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '[1 1 1 ... 1 1 1]' has dtype incompatible with int32, please explicitly cast to a compatible dtype first.\n",
      "  sx.loc[vc.index, 0] = vc.values\n",
      "/opt/anaconda3/lib/python3.12/site-packages/pyranges/methods/intersection.py:95: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '[1 1 1 ... 1 1 1]' has dtype incompatible with int32, please explicitly cast to a compatible dtype first.\n",
      "  sx.loc[vc.index, 0] = vc.values\n",
      "/opt/anaconda3/lib/python3.12/site-packages/pyranges/methods/intersection.py:95: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '[1 1 1 ... 1 1 1]' has dtype incompatible with int32, please explicitly cast to a compatible dtype first.\n",
      "  sx.loc[vc.index, 0] = vc.values\n",
      "/opt/anaconda3/lib/python3.12/site-packages/pyranges/methods/intersection.py:95: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '[1 1 1 ... 1 1 1]' has dtype incompatible with int32, please explicitly cast to a compatible dtype first.\n",
      "  sx.loc[vc.index, 0] = vc.values\n",
      "/opt/anaconda3/lib/python3.12/site-packages/pyranges/methods/intersection.py:95: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '[1 1 1 ... 1 1 1]' has dtype incompatible with int32, please explicitly cast to a compatible dtype first.\n",
      "  sx.loc[vc.index, 0] = vc.values\n",
      "/opt/anaconda3/lib/python3.12/site-packages/pyranges/methods/intersection.py:95: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '[1 1 1 ... 1 1 1]' has dtype incompatible with int32, please explicitly cast to a compatible dtype first.\n",
      "  sx.loc[vc.index, 0] = vc.values\n",
      "/opt/anaconda3/lib/python3.12/site-packages/pyranges/methods/intersection.py:95: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '[1 1 1 ... 1 1 1]' has dtype incompatible with int32, please explicitly cast to a compatible dtype first.\n",
      "  sx.loc[vc.index, 0] = vc.values\n",
      "/opt/anaconda3/lib/python3.12/site-packages/pyranges/methods/intersection.py:95: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '[1 1 1 ... 1 1 1]' has dtype incompatible with int32, please explicitly cast to a compatible dtype first.\n",
      "  sx.loc[vc.index, 0] = vc.values\n",
      "/opt/anaconda3/lib/python3.12/site-packages/pyranges/methods/intersection.py:95: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '[1 1 1 ... 1 1 1]' has dtype incompatible with int32, please explicitly cast to a compatible dtype first.\n",
      "  sx.loc[vc.index, 0] = vc.values\n",
      "/opt/anaconda3/lib/python3.12/site-packages/pyranges/methods/intersection.py:95: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '[1 1 1 ... 1 1 1]' has dtype incompatible with int32, please explicitly cast to a compatible dtype first.\n",
      "  sx.loc[vc.index, 0] = vc.values\n",
      "/opt/anaconda3/lib/python3.12/site-packages/pyranges/methods/intersection.py:95: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '[1 1 1 ... 1 1 1]' has dtype incompatible with int32, please explicitly cast to a compatible dtype first.\n",
      "  sx.loc[vc.index, 0] = vc.values\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "OVERLAPS +--------------+-----------+-----------+------------+------------+-------+\n",
      "| Chromosome   | Start     | End       | offset_0   | offset_5   | +18   |\n",
      "| (object)     | (int64)   | (int64)   | (int64)    | (int64)    | ...   |\n",
      "|--------------+-----------+-----------+------------+------------+-------|\n",
      "| 18           | 3002346   | 3002351   | 0          | 0          | ...   |\n",
      "| 18           | 3002351   | 3002361   | 0          | 0          | ...   |\n",
      "| 18           | 3002361   | 3002366   | 0          | 0          | ...   |\n",
      "| 18           | 3002366   | 3002371   | 0          | 0          | ...   |\n",
      "| ...          | ...       | ...       | ...        | ...        | ...   |\n",
      "| 18           | 9709416   | 9709421   | 0          | 1          | ...   |\n",
      "| 18           | 9709421   | 9709431   | 0          | 0          | ...   |\n",
      "| 18           | 9709431   | 9713761   | 0          | 0          | ...   |\n",
      "| 18           | 9713761   | 9713861   | 0          | 0          | ...   |\n",
      "+--------------+-----------+-----------+------------+------------+-------+\n",
      "Unstranded PyRanges object has 316,561 rows and 23 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "18 hidden columns: offset_10, offset_15, offset_20, offset_25, offset_30, offset_35, ... (+ 12 more.)\n",
      "OVERLAPS FILTERED +--------------+-----------+-----------+\n",
      "| Chromosome   | Start     | End       |\n",
      "| (category)   | (int64)   | (int64)   |\n",
      "|--------------+-----------+-----------|\n",
      "| 18           | 3002401   | 3002466   |\n",
      "| 18           | 3005071   | 3005316   |\n",
      "| 18           | 3005436   | 3005541   |\n",
      "| 18           | 3005601   | 3006001   |\n",
      "| ...          | ...       | ...       |\n",
      "| 18           | 9708001   | 9708431   |\n",
      "| 18           | 9708556   | 9708846   |\n",
      "| 18           | 9708991   | 9709096   |\n",
      "| 18           | 9709311   | 9709361   |\n",
      "+--------------+-----------+-----------+\n",
      "Unstranded PyRanges object has 8,182 rows and 3 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "count                         8182.0\n",
      "mean                      182.306282\n",
      "std                       197.351081\n",
      "min                              5.0\n",
      "25%                             80.0\n",
      "50%                            115.0\n",
      "75%                            210.0\n",
      "max                           2085.0\n",
      "ATACseq_overlap             0.751796\n",
      "ATACseq_ranges_overlap      0.013469\n",
      "DNAseq_overlap              0.639008\n",
      "DNAseq_ranges_overlap       0.022784\n",
      "genes_overlap                0.54393\n",
      "percentage_open             0.153498\n",
      "percentage_open_adj         0.222048\n",
      "model_name                          \n",
      "Name: Length, dtype: object\n"
     ]
    }
   ],
   "source": [
    "import generate_consensus\n",
    "\n",
    "ranges = generate_consensus.generate_consensus_5x20(chromatin_algos.algo5d_aggregated, [1, 10])\n",
    "result = evaluator.evaluate(merged_ranges = ranges)\n",
    "print(result)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "6b5c5de3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "count                         8182.0\n",
      "mean                      182.306282\n",
      "std                       197.351081\n",
      "min                              5.0\n",
      "25%                             80.0\n",
      "50%                            115.0\n",
      "75%                            210.0\n",
      "max                           2085.0\n",
      "ATACseq_overlap             0.751796\n",
      "ATACseq_ranges_overlap      0.013469\n",
      "DNAseq_overlap              0.639008\n",
      "DNAseq_ranges_overlap       0.022784\n",
      "genes_overlap                0.54393\n",
      "percentage_open             0.153498\n",
      "percentage_open_adj         0.222048\n",
      "model_name                          \n",
      "Name: Length, dtype: object\n",
      "count                         6327.0\n",
      "mean                      225.218113\n",
      "std                       221.853931\n",
      "min                             75.0\n",
      "25%                            100.0\n",
      "50%                            140.0\n",
      "75%                            260.0\n",
      "max                           2215.0\n",
      "ATACseq_overlap             0.754341\n",
      "ATACseq_ranges_overlap      0.014147\n",
      "DNAseq_overlap              0.634176\n",
      "DNAseq_ranges_overlap        0.02367\n",
      "genes_overlap               0.546901\n",
      "percentage_open             0.146637\n",
      "percentage_open_adj         0.212123\n",
      "model_name                          \n",
      "Name: Length, dtype: object\n"
     ]
    }
   ],
   "source": [
    "result_p = evaluator.evaluate_processed(merged_ranges = ranges)\n",
    "print(result)\n",
    "print(result_p)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "9435baf8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1615\n",
      "0\n"
     ]
    }
   ],
   "source": [
    "import pyranges as pr\n",
    "\n",
    "ranges.Length = ranges.End - ranges.Start\n",
    "ranges_df = ranges.df\n",
    "ranges_df['']\n",
    "\n",
    "print(len(ranges[ranges.Length < 75]))\n",
    "ranges2 = evaluator.process_ranges(ranges)\n",
    "ranges2.Length = ranges2.End - ranges2.Start\n",
    "print(len(ranges2[ranges2.Length < 75]))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "61959364",
   "metadata": {},
   "outputs": [],
   "source": [
    "def calculate_gaps(pyranges_obj):\n",
    "    \"\"\"\n",
    "    Calculate gaps between ranges in a PyRanges object.\n",
    "\n",
    "    Parameters:\n",
    "        pyranges_obj (pr.PyRanges): A PyRanges object with genomic ranges.\n",
    "\n",
    "    Returns:\n",
    "        pr.PyRanges: A PyRanges object containing the gaps.\n",
    "    \"\"\"\n",
    "    # Convert PyRanges to a DataFrame\n",
    "    df = pyranges_obj.df\n",
    "    \n",
    "    # Sort the ranges by Chromosome and Start position\n",
    "    df = df.sort_values([\"Chromosome\", \"Start\"]).reset_index(drop=True)\n",
    "    \n",
    "    # Shift End and Start columns to calculate gaps\n",
    "    prev_end = df.groupby(\"Chromosome\")[\"End\"].shift(0)   # Current End\n",
    "    next_start = df.groupby(\"Chromosome\")[\"Start\"].shift(-1)  # Next Start\n",
    "    \n",
    "    # Identify gaps where End < Next Start\n",
    "    gaps_df = df[(prev_end < next_start)].assign(Start=prev_end, End=next_start)[[\"Chromosome\", \"Start\", \"End\"]]\n",
    "    \n",
    "    # Drop NaN rows caused by shifting and ensure integer type\n",
    "    gaps_df = gaps_df.dropna().astype({\"Start\": \"int\", \"End\": \"int\"})\n",
    "    \n",
    "    # Convert back to PyRanges\n",
    "    return pr.PyRanges(gaps_df)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "cbaf97f0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "+--------------+-----------+-----------+--------------+------------+\n",
      "| Chromosome   |     Start |       End | Strand       | title      |\n",
      "| (category)   |   (int64) |   (int64) | (category)   | (object)   |\n",
      "|--------------+-----------+-----------+--------------+------------|\n",
      "| chr1         |         5 |        10 | +            | a          |\n",
      "| chr1         |        20 |        28 | +            | b          |\n",
      "| chr1         |        80 |        95 | -            | c          |\n",
      "| chr3         |        10 |        38 | +            | d          |\n",
      "+--------------+-----------+-----------+--------------+------------+\n",
      "Stranded PyRanges object has 4 rows and 5 columns from 2 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome and Strand.\n",
      "+--------------+-----------+-----------+--------------+\n",
      "| Chromosome   |     Start |       End | Strand       |\n",
      "| (category)   |   (int64) |   (int64) | (category)   |\n",
      "|--------------+-----------+-----------+--------------|\n",
      "| chr1         |         5 |        28 | +            |\n",
      "| chr1         |        80 |        95 | -            |\n",
      "| chr3         |        10 |        38 | +            |\n",
      "+--------------+-----------+-----------+--------------+\n",
      "Stranded PyRanges object has 3 rows and 4 columns from 2 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome and Strand.\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "df=pd.DataFrame(\n",
    "{'Chromosome':['chr1', 'chr1', 'chr1', 'chr3'],\n",
    " 'Start': [5, 20, 80, 10],\n",
    " 'End':   [10, 28, 95, 38],\n",
    " 'Strand':['+', '+', '-', '+'],\n",
    " 'title': ['a', 'b', 'c', 'd']}\n",
    ")\n",
    "\n",
    "p=pr.PyRanges(df)\n",
    "print(p)\n",
    "p2=p.merge(slack = 10)\n",
    "print(p2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "847f9a77",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "443\n",
      "0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/y9/8yssfmz161s4ghy99nf1f2cw0000gn/T/ipykernel_27239/19855938.py:18: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  prev_end = df.groupby(\"Chromosome\")[\"End\"].shift(0)   # Current End\n",
      "/var/folders/y9/8yssfmz161s4ghy99nf1f2cw0000gn/T/ipykernel_27239/19855938.py:19: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  next_start = df.groupby(\"Chromosome\")[\"Start\"].shift(-1)  # Next Start\n",
      "/var/folders/y9/8yssfmz161s4ghy99nf1f2cw0000gn/T/ipykernel_27239/19855938.py:18: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  prev_end = df.groupby(\"Chromosome\")[\"End\"].shift(0)   # Current End\n",
      "/var/folders/y9/8yssfmz161s4ghy99nf1f2cw0000gn/T/ipykernel_27239/19855938.py:19: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  next_start = df.groupby(\"Chromosome\")[\"Start\"].shift(-1)  # Next Start\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import pyranges as pr\n",
    "\n",
    "ranges_df = ranges.df\n",
    "gaps = calculate_gaps(ranges)\n",
    "gaps.Length = gaps.End - gaps.Start\n",
    "print(len(gaps[gaps.Length <= 50]))\n",
    "\n",
    "ranges2 = evaluator.process_ranges(ranges)\n",
    "gaps2 = calculate_gaps(ranges2)\n",
    "gaps2.Length = gaps2.End - gaps2.Start\n",
    "print(len(gaps2[gaps2.Length <= 50]))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "28e1cdda",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Runnning scale 1\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0\n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "BINS  IntervalIndex([(3000016, 3000016], (3000016, 3000116], (3000116, 3000216],\n",
      "               (3000216, 3000316], (3000316, 3000416], (3000416, 3000516],\n",
      "               (3000516, 3000616], (3000616, 3000716], (3000716, 3000816],\n",
      "               (3000816, 3000916],\n",
      "               ...\n",
      "               (9716616, 9716716], (9716716, 9716816], (9716816, 9716916],\n",
      "               (9716916, 9717016], (9717016, 9717116], (9717116, 9717216],\n",
      "               (9717216, 9717316], (9717316, 9717416], (9717416, 9717516],\n",
      "               (9717516, 9717593]],\n",
      "              dtype='interval[int64, right]', length=67177)\n",
      "HEAD w BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0   \n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "\n",
      "         bin_offset_0  \n",
      "0  (3000016, 3000116]  \n",
      "1  (3000016, 3000116]  \n",
      "2  (3000016, 3000116]  \n",
      "3  (3000016, 3000116]  \n",
      "4  (3000016, 3000116]  \n",
      "TAIL w BINS\n",
      "       seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  9717555  9717555  18_9717555    1.0    0.0    0.0   \n",
      "699996       19  9717571  9717571  19_9717571    0.0    0.0    0.0   \n",
      "699997       19  9717581  9717581  19_9717581    0.0    0.0    0.0   \n",
      "699998       18  9717582  9717582  18_9717582    0.0    0.0    0.0   \n",
      "699999       18  9717593  9717593  18_9717593    0.0    0.0    0.0   \n",
      "\n",
      "              bin_offset_0  \n",
      "699995  (9717516, 9717593]  \n",
      "699996  (9717516, 9717593]  \n",
      "699997  (9717516, 9717593]  \n",
      "699998  (9717516, 9717593]  \n",
      "699999  (9717516, 9717593]  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:151: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "    TT_S0  starting_nt\n",
      "0     0.0      3000016\n",
      "1     7.0      3000016\n",
      "2     1.0      3000116\n",
      "3     2.0      3000216\n",
      "4     0.0      3000316\n",
      "5     0.0      3000416\n",
      "6     0.0      3000516\n",
      "7     0.0      3000616\n",
      "8     0.0      3000716\n",
      "9     0.0      3000816\n",
      "10    0.0      3000916\n",
      "11    2.0      3001016\n",
      "12    0.0      3001116\n",
      "13    0.0      3001216\n",
      "14    0.0      3001316\n",
      "15    1.0      3001416\n",
      "16    3.0      3001516\n",
      "17    3.0      3001616\n",
      "18    0.0      3001716\n",
      "19    0.0      3001816\n",
      "20    1.0      3001916\n",
      "21    0.0      3002016\n",
      "22    1.0      3002116\n",
      "23   10.0      3002216\n",
      "24    2.0      3002316\n",
      "25   19.0      3002416\n",
      "26    4.0      3002516\n",
      "27    0.0      3002616\n",
      "28    0.0      3002716\n",
      "29    0.0      3002816\n",
      "30    1.0      3002916\n",
      "31    2.0      3003016\n",
      "32    0.0      3003116\n",
      "33    0.0      3003216\n",
      "34    1.0      3003316\n",
      "35    0.0      3003416\n",
      "36    1.0      3003516\n",
      "37    1.0      3003616\n",
      "38    3.0      3003716\n",
      "39    0.0      3003816\n",
      "40    8.0      3003916\n",
      "41    0.0      3004016\n",
      "42    0.0      3004116\n",
      "43    5.0      3004216\n",
      "44    0.0      3004316\n",
      "45    0.0      3004416\n",
      "46    1.0      3004516\n",
      "47    0.0      3004616\n",
      "48    0.0      3004716\n",
      "49   13.0      3004816\n",
      "(67177, 2)\n",
      "TT_S0          float64\n",
      "starting_nt      int64\n",
      "dtype: object\n",
      "sumos ilgis: 67177\n",
      "ranges ilgis: 67177\n",
      "Model input - NO SCALE\n",
      "       TT_S0  genes  transcripts  exons  CDSs\n",
      "0        0.0      0            0      0     0\n",
      "1        7.0      0            0      0     0\n",
      "2        1.0      0            0      0     0\n",
      "3        2.0      0            0      0     0\n",
      "4        0.0      0            0      0     0\n",
      "...      ...    ...          ...    ...   ...\n",
      "67172    0.0      0            0      0     0\n",
      "67173    0.0      0            0      0     0\n",
      "67174    0.0      0            0      0     0\n",
      "67175    1.0      0            0      0     0\n",
      "67176    1.0      0            0      0     0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "       TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0        0.0             0                   0             0            0\n",
      "1        7.0             0                   0             0            0\n",
      "2        1.0             0                   0             0            0\n",
      "3        2.0             0                   0             0            0\n",
      "4        0.0             0                   0             0            0\n",
      "...      ...           ...                 ...           ...          ...\n",
      "67172    0.0             0                   0             0            0\n",
      "67173    0.0             0                   0             0            0\n",
      "67174    0.0             0                   0             0            0\n",
      "67175    1.0             0                   0             0            0\n",
      "67176    1.0             0                   0             0            0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "0.17092159518882952\n",
      "[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 1 0 1 1 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 1 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0]\n",
      "Runnning scale 2\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0\n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "BINS  IntervalIndex([(3000016, 3000016], (3000016, 3000116], (3000116, 3000216],\n",
      "               (3000216, 3000316], (3000316, 3000416], (3000416, 3000516],\n",
      "               (3000516, 3000616], (3000616, 3000716], (3000716, 3000816],\n",
      "               (3000816, 3000916],\n",
      "               ...\n",
      "               (9716616, 9716716], (9716716, 9716816], (9716816, 9716916],\n",
      "               (9716916, 9717016], (9717016, 9717116], (9717116, 9717216],\n",
      "               (9717216, 9717316], (9717316, 9717416], (9717416, 9717516],\n",
      "               (9717516, 9717593]],\n",
      "              dtype='interval[int64, right]', length=67177)\n",
      "HEAD w BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0   \n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "\n",
      "         bin_offset_0  \n",
      "0  (3000016, 3000116]  \n",
      "1  (3000016, 3000116]  \n",
      "2  (3000016, 3000116]  \n",
      "3  (3000016, 3000116]  \n",
      "4  (3000016, 3000116]  \n",
      "TAIL w BINS\n",
      "       seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  9717555  9717555  18_9717555    1.0    0.0    0.0   \n",
      "699996       19  9717571  9717571  19_9717571    0.0    0.0    0.0   \n",
      "699997       19  9717581  9717581  19_9717581    0.0    0.0    0.0   \n",
      "699998       18  9717582  9717582  18_9717582    0.0    0.0    0.0   \n",
      "699999       18  9717593  9717593  18_9717593    0.0    0.0    0.0   \n",
      "\n",
      "              bin_offset_0  \n",
      "699995  (9717516, 9717593]  \n",
      "699996  (9717516, 9717593]  \n",
      "699997  (9717516, 9717593]  \n",
      "699998  (9717516, 9717593]  \n",
      "699999  (9717516, 9717593]  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:151: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "    TT_S0  starting_nt\n",
      "0     0.0      3000016\n",
      "1     7.0      3000016\n",
      "2     1.0      3000116\n",
      "3     2.0      3000216\n",
      "4     0.0      3000316\n",
      "5     0.0      3000416\n",
      "6     0.0      3000516\n",
      "7     0.0      3000616\n",
      "8     0.0      3000716\n",
      "9     0.0      3000816\n",
      "10    0.0      3000916\n",
      "11    2.0      3001016\n",
      "12    0.0      3001116\n",
      "13    0.0      3001216\n",
      "14    0.0      3001316\n",
      "15    1.0      3001416\n",
      "16    3.0      3001516\n",
      "17    3.0      3001616\n",
      "18    0.0      3001716\n",
      "19    0.0      3001816\n",
      "20    1.0      3001916\n",
      "21    0.0      3002016\n",
      "22    1.0      3002116\n",
      "23   10.0      3002216\n",
      "24    2.0      3002316\n",
      "25   19.0      3002416\n",
      "26    4.0      3002516\n",
      "27    0.0      3002616\n",
      "28    0.0      3002716\n",
      "29    0.0      3002816\n",
      "30    1.0      3002916\n",
      "31    2.0      3003016\n",
      "32    0.0      3003116\n",
      "33    0.0      3003216\n",
      "34    1.0      3003316\n",
      "35    0.0      3003416\n",
      "36    1.0      3003516\n",
      "37    1.0      3003616\n",
      "38    3.0      3003716\n",
      "39    0.0      3003816\n",
      "40    8.0      3003916\n",
      "41    0.0      3004016\n",
      "42    0.0      3004116\n",
      "43    5.0      3004216\n",
      "44    0.0      3004316\n",
      "45    0.0      3004416\n",
      "46    1.0      3004516\n",
      "47    0.0      3004616\n",
      "48    0.0      3004716\n",
      "49   13.0      3004816\n",
      "(67177, 2)\n",
      "TT_S0          float64\n",
      "starting_nt      int64\n",
      "dtype: object\n",
      "sumos ilgis: 67177\n",
      "ranges ilgis: 67177\n",
      "Model input - NO SCALE\n",
      "       TT_S0  genes  transcripts  exons  CDSs\n",
      "0        0.0      0            0      0     0\n",
      "1        7.0      0            0      0     0\n",
      "2        1.0      0            0      0     0\n",
      "3        2.0      0            0      0     0\n",
      "4        0.0      0            0      0     0\n",
      "...      ...    ...          ...    ...   ...\n",
      "67172    0.0      0            0      0     0\n",
      "67173    0.0      0            0      0     0\n",
      "67174    0.0      0            0      0     0\n",
      "67175    1.0      0            0      0     0\n",
      "67176    1.0      0            0      0     0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "       TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0        0.0             0                   0             0            0\n",
      "1        7.0             0                   0             0            0\n",
      "2        1.0             0                   0             0            0\n",
      "3        2.0             0                   0             0            0\n",
      "4        0.0             0                   0             0            0\n",
      "...      ...           ...                 ...           ...          ...\n",
      "67172    0.0             0                   0             0            0\n",
      "67173    0.0             0                   0             0            0\n",
      "67174    0.0             0                   0             0            0\n",
      "67175    1.0             0                   0             0            0\n",
      "67176    1.0             0                   0             0            0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "0.20496598538190156\n",
      "[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 1 0 1 1 1 1 0 0 1 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 1 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0]\n",
      "Runnning scale 4\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0\n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "BINS  IntervalIndex([(3000016, 3000016], (3000016, 3000116], (3000116, 3000216],\n",
      "               (3000216, 3000316], (3000316, 3000416], (3000416, 3000516],\n",
      "               (3000516, 3000616], (3000616, 3000716], (3000716, 3000816],\n",
      "               (3000816, 3000916],\n",
      "               ...\n",
      "               (9716616, 9716716], (9716716, 9716816], (9716816, 9716916],\n",
      "               (9716916, 9717016], (9717016, 9717116], (9717116, 9717216],\n",
      "               (9717216, 9717316], (9717316, 9717416], (9717416, 9717516],\n",
      "               (9717516, 9717593]],\n",
      "              dtype='interval[int64, right]', length=67177)\n",
      "HEAD w BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0   \n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "\n",
      "         bin_offset_0  \n",
      "0  (3000016, 3000116]  \n",
      "1  (3000016, 3000116]  \n",
      "2  (3000016, 3000116]  \n",
      "3  (3000016, 3000116]  \n",
      "4  (3000016, 3000116]  \n",
      "TAIL w BINS\n",
      "       seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  9717555  9717555  18_9717555    1.0    0.0    0.0   \n",
      "699996       19  9717571  9717571  19_9717571    0.0    0.0    0.0   \n",
      "699997       19  9717581  9717581  19_9717581    0.0    0.0    0.0   \n",
      "699998       18  9717582  9717582  18_9717582    0.0    0.0    0.0   \n",
      "699999       18  9717593  9717593  18_9717593    0.0    0.0    0.0   \n",
      "\n",
      "              bin_offset_0  \n",
      "699995  (9717516, 9717593]  \n",
      "699996  (9717516, 9717593]  \n",
      "699997  (9717516, 9717593]  \n",
      "699998  (9717516, 9717593]  \n",
      "699999  (9717516, 9717593]  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:151: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "    TT_S0  starting_nt\n",
      "0     0.0      3000016\n",
      "1     7.0      3000016\n",
      "2     1.0      3000116\n",
      "3     2.0      3000216\n",
      "4     0.0      3000316\n",
      "5     0.0      3000416\n",
      "6     0.0      3000516\n",
      "7     0.0      3000616\n",
      "8     0.0      3000716\n",
      "9     0.0      3000816\n",
      "10    0.0      3000916\n",
      "11    2.0      3001016\n",
      "12    0.0      3001116\n",
      "13    0.0      3001216\n",
      "14    0.0      3001316\n",
      "15    1.0      3001416\n",
      "16    3.0      3001516\n",
      "17    3.0      3001616\n",
      "18    0.0      3001716\n",
      "19    0.0      3001816\n",
      "20    1.0      3001916\n",
      "21    0.0      3002016\n",
      "22    1.0      3002116\n",
      "23   10.0      3002216\n",
      "24    2.0      3002316\n",
      "25   19.0      3002416\n",
      "26    4.0      3002516\n",
      "27    0.0      3002616\n",
      "28    0.0      3002716\n",
      "29    0.0      3002816\n",
      "30    1.0      3002916\n",
      "31    2.0      3003016\n",
      "32    0.0      3003116\n",
      "33    0.0      3003216\n",
      "34    1.0      3003316\n",
      "35    0.0      3003416\n",
      "36    1.0      3003516\n",
      "37    1.0      3003616\n",
      "38    3.0      3003716\n",
      "39    0.0      3003816\n",
      "40    8.0      3003916\n",
      "41    0.0      3004016\n",
      "42    0.0      3004116\n",
      "43    5.0      3004216\n",
      "44    0.0      3004316\n",
      "45    0.0      3004416\n",
      "46    1.0      3004516\n",
      "47    0.0      3004616\n",
      "48    0.0      3004716\n",
      "49   13.0      3004816\n",
      "(67177, 2)\n",
      "TT_S0          float64\n",
      "starting_nt      int64\n",
      "dtype: object\n",
      "sumos ilgis: 67177\n",
      "ranges ilgis: 67177\n",
      "Model input - NO SCALE\n",
      "       TT_S0  genes  transcripts  exons  CDSs\n",
      "0        0.0      0            0      0     0\n",
      "1        7.0      0            0      0     0\n",
      "2        1.0      0            0      0     0\n",
      "3        2.0      0            0      0     0\n",
      "4        0.0      0            0      0     0\n",
      "...      ...    ...          ...    ...   ...\n",
      "67172    0.0      0            0      0     0\n",
      "67173    0.0      0            0      0     0\n",
      "67174    0.0      0            0      0     0\n",
      "67175    1.0      0            0      0     0\n",
      "67176    1.0      0            0      0     0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "       TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0        0.0             0                   0             0            0\n",
      "1        7.0             0                   0             0            0\n",
      "2        1.0             0                   0             0            0\n",
      "3        2.0             0                   0             0            0\n",
      "4        0.0             0                   0             0            0\n",
      "...      ...           ...                 ...           ...          ...\n",
      "67172    0.0             0                   0             0            0\n",
      "67173    0.0             0                   0             0            0\n",
      "67174    0.0             0                   0             0            0\n",
      "67175    1.0             0                   0             0            0\n",
      "67176    1.0             0                   0             0            0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "0.20496598538190156\n",
      "[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 1 0 1 1 1 1 0 0 1 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 1 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0]\n",
      "Runnning scale 8\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0\n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "BINS  IntervalIndex([(3000016, 3000016], (3000016, 3000116], (3000116, 3000216],\n",
      "               (3000216, 3000316], (3000316, 3000416], (3000416, 3000516],\n",
      "               (3000516, 3000616], (3000616, 3000716], (3000716, 3000816],\n",
      "               (3000816, 3000916],\n",
      "               ...\n",
      "               (9716616, 9716716], (9716716, 9716816], (9716816, 9716916],\n",
      "               (9716916, 9717016], (9717016, 9717116], (9717116, 9717216],\n",
      "               (9717216, 9717316], (9717316, 9717416], (9717416, 9717516],\n",
      "               (9717516, 9717593]],\n",
      "              dtype='interval[int64, right]', length=67177)\n",
      "HEAD w BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0   \n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "\n",
      "         bin_offset_0  \n",
      "0  (3000016, 3000116]  \n",
      "1  (3000016, 3000116]  \n",
      "2  (3000016, 3000116]  \n",
      "3  (3000016, 3000116]  \n",
      "4  (3000016, 3000116]  \n",
      "TAIL w BINS\n",
      "       seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  9717555  9717555  18_9717555    1.0    0.0    0.0   \n",
      "699996       19  9717571  9717571  19_9717571    0.0    0.0    0.0   \n",
      "699997       19  9717581  9717581  19_9717581    0.0    0.0    0.0   \n",
      "699998       18  9717582  9717582  18_9717582    0.0    0.0    0.0   \n",
      "699999       18  9717593  9717593  18_9717593    0.0    0.0    0.0   \n",
      "\n",
      "              bin_offset_0  \n",
      "699995  (9717516, 9717593]  \n",
      "699996  (9717516, 9717593]  \n",
      "699997  (9717516, 9717593]  \n",
      "699998  (9717516, 9717593]  \n",
      "699999  (9717516, 9717593]  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:151: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "    TT_S0  starting_nt\n",
      "0     0.0      3000016\n",
      "1     7.0      3000016\n",
      "2     1.0      3000116\n",
      "3     2.0      3000216\n",
      "4     0.0      3000316\n",
      "5     0.0      3000416\n",
      "6     0.0      3000516\n",
      "7     0.0      3000616\n",
      "8     0.0      3000716\n",
      "9     0.0      3000816\n",
      "10    0.0      3000916\n",
      "11    2.0      3001016\n",
      "12    0.0      3001116\n",
      "13    0.0      3001216\n",
      "14    0.0      3001316\n",
      "15    1.0      3001416\n",
      "16    3.0      3001516\n",
      "17    3.0      3001616\n",
      "18    0.0      3001716\n",
      "19    0.0      3001816\n",
      "20    1.0      3001916\n",
      "21    0.0      3002016\n",
      "22    1.0      3002116\n",
      "23   10.0      3002216\n",
      "24    2.0      3002316\n",
      "25   19.0      3002416\n",
      "26    4.0      3002516\n",
      "27    0.0      3002616\n",
      "28    0.0      3002716\n",
      "29    0.0      3002816\n",
      "30    1.0      3002916\n",
      "31    2.0      3003016\n",
      "32    0.0      3003116\n",
      "33    0.0      3003216\n",
      "34    1.0      3003316\n",
      "35    0.0      3003416\n",
      "36    1.0      3003516\n",
      "37    1.0      3003616\n",
      "38    3.0      3003716\n",
      "39    0.0      3003816\n",
      "40    8.0      3003916\n",
      "41    0.0      3004016\n",
      "42    0.0      3004116\n",
      "43    5.0      3004216\n",
      "44    0.0      3004316\n",
      "45    0.0      3004416\n",
      "46    1.0      3004516\n",
      "47    0.0      3004616\n",
      "48    0.0      3004716\n",
      "49   13.0      3004816\n",
      "(67177, 2)\n",
      "TT_S0          float64\n",
      "starting_nt      int64\n",
      "dtype: object\n",
      "sumos ilgis: 67177\n",
      "ranges ilgis: 67177\n",
      "Model input - NO SCALE\n",
      "       TT_S0  genes  transcripts  exons  CDSs\n",
      "0        0.0      0            0      0     0\n",
      "1        7.0      0            0      0     0\n",
      "2        1.0      0            0      0     0\n",
      "3        2.0      0            0      0     0\n",
      "4        0.0      0            0      0     0\n",
      "...      ...    ...          ...    ...   ...\n",
      "67172    0.0      0            0      0     0\n",
      "67173    0.0      0            0      0     0\n",
      "67174    0.0      0            0      0     0\n",
      "67175    1.0      0            0      0     0\n",
      "67176    1.0      0            0      0     0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "       TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0        0.0             0                   0             0            0\n",
      "1        7.0             0                   0             0            0\n",
      "2        1.0             0                   0             0            0\n",
      "3        2.0             0                   0             0            0\n",
      "4        0.0             0                   0             0            0\n",
      "...      ...           ...                 ...           ...          ...\n",
      "67172    0.0             0                   0             0            0\n",
      "67173    0.0             0                   0             0            0\n",
      "67174    0.0             0                   0             0            0\n",
      "67175    1.0             0                   0             0            0\n",
      "67176    1.0             0                   0             0            0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "0.20496598538190156\n",
      "[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 1 0 1 1 1 1 0 0 1 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 1 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0]\n",
      "Runnning scale 16\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0\n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "BINS  IntervalIndex([(3000016, 3000016], (3000016, 3000116], (3000116, 3000216],\n",
      "               (3000216, 3000316], (3000316, 3000416], (3000416, 3000516],\n",
      "               (3000516, 3000616], (3000616, 3000716], (3000716, 3000816],\n",
      "               (3000816, 3000916],\n",
      "               ...\n",
      "               (9716616, 9716716], (9716716, 9716816], (9716816, 9716916],\n",
      "               (9716916, 9717016], (9717016, 9717116], (9717116, 9717216],\n",
      "               (9717216, 9717316], (9717316, 9717416], (9717416, 9717516],\n",
      "               (9717516, 9717593]],\n",
      "              dtype='interval[int64, right]', length=67177)\n",
      "HEAD w BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0   \n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "\n",
      "         bin_offset_0  \n",
      "0  (3000016, 3000116]  \n",
      "1  (3000016, 3000116]  \n",
      "2  (3000016, 3000116]  \n",
      "3  (3000016, 3000116]  \n",
      "4  (3000016, 3000116]  \n",
      "TAIL w BINS\n",
      "       seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  9717555  9717555  18_9717555    1.0    0.0    0.0   \n",
      "699996       19  9717571  9717571  19_9717571    0.0    0.0    0.0   \n",
      "699997       19  9717581  9717581  19_9717581    0.0    0.0    0.0   \n",
      "699998       18  9717582  9717582  18_9717582    0.0    0.0    0.0   \n",
      "699999       18  9717593  9717593  18_9717593    0.0    0.0    0.0   \n",
      "\n",
      "              bin_offset_0  \n",
      "699995  (9717516, 9717593]  \n",
      "699996  (9717516, 9717593]  \n",
      "699997  (9717516, 9717593]  \n",
      "699998  (9717516, 9717593]  \n",
      "699999  (9717516, 9717593]  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:151: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "    TT_S0  starting_nt\n",
      "0     0.0      3000016\n",
      "1     7.0      3000016\n",
      "2     1.0      3000116\n",
      "3     2.0      3000216\n",
      "4     0.0      3000316\n",
      "5     0.0      3000416\n",
      "6     0.0      3000516\n",
      "7     0.0      3000616\n",
      "8     0.0      3000716\n",
      "9     0.0      3000816\n",
      "10    0.0      3000916\n",
      "11    2.0      3001016\n",
      "12    0.0      3001116\n",
      "13    0.0      3001216\n",
      "14    0.0      3001316\n",
      "15    1.0      3001416\n",
      "16    3.0      3001516\n",
      "17    3.0      3001616\n",
      "18    0.0      3001716\n",
      "19    0.0      3001816\n",
      "20    1.0      3001916\n",
      "21    0.0      3002016\n",
      "22    1.0      3002116\n",
      "23   10.0      3002216\n",
      "24    2.0      3002316\n",
      "25   19.0      3002416\n",
      "26    4.0      3002516\n",
      "27    0.0      3002616\n",
      "28    0.0      3002716\n",
      "29    0.0      3002816\n",
      "30    1.0      3002916\n",
      "31    2.0      3003016\n",
      "32    0.0      3003116\n",
      "33    0.0      3003216\n",
      "34    1.0      3003316\n",
      "35    0.0      3003416\n",
      "36    1.0      3003516\n",
      "37    1.0      3003616\n",
      "38    3.0      3003716\n",
      "39    0.0      3003816\n",
      "40    8.0      3003916\n",
      "41    0.0      3004016\n",
      "42    0.0      3004116\n",
      "43    5.0      3004216\n",
      "44    0.0      3004316\n",
      "45    0.0      3004416\n",
      "46    1.0      3004516\n",
      "47    0.0      3004616\n",
      "48    0.0      3004716\n",
      "49   13.0      3004816\n",
      "(67177, 2)\n",
      "TT_S0          float64\n",
      "starting_nt      int64\n",
      "dtype: object\n",
      "sumos ilgis: 67177\n",
      "ranges ilgis: 67177\n",
      "Model input - NO SCALE\n",
      "       TT_S0  genes  transcripts  exons  CDSs\n",
      "0        0.0      0            0      0     0\n",
      "1        7.0      0            0      0     0\n",
      "2        1.0      0            0      0     0\n",
      "3        2.0      0            0      0     0\n",
      "4        0.0      0            0      0     0\n",
      "...      ...    ...          ...    ...   ...\n",
      "67172    0.0      0            0      0     0\n",
      "67173    0.0      0            0      0     0\n",
      "67174    0.0      0            0      0     0\n",
      "67175    1.0      0            0      0     0\n",
      "67176    1.0      0            0      0     0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "       TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0        0.0             0                   0             0            0\n",
      "1        7.0             0                   0             0            0\n",
      "2        1.0             0                   0             0            0\n",
      "3        2.0             0                   0             0            0\n",
      "4        0.0             0                   0             0            0\n",
      "...      ...           ...                 ...           ...          ...\n",
      "67172    0.0             0                   0             0            0\n",
      "67173    0.0             0                   0             0            0\n",
      "67174    0.0             0                   0             0            0\n",
      "67175    1.0             0                   0             0            0\n",
      "67176    1.0             0                   0             0            0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "0.20496598538190156\n",
      "[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 1 0 1 1 1 1 0 0 1 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 1 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0]\n",
      "Runnning scale 32\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0\n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "BINS  IntervalIndex([(3000016, 3000016], (3000016, 3000116], (3000116, 3000216],\n",
      "               (3000216, 3000316], (3000316, 3000416], (3000416, 3000516],\n",
      "               (3000516, 3000616], (3000616, 3000716], (3000716, 3000816],\n",
      "               (3000816, 3000916],\n",
      "               ...\n",
      "               (9716616, 9716716], (9716716, 9716816], (9716816, 9716916],\n",
      "               (9716916, 9717016], (9717016, 9717116], (9717116, 9717216],\n",
      "               (9717216, 9717316], (9717316, 9717416], (9717416, 9717516],\n",
      "               (9717516, 9717593]],\n",
      "              dtype='interval[int64, right]', length=67177)\n",
      "HEAD w BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0   \n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "\n",
      "         bin_offset_0  \n",
      "0  (3000016, 3000116]  \n",
      "1  (3000016, 3000116]  \n",
      "2  (3000016, 3000116]  \n",
      "3  (3000016, 3000116]  \n",
      "4  (3000016, 3000116]  \n",
      "TAIL w BINS\n",
      "       seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  9717555  9717555  18_9717555    1.0    0.0    0.0   \n",
      "699996       19  9717571  9717571  19_9717571    0.0    0.0    0.0   \n",
      "699997       19  9717581  9717581  19_9717581    0.0    0.0    0.0   \n",
      "699998       18  9717582  9717582  18_9717582    0.0    0.0    0.0   \n",
      "699999       18  9717593  9717593  18_9717593    0.0    0.0    0.0   \n",
      "\n",
      "              bin_offset_0  \n",
      "699995  (9717516, 9717593]  \n",
      "699996  (9717516, 9717593]  \n",
      "699997  (9717516, 9717593]  \n",
      "699998  (9717516, 9717593]  \n",
      "699999  (9717516, 9717593]  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:151: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "    TT_S0  starting_nt\n",
      "0     0.0      3000016\n",
      "1     7.0      3000016\n",
      "2     1.0      3000116\n",
      "3     2.0      3000216\n",
      "4     0.0      3000316\n",
      "5     0.0      3000416\n",
      "6     0.0      3000516\n",
      "7     0.0      3000616\n",
      "8     0.0      3000716\n",
      "9     0.0      3000816\n",
      "10    0.0      3000916\n",
      "11    2.0      3001016\n",
      "12    0.0      3001116\n",
      "13    0.0      3001216\n",
      "14    0.0      3001316\n",
      "15    1.0      3001416\n",
      "16    3.0      3001516\n",
      "17    3.0      3001616\n",
      "18    0.0      3001716\n",
      "19    0.0      3001816\n",
      "20    1.0      3001916\n",
      "21    0.0      3002016\n",
      "22    1.0      3002116\n",
      "23   10.0      3002216\n",
      "24    2.0      3002316\n",
      "25   19.0      3002416\n",
      "26    4.0      3002516\n",
      "27    0.0      3002616\n",
      "28    0.0      3002716\n",
      "29    0.0      3002816\n",
      "30    1.0      3002916\n",
      "31    2.0      3003016\n",
      "32    0.0      3003116\n",
      "33    0.0      3003216\n",
      "34    1.0      3003316\n",
      "35    0.0      3003416\n",
      "36    1.0      3003516\n",
      "37    1.0      3003616\n",
      "38    3.0      3003716\n",
      "39    0.0      3003816\n",
      "40    8.0      3003916\n",
      "41    0.0      3004016\n",
      "42    0.0      3004116\n",
      "43    5.0      3004216\n",
      "44    0.0      3004316\n",
      "45    0.0      3004416\n",
      "46    1.0      3004516\n",
      "47    0.0      3004616\n",
      "48    0.0      3004716\n",
      "49   13.0      3004816\n",
      "(67177, 2)\n",
      "TT_S0          float64\n",
      "starting_nt      int64\n",
      "dtype: object\n",
      "sumos ilgis: 67177\n",
      "ranges ilgis: 67177\n",
      "Model input - NO SCALE\n",
      "       TT_S0  genes  transcripts  exons  CDSs\n",
      "0        0.0      0            0      0     0\n",
      "1        7.0      0            0      0     0\n",
      "2        1.0      0            0      0     0\n",
      "3        2.0      0            0      0     0\n",
      "4        0.0      0            0      0     0\n",
      "...      ...    ...          ...    ...   ...\n",
      "67172    0.0      0            0      0     0\n",
      "67173    0.0      0            0      0     0\n",
      "67174    0.0      0            0      0     0\n",
      "67175    1.0      0            0      0     0\n",
      "67176    1.0      0            0      0     0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "       TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0        0.0             0                   0             0            0\n",
      "1        7.0             0                   0             0            0\n",
      "2        1.0             0                   0             0            0\n",
      "3        2.0             0                   0             0            0\n",
      "4        0.0             0                   0             0            0\n",
      "...      ...           ...                 ...           ...          ...\n",
      "67172    0.0             0                   0             0            0\n",
      "67173    0.0             0                   0             0            0\n",
      "67174    0.0             0                   0             0            0\n",
      "67175    1.0             0                   0             0            0\n",
      "67176    1.0             0                   0             0            0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "0.17092159518882952\n",
      "[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 1 0 1 1 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 1 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0]\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "for i in range(6):\n",
    "    scale = 2 ** i\n",
    "    print(\"Runnning scale\", scale)\n",
    "    evaluation = evaluator.evaluate(dataframe = chromatin_algos.algo5d(scale), model_name=f\"k-means 5 features scale={scale}\")\n",
    "    result = pd.concat([result, evaluation], axis=1)\n",
    "\n",
    "result.to_csv('../outputs/results1.csv', sep = '\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ca1280ed",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                         Length                      Length  \\\n",
      "count                    7301.0                      6456.0   \n",
      "mean                 188.590604                  177.850062   \n",
      "std                  174.998864                  157.257952   \n",
      "min                       100.0                       100.0   \n",
      "25%                       100.0                       100.0   \n",
      "50%                       100.0                       100.0   \n",
      "75%                       200.0                       200.0   \n",
      "max                      2400.0                      2000.0   \n",
      "ATACseq_overlap        0.697949                    0.668276   \n",
      "DNAseq_overlap         0.595329                    0.553813   \n",
      "genes_overlap          0.544833                    0.550809   \n",
      "percentage_open        0.141691                    0.118157   \n",
      "percentage_open_adj    0.204969                    0.170924   \n",
      "model_name                       k-means 5 features scale=1   \n",
      "min_open_signal            17.0                        19.0   \n",
      "max_open_signal           393.0                       393.0   \n",
      "min_closed_signal           0.0                         0.0   \n",
      "max_closed_signal          16.0                        18.0   \n",
      "\n",
      "                                         Length                      Length  \\\n",
      "count                                    7301.0                      7301.0   \n",
      "mean                                 188.590604                  188.590604   \n",
      "std                                  174.998864                  174.998864   \n",
      "min                                       100.0                       100.0   \n",
      "25%                                       100.0                       100.0   \n",
      "50%                                       100.0                       100.0   \n",
      "75%                                       200.0                       200.0   \n",
      "max                                      2400.0                      2400.0   \n",
      "ATACseq_overlap                        0.697949                    0.697949   \n",
      "DNAseq_overlap                         0.595329                    0.595329   \n",
      "genes_overlap                          0.544833                    0.544833   \n",
      "percentage_open                        0.141691                    0.141691   \n",
      "percentage_open_adj                    0.204969                    0.204969   \n",
      "model_name           k-means 5 features scale=2  k-means 5 features scale=4   \n",
      "min_open_signal                            17.0                        17.0   \n",
      "max_open_signal                           393.0                       393.0   \n",
      "min_closed_signal                           0.0                         0.0   \n",
      "max_closed_signal                          16.0                        16.0   \n",
      "\n",
      "                                         Length                       Length  \\\n",
      "count                                    7301.0                       7301.0   \n",
      "mean                                 188.590604                   188.590604   \n",
      "std                                  174.998864                   174.998864   \n",
      "min                                       100.0                        100.0   \n",
      "25%                                       100.0                        100.0   \n",
      "50%                                       100.0                        100.0   \n",
      "75%                                       200.0                        200.0   \n",
      "max                                      2400.0                       2400.0   \n",
      "ATACseq_overlap                        0.697949                     0.697949   \n",
      "DNAseq_overlap                         0.595329                     0.595329   \n",
      "genes_overlap                          0.544833                     0.544833   \n",
      "percentage_open                        0.141691                     0.141691   \n",
      "percentage_open_adj                    0.204969                     0.204969   \n",
      "model_name           k-means 5 features scale=8  k-means 5 features scale=16   \n",
      "min_open_signal                            17.0                         17.0   \n",
      "max_open_signal                           393.0                        393.0   \n",
      "min_closed_signal                           0.0                          0.0   \n",
      "max_closed_signal                          16.0                         16.0   \n",
      "\n",
      "                                          Length  \n",
      "count                                     6456.0  \n",
      "mean                                  177.850062  \n",
      "std                                   157.257952  \n",
      "min                                        100.0  \n",
      "25%                                        100.0  \n",
      "50%                                        100.0  \n",
      "75%                                        200.0  \n",
      "max                                       2000.0  \n",
      "ATACseq_overlap                         0.668276  \n",
      "DNAseq_overlap                          0.553813  \n",
      "genes_overlap                           0.550809  \n",
      "percentage_open                         0.118157  \n",
      "percentage_open_adj                     0.170924  \n",
      "model_name           k-means 5 features scale=32  \n",
      "min_open_signal                             19.0  \n",
      "max_open_signal                            393.0  \n",
      "min_closed_signal                            0.0  \n",
      "max_closed_signal                           18.0  \n"
     ]
    }
   ],
   "source": [
    "print(result)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
