{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "d001e5b1",
   "metadata": {},
   "outputs": [],
   "source": [
    "import chromatin_algos\n",
    "import evaluator\n",
    "import generate_consensus\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "8f6a7196",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "       Chromosome     Start       End  width strand    study  Length\n",
      "0               1   3062536   3063132    597      *  ATACseq     596\n",
      "1               1   3482735   3483063    329      *  ATACseq     328\n",
      "2               1   3549538   3549654    117      *  ATACseq     116\n",
      "3               1   3649136   3649335    200      *  ATACseq     199\n",
      "4               1   3670656   3672554   1899      *  ATACseq    1898\n",
      "...           ...       ...       ...    ...    ...      ...     ...\n",
      "49305          19  61257841  61258165    325      *  ATACseq     324\n",
      "49306          19  61265420  61265595    176      *  ATACseq     175\n",
      "49307          19  61266516  61266619    104      *  ATACseq     103\n",
      "49308          19  61313296  61313814    519      *  ATACseq     518\n",
      "49309          19  61320979  61321177    199      *  ATACseq     198\n",
      "\n",
      "[49310 rows x 7 columns]\n"
     ]
    }
   ],
   "source": [
    "df = pd.read_csv(\"../outputs/ATAC.tsv\", sep='\\t')\n",
    "df['Length'] = df['End'] - df['Start']\n",
    "print(df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "1b4f8a2c",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:95: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "sumos ilgis: 152618\n",
      "ranges ilgis: 152618\n",
      "0.03080894783053113\n",
      "[6 2 0 0 6 6 6 6 6 6 6 0 6 6 6 0 0 0 6 6 0 6 0 7 0 1 8 6 6 6 0 0 6 6 0 6 0\n",
      " 0 0 6 2 6 6 8 6 6 0 6 6 7 6 4 1 3 6 1 7 9 9 5 5 6 1 1 8 6 6 2 6 0 8 0 6 8\n",
      " 6 6 0 6 6 6 0 2 0 6 0 8 0 8 8 6 6 7 1 0 0 6 0 6 0 7 7 7 4 4 1 2 8 7 4 8 7\n",
      " 6 1 0 6 6 0 6 6 6 6 0 8 0 0 1 0 1 1 7 1 0 2 2 8 0 8 2 0 6 6 6 6 6 0 0 6 7\n",
      " 2 0 7 0 6 0 8 6 7 0 0 0 1 6 7 0 6 7 7 0 6 6 6 6 0 6 0 6 6 6 6 0 8 2 0 0 6\n",
      " 6 0 6 6 6 6 6 0 6 6 6 6 6 6 6 6 6 6 6 6 6 6 7 1 0 0 6 6 0 6 6 6 0 0 6 6 6\n",
      " 6 6 6 6 6 6 6 6 6 6 6 6 6 6 0 6 6 6 0 6 6 6 6 6 6 6 6 6 6 6 0 6 6 6 0 6 0\n",
      " 6 6 6 6 6 6 0 6 6 0 0 0 6 0 0 0 0 6 0 6 6 0 0 6 6 0 6 6 6 6 6 6 6 0 6 6 6\n",
      " 8 6 6 6 0 0 0 6 0 6 6 6 6 8 0 8 2 0 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6\n",
      " 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6\n",
      " 6 6 6 6 6 6 6 6 6 6 6 8 6 0 6 7 0 0 0 0 6 6 6 6 0 7 1 0 7 1 2 2 0 0 0 8 2\n",
      " 0 6 8 0 0 6 7 8 1 0 7 8 6 0 6 6 0 6 8 6 0 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6\n",
      " 0 6 6 0 8 6 6 6 6 6 0 6 6 0 6 0 0 0 0 2 0 6 0 8 8 8 6 8 8 8 0 2 0 0 0 0 6\n",
      " 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6\n",
      " 6 1 8 6 6 6 8 6 0 2 6 0 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6\n",
      " 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6\n",
      " 6 6 6 6 0 6 0 8 6 6 6 6 6 0 6 6 6 6 0 6 0 0 0 0 6 2 6 6 6 6 0 8 6 6 6 6 8\n",
      " 6 6 8 6 6 6 6 6 0 0 2 6 0 0 2 0 6 6 6 6 0 8 0 0 6 6 6 6 0 8 8 6 0 8 0 6 2\n",
      " 6 2 6 6 6 6 0 0 6 6 6 6 6 6 6 6 6 6 8 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6\n",
      " 6 6 6 6 6 6 6 6 6 6 6 6 6 0 0 2 8 0 0 6 0 6 0 6 0 6 6 6 6 0 6 6 6 6 0 6 6\n",
      " 6 6 0 6 6 0 6 6 0 6 6 6 6 6 6 0 6 6 6 6 0 6 6 6 6 6 6 0 6 6 6 6 0 0 6 6 6\n",
      " 6 6 6 6 6 6 6 0 6 6 6 6 6 0 0 6 6 6 0 6 0 6 6 8 6 6 6 0 2 8 7 6 0 6 0 0 6\n",
      " 6 6 6 6 0 6 6 6 6 6 6 6 6 6 6 6 6 6 0 0 0 0 0 6 0 0 6 0 6 0 6 0 0 6 6 6 0\n",
      " 6 6 6 6 6 6 6 6 6 6 6 0 6 6 0 6 0 6 0 8 6 6 6 6 6 0 8 6 6 6 6 6 6 6 6 6 6\n",
      " 6 6 0 6 6 6 6 6 0 6 6 6 6 6 6 6 6 6 6 6 0 0 6 6 0 0 0 8 6 6 6 6 0 6 6 6 0\n",
      " 8 0 8 8 0 0 6 6 6 6 6 6 6 6 6 0 6 6 6 0 0 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6\n",
      " 6 6 6 6 6 6 6 6 8 6 6 6 6 6 6 6 0 6 6 6 6 6 6 6 0 6 6 0 6 8 0 6 6 0 6 6 6\n",
      " 6]\n",
      "threshold density 1.95819\n",
      "cluster signal 0.0\n",
      "cluster target count 227797\n",
      "!!cluster density 0.0\n",
      "cluster signal 77249.0\n",
      "cluster target count 64504\n",
      "!!cluster density 1.1975846459134318\n",
      "cluster signal 72096.0\n",
      "cluster target count 186822\n",
      "!!cluster density 0.3859074413077689\n",
      "cluster signal 75840.0\n",
      "cluster target count 49807\n",
      "!!cluster density 1.5226775352862048\n",
      "cluster signal 78361.0\n",
      "cluster target count 39660\n",
      "!!cluster density 1.9758194654563792\n",
      "cluster signal 93575.0\n",
      "cluster target count 111040\n",
      "!!cluster density 0.8427143371757925\n",
      "cluster signal 43239.0\n",
      "cluster target count 16286\n",
      "!!cluster density 2.654979737197593\n",
      "cluster signal 13559.0\n",
      "cluster target count 3706\n",
      "!!cluster density 3.6586616297895307\n",
      "cluster signal 2328.0\n",
      "cluster target count 367\n",
      "!!cluster density 6.343324250681199\n",
      "cluster signal 664.0\n",
      "cluster target count 11\n",
      "!!cluster density 60.36363636363637\n",
      "\n",
      "Labels mapping\n",
      " {6: 0, 2: 0, 0: 0, 7: 0, 1: 1, 8: 0, 4: 1, 3: 1, 9: 1, 5: 1}\n",
      "DEBUG: 146000 6618 \n",
      "\n",
      " +----------------------+-------------+----------------+-------+\n",
      "| bin_offset_0         | TT_S0       | target_count   | +9    |\n",
      "| (category)           | (float64)   | (int64)        | ...   |\n",
      "|----------------------+-------------+----------------+-------|\n",
      "| (3000018, 3000018]   | 0.0         | 0              | ...   |\n",
      "| (3000018, 3000118]   | 7.0         | 11             | ...   |\n",
      "| (3000118, 3000218]   | 1.0         | 4              | ...   |\n",
      "| (3000218, 3000318]   | 2.0         | 4              | ...   |\n",
      "| ...                  | ...         | ...            | ...   |\n",
      "| (18261318, 18261418] | 3.0         | 15             | ...   |\n",
      "| (18261418, 18261518] | 4.0         | 4              | ...   |\n",
      "| (18261518, 18261618] | 0.0         | 3              | ...   |\n",
      "| (18261618, 18261710] | 4.0         | 3              | ...   |\n",
      "+----------------------+-------------+----------------+-------+\n",
      "Unstranded PyRanges object has 146,000 rows and 12 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "9 hidden columns: starting_nt, Chromosome, Start, End, genes, transcripts, ... (+ 3 more.)\n"
     ]
    }
   ],
   "source": [
    "out = chromatin_algos.algo5d_aggregated(16, 10, 3)\n",
    "result = evaluator.evaluate(merged_ranges = out, model_name=\"test run\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "ca449fd3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "count                                                                5110.0\n",
      "mean                                                             129.510763\n",
      "std                                                               86.322143\n",
      "min                                                                   100.0\n",
      "25%                                                                   100.0\n",
      "50%                                                                   100.0\n",
      "75%                                                                   100.0\n",
      "max                                                                  1400.0\n",
      "mode                                                                  100.0\n",
      "ATACseq_overlap                                                    0.618208\n",
      "ATACseq_ranges_overlap                                             0.068716\n",
      "ATACseq_Jaccard                                                    0.065918\n",
      "DNAseq_overlap                                                     0.522526\n",
      "DNAseq_ranges_overlap                                              0.095496\n",
      "DNAseq_Jaccard                                                     0.087831\n",
      "genes_overlap                                                      0.578253\n",
      "percentage_open                                                     0.03624\n",
      "percentage_open_adj                                                0.043363\n",
      "model_name                                                         test run\n",
      "boxplot_json              {\"mean\":129.5107632094,\"iqr\":0.0,\"cilo\":100.0,...\n",
      "Name: Length, dtype: object\n"
     ]
    }
   ],
   "source": [
    "print(result)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "81f5a277",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:39: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "sumos ilgis: 152618\n",
      "ranges ilgis: 152618\n",
      "0.01940793353339711\n",
      "[0 0 0 0 0 0 6 0 0 0 6 0 0 0 0 0 6 6 0 0 0 6 6 0 0 6 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 6 0 0 6 0 0 0 0 0 0 0 2 6 6 0 7 6 8 8 3 4 0 0 6 0 0 0 0 0 0 0 6 0 0\n",
      " 0 6 0 0 0 0 6 2 2 0 6 0 0 6 0 0 0 1 6 0 0 2 6 0 0 6 6 7 7 2 2 0 6 6 2 0 0\n",
      " 0 6 6 0 0 0 0 0 0 0 0 2 6 0 2 0 6 0 6 7 0 0 0 2 0 6 2 2 0 6 0 0 0 6 6 0 6\n",
      " 0 0 6 6 0 6 6 0 0 6 0 0 0 0 2 0 2 6 0 0 0 0 0 0 0 0 0 0 0 0 0 2 6 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 6 0 6 6 6 0 0 0 6 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 2 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 6 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 6\n",
      " 0 0 0 0 6 0 2 0 0 0 0 0 0 0 0 7 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 6 6 0 6 6 2 0 0 0 0 0 0 0 0 0 7 7 6 2 0 2 0 0 0 6 2 6\n",
      " 2 2 6 6 0 0 6 6 7 7 2 6 0 0 0 0 0 0 0 0 6 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 2 0 0 0 6 6 2 0 0 0 0 0 0 6 0 6 0 0 0 2 6 0 6 0 6 2 6 2 6 6 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 7 7 6 7 0 0 0 0 6 6 0 0 0 0 0 6 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 6 0 0 6 2 0 0 0 0 2 0 0 0 0 0 0 0 0 0 6 0 6 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 2 6 0 0 0 6 0 6 0 2 0 0 0 0 6 0 6 0 0 2 0 0 2\n",
      " 2 7 2 0 0 0 0 0 0 0 0 0 0 0 0 0 0 6 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 6 0 0 0 6 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 6 0 0 0 0 0 0 6 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 6 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 2 0 0 0 0 0 0 0 0 0 0 6 0 0 0 6 0 6 6 0 0 0 0\n",
      " 0 6 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 2 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 2 0 0 0 0 0 0 0 0 0\n",
      " 0 2 0 6 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 2 0 0 0 0 0 0 0\n",
      " 0]\n",
      "threshold density 2.044504285714286\n",
      "cluster signal 13066.0\n",
      "cluster target count 325909\n",
      "!!cluster density 0.04009094563206294\n",
      "cluster signal 78567.0\n",
      "cluster target count 141959\n",
      "!!cluster density 0.5534485309138554\n",
      "cluster signal 129740.0\n",
      "cluster target count 120167\n",
      "!!cluster density 1.079664134080072\n",
      "cluster signal 109115.0\n",
      "cluster target count 64658\n",
      "!!cluster density 1.687571530205079\n",
      "cluster signal 39267.0\n",
      "cluster target count 12796\n",
      "!!cluster density 3.0686933416692717\n",
      "cluster signal 15296.0\n",
      "cluster target count 2836\n",
      "!!cluster density 5.393511988716502\n",
      "cluster signal 23653.0\n",
      "cluster target count 5747\n",
      "!!cluster density 4.115712545676005\n",
      "cluster signal 58463.0\n",
      "cluster target count 24742\n",
      "!!cluster density 2.3629051814727995\n",
      "cluster signal 6442.0\n",
      "cluster target count 862\n",
      "!!cluster density 7.473317865429235\n",
      "cluster signal 3442.0\n",
      "cluster target count 324\n",
      "!!cluster density 10.623456790123457\n",
      "\n",
      "Labels mapping\n",
      " {0: 0, 6: 0, 2: 0, 7: 0, 8: 1, 3: 1, 4: 1, 1: 1, 9: 1, 5: 1}\n",
      "count                                                                3668.0\n",
      "mean                                                             136.968375\n",
      "std                                                              120.138871\n",
      "min                                                                   100.0\n",
      "25%                                                                   100.0\n",
      "50%                                                                   100.0\n",
      "75%                                                                   100.0\n",
      "max                                                                  2200.0\n",
      "mode                                                                  100.0\n",
      "ATACseq_overlap                                                    0.639714\n",
      "ATACseq_ranges_overlap                                             0.093666\n",
      "ATACseq_Jaccard                                                    0.088973\n",
      "DNAseq_overlap                                                     0.455324\n",
      "DNAseq_ranges_overlap                                              0.109616\n",
      "DNAseq_Jaccard                                                     0.096909\n",
      "genes_overlap                                                      0.594112\n",
      "percentage_open                                                    0.027511\n",
      "percentage_open_adj                                                0.032919\n",
      "model_name                                                         test run\n",
      "boxplot_json              {\"mean\":136.9683751363,\"iqr\":0.0,\"cilo\":100.0,...\n",
      "Name: Length, dtype: object\n"
     ]
    }
   ],
   "source": [
    "out = chromatin_algos.algo5d_aggregated(16, 10, 3, 0, 'TT_S2')\n",
    "result = evaluator.evaluate(merged_ranges = out, model_name=\"test run\")\n",
    "print(result)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "id": "0d5d91c9",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[{'mean': 338.16451024133033, 'iqr': 289.0, 'cilo': 242.95671007358655, 'cihi': 247.04328992641345, 'whishi': 879, 'whislo': 100, 'fliers': array([1898, 1906,  882, ..., 1170, 1102, 1045]), 'q1': 157.0, 'med': 245.0, 'q3': 446.0}]\n",
      "{\"mean\":338.1645102413,\"iqr\":289.0,\"cilo\":242.9567100736,\"cihi\":247.0432899264,\"whishi\":879,\"whislo\":100,\"fliers\":[1898,1906,882,1168,1047,928,1054,1203,892,1125,1231,1183,939,1059,902,1421,976,1384,1071,881,1022,935,1827,903,960,1714,903,1115,954,990,1253,1032,1031,923,974,905,936,1084,1776,1213,985,1322,1141,924,1355,1021,968,924,1003,1286,999,977,1588,904,930,1012,938,1142,900,1074,1168,1092,899,961,892,1170,1138,888,1240,1038,912,1159,901,979,886,1087,1180,1069,1015,943,1266,935,1061,1130,1278,1267,937,1258,943,1051,1493,972,1144,1030,998,1121,943,1412,885,959,1754,914,971,1167,989,1103,1014,1119,909,961,1169,921,1423,1154,912,1236,887,920,1227,890,1026,885,1102,1140,1060,927,1114,1035,1110,1135,1311,882,1165,904,1018,936,1025,1031,1126,1002,1111,1096,1114,958,1104,1074,910,892,944,975,1245,1139,920,947,1043,1068,951,1009,1051,905,1046,977,991,964,1022,1581,998,894,939,1227,1504,978,911,903,884,1165,979,950,904,975,1302,983,1109,1006,1135,982,1380,897,1108,920,1093,937,1231,1147,1034,1304,918,1326,1051,936,1025,999,1001,916,1053,1008,957,1314,1001,1601,989,1368,883,1356,1135,946,1042,892,1188,1010,922,1501,970,898,916,909,916,918,943,969,1151,1095,899,920,886,1232,942,1194,1022,976,1513,1008,1108,961,1172,1063,1057,1175,1110,1363,1182,1008,900,1013,1099,1012,1316,1293,1101,1003,1072,1050,1270,972,1020,1592,1288,1132,1258,1109,944,1083,1026,992,975,993,960,1011,917,1036,880,978,1087,1046,954,1202,962,1445,1511,1102,1187,996,1418,892,961,1267,1035,1500,910,894,1509,960,961,1080,1035,905,892,900,1006,982,1209,986,1132,902,966,940,971,1052,908,886,983,1138,989,1038,1055,1020,969,1845,891,1246,1060,893,940,926,1487,1313,903,1097,892,1412,1172,985,929,970,1079,1003,918,927,1008,1239,1032,960,911,900,909,1196,982,1061,1088,896,1044,904,968,1373,1001,900,1335,1323,1017,991,1305,928,1056,1104,1052,902,1036,1269,1157,1062,894,1039,1196,1118,1501,916,1403,993,1043,917,1410,1139,1192,930,914,1845,968,1230,1794,1397,974,968,966,896,996,880,1219,1020,1379,1458,1120,1332,1066,1153,999,880,970,987,1251,1118,908,880,1006,1053,915,944,1040,957,1042,1485,1056,919,1186,924,893,1021,972,1099,948,1069,910,918,954,900,951,1247,1139,932,1233,891,939,945,999,1169,1369,1372,1137,967,973,1215,1207,1247,891,1024,1037,1003,1050,1168,937,1472,1174,1464,1110,881,1256,1082,1217,988,945,1566,1047,1121,971,1012,1426,1113,973,1119,959,1431,1006,1245,909,903,1388,880,966,968,1034,1498,1050,1289,1303,1227,992,1326,1255,1399,1272,1029,900,928,880,1235,925,1220,1015,887,977,1161,900,1232,983,961,1081,1192,933,1248,1640,1020,957,1064,940,1341,1917,942,1230,1186,1089,1133,1154,994,940,1032,971,1334,945,1009,963,1018,955,1184,938,992,1364,926,1174,889,1479,1088,942,926,1340,1387,895,1193,946,929,1208,963,1158,1058,981,1305,1067,991,988,973,927,880,905,1244,1072,989,1332,969,880,891,1253,972,1173,964,984,1381,1155,895,1008,1145,1010,938,1201,918,997,964,1368,936,1060,942,1003,1040,1085,1085,926,1015,956,1022,919,898,931,958,886,1000,943,1212,902,1182,971,1517,1024,1077,910,1090,938,945,1230,969,928,1261,892,1149,1146,1115,1153,1163,1012,1005,1074,901,966,1142,1125,882,1278,987,1093,966,1054,1039,1411,1751,1121,1171,1377,899,1220,1143,910,901,1153,918,1452,1066,1011,1320,918,903,1795,1104,1111,889,952,883,1032,934,1304,948,1172,971,1021,1012,891,897,1286,1062,946,911,944,895,990,1607,1032,1023,1042,975,954,987,1223,1104,930,952,885,1208,958,890,941,920,1007,906,1259,963,898,1069,1058,922,1043,1006,885,965,978,914,891,1155,946,1246,1452,930,1166,1031,1218,1376,904,909,1002,1001,1179,995,909,947,1068,1584,1231,1329,1327,1004,897,969,1136,1133,1222,948,936,917,1002,957,1227,1307,1067,1360,1261,894,1164,975,880,885,1192,887,926,1236,1149,1315,969,1030,982,927,1295,1223,1142,1041,905,1038,939,887,1117,963,989,886,1140,1077,989,954,1075,1087,1031,1319,1281,924,1264,1007,1273,1043,917,917,941,1115,1280,1022,1065,1086,952,970,1147,962,1089,896,1140,1181,1009,1065,1036,1025,909,934,973,966,1083,1004,914,966,1445,883,946,1431,1097,1025,1453,906,889,1083,883,924,916,899,1095,918,892,1022,1316,910,1042,1189,882,953,979,1097,1282,901,993,937,1100,884,902,1111,1202,931,982,881,973,1074,1012,996,1094,965,1040,951,1108,1018,1165,1125,1562,971,890,925,933,901,1030,956,1028,965,1159,1318,1011,1044,1194,1460,1040,887,1071,1040,899,925,1368,1408,915,952,910,925,1209,942,887,912,1160,1063,1455,1116,1293,1190,915,888,1027,886,982,1082,971,1593,1478,1074,958,1016,955,1172,1150,917,948,940,901,1029,997,899,1022,977,921,905,880,914,929,982,903,967,1039,927,1085,1238,1102,1076,1396,1031,931,1665,1056,1209,1236,1011,1140,916,1039,981,1074,1018,912,951,934,1172,945,954,904,1390,1051,1108,924,939,1070,1094,1342,1598,1376,1164,957,1042,1738,894,948,938,1002,1127,1235,1209,1126,1052,955,989,1060,1532,1140,984,925,988,985,1157,892,1179,949,1636,1070,1146,1018,1074,1042,1076,1686,1311,1453,881,1049,916,1082,1121,990,1099,966,1189,1187,978,919,950,944,954,987,1020,1030,1019,924,1040,902,1170,1338,925,953,1239,1011,889,1353,881,944,908,1349,1039,935,894,943,888,1179,954,1232,1345,1364,1309,1007,1276,959,887,952,973,999,1121,1102,926,988,976,1098,919,1376,888,938,1016,1006,926,1036,1106,905,1104,988,1029,1535,1041,1194,1117,1122,972,1176,946,1215,930,1242,1142,910,1014,965,889,887,921,1169,925,991,882,924,1031,880,1048,1064,979,968,1141,1026,881,910,1021,953,932,999,1002,1259,888,933,973,1168,951,894,880,1546,943,989,980,995,1217,1030,993,1148,1044,961,1011,968,944,906,1821,979,947,1677,945,916,922,1194,990,905,1345,1141,893,890,924,924,1301,981,971,1134,1174,896,1025,1044,932,1233,937,927,1536,1123,1190,911,964,1020,1000,1191,907,1005,896,997,1057,947,990,978,1025,1258,998,940,1182,1222,907,890,909,962,1160,912,898,910,916,892,987,941,1198,1229,1307,911,911,1362,1009,1073,893,1753,959,1310,1008,1395,948,1313,1402,1007,1078,1072,1207,1178,1810,2225,1134,911,940,1080,1046,1016,993,937,881,1219,1090,904,1218,992,932,1352,888,1153,1046,1080,1083,895,890,1099,1009,975,1241,1090,1210,1133,1038,1427,905,944,1083,1034,973,894,1096,1252,1101,985,1197,1118,1087,984,1218,1063,1081,972,1181,1574,970,926,1367,977,1084,1059,1218,1299,985,952,1354,910,967,948,1220,980,942,889,966,937,1212,904,1243,1715,1064,1038,943,1414,999,952,1233,1043,1037,1090,1170,931,1004,894,934,993,1197,902,1013,888,1041,889,985,1188,1001,1089,1076,1208,1024,1072,1286,1496,2006,1830,1143,889,1087,1562,946,902,985,1037,1254,977,1043,885,922,997,948,949,937,995,983,1399,1186,1159,1105,1201,1069,986,945,910,939,947,920,1087,1015,892,982,924,1732,1037,886,1195,1097,1024,1091,903,1153,1036,1112,1205,1081,1598,887,1267,1327,941,1192,995,967,999,1150,1108,1216,1143,939,908,1133,1063,1035,1301,909,882,1022,1020,1094,895,1342,1161,1507,989,1537,1559,920,1606,966,975,970,978,1024,1034,912,1554,1105,922,892,1037,1115,1170,1028,900,1249,948,1140,1039,1283,1098,961,1278,964,1032,977,975,1139,958,890,903,970,882,926,1089,987,905,1045,1250,921,1074,1064,1022,935,1114,957,1017,920,1695,944,987,1829,1435,975,944,983,1000,1031,896,895,976,1318,986,1011,967,1322,1017,1197,1071,988,931,884,1179,882,1076,1165,970,921,955,1068,1099,968,919,1117,1204,1077,1522,1224,1148,890,1021,1033,1063,990,903,1281,921,957,1044,993,1058,1311,1046,891,891,916,1115,1105,1523,906,1332,1133,942,1237,1096,922,1151,1063,1365,993,888,953,1357,981,933,895,929,904,1071,1005,1010,941,1009,987,942,996,1399,907,1315,1004,1286,1130,915,1120,1024,1233,1205,1098,1230,920,1770,1259,1254,1247,1344,1022,1056,926,880,929,909,1206,1081,1457,1026,1467,1087,1347,893,896,971,882,1037,1975,1130,1365,977,1032,1066,1205,935,978,1078,1193,969,958,1086,1061,1467,1352,1275,1039,1021,996,1230,1225,1151,983,884,986,1132,981,1048,1031,956,930,924,887,951,978,952,1070,939,940,1169,1272,1144,901,1289,1082,1202,1420,1200,913,1011,885,1054,1131,907,910,955,1136,1170,1067,1093,944,1101,1341,1238,1073,1566,1021,1325,1332,985,1159,895,884,912,1172,1272,898,963,923,977,1166,1156,899,925,1235,1042,999,1210,1484,948,1000,1007,912,913,886,1039,1138,1526,1151,894,1066,1073,924,928,1214,933,932,912,937,1094,930,1052,914,930,1111,1005,1005,904,1036,892,952,886,1197,1173,1303,1165,1342,917,946,894,1192,1007,1019,1043,1667,964,1312,993,1037,1047,1043,941,927,1124,1113,1107,909,886,1029,1446,1083,939,1263,1309,1313,1181,1404,917,949,1039,1137,1066,953,1048,1024,1092,1146,1352,1378,1568,1098,1063,1027,1201,1188,1084,958,1406,998,1237,918,1424,1147,1323,1109,1069,1015,893,1068,922,892,1134,1062,1129,1263,910,1645,1144,1158,1526,956,1222,1227,916,903,972,908,906,1087,994,1118,934,985,924,930,908,1031,968,1013,959,1268,1177,967,1274,1137,1448,898,954,1480,890,1415,978,1602,959,1070,975,1349,1117,948,999,1162,904,910,1574,956,1071,960,1364,1484,1088,1118,1471,1234,1052,1227,1358,976,1112,954,1038,903,942,903,1066,1223,945,1215,1034,887,1069,962,1048,965,891,907,1268,942,1019,947,1030,980,931,1263,1461,952,1146,989,1114,1001,1022,978,891,896,1224,1174,1029,1074,1105,1073,959,966,1046,1071,977,1814,1380,931,1079,1096,1197,895,917,1056,888,1067,990,1120,964,941,1097,1132,935,966,1036,1166,924,905,1287,958,982,971,1090,1099,1070,979,1178,1075,1077,947,957,1041,1036,897,910,891,997,938,1135,1029,1223,902,912,1060,918,1136,911,1467,961,887,904,1126,1018,1415,1108,930,926,1434,1025,974,1452,1124,1053,935,1237,1091,998,1077,988,893,951,1463,904,1416,1255,1151,1413,929,1477,979,1015,999,1097,1163,1616,1224,1165,1096,990,1096,1077,916,918,1240,1205,913,945,1100,897,887,1109,937,1201,886,1332,898,997,985,973,1197,1189,915,1202,946,987,956,1050,898,881,1211,925,882,916,934,924,929,1016,1422,970,934,968,1805,1396,989,885,1258,914,959,1183,958,1004,1020,1255,1073,1060,937,1052,1038,939,992,939,973,1256,1589,1165,981,1263,1891,1160,910,885,1020,936,1042,1092,881,930,1090,884,920,1346,918,1146,1196,1248,896,910,1136,1136,1022,1192,1026,1258,905,1477,1086,903,901,952,929,947,990,1022,1128,1179,906,1450,991,1072,1351,886,1041,1032,1133,986,911,1009,998,941,1273,888,991,1004,1013,975,1337,920,902,1005,999,1097,1388,1297,1274,1010,904,1353,1361,890,1498,1172,1181,1112,975,988,946,1151,924,1018,1040,1293,1005,1065,1025,889,1099,888,883,1119,1122,1085,1123,1252,989,900,957,1048,974,1194,934,931,905,951,950,1052,1079,926,1055,939,1084,1029,922,1103,941,916,1196,935,925,896,881,919,1527,971,991,1009,894,956,995,1174,1238,897,1191,921,1187,964,1316,1150,1029,889,927,1050,1077,1092,953,966,1149,1723,1170,1102,1045],\"q1\":157.0,\"med\":245.0,\"q3\":446.0}\n"
     ]
    }
   ],
   "source": [
    "import matplotlib\n",
    "from matplotlib import cbook\n",
    "\n",
    "stats = cbook.boxplot_stats(df['Length'])\n",
    "print(stats)\n",
    "stats_df = pd.DataFrame(stats)\n",
    "testukas = stats_df.loc[0].to_json()\n",
    "print(testukas)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "id": "b8f9de17",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{\"mean\":338.1645102413,\"iqr\":289.0,\"cilo\":242.9567100736,\"cihi\":247.0432899264,\"whishi\":879,\"whislo\":100,\"fliers\":[1898,1906,882,1168,1047,928,1054,1203,892,1125,1231,1183,939,1059,902,1421,976,1384,1071,881,1022,935,1827,903,960,1714,903,1115,954,990,1253,1032,1031,923,974,905,936,1084,1776,1213,985,1322,1141,924,1355,1021,968,924,1003,1286,999,977,1588,904,930,1012,938,1142,900,1074,1168,1092,899,961,892,1170,1138,888,1240,1038,912,1159,901,979,886,1087,1180,1069,1015,943,1266,935,1061,1130,1278,1267,937,1258,943,1051,1493,972,1144,1030,998,1121,943,1412,885,959,1754,914,971,1167,989,1103,1014,1119,909,961,1169,921,1423,1154,912,1236,887,920,1227,890,1026,885,1102,1140,1060,927,1114,1035,1110,1135,1311,882,1165,904,1018,936,1025,1031,1126,1002,1111,1096,1114,958,1104,1074,910,892,944,975,1245,1139,920,947,1043,1068,951,1009,1051,905,1046,977,991,964,1022,1581,998,894,939,1227,1504,978,911,903,884,1165,979,950,904,975,1302,983,1109,1006,1135,982,1380,897,1108,920,1093,937,1231,1147,1034,1304,918,1326,1051,936,1025,999,1001,916,1053,1008,957,1314,1001,1601,989,1368,883,1356,1135,946,1042,892,1188,1010,922,1501,970,898,916,909,916,918,943,969,1151,1095,899,920,886,1232,942,1194,1022,976,1513,1008,1108,961,1172,1063,1057,1175,1110,1363,1182,1008,900,1013,1099,1012,1316,1293,1101,1003,1072,1050,1270,972,1020,1592,1288,1132,1258,1109,944,1083,1026,992,975,993,960,1011,917,1036,880,978,1087,1046,954,1202,962,1445,1511,1102,1187,996,1418,892,961,1267,1035,1500,910,894,1509,960,961,1080,1035,905,892,900,1006,982,1209,986,1132,902,966,940,971,1052,908,886,983,1138,989,1038,1055,1020,969,1845,891,1246,1060,893,940,926,1487,1313,903,1097,892,1412,1172,985,929,970,1079,1003,918,927,1008,1239,1032,960,911,900,909,1196,982,1061,1088,896,1044,904,968,1373,1001,900,1335,1323,1017,991,1305,928,1056,1104,1052,902,1036,1269,1157,1062,894,1039,1196,1118,1501,916,1403,993,1043,917,1410,1139,1192,930,914,1845,968,1230,1794,1397,974,968,966,896,996,880,1219,1020,1379,1458,1120,1332,1066,1153,999,880,970,987,1251,1118,908,880,1006,1053,915,944,1040,957,1042,1485,1056,919,1186,924,893,1021,972,1099,948,1069,910,918,954,900,951,1247,1139,932,1233,891,939,945,999,1169,1369,1372,1137,967,973,1215,1207,1247,891,1024,1037,1003,1050,1168,937,1472,1174,1464,1110,881,1256,1082,1217,988,945,1566,1047,1121,971,1012,1426,1113,973,1119,959,1431,1006,1245,909,903,1388,880,966,968,1034,1498,1050,1289,1303,1227,992,1326,1255,1399,1272,1029,900,928,880,1235,925,1220,1015,887,977,1161,900,1232,983,961,1081,1192,933,1248,1640,1020,957,1064,940,1341,1917,942,1230,1186,1089,1133,1154,994,940,1032,971,1334,945,1009,963,1018,955,1184,938,992,1364,926,1174,889,1479,1088,942,926,1340,1387,895,1193,946,929,1208,963,1158,1058,981,1305,1067,991,988,973,927,880,905,1244,1072,989,1332,969,880,891,1253,972,1173,964,984,1381,1155,895,1008,1145,1010,938,1201,918,997,964,1368,936,1060,942,1003,1040,1085,1085,926,1015,956,1022,919,898,931,958,886,1000,943,1212,902,1182,971,1517,1024,1077,910,1090,938,945,1230,969,928,1261,892,1149,1146,1115,1153,1163,1012,1005,1074,901,966,1142,1125,882,1278,987,1093,966,1054,1039,1411,1751,1121,1171,1377,899,1220,1143,910,901,1153,918,1452,1066,1011,1320,918,903,1795,1104,1111,889,952,883,1032,934,1304,948,1172,971,1021,1012,891,897,1286,1062,946,911,944,895,990,1607,1032,1023,1042,975,954,987,1223,1104,930,952,885,1208,958,890,941,920,1007,906,1259,963,898,1069,1058,922,1043,1006,885,965,978,914,891,1155,946,1246,1452,930,1166,1031,1218,1376,904,909,1002,1001,1179,995,909,947,1068,1584,1231,1329,1327,1004,897,969,1136,1133,1222,948,936,917,1002,957,1227,1307,1067,1360,1261,894,1164,975,880,885,1192,887,926,1236,1149,1315,969,1030,982,927,1295,1223,1142,1041,905,1038,939,887,1117,963,989,886,1140,1077,989,954,1075,1087,1031,1319,1281,924,1264,1007,1273,1043,917,917,941,1115,1280,1022,1065,1086,952,970,1147,962,1089,896,1140,1181,1009,1065,1036,1025,909,934,973,966,1083,1004,914,966,1445,883,946,1431,1097,1025,1453,906,889,1083,883,924,916,899,1095,918,892,1022,1316,910,1042,1189,882,953,979,1097,1282,901,993,937,1100,884,902,1111,1202,931,982,881,973,1074,1012,996,1094,965,1040,951,1108,1018,1165,1125,1562,971,890,925,933,901,1030,956,1028,965,1159,1318,1011,1044,1194,1460,1040,887,1071,1040,899,925,1368,1408,915,952,910,925,1209,942,887,912,1160,1063,1455,1116,1293,1190,915,888,1027,886,982,1082,971,1593,1478,1074,958,1016,955,1172,1150,917,948,940,901,1029,997,899,1022,977,921,905,880,914,929,982,903,967,1039,927,1085,1238,1102,1076,1396,1031,931,1665,1056,1209,1236,1011,1140,916,1039,981,1074,1018,912,951,934,1172,945,954,904,1390,1051,1108,924,939,1070,1094,1342,1598,1376,1164,957,1042,1738,894,948,938,1002,1127,1235,1209,1126,1052,955,989,1060,1532,1140,984,925,988,985,1157,892,1179,949,1636,1070,1146,1018,1074,1042,1076,1686,1311,1453,881,1049,916,1082,1121,990,1099,966,1189,1187,978,919,950,944,954,987,1020,1030,1019,924,1040,902,1170,1338,925,953,1239,1011,889,1353,881,944,908,1349,1039,935,894,943,888,1179,954,1232,1345,1364,1309,1007,1276,959,887,952,973,999,1121,1102,926,988,976,1098,919,1376,888,938,1016,1006,926,1036,1106,905,1104,988,1029,1535,1041,1194,1117,1122,972,1176,946,1215,930,1242,1142,910,1014,965,889,887,921,1169,925,991,882,924,1031,880,1048,1064,979,968,1141,1026,881,910,1021,953,932,999,1002,1259,888,933,973,1168,951,894,880,1546,943,989,980,995,1217,1030,993,1148,1044,961,1011,968,944,906,1821,979,947,1677,945,916,922,1194,990,905,1345,1141,893,890,924,924,1301,981,971,1134,1174,896,1025,1044,932,1233,937,927,1536,1123,1190,911,964,1020,1000,1191,907,1005,896,997,1057,947,990,978,1025,1258,998,940,1182,1222,907,890,909,962,1160,912,898,910,916,892,987,941,1198,1229,1307,911,911,1362,1009,1073,893,1753,959,1310,1008,1395,948,1313,1402,1007,1078,1072,1207,1178,1810,2225,1134,911,940,1080,1046,1016,993,937,881,1219,1090,904,1218,992,932,1352,888,1153,1046,1080,1083,895,890,1099,1009,975,1241,1090,1210,1133,1038,1427,905,944,1083,1034,973,894,1096,1252,1101,985,1197,1118,1087,984,1218,1063,1081,972,1181,1574,970,926,1367,977,1084,1059,1218,1299,985,952,1354,910,967,948,1220,980,942,889,966,937,1212,904,1243,1715,1064,1038,943,1414,999,952,1233,1043,1037,1090,1170,931,1004,894,934,993,1197,902,1013,888,1041,889,985,1188,1001,1089,1076,1208,1024,1072,1286,1496,2006,1830,1143,889,1087,1562,946,902,985,1037,1254,977,1043,885,922,997,948,949,937,995,983,1399,1186,1159,1105,1201,1069,986,945,910,939,947,920,1087,1015,892,982,924,1732,1037,886,1195,1097,1024,1091,903,1153,1036,1112,1205,1081,1598,887,1267,1327,941,1192,995,967,999,1150,1108,1216,1143,939,908,1133,1063,1035,1301,909,882,1022,1020,1094,895,1342,1161,1507,989,1537,1559,920,1606,966,975,970,978,1024,1034,912,1554,1105,922,892,1037,1115,1170,1028,900,1249,948,1140,1039,1283,1098,961,1278,964,1032,977,975,1139,958,890,903,970,882,926,1089,987,905,1045,1250,921,1074,1064,1022,935,1114,957,1017,920,1695,944,987,1829,1435,975,944,983,1000,1031,896,895,976,1318,986,1011,967,1322,1017,1197,1071,988,931,884,1179,882,1076,1165,970,921,955,1068,1099,968,919,1117,1204,1077,1522,1224,1148,890,1021,1033,1063,990,903,1281,921,957,1044,993,1058,1311,1046,891,891,916,1115,1105,1523,906,1332,1133,942,1237,1096,922,1151,1063,1365,993,888,953,1357,981,933,895,929,904,1071,1005,1010,941,1009,987,942,996,1399,907,1315,1004,1286,1130,915,1120,1024,1233,1205,1098,1230,920,1770,1259,1254,1247,1344,1022,1056,926,880,929,909,1206,1081,1457,1026,1467,1087,1347,893,896,971,882,1037,1975,1130,1365,977,1032,1066,1205,935,978,1078,1193,969,958,1086,1061,1467,1352,1275,1039,1021,996,1230,1225,1151,983,884,986,1132,981,1048,1031,956,930,924,887,951,978,952,1070,939,940,1169,1272,1144,901,1289,1082,1202,1420,1200,913,1011,885,1054,1131,907,910,955,1136,1170,1067,1093,944,1101,1341,1238,1073,1566,1021,1325,1332,985,1159,895,884,912,1172,1272,898,963,923,977,1166,1156,899,925,1235,1042,999,1210,1484,948,1000,1007,912,913,886,1039,1138,1526,1151,894,1066,1073,924,928,1214,933,932,912,937,1094,930,1052,914,930,1111,1005,1005,904,1036,892,952,886,1197,1173,1303,1165,1342,917,946,894,1192,1007,1019,1043,1667,964,1312,993,1037,1047,1043,941,927,1124,1113,1107,909,886,1029,1446,1083,939,1263,1309,1313,1181,1404,917,949,1039,1137,1066,953,1048,1024,1092,1146,1352,1378,1568,1098,1063,1027,1201,1188,1084,958,1406,998,1237,918,1424,1147,1323,1109,1069,1015,893,1068,922,892,1134,1062,1129,1263,910,1645,1144,1158,1526,956,1222,1227,916,903,972,908,906,1087,994,1118,934,985,924,930,908,1031,968,1013,959,1268,1177,967,1274,1137,1448,898,954,1480,890,1415,978,1602,959,1070,975,1349,1117,948,999,1162,904,910,1574,956,1071,960,1364,1484,1088,1118,1471,1234,1052,1227,1358,976,1112,954,1038,903,942,903,1066,1223,945,1215,1034,887,1069,962,1048,965,891,907,1268,942,1019,947,1030,980,931,1263,1461,952,1146,989,1114,1001,1022,978,891,896,1224,1174,1029,1074,1105,1073,959,966,1046,1071,977,1814,1380,931,1079,1096,1197,895,917,1056,888,1067,990,1120,964,941,1097,1132,935,966,1036,1166,924,905,1287,958,982,971,1090,1099,1070,979,1178,1075,1077,947,957,1041,1036,897,910,891,997,938,1135,1029,1223,902,912,1060,918,1136,911,1467,961,887,904,1126,1018,1415,1108,930,926,1434,1025,974,1452,1124,1053,935,1237,1091,998,1077,988,893,951,1463,904,1416,1255,1151,1413,929,1477,979,1015,999,1097,1163,1616,1224,1165,1096,990,1096,1077,916,918,1240,1205,913,945,1100,897,887,1109,937,1201,886,1332,898,997,985,973,1197,1189,915,1202,946,987,956,1050,898,881,1211,925,882,916,934,924,929,1016,1422,970,934,968,1805,1396,989,885,1258,914,959,1183,958,1004,1020,1255,1073,1060,937,1052,1038,939,992,939,973,1256,1589,1165,981,1263,1891,1160,910,885,1020,936,1042,1092,881,930,1090,884,920,1346,918,1146,1196,1248,896,910,1136,1136,1022,1192,1026,1258,905,1477,1086,903,901,952,929,947,990,1022,1128,1179,906,1450,991,1072,1351,886,1041,1032,1133,986,911,1009,998,941,1273,888,991,1004,1013,975,1337,920,902,1005,999,1097,1388,1297,1274,1010,904,1353,1361,890,1498,1172,1181,1112,975,988,946,1151,924,1018,1040,1293,1005,1065,1025,889,1099,888,883,1119,1122,1085,1123,1252,989,900,957,1048,974,1194,934,931,905,951,950,1052,1079,926,1055,939,1084,1029,922,1103,941,916,1196,935,925,896,881,919,1527,971,991,1009,894,956,995,1174,1238,897,1191,921,1187,964,1316,1150,1029,889,927,1050,1077,1092,953,966,1149,1723,1170,1102,1045],\"q1\":157.0,\"med\":245.0,\"q3\":446.0}\n",
      "{'mean': 338.1645102413, 'iqr': 289.0, 'cilo': 242.9567100736, 'cihi': 247.0432899264, 'whishi': 879, 'whislo': 100, 'fliers': [1898, 1906, 882, 1168, 1047, 928, 1054, 1203, 892, 1125, 1231, 1183, 939, 1059, 902, 1421, 976, 1384, 1071, 881, 1022, 935, 1827, 903, 960, 1714, 903, 1115, 954, 990, 1253, 1032, 1031, 923, 974, 905, 936, 1084, 1776, 1213, 985, 1322, 1141, 924, 1355, 1021, 968, 924, 1003, 1286, 999, 977, 1588, 904, 930, 1012, 938, 1142, 900, 1074, 1168, 1092, 899, 961, 892, 1170, 1138, 888, 1240, 1038, 912, 1159, 901, 979, 886, 1087, 1180, 1069, 1015, 943, 1266, 935, 1061, 1130, 1278, 1267, 937, 1258, 943, 1051, 1493, 972, 1144, 1030, 998, 1121, 943, 1412, 885, 959, 1754, 914, 971, 1167, 989, 1103, 1014, 1119, 909, 961, 1169, 921, 1423, 1154, 912, 1236, 887, 920, 1227, 890, 1026, 885, 1102, 1140, 1060, 927, 1114, 1035, 1110, 1135, 1311, 882, 1165, 904, 1018, 936, 1025, 1031, 1126, 1002, 1111, 1096, 1114, 958, 1104, 1074, 910, 892, 944, 975, 1245, 1139, 920, 947, 1043, 1068, 951, 1009, 1051, 905, 1046, 977, 991, 964, 1022, 1581, 998, 894, 939, 1227, 1504, 978, 911, 903, 884, 1165, 979, 950, 904, 975, 1302, 983, 1109, 1006, 1135, 982, 1380, 897, 1108, 920, 1093, 937, 1231, 1147, 1034, 1304, 918, 1326, 1051, 936, 1025, 999, 1001, 916, 1053, 1008, 957, 1314, 1001, 1601, 989, 1368, 883, 1356, 1135, 946, 1042, 892, 1188, 1010, 922, 1501, 970, 898, 916, 909, 916, 918, 943, 969, 1151, 1095, 899, 920, 886, 1232, 942, 1194, 1022, 976, 1513, 1008, 1108, 961, 1172, 1063, 1057, 1175, 1110, 1363, 1182, 1008, 900, 1013, 1099, 1012, 1316, 1293, 1101, 1003, 1072, 1050, 1270, 972, 1020, 1592, 1288, 1132, 1258, 1109, 944, 1083, 1026, 992, 975, 993, 960, 1011, 917, 1036, 880, 978, 1087, 1046, 954, 1202, 962, 1445, 1511, 1102, 1187, 996, 1418, 892, 961, 1267, 1035, 1500, 910, 894, 1509, 960, 961, 1080, 1035, 905, 892, 900, 1006, 982, 1209, 986, 1132, 902, 966, 940, 971, 1052, 908, 886, 983, 1138, 989, 1038, 1055, 1020, 969, 1845, 891, 1246, 1060, 893, 940, 926, 1487, 1313, 903, 1097, 892, 1412, 1172, 985, 929, 970, 1079, 1003, 918, 927, 1008, 1239, 1032, 960, 911, 900, 909, 1196, 982, 1061, 1088, 896, 1044, 904, 968, 1373, 1001, 900, 1335, 1323, 1017, 991, 1305, 928, 1056, 1104, 1052, 902, 1036, 1269, 1157, 1062, 894, 1039, 1196, 1118, 1501, 916, 1403, 993, 1043, 917, 1410, 1139, 1192, 930, 914, 1845, 968, 1230, 1794, 1397, 974, 968, 966, 896, 996, 880, 1219, 1020, 1379, 1458, 1120, 1332, 1066, 1153, 999, 880, 970, 987, 1251, 1118, 908, 880, 1006, 1053, 915, 944, 1040, 957, 1042, 1485, 1056, 919, 1186, 924, 893, 1021, 972, 1099, 948, 1069, 910, 918, 954, 900, 951, 1247, 1139, 932, 1233, 891, 939, 945, 999, 1169, 1369, 1372, 1137, 967, 973, 1215, 1207, 1247, 891, 1024, 1037, 1003, 1050, 1168, 937, 1472, 1174, 1464, 1110, 881, 1256, 1082, 1217, 988, 945, 1566, 1047, 1121, 971, 1012, 1426, 1113, 973, 1119, 959, 1431, 1006, 1245, 909, 903, 1388, 880, 966, 968, 1034, 1498, 1050, 1289, 1303, 1227, 992, 1326, 1255, 1399, 1272, 1029, 900, 928, 880, 1235, 925, 1220, 1015, 887, 977, 1161, 900, 1232, 983, 961, 1081, 1192, 933, 1248, 1640, 1020, 957, 1064, 940, 1341, 1917, 942, 1230, 1186, 1089, 1133, 1154, 994, 940, 1032, 971, 1334, 945, 1009, 963, 1018, 955, 1184, 938, 992, 1364, 926, 1174, 889, 1479, 1088, 942, 926, 1340, 1387, 895, 1193, 946, 929, 1208, 963, 1158, 1058, 981, 1305, 1067, 991, 988, 973, 927, 880, 905, 1244, 1072, 989, 1332, 969, 880, 891, 1253, 972, 1173, 964, 984, 1381, 1155, 895, 1008, 1145, 1010, 938, 1201, 918, 997, 964, 1368, 936, 1060, 942, 1003, 1040, 1085, 1085, 926, 1015, 956, 1022, 919, 898, 931, 958, 886, 1000, 943, 1212, 902, 1182, 971, 1517, 1024, 1077, 910, 1090, 938, 945, 1230, 969, 928, 1261, 892, 1149, 1146, 1115, 1153, 1163, 1012, 1005, 1074, 901, 966, 1142, 1125, 882, 1278, 987, 1093, 966, 1054, 1039, 1411, 1751, 1121, 1171, 1377, 899, 1220, 1143, 910, 901, 1153, 918, 1452, 1066, 1011, 1320, 918, 903, 1795, 1104, 1111, 889, 952, 883, 1032, 934, 1304, 948, 1172, 971, 1021, 1012, 891, 897, 1286, 1062, 946, 911, 944, 895, 990, 1607, 1032, 1023, 1042, 975, 954, 987, 1223, 1104, 930, 952, 885, 1208, 958, 890, 941, 920, 1007, 906, 1259, 963, 898, 1069, 1058, 922, 1043, 1006, 885, 965, 978, 914, 891, 1155, 946, 1246, 1452, 930, 1166, 1031, 1218, 1376, 904, 909, 1002, 1001, 1179, 995, 909, 947, 1068, 1584, 1231, 1329, 1327, 1004, 897, 969, 1136, 1133, 1222, 948, 936, 917, 1002, 957, 1227, 1307, 1067, 1360, 1261, 894, 1164, 975, 880, 885, 1192, 887, 926, 1236, 1149, 1315, 969, 1030, 982, 927, 1295, 1223, 1142, 1041, 905, 1038, 939, 887, 1117, 963, 989, 886, 1140, 1077, 989, 954, 1075, 1087, 1031, 1319, 1281, 924, 1264, 1007, 1273, 1043, 917, 917, 941, 1115, 1280, 1022, 1065, 1086, 952, 970, 1147, 962, 1089, 896, 1140, 1181, 1009, 1065, 1036, 1025, 909, 934, 973, 966, 1083, 1004, 914, 966, 1445, 883, 946, 1431, 1097, 1025, 1453, 906, 889, 1083, 883, 924, 916, 899, 1095, 918, 892, 1022, 1316, 910, 1042, 1189, 882, 953, 979, 1097, 1282, 901, 993, 937, 1100, 884, 902, 1111, 1202, 931, 982, 881, 973, 1074, 1012, 996, 1094, 965, 1040, 951, 1108, 1018, 1165, 1125, 1562, 971, 890, 925, 933, 901, 1030, 956, 1028, 965, 1159, 1318, 1011, 1044, 1194, 1460, 1040, 887, 1071, 1040, 899, 925, 1368, 1408, 915, 952, 910, 925, 1209, 942, 887, 912, 1160, 1063, 1455, 1116, 1293, 1190, 915, 888, 1027, 886, 982, 1082, 971, 1593, 1478, 1074, 958, 1016, 955, 1172, 1150, 917, 948, 940, 901, 1029, 997, 899, 1022, 977, 921, 905, 880, 914, 929, 982, 903, 967, 1039, 927, 1085, 1238, 1102, 1076, 1396, 1031, 931, 1665, 1056, 1209, 1236, 1011, 1140, 916, 1039, 981, 1074, 1018, 912, 951, 934, 1172, 945, 954, 904, 1390, 1051, 1108, 924, 939, 1070, 1094, 1342, 1598, 1376, 1164, 957, 1042, 1738, 894, 948, 938, 1002, 1127, 1235, 1209, 1126, 1052, 955, 989, 1060, 1532, 1140, 984, 925, 988, 985, 1157, 892, 1179, 949, 1636, 1070, 1146, 1018, 1074, 1042, 1076, 1686, 1311, 1453, 881, 1049, 916, 1082, 1121, 990, 1099, 966, 1189, 1187, 978, 919, 950, 944, 954, 987, 1020, 1030, 1019, 924, 1040, 902, 1170, 1338, 925, 953, 1239, 1011, 889, 1353, 881, 944, 908, 1349, 1039, 935, 894, 943, 888, 1179, 954, 1232, 1345, 1364, 1309, 1007, 1276, 959, 887, 952, 973, 999, 1121, 1102, 926, 988, 976, 1098, 919, 1376, 888, 938, 1016, 1006, 926, 1036, 1106, 905, 1104, 988, 1029, 1535, 1041, 1194, 1117, 1122, 972, 1176, 946, 1215, 930, 1242, 1142, 910, 1014, 965, 889, 887, 921, 1169, 925, 991, 882, 924, 1031, 880, 1048, 1064, 979, 968, 1141, 1026, 881, 910, 1021, 953, 932, 999, 1002, 1259, 888, 933, 973, 1168, 951, 894, 880, 1546, 943, 989, 980, 995, 1217, 1030, 993, 1148, 1044, 961, 1011, 968, 944, 906, 1821, 979, 947, 1677, 945, 916, 922, 1194, 990, 905, 1345, 1141, 893, 890, 924, 924, 1301, 981, 971, 1134, 1174, 896, 1025, 1044, 932, 1233, 937, 927, 1536, 1123, 1190, 911, 964, 1020, 1000, 1191, 907, 1005, 896, 997, 1057, 947, 990, 978, 1025, 1258, 998, 940, 1182, 1222, 907, 890, 909, 962, 1160, 912, 898, 910, 916, 892, 987, 941, 1198, 1229, 1307, 911, 911, 1362, 1009, 1073, 893, 1753, 959, 1310, 1008, 1395, 948, 1313, 1402, 1007, 1078, 1072, 1207, 1178, 1810, 2225, 1134, 911, 940, 1080, 1046, 1016, 993, 937, 881, 1219, 1090, 904, 1218, 992, 932, 1352, 888, 1153, 1046, 1080, 1083, 895, 890, 1099, 1009, 975, 1241, 1090, 1210, 1133, 1038, 1427, 905, 944, 1083, 1034, 973, 894, 1096, 1252, 1101, 985, 1197, 1118, 1087, 984, 1218, 1063, 1081, 972, 1181, 1574, 970, 926, 1367, 977, 1084, 1059, 1218, 1299, 985, 952, 1354, 910, 967, 948, 1220, 980, 942, 889, 966, 937, 1212, 904, 1243, 1715, 1064, 1038, 943, 1414, 999, 952, 1233, 1043, 1037, 1090, 1170, 931, 1004, 894, 934, 993, 1197, 902, 1013, 888, 1041, 889, 985, 1188, 1001, 1089, 1076, 1208, 1024, 1072, 1286, 1496, 2006, 1830, 1143, 889, 1087, 1562, 946, 902, 985, 1037, 1254, 977, 1043, 885, 922, 997, 948, 949, 937, 995, 983, 1399, 1186, 1159, 1105, 1201, 1069, 986, 945, 910, 939, 947, 920, 1087, 1015, 892, 982, 924, 1732, 1037, 886, 1195, 1097, 1024, 1091, 903, 1153, 1036, 1112, 1205, 1081, 1598, 887, 1267, 1327, 941, 1192, 995, 967, 999, 1150, 1108, 1216, 1143, 939, 908, 1133, 1063, 1035, 1301, 909, 882, 1022, 1020, 1094, 895, 1342, 1161, 1507, 989, 1537, 1559, 920, 1606, 966, 975, 970, 978, 1024, 1034, 912, 1554, 1105, 922, 892, 1037, 1115, 1170, 1028, 900, 1249, 948, 1140, 1039, 1283, 1098, 961, 1278, 964, 1032, 977, 975, 1139, 958, 890, 903, 970, 882, 926, 1089, 987, 905, 1045, 1250, 921, 1074, 1064, 1022, 935, 1114, 957, 1017, 920, 1695, 944, 987, 1829, 1435, 975, 944, 983, 1000, 1031, 896, 895, 976, 1318, 986, 1011, 967, 1322, 1017, 1197, 1071, 988, 931, 884, 1179, 882, 1076, 1165, 970, 921, 955, 1068, 1099, 968, 919, 1117, 1204, 1077, 1522, 1224, 1148, 890, 1021, 1033, 1063, 990, 903, 1281, 921, 957, 1044, 993, 1058, 1311, 1046, 891, 891, 916, 1115, 1105, 1523, 906, 1332, 1133, 942, 1237, 1096, 922, 1151, 1063, 1365, 993, 888, 953, 1357, 981, 933, 895, 929, 904, 1071, 1005, 1010, 941, 1009, 987, 942, 996, 1399, 907, 1315, 1004, 1286, 1130, 915, 1120, 1024, 1233, 1205, 1098, 1230, 920, 1770, 1259, 1254, 1247, 1344, 1022, 1056, 926, 880, 929, 909, 1206, 1081, 1457, 1026, 1467, 1087, 1347, 893, 896, 971, 882, 1037, 1975, 1130, 1365, 977, 1032, 1066, 1205, 935, 978, 1078, 1193, 969, 958, 1086, 1061, 1467, 1352, 1275, 1039, 1021, 996, 1230, 1225, 1151, 983, 884, 986, 1132, 981, 1048, 1031, 956, 930, 924, 887, 951, 978, 952, 1070, 939, 940, 1169, 1272, 1144, 901, 1289, 1082, 1202, 1420, 1200, 913, 1011, 885, 1054, 1131, 907, 910, 955, 1136, 1170, 1067, 1093, 944, 1101, 1341, 1238, 1073, 1566, 1021, 1325, 1332, 985, 1159, 895, 884, 912, 1172, 1272, 898, 963, 923, 977, 1166, 1156, 899, 925, 1235, 1042, 999, 1210, 1484, 948, 1000, 1007, 912, 913, 886, 1039, 1138, 1526, 1151, 894, 1066, 1073, 924, 928, 1214, 933, 932, 912, 937, 1094, 930, 1052, 914, 930, 1111, 1005, 1005, 904, 1036, 892, 952, 886, 1197, 1173, 1303, 1165, 1342, 917, 946, 894, 1192, 1007, 1019, 1043, 1667, 964, 1312, 993, 1037, 1047, 1043, 941, 927, 1124, 1113, 1107, 909, 886, 1029, 1446, 1083, 939, 1263, 1309, 1313, 1181, 1404, 917, 949, 1039, 1137, 1066, 953, 1048, 1024, 1092, 1146, 1352, 1378, 1568, 1098, 1063, 1027, 1201, 1188, 1084, 958, 1406, 998, 1237, 918, 1424, 1147, 1323, 1109, 1069, 1015, 893, 1068, 922, 892, 1134, 1062, 1129, 1263, 910, 1645, 1144, 1158, 1526, 956, 1222, 1227, 916, 903, 972, 908, 906, 1087, 994, 1118, 934, 985, 924, 930, 908, 1031, 968, 1013, 959, 1268, 1177, 967, 1274, 1137, 1448, 898, 954, 1480, 890, 1415, 978, 1602, 959, 1070, 975, 1349, 1117, 948, 999, 1162, 904, 910, 1574, 956, 1071, 960, 1364, 1484, 1088, 1118, 1471, 1234, 1052, 1227, 1358, 976, 1112, 954, 1038, 903, 942, 903, 1066, 1223, 945, 1215, 1034, 887, 1069, 962, 1048, 965, 891, 907, 1268, 942, 1019, 947, 1030, 980, 931, 1263, 1461, 952, 1146, 989, 1114, 1001, 1022, 978, 891, 896, 1224, 1174, 1029, 1074, 1105, 1073, 959, 966, 1046, 1071, 977, 1814, 1380, 931, 1079, 1096, 1197, 895, 917, 1056, 888, 1067, 990, 1120, 964, 941, 1097, 1132, 935, 966, 1036, 1166, 924, 905, 1287, 958, 982, 971, 1090, 1099, 1070, 979, 1178, 1075, 1077, 947, 957, 1041, 1036, 897, 910, 891, 997, 938, 1135, 1029, 1223, 902, 912, 1060, 918, 1136, 911, 1467, 961, 887, 904, 1126, 1018, 1415, 1108, 930, 926, 1434, 1025, 974, 1452, 1124, 1053, 935, 1237, 1091, 998, 1077, 988, 893, 951, 1463, 904, 1416, 1255, 1151, 1413, 929, 1477, 979, 1015, 999, 1097, 1163, 1616, 1224, 1165, 1096, 990, 1096, 1077, 916, 918, 1240, 1205, 913, 945, 1100, 897, 887, 1109, 937, 1201, 886, 1332, 898, 997, 985, 973, 1197, 1189, 915, 1202, 946, 987, 956, 1050, 898, 881, 1211, 925, 882, 916, 934, 924, 929, 1016, 1422, 970, 934, 968, 1805, 1396, 989, 885, 1258, 914, 959, 1183, 958, 1004, 1020, 1255, 1073, 1060, 937, 1052, 1038, 939, 992, 939, 973, 1256, 1589, 1165, 981, 1263, 1891, 1160, 910, 885, 1020, 936, 1042, 1092, 881, 930, 1090, 884, 920, 1346, 918, 1146, 1196, 1248, 896, 910, 1136, 1136, 1022, 1192, 1026, 1258, 905, 1477, 1086, 903, 901, 952, 929, 947, 990, 1022, 1128, 1179, 906, 1450, 991, 1072, 1351, 886, 1041, 1032, 1133, 986, 911, 1009, 998, 941, 1273, 888, 991, 1004, 1013, 975, 1337, 920, 902, 1005, 999, 1097, 1388, 1297, 1274, 1010, 904, 1353, 1361, 890, 1498, 1172, 1181, 1112, 975, 988, 946, 1151, 924, 1018, 1040, 1293, 1005, 1065, 1025, 889, 1099, 888, 883, 1119, 1122, 1085, 1123, 1252, 989, 900, 957, 1048, 974, 1194, 934, 931, 905, 951, 950, 1052, 1079, 926, 1055, 939, 1084, 1029, 922, 1103, 941, 916, 1196, 935, 925, 896, 881, 919, 1527, 971, 991, 1009, 894, 956, 995, 1174, 1238, 897, 1191, 921, 1187, 964, 1316, 1150, 1029, 889, 927, 1050, 1077, 1092, 953, 966, 1149, 1723, 1170, 1102, 1045], 'q1': 157.0, 'med': 245.0, 'q3': 446.0}\n"
     ]
    }
   ],
   "source": [
    "import json\n",
    "print(testukas)\n",
    "parsintas = json.loads(testukas)\n",
    "print(parsintas)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "688f8b89",
   "metadata": {},
   "outputs": [],
   "source": [
    "dfamas = pd.read_csv('data.csv', sep='\\t')\n",
    "yess = dfamas.iloc[0].to_dict()\n",
    "\n",
    "print(yess)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "21094120",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'mean': 338.16451024133033, 'iqr': 289.0, 'cilo': 242.95671007358652, 'cihi': 247.04328992641345, 'whishi': 879, 'whislo': 100, 'fliers': '[1898 1906  882 ... 1170 1102 1045]', 'q1': 157.0, 'med': 245.0, 'q3': 446.0}\n"
     ]
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "id": "1f91758e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'whiskers': [<matplotlib.lines.Line2D at 0x13ecf0fb0>,\n",
       "  <matplotlib.lines.Line2D at 0x13ecf1220>],\n",
       " 'caps': [<matplotlib.lines.Line2D at 0x13ecf1490>,\n",
       "  <matplotlib.lines.Line2D at 0x13ecf1700>],\n",
       " 'boxes': [<matplotlib.patches.PathPatch at 0x13ecf0cb0>],\n",
       " 'medians': [<matplotlib.lines.Line2D at 0x13ecf19a0>],\n",
       " 'fliers': [],\n",
       " 'means': []}"
      ]
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAjMAAAGiCAYAAAASgEe5AAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjkuMiwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy8hTgPZAAAACXBIWXMAAA9hAAAPYQGoP6dpAAA4GUlEQVR4nO3df3RU9Z3/8deQH5MfJiMkmmEkQLBR0KDCUCmhFFpCOChQD91CixbUsAdOFIyAaJquBqrJSteQCgWrBkKJCKfVqLtLlbCtkYi2MIDLj1htiSSpGXNC4ySR7ATC/f7Bl9kdkiAT8uuS5+Oce9r53Pe9874e7/WVz9yZazEMwxAAAIBJDejtBgAAAK4EYQYAAJgaYQYAAJgaYQYAAJgaYQYAAJgaYQYAAJgaYQYAAJgaYQYAAJgaYQYAAJgaYQYAAJgaYQZAr3jvvfc0a9YsORwOWSwWvfHGG1+7TWlpqZxOp8LCwjRixAi98MIL3d8ogD6PMAOgV3z11Ve6/fbbtWHDhsuqr6io0F133aVJkybp0KFD+ulPf6ply5bptdde6+ZOAfR1Fh40CaC3WSwWFRcX65577umw5vHHH9dbb72l8vJy39iSJUv00Ucf6YMPPuiBLgH0VcG93UBnnDt3Tp9//rmioqJksVh6ux2gXzIMQ42NjXI4HBowoPsneT/44AOlpqb6jU2fPl0FBQU6c+aMQkJC2mzj9Xrl9Xp9r8+dO6d//OMfiomJ4doB9ILuum6YMsx8/vnnio+P7+02AEiqqqrSkCFDuv193G634uLi/Mbi4uJ09uxZ1dXVafDgwW22yc3N1erVq7u9NwCB6errhinDTFRUlKTz/zCio6N7uRugf2poaFB8fLzvfOwJF8+mXPiUvKNZlszMTC1fvtz32uPxaOjQoVw7gF7SXdcNU4aZCxeu6OhoLkhAL+upj2vsdrvcbrffWG1trYKDgxUTE9PuNlarVVartc041w6gd3X1dYNvMwEwhQkTJqikpMRvbPfu3Ro3bly798sA6D8CDjONjY3KyMjQsGHDFB4eruTkZO3fv9+33jAMZWdny+FwKDw8XFOmTNGxY8f89uH1erV06VLFxsYqMjJSs2fPVnV19ZUfDQDTaGpq0uHDh3X48GFJ5796ffjwYVVWVko6/xHRggULfPVLlizRyZMntXz5cpWXl2vz5s0qKCjQypUre6N9AH1IwGFm0aJFKikp0bZt23TkyBGlpqYqJSVFf//73yVJa9euVV5enjZs2KD9+/fLbrdr2rRpamxs9O0jIyNDxcXF2rFjh8rKytTU1KSZM2eqtbW1644MQJ924MABjRkzRmPGjJEkLV++XGPGjNGTTz4pSaqpqfEFG0lKSEjQrl279O677+qOO+7Qz3/+cz3//PP6wQ9+0Cv9A+g7AvqdmebmZkVFRenNN9/U3Xff7Ru/4447NHPmTP385z+Xw+FQRkaGHn/8cUnnZ2Hi4uL07LPPavHixfJ4PLruuuu0bds2zZs3T9L/fjtp165dmj59epv3vfjrlRduIPJ4PHzuDfSShoYG2Ww2U52HZuwZuJp01zkY0MzM2bNn1draqrCwML/x8PBwlZWVqaKiQm632++3IKxWqyZPnqx9+/ZJklwul86cOeNX43A4lJSU5Ku5WG5urmw2m2/ha9kAAOCCgMJMVFSUJkyYoJ///Of6/PPP1draqqKiIv3pT39STU2N75sG7f0WxIV1brdboaGhGjhwYIc1F8vMzJTH4/EtVVVVgbQNAACuYgHfM7Nt2zYZhqEbbrhBVqtVzz//vObPn6+goCBfTXu/BfF1X8O6VI3VavV9lZKvVAIAgP8r4DBz4403qrS0VE1NTaqqqtKf//xnnTlzRgkJCbLb7ZLU7m9BXJitsdvtamlpUX19fYc1AAAAl6vTvzMTGRmpwYMHq76+Xu+8846+//3v+wLN//0tiJaWFpWWlio5OVmS5HQ6FRIS4ldTU1Ojo0eP+moAAAAuV8C/APzOO+/IMAzdfPPN+utf/6rHHntMN998sx544AFZLBZlZGQoJydHiYmJSkxMVE5OjiIiIjR//nxJks1mU1pamlasWKGYmBgNGjRIK1eu1OjRo5WSktLlBwgAAK5uAYcZj8ejzMxMVVdXa9CgQfrBD36gZ555xvcLnKtWrVJzc7PS09NVX1+v8ePHa/fu3X7PYVi3bp2Cg4M1d+5cNTc3a+rUqSosLPS77wYAAOByBPQ7M30FvxUB9D4znodm7Bm4mnTXOWjKB02i7zt9+rQ+/vjjDtc3Nzfrs88+0/DhwxUeHn7JfY0cOVIRERFd3SIA4CpBmEG3+Pjjj+V0OrtkXy6XS2PHju2SfQEArj6EGXSLkSNHyuVydbi+vLxc9913n4qKijRq1Kiv3RcAAB0hzKBbREREXNZsyqhRo5h1AQBckU7/zgwAAEBfQJgBAACmRpgBAACmRpgBAACmRpgBAACmRpgBAACmRpgBAACmRpgBAACmRpgBAACmRpgBAACmRpgBAACmRpgBAACmRpgBAACmRpgBAACmRpgBAACmRpgBAACmRpgBAACmRpgBAACmRpgBAACmRpgBAACmRpgBAACmRpgBAACmRpgBAACmRpgBAACmRpgBAACmRpgBAACmRpgBAACmRpgBAACmFlCYOXv2rH72s58pISFB4eHhGjFihNasWaNz5875agzDUHZ2thwOh8LDwzVlyhQdO3bMbz9er1dLly5VbGysIiMjNXv2bFVXV3fNEQEAgH4loDDz7LPP6oUXXtCGDRtUXl6utWvX6he/+IXWr1/vq1m7dq3y8vK0YcMG7d+/X3a7XdOmTVNjY6OvJiMjQ8XFxdqxY4fKysrU1NSkmTNnqrW1teuODAAA9AvBgRR/8MEH+v73v6+7775bkjR8+HC9+uqrOnDggKTzszL5+fnKysrSnDlzJElbt25VXFyctm/frsWLF8vj8aigoEDbtm1TSkqKJKmoqEjx8fHas2ePpk+f3pXHBwAArnIBzcx8+9vf1n/913/pk08+kSR99NFHKisr01133SVJqqiokNvtVmpqqm8bq9WqyZMna9++fZIkl8ulM2fO+NU4HA4lJSX5ai7m9XrV0NDgtwAAAEgBzsw8/vjj8ng8GjlypIKCgtTa2qpnnnlGP/7xjyVJbrdbkhQXF+e3XVxcnE6ePOmrCQ0N1cCBA9vUXNj+Yrm5uVq9enUgrQIAgH4ioJmZnTt3qqioSNu3b9fBgwe1detW/du//Zu2bt3qV2exWPxeG4bRZuxil6rJzMyUx+PxLVVVVYG0DQAArmIBzcw89thjeuKJJ/SjH/1IkjR69GidPHlSubm5Wrhwoex2u6Tzsy+DBw/2bVdbW+ubrbHb7WppaVF9fb3f7Extba2Sk5PbfV+r1Sqr1RrYkQEAgH4hoJmZ06dPa8AA/02CgoJ8X81OSEiQ3W5XSUmJb31LS4tKS0t9QcXpdCokJMSvpqamRkePHu0wzAAAAHQkoJmZWbNm6ZlnntHQoUN166236tChQ8rLy9ODDz4o6fzHSxkZGcrJyVFiYqISExOVk5OjiIgIzZ8/X5Jks9mUlpamFStWKCYmRoMGDdLKlSs1evRo37ebAAAALldAYWb9+vX6l3/5F6Wnp6u2tlYOh0OLFy/Wk08+6atZtWqVmpublZ6ervr6eo0fP167d+9WVFSUr2bdunUKDg7W3Llz1dzcrKlTp6qwsFBBQUFdd2QAAKBfsBiGYfR2E4FqaGiQzWaTx+NRdHR0b7eDTjh48KCcTqdcLpfGjh3b2+2gE8x4HpqxZ+Bq0l3nIM9mAgAApkaYAQAApkaYAQAApkaYAQAApkaYAQAApkaYAQAApkaYAQAApkaYAQAApkaYAQAApkaYAQAApkaYAQAApkaYAQAApkaYAdBrNm7cqISEBIWFhcnpdGrv3r2XrH/llVd0++23KyIiQoMHD9YDDzygU6dO9VC3APoqwgyAXrFz505lZGQoKytLhw4d0qRJkzRjxgxVVla2W19WVqYFCxYoLS1Nx44d029/+1vt379fixYt6uHOAfQ1hBkAvSIvL09paWlatGiRRo0apfz8fMXHx2vTpk3t1n/44YcaPny4li1bpoSEBH3729/W4sWLdeDAgR7uHEBfQ5gB0ONaWlrkcrmUmprqN56amqp9+/a1u01ycrKqq6u1a9cuGYahL774Qr/73e909913d/g+Xq9XDQ0NfguAqw9hBkCPq6urU2trq+Li4vzG4+Li5Ha7290mOTlZr7zyiubNm6fQ0FDZ7XZde+21Wr9+fYfvk5ubK5vN5lvi4+O79DgA9A2EGQC9xmKx+L02DKPN2AXHjx/XsmXL9OSTT8rlcuntt99WRUWFlixZ0uH+MzMz5fF4fEtVVVWX9g+gbwju7QYA9D+xsbEKCgpqMwtTW1vbZrbmgtzcXE2cOFGPPfaYJOm2225TZGSkJk2apKefflqDBw9us43VapXVau36AwDQpzAzA6DHhYaGyul0qqSkxG+8pKREycnJ7W5z+vRpDRjgf8kKCgqSdH5GB0D/RZgB0CuWL1+ul19+WZs3b1Z5ebkeffRRVVZW+j42yszM1IIFC3z1s2bN0uuvv65NmzbpxIkTev/997Vs2TLdeeedcjgcvXUYAPoAPmYC0CvmzZunU6dOac2aNaqpqVFSUpJ27dqlYcOGSZJqamr8fnPm/vvvV2NjozZs2KAVK1bo2muv1fe+9z09++yzvXUIAPoIi2HC+dmGhgbZbDZ5PB5FR0f3djvohIMHD8rpdMrlcmns2LG93Q46wYznoRl7Bq4m3XUO8jETAAAwNcIMAAAwNcIMAAAwNcIMAAAwNcIMAAAwNcIMAAAwNcIMAAAwNcIMAAAwNcIMAAAwtYDCzPDhw2WxWNosDz30kKTzD3vLzs6Ww+FQeHi4pkyZomPHjvntw+v1aunSpYqNjVVkZKRmz56t6urqrjsiAADQrwQUZvbv36+amhrfcuGJtz/84Q8lSWvXrlVeXp42bNig/fv3y263a9q0aWpsbPTtIyMjQ8XFxdqxY4fKysrU1NSkmTNnqrW1tQsPCwAA9BcBhZnrrrtOdrvdt/zHf/yHbrzxRk2ePFmGYSg/P19ZWVmaM2eOkpKStHXrVp0+fVrbt2+XJHk8HhUUFOi5555TSkqKxowZo6KiIh05ckR79uzplgMEAABXt07fM9PS0qKioiI9+OCDslgsqqiokNvtVmpqqq/GarVq8uTJ2rdvnyTJ5XLpzJkzfjUOh0NJSUm+mvZ4vV41NDT4LQAAANIVhJk33nhDX375pe6//35JktvtliTFxcX51cXFxfnWud1uhYaGauDAgR3WtCc3N1c2m823xMfHd7ZtAABwlel0mCkoKNCMGTPkcDj8xi0Wi99rwzDajF3s62oyMzPl8Xh8S1VVVWfbBgAAV5lOhZmTJ09qz549WrRokW/MbrdLUpsZltraWt9sjd1uV0tLi+rr6zusaY/ValV0dLTfAgAAIHUyzGzZskXXX3+97r77bt9YQkKC7Ha77xtO0vn7akpLS5WcnCxJcjqdCgkJ8aupqanR0aNHfTUAAACBCA50g3PnzmnLli1auHChgoP/d3OLxaKMjAzl5OQoMTFRiYmJysnJUUREhObPny9JstlsSktL04oVKxQTE6NBgwZp5cqVGj16tFJSUrruqAAAQL8RcJjZs2ePKisr9eCDD7ZZt2rVKjU3Nys9PV319fUaP368du/eraioKF/NunXrFBwcrLlz56q5uVlTp05VYWGhgoKCruxIAABAv2QxDMPo7SYC1dDQIJvNJo/Hw/0zJnXw4EE5nU65XC6NHTu2t9tBJ5jxPDRjz8DVpLvOQZ7NBAAATI0wAwAATI0wAwAATI0wAwAATI0wAwAATI0wAwAATI0wAwAATI0wAwAATI0wAwAATI0wAwAATI0wAwAATI0wAwAATI0wAwAATI0wAwAATI0wAwAATI0wAwAATI0wAwAATI0wAwAATI0wAwAATI0wAwAATI0wAwAATI0wAwAATI0wAwAATI0wAwAATI0wAwAATI0wAwAATI0wAwAATI0wAwAATI0wAwAATI0wAwAATI0wAwAATI0wAwAATC3gMPP3v/9d9913n2JiYhQREaE77rhDLpfLt94wDGVnZ8vhcCg8PFxTpkzRsWPH/Pbh9Xq1dOlSxcbGKjIyUrNnz1Z1dfWVHw0AAOh3Agoz9fX1mjhxokJCQvT73/9ex48f13PPPadrr73WV7N27Vrl5eVpw4YN2r9/v+x2u6ZNm6bGxkZfTUZGhoqLi7Vjxw6VlZWpqalJM2fOVGtra5cdGAAA6B+CAyl+9tlnFR8fry1btvjGhg8f7vv/hmEoPz9fWVlZmjNnjiRp69atiouL0/bt27V48WJ5PB4VFBRo27ZtSklJkSQVFRUpPj5ee/bs0fTp07vgsAAAQH8R0MzMW2+9pXHjxumHP/yhrr/+eo0ZM0YvvfSSb31FRYXcbrdSU1N9Y1arVZMnT9a+ffskSS6XS2fOnPGrcTgcSkpK8tVczOv1qqGhwW8BAACQAgwzJ06c0KZNm5SYmKh33nlHS5Ys0bJly/Sb3/xGkuR2uyVJcXFxftvFxcX51rndboWGhmrgwIEd1lwsNzdXNpvNt8THxwfSNgAAuIoFFGbOnTunsWPHKicnR2PGjNHixYv1z//8z9q0aZNfncVi8XttGEabsYtdqiYzM1Mej8e3VFVVBdI2AAC4igUUZgYPHqxbbrnFb2zUqFGqrKyUJNntdklqM8NSW1vrm62x2+1qaWlRfX19hzUXs1qtio6O9lsAAACkAMPMxIkT9Ze//MVv7JNPPtGwYcMkSQkJCbLb7SopKfGtb2lpUWlpqZKTkyVJTqdTISEhfjU1NTU6evSorwYAAOByBfRtpkcffVTJycnKycnR3Llz9ec//1kvvviiXnzxRUnnP17KyMhQTk6OEhMTlZiYqJycHEVERGj+/PmSJJvNprS0NK1YsUIxMTEaNGiQVq5cqdGjR/u+3QQAAHC5Agoz3/zmN1VcXKzMzEytWbNGCQkJys/P17333uurWbVqlZqbm5Wenq76+nqNHz9eu3fvVlRUlK9m3bp1Cg4O1ty5c9Xc3KypU6eqsLBQQUFBXXdkAACgX7AYhmH0dhOBamhokM1mk8fj4f4Zkzp48KCcTqdcLpfGjh3b2+2gE8x4HpqxZ+Bq0l3nIM9mAtBrNm7cqISEBIWFhcnpdGrv3r2XrPd6vcrKytKwYcNktVp14403avPmzT3ULYC+KqCPmQCgq+zcuVMZGRnauHGjJk6cqF//+teaMWOGjh8/rqFDh7a7zdy5c/XFF1+ooKBA3/jGN1RbW6uzZ8/2cOcA+hrCDIBekZeXp7S0NC1atEiSlJ+fr3feeUebNm1Sbm5um/q3335bpaWlOnHihAYNGiTJ/3EqAPovPmYC0ONaWlrkcrn8HmsiSampqR0+1uTC41TWrl2rG264QTfddJNWrlyp5ubmDt+HR6EA/QMzMwB6XF1dnVpbWy/56JOLnThxQmVlZQoLC1NxcbHq6uqUnp6uf/zjHx3eN5Obm6vVq1d3ef8A+hZmZgD0mkAefXLu3DlZLBa98soruvPOO3XXXXcpLy9PhYWFHc7O8CgUoH9gZgZAj4uNjVVQUNAlH31yscGDB+uGG26QzWbzjY0aNUqGYai6ulqJiYlttrFarbJarV3bPIA+h5kZAD0uNDRUTqfT77EmklRSUtLhY00mTpyozz//XE1NTb6xTz75RAMGDNCQIUO6tV8AfRthBkCvWL58uV5++WVt3rxZ5eXlevTRR1VZWaklS5ZIOv8R0YIFC3z18+fPV0xMjB544AEdP35c7733nh577DE9+OCDCg8P763DANAH8DETgF4xb948nTp1SmvWrFFNTY2SkpK0a9cu34Nra2pqVFlZ6au/5pprVFJSoqVLl2rcuHGKiYnR3Llz9fTTT/fWIQDoIwgzAHpNenq60tPT211XWFjYZmzkyJFtPpoCAD5mAgAApkaYAQAApkaYAQAApkaYAQAApkaYAQAApkaYAQAApkaYAQAApkaYAQAApkaYAQAApkaYAQAApkaYAQAApkaYAQAApkaYAQAApkaYAQAApkaYAQAApkaYAQAApkaYAQAApkaYAQAApkaYAQAApkaYAQAApkaYAQAAphZQmMnOzpbFYvFb7Ha7b71hGMrOzpbD4VB4eLimTJmiY8eO+e3D6/Vq6dKlio2NVWRkpGbPnq3q6uquORoAANDvBDwzc+utt6qmpsa3HDlyxLdu7dq1ysvL04YNG7R//37Z7XZNmzZNjY2NvpqMjAwVFxdrx44dKisrU1NTk2bOnKnW1tauOSIAANCvBAe8QXCw32zMBYZhKD8/X1lZWZozZ44kaevWrYqLi9P27du1ePFieTweFRQUaNu2bUpJSZEkFRUVKT4+Xnv27NH06dOv8HAAAEB/E/DMzKeffiqHw6GEhAT96Ec/0okTJyRJFRUVcrvdSk1N9dVarVZNnjxZ+/btkyS5XC6dOXPGr8bhcCgpKclX0x6v16uGhga/BQAAQAowzIwfP16/+c1v9M477+ill16S2+1WcnKyTp06JbfbLUmKi4vz2yYuLs63zu12KzQ0VAMHDuywpj25ubmy2Wy+JT4+PpC2AQDAVSygMDNjxgz94Ac/0OjRo5WSkqL//M//lHT+46QLLBaL3zaGYbQZu9jX1WRmZsrj8fiWqqqqQNoGAABXsSv6anZkZKRGjx6tTz/91HcfzcUzLLW1tb7ZGrvdrpaWFtXX13dY0x6r1aro6Gi/BQAAQLrCMOP1elVeXq7BgwcrISFBdrtdJSUlvvUtLS0qLS1VcnKyJMnpdCokJMSvpqamRkePHvXVAAAABCKgbzOtXLlSs2bN0tChQ1VbW6unn35aDQ0NWrhwoSwWizIyMpSTk6PExEQlJiYqJydHERERmj9/viTJZrMpLS1NK1asUExMjAYNGqSVK1f6PrYCAAAIVEBhprq6Wj/+8Y9VV1en6667Tt/61rf04YcfatiwYZKkVatWqbm5Wenp6aqvr9f48eO1e/duRUVF+faxbt06BQcHa+7cuWpubtbUqVNVWFiooKCgrj0yAADQL1gMwzB6u4lANTQ0yGazyePxcP+MSR08eFBOp1Mul0tjx47t7XbQCWY8D83YM3A16a5zkGczAQAAUyPMAAAAUyPMAAAAUyPMAAAAUyPMAAAAUyPMAAAAUyPMAAAAUyPMAAAAUyPMAAAAUyPMAAAAUyPMAAAAUyPMAAAAUyPMAAAAUyPMAAAAUyPMAAAAUwvu7QZgTpWVlaqrq+v09uXl5X7/21mxsbEaOnToFe0DAGBuhBkErLKyUqNGjdLp06eveF/33XffFW0fERGh8vJyAg0A9GOEGQSsrq5Op0+fVtHzT2pU4rBO7aP5f7z6rNqt4UPsCg+zdmof5Z+e1H3L1qiuro4wAwD9GGEGnTYqcZjGjr6509tP/OZtXdgNAKC/4gZgAABgaoQZAABgaoQZAABgaoQZAABgaoQZAABgaoQZAABgaoQZAABgaoQZAABgaoQZAABgaoQZAABgaoQZAABgaoQZAABgaoQZAL1m48aNSkhIUFhYmJxOp/bu3XtZ273//vsKDg7WHXfc0b0NAjCFKwozubm5slgsysjI8I0ZhqHs7Gw5HA6Fh4drypQpOnbsmN92Xq9XS5cuVWxsrCIjIzV79mxVV1dfSSsATGbnzp3KyMhQVlaWDh06pEmTJmnGjBmqrKy85HYej0cLFizQ1KlTe6hTAH1dp8PM/v379eKLL+q2227zG1+7dq3y8vK0YcMG7d+/X3a7XdOmTVNjY6OvJiMjQ8XFxdqxY4fKysrU1NSkmTNnqrW1tfNHAsBU8vLylJaWpkWLFmnUqFHKz89XfHy8Nm3adMntFi9erPnz52vChAlf+x5er1cNDQ1+C4CrT6fCTFNTk+6991699NJLGjhwoG/cMAzl5+crKytLc+bMUVJSkrZu3arTp09r+/btks7/VVVQUKDnnntOKSkpGjNmjIqKinTkyBHt2bOna44KQJ/W0tIil8ul1NRUv/HU1FTt27evw+22bNmiv/3tb3rqqacu631yc3Nls9l8S3x8/BX1DaBv6lSYeeihh3T33XcrJSXFb7yiokJut9vvAmW1WjV58mTfBcrlcunMmTN+NQ6HQ0lJSR1exPjrCri61NXVqbW1VXFxcX7jcXFxcrvd7W7z6aef6oknntArr7yi4ODgy3qfzMxMeTwe31JVVXXFvQPoey7vivB/7NixQwcPHtT+/fvbrLtwEWrvAnXy5ElfTWhoqN+MzoWaji5iubm5Wr16daCtAujjLBaL32vDMNqMSVJra6vmz5+v1atX66abbrrs/VutVlmt1ivuE0DfFtDMTFVVlR555BEVFRUpLCysw7rLvUBdbg1/XQFXl9jYWAUFBbX5A6a2trbNH0OS1NjYqAMHDujhhx9WcHCwgoODtWbNGn300UcKDg7WH/7wh55qHUAfFFCYcblcqq2tldPp9F1QSktL9fzzzys4ONh3EbrUBcput6ulpUX19fUd1lzMarUqOjrabwFgXqGhoXI6nSopKfEbLykpUXJycpv66OhoHTlyRIcPH/YtS5Ys0c0336zDhw9r/PjxPdU6gD4ooDAzderUNheUcePG6d5779Xhw4c1YsQI2e12vwtUS0uLSktLfRcop9OpkJAQv5qamhodPXq03YsYgKvT8uXL9fLLL2vz5s0qLy/Xo48+qsrKSi1ZskTS+RnZBQsWSJIGDBigpKQkv+X6669XWFiYkpKSFBkZ2ZuHAqCXBXTPTFRUlJKSkvzGIiMjFRMT4xvPyMhQTk6OEhMTlZiYqJycHEVERGj+/PmSJJvNprS0NK1YsUIxMTEaNGiQVq5cqdGjR7e5oRjA1WvevHk6deqU1qxZo5qaGiUlJWnXrl0aNmyYpPN/5Hzdb84AgNSJG4C/zqpVq9Tc3Kz09HTV19dr/Pjx2r17t6Kionw169atU3BwsObOnavm5mZNnTpVhYWFCgoK6up2APRh6enpSk9Pb3ddYWHhJbfNzs5WdnZ21zcFwHSuOMy8++67fq8tFsvXXmTCwsK0fv16rV+//krfHgAA9HM8mwkAAJgaYQYAAJgaYQYAAJgaYQYAAJgaYQYAAJgaYQYAAJgaYQYAAJgaYQYAAJgaYQYAAJgaYQYAAJgaYQYAAJgaYQYAAJgaYQYAAJgaYQYAAJgaYQYAAJgaYQYAAJgaYQYAAJgaYQYAAJgaYQYAAJgaYQYAAJgaYQYAAJgaYQYAAJgaYQYAAJgaYQYAAJgaYQYAAJgaYQYAAJgaYQYAAJgaYQYAAJgaYQYAAJgaYQYAAJgaYQYAAJgaYQYAAJgaYQYAAJhaQGFm06ZNuu222xQdHa3o6GhNmDBBv//9733rDcNQdna2HA6HwsPDNWXKFB07dsxvH16vV0uXLlVsbKwiIyM1e/ZsVVdXd83RAACAfiegMDNkyBD967/+qw4cOKADBw7oe9/7nr7//e/7AsvatWuVl5enDRs2aP/+/bLb7Zo2bZoaGxt9+8jIyFBxcbF27NihsrIyNTU1aebMmWptbe3aIwMAAP1CQGFm1qxZuuuuu3TTTTfppptu0jPPPKNrrrlGH374oQzDUH5+vrKysjRnzhwlJSVp69atOn36tLZv3y5J8ng8Kigo0HPPPaeUlBSNGTNGRUVFOnLkiPbs2dPh+3q9XjU0NPgtAAAA0hXcM9Pa2qodO3boq6++0oQJE1RRUSG3263U1FRfjdVq1eTJk7Vv3z5Jksvl0pkzZ/xqHA6HkpKSfDXtyc3Nlc1m8y3x8fGdbRsAAFxlAg4zR44c0TXXXCOr1aolS5aouLhYt9xyi9xutyQpLi7Orz4uLs63zu12KzQ0VAMHDuywpj2ZmZnyeDy+paqqKtC2AQDAVSo40A1uvvlmHT58WF9++aVee+01LVy4UKWlpb71FovFr94wjDZjF/u6GqvVKqvVGmirAACgHwh4ZiY0NFTf+MY3NG7cOOXm5ur222/XL3/5S9ntdklqM8NSW1vrm62x2+1qaWlRfX19hzUAAACBuOLfmTEMQ16vVwkJCbLb7SopKfGta2lpUWlpqZKTkyVJTqdTISEhfjU1NTU6evSorwYAACAQAX3M9NOf/lQzZsxQfHy8GhsbtWPHDr377rt6++23ZbFYlJGRoZycHCUmJioxMVE5OTmKiIjQ/PnzJUk2m01paWlasWKFYmJiNGjQIK1cuVKjR49WSkpKtxwgAAC4ugUUZr744gv95Cc/UU1NjWw2m2677Ta9/fbbmjZtmiRp1apVam5uVnp6uurr6zV+/Hjt3r1bUVFRvn2sW7dOwcHBmjt3rpqbmzV16lQVFhYqKCioa48MAAD0CwGFmYKCgkuut1gsys7OVnZ2doc1YWFhWr9+vdavXx/IWwMAALSLZzMBAABTI8wAAABTI8wAAABTI8wAAABTI8wAAABTI8wAAABTI8wAAABTI8wAAABTI8wAAABTI8wAAABTI8wAAABTI8wAAABTI8wAAABTI8wAAABTI8wAAABTC+7tBmBO9mssCv+qSqoL6rUewr+qkv0aS6+9PwCgbyDMoFMWO0M16kiedKT3ehj1//sAAPRvhBl0yq9dLZq3NFOjvjG813oo/+tn+vVzqzW71zoAAPQFhBl0irvJUHNkvBT7jV7robmmVe4mo9feH1du48aN+sUvfqGamhrdeuutys/P16RJk9qtff3117Vp0yYdPnxYXq9Xt956q7KzszV9+vQe7hpAX8MNwAB6xc6dO5WRkaGsrCwdOnRIkyZN0owZM1RZWdlu/Xvvvadp06Zp165dcrlc+u53v6tZs2bp0KFDPdw5gL6GmRkAvSIvL09paWlatGiRJCk/P1/vvPOONm3apNzc3Db1+fn5fq9zcnL05ptv6t///d81ZsyYdt/D6/XK6/X6Xjc0NHTdAQDoM5iZAdDjWlpa5HK5lJqa6jeempqqffv2XdY+zp07p8bGRg0aNKjDmtzcXNlsNt8SHx9/RX0D6JsIMwB6XF1dnVpbWxUXF+c3HhcXJ7fbfVn7eO655/TVV19p7ty5HdZkZmbK4/H4lqqqqivqG0DfxMdMAHqNxeL/O0GGYbQZa8+rr76q7Oxsvfnmm7r++us7rLNarbJarVfcJ4C+jTADoMfFxsYqKCiozSxMbW1tm9mai+3cuVNpaWn67W9/q5SUlO5sE4BJ8DETgB4XGhoqp9OpkpISv/GSkhIlJyd3uN2rr76q+++/X9u3b9fdd9/d3W0CMAlmZgD0iuXLl+snP/mJxo0bpwkTJujFF19UZWWllixZIun8/S5///vf9Zvf/EbS+SCzYMEC/fKXv9S3vvUt36xOeHi4bDZbrx0HgN5HmAHQK+bNm6dTp05pzZo1qqmpUVJSknbt2qVhw4ZJkmpqavx+c+bXv/61zp49q4ceekgPPfSQb3zhwoUqLCzs6fYB9CGEGQC9Jj09Xenp6e2uuzigvPvuu93fEABT4p4ZAABgaoQZAABgaoQZAABgagGFmdzcXH3zm99UVFSUrr/+et1zzz36y1/+4ldjGIays7PlcDgUHh6uKVOm6NixY341Xq9XS5cuVWxsrCIjIzV79mxVV1df+dEAAIB+J6AwU1paqoceekgffvihSkpKdPbsWaWmpuqrr77y1axdu1Z5eXnasGGD9u/fL7vdrmnTpqmxsdFXk5GRoeLiYu3YsUNlZWVqamrSzJkz1dra2nVHBgAA+oWAvs309ttv+73esmWLrr/+erlcLn3nO9+RYRjKz89XVlaW5syZI0naunWr4uLitH37di1evFgej0cFBQXatm2b79c7i4qKFB8frz179mj69Olt3pcn3wIAgI5c0T0zHo9HknxPra2oqJDb7fZ7Eq7VatXkyZN9T8J1uVw6c+aMX43D4VBSUlKHT8vlybcAAKAjnQ4zhmFo+fLl+va3v62kpCRJ8v0i56WehOt2uxUaGqqBAwd2WHMxnnwLAAA60ukfzXv44Yf13//93yorK2uzrjNPwr1UDU++BQAAHenUzMzSpUv11ltv6Y9//KOGDBniG7fb7ZJ0ySfh2u12tbS0qL6+vsMaAACAyxVQmDEMQw8//LBef/11/eEPf1BCQoLf+oSEBNntdr8n4ba0tKi0tNT3JFyn06mQkBC/mpqaGh09evSST8sFAABoT0AfMz300EPavn273nzzTUVFRflmYGw2m8LDw2WxWJSRkaGcnBwlJiYqMTFROTk5ioiI0Pz58321aWlpWrFihWJiYjRo0CCtXLlSo0eP9n27CQAA4HIFFGY2bdokSZoyZYrf+JYtW3T//fdLklatWqXm5malp6ervr5e48eP1+7duxUVFeWrX7dunYKDgzV37lw1Nzdr6tSpKiwsVFBQ0JUdDQAA6HcCCjOGYXxtjcViUXZ2trKzszusCQsL0/r167V+/fpA3h4AAKANns0EAABMjTADAABMjTADAABMjTADAABMjTADAABMjTADAABMjTADAABMjTADAABMjTADAABMjTADAABMLaDHGQD/V/mnJzu9bfP/ePVZtVvDh9gVHmbt8fcHAFw9CDMIWGxsrCIiInTfsjW93YoiIiIUGxvb220AAHoRYQYBGzp0qMrLy1VXV9fpfZSXl+u+++5TUVGRRo0a1en9xMbGaujQoZ3eHgBgfoQZdMrQoUO7JESMGjVKY8eO7YKOAAD9FTcAAwAAUyPMAAAAUyPMAAAAUyPMAAAAUyPMAAAAUyPMAAAAUyPMAAAAUyPMAAAAUyPMAAAAUyPMAAAAUyPMAAAAUyPMAAAAUyPMAAAAUyPMAAAAUyPMAAAAUyPMAAAAUyPMAAAAUyPMAAAAUws4zLz33nuaNWuWHA6HLBaL3njjDb/1hmEoOztbDodD4eHhmjJlio4dO+ZX4/V6tXTpUsXGxioyMlKzZ89WdXX1FR0IAADonwIOM1999ZVuv/12bdiwod31a9euVV5enjZs2KD9+/fLbrdr2rRpamxs9NVkZGSouLhYO3bsUFlZmZqamjRz5ky1trZ2/kgAAEC/FBzoBjNmzNCMGTPaXWcYhvLz85WVlaU5c+ZIkrZu3aq4uDht375dixcvlsfjUUFBgbZt26aUlBRJUlFRkeLj47Vnzx5Nnz69zX69Xq+8Xq/vdUNDQ6Bto4edPn1aH3/8cYfry8vL/f73UkaOHKmIiIgu6w0AcHUJOMxcSkVFhdxut1JTU31jVqtVkydP1r59+7R48WK5XC6dOXPGr8bhcCgpKUn79u1rN8zk5uZq9erVXdkqutnHH38sp9P5tXX33Xff19a4XC6NHTu2K9oCAFyFujTMuN1uSVJcXJzfeFxcnE6ePOmrCQ0N1cCBA9vUXNj+YpmZmVq+fLnvdUNDg+Lj47uydXSxkSNHyuVydbi+ublZn332mYYPH67w8PCv3RcAAB3p0jBzgcVi8XttGEabsYtdqsZqtcpqtXZZf+h+ERERXzubMnHixB7qBgBwNevSr2bb7XZJajPDUltb65utsdvtamlpUX19fYc1AAAAl6tLw0xCQoLsdrtKSkp8Yy0tLSotLVVycrIkyel0KiQkxK+mpqZGR48e9dUAAABcroA/ZmpqatJf//pX3+uKigodPnxYgwYN0tChQ5WRkaGcnBwlJiYqMTFROTk5ioiI0Pz58yVJNptNaWlpWrFihWJiYjRo0CCtXLlSo0eP9n27CQAA4HIFHGYOHDig7373u77XF27MXbhwoQoLC7Vq1So1NzcrPT1d9fX1Gj9+vHbv3q2oqCjfNuvWrVNwcLDmzp2r5uZmTZ06VYWFhQoKCuqCQwIAAP2JxTAMo7ebCFRDQ4NsNps8Ho+io6N7ux2gXzLjeWjGnoGrSXedgzybCQAAmBphBgAAmBphBgAAmBphBgAAmBphBgAAmBphBgAAmBphBkCv2bhxoxISEhQWFian06m9e/desr60tFROp1NhYWEaMWKEXnjhhR7qFEBfRpgB0Ct27typjIwMZWVl6dChQ5o0aZJmzJihysrKdusrKip01113adKkSTp06JB++tOfatmyZXrttdd6uHMAfY0pfzTP4/Ho2muvVVVVFT98BfSShoYGxcfH68svv5TNZgt4+/Hjx2vs2LHatGmTb2zUqFG65557lJub26b+8ccf11tvvaXy8nLf2JIlS/TRRx/pgw8+aPc9vF6vvF6v77XH49HQoUO5dgC95EqvGx0J+HEGfUFjY6MkKT4+vpc7AdDY2BjwRamlpUUul0tPPPGE33hqaqr27dvX7jYffPCBUlNT/camT5+ugoICnTlzRiEhIW22yc3N1erVq9uMc+0AetepU6cIMw6HQ1VVVYqKipLFYuntdtAJF9I5fyGbl2EYamxslMPhCHjburo6tba2Ki4uzm88Li5Obre73W3cbne79WfPnlVdXZ0GDx7cZpvMzEzf8+Mk6csvv9SwYcNUWVnZpRfS7mTGc4Wee4YZe74wOzpo0KAu3a8pw8yAAQM0ZMiQ3m4DXSA6Oto0JyHautJAcPEfI4ZhXPIPlPbq2xu/wGq1ymq1thm32Wym+/fOjOcKPfcMM/Y8YEDX3rLLDcAAelxsbKyCgoLazMLU1ta2mX25wG63t1sfHBysmJiYbusVQN9HmAHQ40JDQ+V0OlVSUuI3XlJSouTk5Ha3mTBhQpv63bt3a9y4ce3eLwOg/yDMoFdYrVY99dRT7X4EgP5h+fLlevnll7V582aVl5fr0UcfVWVlpZYsWSLp/P0uCxYs8NUvWbJEJ0+e1PLly1VeXq7NmzeroKBAK1euvOz3NOO/d/TcM+i5Z3RXz6b8ajaAq8PGjRu1du1a1dTUKCkpSevWrdN3vvMdSdL999+vzz77TO+++66vvrS0VI8++qiOHTsmh8Ohxx9/3Bd+APRfhBkAAGBqfMwEAABMjTADAABMjTADAABMjTADAABMjTCDHvXee+9p1qxZcjgcslgseuONN3q7JVxlNm7cqISEBIWFhcnpdGrv3r2XrC8tLZXT6VRYWJhGjBihF154oYc6/V+B9Pz6669r2rRpuu666xQdHa0JEybonXfe6cFuzwv0n/MF77//voKDg3XHHXd0b4PtCLRnr9errKwsDRs2TFarVTfeeKM2b97cQ92eF2jPr7zyim6//XZFRERo8ODBeuCBB3Tq1Kke6rZz1/guOQcNoAft2rXLyMrKMl577TVDklFcXNzbLeEqsmPHDiMkJMR46aWXjOPHjxuPPPKIERkZaZw8ebLd+hMnThgRERHGI488Yhw/ftx46aWXjJCQEON3v/tdn+35kUceMZ599lnjz3/+s/HJJ58YmZmZRkhIiHHw4ME+2/MFX375pTFixAgjNTXVuP3223um2f+vMz3Pnj3bGD9+vFFSUmJUVFQYf/rTn4z333+/z/a8d+9eY8CAAcYvf/lL48SJE8bevXuNW2+91bjnnnt6rOdAr/FddQ4SZtBrCDPoanfeeaexZMkSv7GRI0caTzzxRLv1q1atMkaOHOk3tnjxYuNb3/pWt/V4sUB7bs8tt9xirF69uqtb61Bne543b57xs5/9zHjqqad6PMwE2vPvf/97w2azGadOneqJ9toVaM+/+MUvjBEjRviNPf/888aQIUO6rcdLuZxrfFedg3zMBOCq0NLSIpfLpdTUVL/x1NRU7du3r91tPvjggzb106dP14EDB3TmzJlu6/WCzvR8sXPnzqmxsbHLn0Lckc72vGXLFv3tb3/TU0891d0tttGZnt966y2NGzdOa9eu1Q033KCbbrpJK1euVHNzc0+03Kmek5OTVV1drV27dskwDH3xxRf63e9+p7vvvrsnWu6UrjoHTfnUbAC4WF1dnVpbW9s8qDIuLq7NAyovcLvd7dafPXtWdXV1Gjx4cLf1K3Wu54s999xz+uqrrzR37tzuaLGNzvT86aef6oknntDevXsVHNzz/9npTM8nTpxQWVmZwsLCVFxcrLq6OqWnp+sf//hHj9w305mek5OT9corr2jevHn6n//5H509e1azZ8/W+vXru73fzuqqc5CZGQBXFYvF4vfaMIw2Y19X3954dwq05wteffVVZWdna+fOnbr++uu7q712XW7Pra2tmj9/vlavXq2bbrqpp9prVyD/nM+dOyeLxaJXXnlFd955p+666y7l5eWpsLCwx2ZnpMB6Pn78uJYtW6Ynn3xSLpdLb7/9tioqKvr8Iz+64hxkZgbAVSE2NlZBQUFt/mqtra1t85ffBXa7vd364OBgxcTEdFuvF3Sm5wt27typtLQ0/fa3v1VKSkp3tukn0J4bGxt14MABHTp0SA8//LCk80HBMAwFBwdr9+7d+t73vtenepakwYMH64YbbpDNZvONjRo1SoZhqLq6WomJiX2u59zcXE2cOFGPPfaYJOm2225TZGSkJk2apKeffrrbZxo7o6vOQWZmAFwVQkND5XQ6VVJS4jdeUlKi5OTkdreZMGFCm/rdu3dr3LhxCgkJ6bZeL+hMz9L5GZn7779f27dv7/H7IQLtOTo6WkeOHNHhw4d9y5IlS3TzzTfr8OHDGj9+fJ/rWZImTpyozz//XE1NTb6xTz75RAMGDNCQIUO6tV+pcz2fPn1aAwb4/2c9KChI0v/OdvQ1XXYOBnS7MHCFGhsbjUOHDhmHDh0yJBl5eXnGoUOHvvYrncDluPBV1oKCAuP48eNGRkaGERkZaXz22WeGYRjGE088YfzkJz/x1V/4Wuijjz5qHD9+3CgoKOi1r2Zfbs/bt283goODjV/96ldGTU2Nb/nyyy/7bM8X641vMwXac2NjozFkyBDjn/7pn4xjx44ZpaWlRmJiorFo0aI+2/OWLVuM4OBgY+PGjcbf/vY3o6yszBg3bpxx55139ljPX3eN765zkDCDHvXHP/7RkNRmWbhwYW+3hqvEr371K2PYsGFGaGioMXbsWKO0tNS3buHChcbkyZP96t99911jzJgxRmhoqDF8+HBj06ZNPdxxYD1Pnjy5T5xDgf5z/r96I8wYRuA9l5eXGykpKUZ4eLgxZMgQY/ny5cbp06f7dM/PP/+8ccsttxjh4eHG4MGDjXvvvdeorq7usX6/7hrfXeegxTD66NwTAADAZeCeGQAAYGqEGQAAYGqEGQAAYGqEGQAAYGqEGQAAYGqEGQAAYGqEGQAAYGqEGQAAYGqEGQAAYGqEGQAAYGqEGQAAYGr/DwnQO8Cys2sDAAAAAElFTkSuQmCC",
      "text/plain": [
       "<Figure size 640x480 with 2 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import matplotlib.pyplot as plt\n",
    "import numpy as np\n",
    "\n",
    "from matplotlib import cbook\n",
    "\n",
    "\n",
    "fig, (ax1, ax2) = plt.subplots(1, 2)\n",
    "\n",
    "# separate calculation of statistics and plotting\n",
    "ax1.bxp([parsintas], patch_artist=True, showfliers=False, boxprops={'facecolor': 'bisque'})\n",
    "#ax2.bxp([yess], patch_artist=True, boxprops={'facecolor': 'bisque'})\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "id": "8bd0a48d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0    338.16451\n",
      "Name: mean, dtype: float64\n"
     ]
    }
   ],
   "source": [
    "print(dfamas['mean'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "6c1cd467",
   "metadata": {},
   "outputs": [
    {
     "ename": "TypeError",
     "evalue": "Object of type int64 is not JSON serializable",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[17], line 4\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[38;5;28;01mimport\u001b[39;00m \u001b[38;5;21;01mjson\u001b[39;00m\n\u001b[1;32m      3\u001b[0m \u001b[38;5;28;01mwith\u001b[39;00m \u001b[38;5;28mopen\u001b[39m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mdata.json\u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mw\u001b[39m\u001b[38;5;124m\"\u001b[39m) \u001b[38;5;28;01mas\u001b[39;00m file:\n\u001b[0;32m----> 4\u001b[0m     json\u001b[38;5;241m.\u001b[39mdump(stats, file)\n",
      "File \u001b[0;32m/opt/anaconda3/lib/python3.12/json/__init__.py:179\u001b[0m, in \u001b[0;36mdump\u001b[0;34m(obj, fp, skipkeys, ensure_ascii, check_circular, allow_nan, cls, indent, separators, default, sort_keys, **kw)\u001b[0m\n\u001b[1;32m    173\u001b[0m     iterable \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mcls\u001b[39m(skipkeys\u001b[38;5;241m=\u001b[39mskipkeys, ensure_ascii\u001b[38;5;241m=\u001b[39mensure_ascii,\n\u001b[1;32m    174\u001b[0m         check_circular\u001b[38;5;241m=\u001b[39mcheck_circular, allow_nan\u001b[38;5;241m=\u001b[39mallow_nan, indent\u001b[38;5;241m=\u001b[39mindent,\n\u001b[1;32m    175\u001b[0m         separators\u001b[38;5;241m=\u001b[39mseparators,\n\u001b[1;32m    176\u001b[0m         default\u001b[38;5;241m=\u001b[39mdefault, sort_keys\u001b[38;5;241m=\u001b[39msort_keys, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkw)\u001b[38;5;241m.\u001b[39miterencode(obj)\n\u001b[1;32m    177\u001b[0m \u001b[38;5;66;03m# could accelerate with writelines in some versions of Python, at\u001b[39;00m\n\u001b[1;32m    178\u001b[0m \u001b[38;5;66;03m# a debuggability cost\u001b[39;00m\n\u001b[0;32m--> 179\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m chunk \u001b[38;5;129;01min\u001b[39;00m iterable:\n\u001b[1;32m    180\u001b[0m     fp\u001b[38;5;241m.\u001b[39mwrite(chunk)\n",
      "File \u001b[0;32m/opt/anaconda3/lib/python3.12/json/encoder.py:430\u001b[0m, in \u001b[0;36m_make_iterencode.<locals>._iterencode\u001b[0;34m(o, _current_indent_level)\u001b[0m\n\u001b[1;32m    428\u001b[0m     \u001b[38;5;28;01myield\u001b[39;00m _floatstr(o)\n\u001b[1;32m    429\u001b[0m \u001b[38;5;28;01melif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(o, (\u001b[38;5;28mlist\u001b[39m, \u001b[38;5;28mtuple\u001b[39m)):\n\u001b[0;32m--> 430\u001b[0m     \u001b[38;5;28;01myield from\u001b[39;00m _iterencode_list(o, _current_indent_level)\n\u001b[1;32m    431\u001b[0m \u001b[38;5;28;01melif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(o, \u001b[38;5;28mdict\u001b[39m):\n\u001b[1;32m    432\u001b[0m     \u001b[38;5;28;01myield from\u001b[39;00m _iterencode_dict(o, _current_indent_level)\n",
      "File \u001b[0;32m/opt/anaconda3/lib/python3.12/json/encoder.py:326\u001b[0m, in \u001b[0;36m_make_iterencode.<locals>._iterencode_list\u001b[0;34m(lst, _current_indent_level)\u001b[0m\n\u001b[1;32m    324\u001b[0m         \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m    325\u001b[0m             chunks \u001b[38;5;241m=\u001b[39m _iterencode(value, _current_indent_level)\n\u001b[0;32m--> 326\u001b[0m         \u001b[38;5;28;01myield from\u001b[39;00m chunks\n\u001b[1;32m    327\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m newline_indent \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[1;32m    328\u001b[0m     _current_indent_level \u001b[38;5;241m-\u001b[39m\u001b[38;5;241m=\u001b[39m \u001b[38;5;241m1\u001b[39m\n",
      "File \u001b[0;32m/opt/anaconda3/lib/python3.12/json/encoder.py:406\u001b[0m, in \u001b[0;36m_make_iterencode.<locals>._iterencode_dict\u001b[0;34m(dct, _current_indent_level)\u001b[0m\n\u001b[1;32m    404\u001b[0m         \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m    405\u001b[0m             chunks \u001b[38;5;241m=\u001b[39m _iterencode(value, _current_indent_level)\n\u001b[0;32m--> 406\u001b[0m         \u001b[38;5;28;01myield from\u001b[39;00m chunks\n\u001b[1;32m    407\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m newline_indent \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[1;32m    408\u001b[0m     _current_indent_level \u001b[38;5;241m-\u001b[39m\u001b[38;5;241m=\u001b[39m \u001b[38;5;241m1\u001b[39m\n",
      "File \u001b[0;32m/opt/anaconda3/lib/python3.12/json/encoder.py:439\u001b[0m, in \u001b[0;36m_make_iterencode.<locals>._iterencode\u001b[0;34m(o, _current_indent_level)\u001b[0m\n\u001b[1;32m    437\u001b[0m         \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mValueError\u001b[39;00m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mCircular reference detected\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[1;32m    438\u001b[0m     markers[markerid] \u001b[38;5;241m=\u001b[39m o\n\u001b[0;32m--> 439\u001b[0m o \u001b[38;5;241m=\u001b[39m _default(o)\n\u001b[1;32m    440\u001b[0m \u001b[38;5;28;01myield from\u001b[39;00m _iterencode(o, _current_indent_level)\n\u001b[1;32m    441\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m markers \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n",
      "File \u001b[0;32m/opt/anaconda3/lib/python3.12/json/encoder.py:180\u001b[0m, in \u001b[0;36mJSONEncoder.default\u001b[0;34m(self, o)\u001b[0m\n\u001b[1;32m    161\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mdefault\u001b[39m(\u001b[38;5;28mself\u001b[39m, o):\n\u001b[1;32m    162\u001b[0m \u001b[38;5;250m    \u001b[39m\u001b[38;5;124;03m\"\"\"Implement this method in a subclass such that it returns\u001b[39;00m\n\u001b[1;32m    163\u001b[0m \u001b[38;5;124;03m    a serializable object for ``o``, or calls the base implementation\u001b[39;00m\n\u001b[1;32m    164\u001b[0m \u001b[38;5;124;03m    (to raise a ``TypeError``).\u001b[39;00m\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m    178\u001b[0m \n\u001b[1;32m    179\u001b[0m \u001b[38;5;124;03m    \"\"\"\u001b[39;00m\n\u001b[0;32m--> 180\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mTypeError\u001b[39;00m(\u001b[38;5;124mf\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mObject of type \u001b[39m\u001b[38;5;132;01m{\u001b[39;00mo\u001b[38;5;241m.\u001b[39m\u001b[38;5;18m__class__\u001b[39m\u001b[38;5;241m.\u001b[39m\u001b[38;5;18m__name__\u001b[39m\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m \u001b[39m\u001b[38;5;124m'\u001b[39m\n\u001b[1;32m    181\u001b[0m                     \u001b[38;5;124mf\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mis not JSON serializable\u001b[39m\u001b[38;5;124m'\u001b[39m)\n",
      "\u001b[0;31mTypeError\u001b[0m: Object of type int64 is not JSON serializable"
     ]
    }
   ],
   "source": [
    "import json\n",
    "\n",
    "with open(\"data.json\", \"w\") as file:\n",
    "    json.dump(stats, file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "0558c7ce",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0\n",
      "BINS  IntervalIndex([  (3000018, 3000018],   (3000018, 3000118],\n",
      "                 (3000118, 3000218],   (3000218, 3000318],\n",
      "                 (3000318, 3000418],   (3000418, 3000518],\n",
      "                 (3000518, 3000618],   (3000618, 3000718],\n",
      "                 (3000718, 3000818],   (3000818, 3000918],\n",
      "               ...\n",
      "               (18260718, 18260818], (18260818, 18260918],\n",
      "               (18260918, 18261018], (18261018, 18261118],\n",
      "               (18261118, 18261218], (18261218, 18261318],\n",
      "               (18261318, 18261418], (18261418, 18261518],\n",
      "               (18261518, 18261618], (18261618, 18261710]],\n",
      "              dtype='interval[int64, right]', length=152618)\n",
      "HEAD with BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0   \n",
      "\n",
      "         bin_offset_0  target_count  \n",
      "0  (3000018, 3000118]             1  \n",
      "1  (3000018, 3000118]             1  \n",
      "2  (3000018, 3000118]             1  \n",
      "3  (3000018, 3000118]             1  \n",
      "4  (3000018, 3000118]             1  \n",
      "TAIL with BINS\n",
      "       seqnames     start       end        CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  18261611  18261611  18_18261611    0.0    0.0    0.0   \n",
      "699996       18  18261615  18261615  18_18261615    0.0    0.0    0.0   \n",
      "699997       18  18261682  18261682  18_18261682    0.0    0.0    0.0   \n",
      "699998       18  18261702  18261702  18_18261702    4.0    0.0    0.0   \n",
      "699999       18  18261710  18261710  18_18261710    0.0    0.0    0.0   \n",
      "\n",
      "                bin_offset_0  target_count  \n",
      "699995  (18261518, 18261618]             1  \n",
      "699996  (18261518, 18261618]             1  \n",
      "699997  (18261618, 18261710]             1  \n",
      "699998  (18261618, 18261710]             1  \n",
      "699999  (18261618, 18261710]             1  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:137: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "          bin_offset_0  TT_S0  target_count  starting_nt\n",
      "0   (3000018, 3000018]    0.0             0      3000018\n",
      "1   (3000018, 3000118]    7.0            11      3000018\n",
      "2   (3000118, 3000218]    1.0             4      3000118\n",
      "3   (3000218, 3000318]    2.0             4      3000218\n",
      "4   (3000318, 3000418]    0.0             4      3000318\n",
      "5   (3000418, 3000518]    0.0             2      3000418\n",
      "6   (3000518, 3000618]    0.0             2      3000518\n",
      "7   (3000618, 3000718]    0.0             4      3000618\n",
      "8   (3000718, 3000818]    0.0             7      3000718\n",
      "9   (3000818, 3000918]    0.0             3      3000818\n",
      "10  (3000918, 3001018]    0.0             3      3000918\n",
      "11  (3001018, 3001118]    2.0             6      3001018\n",
      "12  (3001118, 3001218]    0.0             2      3001118\n",
      "13  (3001218, 3001318]    0.0             4      3001218\n",
      "14  (3001318, 3001418]    0.0             3      3001318\n",
      "15  (3001418, 3001518]    1.0             4      3001418\n",
      "16  (3001518, 3001618]    3.0            10      3001518\n",
      "17  (3001618, 3001718]    3.0             7      3001618\n",
      "18  (3001718, 3001818]    0.0             6      3001718\n",
      "19  (3001818, 3001918]    0.0             2      3001818\n",
      "20  (3001918, 3002018]    1.0             3      3001918\n",
      "21  (3002018, 3002118]    0.0             2      3002018\n",
      "22  (3002118, 3002218]    1.0             4      3002118\n",
      "23  (3002218, 3002318]   10.0             6      3002218\n",
      "24  (3002318, 3002418]    2.0             1      3002318\n",
      "25  (3002418, 3002518]   19.0             5      3002418\n",
      "26  (3002518, 3002618]    4.0             2      3002518\n",
      "27  (3002618, 3002718]    0.0             3      3002618\n",
      "28  (3002718, 3002818]    0.0             1      3002718\n",
      "29  (3002818, 3002918]    0.0             2      3002818\n",
      "30  (3002918, 3003018]    1.0             6      3002918\n",
      "31  (3003018, 3003118]    2.0             4      3003018\n",
      "32  (3003118, 3003218]    0.0             0      3003118\n",
      "33  (3003218, 3003318]    0.0             1      3003218\n",
      "34  (3003318, 3003418]    1.0             8      3003318\n",
      "35  (3003418, 3003518]    0.0             3      3003418\n",
      "36  (3003518, 3003618]    1.0             2      3003518\n",
      "37  (3003618, 3003718]    3.0             4      3003618\n",
      "38  (3003718, 3003818]    1.0             6      3003718\n",
      "39  (3003818, 3003918]    0.0             3      3003818\n",
      "40  (3003918, 3004018]    8.0             6      3003918\n",
      "41  (3004018, 3004118]    0.0             0      3004018\n",
      "42  (3004118, 3004218]    0.0             1      3004118\n",
      "43  (3004218, 3004318]    5.0             6      3004218\n",
      "44  (3004318, 3004418]    0.0             1      3004318\n",
      "45  (3004418, 3004518]    0.0             1      3004418\n",
      "46  (3004518, 3004618]    1.0             7      3004518\n",
      "47  (3004618, 3004718]    0.0             0      3004618\n",
      "48  (3004718, 3004818]    0.0             1      3004718\n",
      "49  (3004818, 3004918]   13.0             8      3004818\n",
      "(152618, 4)\n",
      "bin_offset_0    category\n",
      "TT_S0            float64\n",
      "target_count       int64\n",
      "starting_nt        int64\n",
      "dtype: object\n",
      "sumos ilgis: 152618\n",
      "ranges ilgis: 152618\n",
      "Model input - NO SCALE\n",
      "        TT_S0  genes  transcripts  exons  CDSs\n",
      "0         0.0      0            0      0     0\n",
      "1         7.0      0            0      0     0\n",
      "2         1.0      0            0      0     0\n",
      "3         2.0      0            0      0     0\n",
      "4         0.0      0            0      0     0\n",
      "...       ...    ...          ...    ...   ...\n",
      "152613   10.0      0            0      0     0\n",
      "152614    3.0      0            0      0     0\n",
      "152615    4.0      0            0      0     0\n",
      "152616    0.0      0            0      0     0\n",
      "152617    4.0      0            0      0     0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "        TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0         0.0             0                   0             0            0\n",
      "1         7.0             0                   0             0            0\n",
      "2         1.0             0                   0             0            0\n",
      "3         2.0             0                   0             0            0\n",
      "4         0.0             0                   0             0            0\n",
      "...       ...           ...                 ...           ...          ...\n",
      "152613   10.0             0                   0             0            0\n",
      "152614    3.0             0                   0             0            0\n",
      "152615    4.0             0                   0             0            0\n",
      "152616    0.0             0                   0             0            0\n",
      "152617    4.0             0                   0             0            0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "0.5721802146535795\n",
      "[1 0 1 5 1 1 1 1 1 1 1 5 1 1 1 1 5 5 1 1 1 1 1 7 5 4 5 1 1 1 1 5 1 1 1 1 1\n",
      " 5 1 1 0 1 1 9 1 1 1 1 1 7 1 2 4 3 1 4 7 8 8 6 6 1 4 4 5 1 1 0 1 1 9 5 1 9\n",
      " 1 1 5 1 1 1 5 0 5 1 5 5 5 5 9 1 1 7 7 5 5 1 5 1 5 7 7 7 2 2 7 0 9 7 2 5 7\n",
      " 1 4 5 1 1 5 1 1 1 1 1 9 1 1 7 1 4 4 7 4 1 0 0 5 1 9 0 5 1 1 1 1 1 5 1 1 7\n",
      " 0 1 7 5 1 1 9 1 7 5 5 1 4 1 7 1 1 7 7 5 1 1 1 1 1 1 5 1 1 1 1 1 9 0 5 5 1\n",
      " 1 5 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 7 4 1 1 1 1 5 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 5 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 5 1 1\n",
      " 1 1 1 1 1 1 5 1 1 5 1 5 1 5 5 5 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 9 1 1 1 1 5 5 1 5 1 1 1 1 5 1 9 0 5 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 9 1 5 1 7 1 1 1 1 1 1 1 1 1 7 4 1 7 4 0 0 5 1 5 9 0\n",
      " 5 1 9 1 1 1 7 9 7 5 7 9 1 5 1 1 5 1 9 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 5 1 1 1 9 1 1 1 1 1 5 1 1 1 1 1 1 1 5 0 1 1 1 9 9 5 1 5 9 9 5 0 5 5 5 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 7 9 1 1 1 9 1 1 0 1 5 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 5 1 5 5 1 1 1 1 1 1 1 1 1 1 1 1 1 5 1 1 1 0 1 1 1 1 5 9 1 1 1 1 9\n",
      " 1 1 5 1 1 1 1 1 1 5 0 1 1 5 0 5 1 1 1 1 1 9 1 5 1 1 1 1 1 5 9 1 1 9 5 1 0\n",
      " 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 5 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 5 5 0 5 5 5 1 5 1 1 1 1 1 1 1 1 5 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 5 1 1 1 1 5 5 1 1 1\n",
      " 1 1 1 1 1 1 1 5 1 1 1 1 1 5 5 1 1 1 1 1 5 1 1 5 1 1 1 5 0 9 7 1 5 1 5 5 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 5 1 1 1 5 1 1 5 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 5 1 1 1 1 5 1 5 9 1 1 1 1 1 5 5 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 5 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 5 1 1 1 1 5 9 1 1 1 1 1 1 1 1 1\n",
      " 9 5 9 9 5 5 1 1 1 1 1 1 1 1 1 5 1 1 1 5 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 5 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 5 1 9 5 1 1 1 1 1 1\n",
      " 1]\n",
      "threshold density 1.95819\n",
      "cluster signal 16270.0\n",
      "cluster target count 300673\n",
      "!!cluster density 0.054111942209643035\n",
      "cluster signal 77249.0\n",
      "cluster target count 64504\n",
      "!!cluster density 1.1975846459134318\n",
      "cluster signal 87190.0\n",
      "cluster target count 157019\n",
      "!!cluster density 0.5552831186034811\n",
      "cluster signal 89658.0\n",
      "cluster target count 57682\n",
      "!!cluster density 1.554349710481606\n",
      "cluster signal 69119.0\n",
      "cluster target count 33742\n",
      "!!cluster density 2.048455930294588\n",
      "cluster signal 62211.0\n",
      "cluster target count 67967\n",
      "!!cluster density 0.9153118425118072\n",
      "cluster signal 37405.0\n",
      "cluster target count 13932\n",
      "!!cluster density 2.6848262991673844\n",
      "cluster signal 14601.0\n",
      "cluster target count 4062\n",
      "!!cluster density 3.5945347119645494\n",
      "cluster signal 2544.0\n",
      "cluster target count 408\n",
      "!!cluster density 6.235294117647059\n",
      "cluster signal 664.0\n",
      "cluster target count 11\n",
      "!!cluster density 60.36363636363637\n",
      "\n",
      "Labels mapping\n",
      " {1: 0, 0: 0, 5: 0, 7: 0, 4: 1, 9: 0, 2: 1, 3: 1, 8: 1, 6: 1}\n",
      "DEBUG: 146987 5631 \n",
      "\n",
      " +----------------------+-------------+----------------+-------+\n",
      "| bin_offset_0         | TT_S0       | target_count   | +9    |\n",
      "| (category)           | (float64)   | (int64)        | ...   |\n",
      "|----------------------+-------------+----------------+-------|\n",
      "| (3000018, 3000018]   | 0.0         | 0              | ...   |\n",
      "| (3000018, 3000118]   | 7.0         | 11             | ...   |\n",
      "| (3000118, 3000218]   | 1.0         | 4              | ...   |\n",
      "| (3000218, 3000318]   | 2.0         | 4              | ...   |\n",
      "| ...                  | ...         | ...            | ...   |\n",
      "| (18261318, 18261418] | 3.0         | 15             | ...   |\n",
      "| (18261418, 18261518] | 4.0         | 4              | ...   |\n",
      "| (18261518, 18261618] | 0.0         | 3              | ...   |\n",
      "| (18261618, 18261710] | 4.0         | 3              | ...   |\n",
      "+----------------------+-------------+----------------+-------+\n",
      "Unstranded PyRanges object has 146,987 rows and 12 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "9 hidden columns: starting_nt, Chromosome, Start, End, genes, transcripts, ... (+ 3 more.)\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0\n",
      "BINS  IntervalIndex([  (3000018, 3000023],   (3000023, 3000123],\n",
      "                 (3000123, 3000223],   (3000223, 3000323],\n",
      "                 (3000323, 3000423],   (3000423, 3000523],\n",
      "                 (3000523, 3000623],   (3000623, 3000723],\n",
      "                 (3000723, 3000823],   (3000823, 3000923],\n",
      "               ...\n",
      "               (18260723, 18260823], (18260823, 18260923],\n",
      "               (18260923, 18261023], (18261023, 18261123],\n",
      "               (18261123, 18261223], (18261223, 18261323],\n",
      "               (18261323, 18261423], (18261423, 18261523],\n",
      "               (18261523, 18261623], (18261623, 18261710]],\n",
      "              dtype='interval[int64, right]', length=152618)\n",
      "HEAD with BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0   \n",
      "\n",
      "         bin_offset_5  target_count  \n",
      "0  (3000018, 3000023]             1  \n",
      "1  (3000018, 3000023]             1  \n",
      "2  (3000023, 3000123]             1  \n",
      "3  (3000023, 3000123]             1  \n",
      "4  (3000023, 3000123]             1  \n",
      "TAIL with BINS\n",
      "       seqnames     start       end        CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  18261611  18261611  18_18261611    0.0    0.0    0.0   \n",
      "699996       18  18261615  18261615  18_18261615    0.0    0.0    0.0   \n",
      "699997       18  18261682  18261682  18_18261682    0.0    0.0    0.0   \n",
      "699998       18  18261702  18261702  18_18261702    4.0    0.0    0.0   \n",
      "699999       18  18261710  18261710  18_18261710    0.0    0.0    0.0   \n",
      "\n",
      "                bin_offset_5  target_count  \n",
      "699995  (18261523, 18261623]             1  \n",
      "699996  (18261523, 18261623]             1  \n",
      "699997  (18261623, 18261710]             1  \n",
      "699998  (18261623, 18261710]             1  \n",
      "699999  (18261623, 18261710]             1  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:137: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "          bin_offset_5  TT_S0  target_count starting_nt\n",
      "0   (3000018, 3000023]    0.0             2     3000018\n",
      "1   (3000023, 3000123]    7.0             9     3000023\n",
      "2   (3000123, 3000223]    1.0             5     3000123\n",
      "3   (3000223, 3000323]    2.0             3     3000223\n",
      "4   (3000323, 3000423]    0.0             4     3000323\n",
      "5   (3000423, 3000523]    0.0             2     3000423\n",
      "6   (3000523, 3000623]    0.0             2     3000523\n",
      "7   (3000623, 3000723]    0.0             4     3000623\n",
      "8   (3000723, 3000823]    0.0             7     3000723\n",
      "9   (3000823, 3000923]    0.0             4     3000823\n",
      "10  (3000923, 3001023]    0.0             2     3000923\n",
      "11  (3001023, 3001123]    2.0             6     3001023\n",
      "12  (3001123, 3001223]    0.0             2     3001123\n",
      "13  (3001223, 3001323]    0.0             5     3001223\n",
      "14  (3001323, 3001423]    0.0             2     3001323\n",
      "15  (3001423, 3001523]    1.0             6     3001423\n",
      "16  (3001523, 3001623]    3.0             8     3001523\n",
      "17  (3001623, 3001723]    3.0             8     3001623\n",
      "18  (3001723, 3001823]    0.0             5     3001723\n",
      "19  (3001823, 3001923]    0.0             3     3001823\n",
      "20  (3001923, 3002023]    1.0             2     3001923\n",
      "21  (3002023, 3002123]    0.0             2     3002023\n",
      "22  (3002123, 3002223]    1.0             4     3002123\n",
      "23  (3002223, 3002323]   10.0             6     3002223\n",
      "24  (3002323, 3002423]   12.0             2     3002323\n",
      "25  (3002423, 3002523]    9.0             5     3002423\n",
      "26  (3002523, 3002623]    4.0             1     3002523\n",
      "27  (3002623, 3002723]    0.0             3     3002623\n",
      "28  (3002723, 3002823]    0.0             1     3002723\n",
      "29  (3002823, 3002923]    0.0             2     3002823\n",
      "30  (3002923, 3003023]    1.0             6     3002923\n",
      "31  (3003023, 3003123]    2.0             4     3003023\n",
      "32  (3003123, 3003223]    0.0             0     3003123\n",
      "33  (3003223, 3003323]    0.0             2     3003223\n",
      "34  (3003323, 3003423]    1.0             7     3003323\n",
      "35  (3003423, 3003523]    0.0             3     3003423\n",
      "36  (3003523, 3003623]    1.0             2     3003523\n",
      "37  (3003623, 3003723]    3.0             4     3003623\n",
      "38  (3003723, 3003823]    1.0             6     3003723\n",
      "39  (3003823, 3003923]    1.0             4     3003823\n",
      "40  (3003923, 3004023]    7.0             5     3003923\n",
      "41  (3004023, 3004123]    0.0             0     3004023\n",
      "42  (3004123, 3004223]    0.0             2     3004123\n",
      "43  (3004223, 3004323]    5.0             5     3004223\n",
      "44  (3004323, 3004423]    0.0             1     3004323\n",
      "45  (3004423, 3004523]    0.0             2     3004423\n",
      "46  (3004523, 3004623]    1.0             6     3004523\n",
      "47  (3004623, 3004723]    0.0             0     3004623\n",
      "48  (3004723, 3004823]    0.0             1     3004723\n",
      "49  (3004823, 3004923]   13.0             8     3004823\n",
      "(152618, 4)\n",
      "bin_offset_5    category\n",
      "TT_S0            float64\n",
      "target_count       int64\n",
      "starting_nt     category\n",
      "dtype: object\n",
      "sumos ilgis: 152618\n",
      "ranges ilgis: 152618\n",
      "Model input - NO SCALE\n",
      "        TT_S0  genes  transcripts  exons  CDSs\n",
      "0         0.0      0            0      0     0\n",
      "1         7.0      0            0      0     0\n",
      "2         1.0      0            0      0     0\n",
      "3         2.0      0            0      0     0\n",
      "4         0.0      0            0      0     0\n",
      "...       ...    ...          ...    ...   ...\n",
      "152613   10.0      0            0      0     0\n",
      "152614    3.0      0            0      0     0\n",
      "152615    4.0      0            0      0     0\n",
      "152616    0.0      0            0      0     0\n",
      "152617    4.0      0            0      0     0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "        TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0         0.0             0                   0             0            0\n",
      "1         7.0             0                   0             0            0\n",
      "2         1.0             0                   0             0            0\n",
      "3         2.0             0                   0             0            0\n",
      "4         0.0             0                   0             0            0\n",
      "...       ...           ...                 ...           ...          ...\n",
      "152613   10.0             0                   0             0            0\n",
      "152614    3.0             0                   0             0            0\n",
      "152615    4.0             0                   0             0            0\n",
      "152616    0.0             0                   0             0            0\n",
      "152617    4.0             0                   0             0            0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "0.0744866267412756\n",
      "[4 7 4 0 4 4 4 4 4 4 4 0 4 4 4 4 0 0 4 4 4 4 4 1 1 1 0 4 4 4 4 0 4 4 4 4 4\n",
      " 0 4 4 7 4 4 7 4 4 4 4 4 8 4 9 3 2 4 3 8 6 6 5 5 4 8 8 0 4 4 7 4 4 7 0 4 7\n",
      " 4 4 0 4 4 4 0 1 0 4 0 0 0 0 7 4 4 1 8 0 0 4 0 4 0 1 1 8 9 3 8 1 7 1 9 0 1\n",
      " 4 8 0 4 4 0 4 4 4 4 4 7 4 4 8 4 8 8 1 8 4 7 1 0 4 7 1 0 4 4 4 4 4 0 4 4 1\n",
      " 1 7 0 0 4 4 7 0 1 0 0 4 8 4 8 4 4 1 1 0 4 4 4 4 4 4 0 4 4 4 4 4 7 7 0 0 4\n",
      " 4 0 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 1 3 4 4 4 0 4 4 4 4 4 4 4 4 4\n",
      " 4 4 4 4 4 4 4 4 4 4 4 4 4 4 0 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 0 4 4\n",
      " 4 4 4 4 4 4 0 4 4 0 4 0 4 0 0 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4\n",
      " 7 4 4 4 4 0 0 4 0 4 4 4 4 0 4 7 7 0 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4\n",
      " 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4\n",
      " 4 4 4 4 4 4 4 4 4 4 4 7 4 0 4 1 4 4 4 4 4 4 4 4 4 8 8 4 1 8 1 7 0 0 4 7 7\n",
      " 0 4 7 0 4 4 8 7 8 0 8 0 4 0 4 4 0 4 7 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4\n",
      " 0 4 4 4 7 4 4 4 4 4 0 4 4 4 4 4 0 4 0 1 4 4 4 7 7 0 4 0 7 7 0 1 0 0 0 4 4\n",
      " 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4\n",
      " 7 1 7 4 4 4 7 4 4 1 0 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4\n",
      " 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4\n",
      " 4 4 4 4 0 4 0 0 4 4 4 4 4 4 4 4 4 4 4 4 4 0 0 4 4 7 4 4 4 4 0 7 4 4 4 4 7\n",
      " 4 4 0 4 4 4 4 4 4 0 1 4 4 0 7 0 4 4 4 4 4 7 4 0 4 4 4 4 4 1 4 4 4 7 0 4 7\n",
      " 0 7 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 0 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4\n",
      " 4 4 4 4 4 4 4 4 4 4 4 4 4 0 0 7 0 0 0 4 0 4 4 4 4 4 4 4 4 0 4 4 4 4 4 4 4\n",
      " 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 0 4 4 4 4 0 0 4 4 4\n",
      " 4 4 4 4 4 4 4 0 4 4 4 4 4 0 0 4 4 4 4 0 4 4 0 0 4 4 4 0 1 7 1 4 0 4 0 0 4\n",
      " 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 0 4 4 4 0 4 4 0 4 4 4 4\n",
      " 4 4 4 4 4 4 4 4 4 4 4 0 4 4 4 4 0 4 0 7 4 4 4 4 4 0 0 4 4 4 4 4 4 4 4 4 4\n",
      " 4 4 0 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 0 4 4 4 4 0 7 4 4 4 4 4 4 4 4 4\n",
      " 7 0 7 7 0 0 4 4 4 4 4 4 4 4 4 0 4 4 4 0 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4\n",
      " 4 4 4 4 4 4 4 4 0 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 0 4 7 0 4 4 4 4 4 4\n",
      " 4]\n",
      "threshold density 1.95819\n",
      "cluster signal 16194.0\n",
      "cluster target count 300236\n",
      "!!cluster density 0.05393756911229833\n",
      "cluster signal 89658.0\n",
      "cluster target count 92609\n",
      "!!cluster density 0.9681348465051993\n",
      "cluster signal 87589.0\n",
      "cluster target count 157669\n",
      "!!cluster density 0.5555245482624993\n",
      "cluster signal 109287.0\n",
      "cluster target count 79785\n",
      "!!cluster density 1.3697687535250986\n",
      "cluster signal 64283.0\n",
      "cluster target count 35287\n",
      "!!cluster density 1.8217190466744126\n",
      "cluster signal 27899.0\n",
      "cluster target count 10120\n",
      "!!cluster density 2.756818181818182\n",
      "cluster signal 44002.0\n",
      "cluster target count 19713\n",
      "!!cluster density 2.23213108101253\n",
      "cluster signal 13432.0\n",
      "cluster target count 3828\n",
      "!!cluster density 3.5088819226750263\n",
      "cluster signal 3903.0\n",
      "cluster target count 742\n",
      "!!cluster density 5.2601078167115904\n",
      "cluster signal 664.0\n",
      "cluster target count 11\n",
      "!!cluster density 60.36363636363637\n",
      "\n",
      "Labels mapping\n",
      " {4: 0, 7: 0, 0: 0, 1: 0, 8: 0, 9: 1, 3: 1, 2: 1, 6: 1, 5: 1}\n",
      "DEBUG: 149133 3485 \n",
      "\n",
      " +----------------------+-------------+----------------+-------+\n",
      "| bin_offset_5         | TT_S0       | target_count   | +9    |\n",
      "| (category)           | (float64)   | (int64)        | ...   |\n",
      "|----------------------+-------------+----------------+-------|\n",
      "| (3000018, 3000023]   | 0.0         | 2              | ...   |\n",
      "| (3000023, 3000123]   | 7.0         | 9              | ...   |\n",
      "| (3000123, 3000223]   | 1.0         | 5              | ...   |\n",
      "| (3000223, 3000323]   | 2.0         | 3              | ...   |\n",
      "| ...                  | ...         | ...            | ...   |\n",
      "| (18261323, 18261423] | 3.0         | 14             | ...   |\n",
      "| (18261423, 18261523] | 4.0         | 4              | ...   |\n",
      "| (18261523, 18261623] | 0.0         | 3              | ...   |\n",
      "| (18261623, 18261710] | 4.0         | 3              | ...   |\n",
      "+----------------------+-------------+----------------+-------+\n",
      "Unstranded PyRanges object has 149,133 rows and 12 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "9 hidden columns: starting_nt, Chromosome, Start, End, genes, transcripts, ... (+ 3 more.)\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0\n",
      "BINS  IntervalIndex([  (3000018, 3000028],   (3000028, 3000128],\n",
      "                 (3000128, 3000228],   (3000228, 3000328],\n",
      "                 (3000328, 3000428],   (3000428, 3000528],\n",
      "                 (3000528, 3000628],   (3000628, 3000728],\n",
      "                 (3000728, 3000828],   (3000828, 3000928],\n",
      "               ...\n",
      "               (18260728, 18260828], (18260828, 18260928],\n",
      "               (18260928, 18261028], (18261028, 18261128],\n",
      "               (18261128, 18261228], (18261228, 18261328],\n",
      "               (18261328, 18261428], (18261428, 18261528],\n",
      "               (18261528, 18261628], (18261628, 18261710]],\n",
      "              dtype='interval[int64, right]', length=152618)\n",
      "HEAD with BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0   \n",
      "\n",
      "        bin_offset_10  target_count  \n",
      "0  (3000018, 3000028]             1  \n",
      "1  (3000018, 3000028]             1  \n",
      "2  (3000018, 3000028]             1  \n",
      "3  (3000028, 3000128]             1  \n",
      "4  (3000028, 3000128]             1  \n",
      "TAIL with BINS\n",
      "       seqnames     start       end        CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  18261611  18261611  18_18261611    0.0    0.0    0.0   \n",
      "699996       18  18261615  18261615  18_18261615    0.0    0.0    0.0   \n",
      "699997       18  18261682  18261682  18_18261682    0.0    0.0    0.0   \n",
      "699998       18  18261702  18261702  18_18261702    4.0    0.0    0.0   \n",
      "699999       18  18261710  18261710  18_18261710    0.0    0.0    0.0   \n",
      "\n",
      "               bin_offset_10  target_count  \n",
      "699995  (18261528, 18261628]             1  \n",
      "699996  (18261528, 18261628]             1  \n",
      "699997  (18261628, 18261710]             1  \n",
      "699998  (18261628, 18261710]             1  \n",
      "699999  (18261628, 18261710]             1  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:137: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "         bin_offset_10  TT_S0  target_count starting_nt\n",
      "0   (3000018, 3000028]    0.0             3     3000018\n",
      "1   (3000028, 3000128]    7.0             8     3000028\n",
      "2   (3000128, 3000228]    1.0             5     3000128\n",
      "3   (3000228, 3000328]    2.0             3     3000228\n",
      "4   (3000328, 3000428]    0.0             5     3000328\n",
      "5   (3000428, 3000528]    0.0             1     3000428\n",
      "6   (3000528, 3000628]    0.0             2     3000528\n",
      "7   (3000628, 3000728]    0.0             4     3000628\n",
      "8   (3000728, 3000828]    0.0             7     3000728\n",
      "9   (3000828, 3000928]    0.0             4     3000828\n",
      "10  (3000928, 3001028]    0.0             2     3000928\n",
      "11  (3001028, 3001128]    2.0             6     3001028\n",
      "12  (3001128, 3001228]    0.0             2     3001128\n",
      "13  (3001228, 3001328]    0.0             5     3001228\n",
      "14  (3001328, 3001428]    0.0             2     3001328\n",
      "15  (3001428, 3001528]    1.0             6     3001428\n",
      "16  (3001528, 3001628]    3.0             9     3001528\n",
      "17  (3001628, 3001728]    3.0             7     3001628\n",
      "18  (3001728, 3001828]    0.0             5     3001728\n",
      "19  (3001828, 3001928]    0.0             3     3001828\n",
      "20  (3001928, 3002028]    1.0             2     3001928\n",
      "21  (3002028, 3002128]    0.0             2     3002028\n",
      "22  (3002128, 3002228]    1.0             4     3002128\n",
      "23  (3002228, 3002328]   10.0             6     3002228\n",
      "24  (3002328, 3002428]   12.0             2     3002328\n",
      "25  (3002428, 3002528]   13.0             6     3002428\n",
      "26  (3002528, 3002628]    0.0             0     3002528\n",
      "27  (3002628, 3002728]    0.0             3     3002628\n",
      "28  (3002728, 3002828]    0.0             2     3002728\n",
      "29  (3002828, 3002928]    0.0             1     3002828\n",
      "30  (3002928, 3003028]    1.0             6     3002928\n",
      "31  (3003028, 3003128]    2.0             4     3003028\n",
      "32  (3003128, 3003228]    0.0             0     3003128\n",
      "33  (3003228, 3003328]    0.0             3     3003228\n",
      "34  (3003328, 3003428]    1.0             6     3003328\n",
      "35  (3003428, 3003528]    0.0             3     3003428\n",
      "36  (3003528, 3003628]    1.0             2     3003528\n",
      "37  (3003628, 3003728]    3.0             4     3003628\n",
      "38  (3003728, 3003828]    1.0             6     3003728\n",
      "39  (3003828, 3003928]    1.0             4     3003828\n",
      "40  (3003928, 3004028]    7.0             5     3003928\n",
      "41  (3004028, 3004128]    0.0             0     3004028\n",
      "42  (3004128, 3004228]    0.0             3     3004128\n",
      "43  (3004228, 3004328]    5.0             4     3004228\n",
      "44  (3004328, 3004428]    0.0             1     3004328\n",
      "45  (3004428, 3004528]    0.0             3     3004428\n",
      "46  (3004528, 3004628]    1.0             5     3004528\n",
      "47  (3004628, 3004728]    0.0             0     3004628\n",
      "48  (3004728, 3004828]    0.0             1     3004728\n",
      "49  (3004828, 3004928]   13.0             8     3004828\n",
      "(152618, 4)\n",
      "bin_offset_10    category\n",
      "TT_S0             float64\n",
      "target_count        int64\n",
      "starting_nt      category\n",
      "dtype: object\n",
      "sumos ilgis: 152618\n",
      "ranges ilgis: 152618\n",
      "Model input - NO SCALE\n",
      "        TT_S0  genes  transcripts  exons  CDSs\n",
      "0         0.0      0            0      0     0\n",
      "1         7.0      0            0      0     0\n",
      "2         1.0      0            0      0     0\n",
      "3         2.0      0            0      0     0\n",
      "4         0.0      0            0      0     0\n",
      "...       ...    ...          ...    ...   ...\n",
      "152613   10.0      0            0      0     0\n",
      "152614    3.0      0            0      0     0\n",
      "152615    4.0      0            0      0     0\n",
      "152616    0.0      0            0      0     0\n",
      "152617    4.0      0            0      0     0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "        TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0         0.0             0                   0             0            0\n",
      "1         7.0             0                   0             0            0\n",
      "2         1.0             0                   0             0            0\n",
      "3         2.0             0                   0             0            0\n",
      "4         0.0             0                   0             0            0\n",
      "...       ...           ...                 ...           ...          ...\n",
      "152613   10.0             0                   0             0            0\n",
      "152614    3.0             0                   0             0            0\n",
      "152615    4.0             0                   0             0            0\n",
      "152616    0.0             0                   0             0            0\n",
      "152617    4.0             0                   0             0            0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "0.03477964591332608\n",
      "[4 2 0 0 4 4 4 4 4 4 4 0 4 4 4 0 0 0 4 4 0 4 0 2 2 1 4 4 4 4 0 0 4 4 0 4 0\n",
      " 0 0 0 2 4 4 7 4 4 0 4 4 1 4 9 1 9 4 5 1 3 8 8 6 4 1 1 7 4 4 2 4 0 7 0 4 7\n",
      " 4 4 0 4 4 4 0 2 0 4 0 7 0 7 7 4 0 1 2 0 0 4 0 0 4 2 2 1 5 5 1 2 2 2 5 0 2\n",
      " 4 1 0 4 4 0 4 4 4 4 0 7 0 0 1 0 1 1 2 1 0 2 2 7 4 7 2 0 4 4 4 4 4 0 0 4 2\n",
      " 2 2 7 0 4 0 7 0 2 0 0 0 1 4 1 0 4 2 2 0 4 4 4 4 0 4 0 4 4 4 4 0 2 7 0 0 4\n",
      " 4 0 4 4 4 4 4 0 4 4 4 4 4 4 4 4 4 4 4 4 4 7 7 5 0 0 4 0 4 4 4 4 0 0 4 4 4\n",
      " 4 4 4 4 4 4 4 4 4 4 4 4 4 4 0 4 4 0 4 4 4 4 4 4 4 4 4 4 4 4 0 4 4 4 0 4 0\n",
      " 4 4 4 4 4 4 0 4 4 0 4 0 4 0 7 4 0 4 0 4 4 0 4 4 4 0 4 4 4 4 4 4 4 0 4 4 4\n",
      " 7 4 4 4 0 0 0 4 0 4 4 4 4 7 0 7 2 0 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4\n",
      " 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4\n",
      " 4 4 4 4 4 4 4 4 4 4 4 7 4 0 4 2 0 0 0 0 4 4 4 4 0 1 1 0 2 1 2 7 0 0 0 2 2\n",
      " 0 4 7 0 4 4 1 2 2 0 1 0 4 0 4 4 0 4 7 4 0 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4\n",
      " 0 4 4 0 7 4 4 4 4 4 0 4 4 0 4 0 0 4 7 2 4 4 0 7 7 7 4 7 7 7 7 7 0 0 0 0 4\n",
      " 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4\n",
      " 7 2 0 4 4 4 7 4 0 2 0 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4\n",
      " 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4\n",
      " 4 4 4 4 0 4 0 7 4 4 4 4 4 0 4 4 4 4 0 4 0 0 0 4 4 2 4 4 4 4 0 7 4 4 4 4 7\n",
      " 4 4 7 4 4 4 4 0 4 0 2 4 0 0 2 0 4 4 4 0 4 7 0 0 4 4 4 4 0 2 4 4 7 0 0 4 2\n",
      " 0 7 4 4 4 4 0 0 4 4 4 4 4 4 4 4 4 4 7 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4\n",
      " 4 4 4 4 4 4 4 4 4 4 4 4 4 0 0 2 7 0 0 4 0 4 0 4 0 4 4 4 0 4 4 4 4 4 0 4 4\n",
      " 4 4 0 4 4 0 4 4 0 4 4 4 4 4 4 0 4 4 4 4 0 4 4 4 4 4 4 0 4 4 4 4 0 0 4 4 4\n",
      " 4 4 4 4 4 4 4 0 4 4 4 4 4 0 0 4 4 4 0 0 4 4 0 0 4 4 4 0 2 7 2 4 0 4 0 0 4\n",
      " 4 4 4 4 0 4 4 4 4 4 4 4 4 4 4 4 4 4 0 0 0 0 0 4 7 4 4 0 4 0 4 0 0 4 4 4 0\n",
      " 4 4 4 4 4 4 4 4 4 4 4 0 4 4 0 4 0 4 0 7 4 4 4 4 4 0 7 4 4 4 4 4 4 4 4 4 4\n",
      " 4 4 0 4 4 4 4 4 0 4 4 4 4 4 4 4 4 4 4 4 0 0 4 4 0 0 0 7 4 4 4 4 0 4 4 4 0\n",
      " 7 0 2 4 0 0 4 4 4 4 4 4 4 4 4 0 4 4 4 0 0 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4\n",
      " 4 4 4 4 4 4 4 4 7 4 4 4 4 4 4 4 0 4 4 4 4 4 4 4 0 4 4 0 4 7 0 4 4 0 4 4 4\n",
      " 4]\n",
      "threshold density 1.95819\n",
      "cluster signal 0.0\n",
      "cluster target count 227951\n",
      "!!cluster density 0.0\n",
      "cluster signal 136309.0\n",
      "cluster target count 103828\n",
      "!!cluster density 1.3128346881380746\n",
      "cluster signal 71912.0\n",
      "cluster target count 186260\n",
      "!!cluster density 0.3860839686459787\n",
      "cluster signal 81077.0\n",
      "cluster target count 43429\n",
      "!!cluster density 1.8668861820442562\n",
      "cluster signal 94213.0\n",
      "cluster target count 112085\n",
      "!!cluster density 0.8405495829058304\n",
      "cluster signal 18694.0\n",
      "cluster target count 5960\n",
      "!!cluster density 3.136577181208054\n",
      "cluster signal 45697.0\n",
      "cluster target count 18529\n",
      "!!cluster density 2.4662421069674565\n",
      "cluster signal 8068.0\n",
      "cluster target count 1924\n",
      "!!cluster density 4.1933471933471935\n",
      "cluster signal 548.0\n",
      "cluster target count 29\n",
      "!!cluster density 18.896551724137932\n",
      "cluster signal 393.0\n",
      "cluster target count 5\n",
      "!!cluster density 78.6\n",
      "\n",
      "Labels mapping\n",
      " {4: 0, 2: 0, 0: 0, 1: 0, 7: 0, 9: 1, 5: 1, 3: 1, 8: 1, 6: 1}\n",
      "DEBUG: 150040 2578 \n",
      "\n",
      " +----------------------+-------------+----------------+-------+\n",
      "| bin_offset_10        | TT_S0       | target_count   | +9    |\n",
      "| (category)           | (float64)   | (int64)        | ...   |\n",
      "|----------------------+-------------+----------------+-------|\n",
      "| (3000018, 3000028]   | 0.0         | 3              | ...   |\n",
      "| (3000028, 3000128]   | 7.0         | 8              | ...   |\n",
      "| (3000128, 3000228]   | 1.0         | 5              | ...   |\n",
      "| (3000228, 3000328]   | 2.0         | 3              | ...   |\n",
      "| ...                  | ...         | ...            | ...   |\n",
      "| (18261328, 18261428] | 3.0         | 14             | ...   |\n",
      "| (18261428, 18261528] | 4.0         | 4              | ...   |\n",
      "| (18261528, 18261628] | 0.0         | 3              | ...   |\n",
      "| (18261628, 18261710] | 4.0         | 3              | ...   |\n",
      "+----------------------+-------------+----------------+-------+\n",
      "Unstranded PyRanges object has 150,040 rows and 12 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "9 hidden columns: starting_nt, Chromosome, Start, End, genes, transcripts, ... (+ 3 more.)\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0\n",
      "BINS  IntervalIndex([  (3000018, 3000033],   (3000033, 3000133],\n",
      "                 (3000133, 3000233],   (3000233, 3000333],\n",
      "                 (3000333, 3000433],   (3000433, 3000533],\n",
      "                 (3000533, 3000633],   (3000633, 3000733],\n",
      "                 (3000733, 3000833],   (3000833, 3000933],\n",
      "               ...\n",
      "               (18260733, 18260833], (18260833, 18260933],\n",
      "               (18260933, 18261033], (18261033, 18261133],\n",
      "               (18261133, 18261233], (18261233, 18261333],\n",
      "               (18261333, 18261433], (18261433, 18261533],\n",
      "               (18261533, 18261633], (18261633, 18261710]],\n",
      "              dtype='interval[int64, right]', length=152618)\n",
      "HEAD with BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0   \n",
      "\n",
      "        bin_offset_15  target_count  \n",
      "0  (3000018, 3000033]             1  \n",
      "1  (3000018, 3000033]             1  \n",
      "2  (3000018, 3000033]             1  \n",
      "3  (3000033, 3000133]             1  \n",
      "4  (3000033, 3000133]             1  \n",
      "TAIL with BINS\n",
      "       seqnames     start       end        CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  18261611  18261611  18_18261611    0.0    0.0    0.0   \n",
      "699996       18  18261615  18261615  18_18261615    0.0    0.0    0.0   \n",
      "699997       18  18261682  18261682  18_18261682    0.0    0.0    0.0   \n",
      "699998       18  18261702  18261702  18_18261702    4.0    0.0    0.0   \n",
      "699999       18  18261710  18261710  18_18261710    0.0    0.0    0.0   \n",
      "\n",
      "               bin_offset_15  target_count  \n",
      "699995  (18261533, 18261633]             1  \n",
      "699996  (18261533, 18261633]             1  \n",
      "699997  (18261633, 18261710]             1  \n",
      "699998  (18261633, 18261710]             1  \n",
      "699999  (18261633, 18261710]             1  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:137: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "         bin_offset_15  TT_S0  target_count starting_nt\n",
      "0   (3000018, 3000033]    0.0             3     3000018\n",
      "1   (3000033, 3000133]    8.0             9     3000033\n",
      "2   (3000133, 3000233]    0.0             4     3000133\n",
      "3   (3000233, 3000333]    2.0             3     3000233\n",
      "4   (3000333, 3000433]    0.0             5     3000333\n",
      "5   (3000433, 3000533]    0.0             1     3000433\n",
      "6   (3000533, 3000633]    0.0             2     3000533\n",
      "7   (3000633, 3000733]    0.0             4     3000633\n",
      "8   (3000733, 3000833]    0.0             7     3000733\n",
      "9   (3000833, 3000933]    0.0             4     3000833\n",
      "10  (3000933, 3001033]    0.0             3     3000933\n",
      "11  (3001033, 3001133]    2.0             5     3001033\n",
      "12  (3001133, 3001233]    0.0             2     3001133\n",
      "13  (3001233, 3001333]    0.0             5     3001233\n",
      "14  (3001333, 3001433]    0.0             2     3001333\n",
      "15  (3001433, 3001533]    1.0             7     3001433\n",
      "16  (3001533, 3001633]    3.0             8     3001533\n",
      "17  (3001633, 3001733]    3.0             7     3001633\n",
      "18  (3001733, 3001833]    0.0             6     3001733\n",
      "19  (3001833, 3001933]    0.0             2     3001833\n",
      "20  (3001933, 3002033]    1.0             2     3001933\n",
      "21  (3002033, 3002133]    0.0             3     3002033\n",
      "22  (3002133, 3002233]    4.0             4     3002133\n",
      "23  (3002233, 3002333]    7.0             5     3002233\n",
      "24  (3002333, 3002433]   12.0             2     3002333\n",
      "25  (3002433, 3002533]   13.0             6     3002433\n",
      "26  (3002533, 3002633]    0.0             0     3002533\n",
      "27  (3002633, 3002733]    0.0             3     3002633\n",
      "28  (3002733, 3002833]    0.0             2     3002733\n",
      "29  (3002833, 3002933]    0.0             1     3002833\n",
      "30  (3002933, 3003033]    1.0             6     3002933\n",
      "31  (3003033, 3003133]    2.0             4     3003033\n",
      "32  (3003133, 3003233]    0.0             0     3003133\n",
      "33  (3003233, 3003333]    0.0             4     3003233\n",
      "34  (3003333, 3003433]    1.0             5     3003333\n",
      "35  (3003433, 3003533]    0.0             3     3003433\n",
      "36  (3003533, 3003633]    1.0             2     3003533\n",
      "37  (3003633, 3003733]    3.0             5     3003633\n",
      "38  (3003733, 3003833]    1.0             5     3003733\n",
      "39  (3003833, 3003933]    1.0             4     3003833\n",
      "40  (3003933, 3004033]    7.0             5     3003933\n",
      "41  (3004033, 3004133]    0.0             0     3004033\n",
      "42  (3004133, 3004233]    0.0             3     3004133\n",
      "43  (3004233, 3004333]    5.0             4     3004233\n",
      "44  (3004333, 3004433]    0.0             1     3004333\n",
      "45  (3004433, 3004533]    0.0             3     3004433\n",
      "46  (3004533, 3004633]    1.0             5     3004533\n",
      "47  (3004633, 3004733]    0.0             0     3004633\n",
      "48  (3004733, 3004833]    0.0             1     3004733\n",
      "49  (3004833, 3004933]   13.0             8     3004833\n",
      "(152618, 4)\n",
      "bin_offset_15    category\n",
      "TT_S0             float64\n",
      "target_count        int64\n",
      "starting_nt      category\n",
      "dtype: object\n",
      "sumos ilgis: 152618\n",
      "ranges ilgis: 152618\n",
      "Model input - NO SCALE\n",
      "        TT_S0  genes  transcripts  exons  CDSs\n",
      "0         0.0      0            0      0     0\n",
      "1         8.0      0            0      0     0\n",
      "2         0.0      0            0      0     0\n",
      "3         2.0      0            0      0     0\n",
      "4         0.0      0            0      0     0\n",
      "...       ...    ...          ...    ...   ...\n",
      "152613   13.0      0            0      0     0\n",
      "152614    0.0      0            0      0     0\n",
      "152615    4.0      0            0      0     0\n",
      "152616    0.0      0            0      0     0\n",
      "152617    4.0      0            0      0     0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "        TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0         0.0             0                   0             0            0\n",
      "1         8.0             0                   0             0            0\n",
      "2         0.0             0                   0             0            0\n",
      "3         2.0             0                   0             0            0\n",
      "4         0.0             0                   0             0            0\n",
      "...       ...           ...                 ...           ...          ...\n",
      "152613   13.0             0                   0             0            0\n",
      "152614    0.0             0                   0             0            0\n",
      "152615    4.0             0                   0             0            0\n",
      "152616    0.0             0                   0             0            0\n",
      "152617    4.0             0                   0             0            0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "0.01924412585671415\n",
      "[2 9 2 7 2 2 2 2 2 2 2 7 2 2 2 2 0 0 2 2 2 2 0 8 3 3 2 2 2 2 2 7 2 2 2 2 2\n",
      " 0 2 2 8 2 2 8 2 2 2 2 2 3 2 6 1 6 2 1 3 4 5 4 5 2 6 2 0 2 2 8 2 2 8 7 7 0\n",
      " 2 2 7 2 2 2 7 9 7 2 0 0 7 0 8 2 2 3 9 7 7 2 0 0 2 9 3 3 6 6 8 0 9 8 6 2 3\n",
      " 2 3 0 2 2 7 2 2 2 2 2 8 2 2 3 7 3 1 9 3 2 9 8 8 2 8 9 0 2 2 2 2 2 7 2 2 9\n",
      " 9 8 8 7 2 2 8 0 9 0 7 2 1 2 3 2 2 9 3 0 2 2 2 2 2 2 7 2 2 2 2 2 8 9 2 0 2\n",
      " 2 0 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 8 3 3 2 2 2 7 2 2 2 2 2 2 2 2 2\n",
      " 2 2 2 2 2 2 2 2 2 2 2 2 2 2 0 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 7 2 2\n",
      " 2 2 2 2 2 2 7 2 2 0 2 7 2 7 0 2 2 2 2 2 2 7 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2\n",
      " 8 2 2 2 2 7 0 2 7 2 2 2 2 0 0 0 8 7 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2\n",
      " 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2\n",
      " 2 2 2 2 2 2 2 2 2 2 2 8 2 7 2 3 2 2 2 2 2 2 2 2 2 3 3 7 9 3 9 8 0 7 7 9 8\n",
      " 0 2 8 7 2 2 3 9 3 2 3 7 2 0 2 2 7 2 8 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2\n",
      " 0 2 2 2 8 2 2 2 2 2 0 2 2 2 2 2 7 2 8 8 2 2 2 8 8 0 2 0 8 8 8 8 7 7 0 2 2\n",
      " 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2\n",
      " 8 3 2 2 2 2 8 2 2 9 0 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2\n",
      " 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2\n",
      " 2 2 2 2 0 2 7 0 2 2 2 2 2 2 2 2 2 2 2 2 2 7 7 2 2 8 2 2 2 2 7 8 2 2 2 2 8\n",
      " 2 2 0 2 2 2 2 2 2 0 9 2 2 0 8 7 2 2 2 2 2 8 2 7 2 2 2 2 2 9 2 2 8 0 2 2 8\n",
      " 7 8 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 0 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2\n",
      " 2 2 2 2 2 2 2 2 2 2 2 2 2 0 0 8 0 7 0 2 7 2 2 2 2 2 2 2 0 2 2 2 2 2 2 2 2\n",
      " 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 0 2 2 2 2 7 0 2 2 2\n",
      " 2 2 2 2 2 2 2 7 2 2 2 2 2 7 0 2 2 2 2 7 2 2 7 7 2 2 2 7 9 8 9 2 0 2 2 0 2\n",
      " 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 0 2 2 2 2 7 2 2 7 2 2 2 2\n",
      " 2 2 2 2 2 2 2 2 2 2 2 7 2 2 2 2 7 2 0 0 2 2 2 2 2 7 0 2 2 2 2 2 2 2 2 2 2\n",
      " 2 2 0 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 7 2 2 2 2 0 8 2 2 2 2 2 2 2 2 2\n",
      " 8 0 3 2 2 0 2 2 2 2 2 2 2 2 2 7 2 2 2 0 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2\n",
      " 2 2 2 2 2 2 2 2 0 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 0 2 8 0 2 2 2 2 2 2\n",
      " 2]\n",
      "threshold density 1.95819\n",
      "cluster signal 16259.0\n",
      "cluster target count 300608\n",
      "!!cluster density 0.05408705024483713\n",
      "cluster signal 71426.0\n",
      "cluster target count 55010\n",
      "!!cluster density 1.2984184693692056\n",
      "cluster signal 25594.0\n",
      "cluster target count 62061\n",
      "!!cluster density 0.41240070253460304\n",
      "cluster signal 62090.0\n",
      "cluster target count 95857\n",
      "!!cluster density 0.6477356896209979\n",
      "cluster signal 89599.0\n",
      "cluster target count 92575\n",
      "!!cluster density 0.9678530920874966\n",
      "cluster signal 92006.0\n",
      "cluster target count 54503\n",
      "!!cluster density 1.6880905638221748\n",
      "cluster signal 30391.0\n",
      "cluster target count 10519\n",
      "!!cluster density 2.8891529613081093\n",
      "cluster signal 58611.0\n",
      "cluster target count 26400\n",
      "!!cluster density 2.2201136363636365\n",
      "cluster signal 10298.0\n",
      "cluster target count 2457\n",
      "!!cluster density 4.191290191290191\n",
      "cluster signal 637.0\n",
      "cluster target count 10\n",
      "!!cluster density 63.7\n",
      "\n",
      "Labels mapping\n",
      " {2: 0, 9: 0, 7: 0, 0: 0, 8: 0, 3: 0, 6: 1, 1: 1, 4: 1, 5: 1}\n",
      "DEBUG: 148548 4070 \n",
      "\n",
      " +----------------------+-------------+----------------+-------+\n",
      "| bin_offset_15        | TT_S0       | target_count   | +9    |\n",
      "| (category)           | (float64)   | (int64)        | ...   |\n",
      "|----------------------+-------------+----------------+-------|\n",
      "| (3000018, 3000033]   | 0.0         | 3              | ...   |\n",
      "| (3000033, 3000133]   | 8.0         | 9              | ...   |\n",
      "| (3000133, 3000233]   | 0.0         | 4              | ...   |\n",
      "| (3000233, 3000333]   | 2.0         | 3              | ...   |\n",
      "| ...                  | ...         | ...            | ...   |\n",
      "| (18261333, 18261433] | 0.0         | 12             | ...   |\n",
      "| (18261433, 18261533] | 4.0         | 4              | ...   |\n",
      "| (18261533, 18261633] | 0.0         | 3              | ...   |\n",
      "| (18261633, 18261710] | 4.0         | 3              | ...   |\n",
      "+----------------------+-------------+----------------+-------+\n",
      "Unstranded PyRanges object has 148,548 rows and 12 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "9 hidden columns: starting_nt, Chromosome, Start, End, genes, transcripts, ... (+ 3 more.)\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0\n",
      "BINS  IntervalIndex([  (3000018, 3000038],   (3000038, 3000138],\n",
      "                 (3000138, 3000238],   (3000238, 3000338],\n",
      "                 (3000338, 3000438],   (3000438, 3000538],\n",
      "                 (3000538, 3000638],   (3000638, 3000738],\n",
      "                 (3000738, 3000838],   (3000838, 3000938],\n",
      "               ...\n",
      "               (18260738, 18260838], (18260838, 18260938],\n",
      "               (18260938, 18261038], (18261038, 18261138],\n",
      "               (18261138, 18261238], (18261238, 18261338],\n",
      "               (18261338, 18261438], (18261438, 18261538],\n",
      "               (18261538, 18261638], (18261638, 18261710]],\n",
      "              dtype='interval[int64, right]', length=152618)\n",
      "HEAD with BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0   \n",
      "\n",
      "        bin_offset_20  target_count  \n",
      "0  (3000018, 3000038]             1  \n",
      "1  (3000018, 3000038]             1  \n",
      "2  (3000018, 3000038]             1  \n",
      "3  (3000038, 3000138]             1  \n",
      "4  (3000038, 3000138]             1  \n",
      "TAIL with BINS\n",
      "       seqnames     start       end        CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  18261611  18261611  18_18261611    0.0    0.0    0.0   \n",
      "699996       18  18261615  18261615  18_18261615    0.0    0.0    0.0   \n",
      "699997       18  18261682  18261682  18_18261682    0.0    0.0    0.0   \n",
      "699998       18  18261702  18261702  18_18261702    4.0    0.0    0.0   \n",
      "699999       18  18261710  18261710  18_18261710    0.0    0.0    0.0   \n",
      "\n",
      "               bin_offset_20  target_count  \n",
      "699995  (18261538, 18261638]             1  \n",
      "699996  (18261538, 18261638]             1  \n",
      "699997  (18261638, 18261710]             1  \n",
      "699998  (18261638, 18261710]             1  \n",
      "699999  (18261638, 18261710]             1  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:137: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "         bin_offset_20  TT_S0  target_count starting_nt\n",
      "0   (3000018, 3000038]    0.0             3     3000018\n",
      "1   (3000038, 3000138]    8.0            10     3000038\n",
      "2   (3000138, 3000238]    0.0             3     3000138\n",
      "3   (3000238, 3000338]    2.0             3     3000238\n",
      "4   (3000338, 3000438]    0.0             5     3000338\n",
      "5   (3000438, 3000538]    0.0             1     3000438\n",
      "6   (3000538, 3000638]    0.0             2     3000538\n",
      "7   (3000638, 3000738]    0.0             5     3000638\n",
      "8   (3000738, 3000838]    0.0             6     3000738\n",
      "9   (3000838, 3000938]    0.0             4     3000838\n",
      "10  (3000938, 3001038]    0.0             3     3000938\n",
      "11  (3001038, 3001138]    2.0             5     3001038\n",
      "12  (3001138, 3001238]    0.0             2     3001138\n",
      "13  (3001238, 3001338]    0.0             5     3001238\n",
      "14  (3001338, 3001438]    0.0             2     3001338\n",
      "15  (3001438, 3001538]    1.0             7     3001438\n",
      "16  (3001538, 3001638]    3.0             8     3001538\n",
      "17  (3001638, 3001738]    3.0             7     3001638\n",
      "18  (3001738, 3001838]    0.0             6     3001738\n",
      "19  (3001838, 3001938]    0.0             2     3001838\n",
      "20  (3001938, 3002038]    1.0             2     3001938\n",
      "21  (3002038, 3002138]    0.0             3     3002038\n",
      "22  (3002138, 3002238]    4.0             5     3002138\n",
      "23  (3002238, 3002338]    7.0             4     3002238\n",
      "24  (3002338, 3002438]   12.0             2     3002338\n",
      "25  (3002438, 3002538]   13.0             6     3002438\n",
      "26  (3002538, 3002638]    0.0             0     3002538\n",
      "27  (3002638, 3002738]    0.0             3     3002638\n",
      "28  (3002738, 3002838]    0.0             3     3002738\n",
      "29  (3002838, 3002938]    0.0             0     3002838\n",
      "30  (3002938, 3003038]    1.0             8     3002938\n",
      "31  (3003038, 3003138]    2.0             2     3003038\n",
      "32  (3003138, 3003238]    0.0             0     3003138\n",
      "33  (3003238, 3003338]    0.0             4     3003238\n",
      "34  (3003338, 3003438]    1.0             5     3003338\n",
      "35  (3003438, 3003538]    0.0             3     3003438\n",
      "36  (3003538, 3003638]    1.0             2     3003538\n",
      "37  (3003638, 3003738]    3.0             5     3003638\n",
      "38  (3003738, 3003838]    1.0             5     3003738\n",
      "39  (3003838, 3003938]    1.0             4     3003838\n",
      "40  (3003938, 3004038]    7.0             5     3003938\n",
      "41  (3004038, 3004138]    0.0             0     3004038\n",
      "42  (3004138, 3004238]    1.0             4     3004138\n",
      "43  (3004238, 3004338]    4.0             3     3004238\n",
      "44  (3004338, 3004438]    0.0             1     3004338\n",
      "45  (3004438, 3004538]    0.0             3     3004438\n",
      "46  (3004538, 3004638]    1.0             5     3004538\n",
      "47  (3004638, 3004738]    0.0             0     3004638\n",
      "48  (3004738, 3004838]    1.0             2     3004738\n",
      "49  (3004838, 3004938]   12.0             7     3004838\n",
      "(152618, 4)\n",
      "bin_offset_20    category\n",
      "TT_S0             float64\n",
      "target_count        int64\n",
      "starting_nt      category\n",
      "dtype: object\n",
      "sumos ilgis: 152618\n",
      "ranges ilgis: 152618\n",
      "Model input - NO SCALE\n",
      "        TT_S0  genes  transcripts  exons  CDSs\n",
      "0         0.0      0            0      0     0\n",
      "1         8.0      0            0      0     0\n",
      "2         0.0      0            0      0     0\n",
      "3         2.0      0            0      0     0\n",
      "4         0.0      0            0      0     0\n",
      "...       ...    ...          ...    ...   ...\n",
      "152613   13.0      0            0      0     0\n",
      "152614    1.0      0            0      0     0\n",
      "152615    3.0      0            0      0     0\n",
      "152616    0.0      0            0      0     0\n",
      "152617    4.0      0            0      0     0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "        TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0         0.0             0                   0             0            0\n",
      "1         8.0             0                   0             0            0\n",
      "2         0.0             0                   0             0            0\n",
      "3         2.0             0                   0             0            0\n",
      "4         0.0             0                   0             0            0\n",
      "...       ...           ...                 ...           ...          ...\n",
      "152613   13.0             0                   0             0            0\n",
      "152614    1.0             0                   0             0            0\n",
      "152615    3.0             0                   0             0            0\n",
      "152616    0.0             0                   0             0            0\n",
      "152617    4.0             0                   0             0            0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "0.038501356327562936\n",
      "[9 8 9 7 9 9 9 9 9 9 9 7 9 9 9 0 7 7 9 9 0 9 7 3 1 1 9 9 9 9 0 7 9 9 0 9 0\n",
      " 7 0 0 3 9 0 7 9 9 0 9 0 1 9 2 5 2 9 5 1 4 4 4 6 7 5 0 7 9 9 3 9 0 3 7 7 0\n",
      " 9 9 7 9 9 9 7 8 7 9 7 7 7 7 3 9 7 1 8 7 7 9 7 7 9 8 1 5 2 2 3 7 8 3 2 0 8\n",
      " 9 1 7 9 0 0 9 9 9 9 0 3 0 0 1 7 1 5 8 1 0 8 3 3 9 3 8 7 9 9 9 9 9 7 0 9 8\n",
      " 8 3 3 7 9 0 3 7 8 7 7 0 1 9 1 0 3 8 3 7 9 9 9 9 0 9 7 9 9 9 9 0 8 3 9 7 9\n",
      " 9 7 9 9 9 9 9 0 9 9 9 9 9 9 9 9 9 9 9 9 9 3 8 1 9 0 9 7 9 9 9 9 0 0 9 9 9\n",
      " 9 9 9 9 9 9 9 9 9 9 9 9 9 7 9 9 9 0 9 9 9 9 9 9 9 9 9 9 9 9 0 9 9 9 7 9 0\n",
      " 9 9 9 9 9 9 7 9 9 7 9 7 9 7 7 9 0 9 0 9 9 7 9 9 9 0 9 9 9 9 9 9 0 9 9 9 9\n",
      " 3 9 9 9 0 7 7 7 9 9 9 9 9 7 7 7 3 7 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9\n",
      " 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9\n",
      " 9 9 9 9 9 9 9 9 9 9 9 3 9 7 7 8 0 0 0 0 9 9 9 9 7 8 1 1 9 1 8 3 7 0 7 8 3\n",
      " 7 9 3 7 9 9 1 8 1 7 1 7 9 7 9 9 7 9 3 9 0 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9\n",
      " 7 9 9 0 3 9 9 9 9 7 0 9 9 0 9 0 7 9 3 3 9 9 7 7 3 7 9 7 8 7 8 7 7 9 7 0 9\n",
      " 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9\n",
      " 3 1 0 9 9 9 3 9 0 8 7 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9\n",
      " 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9\n",
      " 9 9 9 9 7 9 7 7 9 9 9 9 9 0 9 9 9 9 0 9 0 7 7 9 9 3 9 9 9 9 7 3 9 9 9 9 3\n",
      " 9 9 7 9 9 9 9 0 9 7 8 9 0 7 3 7 9 9 9 0 9 3 0 7 9 9 9 9 0 8 9 0 3 7 9 0 3\n",
      " 7 3 9 9 9 9 0 0 9 9 9 9 9 9 9 9 9 9 7 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9\n",
      " 9 9 9 9 9 9 9 9 9 9 9 9 9 7 7 3 7 7 7 9 7 0 9 9 0 9 9 9 7 9 9 9 9 9 0 9 9\n",
      " 9 9 0 9 9 0 9 9 0 9 9 9 9 9 9 0 9 9 9 9 0 9 9 9 9 9 9 7 9 9 9 9 7 7 9 9 9\n",
      " 9 9 9 9 9 9 9 7 9 9 9 9 9 7 7 9 9 9 0 7 9 9 7 7 9 9 9 7 8 3 8 9 7 0 0 7 9\n",
      " 9 9 9 9 0 9 9 9 9 9 9 9 9 9 9 9 9 9 0 0 0 0 0 9 7 9 9 0 9 7 9 0 7 9 9 9 0\n",
      " 9 9 9 9 9 9 9 9 9 9 7 9 9 9 0 9 7 9 7 7 9 9 9 9 9 7 7 9 9 9 9 9 9 9 9 9 9\n",
      " 9 9 7 9 9 9 9 9 0 9 9 9 9 9 9 9 9 9 9 9 0 7 9 9 0 0 7 3 9 9 9 9 0 9 9 9 0\n",
      " 3 7 8 0 7 7 9 9 9 9 9 9 9 9 9 7 9 9 9 7 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9\n",
      " 9 9 9 9 9 9 9 9 7 9 9 9 9 9 9 9 0 9 9 9 9 9 9 9 0 9 9 7 9 3 0 9 9 0 9 9 9\n",
      " 9]\n",
      "threshold density 1.95819\n",
      "cluster signal 0.0\n",
      "cluster target count 228091\n",
      "!!cluster density 0.0\n",
      "cluster signal 90634.0\n",
      "cluster target count 67570\n",
      "!!cluster density 1.3413349119431701\n",
      "cluster signal 87428.0\n",
      "cluster target count 157431\n",
      "!!cluster density 0.5553417052550006\n",
      "cluster signal 16266.0\n",
      "cluster target count 72663\n",
      "!!cluster density 0.22385533214978737\n",
      "cluster signal 90293.0\n",
      "cluster target count 93435\n",
      "!!cluster density 0.9663723444105528\n",
      "cluster signal 82183.0\n",
      "cluster target count 46123\n",
      "!!cluster density 1.781822518049563\n",
      "cluster signal 24687.0\n",
      "cluster target count 8023\n",
      "!!cluster density 3.07702854293905\n",
      "cluster signal 58010.0\n",
      "cluster target count 25184\n",
      "!!cluster density 2.303446632782719\n",
      "cluster signal 7017.0\n",
      "cluster target count 1475\n",
      "!!cluster density 4.75728813559322\n",
      "cluster signal 393.0\n",
      "cluster target count 5\n",
      "!!cluster density 78.6\n",
      "\n",
      "Labels mapping\n",
      " {9: 0, 8: 0, 7: 0, 0: 0, 3: 0, 1: 0, 2: 1, 5: 1, 4: 1, 6: 1}\n",
      "DEBUG: 149127 3491 \n",
      "\n",
      " +----------------------+-------------+----------------+-------+\n",
      "| bin_offset_20        | TT_S0       | target_count   | +9    |\n",
      "| (category)           | (float64)   | (int64)        | ...   |\n",
      "|----------------------+-------------+----------------+-------|\n",
      "| (3000018, 3000038]   | 0.0         | 3              | ...   |\n",
      "| (3000038, 3000138]   | 8.0         | 10             | ...   |\n",
      "| (3000138, 3000238]   | 0.0         | 3              | ...   |\n",
      "| (3000238, 3000338]   | 2.0         | 3              | ...   |\n",
      "| ...                  | ...         | ...            | ...   |\n",
      "| (18261338, 18261438] | 1.0         | 12             | ...   |\n",
      "| (18261438, 18261538] | 3.0         | 4              | ...   |\n",
      "| (18261538, 18261638] | 0.0         | 2              | ...   |\n",
      "| (18261638, 18261710] | 4.0         | 3              | ...   |\n",
      "+----------------------+-------------+----------------+-------+\n",
      "Unstranded PyRanges object has 149,127 rows and 12 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "9 hidden columns: starting_nt, Chromosome, Start, End, genes, transcripts, ... (+ 3 more.)\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0\n",
      "BINS  IntervalIndex([  (3000018, 3000043],   (3000043, 3000143],\n",
      "                 (3000143, 3000243],   (3000243, 3000343],\n",
      "                 (3000343, 3000443],   (3000443, 3000543],\n",
      "                 (3000543, 3000643],   (3000643, 3000743],\n",
      "                 (3000743, 3000843],   (3000843, 3000943],\n",
      "               ...\n",
      "               (18260743, 18260843], (18260843, 18260943],\n",
      "               (18260943, 18261043], (18261043, 18261143],\n",
      "               (18261143, 18261243], (18261243, 18261343],\n",
      "               (18261343, 18261443], (18261443, 18261543],\n",
      "               (18261543, 18261643], (18261643, 18261710]],\n",
      "              dtype='interval[int64, right]', length=152618)\n",
      "HEAD with BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0   \n",
      "\n",
      "        bin_offset_25  target_count  \n",
      "0  (3000018, 3000043]             1  \n",
      "1  (3000018, 3000043]             1  \n",
      "2  (3000018, 3000043]             1  \n",
      "3  (3000018, 3000043]             1  \n",
      "4  (3000043, 3000143]             1  \n",
      "TAIL with BINS\n",
      "       seqnames     start       end        CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  18261611  18261611  18_18261611    0.0    0.0    0.0   \n",
      "699996       18  18261615  18261615  18_18261615    0.0    0.0    0.0   \n",
      "699997       18  18261682  18261682  18_18261682    0.0    0.0    0.0   \n",
      "699998       18  18261702  18261702  18_18261702    4.0    0.0    0.0   \n",
      "699999       18  18261710  18261710  18_18261710    0.0    0.0    0.0   \n",
      "\n",
      "               bin_offset_25  target_count  \n",
      "699995  (18261543, 18261643]             1  \n",
      "699996  (18261543, 18261643]             1  \n",
      "699997  (18261643, 18261710]             1  \n",
      "699998  (18261643, 18261710]             1  \n",
      "699999  (18261643, 18261710]             1  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:137: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "         bin_offset_25  TT_S0  target_count starting_nt\n",
      "0   (3000018, 3000043]    4.0             4     3000018\n",
      "1   (3000043, 3000143]    4.0             9     3000043\n",
      "2   (3000143, 3000243]    0.0             3     3000143\n",
      "3   (3000243, 3000343]    2.0             3     3000243\n",
      "4   (3000343, 3000443]    0.0             5     3000343\n",
      "5   (3000443, 3000543]    0.0             1     3000443\n",
      "6   (3000543, 3000643]    0.0             2     3000543\n",
      "7   (3000643, 3000743]    0.0             5     3000643\n",
      "8   (3000743, 3000843]    0.0             6     3000743\n",
      "9   (3000843, 3000943]    0.0             4     3000843\n",
      "10  (3000943, 3001043]    2.0             4     3000943\n",
      "11  (3001043, 3001143]    0.0             4     3001043\n",
      "12  (3001143, 3001243]    0.0             2     3001143\n",
      "13  (3001243, 3001343]    0.0             5     3001243\n",
      "14  (3001343, 3001443]    0.0             2     3001343\n",
      "15  (3001443, 3001543]    1.0             7     3001443\n",
      "16  (3001543, 3001643]    3.0             8     3001543\n",
      "17  (3001643, 3001743]    3.0             8     3001643\n",
      "18  (3001743, 3001843]    0.0             5     3001743\n",
      "19  (3001843, 3001943]    0.0             2     3001843\n",
      "20  (3001943, 3002043]    1.0             2     3001943\n",
      "21  (3002043, 3002143]    0.0             3     3002043\n",
      "22  (3002143, 3002243]    4.0             5     3002143\n",
      "23  (3002243, 3002343]    7.0             4     3002243\n",
      "24  (3002343, 3002443]   12.0             3     3002343\n",
      "25  (3002443, 3002543]   13.0             5     3002443\n",
      "26  (3002543, 3002643]    0.0             0     3002543\n",
      "27  (3002643, 3002743]    0.0             3     3002643\n",
      "28  (3002743, 3002843]    0.0             3     3002743\n",
      "29  (3002843, 3002943]    0.0             0     3002843\n",
      "30  (3002943, 3003043]    1.0             8     3002943\n",
      "31  (3003043, 3003143]    2.0             2     3003043\n",
      "32  (3003143, 3003243]    0.0             0     3003143\n",
      "33  (3003243, 3003343]    1.0             5     3003243\n",
      "34  (3003343, 3003443]    0.0             4     3003343\n",
      "35  (3003443, 3003543]    0.0             3     3003443\n",
      "36  (3003543, 3003643]    1.0             2     3003543\n",
      "37  (3003643, 3003743]    3.0             5     3003643\n",
      "38  (3003743, 3003843]    1.0             5     3003743\n",
      "39  (3003843, 3003943]    1.0             5     3003843\n",
      "40  (3003943, 3004043]    7.0             4     3003943\n",
      "41  (3004043, 3004143]    0.0             0     3004043\n",
      "42  (3004143, 3004243]    1.0             4     3004143\n",
      "43  (3004243, 3004343]    4.0             3     3004243\n",
      "44  (3004343, 3004443]    0.0             1     3004343\n",
      "45  (3004443, 3004543]    0.0             3     3004443\n",
      "46  (3004543, 3004643]    1.0             5     3004543\n",
      "47  (3004643, 3004743]    0.0             0     3004643\n",
      "48  (3004743, 3004843]    2.0             3     3004743\n",
      "49  (3004843, 3004943]   11.0             6     3004843\n",
      "(152618, 4)\n",
      "bin_offset_25    category\n",
      "TT_S0             float64\n",
      "target_count        int64\n",
      "starting_nt      category\n",
      "dtype: object\n",
      "sumos ilgis: 152618\n",
      "ranges ilgis: 152618\n",
      "Model input - NO SCALE\n",
      "        TT_S0  genes  transcripts  exons  CDSs\n",
      "0         4.0      0            0      0     0\n",
      "1         4.0      0            0      0     0\n",
      "2         0.0      0            0      0     0\n",
      "3         2.0      0            0      0     0\n",
      "4         0.0      0            0      0     0\n",
      "...       ...    ...          ...    ...   ...\n",
      "152613   13.0      0            0      0     0\n",
      "152614    1.0      0            0      0     0\n",
      "152615    3.0      0            0      0     0\n",
      "152616    0.0      0            0      0     0\n",
      "152617    4.0      0            0      0     0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "        TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0         4.0             0                   0             0            0\n",
      "1         4.0             0                   0             0            0\n",
      "2         0.0             0                   0             0            0\n",
      "3         2.0             0                   0             0            0\n",
      "4         0.0             0                   0             0            0\n",
      "...       ...           ...                 ...           ...          ...\n",
      "152613   13.0             0                   0             0            0\n",
      "152614    1.0             0                   0             0            0\n",
      "152615    3.0             0                   0             0            0\n",
      "152616    0.0             0                   0             0            0\n",
      "152617    4.0             0                   0             0            0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "0.0037741288707754\n",
      "[9 9 3 0 3 3 3 3 3 3 0 3 3 3 3 0 9 9 3 3 0 3 9 5 2 7 3 3 3 3 0 0 3 0 3 3 0\n",
      " 9 0 0 5 3 0 9 3 3 0 3 0 2 3 1 2 1 3 4 2 6 6 6 8 5 4 0 9 3 3 5 3 0 5 0 9 0\n",
      " 3 3 0 3 3 3 9 5 0 3 9 9 0 5 0 3 9 7 2 0 0 3 9 9 3 2 2 7 4 1 9 9 2 5 1 3 2\n",
      " 3 7 9 3 0 0 3 3 3 3 0 5 0 0 7 0 7 4 2 7 0 2 5 5 3 5 2 9 3 3 3 3 3 0 0 3 2\n",
      " 5 5 5 0 3 0 5 2 9 9 0 0 7 3 7 0 5 7 3 9 3 3 3 0 3 3 0 3 3 3 3 0 5 5 3 9 3\n",
      " 3 9 3 3 3 3 3 0 3 3 3 3 3 3 3 3 3 3 3 3 3 5 2 7 3 0 3 0 3 3 3 3 0 0 3 3 3\n",
      " 3 3 3 3 3 3 3 3 3 3 3 3 3 9 3 3 3 0 3 3 3 3 3 3 3 3 3 3 3 0 3 3 3 3 0 3 0\n",
      " 3 3 3 3 3 3 0 3 3 9 3 0 3 0 9 3 0 3 0 3 3 0 3 3 3 0 3 3 3 3 3 3 0 3 3 3 3\n",
      " 5 3 3 3 0 0 9 0 3 3 3 3 3 5 0 9 5 0 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3\n",
      " 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3\n",
      " 3 3 3 3 3 3 3 3 3 3 3 5 3 0 9 2 0 0 0 0 3 3 3 3 5 7 5 2 3 7 2 5 9 0 0 5 5\n",
      " 9 3 5 0 3 0 7 5 7 0 2 0 3 9 3 3 0 3 5 3 0 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3\n",
      " 9 3 0 3 5 3 3 3 3 0 0 3 3 0 3 0 0 3 5 5 3 0 0 9 5 9 3 9 5 5 5 9 9 3 9 0 3\n",
      " 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3\n",
      " 5 2 0 3 3 0 9 3 0 5 9 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3\n",
      " 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3\n",
      " 3 3 3 3 9 3 0 9 3 3 3 3 3 0 3 3 3 3 0 3 0 0 0 3 3 5 3 3 3 3 0 5 3 3 3 3 5\n",
      " 3 3 9 3 3 3 3 0 3 9 2 3 0 2 9 3 3 3 3 0 3 5 0 0 3 3 3 3 0 2 3 0 5 9 3 0 5\n",
      " 9 9 3 3 3 3 0 3 3 3 3 3 3 3 3 3 3 3 9 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3\n",
      " 3 3 3 3 3 3 3 3 3 3 3 3 3 9 9 2 9 3 9 3 0 0 3 3 0 3 3 3 9 3 3 3 3 3 0 3 3\n",
      " 3 3 0 3 3 0 3 0 3 3 3 3 3 3 0 3 3 3 3 3 0 3 3 3 3 3 9 3 3 3 3 0 3 9 3 3 3\n",
      " 3 3 3 3 3 3 3 0 3 3 3 3 3 0 9 3 3 3 0 0 3 3 0 0 3 3 3 2 0 5 2 3 9 0 0 9 3\n",
      " 3 3 3 3 0 3 3 3 3 3 3 3 3 3 3 3 3 3 0 0 0 0 0 3 9 3 3 0 3 0 3 0 0 3 3 3 0\n",
      " 3 3 3 3 3 3 3 3 3 3 0 3 3 3 0 3 0 3 5 0 3 3 3 3 3 0 9 3 3 3 3 3 3 3 3 3 3\n",
      " 3 3 9 3 3 3 3 3 0 3 3 3 3 3 3 3 3 3 3 3 0 0 3 3 0 3 9 5 3 3 3 3 0 3 3 3 0\n",
      " 5 9 2 0 0 0 3 3 3 3 3 3 3 3 3 0 3 3 3 9 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3\n",
      " 3 3 3 3 3 3 3 3 9 3 3 3 3 3 3 3 0 3 3 3 3 3 3 3 0 3 3 9 3 5 0 3 3 0 3 3 3\n",
      " 3]\n",
      "threshold density 1.95819\n",
      "cluster signal 61766.0\n",
      "cluster target count 95106\n",
      "!!cluster density 0.6494437785208084\n",
      "cluster signal 0.0\n",
      "cluster target count 227860\n",
      "!!cluster density 0.0\n",
      "cluster signal 41895.0\n",
      "cluster target count 135102\n",
      "!!cluster density 0.3100990362836968\n",
      "cluster signal 115533.0\n",
      "cluster target count 114480\n",
      "!!cluster density 1.0091981132075472\n",
      "cluster signal 83671.0\n",
      "cluster target count 57809\n",
      "!!cluster density 1.4473697867114117\n",
      "cluster signal 73503.0\n",
      "cluster target count 39579\n",
      "!!cluster density 1.8571212006367013\n",
      "cluster signal 20548.0\n",
      "cluster target count 6759\n",
      "!!cluster density 3.040094688563397\n",
      "cluster signal 50439.0\n",
      "cluster target count 21275\n",
      "!!cluster density 2.370810810810811\n",
      "cluster signal 1016.0\n",
      "cluster target count 38\n",
      "!!cluster density 26.736842105263158\n",
      "cluster signal 8540.0\n",
      "cluster target count 1992\n",
      "!!cluster density 4.28714859437751\n",
      "\n",
      "Labels mapping\n",
      " {9: 0, 3: 0, 0: 0, 5: 0, 2: 0, 7: 0, 1: 1, 4: 1, 6: 1, 8: 1}\n",
      "DEBUG: 149654 2964 \n",
      "\n",
      " +----------------------+-------------+----------------+-------+\n",
      "| bin_offset_25        | TT_S0       | target_count   | +9    |\n",
      "| (category)           | (float64)   | (int64)        | ...   |\n",
      "|----------------------+-------------+----------------+-------|\n",
      "| (3000018, 3000043]   | 4.0         | 4              | ...   |\n",
      "| (3000043, 3000143]   | 4.0         | 9              | ...   |\n",
      "| (3000143, 3000243]   | 0.0         | 3              | ...   |\n",
      "| (3000243, 3000343]   | 2.0         | 3              | ...   |\n",
      "| ...                  | ...         | ...            | ...   |\n",
      "| (18261343, 18261443] | 1.0         | 12             | ...   |\n",
      "| (18261443, 18261543] | 3.0         | 4              | ...   |\n",
      "| (18261543, 18261643] | 0.0         | 2              | ...   |\n",
      "| (18261643, 18261710] | 4.0         | 3              | ...   |\n",
      "+----------------------+-------------+----------------+-------+\n",
      "Unstranded PyRanges object has 149,654 rows and 12 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "9 hidden columns: starting_nt, Chromosome, Start, End, genes, transcripts, ... (+ 3 more.)\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0\n",
      "BINS  IntervalIndex([  (3000018, 3000048],   (3000048, 3000148],\n",
      "                 (3000148, 3000248],   (3000248, 3000348],\n",
      "                 (3000348, 3000448],   (3000448, 3000548],\n",
      "                 (3000548, 3000648],   (3000648, 3000748],\n",
      "                 (3000748, 3000848],   (3000848, 3000948],\n",
      "               ...\n",
      "               (18260748, 18260848], (18260848, 18260948],\n",
      "               (18260948, 18261048], (18261048, 18261148],\n",
      "               (18261148, 18261248], (18261248, 18261348],\n",
      "               (18261348, 18261448], (18261448, 18261548],\n",
      "               (18261548, 18261648], (18261648, 18261710]],\n",
      "              dtype='interval[int64, right]', length=152618)\n",
      "HEAD with BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0   \n",
      "\n",
      "        bin_offset_30  target_count  \n",
      "0  (3000018, 3000048]             1  \n",
      "1  (3000018, 3000048]             1  \n",
      "2  (3000018, 3000048]             1  \n",
      "3  (3000018, 3000048]             1  \n",
      "4  (3000048, 3000148]             1  \n",
      "TAIL with BINS\n",
      "       seqnames     start       end        CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  18261611  18261611  18_18261611    0.0    0.0    0.0   \n",
      "699996       18  18261615  18261615  18_18261615    0.0    0.0    0.0   \n",
      "699997       18  18261682  18261682  18_18261682    0.0    0.0    0.0   \n",
      "699998       18  18261702  18261702  18_18261702    4.0    0.0    0.0   \n",
      "699999       18  18261710  18261710  18_18261710    0.0    0.0    0.0   \n",
      "\n",
      "               bin_offset_30  target_count  \n",
      "699995  (18261548, 18261648]             1  \n",
      "699996  (18261548, 18261648]             1  \n",
      "699997  (18261648, 18261710]             1  \n",
      "699998  (18261648, 18261710]             1  \n",
      "699999  (18261648, 18261710]             1  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:137: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "         bin_offset_30  TT_S0  target_count starting_nt\n",
      "0   (3000018, 3000048]    4.0             4     3000018\n",
      "1   (3000048, 3000148]    4.0             9     3000048\n",
      "2   (3000148, 3000248]    0.0             3     3000148\n",
      "3   (3000248, 3000348]    2.0             3     3000248\n",
      "4   (3000348, 3000448]    0.0             5     3000348\n",
      "5   (3000448, 3000548]    0.0             1     3000448\n",
      "6   (3000548, 3000648]    0.0             2     3000548\n",
      "7   (3000648, 3000748]    0.0             5     3000648\n",
      "8   (3000748, 3000848]    0.0             6     3000748\n",
      "9   (3000848, 3000948]    0.0             5     3000848\n",
      "10  (3000948, 3001048]    2.0             3     3000948\n",
      "11  (3001048, 3001148]    0.0             4     3001048\n",
      "12  (3001148, 3001248]    0.0             2     3001148\n",
      "13  (3001248, 3001348]    0.0             5     3001248\n",
      "14  (3001348, 3001448]    0.0             2     3001348\n",
      "15  (3001448, 3001548]    1.0             8     3001448\n",
      "16  (3001548, 3001648]    5.0             8     3001548\n",
      "17  (3001648, 3001748]    1.0             7     3001648\n",
      "18  (3001748, 3001848]    0.0             5     3001748\n",
      "19  (3001848, 3001948]    0.0             2     3001848\n",
      "20  (3001948, 3002048]    1.0             2     3001948\n",
      "21  (3002048, 3002148]    0.0             3     3002048\n",
      "22  (3002148, 3002248]    4.0             5     3002148\n",
      "23  (3002248, 3002348]    7.0             4     3002248\n",
      "24  (3002348, 3002448]   14.0             4     3002348\n",
      "25  (3002448, 3002548]   11.0             4     3002448\n",
      "26  (3002548, 3002648]    0.0             1     3002548\n",
      "27  (3002648, 3002748]    0.0             2     3002648\n",
      "28  (3002748, 3002848]    0.0             3     3002748\n",
      "29  (3002848, 3002948]    0.0             0     3002848\n",
      "30  (3002948, 3003048]    1.0             8     3002948\n",
      "31  (3003048, 3003148]    2.0             2     3003048\n",
      "32  (3003148, 3003248]    0.0             0     3003148\n",
      "33  (3003248, 3003348]    1.0             5     3003248\n",
      "34  (3003348, 3003448]    0.0             4     3003348\n",
      "35  (3003448, 3003548]    0.0             4     3003448\n",
      "36  (3003548, 3003648]    1.0             1     3003548\n",
      "37  (3003648, 3003748]    3.0             5     3003648\n",
      "38  (3003748, 3003848]    1.0             5     3003748\n",
      "39  (3003848, 3003948]    1.0             5     3003848\n",
      "40  (3003948, 3004048]    7.0             4     3003948\n",
      "41  (3004048, 3004148]    0.0             0     3004048\n",
      "42  (3004148, 3004248]    1.0             4     3004148\n",
      "43  (3004248, 3004348]    4.0             3     3004248\n",
      "44  (3004348, 3004448]    0.0             1     3004348\n",
      "45  (3004448, 3004548]    0.0             3     3004448\n",
      "46  (3004548, 3004648]    1.0             5     3004548\n",
      "47  (3004648, 3004748]    0.0             0     3004648\n",
      "48  (3004748, 3004848]    2.0             3     3004748\n",
      "49  (3004848, 3004948]   11.0             6     3004848\n",
      "(152618, 4)\n",
      "bin_offset_30    category\n",
      "TT_S0             float64\n",
      "target_count        int64\n",
      "starting_nt      category\n",
      "dtype: object\n",
      "sumos ilgis: 152618\n",
      "ranges ilgis: 152618\n",
      "Model input - NO SCALE\n",
      "        TT_S0  genes  transcripts  exons  CDSs\n",
      "0         4.0      0            0      0     0\n",
      "1         4.0      0            0      0     0\n",
      "2         0.0      0            0      0     0\n",
      "3         2.0      0            0      0     0\n",
      "4         0.0      0            0      0     0\n",
      "...       ...    ...          ...    ...   ...\n",
      "152613   13.0      0            0      0     0\n",
      "152614    1.0      0            0      0     0\n",
      "152615    3.0      0            0      0     0\n",
      "152616    0.0      0            0      0     0\n",
      "152617    4.0      0            0      0     0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "        TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0         4.0             0                   0             0            0\n",
      "1         4.0             0                   0             0            0\n",
      "2         0.0             0                   0             0            0\n",
      "3         2.0             0                   0             0            0\n",
      "4         0.0             0                   0             0            0\n",
      "...       ...           ...                 ...           ...          ...\n",
      "152613   13.0             0                   0             0            0\n",
      "152614    1.0             0                   0             0            0\n",
      "152615    3.0             0                   0             0            0\n",
      "152616    0.0             0                   0             0            0\n",
      "152617    4.0             0                   0             0            0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "0.05010549214378383\n",
      "[6 6 8 0 8 8 8 8 8 8 0 8 8 8 8 0 6 0 8 8 0 8 6 3 1 1 8 8 8 8 0 0 8 0 8 8 0\n",
      " 6 0 0 3 8 0 6 8 8 0 8 0 1 8 2 1 2 8 4 2 5 5 5 9 3 4 0 6 8 8 3 8 0 3 8 6 0\n",
      " 8 8 0 8 8 8 6 3 0 6 8 6 0 3 0 8 6 1 3 0 0 8 6 6 8 1 7 7 2 4 6 3 1 6 4 6 3\n",
      " 8 7 6 8 0 0 8 8 8 8 0 3 0 0 1 0 7 7 1 1 0 1 6 6 8 6 3 6 8 8 8 8 8 0 0 8 1\n",
      " 3 3 6 0 8 0 3 3 6 6 0 1 3 8 1 0 6 7 0 0 8 8 8 0 8 8 0 8 8 8 8 0 3 3 8 6 8\n",
      " 8 6 8 8 8 8 8 0 8 8 8 8 8 8 8 8 8 8 8 8 8 3 1 7 8 0 8 0 8 8 8 8 0 0 8 8 8\n",
      " 8 8 8 8 8 8 8 8 8 8 8 8 8 6 8 8 8 0 8 8 8 8 8 8 8 8 8 8 8 0 8 8 8 8 0 8 0\n",
      " 8 8 8 8 8 8 0 8 8 6 8 0 8 0 6 8 0 8 0 8 0 0 8 8 8 0 8 8 8 8 8 8 0 8 8 8 8\n",
      " 6 8 8 8 0 0 6 0 8 8 8 8 8 6 6 3 6 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8\n",
      " 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8\n",
      " 8 8 8 8 8 8 8 8 8 8 8 6 8 0 6 3 0 0 0 0 8 8 8 8 3 1 3 1 8 7 6 3 6 0 0 3 6\n",
      " 6 8 6 0 8 0 1 3 1 0 1 0 8 6 8 8 0 8 6 8 0 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8\n",
      " 6 8 0 8 6 8 8 8 8 0 0 8 8 0 8 0 0 8 1 0 8 0 0 6 1 8 8 6 3 3 3 0 6 8 6 0 8\n",
      " 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8\n",
      " 3 1 0 8 8 0 6 8 0 3 6 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8\n",
      " 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8\n",
      " 8 8 8 8 6 8 0 6 8 8 8 8 8 0 8 8 8 8 0 8 0 0 0 8 6 0 8 8 8 8 0 3 8 8 8 8 3\n",
      " 8 8 6 8 8 8 8 0 8 3 3 8 0 3 6 8 8 8 8 0 8 3 0 0 8 8 8 8 0 3 8 0 6 6 8 0 3\n",
      " 6 6 8 8 8 8 0 8 8 8 8 8 8 8 8 8 8 8 6 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8\n",
      " 8 8 8 8 8 8 8 8 8 8 8 8 8 6 6 3 6 8 6 8 0 0 8 8 0 8 8 8 6 8 8 8 8 8 0 8 8\n",
      " 8 8 0 8 8 0 8 0 8 8 8 8 8 8 0 8 8 8 8 8 0 8 8 8 8 8 6 8 8 8 8 0 8 6 8 8 8\n",
      " 8 8 8 8 8 8 8 0 8 8 8 8 8 0 6 8 8 8 0 0 8 8 0 0 8 8 8 1 0 6 1 8 6 0 0 6 8\n",
      " 8 8 8 8 0 8 8 8 8 8 8 8 8 8 8 8 8 8 0 0 8 0 0 8 6 8 8 0 8 0 8 6 8 8 8 8 0\n",
      " 8 8 8 8 8 8 8 8 8 8 0 8 8 8 0 8 0 8 3 0 8 8 8 8 8 0 6 8 8 8 8 8 8 8 8 8 8\n",
      " 8 8 6 8 8 8 8 8 0 8 8 8 8 8 8 8 8 8 8 8 0 0 8 8 0 8 6 6 8 8 8 8 0 8 8 8 0\n",
      " 3 6 1 0 0 0 8 8 8 8 8 8 8 8 8 0 8 8 8 6 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8\n",
      " 8 8 8 8 8 8 8 8 6 8 8 8 8 8 8 8 0 8 8 8 8 8 8 8 0 8 8 6 8 3 0 8 8 0 8 8 8\n",
      " 8]\n",
      "threshold density 1.95819\n",
      "cluster signal 93004.0\n",
      "cluster target count 131376\n",
      "!!cluster density 0.7079222993545244\n",
      "cluster signal 0.0\n",
      "cluster target count 228069\n",
      "!!cluster density 0.0\n",
      "cluster signal 41925.0\n",
      "cluster target count 135138\n",
      "!!cluster density 0.3102384229454336\n",
      "cluster signal 108587.0\n",
      "cluster target count 96310\n",
      "!!cluster density 1.1274737825770949\n",
      "cluster signal 88886.0\n",
      "cluster target count 56667\n",
      "!!cluster density 1.5685672437220957\n",
      "cluster signal 16927.0\n",
      "cluster target count 5366\n",
      "!!cluster density 3.154491241147969\n",
      "cluster signal 34819.0\n",
      "cluster target count 13605\n",
      "!!cluster density 2.559279676589489\n",
      "cluster signal 985.0\n",
      "cluster target count 37\n",
      "!!cluster density 26.62162162162162\n",
      "cluster signal 7065.0\n",
      "cluster target count 1619\n",
      "!!cluster density 4.3638048177887585\n",
      "cluster signal 64713.0\n",
      "cluster target count 31813\n",
      "!!cluster density 2.0341684217143934\n",
      "\n",
      "Labels mapping\n",
      " {6: 0, 8: 0, 0: 0, 3: 0, 1: 0, 2: 1, 4: 1, 5: 1, 9: 1, 7: 1}\n",
      "DEBUG: 146964 5654 \n",
      "\n",
      " +----------------------+-------------+----------------+-------+\n",
      "| bin_offset_30        | TT_S0       | target_count   | +9    |\n",
      "| (category)           | (float64)   | (int64)        | ...   |\n",
      "|----------------------+-------------+----------------+-------|\n",
      "| (3000018, 3000048]   | 4.0         | 4              | ...   |\n",
      "| (3000048, 3000148]   | 4.0         | 9              | ...   |\n",
      "| (3000148, 3000248]   | 0.0         | 3              | ...   |\n",
      "| (3000248, 3000348]   | 2.0         | 3              | ...   |\n",
      "| ...                  | ...         | ...            | ...   |\n",
      "| (18261348, 18261448] | 1.0         | 10             | ...   |\n",
      "| (18261448, 18261548] | 3.0         | 4              | ...   |\n",
      "| (18261548, 18261648] | 0.0         | 2              | ...   |\n",
      "| (18261648, 18261710] | 4.0         | 3              | ...   |\n",
      "+----------------------+-------------+----------------+-------+\n",
      "Unstranded PyRanges object has 146,964 rows and 12 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "9 hidden columns: starting_nt, Chromosome, Start, End, genes, transcripts, ... (+ 3 more.)\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0\n",
      "BINS  IntervalIndex([  (3000018, 3000053],   (3000053, 3000153],\n",
      "                 (3000153, 3000253],   (3000253, 3000353],\n",
      "                 (3000353, 3000453],   (3000453, 3000553],\n",
      "                 (3000553, 3000653],   (3000653, 3000753],\n",
      "                 (3000753, 3000853],   (3000853, 3000953],\n",
      "               ...\n",
      "               (18260753, 18260853], (18260853, 18260953],\n",
      "               (18260953, 18261053], (18261053, 18261153],\n",
      "               (18261153, 18261253], (18261253, 18261353],\n",
      "               (18261353, 18261453], (18261453, 18261553],\n",
      "               (18261553, 18261653], (18261653, 18261710]],\n",
      "              dtype='interval[int64, right]', length=152618)\n",
      "HEAD with BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0   \n",
      "\n",
      "        bin_offset_35  target_count  \n",
      "0  (3000018, 3000053]             1  \n",
      "1  (3000018, 3000053]             1  \n",
      "2  (3000018, 3000053]             1  \n",
      "3  (3000018, 3000053]             1  \n",
      "4  (3000053, 3000153]             1  \n",
      "TAIL with BINS\n",
      "       seqnames     start       end        CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  18261611  18261611  18_18261611    0.0    0.0    0.0   \n",
      "699996       18  18261615  18261615  18_18261615    0.0    0.0    0.0   \n",
      "699997       18  18261682  18261682  18_18261682    0.0    0.0    0.0   \n",
      "699998       18  18261702  18261702  18_18261702    4.0    0.0    0.0   \n",
      "699999       18  18261710  18261710  18_18261710    0.0    0.0    0.0   \n",
      "\n",
      "               bin_offset_35  target_count  \n",
      "699995  (18261553, 18261653]             1  \n",
      "699996  (18261553, 18261653]             1  \n",
      "699997  (18261653, 18261710]             1  \n",
      "699998  (18261653, 18261710]             1  \n",
      "699999  (18261653, 18261710]             1  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:137: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "         bin_offset_35  TT_S0  target_count starting_nt\n",
      "0   (3000018, 3000053]    4.0             4     3000018\n",
      "1   (3000053, 3000153]    4.0             9     3000053\n",
      "2   (3000153, 3000253]    0.0             3     3000153\n",
      "3   (3000253, 3000353]    2.0             4     3000253\n",
      "4   (3000353, 3000453]    0.0             4     3000353\n",
      "5   (3000453, 3000553]    0.0             1     3000453\n",
      "6   (3000553, 3000653]    0.0             2     3000553\n",
      "7   (3000653, 3000753]    0.0             6     3000653\n",
      "8   (3000753, 3000853]    0.0             5     3000753\n",
      "9   (3000853, 3000953]    0.0             5     3000853\n",
      "10  (3000953, 3001053]    2.0             3     3000953\n",
      "11  (3001053, 3001153]    0.0             4     3001053\n",
      "12  (3001153, 3001253]    0.0             2     3001153\n",
      "13  (3001253, 3001353]    0.0             5     3001253\n",
      "14  (3001353, 3001453]    0.0             3     3001353\n",
      "15  (3001453, 3001553]    1.0             8     3001453\n",
      "16  (3001553, 3001653]    5.0             7     3001553\n",
      "17  (3001653, 3001753]    1.0             7     3001653\n",
      "18  (3001753, 3001853]    0.0             5     3001753\n",
      "19  (3001853, 3001953]    0.0             2     3001853\n",
      "20  (3001953, 3002053]    1.0             2     3001953\n",
      "21  (3002053, 3002153]    0.0             3     3002053\n",
      "22  (3002153, 3002253]    4.0             6     3002153\n",
      "23  (3002253, 3002353]    7.0             3     3002253\n",
      "24  (3002353, 3002453]   14.0             4     3002353\n",
      "25  (3002453, 3002553]   11.0             4     3002453\n",
      "26  (3002553, 3002653]    0.0             1     3002553\n",
      "27  (3002653, 3002753]    0.0             2     3002653\n",
      "28  (3002753, 3002853]    0.0             3     3002753\n",
      "29  (3002853, 3002953]    0.0             1     3002853\n",
      "30  (3002953, 3003053]    1.0             7     3002953\n",
      "31  (3003053, 3003153]    2.0             2     3003053\n",
      "32  (3003153, 3003253]    0.0             0     3003153\n",
      "33  (3003253, 3003353]    1.0             5     3003253\n",
      "34  (3003353, 3003453]    0.0             4     3003353\n",
      "35  (3003453, 3003553]    0.0             4     3003453\n",
      "36  (3003553, 3003653]    1.0             1     3003553\n",
      "37  (3003653, 3003753]    3.0             5     3003653\n",
      "38  (3003753, 3003853]    1.0             5     3003753\n",
      "39  (3003853, 3003953]    1.0             6     3003853\n",
      "40  (3003953, 3004053]    7.0             3     3003953\n",
      "41  (3004053, 3004153]    0.0             0     3004053\n",
      "42  (3004153, 3004253]    1.0             4     3004153\n",
      "43  (3004253, 3004353]    4.0             3     3004253\n",
      "44  (3004353, 3004453]    0.0             1     3004353\n",
      "45  (3004453, 3004553]    0.0             3     3004453\n",
      "46  (3004553, 3004653]    1.0             5     3004553\n",
      "47  (3004653, 3004753]    0.0             0     3004653\n",
      "48  (3004753, 3004853]   10.0             4     3004753\n",
      "49  (3004853, 3004953]    3.0             5     3004853\n",
      "(152618, 4)\n",
      "bin_offset_35    category\n",
      "TT_S0             float64\n",
      "target_count        int64\n",
      "starting_nt      category\n",
      "dtype: object\n",
      "sumos ilgis: 152618\n",
      "ranges ilgis: 152618\n",
      "Model input - NO SCALE\n",
      "        TT_S0  genes  transcripts  exons  CDSs\n",
      "0         4.0      0            0      0     0\n",
      "1         4.0      0            0      0     0\n",
      "2         0.0      0            0      0     0\n",
      "3         2.0      0            0      0     0\n",
      "4         0.0      0            0      0     0\n",
      "...       ...    ...          ...    ...   ...\n",
      "152613   13.0      0            0      0     0\n",
      "152614    1.0      0            0      0     0\n",
      "152615    3.0      0            0      0     0\n",
      "152616    0.0      0            0      0     0\n",
      "152617    4.0      0            0      0     0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "        TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0         4.0             0                   0             0            0\n",
      "1         4.0             0                   0             0            0\n",
      "2         0.0             0                   0             0            0\n",
      "3         2.0             0                   0             0            0\n",
      "4         0.0             0                   0             0            0\n",
      "...       ...           ...                 ...           ...          ...\n",
      "152613   13.0             0                   0             0            0\n",
      "152614    1.0             0                   0             0            0\n",
      "152615    3.0             0                   0             0            0\n",
      "152616    0.0             0                   0             0            0\n",
      "152617    4.0             0                   0             0            0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "0.10697296518104024\n",
      "[8 8 7 8 7 7 7 7 7 7 8 7 7 7 7 1 0 1 7 7 1 7 8 0 6 6 7 7 7 7 1 8 7 1 7 7 1\n",
      " 8 1 1 0 7 1 8 7 7 1 7 6 8 7 3 6 3 7 3 3 4 4 9 3 0 2 1 8 7 7 0 7 1 0 7 8 1\n",
      " 7 7 8 7 7 7 8 0 8 8 7 8 8 0 1 7 8 6 6 8 8 7 8 8 7 6 2 6 3 3 8 0 6 8 3 8 0\n",
      " 8 6 8 7 1 1 7 7 7 7 1 0 1 1 6 8 2 2 6 6 1 6 8 0 7 0 6 8 7 7 7 7 7 8 1 7 6\n",
      " 0 0 0 8 7 1 0 6 8 8 8 6 0 8 6 1 0 2 1 8 7 7 7 1 7 7 8 7 7 7 7 1 6 0 7 8 7\n",
      " 7 8 7 7 7 7 7 1 7 7 7 7 7 7 7 7 7 7 7 7 7 0 6 2 1 7 7 8 7 7 7 7 1 1 7 7 7\n",
      " 7 7 7 7 7 7 7 7 7 7 7 7 7 8 7 7 7 1 7 7 7 7 7 7 7 7 7 7 7 1 7 7 7 7 8 7 1\n",
      " 7 7 7 7 7 7 8 7 7 8 7 8 7 8 8 7 1 7 1 7 1 1 7 7 1 7 7 7 7 7 7 7 1 7 7 7 7\n",
      " 0 7 7 7 1 8 8 8 7 7 7 7 7 0 8 0 0 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7\n",
      " 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7\n",
      " 7 7 7 7 7 7 7 7 7 7 7 0 7 8 8 6 1 1 1 1 7 7 7 7 6 6 0 6 7 2 0 8 8 1 8 0 0\n",
      " 8 8 8 8 7 1 6 2 8 0 0 8 7 8 7 7 8 7 0 7 1 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7\n",
      " 8 7 1 7 0 7 7 7 7 8 1 7 7 1 7 1 8 7 6 8 7 1 1 8 6 7 7 8 0 0 0 8 8 7 8 1 7\n",
      " 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7\n",
      " 0 6 7 7 7 1 8 7 1 0 8 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7\n",
      " 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7\n",
      " 7 7 7 7 8 8 7 8 7 7 7 7 7 1 7 7 7 7 1 7 1 8 8 7 0 8 7 7 7 7 0 1 7 7 7 7 0\n",
      " 7 7 8 7 7 7 7 1 7 6 8 7 1 6 8 7 7 7 7 1 7 0 1 8 7 7 7 7 1 6 7 1 0 8 7 1 0\n",
      " 8 8 7 7 7 7 8 7 7 7 7 7 7 7 7 7 7 8 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7\n",
      " 7 7 7 7 7 7 7 7 7 7 7 7 7 8 8 6 8 7 8 7 8 1 7 7 1 7 7 7 8 7 7 7 7 7 1 7 7\n",
      " 7 7 1 7 7 1 7 1 7 7 7 7 7 7 1 7 7 7 7 7 1 7 7 7 7 7 8 7 7 7 7 8 7 8 7 7 7\n",
      " 7 7 7 7 7 7 7 8 7 7 7 7 7 8 8 7 7 7 1 8 7 7 8 7 7 7 7 6 1 0 6 7 8 1 8 7 7\n",
      " 7 7 7 7 1 7 7 7 7 7 7 7 7 7 7 7 7 7 1 8 7 1 1 7 8 7 7 1 7 8 7 8 7 7 7 7 1\n",
      " 7 7 7 7 7 7 7 7 7 7 8 7 7 7 1 7 8 7 0 1 7 7 7 7 7 8 8 7 7 7 7 7 7 7 7 7 7\n",
      " 7 7 8 7 7 7 7 7 1 7 7 7 7 7 7 7 7 7 7 7 1 8 7 7 8 7 8 0 7 7 7 7 1 7 7 7 0\n",
      " 8 8 6 1 8 8 7 7 7 7 7 7 7 7 7 8 7 7 7 8 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7\n",
      " 7 7 7 7 7 7 7 7 8 7 7 7 7 7 7 7 1 7 7 7 7 7 7 7 1 7 7 8 7 0 1 7 7 1 7 7 7\n",
      " 7]\n",
      "threshold density 1.95819\n",
      "cluster signal 87080.0\n",
      "cluster target count 156624\n",
      "!!cluster density 0.555981203391562\n",
      "cluster signal 0.0\n",
      "cluster target count 228334\n",
      "!!cluster density 0.0\n",
      "cluster signal 16326.0\n",
      "cluster target count 72953\n",
      "!!cluster density 0.22378791824873548\n",
      "cluster signal 115754.0\n",
      "cluster target count 114612\n",
      "!!cluster density 1.009964052629742\n",
      "cluster signal 113463.0\n",
      "cluster target count 75113\n",
      "!!cluster density 1.5105640834476055\n",
      "cluster signal 31051.0\n",
      "cluster target count 10798\n",
      "!!cluster density 2.8756251157621784\n",
      "cluster signal 600.0\n",
      "cluster target count 35\n",
      "!!cluster density 17.142857142857142\n",
      "cluster signal 461.0\n",
      "cluster target count 4\n",
      "!!cluster density 115.25\n",
      "cluster signal 82159.0\n",
      "cluster target count 39037\n",
      "!!cluster density 2.1046443118067475\n",
      "cluster signal 10017.0\n",
      "cluster target count 2490\n",
      "!!cluster density 4.02289156626506\n",
      "\n",
      "Labels mapping\n",
      " {8: 0, 7: 0, 1: 0, 0: 0, 6: 0, 3: 1, 4: 1, 9: 1, 2: 1, 5: 1}\n",
      "DEBUG: 146970 5648 \n",
      "\n",
      " +----------------------+-------------+----------------+-------+\n",
      "| bin_offset_35        | TT_S0       | target_count   | +9    |\n",
      "| (category)           | (float64)   | (int64)        | ...   |\n",
      "|----------------------+-------------+----------------+-------|\n",
      "| (3000018, 3000053]   | 4.0         | 4              | ...   |\n",
      "| (3000053, 3000153]   | 4.0         | 9              | ...   |\n",
      "| (3000153, 3000253]   | 0.0         | 3              | ...   |\n",
      "| (3000253, 3000353]   | 2.0         | 4              | ...   |\n",
      "| ...                  | ...         | ...            | ...   |\n",
      "| (18261353, 18261453] | 1.0         | 9              | ...   |\n",
      "| (18261453, 18261553] | 3.0         | 4              | ...   |\n",
      "| (18261553, 18261653] | 0.0         | 2              | ...   |\n",
      "| (18261653, 18261710] | 4.0         | 3              | ...   |\n",
      "+----------------------+-------------+----------------+-------+\n",
      "Unstranded PyRanges object has 146,970 rows and 12 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "9 hidden columns: starting_nt, Chromosome, Start, End, genes, transcripts, ... (+ 3 more.)\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0\n",
      "BINS  IntervalIndex([  (3000018, 3000058],   (3000058, 3000158],\n",
      "                 (3000158, 3000258],   (3000258, 3000358],\n",
      "                 (3000358, 3000458],   (3000458, 3000558],\n",
      "                 (3000558, 3000658],   (3000658, 3000758],\n",
      "                 (3000758, 3000858],   (3000858, 3000958],\n",
      "               ...\n",
      "               (18260758, 18260858], (18260858, 18260958],\n",
      "               (18260958, 18261058], (18261058, 18261158],\n",
      "               (18261158, 18261258], (18261258, 18261358],\n",
      "               (18261358, 18261458], (18261458, 18261558],\n",
      "               (18261558, 18261658], (18261658, 18261710]],\n",
      "              dtype='interval[int64, right]', length=152618)\n",
      "HEAD with BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0   \n",
      "\n",
      "        bin_offset_40  target_count  \n",
      "0  (3000018, 3000058]             1  \n",
      "1  (3000018, 3000058]             1  \n",
      "2  (3000018, 3000058]             1  \n",
      "3  (3000018, 3000058]             1  \n",
      "4  (3000058, 3000158]             1  \n",
      "TAIL with BINS\n",
      "       seqnames     start       end        CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  18261611  18261611  18_18261611    0.0    0.0    0.0   \n",
      "699996       18  18261615  18261615  18_18261615    0.0    0.0    0.0   \n",
      "699997       18  18261682  18261682  18_18261682    0.0    0.0    0.0   \n",
      "699998       18  18261702  18261702  18_18261702    4.0    0.0    0.0   \n",
      "699999       18  18261710  18261710  18_18261710    0.0    0.0    0.0   \n",
      "\n",
      "               bin_offset_40  target_count  \n",
      "699995  (18261558, 18261658]             1  \n",
      "699996  (18261558, 18261658]             1  \n",
      "699997  (18261658, 18261710]             1  \n",
      "699998  (18261658, 18261710]             1  \n",
      "699999  (18261658, 18261710]             1  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:137: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "         bin_offset_40  TT_S0  target_count starting_nt\n",
      "0   (3000018, 3000058]    4.0             4     3000018\n",
      "1   (3000058, 3000158]    4.0             9     3000058\n",
      "2   (3000158, 3000258]    0.0             3     3000158\n",
      "3   (3000258, 3000358]    2.0             5     3000258\n",
      "4   (3000358, 3000458]    0.0             3     3000358\n",
      "5   (3000458, 3000558]    0.0             1     3000458\n",
      "6   (3000558, 3000658]    0.0             2     3000558\n",
      "7   (3000658, 3000758]    0.0             6     3000658\n",
      "8   (3000758, 3000858]    0.0             5     3000758\n",
      "9   (3000858, 3000958]    0.0             5     3000858\n",
      "10  (3000958, 3001058]    2.0             5     3000958\n",
      "11  (3001058, 3001158]    0.0             2     3001058\n",
      "12  (3001158, 3001258]    0.0             2     3001158\n",
      "13  (3001258, 3001358]    0.0             5     3001258\n",
      "14  (3001358, 3001458]    0.0             4     3001358\n",
      "15  (3001458, 3001558]    1.0             7     3001458\n",
      "16  (3001558, 3001658]    5.0             7     3001558\n",
      "17  (3001658, 3001758]    1.0             7     3001658\n",
      "18  (3001758, 3001858]    0.0             5     3001758\n",
      "19  (3001858, 3001958]    0.0             2     3001858\n",
      "20  (3001958, 3002058]    1.0             2     3001958\n",
      "21  (3002058, 3002158]    0.0             3     3002058\n",
      "22  (3002158, 3002258]    4.0             6     3002158\n",
      "23  (3002258, 3002358]    9.0             4     3002258\n",
      "24  (3002358, 3002458]   19.0             4     3002358\n",
      "25  (3002458, 3002558]    4.0             3     3002458\n",
      "26  (3002558, 3002658]    0.0             1     3002558\n",
      "27  (3002658, 3002758]    0.0             3     3002658\n",
      "28  (3002758, 3002858]    0.0             2     3002758\n",
      "29  (3002858, 3002958]    0.0             1     3002858\n",
      "30  (3002958, 3003058]    3.0             8     3002958\n",
      "31  (3003058, 3003158]    0.0             1     3003058\n",
      "32  (3003158, 3003258]    0.0             0     3003158\n",
      "33  (3003258, 3003358]    1.0             6     3003258\n",
      "34  (3003358, 3003458]    0.0             3     3003358\n",
      "35  (3003458, 3003558]    0.0             4     3003458\n",
      "36  (3003558, 3003658]    1.0             1     3003558\n",
      "37  (3003658, 3003758]    4.0             6     3003658\n",
      "38  (3003758, 3003858]    0.0             4     3003758\n",
      "39  (3003858, 3003958]    1.0             6     3003858\n",
      "40  (3003958, 3004058]    7.0             3     3003958\n",
      "41  (3004058, 3004158]    0.0             0     3004058\n",
      "42  (3004158, 3004258]    1.0             4     3004158\n",
      "43  (3004258, 3004358]    4.0             3     3004258\n",
      "44  (3004358, 3004458]    0.0             1     3004358\n",
      "45  (3004458, 3004558]    0.0             5     3004458\n",
      "46  (3004558, 3004658]    1.0             3     3004558\n",
      "47  (3004658, 3004758]    0.0             0     3004658\n",
      "48  (3004758, 3004858]   10.0             4     3004758\n",
      "49  (3004858, 3004958]    3.0             5     3004858\n",
      "(152618, 4)\n",
      "bin_offset_40    category\n",
      "TT_S0             float64\n",
      "target_count        int64\n",
      "starting_nt      category\n",
      "dtype: object\n",
      "sumos ilgis: 152618\n",
      "ranges ilgis: 152618\n",
      "Model input - NO SCALE\n",
      "        TT_S0  genes  transcripts  exons  CDSs\n",
      "0         4.0      0            0      0     0\n",
      "1         4.0      0            0      0     0\n",
      "2         0.0      0            0      0     0\n",
      "3         2.0      0            0      0     0\n",
      "4         0.0      0            0      0     0\n",
      "...       ...    ...          ...    ...   ...\n",
      "152613   13.0      0            0      0     0\n",
      "152614    4.0      0            0      0     0\n",
      "152615    0.0      0            0      0     0\n",
      "152616    0.0      0            0      0     0\n",
      "152617    4.0      0            0      0     0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "        TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0         4.0             0                   0             0            0\n",
      "1         4.0             0                   0             0            0\n",
      "2         0.0             0                   0             0            0\n",
      "3         2.0             0                   0             0            0\n",
      "4         0.0             0                   0             0            0\n",
      "...       ...           ...                 ...           ...          ...\n",
      "152613   13.0             0                   0             0            0\n",
      "152614    4.0             0                   0             0            0\n",
      "152615    0.0             0                   0             0            0\n",
      "152616    0.0             0                   0             0            0\n",
      "152617    4.0             0                   0             0            0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "0.011027532794296873\n",
      "[4 4 8 0 8 8 8 8 8 8 0 8 8 8 8 0 4 0 8 8 0 8 4 2 9 4 8 8 8 8 4 8 8 0 8 8 0\n",
      " 4 8 0 2 8 0 4 8 8 0 8 2 4 8 7 4 7 8 7 1 3 3 5 0 2 1 0 4 8 8 2 8 0 2 8 4 0\n",
      " 8 8 0 8 8 0 0 2 0 4 8 4 0 2 8 8 4 6 2 0 0 8 4 4 8 2 9 9 1 1 4 4 2 1 4 4 2\n",
      " 4 6 4 8 0 0 8 8 8 8 0 4 0 0 6 0 9 9 6 6 0 6 0 4 4 0 2 4 8 8 8 8 0 8 0 8 9\n",
      " 0 2 4 0 8 0 4 2 4 4 0 2 2 4 2 0 2 6 0 0 8 8 8 0 8 8 0 8 8 8 0 8 2 4 8 4 8\n",
      " 8 4 8 8 8 8 8 0 8 8 8 8 8 8 8 8 8 8 8 8 8 2 6 9 0 8 8 0 8 8 8 8 0 0 8 8 8\n",
      " 8 8 8 8 8 8 8 8 8 8 8 8 8 4 8 8 8 0 8 8 8 8 8 8 8 8 8 8 8 0 8 8 8 8 0 8 0\n",
      " 8 8 8 8 8 8 0 8 8 4 8 0 8 4 0 0 8 8 0 8 0 0 8 8 0 8 8 8 8 8 8 8 0 8 8 8 8\n",
      " 4 8 8 0 8 4 0 0 8 8 8 8 8 4 4 4 4 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8\n",
      " 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8\n",
      " 8 8 8 8 8 8 8 8 8 8 8 4 8 0 4 2 0 0 8 0 8 8 8 8 2 6 2 6 8 9 2 4 4 0 0 6 8\n",
      " 4 4 0 0 8 0 6 9 4 4 2 0 8 4 8 8 0 8 4 8 0 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8\n",
      " 4 8 0 8 4 8 8 8 8 0 0 8 0 8 8 0 0 8 6 0 8 0 0 4 2 8 8 4 2 4 2 4 0 8 4 0 8\n",
      " 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8\n",
      " 2 6 8 8 8 4 8 8 0 2 4 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8\n",
      " 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8\n",
      " 8 8 8 8 4 0 8 4 8 8 8 8 8 0 8 8 8 8 0 8 0 0 0 8 4 0 8 8 8 8 2 0 8 8 8 8 4\n",
      " 8 8 4 8 8 8 8 0 8 2 0 8 0 2 4 8 8 8 8 0 8 4 0 0 8 8 8 0 8 2 8 0 4 4 8 0 4\n",
      " 4 4 8 8 8 8 0 8 8 8 8 8 8 8 8 8 8 4 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8\n",
      " 8 8 8 8 8 8 8 8 8 8 8 8 8 4 4 2 4 8 4 8 0 0 8 8 0 8 8 8 4 8 8 8 8 8 0 8 8\n",
      " 8 0 8 8 8 0 8 0 8 8 8 8 8 8 0 8 8 8 8 8 0 8 8 8 8 8 4 8 8 8 8 0 4 8 8 8 8\n",
      " 8 8 8 8 8 8 8 0 8 8 8 8 8 0 4 8 8 8 0 0 8 8 4 8 8 8 8 2 0 4 2 8 4 0 4 8 8\n",
      " 8 8 8 8 0 8 8 8 8 8 8 8 8 8 8 8 8 8 0 0 8 0 0 8 4 8 8 0 8 0 8 4 8 8 8 0 8\n",
      " 8 8 8 8 8 8 8 8 8 8 0 8 8 8 0 8 0 8 2 0 8 8 8 8 0 0 4 8 8 8 8 8 8 8 8 8 8\n",
      " 8 8 4 8 8 8 8 8 0 8 8 8 8 8 8 8 8 8 8 8 0 0 8 0 0 8 4 4 8 8 8 8 0 8 8 0 4\n",
      " 4 4 2 0 0 0 8 8 8 8 8 8 8 8 8 0 8 8 8 4 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8\n",
      " 8 8 8 8 8 8 8 8 4 8 8 8 8 8 8 8 0 8 8 8 8 8 8 8 0 8 8 4 8 2 0 8 8 0 8 8 8\n",
      " 8]\n",
      "threshold density 1.95819\n",
      "cluster signal 123700.0\n",
      "cluster target count 163027\n",
      "!!cluster density 0.75877001968999\n",
      "cluster signal 0.0\n",
      "cluster target count 228239\n",
      "!!cluster density 0.0\n",
      "cluster signal 41894.0\n",
      "cluster target count 134691\n",
      "!!cluster density 0.3110378570208848\n",
      "cluster signal 99646.0\n",
      "cluster target count 80289\n",
      "!!cluster density 1.2410915567512362\n",
      "cluster signal 58374.0\n",
      "cluster target count 29017\n",
      "!!cluster density 2.011717269187028\n",
      "cluster signal 18308.0\n",
      "cluster target count 5569\n",
      "!!cluster density 3.287484288022984\n",
      "cluster signal 41814.0\n",
      "cluster target count 16701\n",
      "!!cluster density 2.503682414226693\n",
      "cluster signal 5415.0\n",
      "cluster target count 1075\n",
      "!!cluster density 5.037209302325581\n",
      "cluster signal 490.0\n",
      "cluster target count 5\n",
      "!!cluster density 98.0\n",
      "cluster signal 67270.0\n",
      "cluster target count 41387\n",
      "!!cluster density 1.625389615096528\n",
      "\n",
      "Labels mapping\n",
      " {4: 0, 8: 0, 0: 0, 2: 0, 9: 1, 7: 1, 1: 1, 3: 1, 5: 1, 6: 0}\n",
      "DEBUG: 146968 5650 \n",
      "\n",
      " +----------------------+-------------+----------------+-------+\n",
      "| bin_offset_40        | TT_S0       | target_count   | +9    |\n",
      "| (category)           | (float64)   | (int64)        | ...   |\n",
      "|----------------------+-------------+----------------+-------|\n",
      "| (3000018, 3000058]   | 4.0         | 4              | ...   |\n",
      "| (3000058, 3000158]   | 4.0         | 9              | ...   |\n",
      "| (3000158, 3000258]   | 0.0         | 3              | ...   |\n",
      "| (3000258, 3000358]   | 2.0         | 5              | ...   |\n",
      "| ...                  | ...         | ...            | ...   |\n",
      "| (18261358, 18261458] | 4.0         | 10             | ...   |\n",
      "| (18261458, 18261558] | 0.0         | 2              | ...   |\n",
      "| (18261558, 18261658] | 0.0         | 2              | ...   |\n",
      "| (18261658, 18261710] | 4.0         | 3              | ...   |\n",
      "+----------------------+-------------+----------------+-------+\n",
      "Unstranded PyRanges object has 146,968 rows and 12 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "9 hidden columns: starting_nt, Chromosome, Start, End, genes, transcripts, ... (+ 3 more.)\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0\n",
      "BINS  IntervalIndex([  (3000018, 3000063],   (3000063, 3000163],\n",
      "                 (3000163, 3000263],   (3000263, 3000363],\n",
      "                 (3000363, 3000463],   (3000463, 3000563],\n",
      "                 (3000563, 3000663],   (3000663, 3000763],\n",
      "                 (3000763, 3000863],   (3000863, 3000963],\n",
      "               ...\n",
      "               (18260763, 18260863], (18260863, 18260963],\n",
      "               (18260963, 18261063], (18261063, 18261163],\n",
      "               (18261163, 18261263], (18261263, 18261363],\n",
      "               (18261363, 18261463], (18261463, 18261563],\n",
      "               (18261563, 18261663], (18261663, 18261710]],\n",
      "              dtype='interval[int64, right]', length=152618)\n",
      "HEAD with BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0   \n",
      "\n",
      "        bin_offset_45  target_count  \n",
      "0  (3000018, 3000063]             1  \n",
      "1  (3000018, 3000063]             1  \n",
      "2  (3000018, 3000063]             1  \n",
      "3  (3000018, 3000063]             1  \n",
      "4  (3000063, 3000163]             1  \n",
      "TAIL with BINS\n",
      "       seqnames     start       end        CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  18261611  18261611  18_18261611    0.0    0.0    0.0   \n",
      "699996       18  18261615  18261615  18_18261615    0.0    0.0    0.0   \n",
      "699997       18  18261682  18261682  18_18261682    0.0    0.0    0.0   \n",
      "699998       18  18261702  18261702  18_18261702    4.0    0.0    0.0   \n",
      "699999       18  18261710  18261710  18_18261710    0.0    0.0    0.0   \n",
      "\n",
      "               bin_offset_45  target_count  \n",
      "699995  (18261563, 18261663]             1  \n",
      "699996  (18261563, 18261663]             1  \n",
      "699997  (18261663, 18261710]             1  \n",
      "699998  (18261663, 18261710]             1  \n",
      "699999  (18261663, 18261710]             1  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:137: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "         bin_offset_45  TT_S0  target_count starting_nt\n",
      "0   (3000018, 3000063]    4.0             4     3000018\n",
      "1   (3000063, 3000163]    4.0            10     3000063\n",
      "2   (3000163, 3000263]    0.0             2     3000163\n",
      "3   (3000263, 3000363]    2.0             5     3000263\n",
      "4   (3000363, 3000463]    0.0             3     3000363\n",
      "5   (3000463, 3000563]    0.0             1     3000463\n",
      "6   (3000563, 3000663]    0.0             3     3000563\n",
      "7   (3000663, 3000763]    0.0             5     3000663\n",
      "8   (3000763, 3000863]    0.0             5     3000763\n",
      "9   (3000863, 3000963]    0.0             5     3000863\n",
      "10  (3000963, 3001063]    2.0             6     3000963\n",
      "11  (3001063, 3001163]    0.0             2     3001063\n",
      "12  (3001163, 3001263]    0.0             2     3001163\n",
      "13  (3001263, 3001363]    0.0             5     3001263\n",
      "14  (3001363, 3001463]    0.0             3     3001363\n",
      "15  (3001463, 3001563]    1.0             7     3001463\n",
      "16  (3001563, 3001663]    6.0             8     3001563\n",
      "17  (3001663, 3001763]    0.0             7     3001663\n",
      "18  (3001763, 3001863]    0.0             5     3001763\n",
      "19  (3001863, 3001963]    0.0             1     3001863\n",
      "20  (3001963, 3002063]    1.0             2     3001963\n",
      "21  (3002063, 3002163]    0.0             3     3002063\n",
      "22  (3002163, 3002263]    4.0             6     3002163\n",
      "23  (3002263, 3002363]    9.0             4     3002263\n",
      "24  (3002363, 3002463]   19.0             4     3002363\n",
      "25  (3002463, 3002563]    4.0             3     3002463\n",
      "26  (3002563, 3002663]    0.0             1     3002563\n",
      "27  (3002663, 3002763]    0.0             3     3002663\n",
      "28  (3002763, 3002863]    0.0             2     3002763\n",
      "29  (3002863, 3002963]    0.0             1     3002863\n",
      "30  (3002963, 3003063]    3.0             8     3002963\n",
      "31  (3003063, 3003163]    0.0             1     3003063\n",
      "32  (3003163, 3003263]    0.0             0     3003163\n",
      "33  (3003263, 3003363]    1.0             7     3003263\n",
      "34  (3003363, 3003463]    0.0             3     3003363\n",
      "35  (3003463, 3003563]    0.0             3     3003463\n",
      "36  (3003563, 3003663]    1.0             1     3003563\n",
      "37  (3003663, 3003763]    4.0             8     3003663\n",
      "38  (3003763, 3003863]    0.0             2     3003763\n",
      "39  (3003863, 3003963]    6.0             7     3003863\n",
      "40  (3003963, 3004063]    2.0             2     3003963\n",
      "41  (3004063, 3004163]    0.0             0     3004063\n",
      "42  (3004163, 3004263]    1.0             4     3004163\n",
      "43  (3004263, 3004363]    4.0             3     3004263\n",
      "44  (3004363, 3004463]    0.0             1     3004363\n",
      "45  (3004463, 3004563]    0.0             5     3004463\n",
      "46  (3004563, 3004663]    1.0             3     3004563\n",
      "47  (3004663, 3004763]    0.0             0     3004663\n",
      "48  (3004763, 3004863]   10.0             4     3004763\n",
      "49  (3004863, 3004963]    3.0             5     3004863\n",
      "(152618, 4)\n",
      "bin_offset_45    category\n",
      "TT_S0             float64\n",
      "target_count        int64\n",
      "starting_nt      category\n",
      "dtype: object\n",
      "sumos ilgis: 152618\n",
      "ranges ilgis: 152618\n",
      "Model input - NO SCALE\n",
      "        TT_S0  genes  transcripts  exons  CDSs\n",
      "0         4.0      0            0      0     0\n",
      "1         4.0      0            0      0     0\n",
      "2         0.0      0            0      0     0\n",
      "3         2.0      0            0      0     0\n",
      "4         0.0      0            0      0     0\n",
      "...       ...    ...          ...    ...   ...\n",
      "152613   13.0      0            0      0     0\n",
      "152614    4.0      0            0      0     0\n",
      "152615    0.0      0            0      0     0\n",
      "152616    0.0      0            0      0     0\n",
      "152617    4.0      0            0      0     0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "        TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0         4.0             0                   0             0            0\n",
      "1         4.0             0                   0             0            0\n",
      "2         0.0             0                   0             0            0\n",
      "3         2.0             0                   0             0            0\n",
      "4         0.0             0                   0             0            0\n",
      "...       ...           ...                 ...           ...          ...\n",
      "152613   13.0             0                   0             0            0\n",
      "152614    4.0             0                   0             0            0\n",
      "152615    0.0             0                   0             0            0\n",
      "152616    0.0             0                   0             0            0\n",
      "152617    4.0             0                   0             0            0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "0.19128805252329345\n",
      "[6 6 9 1 9 9 9 9 9 9 1 9 9 9 9 1 6 9 9 9 1 9 6 2 0 6 9 9 9 9 6 9 9 1 9 9 1\n",
      " 6 9 6 1 9 1 6 9 9 1 9 2 6 9 4 6 4 9 4 8 3 3 7 1 2 0 1 6 9 9 2 9 1 2 9 6 1\n",
      " 9 9 1 9 9 1 1 2 1 6 9 6 1 2 9 9 6 5 5 9 1 9 6 6 6 6 0 5 4 4 6 6 2 4 6 2 1\n",
      " 6 5 6 9 1 1 9 9 9 9 6 6 1 1 5 1 0 5 2 5 1 5 1 6 6 1 2 1 9 9 9 9 1 9 1 9 0\n",
      " 1 2 6 1 9 1 6 5 1 6 1 2 2 6 2 1 2 5 1 1 9 9 9 1 9 9 1 9 9 9 1 9 2 6 9 6 9\n",
      " 9 6 9 9 9 9 9 1 9 9 9 9 9 9 9 9 9 9 9 9 9 5 5 2 1 9 9 1 9 9 9 9 1 1 9 9 9\n",
      " 9 9 9 9 9 9 9 9 9 9 9 9 9 6 9 9 9 1 9 9 9 9 9 9 9 9 9 9 9 1 9 9 9 9 1 9 1\n",
      " 9 9 9 9 9 9 1 9 9 6 9 1 9 6 1 1 9 9 1 9 1 1 9 9 1 9 9 9 9 9 9 9 1 9 9 9 9\n",
      " 6 9 9 1 1 1 1 1 9 9 9 9 9 6 6 6 6 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9\n",
      " 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9\n",
      " 9 9 9 9 9 9 9 9 9 9 9 6 1 9 6 2 1 1 9 1 9 9 9 9 2 5 6 5 9 0 2 6 1 1 6 2 9\n",
      " 6 6 1 1 9 1 5 5 6 6 2 9 9 6 9 9 1 9 6 1 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9\n",
      " 6 9 1 9 6 9 9 9 9 1 1 9 1 9 9 1 1 6 2 1 9 1 6 1 2 9 9 6 2 6 2 6 1 9 6 1 9\n",
      " 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9\n",
      " 2 5 9 9 9 6 9 9 1 2 6 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9\n",
      " 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9\n",
      " 9 9 9 6 9 1 9 6 9 9 9 9 9 1 9 9 9 9 1 9 1 1 1 9 6 1 9 9 9 9 2 1 9 9 9 9 6\n",
      " 9 9 6 9 9 9 9 1 9 5 9 9 1 2 6 9 9 9 9 1 9 6 6 9 9 9 9 1 9 2 9 1 6 6 9 1 6\n",
      " 6 6 9 9 9 1 1 9 9 9 9 9 9 9 9 9 9 6 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9\n",
      " 9 9 9 9 9 9 9 9 9 9 9 9 9 6 6 2 6 9 6 9 1 1 9 9 1 9 9 9 6 9 9 9 9 9 1 9 9\n",
      " 9 1 9 9 9 1 9 1 9 9 9 9 9 9 1 9 9 9 9 9 1 9 9 9 9 9 6 9 9 9 9 1 6 9 9 9 9\n",
      " 9 9 9 9 9 9 9 1 9 9 9 9 9 1 6 9 9 9 1 1 9 9 6 9 9 9 9 2 1 6 2 9 6 1 6 9 9\n",
      " 9 9 9 9 1 9 9 9 9 9 9 9 9 9 9 9 9 1 9 1 9 1 1 9 6 9 1 9 9 1 1 1 9 9 9 1 9\n",
      " 9 9 9 9 9 9 9 9 9 9 1 9 9 9 1 9 1 9 2 1 9 9 9 9 1 1 6 9 9 9 9 9 9 9 9 9 9\n",
      " 9 9 6 9 9 9 9 1 9 9 9 9 9 9 9 9 9 9 9 9 1 1 9 1 1 9 6 6 9 9 9 9 1 9 9 1 6\n",
      " 6 6 2 1 1 1 9 9 9 9 9 9 9 9 1 9 9 9 9 6 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9\n",
      " 9 9 9 9 9 9 9 9 6 9 9 9 9 9 9 1 9 9 9 9 9 9 9 9 1 9 9 6 9 2 1 9 9 1 9 9 9\n",
      " 9]\n",
      "threshold density 1.95819\n",
      "cluster signal 123595.0\n",
      "cluster target count 163021\n",
      "!!cluster density 0.7581538574784844\n",
      "cluster signal 0.0\n",
      "cluster target count 227737\n",
      "!!cluster density 0.0\n",
      "cluster signal 41998.0\n",
      "cluster target count 135034\n",
      "!!cluster density 0.3110179658456389\n",
      "cluster signal 99657.0\n",
      "cluster target count 80663\n",
      "!!cluster density 1.2354735132588672\n",
      "cluster signal 60518.0\n",
      "cluster target count 26842\n",
      "!!cluster density 2.254600998435288\n",
      "cluster signal 28393.0\n",
      "cluster target count 9715\n",
      "!!cluster density 2.9225939269171386\n",
      "cluster signal 9535.0\n",
      "cluster target count 2336\n",
      "!!cluster density 4.081763698630137\n",
      "cluster signal 520.0\n",
      "cluster target count 32\n",
      "!!cluster density 16.25\n",
      "cluster signal 490.0\n",
      "cluster target count 5\n",
      "!!cluster density 98.0\n",
      "cluster signal 92205.0\n",
      "cluster target count 54615\n",
      "!!cluster density 1.6882724526229058\n",
      "\n",
      "Labels mapping\n",
      " {6: 0, 9: 0, 1: 0, 2: 0, 0: 1, 4: 1, 8: 1, 3: 1, 7: 1, 5: 0}\n",
      "DEBUG: 148588 4030 \n",
      "\n",
      " +----------------------+-------------+----------------+-------+\n",
      "| bin_offset_45        | TT_S0       | target_count   | +9    |\n",
      "| (category)           | (float64)   | (int64)        | ...   |\n",
      "|----------------------+-------------+----------------+-------|\n",
      "| (3000018, 3000063]   | 4.0         | 4              | ...   |\n",
      "| (3000063, 3000163]   | 4.0         | 10             | ...   |\n",
      "| (3000163, 3000263]   | 0.0         | 2              | ...   |\n",
      "| (3000263, 3000363]   | 2.0         | 5              | ...   |\n",
      "| ...                  | ...         | ...            | ...   |\n",
      "| (18261363, 18261463] | 4.0         | 9              | ...   |\n",
      "| (18261463, 18261563] | 0.0         | 2              | ...   |\n",
      "| (18261563, 18261663] | 0.0         | 2              | ...   |\n",
      "| (18261663, 18261710] | 4.0         | 3              | ...   |\n",
      "+----------------------+-------------+----------------+-------+\n",
      "Unstranded PyRanges object has 148,588 rows and 12 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "9 hidden columns: starting_nt, Chromosome, Start, End, genes, transcripts, ... (+ 3 more.)\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0\n",
      "BINS  IntervalIndex([  (3000018, 3000068],   (3000068, 3000168],\n",
      "                 (3000168, 3000268],   (3000268, 3000368],\n",
      "                 (3000368, 3000468],   (3000468, 3000568],\n",
      "                 (3000568, 3000668],   (3000668, 3000768],\n",
      "                 (3000768, 3000868],   (3000868, 3000968],\n",
      "               ...\n",
      "               (18260768, 18260868], (18260868, 18260968],\n",
      "               (18260968, 18261068], (18261068, 18261168],\n",
      "               (18261168, 18261268], (18261268, 18261368],\n",
      "               (18261368, 18261468], (18261468, 18261568],\n",
      "               (18261568, 18261668], (18261668, 18261710]],\n",
      "              dtype='interval[int64, right]', length=152618)\n",
      "HEAD with BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0   \n",
      "\n",
      "        bin_offset_50  target_count  \n",
      "0  (3000018, 3000068]             1  \n",
      "1  (3000018, 3000068]             1  \n",
      "2  (3000018, 3000068]             1  \n",
      "3  (3000018, 3000068]             1  \n",
      "4  (3000018, 3000068]             1  \n",
      "TAIL with BINS\n",
      "       seqnames     start       end        CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  18261611  18261611  18_18261611    0.0    0.0    0.0   \n",
      "699996       18  18261615  18261615  18_18261615    0.0    0.0    0.0   \n",
      "699997       18  18261682  18261682  18_18261682    0.0    0.0    0.0   \n",
      "699998       18  18261702  18261702  18_18261702    4.0    0.0    0.0   \n",
      "699999       18  18261710  18261710  18_18261710    0.0    0.0    0.0   \n",
      "\n",
      "               bin_offset_50  target_count  \n",
      "699995  (18261568, 18261668]             1  \n",
      "699996  (18261568, 18261668]             1  \n",
      "699997  (18261668, 18261710]             1  \n",
      "699998  (18261668, 18261710]             1  \n",
      "699999  (18261668, 18261710]             1  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:137: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "         bin_offset_50  TT_S0  target_count starting_nt\n",
      "0   (3000018, 3000068]    4.0             5     3000018\n",
      "1   (3000068, 3000168]    4.0             9     3000068\n",
      "2   (3000168, 3000268]    0.0             2     3000168\n",
      "3   (3000268, 3000368]    2.0             5     3000268\n",
      "4   (3000368, 3000468]    0.0             3     3000368\n",
      "5   (3000468, 3000568]    0.0             1     3000468\n",
      "6   (3000568, 3000668]    0.0             3     3000568\n",
      "7   (3000668, 3000768]    0.0             5     3000668\n",
      "8   (3000768, 3000868]    0.0             5     3000768\n",
      "9   (3000868, 3000968]    0.0             5     3000868\n",
      "10  (3000968, 3001068]    2.0             6     3000968\n",
      "11  (3001068, 3001168]    0.0             2     3001068\n",
      "12  (3001168, 3001268]    0.0             2     3001168\n",
      "13  (3001268, 3001368]    0.0             5     3001268\n",
      "14  (3001368, 3001468]    0.0             3     3001368\n",
      "15  (3001468, 3001568]    4.0             9     3001468\n",
      "16  (3001568, 3001668]    3.0             6     3001568\n",
      "17  (3001668, 3001768]    0.0             7     3001668\n",
      "18  (3001768, 3001868]    0.0             5     3001768\n",
      "19  (3001868, 3001968]    0.0             1     3001868\n",
      "20  (3001968, 3002068]    1.0             2     3001968\n",
      "21  (3002068, 3002168]    0.0             3     3002068\n",
      "22  (3002168, 3002268]    4.0             6     3002168\n",
      "23  (3002268, 3002368]    9.0             4     3002268\n",
      "24  (3002368, 3002468]   19.0             4     3002368\n",
      "25  (3002468, 3002568]    4.0             3     3002468\n",
      "26  (3002568, 3002668]    0.0             1     3002568\n",
      "27  (3002668, 3002768]    0.0             3     3002668\n",
      "28  (3002768, 3002868]    0.0             2     3002768\n",
      "29  (3002868, 3002968]    0.0             1     3002868\n",
      "30  (3002968, 3003068]    3.0             9     3002968\n",
      "31  (3003068, 3003168]    0.0             0     3003068\n",
      "32  (3003168, 3003268]    0.0             0     3003168\n",
      "33  (3003268, 3003368]    1.0             7     3003268\n",
      "34  (3003368, 3003468]    0.0             3     3003368\n",
      "35  (3003468, 3003568]    0.0             3     3003468\n",
      "36  (3003568, 3003668]    1.0             1     3003568\n",
      "37  (3003668, 3003768]    4.0             8     3003668\n",
      "38  (3003768, 3003868]    0.0             2     3003768\n",
      "39  (3003868, 3003968]    6.0             7     3003868\n",
      "40  (3003968, 3004068]    2.0             2     3003968\n",
      "41  (3004068, 3004168]    0.0             0     3004068\n",
      "42  (3004168, 3004268]    3.0             5     3004168\n",
      "43  (3004268, 3004368]    2.0             3     3004268\n",
      "44  (3004368, 3004468]    0.0             0     3004368\n",
      "45  (3004468, 3004568]    0.0             6     3004468\n",
      "46  (3004568, 3004668]    1.0             2     3004568\n",
      "47  (3004668, 3004768]    0.0             0     3004668\n",
      "48  (3004768, 3004868]   10.0             5     3004768\n",
      "49  (3004868, 3004968]    3.0             4     3004868\n",
      "(152618, 4)\n",
      "bin_offset_50    category\n",
      "TT_S0             float64\n",
      "target_count        int64\n",
      "starting_nt      category\n",
      "dtype: object\n",
      "sumos ilgis: 152618\n",
      "ranges ilgis: 152618\n",
      "Model input - NO SCALE\n",
      "        TT_S0  genes  transcripts  exons  CDSs\n",
      "0         4.0      0            0      0     0\n",
      "1         4.0      0            0      0     0\n",
      "2         0.0      0            0      0     0\n",
      "3         2.0      0            0      0     0\n",
      "4         0.0      0            0      0     0\n",
      "...       ...    ...          ...    ...   ...\n",
      "152613   13.0      0            0      0     0\n",
      "152614    4.0      0            0      0     0\n",
      "152615    0.0      0            0      0     0\n",
      "152616    0.0      0            0      0     0\n",
      "152617    4.0      0            0      0     0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "        TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0         4.0             0                   0             0            0\n",
      "1         4.0             0                   0             0            0\n",
      "2         0.0             0                   0             0            0\n",
      "3         2.0             0                   0             0            0\n",
      "4         0.0             0                   0             0            0\n",
      "...       ...           ...                 ...           ...          ...\n",
      "152613   13.0             0                   0             0            0\n",
      "152614    4.0             0                   0             0            0\n",
      "152615    0.0             0                   0             0            0\n",
      "152616    0.0             0                   0             0            0\n",
      "152617    4.0             0                   0             0            0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "0.019702787351426436\n",
      "[0 0 6 2 6 6 6 6 6 6 2 6 6 6 6 0 0 6 6 6 2 6 0 5 1 0 6 6 6 6 0 6 6 2 6 6 2\n",
      " 0 6 9 2 6 0 2 6 6 2 6 5 0 6 3 7 5 6 7 3 3 3 4 2 9 1 0 6 6 6 9 6 2 9 6 0 2\n",
      " 6 6 2 6 6 2 2 9 2 0 0 2 2 5 6 6 0 8 5 6 2 2 2 0 9 9 8 8 7 1 0 9 5 7 0 5 2\n",
      " 0 5 0 6 2 2 6 6 6 6 0 0 2 2 8 2 1 8 5 8 2 8 2 0 0 2 5 2 6 6 6 6 2 6 2 6 1\n",
      " 2 9 0 2 6 2 9 5 2 0 2 8 0 0 5 2 9 8 2 2 6 6 6 2 6 6 2 6 6 6 2 6 5 0 6 0 6\n",
      " 6 0 6 6 6 6 6 2 6 6 6 6 6 6 6 6 6 6 6 6 6 5 8 9 2 6 6 2 6 6 6 6 2 2 6 6 6\n",
      " 6 6 6 6 6 6 6 6 6 6 6 6 6 0 6 6 6 2 6 6 6 6 6 6 6 6 6 6 6 2 6 6 6 6 2 6 2\n",
      " 6 6 6 6 6 2 6 6 6 0 2 6 2 2 2 2 6 6 2 6 2 2 6 6 2 6 6 6 6 6 6 6 2 6 6 6 6\n",
      " 0 6 6 2 2 0 6 2 6 6 6 6 6 0 0 9 0 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6\n",
      " 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6\n",
      " 6 6 6 6 6 6 6 6 6 6 6 0 2 6 0 5 2 2 6 2 6 6 6 6 5 8 0 8 6 1 9 0 2 2 0 5 6\n",
      " 0 0 0 2 6 2 8 1 0 0 5 6 6 0 6 6 2 6 0 2 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6\n",
      " 0 6 2 6 0 6 6 6 6 2 2 6 2 6 2 2 2 0 9 2 6 2 0 2 5 6 6 0 9 9 9 0 2 6 0 2 6\n",
      " 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6\n",
      " 9 8 6 6 6 0 6 6 2 9 0 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6\n",
      " 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6\n",
      " 6 6 6 0 6 2 6 0 6 6 6 6 6 2 6 6 6 6 2 6 2 2 2 6 0 2 6 6 6 6 9 2 6 6 6 6 9\n",
      " 6 6 0 6 6 6 6 2 6 8 6 6 2 5 0 6 6 6 6 2 9 6 0 6 6 6 6 2 6 5 6 2 0 0 6 0 0\n",
      " 9 2 6 6 6 2 2 6 6 6 6 6 6 6 6 6 6 0 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6\n",
      " 6 6 6 6 6 6 6 6 6 6 6 6 6 0 0 5 0 0 6 6 2 2 6 6 2 6 6 6 0 6 6 6 6 6 2 6 6\n",
      " 6 2 6 6 6 2 6 2 6 6 6 6 6 6 2 6 6 6 6 6 2 6 6 6 6 6 0 6 6 6 6 2 0 6 6 6 6\n",
      " 6 6 6 6 6 6 6 2 6 6 6 6 6 0 2 6 6 6 2 2 6 6 0 6 6 6 6 5 2 5 0 6 0 2 0 6 6\n",
      " 6 6 6 6 2 6 6 6 6 6 6 6 6 6 6 6 6 2 6 2 2 6 2 6 0 6 2 6 2 6 2 2 6 6 6 2 6\n",
      " 6 6 6 6 6 6 6 6 6 6 2 6 6 6 2 6 2 6 9 2 6 6 6 6 2 2 0 6 6 6 6 6 6 6 6 6 6\n",
      " 6 6 0 6 6 6 6 2 6 6 6 6 6 6 6 6 6 6 6 6 2 2 6 2 2 6 0 0 6 6 6 6 2 6 6 2 0\n",
      " 0 0 5 2 2 2 6 6 6 6 6 6 6 6 2 6 6 6 6 0 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6\n",
      " 6 6 6 6 6 6 6 6 0 6 6 6 6 6 6 2 6 6 6 6 6 6 6 6 2 6 6 0 6 9 2 6 6 2 6 6 6\n",
      " 6]\n",
      "threshold density 1.95819\n",
      "cluster signal 93106.0\n",
      "cluster target count 132140\n",
      "!!cluster density 0.7046011805660662\n",
      "cluster signal 0.0\n",
      "cluster target count 227660\n",
      "!!cluster density 0.0\n",
      "cluster signal 41843.0\n",
      "cluster target count 134897\n",
      "!!cluster density 0.31018480766807266\n",
      "cluster signal 65440.0\n",
      "cluster target count 46758\n",
      "!!cluster density 1.3995466016510543\n",
      "cluster signal 60647.0\n",
      "cluster target count 26898\n",
      "!!cluster density 2.254702951892334\n",
      "cluster signal 84762.0\n",
      "cluster target count 77923\n",
      "!!cluster density 1.0877661281008175\n",
      "cluster signal 9728.0\n",
      "cluster target count 2205\n",
      "!!cluster density 4.4117913832199545\n",
      "cluster signal 28406.0\n",
      "cluster target count 9728\n",
      "!!cluster density 2.9200246710526314\n",
      "cluster signal 490.0\n",
      "cluster target count 5\n",
      "!!cluster density 98.0\n",
      "cluster signal 72489.0\n",
      "cluster target count 41786\n",
      "!!cluster density 1.7347676255205093\n",
      "\n",
      "Labels mapping\n",
      " {0: 0, 6: 0, 2: 0, 5: 0, 1: 1, 9: 0, 3: 1, 7: 1, 4: 1, 8: 0}\n",
      "DEBUG: 148595 4023 \n",
      "\n",
      " +----------------------+-------------+----------------+-------+\n",
      "| bin_offset_50        | TT_S0       | target_count   | +9    |\n",
      "| (category)           | (float64)   | (int64)        | ...   |\n",
      "|----------------------+-------------+----------------+-------|\n",
      "| (3000018, 3000068]   | 4.0         | 5              | ...   |\n",
      "| (3000068, 3000168]   | 4.0         | 9              | ...   |\n",
      "| (3000168, 3000268]   | 0.0         | 2              | ...   |\n",
      "| (3000268, 3000368]   | 2.0         | 5              | ...   |\n",
      "| ...                  | ...         | ...            | ...   |\n",
      "| (18261368, 18261468] | 4.0         | 8              | ...   |\n",
      "| (18261468, 18261568] | 0.0         | 2              | ...   |\n",
      "| (18261568, 18261668] | 0.0         | 2              | ...   |\n",
      "| (18261668, 18261710] | 4.0         | 3              | ...   |\n",
      "+----------------------+-------------+----------------+-------+\n",
      "Unstranded PyRanges object has 148,595 rows and 12 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "9 hidden columns: starting_nt, Chromosome, Start, End, genes, transcripts, ... (+ 3 more.)\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0\n",
      "BINS  IntervalIndex([  (3000018, 3000073],   (3000073, 3000173],\n",
      "                 (3000173, 3000273],   (3000273, 3000373],\n",
      "                 (3000373, 3000473],   (3000473, 3000573],\n",
      "                 (3000573, 3000673],   (3000673, 3000773],\n",
      "                 (3000773, 3000873],   (3000873, 3000973],\n",
      "               ...\n",
      "               (18260773, 18260873], (18260873, 18260973],\n",
      "               (18260973, 18261073], (18261073, 18261173],\n",
      "               (18261173, 18261273], (18261273, 18261373],\n",
      "               (18261373, 18261473], (18261473, 18261573],\n",
      "               (18261573, 18261673], (18261673, 18261710]],\n",
      "              dtype='interval[int64, right]', length=152618)\n",
      "HEAD with BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0   \n",
      "\n",
      "        bin_offset_55  target_count  \n",
      "0  (3000018, 3000073]             1  \n",
      "1  (3000018, 3000073]             1  \n",
      "2  (3000018, 3000073]             1  \n",
      "3  (3000018, 3000073]             1  \n",
      "4  (3000018, 3000073]             1  \n",
      "TAIL with BINS\n",
      "       seqnames     start       end        CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  18261611  18261611  18_18261611    0.0    0.0    0.0   \n",
      "699996       18  18261615  18261615  18_18261615    0.0    0.0    0.0   \n",
      "699997       18  18261682  18261682  18_18261682    0.0    0.0    0.0   \n",
      "699998       18  18261702  18261702  18_18261702    4.0    0.0    0.0   \n",
      "699999       18  18261710  18261710  18_18261710    0.0    0.0    0.0   \n",
      "\n",
      "               bin_offset_55  target_count  \n",
      "699995  (18261573, 18261673]             1  \n",
      "699996  (18261573, 18261673]             1  \n",
      "699997  (18261673, 18261710]             1  \n",
      "699998  (18261673, 18261710]             1  \n",
      "699999  (18261673, 18261710]             1  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:137: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "         bin_offset_55  TT_S0  target_count starting_nt\n",
      "0   (3000018, 3000073]    4.0             5     3000018\n",
      "1   (3000073, 3000173]    4.0            10     3000073\n",
      "2   (3000173, 3000273]    0.0             1     3000173\n",
      "3   (3000273, 3000373]    2.0             5     3000273\n",
      "4   (3000373, 3000473]    0.0             3     3000373\n",
      "5   (3000473, 3000573]    0.0             1     3000473\n",
      "6   (3000573, 3000673]    0.0             4     3000573\n",
      "7   (3000673, 3000773]    0.0             4     3000673\n",
      "8   (3000773, 3000873]    0.0             6     3000773\n",
      "9   (3000873, 3000973]    0.0             4     3000873\n",
      "10  (3000973, 3001073]    2.0             6     3000973\n",
      "11  (3001073, 3001173]    0.0             2     3001073\n",
      "12  (3001173, 3001273]    0.0             2     3001173\n",
      "13  (3001273, 3001373]    0.0             5     3001273\n",
      "14  (3001373, 3001473]    0.0             3     3001373\n",
      "15  (3001473, 3001573]    4.0             9     3001473\n",
      "16  (3001573, 3001673]    3.0             6     3001573\n",
      "17  (3001673, 3001773]    0.0             7     3001673\n",
      "18  (3001773, 3001873]    0.0             5     3001773\n",
      "19  (3001873, 3001973]    0.0             1     3001873\n",
      "20  (3001973, 3002073]    1.0             2     3001973\n",
      "21  (3002073, 3002173]    0.0             3     3002073\n",
      "22  (3002173, 3002273]    4.0             7     3002173\n",
      "23  (3002273, 3002373]    9.0             3     3002273\n",
      "24  (3002373, 3002473]   19.0             4     3002373\n",
      "25  (3002473, 3002573]    4.0             3     3002473\n",
      "26  (3002573, 3002673]    0.0             1     3002573\n",
      "27  (3002673, 3002773]    0.0             3     3002673\n",
      "28  (3002773, 3002873]    0.0             2     3002773\n",
      "29  (3002873, 3002973]    1.0             2     3002873\n",
      "30  (3002973, 3003073]    2.0             8     3002973\n",
      "31  (3003073, 3003173]    0.0             0     3003073\n",
      "32  (3003173, 3003273]    0.0             0     3003173\n",
      "33  (3003273, 3003373]    1.0             8     3003273\n",
      "34  (3003373, 3003473]    0.0             2     3003373\n",
      "35  (3003473, 3003573]    0.0             3     3003473\n",
      "36  (3003573, 3003673]    1.0             2     3003573\n",
      "37  (3003673, 3003773]    4.0             8     3003673\n",
      "38  (3003773, 3003873]    0.0             1     3003773\n",
      "39  (3003873, 3003973]    7.0             8     3003873\n",
      "40  (3003973, 3004073]    1.0             1     3003973\n",
      "41  (3004073, 3004173]    0.0             0     3004073\n",
      "42  (3004173, 3004273]    3.0             5     3004173\n",
      "43  (3004273, 3004373]    2.0             3     3004273\n",
      "44  (3004373, 3004473]    0.0             0     3004373\n",
      "45  (3004473, 3004573]    0.0             6     3004473\n",
      "46  (3004573, 3004673]    1.0             2     3004573\n",
      "47  (3004673, 3004773]    0.0             0     3004673\n",
      "48  (3004773, 3004873]   10.0             6     3004773\n",
      "49  (3004873, 3004973]    3.0             3     3004873\n",
      "(152618, 4)\n",
      "bin_offset_55    category\n",
      "TT_S0             float64\n",
      "target_count        int64\n",
      "starting_nt      category\n",
      "dtype: object\n",
      "sumos ilgis: 152618\n",
      "ranges ilgis: 152618\n",
      "Model input - NO SCALE\n",
      "        TT_S0  genes  transcripts  exons  CDSs\n",
      "0         4.0      0            0      0     0\n",
      "1         4.0      0            0      0     0\n",
      "2         0.0      0            0      0     0\n",
      "3         2.0      0            0      0     0\n",
      "4         0.0      0            0      0     0\n",
      "...       ...    ...          ...    ...   ...\n",
      "152613   13.0      0            0      0     0\n",
      "152614    4.0      0            0      0     0\n",
      "152615    0.0      0            0      0     0\n",
      "152616    0.0      0            0      0     0\n",
      "152617    4.0      0            0      0     0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "        TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0         4.0             0                   0             0            0\n",
      "1         4.0             0                   0             0            0\n",
      "2         0.0             0                   0             0            0\n",
      "3         2.0             0                   0             0            0\n",
      "4         0.0             0                   0             0            0\n",
      "...       ...           ...                 ...           ...          ...\n",
      "152613   13.0             0                   0             0            0\n",
      "152614    4.0             0                   0             0            0\n",
      "152615    0.0             0                   0             0            0\n",
      "152616    0.0             0                   0             0            0\n",
      "152617    4.0             0                   0             0            0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "0.10128556264660787\n",
      "[3 3 0 7 0 0 0 0 0 0 7 0 0 0 0 3 7 0 0 0 0 0 3 1 4 3 0 0 0 0 7 0 0 0 0 0 0\n",
      " 3 0 1 0 0 7 7 0 0 0 0 1 7 0 6 2 1 0 2 6 9 9 5 7 1 8 3 0 0 0 1 0 0 1 0 3 0\n",
      " 0 0 7 0 0 7 1 0 7 7 7 0 7 1 0 0 3 1 1 0 7 7 0 7 3 3 4 8 8 8 3 3 1 2 3 1 0\n",
      " 1 3 7 0 0 0 0 0 0 0 3 7 0 0 4 3 8 4 1 1 0 4 0 3 3 0 1 7 0 0 0 0 7 0 0 0 4\n",
      " 0 1 3 7 0 3 7 1 7 7 7 4 3 3 1 0 1 4 7 0 0 0 0 0 0 0 7 0 0 0 0 0 1 3 0 7 0\n",
      " 0 7 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 4 1 0 0 0 7 0 0 0 0 7 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 7 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 7 0 0\n",
      " 0 0 0 0 0 7 0 0 0 7 7 0 7 7 7 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 7\n",
      " 7 0 0 0 7 7 0 7 0 0 0 0 3 0 7 3 3 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 3 0 7 0 7 1 0 7 0 0 0 0 0 0 1 4 3 1 0 8 1 3 7 0 3 1 0\n",
      " 7 7 7 0 0 0 4 4 7 3 1 0 0 7 0 0 7 0 3 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 7\n",
      " 0 0 0 0 3 0 0 0 0 7 0 0 0 0 0 0 0 7 1 7 0 0 7 7 1 0 0 3 1 3 1 7 7 0 7 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 1 1 0 0 0 3 0 0 0 1 7 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 7 0 7 0 3 0 0 0 0 0 0 0 0 0 0 0 0 0 7 7 0 3 7 0 0 0 0 1 0 0 0 0 0 3\n",
      " 0 0 3 0 0 0 0 0 0 1 0 0 0 1 7 0 0 0 0 0 3 0 7 0 0 0 0 0 3 3 0 0 3 3 0 3 7\n",
      " 3 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 3 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 7 7 1 3 7 0 0 7 0 0 0 0 0 0 0 7 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 7 0 0 0 0 7 7 0 0 0 0\n",
      " 0 0 0 0 0 0 0 7 0 0 0 0 0 3 0 0 0 0 0 7 0 0 3 0 0 0 0 1 0 1 3 0 7 7 7 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 3 0 0 0 7 0 0 7 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 7 0 0 0 0 0 7 0 1 0 0 0 0 0 0 3 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 7 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 7 0 0 0 0 3 3 0 0 0 0 0 0 0 0 3\n",
      " 7 3 1 0 7 7 0 0 0 0 0 0 0 0 7 0 0 0 0 7 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 3 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 7 0 3 0 0 0 0 0 0 0\n",
      " 0]\n",
      "threshold density 1.95819\n",
      "cluster signal 93243.0\n",
      "cluster target count 111116\n",
      "!!cluster density 0.8391500773965945\n",
      "cluster signal 16392.0\n",
      "cluster target count 300918\n",
      "!!cluster density 0.05447331166630112\n",
      "cluster signal 55504.0\n",
      "cluster target count 113274\n",
      "!!cluster density 0.48999770468068576\n",
      "cluster signal 137967.0\n",
      "cluster target count 105469\n",
      "!!cluster density 1.3081284548066257\n",
      "cluster signal 80407.0\n",
      "cluster target count 42659\n",
      "!!cluster density 1.8848777514709674\n",
      "cluster signal 9763.0\n",
      "cluster target count 2324\n",
      "!!cluster density 4.200946643717728\n",
      "cluster signal 22372.0\n",
      "cluster target count 7542\n",
      "!!cluster density 2.966321930522408\n",
      "cluster signal 508.0\n",
      "cluster target count 31\n",
      "!!cluster density 16.387096774193548\n",
      "cluster signal 490.0\n",
      "cluster target count 6\n",
      "!!cluster density 81.66666666666667\n",
      "cluster signal 40265.0\n",
      "cluster target count 16661\n",
      "!!cluster density 2.4167216853730267\n",
      "\n",
      "Labels mapping\n",
      " {3: 0, 0: 0, 7: 0, 1: 0, 4: 0, 6: 1, 2: 1, 9: 1, 5: 1, 8: 1}\n",
      "DEBUG: 150046 2572 \n",
      "\n",
      " +----------------------+-------------+----------------+-------+\n",
      "| bin_offset_55        | TT_S0       | target_count   | +9    |\n",
      "| (category)           | (float64)   | (int64)        | ...   |\n",
      "|----------------------+-------------+----------------+-------|\n",
      "| (3000018, 3000073]   | 4.0         | 5              | ...   |\n",
      "| (3000073, 3000173]   | 4.0         | 10             | ...   |\n",
      "| (3000173, 3000273]   | 0.0         | 1              | ...   |\n",
      "| (3000273, 3000373]   | 2.0         | 5              | ...   |\n",
      "| ...                  | ...         | ...            | ...   |\n",
      "| (18261373, 18261473] | 4.0         | 7              | ...   |\n",
      "| (18261473, 18261573] | 0.0         | 2              | ...   |\n",
      "| (18261573, 18261673] | 0.0         | 2              | ...   |\n",
      "| (18261673, 18261710] | 4.0         | 3              | ...   |\n",
      "+----------------------+-------------+----------------+-------+\n",
      "Unstranded PyRanges object has 150,046 rows and 12 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "9 hidden columns: starting_nt, Chromosome, Start, End, genes, transcripts, ... (+ 3 more.)\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0\n",
      "BINS  IntervalIndex([  (3000018, 3000078],   (3000078, 3000178],\n",
      "                 (3000178, 3000278],   (3000278, 3000378],\n",
      "                 (3000378, 3000478],   (3000478, 3000578],\n",
      "                 (3000578, 3000678],   (3000678, 3000778],\n",
      "                 (3000778, 3000878],   (3000878, 3000978],\n",
      "               ...\n",
      "               (18260778, 18260878], (18260878, 18260978],\n",
      "               (18260978, 18261078], (18261078, 18261178],\n",
      "               (18261178, 18261278], (18261278, 18261378],\n",
      "               (18261378, 18261478], (18261478, 18261578],\n",
      "               (18261578, 18261678], (18261678, 18261710]],\n",
      "              dtype='interval[int64, right]', length=152618)\n",
      "HEAD with BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0   \n",
      "\n",
      "        bin_offset_60  target_count  \n",
      "0  (3000018, 3000078]             1  \n",
      "1  (3000018, 3000078]             1  \n",
      "2  (3000018, 3000078]             1  \n",
      "3  (3000018, 3000078]             1  \n",
      "4  (3000018, 3000078]             1  \n",
      "TAIL with BINS\n",
      "       seqnames     start       end        CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  18261611  18261611  18_18261611    0.0    0.0    0.0   \n",
      "699996       18  18261615  18261615  18_18261615    0.0    0.0    0.0   \n",
      "699997       18  18261682  18261682  18_18261682    0.0    0.0    0.0   \n",
      "699998       18  18261702  18261702  18_18261702    4.0    0.0    0.0   \n",
      "699999       18  18261710  18261710  18_18261710    0.0    0.0    0.0   \n",
      "\n",
      "               bin_offset_60  target_count  \n",
      "699995  (18261578, 18261678]             1  \n",
      "699996  (18261578, 18261678]             1  \n",
      "699997  (18261678, 18261710]             1  \n",
      "699998  (18261678, 18261710]             1  \n",
      "699999  (18261678, 18261710]             1  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:137: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "         bin_offset_60  TT_S0  target_count starting_nt\n",
      "0   (3000018, 3000078]    4.0             5     3000018\n",
      "1   (3000078, 3000178]    4.0            10     3000078\n",
      "2   (3000178, 3000278]    0.0             1     3000178\n",
      "3   (3000278, 3000378]    2.0             5     3000278\n",
      "4   (3000378, 3000478]    0.0             3     3000378\n",
      "5   (3000478, 3000578]    0.0             1     3000478\n",
      "6   (3000578, 3000678]    0.0             4     3000578\n",
      "7   (3000678, 3000778]    0.0             4     3000678\n",
      "8   (3000778, 3000878]    0.0             6     3000778\n",
      "9   (3000878, 3000978]    0.0             4     3000878\n",
      "10  (3000978, 3001078]    2.0             6     3000978\n",
      "11  (3001078, 3001178]    0.0             2     3001078\n",
      "12  (3001178, 3001278]    0.0             2     3001178\n",
      "13  (3001278, 3001378]    0.0             5     3001278\n",
      "14  (3001378, 3001478]    0.0             3     3001378\n",
      "15  (3001478, 3001578]    4.0             9     3001478\n",
      "16  (3001578, 3001678]    3.0             6     3001578\n",
      "17  (3001678, 3001778]    0.0             7     3001678\n",
      "18  (3001778, 3001878]    0.0             5     3001778\n",
      "19  (3001878, 3001978]    0.0             1     3001878\n",
      "20  (3001978, 3002078]    1.0             4     3001978\n",
      "21  (3002078, 3002178]    0.0             1     3002078\n",
      "22  (3002178, 3002278]    4.0             7     3002178\n",
      "23  (3002278, 3002378]    9.0             3     3002278\n",
      "24  (3002378, 3002478]   19.0             4     3002378\n",
      "25  (3002478, 3002578]    4.0             3     3002478\n",
      "26  (3002578, 3002678]    0.0             1     3002578\n",
      "27  (3002678, 3002778]    0.0             3     3002678\n",
      "28  (3002778, 3002878]    0.0             2     3002778\n",
      "29  (3002878, 3002978]    1.0             2     3002878\n",
      "30  (3002978, 3003078]    2.0             8     3002978\n",
      "31  (3003078, 3003178]    0.0             0     3003078\n",
      "32  (3003178, 3003278]    0.0             0     3003178\n",
      "33  (3003278, 3003378]    1.0             9     3003278\n",
      "34  (3003378, 3003478]    0.0             1     3003378\n",
      "35  (3003478, 3003578]    0.0             3     3003478\n",
      "36  (3003578, 3003678]    1.0             2     3003578\n",
      "37  (3003678, 3003778]    4.0             8     3003678\n",
      "38  (3003778, 3003878]    0.0             1     3003778\n",
      "39  (3003878, 3003978]    8.0             9     3003878\n",
      "40  (3003978, 3004078]    0.0             0     3003978\n",
      "41  (3004078, 3004178]    0.0             0     3004078\n",
      "42  (3004178, 3004278]    3.0             5     3004178\n",
      "43  (3004278, 3004378]    2.0             3     3004278\n",
      "44  (3004378, 3004478]    0.0             0     3004378\n",
      "45  (3004478, 3004578]    0.0             6     3004478\n",
      "46  (3004578, 3004678]    1.0             2     3004578\n",
      "47  (3004678, 3004778]    0.0             0     3004678\n",
      "48  (3004778, 3004878]   10.0             6     3004778\n",
      "49  (3004878, 3004978]    3.0             3     3004878\n",
      "(152618, 4)\n",
      "bin_offset_60    category\n",
      "TT_S0             float64\n",
      "target_count        int64\n",
      "starting_nt      category\n",
      "dtype: object\n",
      "sumos ilgis: 152618\n",
      "ranges ilgis: 152618\n",
      "Model input - NO SCALE\n",
      "        TT_S0  genes  transcripts  exons  CDSs\n",
      "0         4.0      0            0      0     0\n",
      "1         4.0      0            0      0     0\n",
      "2         0.0      0            0      0     0\n",
      "3         2.0      0            0      0     0\n",
      "4         0.0      0            0      0     0\n",
      "...       ...    ...          ...    ...   ...\n",
      "152613    4.0      0            0      0     0\n",
      "152614    4.0      0            0      0     0\n",
      "152615    0.0      0            0      0     0\n",
      "152616    0.0      0            0      0     0\n",
      "152617    4.0      0            0      0     0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "        TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0         4.0             0                   0             0            0\n",
      "1         4.0             0                   0             0            0\n",
      "2         0.0             0                   0             0            0\n",
      "3         2.0             0                   0             0            0\n",
      "4         0.0             0                   0             0            0\n",
      "...       ...           ...                 ...           ...          ...\n",
      "152613    4.0             0                   0             0            0\n",
      "152614    4.0             0                   0             0            0\n",
      "152615    0.0             0                   0             0            0\n",
      "152616    0.0             0                   0             0            0\n",
      "152617    4.0             0                   0             0            0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "0.19076386795790798\n",
      "[5 5 6 1 6 6 6 6 6 6 1 6 6 6 6 5 5 6 6 6 1 6 5 0 2 5 6 6 6 1 1 6 6 1 6 6 1\n",
      " 5 6 0 6 6 5 1 6 6 1 6 9 5 6 3 7 9 6 7 8 8 8 4 6 9 2 5 6 6 6 0 6 1 0 6 5 1\n",
      " 6 6 1 6 6 1 0 6 1 5 5 1 5 0 6 6 0 9 9 6 1 5 6 5 0 5 2 2 7 2 0 5 9 7 5 9 6\n",
      " 0 0 5 6 1 1 6 6 6 6 5 5 1 1 9 0 2 2 0 9 1 9 1 5 5 5 0 1 6 6 6 6 1 6 1 6 2\n",
      " 1 0 5 1 6 5 1 9 1 5 1 9 5 5 0 1 0 9 1 1 6 6 6 1 6 6 1 6 6 6 1 6 9 5 6 5 6\n",
      " 6 5 6 6 6 6 1 6 6 6 6 6 6 6 6 6 6 6 6 6 6 9 2 1 1 6 6 1 6 6 6 6 1 6 6 6 6\n",
      " 6 6 6 6 6 6 6 6 6 6 6 6 6 5 6 6 6 1 6 6 6 6 6 6 6 6 6 6 6 1 6 6 6 6 1 6 1\n",
      " 6 6 6 6 6 1 6 6 6 5 1 6 1 1 1 1 6 6 1 6 1 1 6 6 1 6 6 6 6 6 6 6 1 6 6 6 1\n",
      " 5 6 6 1 1 5 6 1 6 6 6 6 5 1 5 0 5 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6\n",
      " 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6\n",
      " 6 6 6 6 6 6 6 6 6 6 5 6 1 6 5 0 6 1 6 1 6 6 6 6 9 2 5 9 6 2 0 5 1 1 5 9 6\n",
      " 5 5 5 1 6 1 2 2 5 5 9 6 6 5 6 6 1 6 5 1 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 5\n",
      " 6 6 1 1 5 6 6 6 6 1 1 6 1 6 1 1 1 5 0 1 6 1 5 1 9 6 6 5 0 0 0 5 1 1 1 1 6\n",
      " 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6\n",
      " 0 9 6 6 6 5 6 6 1 0 5 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6\n",
      " 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6\n",
      " 6 6 6 5 6 1 5 1 6 6 6 6 6 1 6 6 6 6 1 6 1 5 1 6 5 1 6 6 6 6 0 6 6 6 6 6 0\n",
      " 6 6 5 6 6 6 6 1 6 9 6 6 1 0 5 6 6 6 6 1 0 6 5 6 6 6 6 1 5 5 6 1 5 5 6 5 1\n",
      " 0 6 6 6 6 1 1 6 6 6 6 6 6 6 6 6 6 5 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6\n",
      " 6 6 6 6 6 6 6 6 6 6 6 6 6 5 5 0 5 5 6 6 1 1 6 6 1 6 6 6 5 6 6 6 6 6 1 6 6\n",
      " 6 1 6 6 6 1 6 1 6 6 6 6 6 6 1 6 6 6 6 6 1 6 6 6 6 6 5 6 6 6 6 1 5 6 6 6 6\n",
      " 6 6 6 6 6 6 6 1 6 6 6 6 6 5 6 6 6 6 1 1 6 6 5 6 6 6 6 9 1 9 5 6 5 1 5 6 6\n",
      " 6 6 6 6 1 6 6 6 6 6 6 6 6 6 6 6 6 1 1 1 1 6 1 6 5 6 1 6 1 6 1 1 6 6 6 1 6\n",
      " 6 6 6 6 6 6 6 6 6 6 1 6 6 6 1 6 1 6 0 1 6 6 6 6 1 5 6 6 6 6 6 6 6 6 6 6 6\n",
      " 6 6 5 6 6 6 6 1 6 6 6 6 6 6 6 6 6 6 6 6 1 1 6 1 1 1 5 5 6 6 6 6 1 6 6 1 5\n",
      " 5 0 0 1 1 1 6 6 6 6 6 6 6 6 1 6 6 6 6 5 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6\n",
      " 6 6 6 6 6 6 6 6 5 6 6 6 6 6 6 1 6 6 6 6 6 6 6 1 6 6 6 5 1 0 1 6 6 1 6 6 6\n",
      " 6]\n",
      "threshold density 1.95819\n",
      "cluster signal 92809.0\n",
      "cluster target count 131728\n",
      "!!cluster density 0.7045502854366573\n",
      "cluster signal 0.0\n",
      "cluster target count 227632\n",
      "!!cluster density 0.0\n",
      "cluster signal 41912.0\n",
      "cluster target count 135495\n",
      "!!cluster density 0.3093250673456585\n",
      "cluster signal 108005.0\n",
      "cluster target count 95611\n",
      "!!cluster density 1.129629435943563\n",
      "cluster signal 68247.0\n",
      "cluster target count 33260\n",
      "!!cluster density 2.051924233313289\n",
      "cluster signal 90107.0\n",
      "cluster target count 57498\n",
      "!!cluster density 1.5671327698354725\n",
      "cluster signal 15503.0\n",
      "cluster target count 4341\n",
      "!!cluster density 3.571296936189818\n",
      "cluster signal 37194.0\n",
      "cluster target count 13987\n",
      "!!cluster density 2.6591835275613067\n",
      "cluster signal 2642.0\n",
      "cluster target count 441\n",
      "!!cluster density 5.99092970521542\n",
      "cluster signal 492.0\n",
      "cluster target count 7\n",
      "!!cluster density 70.28571428571429\n",
      "\n",
      "Labels mapping\n",
      " {5: 0, 6: 0, 1: 0, 0: 0, 2: 1, 9: 0, 3: 1, 7: 1, 8: 1, 4: 1}\n",
      "DEBUG: 147001 5617 \n",
      "\n",
      " +----------------------+-------------+----------------+-------+\n",
      "| bin_offset_60        | TT_S0       | target_count   | +9    |\n",
      "| (category)           | (float64)   | (int64)        | ...   |\n",
      "|----------------------+-------------+----------------+-------|\n",
      "| (3000018, 3000078]   | 4.0         | 5              | ...   |\n",
      "| (3000078, 3000178]   | 4.0         | 10             | ...   |\n",
      "| (3000178, 3000278]   | 0.0         | 1              | ...   |\n",
      "| (3000278, 3000378]   | 2.0         | 5              | ...   |\n",
      "| ...                  | ...         | ...            | ...   |\n",
      "| (18261378, 18261478] | 4.0         | 6              | ...   |\n",
      "| (18261478, 18261578] | 0.0         | 2              | ...   |\n",
      "| (18261578, 18261678] | 0.0         | 2              | ...   |\n",
      "| (18261678, 18261710] | 4.0         | 3              | ...   |\n",
      "+----------------------+-------------+----------------+-------+\n",
      "Unstranded PyRanges object has 147,001 rows and 12 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "9 hidden columns: starting_nt, Chromosome, Start, End, genes, transcripts, ... (+ 3 more.)\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0\n",
      "BINS  IntervalIndex([  (3000018, 3000083],   (3000083, 3000183],\n",
      "                 (3000183, 3000283],   (3000283, 3000383],\n",
      "                 (3000383, 3000483],   (3000483, 3000583],\n",
      "                 (3000583, 3000683],   (3000683, 3000783],\n",
      "                 (3000783, 3000883],   (3000883, 3000983],\n",
      "               ...\n",
      "               (18260783, 18260883], (18260883, 18260983],\n",
      "               (18260983, 18261083], (18261083, 18261183],\n",
      "               (18261183, 18261283], (18261283, 18261383],\n",
      "               (18261383, 18261483], (18261483, 18261583],\n",
      "               (18261583, 18261683], (18261683, 18261710]],\n",
      "              dtype='interval[int64, right]', length=152618)\n",
      "HEAD with BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0   \n",
      "\n",
      "        bin_offset_65  target_count  \n",
      "0  (3000018, 3000083]             1  \n",
      "1  (3000018, 3000083]             1  \n",
      "2  (3000018, 3000083]             1  \n",
      "3  (3000018, 3000083]             1  \n",
      "4  (3000018, 3000083]             1  \n",
      "TAIL with BINS\n",
      "       seqnames     start       end        CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  18261611  18261611  18_18261611    0.0    0.0    0.0   \n",
      "699996       18  18261615  18261615  18_18261615    0.0    0.0    0.0   \n",
      "699997       18  18261682  18261682  18_18261682    0.0    0.0    0.0   \n",
      "699998       18  18261702  18261702  18_18261702    4.0    0.0    0.0   \n",
      "699999       18  18261710  18261710  18_18261710    0.0    0.0    0.0   \n",
      "\n",
      "               bin_offset_65  target_count  \n",
      "699995  (18261583, 18261683]             1  \n",
      "699996  (18261583, 18261683]             1  \n",
      "699997  (18261583, 18261683]             1  \n",
      "699998  (18261683, 18261710]             1  \n",
      "699999  (18261683, 18261710]             1  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:137: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "         bin_offset_65  TT_S0  target_count starting_nt\n",
      "0   (3000018, 3000083]    4.0             5     3000018\n",
      "1   (3000083, 3000183]    4.0            10     3000083\n",
      "2   (3000183, 3000283]    0.0             1     3000183\n",
      "3   (3000283, 3000383]    2.0             5     3000283\n",
      "4   (3000383, 3000483]    0.0             3     3000383\n",
      "5   (3000483, 3000583]    0.0             1     3000483\n",
      "6   (3000583, 3000683]    0.0             4     3000583\n",
      "7   (3000683, 3000783]    0.0             4     3000683\n",
      "8   (3000783, 3000883]    0.0             6     3000783\n",
      "9   (3000883, 3000983]    0.0             4     3000883\n",
      "10  (3000983, 3001083]    2.0             6     3000983\n",
      "11  (3001083, 3001183]    0.0             2     3001083\n",
      "12  (3001183, 3001283]    0.0             2     3001183\n",
      "13  (3001283, 3001383]    0.0             5     3001283\n",
      "14  (3001383, 3001483]    0.0             3     3001383\n",
      "15  (3001483, 3001583]    4.0             9     3001483\n",
      "16  (3001583, 3001683]    3.0             7     3001583\n",
      "17  (3001683, 3001783]    0.0             7     3001683\n",
      "18  (3001783, 3001883]    0.0             4     3001783\n",
      "19  (3001883, 3001983]    0.0             2     3001883\n",
      "20  (3001983, 3002083]    1.0             3     3001983\n",
      "21  (3002083, 3002183]    0.0             1     3002083\n",
      "22  (3002183, 3002283]    4.0             7     3002183\n",
      "23  (3002283, 3002383]    9.0             3     3002283\n",
      "24  (3002383, 3002483]   19.0             4     3002383\n",
      "25  (3002483, 3002583]    4.0             3     3002483\n",
      "26  (3002583, 3002683]    0.0             1     3002583\n",
      "27  (3002683, 3002783]    0.0             3     3002683\n",
      "28  (3002783, 3002883]    0.0             2     3002783\n",
      "29  (3002883, 3002983]    1.0             3     3002883\n",
      "30  (3002983, 3003083]    2.0             7     3002983\n",
      "31  (3003083, 3003183]    0.0             0     3003083\n",
      "32  (3003183, 3003283]    0.0             0     3003183\n",
      "33  (3003283, 3003383]    1.0             9     3003283\n",
      "34  (3003383, 3003483]    0.0             1     3003383\n",
      "35  (3003483, 3003583]    0.0             3     3003483\n",
      "36  (3003583, 3003683]    2.0             3     3003583\n",
      "37  (3003683, 3003783]    3.0             7     3003683\n",
      "38  (3003783, 3003883]    0.0             1     3003783\n",
      "39  (3003883, 3003983]    8.0             9     3003883\n",
      "40  (3003983, 3004083]    0.0             0     3003983\n",
      "41  (3004083, 3004183]    0.0             0     3004083\n",
      "42  (3004183, 3004283]    3.0             5     3004183\n",
      "43  (3004283, 3004383]    2.0             3     3004283\n",
      "44  (3004383, 3004483]    0.0             0     3004383\n",
      "45  (3004483, 3004583]    0.0             6     3004483\n",
      "46  (3004583, 3004683]    1.0             2     3004583\n",
      "47  (3004683, 3004783]    0.0             0     3004683\n",
      "48  (3004783, 3004883]   10.0             7     3004783\n",
      "49  (3004883, 3004983]    3.0             2     3004883\n",
      "(152618, 4)\n",
      "bin_offset_65    category\n",
      "TT_S0             float64\n",
      "target_count        int64\n",
      "starting_nt      category\n",
      "dtype: object\n",
      "sumos ilgis: 152618\n",
      "ranges ilgis: 152618\n",
      "Model input - NO SCALE\n",
      "        TT_S0  genes  transcripts  exons  CDSs\n",
      "0         4.0      0            0      0     0\n",
      "1         4.0      0            0      0     0\n",
      "2         0.0      0            0      0     0\n",
      "3         2.0      0            0      0     0\n",
      "4         0.0      0            0      0     0\n",
      "...       ...    ...          ...    ...   ...\n",
      "152613    4.0      0            0      0     0\n",
      "152614    4.0      0            0      0     0\n",
      "152615    0.0      0            0      0     0\n",
      "152616    0.0      0            0      0     0\n",
      "152617    4.0      0            0      0     0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "        TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0         4.0             0                   0             0            0\n",
      "1         4.0             0                   0             0            0\n",
      "2         0.0             0                   0             0            0\n",
      "3         2.0             0                   0             0            0\n",
      "4         0.0             0                   0             0            0\n",
      "...       ...           ...                 ...           ...          ...\n",
      "152613    4.0             0                   0             0            0\n",
      "152614    4.0             0                   0             0            0\n",
      "152615    0.0             0                   0             0            0\n",
      "152616    0.0             0                   0             0            0\n",
      "152617    4.0             0                   0             0            0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "0.5723702315585317\n",
      "[0 0 1 8 1 1 1 1 1 1 8 1 1 1 1 0 0 1 1 1 1 1 0 3 2 0 1 1 1 1 8 1 1 1 1 1 8\n",
      " 0 1 3 1 1 0 8 1 1 1 1 3 0 1 4 5 3 2 7 4 4 4 6 1 3 2 0 1 1 9 1 1 1 3 1 0 1\n",
      " 1 1 8 1 1 8 3 1 8 0 0 1 0 3 1 1 9 3 3 1 8 0 1 0 9 0 7 2 5 2 9 0 7 2 0 3 1\n",
      " 3 3 1 1 1 1 1 1 1 1 0 0 1 1 7 9 2 3 3 3 1 7 0 0 0 0 3 1 1 1 1 1 8 1 1 8 7\n",
      " 1 9 0 8 1 0 8 3 8 0 1 7 8 3 0 1 9 7 8 1 1 1 1 1 1 1 8 1 1 1 1 1 3 0 1 0 1\n",
      " 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 3 2 1 1 1 1 8 1 1 1 1 8 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 8 1 1\n",
      " 1 1 1 1 1 8 1 1 1 0 8 1 8 8 8 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 8\n",
      " 0 1 1 1 8 0 1 8 1 1 1 1 0 1 0 9 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 0 1 8 1 0 3 1 8 1 1 1 1 1 1 3 7 0 7 1 2 3 0 8 1 0 3 1\n",
      " 8 0 1 1 1 1 7 7 0 0 3 1 1 0 1 1 8 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0\n",
      " 1 1 1 8 0 1 1 1 1 8 1 1 1 1 1 1 1 0 3 1 1 1 0 8 3 1 1 0 3 9 9 0 8 0 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 9 7 1 1 1 0 1 1 0 9 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 0 1 8 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 9 1 1 1 1 1 3 1 1 1 1 1 9\n",
      " 1 1 0 1 1 1 1 1 1 7 1 1 1 3 0 1 1 1 1 1 9 1 0 1 1 1 1 1 0 0 1 1 0 0 1 0 8\n",
      " 9 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 0 1 0 3 8 0 1 1 8 1 1 1 1 1 1 1 0 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 8 0 1 1 1 1\n",
      " 1 1 1 1 1 1 1 8 1 1 1 1 1 0 1 1 1 1 1 8 1 1 0 1 1 1 1 3 1 3 0 1 0 8 0 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 8 1 1 8 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 8 1 1 1 1 1 8 1 9 1 1 1 1 1 1 0 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 1 0 0 1 1 1 1 1 1 1 1 0\n",
      " 0 9 9 1 8 8 1 1 1 1 1 1 1 1 8 1 1 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 9 1 1 1 1 1 1 1\n",
      " 1]\n",
      "threshold density 1.95819\n",
      "cluster signal 93113.0\n",
      "cluster target count 131923\n",
      "!!cluster density 0.7058132395412475\n",
      "cluster signal 16338.0\n",
      "cluster target count 300984\n",
      "!!cluster density 0.05428195518698668\n",
      "cluster signal 25526.0\n",
      "cluster target count 61945\n",
      "!!cluster density 0.41207522802486074\n",
      "cluster signal 90980.0\n",
      "cluster target count 67893\n",
      "!!cluster density 1.3400497842192862\n",
      "cluster signal 52026.0\n",
      "cluster target count 22749\n",
      "!!cluster density 2.286957668468944\n",
      "cluster signal 10315.0\n",
      "cluster target count 2300\n",
      "!!cluster density 4.484782608695652\n",
      "cluster signal 27574.0\n",
      "cluster target count 9405\n",
      "!!cluster density 2.931844763423711\n",
      "cluster signal 81693.0\n",
      "cluster target count 46271\n",
      "!!cluster density 1.7655334874975688\n",
      "cluster signal 492.0\n",
      "cluster target count 7\n",
      "!!cluster density 70.28571428571429\n",
      "cluster signal 58854.0\n",
      "cluster target count 56523\n",
      "!!cluster density 1.041239849264901\n",
      "\n",
      "Labels mapping\n",
      " {0: 0, 1: 0, 8: 0, 3: 0, 2: 1, 4: 1, 5: 1, 7: 0, 6: 1, 9: 0}\n",
      "DEBUG: 149129 3489 \n",
      "\n",
      " +----------------------+-------------+----------------+-------+\n",
      "| bin_offset_65        | TT_S0       | target_count   | +9    |\n",
      "| (category)           | (float64)   | (int64)        | ...   |\n",
      "|----------------------+-------------+----------------+-------|\n",
      "| (3000018, 3000083]   | 4.0         | 5              | ...   |\n",
      "| (3000083, 3000183]   | 4.0         | 10             | ...   |\n",
      "| (3000183, 3000283]   | 0.0         | 1              | ...   |\n",
      "| (3000283, 3000383]   | 2.0         | 5              | ...   |\n",
      "| ...                  | ...         | ...            | ...   |\n",
      "| (18261383, 18261483] | 4.0         | 5              | ...   |\n",
      "| (18261483, 18261583] | 0.0         | 2              | ...   |\n",
      "| (18261583, 18261683] | 0.0         | 3              | ...   |\n",
      "| (18261683, 18261710] | 4.0         | 2              | ...   |\n",
      "+----------------------+-------------+----------------+-------+\n",
      "Unstranded PyRanges object has 149,129 rows and 12 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "9 hidden columns: starting_nt, Chromosome, Start, End, genes, transcripts, ... (+ 3 more.)\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0\n",
      "BINS  IntervalIndex([  (3000018, 3000088],   (3000088, 3000188],\n",
      "                 (3000188, 3000288],   (3000288, 3000388],\n",
      "                 (3000388, 3000488],   (3000488, 3000588],\n",
      "                 (3000588, 3000688],   (3000688, 3000788],\n",
      "                 (3000788, 3000888],   (3000888, 3000988],\n",
      "               ...\n",
      "               (18260788, 18260888], (18260888, 18260988],\n",
      "               (18260988, 18261088], (18261088, 18261188],\n",
      "               (18261188, 18261288], (18261288, 18261388],\n",
      "               (18261388, 18261488], (18261488, 18261588],\n",
      "               (18261588, 18261688], (18261688, 18261710]],\n",
      "              dtype='interval[int64, right]', length=152618)\n",
      "HEAD with BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0   \n",
      "\n",
      "        bin_offset_70  target_count  \n",
      "0  (3000018, 3000088]             1  \n",
      "1  (3000018, 3000088]             1  \n",
      "2  (3000018, 3000088]             1  \n",
      "3  (3000018, 3000088]             1  \n",
      "4  (3000018, 3000088]             1  \n",
      "TAIL with BINS\n",
      "       seqnames     start       end        CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  18261611  18261611  18_18261611    0.0    0.0    0.0   \n",
      "699996       18  18261615  18261615  18_18261615    0.0    0.0    0.0   \n",
      "699997       18  18261682  18261682  18_18261682    0.0    0.0    0.0   \n",
      "699998       18  18261702  18261702  18_18261702    4.0    0.0    0.0   \n",
      "699999       18  18261710  18261710  18_18261710    0.0    0.0    0.0   \n",
      "\n",
      "               bin_offset_70  target_count  \n",
      "699995  (18261588, 18261688]             1  \n",
      "699996  (18261588, 18261688]             1  \n",
      "699997  (18261588, 18261688]             1  \n",
      "699998  (18261688, 18261710]             1  \n",
      "699999  (18261688, 18261710]             1  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:137: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "         bin_offset_70  TT_S0  target_count starting_nt\n",
      "0   (3000018, 3000088]    7.0             8     3000018\n",
      "1   (3000088, 3000188]    1.0             7     3000088\n",
      "2   (3000188, 3000288]    0.0             1     3000188\n",
      "3   (3000288, 3000388]    2.0             7     3000288\n",
      "4   (3000388, 3000488]    0.0             1     3000388\n",
      "5   (3000488, 3000588]    0.0             1     3000488\n",
      "6   (3000588, 3000688]    0.0             5     3000588\n",
      "7   (3000688, 3000788]    0.0             5     3000688\n",
      "8   (3000788, 3000888]    0.0             5     3000788\n",
      "9   (3000888, 3000988]    0.0             3     3000888\n",
      "10  (3000988, 3001088]    2.0             6     3000988\n",
      "11  (3001088, 3001188]    0.0             3     3001088\n",
      "12  (3001188, 3001288]    0.0             1     3001188\n",
      "13  (3001288, 3001388]    0.0             5     3001288\n",
      "14  (3001388, 3001488]    0.0             4     3001388\n",
      "15  (3001488, 3001588]    4.0             8     3001488\n",
      "16  (3001588, 3001688]    3.0             7     3001588\n",
      "17  (3001688, 3001788]    0.0             7     3001688\n",
      "18  (3001788, 3001888]    0.0             4     3001788\n",
      "19  (3001888, 3001988]    0.0             2     3001888\n",
      "20  (3001988, 3002088]    1.0             3     3001988\n",
      "21  (3002088, 3002188]    0.0             1     3002088\n",
      "22  (3002188, 3002288]    4.0             7     3002188\n",
      "23  (3002288, 3002388]    9.0             3     3002288\n",
      "24  (3002388, 3002488]   19.0             4     3002388\n",
      "25  (3002488, 3002588]    4.0             3     3002488\n",
      "26  (3002588, 3002688]    0.0             1     3002588\n",
      "27  (3002688, 3002788]    0.0             3     3002688\n",
      "28  (3002788, 3002888]    0.0             2     3002788\n",
      "29  (3002888, 3002988]    1.0             4     3002888\n",
      "30  (3002988, 3003088]    2.0             6     3002988\n",
      "31  (3003088, 3003188]    0.0             0     3003088\n",
      "32  (3003188, 3003288]    0.0             0     3003188\n",
      "33  (3003288, 3003388]    1.0             9     3003288\n",
      "34  (3003388, 3003488]    0.0             1     3003388\n",
      "35  (3003488, 3003588]    1.0             4     3003488\n",
      "36  (3003588, 3003688]    1.0             3     3003588\n",
      "37  (3003688, 3003788]    3.0             6     3003688\n",
      "38  (3003788, 3003888]    0.0             1     3003788\n",
      "39  (3003888, 3003988]    8.0             9     3003888\n",
      "40  (3003988, 3004088]    0.0             0     3003988\n",
      "41  (3004088, 3004188]    0.0             0     3004088\n",
      "42  (3004188, 3004288]    3.0             5     3004188\n",
      "43  (3004288, 3004388]    2.0             3     3004288\n",
      "44  (3004388, 3004488]    0.0             0     3004388\n",
      "45  (3004488, 3004588]    0.0             6     3004488\n",
      "46  (3004588, 3004688]    1.0             2     3004588\n",
      "47  (3004688, 3004788]    0.0             0     3004688\n",
      "48  (3004788, 3004888]   10.0             7     3004788\n",
      "49  (3004888, 3004988]    3.0             2     3004888\n",
      "(152618, 4)\n",
      "bin_offset_70    category\n",
      "TT_S0             float64\n",
      "target_count        int64\n",
      "starting_nt      category\n",
      "dtype: object\n",
      "sumos ilgis: 152618\n",
      "ranges ilgis: 152618\n",
      "Model input - NO SCALE\n",
      "        TT_S0  genes  transcripts  exons  CDSs\n",
      "0         7.0      0            0      0     0\n",
      "1         1.0      0            0      0     0\n",
      "2         0.0      0            0      0     0\n",
      "3         2.0      0            0      0     0\n",
      "4         0.0      0            0      0     0\n",
      "...       ...    ...          ...    ...   ...\n",
      "152613    4.0      0            0      0     0\n",
      "152614    4.0      0            0      0     0\n",
      "152615    0.0      0            0      0     0\n",
      "152616    0.0      0            0      0     0\n",
      "152617    4.0      0            0      0     0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "        TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0         7.0             0                   0             0            0\n",
      "1         1.0             0                   0             0            0\n",
      "2         0.0             0                   0             0            0\n",
      "3         2.0             0                   0             0            0\n",
      "4         0.0             0                   0             0            0\n",
      "...       ...           ...                 ...           ...          ...\n",
      "152613    4.0             0                   0             0            0\n",
      "152614    4.0             0                   0             0            0\n",
      "152615    0.0             0                   0             0            0\n",
      "152616    0.0             0                   0             0            0\n",
      "152617    4.0             0                   0             0            0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "0.04402495118531235\n",
      "[6 0 0 3 0 0 0 0 0 0 3 0 0 0 0 3 3 0 0 0 0 0 3 6 8 3 0 0 0 0 3 0 0 0 0 0 0\n",
      " 3 0 6 0 0 3 3 0 0 0 0 1 3 0 9 2 6 8 8 9 5 5 7 0 1 4 3 0 0 6 0 0 0 6 0 3 0\n",
      " 0 0 3 0 0 3 6 0 3 3 3 0 3 6 0 0 6 8 3 0 3 3 0 3 6 3 8 4 4 4 6 3 8 4 3 1 0\n",
      " 6 6 0 0 0 0 0 0 0 0 3 3 0 0 8 1 8 1 1 1 6 6 3 3 3 3 6 0 0 0 0 0 3 0 0 3 8\n",
      " 0 6 3 3 0 3 3 1 3 3 0 8 3 6 3 0 6 1 3 0 0 0 0 0 0 0 3 0 0 0 0 0 1 3 0 3 0\n",
      " 3 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 4 0 0 0 0 3 0 0 0 0 3 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 3 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 3 0 0\n",
      " 0 0 0 0 0 3 0 0 0 3 3 0 3 3 3 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 3\n",
      " 3 0 0 0 3 3 0 3 0 0 0 0 3 0 3 6 3 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 3 0 3 0 3 6 0 3 0 0 0 0 0 0 1 8 3 1 0 4 3 3 3 0 3 1 3\n",
      " 0 3 0 0 0 0 8 8 3 3 1 0 0 3 0 0 3 3 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 3\n",
      " 0 0 0 3 3 0 0 0 0 3 0 0 0 0 0 0 0 3 6 0 0 0 3 3 1 0 0 3 6 3 6 3 3 3 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 6 1 0 0 0 3 0 0 6 0 3 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 3 0 3 3 0 0 0 0 0 0 0 0 0 0 0 0 0 3 0 0 0 6 0 0 0 0 0 6 0 0 0 0 0 6\n",
      " 0 0 3 0 0 0 0 0 0 1 0 0 3 6 3 0 0 0 0 0 6 0 3 0 0 0 0 0 3 3 0 0 3 3 0 3 3\n",
      " 6 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 3 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 3 0 3 1 3 3 0 0 3 0 0 0 0 0 0 0 3 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 3 0 0 0 0 3 3 0 0 0 0\n",
      " 0 0 0 0 0 0 0 3 0 0 0 0 0 3 0 0 0 0 0 3 0 0 3 0 0 0 0 1 0 1 3 3 0 3 3 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 3 0 0 0 3 0 0 3 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 3 0 0 0 0 0 3 0 6 0 0 0 0 0 0 3 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 3 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 3 0 0 0 0 0 3 3 0 0 0 0 0 0 0 0 3\n",
      " 3 6 6 0 3 0 0 0 0 0 0 0 0 0 3 0 0 0 0 3 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 3 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 3 0 6 0 0 0 0 0 0 0\n",
      " 0]\n",
      "threshold density 1.95819\n",
      "cluster signal 108776.0\n",
      "cluster target count 96796\n",
      "!!cluster density 1.1237654448530932\n",
      "cluster signal 16228.0\n",
      "cluster target count 300654\n",
      "!!cluster density 0.05397566638062357\n",
      "cluster signal 118243.0\n",
      "cluster target count 193548\n",
      "!!cluster density 0.6109233885134437\n",
      "cluster signal 64274.0\n",
      "cluster target count 33228\n",
      "!!cluster density 1.9343324906705188\n",
      "cluster signal 75667.0\n",
      "cluster target count 49117\n",
      "!!cluster density 1.540546043121526\n",
      "cluster signal 7730.0\n",
      "cluster target count 1786\n",
      "!!cluster density 4.328107502799552\n",
      "cluster signal 18539.0\n",
      "cluster target count 5979\n",
      "!!cluster density 3.1006857334002342\n",
      "cluster signal 830.0\n",
      "cluster target count 84\n",
      "!!cluster density 9.880952380952381\n",
      "cluster signal 492.0\n",
      "cluster target count 7\n",
      "!!cluster density 70.28571428571429\n",
      "cluster signal 46132.0\n",
      "cluster target count 18801\n",
      "!!cluster density 2.4536992713153554\n",
      "\n",
      "Labels mapping\n",
      " {6: 0, 0: 0, 3: 0, 8: 0, 1: 0, 9: 1, 2: 1, 5: 1, 7: 1, 4: 1}\n",
      "DEBUG: 150024 2594 \n",
      "\n",
      " +----------------------+-------------+----------------+-------+\n",
      "| bin_offset_70        | TT_S0       | target_count   | +9    |\n",
      "| (category)           | (float64)   | (int64)        | ...   |\n",
      "|----------------------+-------------+----------------+-------|\n",
      "| (3000018, 3000088]   | 7.0         | 8              | ...   |\n",
      "| (3000088, 3000188]   | 1.0         | 7              | ...   |\n",
      "| (3000188, 3000288]   | 0.0         | 1              | ...   |\n",
      "| (3000288, 3000388]   | 2.0         | 7              | ...   |\n",
      "| ...                  | ...         | ...            | ...   |\n",
      "| (18261388, 18261488] | 4.0         | 6              | ...   |\n",
      "| (18261488, 18261588] | 0.0         | 1              | ...   |\n",
      "| (18261588, 18261688] | 0.0         | 3              | ...   |\n",
      "| (18261688, 18261710] | 4.0         | 2              | ...   |\n",
      "+----------------------+-------------+----------------+-------+\n",
      "Unstranded PyRanges object has 150,024 rows and 12 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "9 hidden columns: starting_nt, Chromosome, Start, End, genes, transcripts, ... (+ 3 more.)\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0\n",
      "BINS  IntervalIndex([  (3000018, 3000093],   (3000093, 3000193],\n",
      "                 (3000193, 3000293],   (3000293, 3000393],\n",
      "                 (3000393, 3000493],   (3000493, 3000593],\n",
      "                 (3000593, 3000693],   (3000693, 3000793],\n",
      "                 (3000793, 3000893],   (3000893, 3000993],\n",
      "               ...\n",
      "               (18260793, 18260893], (18260893, 18260993],\n",
      "               (18260993, 18261093], (18261093, 18261193],\n",
      "               (18261193, 18261293], (18261293, 18261393],\n",
      "               (18261393, 18261493], (18261493, 18261593],\n",
      "               (18261593, 18261693], (18261693, 18261710]],\n",
      "              dtype='interval[int64, right]', length=152618)\n",
      "HEAD with BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0   \n",
      "\n",
      "        bin_offset_75  target_count  \n",
      "0  (3000018, 3000093]             1  \n",
      "1  (3000018, 3000093]             1  \n",
      "2  (3000018, 3000093]             1  \n",
      "3  (3000018, 3000093]             1  \n",
      "4  (3000018, 3000093]             1  \n",
      "TAIL with BINS\n",
      "       seqnames     start       end        CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  18261611  18261611  18_18261611    0.0    0.0    0.0   \n",
      "699996       18  18261615  18261615  18_18261615    0.0    0.0    0.0   \n",
      "699997       18  18261682  18261682  18_18261682    0.0    0.0    0.0   \n",
      "699998       18  18261702  18261702  18_18261702    4.0    0.0    0.0   \n",
      "699999       18  18261710  18261710  18_18261710    0.0    0.0    0.0   \n",
      "\n",
      "               bin_offset_75  target_count  \n",
      "699995  (18261593, 18261693]             1  \n",
      "699996  (18261593, 18261693]             1  \n",
      "699997  (18261593, 18261693]             1  \n",
      "699998  (18261693, 18261710]             1  \n",
      "699999  (18261693, 18261710]             1  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:137: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "         bin_offset_75  TT_S0  target_count starting_nt\n",
      "0   (3000018, 3000093]    7.0             9     3000018\n",
      "1   (3000093, 3000193]    1.0             6     3000093\n",
      "2   (3000193, 3000293]    0.0             1     3000193\n",
      "3   (3000293, 3000393]    2.0             7     3000293\n",
      "4   (3000393, 3000493]    0.0             1     3000393\n",
      "5   (3000493, 3000593]    0.0             1     3000493\n",
      "6   (3000593, 3000693]    0.0             6     3000593\n",
      "7   (3000693, 3000793]    0.0             4     3000693\n",
      "8   (3000793, 3000893]    0.0             6     3000793\n",
      "9   (3000893, 3000993]    0.0             2     3000893\n",
      "10  (3000993, 3001093]    2.0             6     3000993\n",
      "11  (3001093, 3001193]    0.0             3     3001093\n",
      "12  (3001193, 3001293]    0.0             2     3001193\n",
      "13  (3001293, 3001393]    0.0             5     3001293\n",
      "14  (3001393, 3001493]    0.0             3     3001393\n",
      "15  (3001493, 3001593]    4.0             9     3001493\n",
      "16  (3001593, 3001693]    3.0             7     3001593\n",
      "17  (3001693, 3001793]    0.0             7     3001693\n",
      "18  (3001793, 3001893]    0.0             3     3001793\n",
      "19  (3001893, 3001993]    0.0             2     3001893\n",
      "20  (3001993, 3002093]    1.0             3     3001993\n",
      "21  (3002093, 3002193]    0.0             2     3002093\n",
      "22  (3002193, 3002293]    4.0             6     3002193\n",
      "23  (3002293, 3002393]    9.0             3     3002293\n",
      "24  (3002393, 3002493]   19.0             5     3002393\n",
      "25  (3002493, 3002593]    4.0             2     3002493\n",
      "26  (3002593, 3002693]    0.0             2     3002593\n",
      "27  (3002693, 3002793]    0.0             2     3002693\n",
      "28  (3002793, 3002893]    0.0             2     3002793\n",
      "29  (3002893, 3002993]    1.0             5     3002893\n",
      "30  (3002993, 3003093]    2.0             5     3002993\n",
      "31  (3003093, 3003193]    0.0             0     3003093\n",
      "32  (3003193, 3003293]    0.0             0     3003193\n",
      "33  (3003293, 3003393]    1.0             9     3003293\n",
      "34  (3003393, 3003493]    0.0             1     3003393\n",
      "35  (3003493, 3003593]    1.0             4     3003493\n",
      "36  (3003593, 3003693]    1.0             3     3003593\n",
      "37  (3003693, 3003793]    3.0             6     3003693\n",
      "38  (3003793, 3003893]    0.0             2     3003793\n",
      "39  (3003893, 3003993]    8.0             8     3003893\n",
      "40  (3003993, 3004093]    0.0             0     3003993\n",
      "41  (3004093, 3004193]    0.0             1     3004093\n",
      "42  (3004193, 3004293]    3.0             4     3004193\n",
      "43  (3004293, 3004393]    2.0             3     3004293\n",
      "44  (3004393, 3004493]    0.0             1     3004393\n",
      "45  (3004493, 3004593]    0.0             5     3004493\n",
      "46  (3004593, 3004693]    1.0             2     3004593\n",
      "47  (3004693, 3004793]    0.0             0     3004693\n",
      "48  (3004793, 3004893]   10.0             7     3004793\n",
      "49  (3004893, 3004993]    3.0             2     3004893\n",
      "(152618, 4)\n",
      "bin_offset_75    category\n",
      "TT_S0             float64\n",
      "target_count        int64\n",
      "starting_nt      category\n",
      "dtype: object\n",
      "sumos ilgis: 152618\n",
      "ranges ilgis: 152618\n",
      "Model input - NO SCALE\n",
      "        TT_S0  genes  transcripts  exons  CDSs\n",
      "0         7.0      0            0      0     0\n",
      "1         1.0      0            0      0     0\n",
      "2         0.0      0            0      0     0\n",
      "3         2.0      0            0      0     0\n",
      "4         0.0      0            0      0     0\n",
      "...       ...    ...          ...    ...   ...\n",
      "152613    4.0      0            0      0     0\n",
      "152614    4.0      0            0      0     0\n",
      "152615    0.0      0            0      0     0\n",
      "152616    0.0      0            0      0     0\n",
      "152617    4.0      0            0      0     0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "        TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0         7.0             0                   0             0            0\n",
      "1         1.0             0                   0             0            0\n",
      "2         0.0             0                   0             0            0\n",
      "3         2.0             0                   0             0            0\n",
      "4         0.0             0                   0             0            0\n",
      "...       ...           ...                 ...           ...          ...\n",
      "152613    4.0             0                   0             0            0\n",
      "152614    4.0             0                   0             0            0\n",
      "152615    0.0             0                   0             0            0\n",
      "152616    0.0             0                   0             0            0\n",
      "152617    4.0             0                   0             0            0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "0.023457259300999882\n",
      "[2 0 4 0 4 4 4 4 4 4 0 4 4 4 4 2 0 4 4 4 0 4 2 7 1 2 4 4 4 0 0 4 4 0 4 0 0\n",
      " 0 4 7 4 4 0 0 4 4 0 4 7 0 4 3 5 2 1 9 8 8 8 6 4 7 5 2 4 4 2 4 4 2 2 4 2 0\n",
      " 4 4 0 4 4 0 7 4 0 0 0 0 0 7 4 4 2 1 2 4 0 0 4 0 2 2 1 1 5 5 2 2 9 5 2 9 4\n",
      " 9 2 4 4 0 0 4 4 4 4 2 2 4 0 1 9 1 9 9 9 2 7 2 0 2 2 7 4 4 4 4 4 0 0 4 0 1\n",
      " 0 2 2 0 0 2 4 9 0 0 0 1 0 7 2 4 7 9 0 0 4 4 4 0 4 4 0 4 4 4 0 0 9 0 4 0 4\n",
      " 0 4 4 4 4 4 0 4 4 4 4 4 4 4 4 4 4 4 4 4 4 9 1 0 0 4 4 0 4 4 4 4 0 4 4 4 4\n",
      " 4 4 4 4 4 4 4 4 4 4 4 4 4 0 4 4 4 0 4 4 4 4 4 4 4 4 4 4 4 0 4 4 4 0 4 4 0\n",
      " 4 4 4 4 4 0 4 4 4 0 0 4 0 0 0 0 4 0 4 4 0 0 4 4 0 4 4 4 4 4 4 4 0 4 4 4 0\n",
      " 0 4 4 0 0 0 4 0 4 4 4 4 2 0 2 2 2 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4\n",
      " 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4\n",
      " 4 4 4 4 4 4 4 4 4 4 2 4 0 4 2 7 4 0 0 4 4 4 4 0 9 9 2 9 4 5 2 0 0 0 2 7 0\n",
      " 0 2 0 0 4 0 1 9 0 2 7 4 4 0 4 0 4 2 4 0 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 0\n",
      " 4 4 0 0 0 4 4 4 4 0 0 4 0 4 0 0 0 0 7 0 4 0 0 0 7 4 0 0 7 2 2 0 0 0 4 0 4\n",
      " 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4\n",
      " 2 9 4 4 4 2 4 4 7 0 0 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4\n",
      " 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4\n",
      " 4 4 4 0 4 0 0 0 4 4 4 4 0 4 4 4 4 0 4 0 0 0 0 4 2 0 4 4 4 4 7 4 4 4 4 4 2\n",
      " 4 4 2 4 4 4 4 0 4 9 4 4 2 2 0 4 4 4 4 0 2 4 0 4 4 4 4 0 2 2 4 0 2 2 4 2 0\n",
      " 2 4 4 4 4 0 0 4 4 4 4 4 4 4 4 4 4 2 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4\n",
      " 4 4 4 4 4 4 4 4 4 4 4 4 0 4 0 9 0 0 4 4 0 0 4 4 0 4 4 4 0 4 4 4 4 4 0 4 4\n",
      " 4 0 4 4 0 4 4 0 4 4 4 4 4 4 0 4 4 4 4 4 0 4 4 4 4 4 0 4 4 4 4 0 0 4 4 4 4\n",
      " 4 4 4 4 4 4 0 4 4 4 4 4 0 2 4 4 4 0 4 0 4 4 2 4 4 4 4 7 0 1 4 0 4 0 0 4 4\n",
      " 4 4 4 4 0 4 4 4 4 4 4 4 4 4 4 4 4 0 0 0 0 4 0 4 2 4 0 4 0 4 0 0 4 4 4 0 4\n",
      " 4 4 4 4 4 4 4 4 4 4 0 4 4 0 4 4 0 4 2 0 4 4 4 4 0 2 4 4 4 4 4 4 4 4 4 4 4\n",
      " 4 4 0 4 4 4 4 0 4 4 4 4 4 4 4 4 4 4 4 4 0 4 4 0 0 0 2 2 4 4 4 4 0 4 4 0 2\n",
      " 0 2 2 0 2 4 4 4 4 4 4 4 4 4 0 4 4 4 4 0 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4\n",
      " 4 4 4 4 4 4 4 2 4 4 4 4 4 4 4 0 4 4 4 4 4 4 4 0 4 4 4 0 2 0 4 4 0 4 4 4 4\n",
      " 4]\n",
      "threshold density 1.95819\n",
      "cluster signal 121841.0\n",
      "cluster target count 136651\n",
      "!!cluster density 0.8916217224901392\n",
      "cluster signal 72061.0\n",
      "cluster target count 186720\n",
      "!!cluster density 0.3859308054841474\n",
      "cluster signal 0.0\n",
      "cluster target count 227935\n",
      "!!cluster density 0.0\n",
      "cluster signal 71562.0\n",
      "cluster target count 55263\n",
      "!!cluster density 1.294935128386081\n",
      "cluster signal 62181.0\n",
      "cluster target count 30444\n",
      "!!cluster density 2.042471422940481\n",
      "cluster signal 16452.0\n",
      "cluster target count 4661\n",
      "!!cluster density 3.5297146535078308\n",
      "cluster signal 41819.0\n",
      "cluster target count 15989\n",
      "!!cluster density 2.6154856463818876\n",
      "cluster signal 68009.0\n",
      "cluster target count 41937\n",
      "!!cluster density 1.6216944464315521\n",
      "cluster signal 2494.0\n",
      "cluster target count 393\n",
      "!!cluster density 6.346055979643766\n",
      "cluster signal 492.0\n",
      "cluster target count 7\n",
      "!!cluster density 70.28571428571429\n",
      "\n",
      "Labels mapping\n",
      " {2: 0, 0: 0, 4: 0, 7: 0, 1: 1, 3: 1, 5: 1, 9: 0, 8: 1, 6: 1}\n",
      "DEBUG: 147054 5564 \n",
      "\n",
      " +----------------------+-------------+----------------+-------+\n",
      "| bin_offset_75        | TT_S0       | target_count   | +9    |\n",
      "| (category)           | (float64)   | (int64)        | ...   |\n",
      "|----------------------+-------------+----------------+-------|\n",
      "| (3000018, 3000093]   | 7.0         | 9              | ...   |\n",
      "| (3000093, 3000193]   | 1.0         | 6              | ...   |\n",
      "| (3000193, 3000293]   | 0.0         | 1              | ...   |\n",
      "| (3000293, 3000393]   | 2.0         | 7              | ...   |\n",
      "| ...                  | ...         | ...            | ...   |\n",
      "| (18261393, 18261493] | 4.0         | 6              | ...   |\n",
      "| (18261493, 18261593] | 0.0         | 1              | ...   |\n",
      "| (18261593, 18261693] | 0.0         | 3              | ...   |\n",
      "| (18261693, 18261710] | 4.0         | 2              | ...   |\n",
      "+----------------------+-------------+----------------+-------+\n",
      "Unstranded PyRanges object has 147,054 rows and 12 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "9 hidden columns: starting_nt, Chromosome, Start, End, genes, transcripts, ... (+ 3 more.)\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0\n",
      "BINS  IntervalIndex([  (3000018, 3000098],   (3000098, 3000198],\n",
      "                 (3000198, 3000298],   (3000298, 3000398],\n",
      "                 (3000398, 3000498],   (3000498, 3000598],\n",
      "                 (3000598, 3000698],   (3000698, 3000798],\n",
      "                 (3000798, 3000898],   (3000898, 3000998],\n",
      "               ...\n",
      "               (18260798, 18260898], (18260898, 18260998],\n",
      "               (18260998, 18261098], (18261098, 18261198],\n",
      "               (18261198, 18261298], (18261298, 18261398],\n",
      "               (18261398, 18261498], (18261498, 18261598],\n",
      "               (18261598, 18261698], (18261698, 18261710]],\n",
      "              dtype='interval[int64, right]', length=152618)\n",
      "HEAD with BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0   \n",
      "\n",
      "        bin_offset_80  target_count  \n",
      "0  (3000018, 3000098]             1  \n",
      "1  (3000018, 3000098]             1  \n",
      "2  (3000018, 3000098]             1  \n",
      "3  (3000018, 3000098]             1  \n",
      "4  (3000018, 3000098]             1  \n",
      "TAIL with BINS\n",
      "       seqnames     start       end        CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  18261611  18261611  18_18261611    0.0    0.0    0.0   \n",
      "699996       18  18261615  18261615  18_18261615    0.0    0.0    0.0   \n",
      "699997       18  18261682  18261682  18_18261682    0.0    0.0    0.0   \n",
      "699998       18  18261702  18261702  18_18261702    4.0    0.0    0.0   \n",
      "699999       18  18261710  18261710  18_18261710    0.0    0.0    0.0   \n",
      "\n",
      "               bin_offset_80  target_count  \n",
      "699995  (18261598, 18261698]             1  \n",
      "699996  (18261598, 18261698]             1  \n",
      "699997  (18261598, 18261698]             1  \n",
      "699998  (18261698, 18261710]             1  \n",
      "699999  (18261698, 18261710]             1  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:137: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "         bin_offset_80  TT_S0  target_count starting_nt\n",
      "0   (3000018, 3000098]    7.0            10     3000018\n",
      "1   (3000098, 3000198]    1.0             5     3000098\n",
      "2   (3000198, 3000298]    0.0             1     3000198\n",
      "3   (3000298, 3000398]    2.0             7     3000298\n",
      "4   (3000398, 3000498]    0.0             1     3000398\n",
      "5   (3000498, 3000598]    0.0             1     3000498\n",
      "6   (3000598, 3000698]    0.0             6     3000598\n",
      "7   (3000698, 3000798]    0.0             5     3000698\n",
      "8   (3000798, 3000898]    0.0             5     3000798\n",
      "9   (3000898, 3000998]    0.0             2     3000898\n",
      "10  (3000998, 3001098]    2.0             7     3000998\n",
      "11  (3001098, 3001198]    0.0             2     3001098\n",
      "12  (3001198, 3001298]    0.0             2     3001198\n",
      "13  (3001298, 3001398]    0.0             5     3001298\n",
      "14  (3001398, 3001498]    0.0             3     3001398\n",
      "15  (3001498, 3001598]    4.0            10     3001498\n",
      "16  (3001598, 3001698]    3.0             6     3001598\n",
      "17  (3001698, 3001798]    0.0             7     3001698\n",
      "18  (3001798, 3001898]    0.0             3     3001798\n",
      "19  (3001898, 3001998]    0.0             2     3001898\n",
      "20  (3001998, 3002098]    1.0             3     3001998\n",
      "21  (3002098, 3002198]    0.0             2     3002098\n",
      "22  (3002198, 3002298]   11.0             7     3002198\n",
      "23  (3002298, 3002398]    2.0             2     3002298\n",
      "24  (3002398, 3002498]   19.0             5     3002398\n",
      "25  (3002498, 3002598]    4.0             2     3002498\n",
      "26  (3002598, 3002698]    0.0             2     3002598\n",
      "27  (3002698, 3002798]    0.0             2     3002698\n",
      "28  (3002798, 3002898]    0.0             2     3002798\n",
      "29  (3002898, 3002998]    1.0             5     3002898\n",
      "30  (3002998, 3003098]    2.0             5     3002998\n",
      "31  (3003098, 3003198]    0.0             0     3003098\n",
      "32  (3003198, 3003298]    0.0             0     3003198\n",
      "33  (3003298, 3003398]    1.0             9     3003298\n",
      "34  (3003398, 3003498]    0.0             1     3003398\n",
      "35  (3003498, 3003598]    1.0             4     3003498\n",
      "36  (3003598, 3003698]    1.0             3     3003598\n",
      "37  (3003698, 3003798]    3.0             6     3003698\n",
      "38  (3003798, 3003898]    0.0             2     3003798\n",
      "39  (3003898, 3003998]    8.0             8     3003898\n",
      "40  (3003998, 3004098]    0.0             0     3003998\n",
      "41  (3004098, 3004198]    0.0             1     3004098\n",
      "42  (3004198, 3004298]    3.0             5     3004198\n",
      "43  (3004298, 3004398]    2.0             2     3004298\n",
      "44  (3004398, 3004498]    0.0             1     3004398\n",
      "45  (3004498, 3004598]    0.0             6     3004498\n",
      "46  (3004598, 3004698]    1.0             1     3004598\n",
      "47  (3004698, 3004798]    0.0             0     3004698\n",
      "48  (3004798, 3004898]   10.0             7     3004798\n",
      "49  (3004898, 3004998]    3.0             2     3004898\n",
      "(152618, 4)\n",
      "bin_offset_80    category\n",
      "TT_S0             float64\n",
      "target_count        int64\n",
      "starting_nt      category\n",
      "dtype: object\n",
      "sumos ilgis: 152618\n",
      "ranges ilgis: 152618\n",
      "Model input - NO SCALE\n",
      "        TT_S0  genes  transcripts  exons  CDSs\n",
      "0         7.0      0            0      0     0\n",
      "1         1.0      0            0      0     0\n",
      "2         0.0      0            0      0     0\n",
      "3         2.0      0            0      0     0\n",
      "4         0.0      0            0      0     0\n",
      "...       ...    ...          ...    ...   ...\n",
      "152613    3.0      0            0      0     0\n",
      "152614    4.0      0            0      0     0\n",
      "152615    0.0      0            0      0     0\n",
      "152616    0.0      0            0      0     0\n",
      "152617    4.0      0            0      0     0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "        TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0         7.0             0                   0             0            0\n",
      "1         1.0             0                   0             0            0\n",
      "2         0.0             0                   0             0            0\n",
      "3         2.0             0                   0             0            0\n",
      "4         0.0             0                   0             0            0\n",
      "...       ...           ...                 ...           ...          ...\n",
      "152613    3.0             0                   0             0            0\n",
      "152614    4.0             0                   0             0            0\n",
      "152615    0.0             0                   0             0            0\n",
      "152616    0.0             0                   0             0            0\n",
      "152617    4.0             0                   0             0            0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "0.030291315572212978\n",
      "[9 2 2 7 2 2 2 2 2 2 7 2 2 2 2 0 7 2 2 2 2 2 5 7 1 0 2 2 2 2 7 2 2 2 2 2 2\n",
      " 7 2 9 2 2 7 7 2 2 2 2 5 7 2 4 3 9 1 1 4 8 8 6 2 5 3 0 2 2 9 2 2 0 0 2 0 2\n",
      " 2 2 7 2 2 7 5 7 2 7 7 7 7 9 2 2 5 1 7 2 7 7 2 7 9 9 3 5 3 1 9 9 5 3 0 5 2\n",
      " 1 0 2 2 2 2 2 2 2 2 0 0 2 5 7 5 1 5 5 5 9 9 0 2 0 0 5 2 2 2 2 2 7 2 2 7 1\n",
      " 2 9 0 7 2 9 2 5 7 7 2 1 2 9 9 2 9 5 7 2 2 2 2 2 2 2 7 2 2 2 2 2 5 7 7 2 2\n",
      " 7 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 5 1 2 2 2 2 7 2 2 2 2 7 2 2 2 2\n",
      " 2 2 2 2 2 2 2 2 2 2 2 2 2 7 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 7 2 2 2\n",
      " 2 2 2 2 2 7 2 2 2 7 7 2 7 7 7 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 7\n",
      " 7 2 2 2 7 7 2 7 2 2 2 2 0 2 0 0 0 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2\n",
      " 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2\n",
      " 2 2 2 2 2 2 2 2 2 2 0 2 7 2 0 5 2 7 2 2 2 2 2 2 5 5 0 5 2 3 9 7 7 2 9 5 7\n",
      " 2 0 2 2 2 2 1 1 7 9 9 2 2 7 2 7 2 0 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 7\n",
      " 2 2 2 0 2 2 2 2 2 7 2 2 2 2 2 2 2 7 5 2 2 2 7 7 5 2 0 2 5 0 9 7 7 7 2 2 2\n",
      " 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2\n",
      " 9 5 2 2 2 0 2 2 9 2 7 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2\n",
      " 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2\n",
      " 2 2 2 7 2 7 7 2 2 2 2 2 2 2 2 2 2 2 2 2 7 2 2 2 9 2 2 2 2 2 9 2 2 2 2 2 9\n",
      " 2 2 0 2 2 2 2 2 2 5 2 2 0 9 7 2 2 2 2 2 9 2 7 2 2 2 2 2 0 0 2 2 0 0 2 9 2\n",
      " 9 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 0 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2\n",
      " 2 2 2 2 2 2 2 2 2 2 2 2 7 2 5 0 7 7 2 2 7 2 2 2 2 2 2 2 7 2 2 2 2 2 2 2 2\n",
      " 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 7 2 2 2 2 7 7 2 2 2 2\n",
      " 2 2 2 2 2 2 7 2 2 2 2 2 2 0 2 2 2 2 2 7 2 2 0 2 2 2 2 5 9 5 2 7 2 7 7 2 2\n",
      " 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 0 2 2 2 7 2 2 7 2 2 2 2 2\n",
      " 2 2 2 2 2 2 2 2 2 2 7 2 2 2 2 2 7 2 9 2 2 2 2 2 2 0 2 2 2 2 2 2 2 2 2 2 2\n",
      " 2 2 7 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 7 2 2 2 2 2 0 0 2 2 2 2 2 2 2 2 9\n",
      " 2 9 9 2 0 2 2 2 2 2 2 2 2 2 7 2 2 2 7 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2\n",
      " 2 2 2 2 2 2 2 0 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 7 2 0 7 2 2 2 2 2 2 2\n",
      " 2]\n",
      "threshold density 1.95819\n",
      "cluster signal 85138.0\n",
      "cluster target count 78853\n",
      "!!cluster density 1.0797052743712985\n",
      "cluster signal 16250.0\n",
      "cluster target count 300959\n",
      "!!cluster density 0.05399406563684754\n",
      "cluster signal 55658.0\n",
      "cluster target count 113416\n",
      "!!cluster density 0.4907420469774988\n",
      "cluster signal 62726.0\n",
      "cluster target count 79631\n",
      "!!cluster density 0.7877083045547588\n",
      "cluster signal 99387.0\n",
      "cluster target count 67224\n",
      "!!cluster density 1.4784451981435203\n",
      "cluster signal 76812.0\n",
      "cluster target count 38999\n",
      "!!cluster density 1.9695889638195851\n",
      "cluster signal 14917.0\n",
      "cluster target count 3958\n",
      "!!cluster density 3.768822637695806\n",
      "cluster signal 44481.0\n",
      "cluster target count 16835\n",
      "!!cluster density 2.6421740421740423\n",
      "cluster signal 1050.0\n",
      "cluster target count 118\n",
      "!!cluster density 8.898305084745763\n",
      "cluster signal 492.0\n",
      "cluster target count 7\n",
      "!!cluster density 70.28571428571429\n",
      "\n",
      "Labels mapping\n",
      " {9: 0, 2: 0, 7: 0, 0: 0, 5: 0, 1: 1, 4: 1, 3: 1, 8: 1, 6: 1}\n",
      "DEBUG: 146024 6594 \n",
      "\n",
      " +----------------------+-------------+----------------+-------+\n",
      "| bin_offset_80        | TT_S0       | target_count   | +9    |\n",
      "| (category)           | (float64)   | (int64)        | ...   |\n",
      "|----------------------+-------------+----------------+-------|\n",
      "| (3000018, 3000098]   | 7.0         | 10             | ...   |\n",
      "| (3000098, 3000198]   | 1.0         | 5              | ...   |\n",
      "| (3000198, 3000298]   | 0.0         | 1              | ...   |\n",
      "| (3000298, 3000398]   | 2.0         | 7              | ...   |\n",
      "| ...                  | ...         | ...            | ...   |\n",
      "| (18261398, 18261498] | 4.0         | 5              | ...   |\n",
      "| (18261498, 18261598] | 0.0         | 1              | ...   |\n",
      "| (18261598, 18261698] | 0.0         | 3              | ...   |\n",
      "| (18261698, 18261710] | 4.0         | 2              | ...   |\n",
      "+----------------------+-------------+----------------+-------+\n",
      "Unstranded PyRanges object has 146,024 rows and 12 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "9 hidden columns: starting_nt, Chromosome, Start, End, genes, transcripts, ... (+ 3 more.)\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0\n",
      "BINS  IntervalIndex([  (3000018, 3000103],   (3000103, 3000203],\n",
      "                 (3000203, 3000303],   (3000303, 3000403],\n",
      "                 (3000403, 3000503],   (3000503, 3000603],\n",
      "                 (3000603, 3000703],   (3000703, 3000803],\n",
      "                 (3000803, 3000903],   (3000903, 3001003],\n",
      "               ...\n",
      "               (18260803, 18260903], (18260903, 18261003],\n",
      "               (18261003, 18261103], (18261103, 18261203],\n",
      "               (18261203, 18261303], (18261303, 18261403],\n",
      "               (18261403, 18261503], (18261503, 18261603],\n",
      "               (18261603, 18261703], (18261703, 18261710]],\n",
      "              dtype='interval[int64, right]', length=152618)\n",
      "HEAD with BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0   \n",
      "\n",
      "        bin_offset_85  target_count  \n",
      "0  (3000018, 3000103]             1  \n",
      "1  (3000018, 3000103]             1  \n",
      "2  (3000018, 3000103]             1  \n",
      "3  (3000018, 3000103]             1  \n",
      "4  (3000018, 3000103]             1  \n",
      "TAIL with BINS\n",
      "       seqnames     start       end        CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  18261611  18261611  18_18261611    0.0    0.0    0.0   \n",
      "699996       18  18261615  18261615  18_18261615    0.0    0.0    0.0   \n",
      "699997       18  18261682  18261682  18_18261682    0.0    0.0    0.0   \n",
      "699998       18  18261702  18261702  18_18261702    4.0    0.0    0.0   \n",
      "699999       18  18261710  18261710  18_18261710    0.0    0.0    0.0   \n",
      "\n",
      "               bin_offset_85  target_count  \n",
      "699995  (18261603, 18261703]             1  \n",
      "699996  (18261603, 18261703]             1  \n",
      "699997  (18261603, 18261703]             1  \n",
      "699998  (18261603, 18261703]             1  \n",
      "699999  (18261703, 18261710]             1  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:137: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "         bin_offset_85  TT_S0  target_count starting_nt\n",
      "0   (3000018, 3000103]    7.0            11     3000018\n",
      "1   (3000103, 3000203]    1.0             4     3000103\n",
      "2   (3000203, 3000303]    0.0             1     3000203\n",
      "3   (3000303, 3000403]    2.0             7     3000303\n",
      "4   (3000403, 3000503]    0.0             1     3000403\n",
      "5   (3000503, 3000603]    0.0             1     3000503\n",
      "6   (3000603, 3000703]    0.0             6     3000603\n",
      "7   (3000703, 3000803]    0.0             5     3000703\n",
      "8   (3000803, 3000903]    0.0             5     3000803\n",
      "9   (3000903, 3001003]    0.0             3     3000903\n",
      "10  (3001003, 3001103]    2.0             6     3001003\n",
      "11  (3001103, 3001203]    0.0             2     3001103\n",
      "12  (3001203, 3001303]    0.0             3     3001203\n",
      "13  (3001303, 3001403]    0.0             4     3001303\n",
      "14  (3001403, 3001503]    1.0             4     3001403\n",
      "15  (3001503, 3001603]    3.0             9     3001503\n",
      "16  (3001603, 3001703]    3.0             6     3001603\n",
      "17  (3001703, 3001803]    0.0             8     3001703\n",
      "18  (3001803, 3001903]    0.0             2     3001803\n",
      "19  (3001903, 3002003]    0.0             2     3001903\n",
      "20  (3002003, 3002103]    1.0             3     3002003\n",
      "21  (3002103, 3002203]    0.0             2     3002103\n",
      "22  (3002203, 3002303]   11.0             7     3002203\n",
      "23  (3002303, 3002403]    2.0             2     3002303\n",
      "24  (3002403, 3002503]   19.0             5     3002403\n",
      "25  (3002503, 3002603]    4.0             2     3002503\n",
      "26  (3002603, 3002703]    0.0             3     3002603\n",
      "27  (3002703, 3002803]    0.0             1     3002703\n",
      "28  (3002803, 3002903]    0.0             2     3002803\n",
      "29  (3002903, 3003003]    1.0             5     3002903\n",
      "30  (3003003, 3003103]    2.0             5     3003003\n",
      "31  (3003103, 3003203]    0.0             0     3003103\n",
      "32  (3003203, 3003303]    0.0             0     3003203\n",
      "33  (3003303, 3003403]    1.0             9     3003303\n",
      "34  (3003403, 3003503]    0.0             2     3003403\n",
      "35  (3003503, 3003603]    1.0             3     3003503\n",
      "36  (3003603, 3003703]    1.0             3     3003603\n",
      "37  (3003703, 3003803]    3.0             6     3003703\n",
      "38  (3003803, 3003903]    0.0             3     3003803\n",
      "39  (3003903, 3004003]    8.0             7     3003903\n",
      "40  (3004003, 3004103]    0.0             0     3004003\n",
      "41  (3004103, 3004203]    0.0             1     3004103\n",
      "42  (3004203, 3004303]    5.0             6     3004203\n",
      "43  (3004303, 3004403]    0.0             1     3004303\n",
      "44  (3004403, 3004503]    0.0             1     3004403\n",
      "45  (3004503, 3004603]    1.0             7     3004503\n",
      "46  (3004603, 3004703]    0.0             0     3004603\n",
      "47  (3004703, 3004803]    0.0             0     3004703\n",
      "48  (3004803, 3004903]   10.0             7     3004803\n",
      "49  (3004903, 3005003]    3.0             2     3004903\n",
      "(152618, 4)\n",
      "bin_offset_85    category\n",
      "TT_S0             float64\n",
      "target_count        int64\n",
      "starting_nt      category\n",
      "dtype: object\n",
      "sumos ilgis: 152618\n",
      "ranges ilgis: 152618\n",
      "Model input - NO SCALE\n",
      "        TT_S0  genes  transcripts  exons  CDSs\n",
      "0         7.0      0            0      0     0\n",
      "1         1.0      0            0      0     0\n",
      "2         0.0      0            0      0     0\n",
      "3         2.0      0            0      0     0\n",
      "4         0.0      0            0      0     0\n",
      "...       ...    ...          ...    ...   ...\n",
      "152613    3.0      0            0      0     0\n",
      "152614    4.0      0            0      0     0\n",
      "152615    0.0      0            0      0     0\n",
      "152616    4.0      0            0      0     0\n",
      "152617    0.0      0            0      0     0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "        TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0         7.0             0                   0             0            0\n",
      "1         1.0             0                   0             0            0\n",
      "2         0.0             0                   0             0            0\n",
      "3         2.0             0                   0             0            0\n",
      "4         0.0             0                   0             0            0\n",
      "...       ...           ...                 ...           ...          ...\n",
      "152613    3.0             0                   0             0            0\n",
      "152614    4.0             0                   0             0            0\n",
      "152615    0.0             0                   0             0            0\n",
      "152616    4.0             0                   0             0            0\n",
      "152617    0.0             0                   0             0            0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "0.044287043468005084\n",
      "[3 0 0 6 0 0 0 0 0 0 6 0 0 0 0 6 6 0 0 0 0 0 1 6 7 6 0 0 0 0 6 0 0 0 0 0 0\n",
      " 6 0 3 0 0 9 0 0 0 0 0 1 6 0 8 8 3 7 7 4 4 4 5 0 1 2 9 0 0 3 0 0 6 9 0 6 0\n",
      " 0 0 6 0 0 6 3 6 0 6 6 6 6 9 0 0 1 7 6 0 6 6 0 6 1 6 2 2 2 7 9 3 1 2 6 1 0\n",
      " 7 6 0 0 0 0 0 0 0 0 6 6 0 1 6 7 1 1 1 3 9 3 9 0 6 6 3 0 0 0 0 0 6 0 0 6 7\n",
      " 0 3 6 6 0 9 0 1 6 6 0 7 0 3 9 0 1 1 6 0 0 0 0 0 0 0 6 0 0 0 0 0 1 6 6 0 0\n",
      " 6 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 2 0 0 0 0 6 0 0 0 0 6 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 6 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 6 0 0 0\n",
      " 0 0 0 0 0 6 0 0 6 0 6 0 6 6 6 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 6\n",
      " 6 0 0 0 6 6 0 6 0 0 0 0 6 0 9 3 6 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 9 0 6 0 3 6 0 6 0 0 0 0 0 0 1 1 6 1 0 2 9 6 0 0 9 3 6\n",
      " 0 9 0 0 0 9 1 7 6 3 3 0 0 6 0 6 0 9 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 6\n",
      " 0 0 0 9 0 0 0 0 0 6 0 0 0 0 0 0 0 6 3 0 0 0 6 6 1 0 6 0 3 9 3 6 6 6 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 1 3 0 0 0 9 0 0 3 0 6 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 6 0 6 6 0 0 0 0 0 0 0 0 0 0 0 0 0 6 0 0 0 3 0 0 0 0 0 3 0 0 0 0 0 9\n",
      " 0 0 6 0 0 0 0 0 0 1 0 0 6 3 6 0 0 0 0 0 9 0 6 0 0 0 0 0 6 9 0 0 9 6 0 3 0\n",
      " 3 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 6 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 6 0 1 6 6 6 0 0 6 0 0 0 0 0 0 0 6 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 6 0 0 0 0 6 6 0 0 0 0\n",
      " 0 0 0 0 0 0 6 0 0 0 0 0 0 6 0 0 0 0 0 6 0 0 6 0 0 0 0 1 9 1 0 6 0 6 6 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 6 0 0 0 6 0 0 6 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 6 0 0 0 0 6 0 0 3 0 0 0 0 0 6 6 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 6 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 6 0 0 0 0 0 3 0 0 0 0 0 0 0 0 0 9\n",
      " 0 9 3 0 6 0 0 0 0 0 0 0 0 0 6 0 0 0 6 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 6 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 6 0 9 6 0 0 0 0 0 0 0\n",
      " 0]\n",
      "threshold density 1.95819\n",
      "cluster signal 79398.0\n",
      "cluster target count 66757\n",
      "!!cluster density 1.1893584193417919\n",
      "cluster signal 16260.0\n",
      "cluster target count 300802\n",
      "!!cluster density 0.05405549165231614\n",
      "cluster signal 86946.0\n",
      "cluster target count 156981\n",
      "!!cluster density 0.5538632063752938\n",
      "cluster signal 76258.0\n",
      "cluster target count 49644\n",
      "!!cluster density 1.5360970107163001\n",
      "cluster signal 62893.0\n",
      "cluster target count 32798\n",
      "!!cluster density 1.917586438197451\n",
      "cluster signal 61118.0\n",
      "cluster target count 66340\n",
      "!!cluster density 0.9212842930358758\n",
      "cluster signal 20678.0\n",
      "cluster target count 6432\n",
      "!!cluster density 3.214863184079602\n",
      "cluster signal 6344.0\n",
      "cluster target count 1337\n",
      "!!cluster density 4.744951383694839\n",
      "cluster signal 492.0\n",
      "cluster target count 7\n",
      "!!cluster density 70.28571428571429\n",
      "cluster signal 46524.0\n",
      "cluster target count 18902\n",
      "!!cluster density 2.4613268437202414\n",
      "\n",
      "Labels mapping\n",
      " {3: 0, 0: 0, 6: 0, 1: 0, 7: 0, 9: 0, 8: 1, 4: 1, 5: 1, 2: 1}\n",
      "DEBUG: 150010 2608 \n",
      "\n",
      " +----------------------+-------------+----------------+-------+\n",
      "| bin_offset_85        | TT_S0       | target_count   | +9    |\n",
      "| (category)           | (float64)   | (int64)        | ...   |\n",
      "|----------------------+-------------+----------------+-------|\n",
      "| (3000018, 3000103]   | 7.0         | 11             | ...   |\n",
      "| (3000103, 3000203]   | 1.0         | 4              | ...   |\n",
      "| (3000203, 3000303]   | 0.0         | 1              | ...   |\n",
      "| (3000303, 3000403]   | 2.0         | 7              | ...   |\n",
      "| ...                  | ...         | ...            | ...   |\n",
      "| (18261403, 18261503] | 4.0         | 4              | ...   |\n",
      "| (18261503, 18261603] | 0.0         | 1              | ...   |\n",
      "| (18261603, 18261703] | 4.0         | 4              | ...   |\n",
      "| (18261703, 18261710] | 0.0         | 1              | ...   |\n",
      "+----------------------+-------------+----------------+-------+\n",
      "Unstranded PyRanges object has 150,010 rows and 12 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "9 hidden columns: starting_nt, Chromosome, Start, End, genes, transcripts, ... (+ 3 more.)\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0\n",
      "BINS  IntervalIndex([  (3000018, 3000108],   (3000108, 3000208],\n",
      "                 (3000208, 3000308],   (3000308, 3000408],\n",
      "                 (3000408, 3000508],   (3000508, 3000608],\n",
      "                 (3000608, 3000708],   (3000708, 3000808],\n",
      "                 (3000808, 3000908],   (3000908, 3001008],\n",
      "               ...\n",
      "               (18260808, 18260908], (18260908, 18261008],\n",
      "               (18261008, 18261108], (18261108, 18261208],\n",
      "               (18261208, 18261308], (18261308, 18261408],\n",
      "               (18261408, 18261508], (18261508, 18261608],\n",
      "               (18261608, 18261708], (18261708, 18261710]],\n",
      "              dtype='interval[int64, right]', length=152618)\n",
      "HEAD with BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0   \n",
      "\n",
      "        bin_offset_90  target_count  \n",
      "0  (3000018, 3000108]             1  \n",
      "1  (3000018, 3000108]             1  \n",
      "2  (3000018, 3000108]             1  \n",
      "3  (3000018, 3000108]             1  \n",
      "4  (3000018, 3000108]             1  \n",
      "TAIL with BINS\n",
      "       seqnames     start       end        CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  18261611  18261611  18_18261611    0.0    0.0    0.0   \n",
      "699996       18  18261615  18261615  18_18261615    0.0    0.0    0.0   \n",
      "699997       18  18261682  18261682  18_18261682    0.0    0.0    0.0   \n",
      "699998       18  18261702  18261702  18_18261702    4.0    0.0    0.0   \n",
      "699999       18  18261710  18261710  18_18261710    0.0    0.0    0.0   \n",
      "\n",
      "               bin_offset_90  target_count  \n",
      "699995  (18261608, 18261708]             1  \n",
      "699996  (18261608, 18261708]             1  \n",
      "699997  (18261608, 18261708]             1  \n",
      "699998  (18261608, 18261708]             1  \n",
      "699999  (18261708, 18261710]             1  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:137: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "         bin_offset_90  TT_S0  target_count starting_nt\n",
      "0   (3000018, 3000108]    7.0            11     3000018\n",
      "1   (3000108, 3000208]    1.0             4     3000108\n",
      "2   (3000208, 3000308]    0.0             1     3000208\n",
      "3   (3000308, 3000408]    2.0             7     3000308\n",
      "4   (3000408, 3000508]    0.0             1     3000408\n",
      "5   (3000508, 3000608]    0.0             2     3000508\n",
      "6   (3000608, 3000708]    0.0             5     3000608\n",
      "7   (3000708, 3000808]    0.0             6     3000708\n",
      "8   (3000808, 3000908]    0.0             4     3000808\n",
      "9   (3000908, 3001008]    0.0             3     3000908\n",
      "10  (3001008, 3001108]    2.0             6     3001008\n",
      "11  (3001108, 3001208]    0.0             2     3001108\n",
      "12  (3001208, 3001308]    0.0             3     3001208\n",
      "13  (3001308, 3001408]    0.0             4     3001308\n",
      "14  (3001408, 3001508]    1.0             4     3001408\n",
      "15  (3001508, 3001608]    3.0             9     3001508\n",
      "16  (3001608, 3001708]    3.0             6     3001608\n",
      "17  (3001708, 3001808]    0.0             8     3001708\n",
      "18  (3001808, 3001908]    0.0             2     3001808\n",
      "19  (3001908, 3002008]    1.0             3     3001908\n",
      "20  (3002008, 3002108]    0.0             2     3002008\n",
      "21  (3002108, 3002208]    0.0             2     3002108\n",
      "22  (3002208, 3002308]   11.0             7     3002208\n",
      "23  (3002308, 3002408]    2.0             2     3002308\n",
      "24  (3002408, 3002508]   19.0             5     3002408\n",
      "25  (3002508, 3002608]    4.0             2     3002508\n",
      "26  (3002608, 3002708]    0.0             3     3002608\n",
      "27  (3002708, 3002808]    0.0             1     3002708\n",
      "28  (3002808, 3002908]    0.0             2     3002808\n",
      "29  (3002908, 3003008]    1.0             5     3002908\n",
      "30  (3003008, 3003108]    2.0             5     3003008\n",
      "31  (3003108, 3003208]    0.0             0     3003108\n",
      "32  (3003208, 3003308]    0.0             1     3003208\n",
      "33  (3003308, 3003408]    1.0             8     3003308\n",
      "34  (3003408, 3003508]    0.0             3     3003408\n",
      "35  (3003508, 3003608]    1.0             2     3003508\n",
      "36  (3003608, 3003708]    1.0             3     3003608\n",
      "37  (3003708, 3003808]    3.0             6     3003708\n",
      "38  (3003808, 3003908]    0.0             3     3003808\n",
      "39  (3003908, 3004008]    8.0             7     3003908\n",
      "40  (3004008, 3004108]    0.0             0     3004008\n",
      "41  (3004108, 3004208]    0.0             1     3004108\n",
      "42  (3004208, 3004308]    5.0             6     3004208\n",
      "43  (3004308, 3004408]    0.0             1     3004308\n",
      "44  (3004408, 3004508]    0.0             1     3004408\n",
      "45  (3004508, 3004608]    1.0             7     3004508\n",
      "46  (3004608, 3004708]    0.0             0     3004608\n",
      "47  (3004708, 3004808]    0.0             1     3004708\n",
      "48  (3004808, 3004908]   10.0             6     3004808\n",
      "49  (3004908, 3005008]    3.0             2     3004908\n",
      "(152618, 4)\n",
      "bin_offset_90    category\n",
      "TT_S0             float64\n",
      "target_count        int64\n",
      "starting_nt      category\n",
      "dtype: object\n",
      "sumos ilgis: 152618\n",
      "ranges ilgis: 152618\n",
      "Model input - NO SCALE\n",
      "        TT_S0  genes  transcripts  exons  CDSs\n",
      "0         7.0      0            0      0     0\n",
      "1         1.0      0            0      0     0\n",
      "2         0.0      0            0      0     0\n",
      "3         2.0      0            0      0     0\n",
      "4         0.0      0            0      0     0\n",
      "...       ...    ...          ...    ...   ...\n",
      "152613    3.0      0            0      0     0\n",
      "152614    4.0      0            0      0     0\n",
      "152615    0.0      0            0      0     0\n",
      "152616    4.0      0            0      0     0\n",
      "152617    0.0      0            0      0     0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "        TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0         7.0             0                   0             0            0\n",
      "1         1.0             0                   0             0            0\n",
      "2         0.0             0                   0             0            0\n",
      "3         2.0             0                   0             0            0\n",
      "4         0.0             0                   0             0            0\n",
      "...       ...           ...                 ...           ...          ...\n",
      "152613    3.0             0                   0             0            0\n",
      "152614    4.0             0                   0             0            0\n",
      "152615    0.0             0                   0             0            0\n",
      "152616    4.0             0                   0             0            0\n",
      "152617    0.0             0                   0             0            0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "0.5721867669606469\n",
      "[0 1 1 5 1 1 1 1 1 1 5 1 1 1 1 5 5 1 1 1 1 1 8 5 9 5 1 1 1 1 5 1 1 1 1 1 1\n",
      " 5 1 8 1 1 0 1 1 1 1 1 8 5 1 7 2 0 9 3 6 6 6 4 1 8 9 5 1 1 0 1 1 5 0 1 5 1\n",
      " 1 5 1 1 1 5 8 5 1 5 5 5 5 0 1 1 8 3 5 1 5 5 1 5 8 8 3 9 9 3 8 0 8 9 5 8 1\n",
      " 3 5 1 1 5 1 1 1 1 1 0 1 1 3 5 3 3 8 3 8 0 8 5 1 5 5 8 1 1 1 1 1 5 1 1 5 3\n",
      " 1 8 5 5 1 0 1 3 5 5 1 3 1 8 5 1 8 8 5 1 1 1 1 1 1 5 1 1 1 1 1 1 8 5 5 1 1\n",
      " 5 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 8 9 1 1 1 1 5 1 1 1 1 5 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 5 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 5 1 1 1\n",
      " 1 1 1 1 1 5 1 1 5 1 5 1 5 5 5 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 5\n",
      " 5 1 1 1 5 5 1 5 1 1 1 1 5 1 0 0 5 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 0 1 5 1 8 5 1 1 1 1 1 1 1 1 8 3 5 8 1 9 0 5 1 5 0 8 5\n",
      " 1 0 1 1 1 0 3 3 5 8 0 1 5 1 1 5 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 5\n",
      " 1 1 1 0 1 1 1 1 1 5 1 1 1 1 1 1 1 5 8 1 1 1 0 1 8 1 5 0 5 5 8 5 5 5 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 8 8 1 1 1 0 1 1 8 1 5 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 5 1 5 5 1 1 1 1 1 1 1 1 1 1 1 1 1 5 1 1 1 0 1 1 1 1 1 8 1 1 1 1 1 0\n",
      " 1 1 5 1 1 1 1 1 1 8 1 1 5 0 5 1 1 1 1 1 0 1 5 1 1 1 1 1 5 0 1 1 0 5 1 0 1\n",
      " 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 5 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 5 5 8 5 5 5 1 1 5 1 1 1 1 1 1 1 5 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 5 1 1 1 1 5 5 1 1 1 1\n",
      " 1 1 1 1 1 1 5 1 1 1 1 1 1 5 1 1 1 1 1 5 1 1 5 1 1 1 1 8 0 8 1 5 1 5 5 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 5 1 1 1 5 1 1 5 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 5 1 1 1 1 5 1 1 0 1 1 1 1 1 5 5 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 5 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 5 1 1 1 1 1 0 1 1 1 1 1 1 1 1 1 0\n",
      " 1 0 0 1 5 1 1 1 1 1 1 1 1 1 5 1 1 1 5 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 5 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 5 1 0 5 1 1 1 1 1 1 1\n",
      " 1]\n",
      "threshold density 1.95819\n",
      "cluster signal 89902.0\n",
      "cluster target count 92790\n",
      "!!cluster density 0.9688759564608255\n",
      "cluster signal 16275.0\n",
      "cluster target count 301023\n",
      "!!cluster density 0.05406563618062407\n",
      "cluster signal 87250.0\n",
      "cluster target count 157031\n",
      "!!cluster density 0.555622775120836\n",
      "cluster signal 109456.0\n",
      "cluster target count 79484\n",
      "!!cluster density 1.377082180061396\n",
      "cluster signal 47433.0\n",
      "cluster target count 20121\n",
      "!!cluster density 2.357387803787088\n",
      "cluster signal 8978.0\n",
      "cluster target count 2310\n",
      "!!cluster density 3.8865800865800866\n",
      "cluster signal 23135.0\n",
      "cluster target count 7639\n",
      "!!cluster density 3.0285377667233933\n",
      "cluster signal 72309.0\n",
      "cluster target count 39373\n",
      "!!cluster density 1.8365123307850557\n",
      "cluster signal 1681.0\n",
      "cluster target count 222\n",
      "!!cluster density 7.572072072072072\n",
      "cluster signal 492.0\n",
      "cluster target count 7\n",
      "!!cluster density 70.28571428571429\n",
      "\n",
      "Labels mapping\n",
      " {0: 0, 1: 0, 5: 0, 8: 0, 9: 1, 7: 1, 2: 1, 3: 0, 6: 1, 4: 1}\n",
      "DEBUG: 149596 3022 \n",
      "\n",
      " +----------------------+-------------+----------------+-------+\n",
      "| bin_offset_90        | TT_S0       | target_count   | +9    |\n",
      "| (category)           | (float64)   | (int64)        | ...   |\n",
      "|----------------------+-------------+----------------+-------|\n",
      "| (3000018, 3000108]   | 7.0         | 11             | ...   |\n",
      "| (3000108, 3000208]   | 1.0         | 4              | ...   |\n",
      "| (3000208, 3000308]   | 0.0         | 1              | ...   |\n",
      "| (3000308, 3000408]   | 2.0         | 7              | ...   |\n",
      "| ...                  | ...         | ...            | ...   |\n",
      "| (18261408, 18261508] | 4.0         | 4              | ...   |\n",
      "| (18261508, 18261608] | 0.0         | 1              | ...   |\n",
      "| (18261608, 18261708] | 4.0         | 4              | ...   |\n",
      "| (18261708, 18261710] | 0.0         | 1              | ...   |\n",
      "+----------------------+-------------+----------------+-------+\n",
      "Unstranded PyRanges object has 149,596 rows and 12 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "9 hidden columns: starting_nt, Chromosome, Start, End, genes, transcripts, ... (+ 3 more.)\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0\n",
      "BINS  IntervalIndex([  (3000018, 3000113],   (3000113, 3000213],\n",
      "                 (3000213, 3000313],   (3000313, 3000413],\n",
      "                 (3000413, 3000513],   (3000513, 3000613],\n",
      "                 (3000613, 3000713],   (3000713, 3000813],\n",
      "                 (3000813, 3000913],   (3000913, 3001013],\n",
      "               ...\n",
      "               (18260713, 18260813], (18260813, 18260913],\n",
      "               (18260913, 18261013], (18261013, 18261113],\n",
      "               (18261113, 18261213], (18261213, 18261313],\n",
      "               (18261313, 18261413], (18261413, 18261513],\n",
      "               (18261513, 18261613], (18261613, 18261710]],\n",
      "              dtype='interval[int64, right]', length=152617)\n",
      "HEAD with BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0   \n",
      "\n",
      "        bin_offset_95  target_count  \n",
      "0  (3000018, 3000113]             1  \n",
      "1  (3000018, 3000113]             1  \n",
      "2  (3000018, 3000113]             1  \n",
      "3  (3000018, 3000113]             1  \n",
      "4  (3000018, 3000113]             1  \n",
      "TAIL with BINS\n",
      "       seqnames     start       end        CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  18261611  18261611  18_18261611    0.0    0.0    0.0   \n",
      "699996       18  18261615  18261615  18_18261615    0.0    0.0    0.0   \n",
      "699997       18  18261682  18261682  18_18261682    0.0    0.0    0.0   \n",
      "699998       18  18261702  18261702  18_18261702    4.0    0.0    0.0   \n",
      "699999       18  18261710  18261710  18_18261710    0.0    0.0    0.0   \n",
      "\n",
      "               bin_offset_95  target_count  \n",
      "699995  (18261513, 18261613]             1  \n",
      "699996  (18261613, 18261710]             1  \n",
      "699997  (18261613, 18261710]             1  \n",
      "699998  (18261613, 18261710]             1  \n",
      "699999  (18261613, 18261710]             1  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:137: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "         bin_offset_95  TT_S0  target_count starting_nt\n",
      "0   (3000018, 3000113]    7.0            11     3000018\n",
      "1   (3000113, 3000213]    1.0             4     3000113\n",
      "2   (3000213, 3000313]    2.0             2     3000213\n",
      "3   (3000313, 3000413]    0.0             6     3000313\n",
      "4   (3000413, 3000513]    0.0             1     3000413\n",
      "5   (3000513, 3000613]    0.0             3     3000513\n",
      "6   (3000613, 3000713]    0.0             4     3000613\n",
      "7   (3000713, 3000813]    0.0             6     3000713\n",
      "8   (3000813, 3000913]    0.0             4     3000813\n",
      "9   (3000913, 3001013]    0.0             3     3000913\n",
      "10  (3001013, 3001113]    2.0             6     3001013\n",
      "11  (3001113, 3001213]    0.0             2     3001113\n",
      "12  (3001213, 3001313]    0.0             3     3001213\n",
      "13  (3001313, 3001413]    0.0             4     3001313\n",
      "14  (3001413, 3001513]    1.0             4     3001413\n",
      "15  (3001513, 3001613]    3.0            10     3001513\n",
      "16  (3001613, 3001713]    3.0             6     3001613\n",
      "17  (3001713, 3001813]    0.0             7     3001713\n",
      "18  (3001813, 3001913]    0.0             2     3001813\n",
      "19  (3001913, 3002013]    1.0             3     3001913\n",
      "20  (3002013, 3002113]    0.0             2     3002013\n",
      "21  (3002113, 3002213]    1.0             3     3002113\n",
      "22  (3002213, 3002313]   10.0             7     3002213\n",
      "23  (3002313, 3002413]    2.0             1     3002313\n",
      "24  (3002413, 3002513]   19.0             5     3002413\n",
      "25  (3002513, 3002613]    4.0             2     3002513\n",
      "26  (3002613, 3002713]    0.0             3     3002613\n",
      "27  (3002713, 3002813]    0.0             1     3002713\n",
      "28  (3002813, 3002913]    0.0             2     3002813\n",
      "29  (3002913, 3003013]    1.0             6     3002913\n",
      "30  (3003013, 3003113]    2.0             4     3003013\n",
      "31  (3003113, 3003213]    0.0             0     3003113\n",
      "32  (3003213, 3003313]    0.0             1     3003213\n",
      "33  (3003313, 3003413]    1.0             8     3003313\n",
      "34  (3003413, 3003513]    0.0             3     3003413\n",
      "35  (3003513, 3003613]    1.0             2     3003513\n",
      "36  (3003613, 3003713]    1.0             3     3003613\n",
      "37  (3003713, 3003813]    3.0             6     3003713\n",
      "38  (3003813, 3003913]    0.0             4     3003813\n",
      "39  (3003913, 3004013]    8.0             6     3003913\n",
      "40  (3004013, 3004113]    0.0             0     3004013\n",
      "41  (3004113, 3004213]    0.0             1     3004113\n",
      "42  (3004213, 3004313]    5.0             6     3004213\n",
      "43  (3004313, 3004413]    0.0             1     3004313\n",
      "44  (3004413, 3004513]    0.0             1     3004413\n",
      "45  (3004513, 3004613]    1.0             7     3004513\n",
      "46  (3004613, 3004713]    0.0             0     3004613\n",
      "47  (3004713, 3004813]    0.0             1     3004713\n",
      "48  (3004813, 3004913]   10.0             7     3004813\n",
      "49  (3004913, 3005013]    3.0             1     3004913\n",
      "(152617, 4)\n",
      "bin_offset_95    category\n",
      "TT_S0             float64\n",
      "target_count        int64\n",
      "starting_nt      category\n",
      "dtype: object\n",
      "sumos ilgis: 152617\n",
      "ranges ilgis: 152617\n",
      "Model input - NO SCALE\n",
      "        TT_S0  genes  transcripts  exons  CDSs\n",
      "0         7.0      0            0      0     0\n",
      "1         1.0      0            0      0     0\n",
      "2         2.0      0            0      0     0\n",
      "3         0.0      0            0      0     0\n",
      "4         0.0      0            0      0     0\n",
      "...       ...    ...          ...    ...   ...\n",
      "152612   10.0      0            0      0     0\n",
      "152613    3.0      0            0      0     0\n",
      "152614    4.0      0            0      0     0\n",
      "152615    0.0      0            0      0     0\n",
      "152616    4.0      0            0      0     0\n",
      "\n",
      "[152617 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "        TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0         7.0             0                   0             0            0\n",
      "1         1.0             0                   0             0            0\n",
      "2         2.0             0                   0             0            0\n",
      "3         0.0             0                   0             0            0\n",
      "4         0.0             0                   0             0            0\n",
      "...       ...           ...                 ...           ...          ...\n",
      "152612   10.0             0                   0             0            0\n",
      "152613    3.0             0                   0             0            0\n",
      "152614    4.0             0                   0             0            0\n",
      "152615    0.0             0                   0             0            0\n",
      "152616    4.0             0                   0             0            0\n",
      "\n",
      "[152617 rows x 5 columns]\n",
      "0.01878558745093928\n",
      "[3 9 0 5 5 5 5 5 5 5 0 5 5 5 9 0 0 5 5 9 5 9 3 0 1 8 5 5 5 9 0 5 5 9 5 9 9\n",
      " 0 5 3 5 5 8 5 5 9 5 5 3 0 8 2 2 3 1 7 4 4 6 6 5 7 7 8 5 5 3 5 9 8 0 5 8 9\n",
      " 5 0 5 5 5 0 3 0 5 0 8 0 8 8 5 5 7 7 0 5 0 0 5 0 3 3 7 1 1 7 3 8 3 1 8 7 5\n",
      " 7 8 5 5 0 5 5 5 5 9 8 9 9 7 0 7 7 7 1 9 3 3 8 9 8 3 0 5 5 5 5 5 0 9 5 0 7\n",
      " 9 3 0 5 9 8 5 7 0 0 9 7 5 7 9 5 3 7 0 5 5 5 5 9 5 0 5 5 5 5 9 0 3 0 0 5 5\n",
      " 0 5 5 5 5 5 9 5 5 5 5 5 5 5 5 5 5 5 5 5 5 7 1 9 9 5 5 0 5 5 5 5 0 5 5 5 5\n",
      " 5 5 5 5 5 5 5 5 5 5 5 5 5 0 5 5 5 9 5 5 5 5 5 5 5 5 5 5 5 9 5 5 5 0 5 9 5\n",
      " 5 5 5 5 5 0 5 5 0 9 0 5 0 0 0 9 5 9 5 5 9 9 5 5 9 5 5 5 5 5 5 5 9 5 5 5 8\n",
      " 5 5 5 9 0 0 5 0 5 5 5 5 8 9 8 3 0 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5\n",
      " 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5\n",
      " 5 5 5 5 5 5 5 5 5 5 8 5 0 5 7 9 9 9 9 5 5 5 5 9 7 7 9 7 3 7 3 0 9 0 8 3 0\n",
      " 5 8 9 9 5 8 7 7 0 3 8 5 0 5 5 0 5 8 5 9 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 0\n",
      " 5 5 9 8 5 5 5 5 5 0 5 5 9 5 9 9 9 0 3 9 5 9 8 5 3 5 8 8 8 0 3 0 0 0 9 5 5\n",
      " 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5\n",
      " 7 3 5 5 5 8 5 9 3 5 0 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5\n",
      " 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5\n",
      " 5 5 5 0 5 0 8 5 5 5 5 5 9 5 5 5 5 9 5 9 0 9 9 5 3 5 5 5 5 5 3 5 5 5 5 8 5\n",
      " 5 5 8 5 5 5 5 9 0 3 5 5 8 3 0 5 5 5 5 9 8 9 0 5 5 5 5 9 8 8 5 9 8 8 5 3 5\n",
      " 3 5 5 5 5 9 9 5 5 5 5 5 5 5 5 5 5 8 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5\n",
      " 5 5 5 5 5 5 5 5 5 5 5 5 0 0 3 8 0 0 5 5 0 9 5 9 5 5 5 5 0 5 5 5 5 9 5 5 5\n",
      " 5 9 5 5 9 5 5 9 5 5 5 5 5 5 9 5 5 5 5 5 9 5 5 5 5 5 0 5 5 5 5 0 0 5 5 5 5\n",
      " 5 5 5 5 5 5 0 5 5 5 5 5 9 8 5 5 5 9 5 0 5 5 8 5 5 5 9 3 8 3 5 0 5 0 0 5 5\n",
      " 5 5 5 5 9 5 5 5 5 5 5 5 5 5 5 5 5 9 9 9 9 9 5 9 0 5 9 5 0 5 9 0 5 5 5 9 5\n",
      " 5 5 5 5 5 5 5 5 5 5 0 5 5 9 5 0 5 9 3 5 5 5 5 5 0 8 5 5 5 5 5 5 5 5 5 5 5\n",
      " 5 0 5 5 5 5 5 9 5 5 5 5 5 5 5 5 5 5 5 9 0 5 5 9 9 9 3 9 5 5 5 9 5 5 5 9 8\n",
      " 0 8 8 0 0 5 5 5 5 5 5 5 5 5 0 5 5 5 0 9 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5\n",
      " 5 5 5 5 5 5 5 8 5 5 5 5 5 5 5 9 5 5 5 5 5 5 5 9 5 5 0 5 8 0 5 5 9 5 5 5 5\n",
      " 5]\n",
      "threshold density 1.95819\n",
      "cluster signal 98867.0\n",
      "cluster target count 80044\n",
      "!!cluster density 1.2351581630103443\n",
      "cluster signal 16302.0\n",
      "cluster target count 73095\n",
      "!!cluster density 0.22302483069977427\n",
      "cluster signal 55738.0\n",
      "cluster target count 113745\n",
      "!!cluster density 0.4900259352059431\n",
      "cluster signal 0.0\n",
      "cluster target count 227702\n",
      "!!cluster density 0.0\n",
      "cluster signal 62816.0\n",
      "cluster target count 26665\n",
      "!!cluster density 2.3557472342021377\n",
      "cluster signal 93349.0\n",
      "cluster target count 110988\n",
      "!!cluster density 0.8410729087829315\n",
      "cluster signal 21650.0\n",
      "cluster target count 6694\n",
      "!!cluster density 3.2342396175679715\n",
      "cluster signal 102399.0\n",
      "cluster target count 59989\n",
      "!!cluster density 1.7069629432062545\n",
      "cluster signal 5112.0\n",
      "cluster target count 1066\n",
      "!!cluster density 4.795497185741088\n",
      "cluster signal 678.0\n",
      "cluster target count 12\n",
      "!!cluster density 56.5\n",
      "\n",
      "Labels mapping\n",
      " {3: 0, 9: 0, 0: 0, 5: 0, 1: 1, 8: 0, 2: 1, 7: 0, 4: 1, 6: 1}\n",
      "DEBUG: 149116 3501 \n",
      "\n",
      " +----------------------+-------------+----------------+-------+\n",
      "| bin_offset_95        | TT_S0       | target_count   | +9    |\n",
      "| (category)           | (float64)   | (int64)        | ...   |\n",
      "|----------------------+-------------+----------------+-------|\n",
      "| (3000018, 3000113]   | 7.0         | 11             | ...   |\n",
      "| (3000113, 3000213]   | 1.0         | 4              | ...   |\n",
      "| (3000213, 3000313]   | 2.0         | 2              | ...   |\n",
      "| (3000313, 3000413]   | 0.0         | 6              | ...   |\n",
      "| ...                  | ...         | ...            | ...   |\n",
      "| (18261313, 18261413] | 3.0         | 15             | ...   |\n",
      "| (18261413, 18261513] | 4.0         | 4              | ...   |\n",
      "| (18261513, 18261613] | 0.0         | 2              | ...   |\n",
      "| (18261613, 18261710] | 4.0         | 4              | ...   |\n",
      "+----------------------+-------------+----------------+-------+\n",
      "Unstranded PyRanges object has 149,116 rows and 12 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "9 hidden columns: starting_nt, Chromosome, Start, End, genes, transcripts, ... (+ 3 more.)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/anaconda3/lib/python3.12/site-packages/pyranges/methods/intersection.py:95: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '[1 1 1 ... 1 1 1]' has dtype incompatible with int32, please explicitly cast to a compatible dtype first.\n",
      "  sx.loc[vc.index, 0] = vc.values\n",
      "/opt/anaconda3/lib/python3.12/site-packages/pyranges/methods/intersection.py:95: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '[1 1 1 ... 1 1 1]' has dtype incompatible with int32, please explicitly cast to a compatible dtype first.\n",
      "  sx.loc[vc.index, 0] = vc.values\n",
      "/opt/anaconda3/lib/python3.12/site-packages/pyranges/methods/intersection.py:95: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '[1 1 1 ... 1 1 1]' has dtype incompatible with int32, please explicitly cast to a compatible dtype first.\n",
      "  sx.loc[vc.index, 0] = vc.values\n",
      "/opt/anaconda3/lib/python3.12/site-packages/pyranges/methods/intersection.py:95: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '[1 1 1 ... 1 1 1]' has dtype incompatible with int32, please explicitly cast to a compatible dtype first.\n",
      "  sx.loc[vc.index, 0] = vc.values\n",
      "/opt/anaconda3/lib/python3.12/site-packages/pyranges/methods/intersection.py:95: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '[1 1 1 ... 1 1 1]' has dtype incompatible with int32, please explicitly cast to a compatible dtype first.\n",
      "  sx.loc[vc.index, 0] = vc.values\n",
      "/opt/anaconda3/lib/python3.12/site-packages/pyranges/methods/intersection.py:95: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '[1 1 1 ... 1 1 1]' has dtype incompatible with int32, please explicitly cast to a compatible dtype first.\n",
      "  sx.loc[vc.index, 0] = vc.values\n",
      "/opt/anaconda3/lib/python3.12/site-packages/pyranges/methods/intersection.py:95: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '[1 1 1 ... 1 1 1]' has dtype incompatible with int32, please explicitly cast to a compatible dtype first.\n",
      "  sx.loc[vc.index, 0] = vc.values\n",
      "/opt/anaconda3/lib/python3.12/site-packages/pyranges/methods/intersection.py:95: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '[1 1 1 ... 1 1 1]' has dtype incompatible with int32, please explicitly cast to a compatible dtype first.\n",
      "  sx.loc[vc.index, 0] = vc.values\n",
      "/opt/anaconda3/lib/python3.12/site-packages/pyranges/methods/intersection.py:95: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '[1 1 1 ... 1 1 1]' has dtype incompatible with int32, please explicitly cast to a compatible dtype first.\n",
      "  sx.loc[vc.index, 0] = vc.values\n",
      "/opt/anaconda3/lib/python3.12/site-packages/pyranges/methods/intersection.py:95: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '[1 1 1 ... 1 1 1]' has dtype incompatible with int32, please explicitly cast to a compatible dtype first.\n",
      "  sx.loc[vc.index, 0] = vc.values\n",
      "/opt/anaconda3/lib/python3.12/site-packages/pyranges/methods/intersection.py:95: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '[1 1 1 ... 1 1 1]' has dtype incompatible with int32, please explicitly cast to a compatible dtype first.\n",
      "  sx.loc[vc.index, 0] = vc.values\n",
      "/opt/anaconda3/lib/python3.12/site-packages/pyranges/methods/intersection.py:95: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '[1 1 1 ... 1 1 1]' has dtype incompatible with int32, please explicitly cast to a compatible dtype first.\n",
      "  sx.loc[vc.index, 0] = vc.values\n",
      "/opt/anaconda3/lib/python3.12/site-packages/pyranges/methods/intersection.py:95: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '[1 1 1 ... 1 1 1]' has dtype incompatible with int32, please explicitly cast to a compatible dtype first.\n",
      "  sx.loc[vc.index, 0] = vc.values\n",
      "/opt/anaconda3/lib/python3.12/site-packages/pyranges/methods/intersection.py:95: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '[1 1 1 ... 1 1 1]' has dtype incompatible with int32, please explicitly cast to a compatible dtype first.\n",
      "  sx.loc[vc.index, 0] = vc.values\n",
      "/opt/anaconda3/lib/python3.12/site-packages/pyranges/methods/intersection.py:95: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '[1 1 1 ... 1 1 1]' has dtype incompatible with int32, please explicitly cast to a compatible dtype first.\n",
      "  sx.loc[vc.index, 0] = vc.values\n",
      "/opt/anaconda3/lib/python3.12/site-packages/pyranges/methods/intersection.py:95: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '[1 1 1 ... 1 1 1]' has dtype incompatible with int32, please explicitly cast to a compatible dtype first.\n",
      "  sx.loc[vc.index, 0] = vc.values\n",
      "/opt/anaconda3/lib/python3.12/site-packages/pyranges/methods/intersection.py:95: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '[1 1 1 ... 1 1 1]' has dtype incompatible with int32, please explicitly cast to a compatible dtype first.\n",
      "  sx.loc[vc.index, 0] = vc.values\n",
      "/opt/anaconda3/lib/python3.12/site-packages/pyranges/methods/intersection.py:95: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '[1 1 1 ... 1 1 1]' has dtype incompatible with int32, please explicitly cast to a compatible dtype first.\n",
      "  sx.loc[vc.index, 0] = vc.values\n",
      "/opt/anaconda3/lib/python3.12/site-packages/pyranges/methods/intersection.py:95: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '[1 1 1 ... 1 1 1]' has dtype incompatible with int32, please explicitly cast to a compatible dtype first.\n",
      "  sx.loc[vc.index, 0] = vc.values\n",
      "/opt/anaconda3/lib/python3.12/site-packages/pyranges/methods/intersection.py:95: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '[1 1 1 ... 1 1 1]' has dtype incompatible with int32, please explicitly cast to a compatible dtype first.\n",
      "  sx.loc[vc.index, 0] = vc.values\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "OVERLAPS +--------------+-----------+-----------+------------+------------+-------+\n",
      "| Chromosome   | Start     | End       | offset_0   | offset_5   | +18   |\n",
      "| (object)     | (int64)   | (int64)   | (int64)    | (int64)    | ...   |\n",
      "|--------------+-----------+-----------+------------+------------+-------|\n",
      "| 18           | 3002358   | 3002363   | 0          | 0          | ...   |\n",
      "| 18           | 3002363   | 3002368   | 0          | 0          | ...   |\n",
      "| 18           | 3002368   | 3002378   | 0          | 0          | ...   |\n",
      "| 18           | 3002378   | 3002383   | 0          | 0          | ...   |\n",
      "| ...          | ...       | ...       | ...        | ...        | ...   |\n",
      "| 18           | 18244418  | 18244448  | 0          | 0          | ...   |\n",
      "| 18           | 18244448  | 18244453  | 0          | 0          | ...   |\n",
      "| 18           | 18244453  | 18244458  | 0          | 0          | ...   |\n",
      "| 18           | 18244458  | 18244478  | 0          | 0          | ...   |\n",
      "+--------------+-----------+-----------+------------+------------+-------+\n",
      "Unstranded PyRanges object has 130,403 rows and 23 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "18 hidden columns: offset_10, offset_15, offset_20, offset_25, offset_30, offset_35, ... (+ 12 more.)\n",
      "OVERLAPS FILTERED +--------------+-----------+-----------+\n",
      "| Chromosome   | Start     | End       |\n",
      "| (category)   | (int64)   | (int64)   |\n",
      "|--------------+-----------+-----------|\n",
      "| 18           | 3005068   | 3005313   |\n",
      "| 18           | 3005443   | 3005533   |\n",
      "| 18           | 3005593   | 3006003   |\n",
      "| 18           | 3006178   | 3006263   |\n",
      "| ...          | ...       | ...       |\n",
      "| 18           | 18100743  | 18100838  |\n",
      "| 18           | 18106703  | 18106763  |\n",
      "| 18           | 18146728  | 18146828  |\n",
      "| 18           | 18237513  | 18237603  |\n",
      "+--------------+-----------+-----------+\n",
      "Unstranded PyRanges object has 2,937 rows and 3 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n"
     ]
    }
   ],
   "source": [
    "out = generate_consensus.generate_consensus_5x20(chromatin_algos.algo5d_aggregated, [16, 10, 3])\n",
    "result = evaluator.evaluate(merged_ranges = out, model_name=\"test run\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "2c68c72c",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:95: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "sumos ilgis: 152618\n",
      "ranges ilgis: 152618\n",
      "0.050570705945563434\n",
      "[0 9 0 7 0 0 0 0 0 0 0 7 0 0 0 0 7 7 0 0 0 0 0 1 7 5 7 0 0 0 0 7 0 0 0 0 0\n",
      " 7 0 0 9 0 0 2 0 0 0 0 0 1 0 6 5 3 0 5 1 8 8 4 4 0 5 5 7 0 0 9 0 0 2 7 0 2\n",
      " 0 0 7 0 0 0 7 9 7 0 7 7 7 7 2 0 0 1 1 7 7 0 7 0 7 1 1 1 6 6 1 9 2 1 6 7 1\n",
      " 0 5 7 0 0 7 0 0 0 0 0 2 0 0 1 0 5 5 1 5 0 9 9 7 0 2 9 7 0 0 0 0 0 7 0 0 1\n",
      " 9 0 1 7 0 0 2 0 1 7 7 0 5 0 1 0 0 1 1 7 0 0 0 0 0 0 7 0 0 0 0 0 2 9 7 7 0\n",
      " 0 7 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 5 0 0 0 0 7 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 7 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 7 0 0\n",
      " 0 0 0 0 0 0 7 0 0 7 0 7 0 7 7 7 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 2 0 0 0 0 7 7 0 7 0 0 0 0 7 0 2 9 7 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 2 0 7 0 1 0 0 0 0 0 0 0 0 0 1 5 0 1 5 9 9 7 0 7 2 9\n",
      " 7 0 2 0 0 0 1 2 1 7 1 2 0 7 0 0 7 0 2 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 7 0 0 0 2 0 0 0 0 0 7 0 0 0 0 0 0 0 7 9 0 0 0 2 2 7 0 7 2 2 7 9 7 7 7 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 1 2 0 0 0 2 0 0 9 0 7 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 7 0 7 7 0 0 0 0 0 0 0 0 0 0 0 0 0 7 0 0 0 9 0 0 0 0 7 2 0 0 0 0 2\n",
      " 0 0 7 0 0 0 0 0 0 7 9 0 0 7 9 7 0 0 0 0 0 2 0 7 0 0 0 0 0 7 2 0 0 2 7 0 9\n",
      " 0 9 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 7 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 7 7 9 7 7 7 0 7 0 0 0 0 0 0 0 0 7 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 7 0 0 0 0 7 7 0 0 0\n",
      " 0 0 0 0 0 0 0 7 0 0 0 0 0 7 7 0 0 0 0 0 7 0 0 7 0 0 0 7 9 2 1 0 7 0 7 7 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 7 0 0 0 7 0 0 7 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 7 0 0 0 0 7 0 7 2 0 0 0 0 0 7 7 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 7 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 7 0 0 0 0 7 2 0 0 0 0 0 0 0 0 0\n",
      " 2 7 2 2 7 7 0 0 0 0 0 0 0 0 0 7 0 0 0 7 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 7 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 7 0 2 7 0 0 0 0 0 0\n",
      " 0]\n",
      "threshold density 1.95819\n",
      "cluster signal 16270.0\n",
      "cluster target count 300673\n",
      "!!cluster density 0.054111942209643035\n",
      "cluster signal 77249.0\n",
      "cluster target count 64504\n",
      "!!cluster density 1.1975846459134318\n",
      "cluster signal 87190.0\n",
      "cluster target count 157019\n",
      "!!cluster density 0.5552831186034811\n",
      "cluster signal 89658.0\n",
      "cluster target count 57682\n",
      "!!cluster density 1.554349710481606\n",
      "cluster signal 64543.0\n",
      "cluster target count 31785\n",
      "!!cluster density 2.0306119238634577\n",
      "cluster signal 62211.0\n",
      "cluster target count 67967\n",
      "!!cluster density 0.9153118425118072\n",
      "cluster signal 37536.0\n",
      "cluster target count 14421\n",
      "!!cluster density 2.602870813397129\n",
      "cluster signal 15940.0\n",
      "cluster target count 4740\n",
      "!!cluster density 3.3628691983122363\n",
      "cluster signal 5650.0\n",
      "cluster target count 1198\n",
      "!!cluster density 4.716193656093489\n",
      "cluster signal 664.0\n",
      "cluster target count 11\n",
      "!!cluster density 60.36363636363637\n",
      "\n",
      "Labels mapping\n",
      " {0: 0, 9: 0, 7: 0, 1: 0, 5: 1, 2: 0, 6: 1, 3: 1, 8: 1, 4: 1}\n",
      "DEBUG: 146987 5631 \n",
      "\n",
      " +----------------------+-------------+----------------+-------+\n",
      "| bin_offset_0         | TT_S0       | target_count   | +9    |\n",
      "| (category)           | (float64)   | (int64)        | ...   |\n",
      "|----------------------+-------------+----------------+-------|\n",
      "| (3000018, 3000018]   | 0.0         | 0              | ...   |\n",
      "| (3000018, 3000118]   | 7.0         | 11             | ...   |\n",
      "| (3000118, 3000218]   | 1.0         | 4              | ...   |\n",
      "| (3000218, 3000318]   | 2.0         | 4              | ...   |\n",
      "| ...                  | ...         | ...            | ...   |\n",
      "| (18261318, 18261418] | 3.0         | 15             | ...   |\n",
      "| (18261418, 18261518] | 4.0         | 4              | ...   |\n",
      "| (18261518, 18261618] | 0.0         | 3              | ...   |\n",
      "| (18261618, 18261710] | 4.0         | 3              | ...   |\n",
      "+----------------------+-------------+----------------+-------+\n",
      "Unstranded PyRanges object has 146,987 rows and 12 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "9 hidden columns: starting_nt, Chromosome, Start, End, genes, transcripts, ... (+ 3 more.)\n",
      "count                                                                4380.0\n",
      "mean                                                             128.561644\n",
      "std                                                               84.971333\n",
      "min                                                                   100.0\n",
      "25%                                                                   100.0\n",
      "50%                                                                   100.0\n",
      "75%                                                                   100.0\n",
      "max                                                                  1400.0\n",
      "mode                                                                  100.0\n",
      "ATACseq_overlap                                                    0.595016\n",
      "ATACseq_ranges_overlap                                              0.07773\n",
      "ATACseq_Jaccard                                                    0.073825\n",
      "DNAseq_overlap                                                     0.502716\n",
      "DNAseq_ranges_overlap                                              0.107979\n",
      "DNAseq_Jaccard                                                     0.097559\n",
      "genes_overlap                                                      0.586276\n",
      "percentage_open                                                    0.030835\n",
      "percentage_open_adj                                                0.036896\n",
      "model_name                                                         test run\n",
      "boxplot_json              {\"mean\":128.5616438356,\"iqr\":0.0,\"cilo\":100.0,...\n",
      "Name: Length, dtype: object\n"
     ]
    }
   ],
   "source": [
    "import chromatin_algos\n",
    "import evaluator\n",
    "\n",
    "out = chromatin_algos.algo5d_aggregated(16, 10, 3)\n",
    "result = evaluator.evaluate(merged_ranges = out, model_name=\"test run\")\n",
    "print(result)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "6f590d1a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "count                                                               17831.0\n",
      "mean                                                               131.5574\n",
      "std                                                               83.492421\n",
      "min                                                                   100.0\n",
      "25%                                                                   100.0\n",
      "50%                                                                   100.0\n",
      "75%                                                                   100.0\n",
      "max                                                                  1400.0\n",
      "mode                                                                  100.0\n",
      "ATACseq_overlap                                                    0.491533\n",
      "ATACseq_ranges_overlap                                             0.115278\n",
      "ATACseq_Jaccard                                                    0.102955\n",
      "DNAseq_overlap                                                     0.444919\n",
      "DNAseq_ranges_overlap                                              0.168985\n",
      "DNAseq_Jaccard                                                     0.139516\n",
      "genes_overlap                                                      0.673301\n",
      "percentage_open                                                     0.02602\n",
      "percentage_open_adj                                                0.026916\n",
      "model_name                                                         test run\n",
      "boxplot_json              {\"mean\":131.5574000336,\"iqr\":0.0,\"cilo\":100.0,...\n",
      "Name: Length, dtype: object\n"
     ]
    }
   ],
   "source": [
    "print(result)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "c6fefef7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "count                                                                3802.0\n",
      "mean                                                             127.748553\n",
      "std                                                               83.312327\n",
      "min                                                                   100.0\n",
      "25%                                                                   100.0\n",
      "50%                                                                   100.0\n",
      "75%                                                                   100.0\n",
      "max                                                                  1400.0\n",
      "mode                                                                  100.0\n",
      "ATACseq_overlap                                                     0.56708\n",
      "ATACseq_ranges_overlap                                             0.085886\n",
      "ATACseq_Jaccard                                                    0.080602\n",
      "DNAseq_overlap                                                     0.476994\n",
      "DNAseq_ranges_overlap                                              0.118781\n",
      "DNAseq_Jaccard                                                     0.105094\n",
      "genes_overlap                                                      0.640012\n",
      "percentage_open                                                    0.026597\n",
      "percentage_open_adj                                                0.031825\n",
      "model_name                                                         test run\n",
      "stats_json                {\"mean\":127.748553393,\"iqr\":0.0,\"cilo\":100.0,\"...\n",
      "Name: Length, dtype: object\n"
     ]
    }
   ],
   "source": [
    "print(result)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "0e8edc55",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1176\n"
     ]
    }
   ],
   "source": [
    "print(len(result['stats_json']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "50edf0b9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'whiskers': [<matplotlib.lines.Line2D at 0x1284b04d0>,\n",
       "  <matplotlib.lines.Line2D at 0x1284b0da0>],\n",
       " 'caps': [<matplotlib.lines.Line2D at 0x1284b10a0>,\n",
       "  <matplotlib.lines.Line2D at 0x1284b1340>],\n",
       " 'boxes': [<matplotlib.patches.PathPatch at 0x12c001400>],\n",
       " 'medians': [<matplotlib.lines.Line2D at 0x1284b18e0>],\n",
       " 'fliers': [],\n",
       " 'means': []}"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAjMAAAGiCAYAAAASgEe5AAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjkuMiwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy8hTgPZAAAACXBIWXMAAA9hAAAPYQGoP6dpAAAu00lEQVR4nO3df1RVdb7/8dcR5CAO0CDKj0TERkcmHIvD+APHSa0odGyZzpXGSi11YlmZopXkvYmuVqxplmZpmlP+yBGLVZOOd2JSmhI17V5F7ZrSZImBI8SCkh/qHPyxv390Pd97AlTwHA4feD7W2iv3Z38++7x3q717+dn77GOzLMsSAACAoTr5ugAAAIDrQZgBAABGI8wAAACjEWYAAIDRCDMAAMBohBkAAGA0wgwAADAaYQYAABiNMAMAAIxGmAEAAEYjzADwiZ07d2rs2LGKjo6WzWbTli1brjqmoKBADodDgYGB6tOnj1599VXvFwqgzSPMAPCJM2fOaODAgVqxYsU19S8uLtbo0aM1fPhwHTx4UM8884xmzZqlP//5z16uFEBbZ+OHJgH4ms1m0+bNmzVu3Lgm+zz99NPaunWrioqKXG3p6en69NNPtXfv3laoEkBb5e/rAlri0qVLOnXqlIKDg2Wz2XxdDtAhWZal2tpaRUdHq1Mn70/y7t27VykpKW5td911l9asWaPz58+rc+fODcY4nU45nU7X+qVLl/Ttt9+qW7duXDsAH/DWdcPIMHPq1CnFxMT4ugwAkkpLS9WzZ0+vf055ebkiIiLc2iIiInThwgVVVlYqKiqqwZjs7GwtWrTI67UBaB5PXzeMDDPBwcGSvv+XERIS4uNqgI6ppqZGMTExrvOxNfxwNuXyXfKmZlkyMzOVkZHhWq+urlavXr24dgA+4q3rhpFh5vKFKyQkhAsS4GOtdbsmMjJS5eXlbm0VFRXy9/dXt27dGh1jt9tlt9sbtHPtAHzL09cNvs0EwAhDhw5Vfn6+W9v27duVlJTU6PMyADoOwgwAn6irq9OhQ4d06NAhSd9/9frQoUMqKSmR9P0tosmTJ7v6p6en6+uvv1ZGRoaKioq0du1arVmzRvPmzfNF+QDaECNvMwEw3/79+zVy5EjX+uVnW6ZMmaL169errKzMFWwkKS4uTnl5eZozZ45eeeUVRUdH6+WXX9aECRNavXYAbYuR75mpqalRaGioqqurue8N+IiJ56GJNQPtibfOQW4zAQAAoxFmAACA0QgzAADAaIQZAABgNMIMAAAwGmEGAAAYjTADAACMRpgBAABG4w3A8IqzZ8/q888/b3L7uXPndOLECfXu3VtdunS54r769++voKAgT5cIAGgnCDPwis8//1wOh8Mj+yosLFRiYqJH9gUAaH8IM/CK/v37q7CwsMntRUVFeuCBB7Rx40bFx8dfdV8AADSFMAOvCAoKuqbZlPj4eGZdAADXhQeAAQCA0QgzAADAaIQZAABgNMIMAAAwGmEGAAAYjTADAACMRpgBAABGI8wAAACjEWYAAIDRCDMAAMBohBkAAGA0wgwAADBas8PMzp07NXbsWEVHR8tms2nLli1u2202W6PLH/7wB1efESNGNNh+3333XffBAACAjqfZYebMmTMaOHCgVqxY0ej2srIyt2Xt2rWy2WyaMGGCW78ZM2a49Vu9enXLjgAAAHRo/s0dkJqaqtTU1Ca3R0ZGuq3/5S9/0ciRI9WnTx+39qCgoAZ9AQAAmsurz8x88803eu+99zRt2rQG23JychQeHq6bb75Z8+bNU21tbZP7cTqdqqmpcVsAAACkFszMNMcbb7yh4OBgjR8/3q39/vvvV1xcnCIjI/XZZ58pMzNTn376qfLz8xvdT3Z2thYtWuTNUgEAgKG8GmbWrl2r+++/X4GBgW7tM2bMcP05ISFBffv2VVJSkg4cOKDExMQG+8nMzFRGRoZrvaamRjExMd4rHAAAGMNrYWbXrl36xz/+odzc3Kv2TUxMVOfOnXXs2LFGw4zdbpfdbvdGmQAAwHBee2ZmzZo1cjgcGjhw4FX7HjlyROfPn1dUVJS3ygEAAO1Us2dm6urq9OWXX7rWi4uLdejQIYWFhalXr16Svr8N9Pbbb2vJkiUNxn/11VfKycnR6NGjFR4erqNHj2ru3Lm69dZbNWzYsOs4FAAA0BE1O8zs379fI0eOdK1ffpZlypQpWr9+vSTprbfekmVZ+u1vf9tgfEBAgP7+97/rpZdeUl1dnWJiYjRmzBgtXLhQfn5+LTwMAADQUTU7zIwYMUKWZV2xz+9+9zv97ne/a3RbTEyMCgoKmvuxAAAAjeK3mQAAgNEIMwAAwGiEGQAAYDTCDAAAMBphBgAAGI0wAwAAjEaYAQAARiPMAAAAoxFmAACA0QgzAADAaIQZAABgNMIMAAAwGmEGAAAYjTADAACMRpgBAABGI8wAAACjEWYAAIDRCDMAAMBohBkAAGA0wgwAADAaYQYAABiNMAMAAIxGmAEAAEYjzAAAAKMRZgAAgNEIMwAAwGiEGQAAYDTCDAAAMBphBoDPrFy5UnFxcQoMDJTD4dCuXbuu2D8nJ0cDBw5UUFCQoqKi9NBDD6mqqqqVqgXQVhFmAPhEbm6uZs+erQULFujgwYMaPny4UlNTVVJS0mj/3bt3a/LkyZo2bZqOHDmit99+W/v27dP06dNbuXIAbQ1hBoBPLF26VNOmTdP06dMVHx+vZcuWKSYmRqtWrWq0/yeffKLevXtr1qxZiouL0y9/+Us98sgj2r9/fytXDqCtIcwAaHX19fUqLCxUSkqKW3tKSor27NnT6Jjk5GSdPHlSeXl5sixL33zzjd555x2NGTOmyc9xOp2qqalxWwC0P4QZAK2usrJSFy9eVEREhFt7RESEysvLGx2TnJysnJwcpaWlKSAgQJGRkbrhhhu0fPnyJj8nOztboaGhriUmJsajxwGgbSDMAPAZm83mtm5ZVoO2y44ePapZs2bp2WefVWFhod5//30VFxcrPT29yf1nZmaqurratZSWlnq0fgBtg7+vCwDQ8YSHh8vPz6/BLExFRUWD2ZrLsrOzNWzYMD355JOSpJ///Ofq2rWrhg8frueee05RUVENxtjtdtntds8fAIA2hZkZAK0uICBADodD+fn5bu35+flKTk5udMzZs2fVqZP7JcvPz0/S9zM6ADquZoeZnTt3auzYsYqOjpbNZtOWLVvctk+dOlU2m81tGTJkiFsfp9Opxx9/XOHh4eratavuuecenTx58roOBIBZMjIy9Prrr2vt2rUqKirSnDlzVFJS4rptlJmZqcmTJ7v6jx07Vu+++65WrVql48eP6+OPP9asWbM0aNAgRUdH++owALQBzb7NdObMGQ0cOFAPPfSQJkyY0Gifu+++W+vWrXOtBwQEuG2fPXu2/vM//1NvvfWWunXrprlz5+rXv/61CgsLXX/TAtC+paWlqaqqSosXL1ZZWZkSEhKUl5en2NhYSVJZWZnbO2emTp2q2tparVixQnPnztUNN9ygUaNG6fe//72vDgFAG2GzrmN+1mazafPmzRo3bpyrberUqTp9+nSDGZvLqqur1b17d/3pT39SWlqaJOnUqVOKiYlRXl6e7rrrrqt+bk1NjUJDQ1VdXa2QkJCWlg8fOnDggBwOhwoLC5WYmOjrctACJp6HJtYMtCfeOge98szMjh071KNHD/Xr108zZsxQRUWFa1thYaHOnz/v9n6J6OhoJSQkNPl+Cd4VAQAAmuLxMJOamqqcnBx9+OGHWrJkifbt26dRo0bJ6XRKksrLyxUQEKAf//jHbuOu9H4J3hUBAACa4vGvZl++dSRJCQkJSkpKUmxsrN577z2NHz++yXFXer9EZmamMjIyXOs1NTUEGgAAIKkVvpodFRWl2NhYHTt2TJIUGRmp+vp6fffdd279rvR+CbvdrpCQELcFAABAaoUwU1VVpdLSUtcLrRwOhzp37uz2fomysjJ99tlnTb5fAgAAoCnNvs1UV1enL7/80rVeXFysQ4cOKSwsTGFhYcrKytKECRMUFRWlEydO6JlnnlF4eLjuvfdeSVJoaKimTZumuXPnqlu3bgoLC9O8efM0YMAA3XHHHZ47MgAA0CE0O8zs379fI0eOdK1ffpZlypQpWrVqlQ4fPqwNGzbo9OnTioqK0siRI5Wbm6vg4GDXmBdffFH+/v6aOHGizp07p9tvv13r16/nHTMAAKDZmh1mRowYccVXh2/btu2q+wgMDNTy5cuv+Gu3AAAA14LfZgIAAEYjzAAAAKMRZgAAgNEIMwAAwGiEGQAAYDTCDAAAMBphBgAAGI0wAwAAjEaYAQAARiPMAAAAoxFmAACA0QgzAADAaIQZAABgNMIMAAAwGmEGAAAYjTADAACMRpgBAABGI8wAAACjEWYAAIDRCDMAAMBohBkAAGA0wgwAADAaYQYAABiNMAMAAIxGmAEAAEYjzAAAAKMRZgAAgNEIMwAAwGiEGQAAYDTCDAAAMBphBgAAGI0wAwAAjEaYAQAARiPMAAAAoxFmAACA0QgzAADAaM0OMzt37tTYsWMVHR0tm82mLVu2uLadP39eTz/9tAYMGKCuXbsqOjpakydP1qlTp9z2MWLECNlsNrflvvvuu+6DAQAAHU+zw8yZM2c0cOBArVixosG2s2fP6sCBA/qP//gPHThwQO+++66++OIL3XPPPQ36zpgxQ2VlZa5l9erVLTsCAADQofk3d0BqaqpSU1Mb3RYaGqr8/Hy3tuXLl2vQoEEqKSlRr169XO1BQUGKjIy8ps90Op1yOp2u9ZqamuaWDQAA2imvPzNTXV0tm82mG264wa09JydH4eHhuvnmmzVv3jzV1tY2uY/s7GyFhoa6lpiYGC9XDQAATNHsmZnm+Ne//qX58+dr0qRJCgkJcbXff//9iouLU2RkpD777DNlZmbq008/bTCrc1lmZqYyMjJc6zU1NQQaAAAgyYth5vz587rvvvt06dIlrVy50m3bjBkzXH9OSEhQ3759lZSUpAMHDigxMbHBvux2u+x2u7dKBQAABvPKbabz589r4sSJKi4uVn5+vtusTGMSExPVuXNnHTt2zBvlAACAdszjMzOXg8yxY8f00UcfqVu3blcdc+TIEZ0/f15RUVGeLgcAALRzzQ4zdXV1+vLLL13rxcXFOnTokMLCwhQdHa3f/OY3OnDggP7617/q4sWLKi8vlySFhYUpICBAX331lXJycjR69GiFh4fr6NGjmjt3rm699VYNGzbMc0cGAAA6hGaHmf3792vkyJGu9csP5k6ZMkVZWVnaunWrJOmWW25xG/fRRx9pxIgRCggI0N///ne99NJLqqurU0xMjMaMGaOFCxfKz8/vOg4FAAB0RM0OMyNGjJBlWU1uv9I2SYqJiVFBQUFzPxYAAKBR/DYTAAAwGmEGAAAYjTADAACMRpgBAABGI8wAAACjEWYAAIDRCDMAAMBohBkAAGA0wgwAn1m5cqXi4uIUGBgoh8OhXbt2XbG/0+nUggULFBsbK7vdrptuuklr165tpWoBtFUe/6FJALgWubm5mj17tlauXKlhw4Zp9erVSk1N1dGjR9WrV69Gx0ycOFHffPON1qxZo5/85CeqqKjQhQsXWrlyAG0NYQaATyxdulTTpk3T9OnTJUnLli3Ttm3btGrVKmVnZzfo//7776ugoEDHjx9XWFiYJKl3796tWTKANorbTABaXX19vQoLC5WSkuLWnpKSoj179jQ6ZuvWrUpKStILL7ygG2+8Uf369dO8efN07ty5Jj/H6XSqpqbGbQHQ/jAzA6DVVVZW6uLFi4qIiHBrj4iIUHl5eaNjjh8/rt27dyswMFCbN29WZWWlZs6cqW+//bbJ52ays7O1aNEij9cPoG1hZgaAz9hsNrd1y7IatF126dIl2Ww25eTkaNCgQRo9erSWLl2q9evXNzk7k5mZqerqatdSWlrq8WMA4HvMzABodeHh4fLz82swC1NRUdFgtuayqKgo3XjjjQoNDXW1xcfHy7IsnTx5Un379m0wxm63y263e7Z4AG0OMzMAWl1AQIAcDofy8/Pd2vPz85WcnNzomGHDhunUqVOqq6tztX3xxRfq1KmTevbs6dV6AbRthBkAPpGRkaHXX39da9euVVFRkebMmaOSkhKlp6dL+v4W0eTJk139J02apG7duumhhx7S0aNHtXPnTj355JN6+OGH1aVLF18dBoA2gNtMAHwiLS1NVVVVWrx4scrKypSQkKC8vDzFxsZKksrKylRSUuLq/6Mf/Uj5+fl6/PHHlZSUpG7dumnixIl67rnnfHUIANoIwgwAn5k5c6ZmzpzZ6Lb169c3aOvfv3+DW1MAQJhBi5SUlKiysrLF44uKitz+2VLh4eFNvi0WANAxEGbQbCUlJYqPj9fZs2eve18PPPDAdY0PCgpSUVERgQYAOjDCDJqtsrJSZ8+e1caXn1V839gW7ePcv5w6cbJcvXtGqktgy746W3Tsaz0wa7EqKysJMwDQgRFm0GLxfWOVOOCnLR4/7Bc/92A1AICOiq9mAwAAoxFmAACA0QgzAADAaIQZAABgNMIMAAAwGmEGAAAYjTADAACMRpgBAABGI8wAAACjEWYAAIDRCDMAAMBohBkAAGC0ZoeZnTt3auzYsYqOjpbNZtOWLVvctluWpaysLEVHR6tLly4aMWKEjhw54tbH6XTq8ccfV3h4uLp27ap77rlHJ0+evK4DAQAAHVOzw8yZM2c0cOBArVixotHtL7zwgpYuXaoVK1Zo3759ioyM1J133qna2lpXn9mzZ2vz5s166623tHv3btXV1enXv/61Ll682PIjAQAAHZJ/cwekpqYqNTW10W2WZWnZsmVasGCBxo8fL0l64403FBERoU2bNumRRx5RdXW11qxZoz/96U+64447JEkbN25UTEyMPvjgA911113XcTgAAKCj8egzM8XFxSovL1dKSoqrzW6367bbbtOePXskSYWFhTp//rxbn+joaCUkJLj6/JDT6VRNTY3bAgAAIHk4zJSXl0uSIiIi3NojIiJc28rLyxUQEKAf//jHTfb5oezsbIWGhrqWmJgYT5YNAAAM5pVvM9lsNrd1y7IatP3QlfpkZmaqurratZSWlnqsVgAAYDaPhpnIyEhJajDDUlFR4ZqtiYyMVH19vb777rsm+/yQ3W5XSEiI2wIAACB5OMzExcUpMjJS+fn5rrb6+noVFBQoOTlZkuRwONS5c2e3PmVlZfrss89cfQAAAK5Vs7/NVFdXpy+//NK1XlxcrEOHDiksLEy9evXS7Nmz9fzzz6tv377q27evnn/+eQUFBWnSpEmSpNDQUE2bNk1z585Vt27dFBYWpnnz5mnAgAGubzcBAABcq2aHmf3792vkyJGu9YyMDEnSlClTtH79ej311FM6d+6cZs6cqe+++06DBw/W9u3bFRwc7Brz4osvyt/fXxMnTtS5c+d0++23a/369fLz8/PAIQEAgI6k2WFmxIgRsiyrye02m01ZWVnKyspqsk9gYKCWL1+u5cuXN/fjAQAA3PDbTAAAwGiEGQAAYDTCDAAAMBphBgAAGI0wAwAAjEaYAQAARiPMAAAAoxFmAACA0QgzAADAaIQZAABgNMIMAAAwGmEGAAAYjTADAACMRpgBAABGI8wAAACjEWYAAIDRCDMAAMBohBkAAGA0wgwAADAaYQYAABjN39cFwEyRP7Kpy5lSqdLPZzV0OVOqyB/ZfPb5AIC2gTCDFnnEEaD4w0ulw76rIf5/6wAAdGyEGbTI6sJ6pT2eqfif9PZZDUVfntDqJYt0j88qAAC0BYQZtEh5naVzXWOk8J/4rIZzZRdVXmf57PMBAG0DDwADAACjEWYAAIDRCDMAAMBohBkAAGA0wgwAADAaYQYAABiNMAMAAIxGmAEAAEYjzAAAAKMRZgAAgNEIMwAAwGiEGQAAYDSPh5nevXvLZrM1WB599FFJ0tSpUxtsGzJkiKfLAGCAlStXKi4uToGBgXI4HNq1a9c1jfv444/l7++vW265xbsFAjCCx8PMvn37VFZW5lry8/MlSf/2b//m6nP33Xe79cnLy/N0GQDauNzcXM2ePVsLFizQwYMHNXz4cKWmpqqkpOSK46qrqzV58mTdfvvtrVQpgLbO42Gme/fuioyMdC1//etfddNNN+m2225z9bHb7W59wsLCPF0GgDZu6dKlmjZtmqZPn674+HgtW7ZMMTExWrVq1RXHPfLII5o0aZKGDh161c9wOp2qqalxWwC0P159Zqa+vl4bN27Uww8/LJvN5mrfsWOHevTooX79+mnGjBmqqKi44n64IAHtS319vQoLC5WSkuLWnpKSoj179jQ5bt26dfrqq6+0cOHCa/qc7OxshYaGupaYmJjrqhtA2+TVMLNlyxadPn1aU6dOdbWlpqYqJydHH374oZYsWaJ9+/Zp1KhRcjqdTe6HCxLQvlRWVurixYuKiIhwa4+IiFB5eXmjY44dO6b58+crJydH/v7+1/Q5mZmZqq6udi2lpaXXXTuAtufarggttGbNGqWmpio6OtrVlpaW5vpzQkKCkpKSFBsbq/fee0/jx49vdD+ZmZnKyMhwrdfU1BBogHbg/87YSpJlWQ3aJOnixYuaNGmSFi1apH79+l3z/u12u+x2+3XXCaBt81qY+frrr/XBBx/o3XffvWK/qKgoxcbG6tixY0324YIEtC/h4eHy8/NrMAtTUVHRYLZGkmpra7V//34dPHhQjz32mCTp0qVLsixL/v7+2r59u0aNGtUqtQNoe7x2m2ndunXq0aOHxowZc8V+VVVVKi0tVVRUlLdKAdDGBAQEyOFwuL7teFl+fr6Sk5Mb9A8JCdHhw4d16NAh15Kenq6f/vSnOnTokAYPHtxapQNog7wyM3Pp0iWtW7dOU6ZMcbu3XVdXp6ysLE2YMEFRUVE6ceKEnnnmGYWHh+vee+/1RikA2qiMjAw9+OCDSkpK0tChQ/XHP/5RJSUlSk9Pl/T97eV//vOf2rBhgzp16qSEhAS38T169FBgYGCDdgAdj1fCzAcffKCSkhI9/PDDbu1+fn46fPiwNmzYoNOnTysqKkojR45Ubm6ugoODvVEKgDYqLS1NVVVVWrx4scrKypSQkKC8vDzFxsZKksrKyq76zhkAkCSbZVmWr4torpqaGoWGhqq6ulohISG+LqfDOXDggBwOhwr/tkaJA37quzoO/0OO1GkqLCxUYmKiz+roqEw8D02sGWhPvHUO8ttMAADAaIQZAABgNMIMAAAwGmEGAAAYjTADAACMRpgBAABGI8wAAACjEWYAAIDRCDMAAMBohBkAAGA0wgwAADAaYQYAABiNMAMAAIxGmAEAAEYjzAAAAKMRZgAAgNEIMwAAwGiEGQAAYDTCDAAAMBphBgAAGM3f1wXAXEXHvm7x2HP/curEyXL17hmpLoH2Vv98AED7QZhBs4WHhysoKEgPzFrs61IUFBSk8PBwX5cBAPAhwgyarVevXioqKlJlZWWL91FUVKQHHnhAGzduVHx8fIv3Ex4erl69erV4PADAfIQZtEivXr08EiLi4+OVmJjogYoAAB0VDwADAACjEWYAAIDRCDMAAMBohBkAAGA0wgwAADAaYQYAABiNMAMAAIxGmAEAAEYjzAAAAKMRZgAAgNEIMwAAwGiEGQAAYDSPh5msrCzZbDa3JTIy0rXdsixlZWUpOjpaXbp00YgRI3TkyBFPlwEAADoIr8zM3HzzzSorK3Mthw8fdm174YUXtHTpUq1YsUL79u1TZGSk7rzzTtXW1nqjFAAA0M75e2Wn/v5uszGXWZalZcuWacGCBRo/frwk6Y033lBERIQ2bdqkRx55pNH9OZ1OOZ1O13pNTY03ygYAAAbyyszMsWPHFB0drbi4ON133306fvy4JKm4uFjl5eVKSUlx9bXb7brtttu0Z8+eJveXnZ2t0NBQ1xITE+ONsgEAgIE8HmYGDx6sDRs2aNu2bXrttddUXl6u5ORkVVVVqby8XJIUERHhNiYiIsK1rTGZmZmqrq52LaWlpZ4uGwAAGMrjt5lSU1Ndfx4wYICGDh2qm266SW+88YaGDBkiSbLZbG5jLMtq0PZ/2e122e12T5cKAADaAa9/Nbtr164aMGCAjh075nqO5oezMBUVFQ1mawAAAK6F18OM0+lUUVGRoqKiFBcXp8jISOXn57u219fXq6CgQMnJyd4uBQAAtEMev800b948jR07Vr169VJFRYWee+451dTUaMqUKbLZbJo9e7aef/559e3bV3379tXzzz+voKAgTZo0ydOlAACADsDjYebkyZP67W9/q8rKSnXv3l1DhgzRJ598otjYWEnSU089pXPnzmnmzJn67rvvNHjwYG3fvl3BwcGeLgUAAHQAHg8zb7311hW322w2ZWVlKSsry9MfDQAAOiB+mwkAABiNMAMAAIxGmAEAAEYjzAAAAKMRZgAAgNEIMwAAwGiEGQAAYDTCDAAAMBphBgAAGI0wAwAAjEaYAQAARiPMAAAAoxFmAACA0QgzAADAaIQZAABgNMIMAAAwGmEGgM+sXLlScXFxCgwMlMPh0K5du5rs++677+rOO+9U9+7dFRISoqFDh2rbtm2tWC2AtoowA8AncnNzNXv2bC1YsEAHDx7U8OHDlZqaqpKSkkb779y5U3feeafy8vJUWFiokSNHauzYsTp48GArVw6grbFZlmX5uojmqqmpUWhoqKqrqxUSEuLrctACBw4ckMPhUGFhoRITE31dDlrges/DwYMHKzExUatWrXK1xcfHa9y4ccrOzr6mfdx8881KS0vTs88+2+h2p9Mpp9PpVnNMTAzXDsBHvPX/b2ZmALS6+vp6FRYWKiUlxa09JSVFe/bsuaZ9XLp0SbW1tQoLC2uyT3Z2tkJDQ11LTEzMddUNoG0izABodZWVlbp48aIiIiLc2iMiIlReXn5N+1iyZInOnDmjiRMnNtknMzNT1dXVrqW0tPS66gbQNvn7ugAAHZfNZnNbtyyrQVtj3nzzTWVlZekvf/mLevTo0WQ/u90uu91+3XUCaNsIMwBaXXh4uPz8/BrMwlRUVDSYrfmh3NxcTZs2TW+//bbuuOMOb5YJwBDcZgLQ6gICAuRwOJSfn+/Wnp+fr+Tk5CbHvfnmm5o6dao2bdqkMWPGeLtMAIZgZgaAT2RkZOjBBx9UUlKShg4dqj/+8Y8qKSlRenq6pO+fd/nnP/+pDRs2SPo+yEyePFkvvfSShgwZ4prV6dKli0JDQ312HAB8jzADwCfS0tJUVVWlxYsXq6ysTAkJCcrLy1NsbKwkqayszO2dM6tXr9aFCxf06KOP6tFHH3W1T5kyRevXr2/t8gG0IYQZAD4zc+ZMzZw5s9FtPwwoO3bs8H5BAIzEMzMAAMBohBkAAGA0wgwAADAaYQYAABiNMAMAAIxGmAEAAEYjzAAAAKMRZgAAgNE8Hmays7P1i1/8QsHBwerRo4fGjRunf/zjH259pk6dKpvN5rYMGTLE06UAAIAOwONhpqCgQI8++qg++eQT5efn68KFC0pJSdGZM2fc+t19990qKytzLXl5eZ4uBQAAdAAe/zmD999/32193bp16tGjhwoLC/WrX/3K1W632xUZGenpjwcAAB2M15+Zqa6uliSFhYW5te/YsUM9evRQv379NGPGDFVUVDS5D6fTqZqaGrcFAABA8nKYsSxLGRkZ+uUvf6mEhARXe2pqqnJycvThhx9qyZIl2rdvn0aNGiWn09nofrKzsxUaGupaYmJivFk2AAAwiFd/Nfuxxx7T//zP/2j37t1u7Wlpaa4/JyQkKCkpSbGxsXrvvfc0fvz4BvvJzMxURkaGa72mpoZAAwAAJHkxzDz++OPaunWrdu7cqZ49e16xb1RUlGJjY3Xs2LFGt9vtdtntdm+UCQAADOfxMGNZlh5//HFt3rxZO3bsUFxc3FXHVFVVqbS0VFFRUZ4uBwAAtHMef2bm0Ucf1caNG7Vp0yYFBwervLxc5eXlOnfunCSprq5O8+bN0969e3XixAnt2LFDY8eOVXh4uO69915PlwMAANo5j8/MrFq1SpI0YsQIt/Z169Zp6tSp8vPz0+HDh7VhwwadPn1aUVFRGjlypHJzcxUcHOzpcgAAQDvnldtMV9KlSxdt27bN0x8LAAA6KH6bCQAAGI0wAwAAjEaYAQAARiPMAAAAoxFmAACA0QgzAADAaIQZAABgNMIMAAAwGmEGAAAYjTADAACMRpgBAABGI8wAAACjEWYAAIDRCDMAAMBohBkAAGA0wgwAADAaYQYAABiNMAMAAIxGmAEAAEYjzAAAAKMRZgAAgNEIMwAAwGiEGQAAYDTCDAAAMBphBgAAGI0wAwAAjEaYAQAARiPMAAAAoxFmAACA0fx9XQDap7Nnz+rzzz9vcntRUZHbP6+kf//+CgoK8lhtAID2hTADr/j888/lcDiu2u+BBx64ap/CwkIlJiZ6oiwAQDtEmIFX9O/fX4WFhU1uP3funE6cOKHevXurS5cuV90XAABNIczAK4KCgq46mzJs2LBWqgYA0J7xADAAADAaYQYAABiNMAMAAIzm0zCzcuVKxcXFKTAwUA6HQ7t27fJlOQAAwEA+CzO5ubmaPXu2FixYoIMHD2r48OFKTU1VSUmJr0oCAAAG8lmYWbp0qaZNm6bp06crPj5ey5YtU0xMjFatWuWrkgAAgIF8Embq6+tVWFiolJQUt/aUlBTt2bOnQX+n06mamhq3BQAAQPJRmKmsrNTFixcVERHh1h4REaHy8vIG/bOzsxUaGupaYmJiWqtUAADQxvn0AWCbzea2bllWgzZJyszMVHV1tWspLS1trRIBAEAb55M3AIeHh8vPz6/BLExFRUWD2RpJstvtstvtrVUeAAAwiE9mZgICAuRwOJSfn+/Wnp+fr+TkZF+UBAAADOWz20wZGRl6/fXXtXbtWhUVFWnOnDkqKSlRenq6r0oC0Mqa+66pgoICORwOBQYGqk+fPnr11VdbqVIAbZnPfmgyLS1NVVVVWrx4scrKypSQkKC8vDzFxsb6qiQArejyu6ZWrlypYcOGafXq1UpNTdXRo0fVq1evBv2Li4s1evRozZgxQxs3btTHH3+smTNnqnv37powYYIPjgBAW2GzLMvydRHNVV1drRtuuEGlpaUKCQnxdTlAh1RTU6OYmBidPn1aoaGhzR4/ePBgJSYmur1bKj4+XuPGjVN2dnaD/k8//bS2bt2qoqIiV1t6ero+/fRT7d27t9HPcDqdcjqdrvXq6mr16tWLawfgI9d73WiKz2Zmrkdtba0k8RVtoA2ora1t9kXp8rum5s+f79be1LumJGnv3r0N3k111113ac2aNTp//rw6d+7cYEx2drYWLVrUoJ1rB+BbVVVVhJno6GiVlpYqODi40a9yo+27nM75G7K5LMtSbW2toqOjmz22ue+akqTy8vJG+1+4cEGVlZWKiopqMCYzM1MZGRmu9dOnTys2NlYlJSUevZB6k4nnCjW3DhNrvjw7GhYW5tH9GhlmOnXqpJ49e/q6DHhASEiIMSchGrreQHCt75q6Uv/G2i9r6rUOoaGhxv13Z+K5Qs2tw8SaO3Xy7PePfPrSPAAdU3PfNSVJkZGRjfb39/dXt27dvFYrgLaPMAOg1bXkXVNDhw5t0H/79u1KSkpq9HkZAB0HYQY+YbfbtXDhQt7s3IFd7V1TmZmZmjx5sqt/enq6vv76a2VkZKioqEhr167VmjVrNG/evGv+TBP/u6Pm1kHNrcNbNRv51WwA7cPKlSv1wgsvuN419eKLL+pXv/qVJGnq1Kk6ceKEduzY4epfUFCgOXPm6MiRI4qOjtbTTz/NizYBEGYAAIDZuM0EAACMRpgBAABGI8wAAACjEWYAAIDRCDNoVTt37tTYsWMVHR0tm82mLVu2+LoktDMrV65UXFycAgMD5XA4tGvXriv2LygokMPhUGBgoPr06aNXX321lSr9/5pT87vvvqs777xT3bt3V0hIiIYOHapt27a1YrXfa+6/58s+/vhj+fv765ZbbvFugY1obs1Op1MLFixQbGys7Ha7brrpJq1du7aVqv1ec2vOycnRwIEDFRQUpKioKD300EOqqqpqpWpbdo33yDloAa0oLy/PWrBggfXnP//ZkmRt3rzZ1yWhHXnrrbeszp07W6+99pp19OhR64knnrC6du1qff311432P378uBUUFGQ98cQT1tGjR63XXnvN6ty5s/XOO++02ZqfeOIJ6/e//7313//939YXX3xhZWZmWp07d7YOHDjQZmu+7PTp01afPn2slJQUa+DAga1T7P9qSc333HOPNXjwYCs/P98qLi62/uu//sv6+OOP22zNu3btsjp16mS99NJL1vHjx61du3ZZN998szVu3LhWq7m513hPnYOEGfgMYQaeNmjQICs9Pd2trX///tb8+fMb7f/UU09Z/fv3d2t75JFHrCFDhnitxh9qbs2N+dnPfmYtWrTI06U1qaU1p6WlWf/+7/9uLVy4sNXDTHNr/tvf/maFhoZaVVVVrVFeo5pb8x/+8AerT58+bm0vv/yy1bNnT6/VeCXXco331DnIbSYA7UJ9fb0KCwuVkpLi1p6SkqI9e/Y0Ombv3r0N+t91113av3+/zp8/77VaL2tJzT906dIl1dbWevxXiJvS0prXrVunr776SgsXLvR2iQ20pOatW7cqKSlJL7zwgm688Ub169dP8+bN07lz51qj5BbVnJycrJMnTyovL0+WZembb77RO++8ozFjxrRGyS3iqXPQyF/NBoAfqqys1MWLFxv8UGVERESDH6i8rLy8vNH+Fy5cUGVlpaKiorxWr9Symn9oyZIlOnPmjCZOnOiNEhtoSc3Hjh3T/PnztWvXLvn7t/7/dlpS8/Hjx7V7924FBgZq8+bNqqys1MyZM/Xtt9+2ynMzLak5OTlZOTk5SktL07/+9S9duHBB99xzj5YvX+71elvKU+cgMzMA2hWbzea2bllWg7ar9W+s3ZuaW/Nlb775prKyspSbm6sePXp4q7xGXWvNFy9e1KRJk7Ro0SL169evtcprVHP+PV+6dEk2m005OTkaNGiQRo8eraVLl2r9+vWtNjsjNa/mo0ePatasWXr22WdVWFio999/X8XFxW3+Jz88cQ4yMwOgXQgPD5efn1+Dv7VWVFQ0+JvfZZGRkY329/f3V7du3bxW62Utqfmy3NxcTZs2TW+//bbuuOMOb5bpprk119bWav/+/Tp48KAee+wxSd8HBcuy5O/vr+3bt2vUqFFtqmZJioqK0o033qjQ0FBXW3x8vCzL0smTJ9W3b982V3N2draGDRumJ598UpL085//XF27dtXw4cP13HPPeX2msSU8dQ4yMwOgXQgICJDD4VB+fr5be35+vpKTkxsdM3To0Ab9t2/frqSkJHXu3NlrtV7Wkpql72dkpk6dqk2bNrX68xDNrTkkJESHDx/WoUOHXEt6erp++tOf6tChQxo8eHCbq1mShg0bplOnTqmurs7V9sUXX6hTp07q2bOnV+uVWlbz2bNn1amT+//W/fz8JP3/2Y62xmPnYLMeFwauU21trXXw4EHr4MGDliRr6dKl1sGDB6/6lU7gWlz+KuuaNWuso0ePWrNnz7a6du1qnThxwrIsy5o/f7714IMPuvpf/lronDlzrKNHj1pr1qzx2Vezr7XmTZs2Wf7+/tYrr7xilZWVuZbTp0+32Zp/yBffZmpuzbW1tVbPnj2t3/zmN9aRI0esgoICq2/fvtb06dPbbM3r1q2z/P39rZUrV1pfffWVtXv3bispKckaNGhQq9V8tWu8t85Bwgxa1UcffWRJarBMmTLF16WhnXjllVes2NhYKyAgwEpMTLQKCgpc26ZMmWLddtttbv137Nhh3XrrrVZAQIDVu3dva9WqVa1ccfNqvu2229rEOdTcf8//ly/CjGU1v+aioiLrjjvusLp06WL17NnTysjIsM6ePduma3755Zetn/3sZ1aXLl2sqKgo6/7777dOnjzZavVe7RrvrXPQZlltdO4JAADgGvDMDAAAMBphBgAAGI0wAwAAjEaYAQAARiPMAAAAoxFmAACA0QgzAADAaIQZAABgNMIMAAAwGmEGAAAYjTADAACM9v8AKlXRsS1EkvwAAAAASUVORK5CYII=",
      "text/plain": [
       "<Figure size 640x480 with 2 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import matplotlib.pyplot as plt\n",
    "import numpy as np\n",
    "import json\n",
    "from matplotlib import cbook\n",
    "\n",
    "data = json.loads(result['stats_json'])\n",
    "fig, (ax1, ax2) = plt.subplots(1, 2)\n",
    "\n",
    "# separate calculation of statistics and plotting\n",
    "ax1.bxp([data], patch_artist=True, showfliers=False, boxprops={'facecolor': 'bisque'})\n",
    "#ax2.bxp([yess], patch_artist=True, boxprops={'facecolor': 'bisque'})\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "c796fe24",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "count                         3249.0\n",
      "mean                      126.992921\n",
      "std                        83.678368\n",
      "min                            100.0\n",
      "25%                            100.0\n",
      "50%                            100.0\n",
      "75%                            100.0\n",
      "max                           1400.0\n",
      "mode                           100.0\n",
      "ATACseq_overlap             0.543807\n",
      "ATACseq_ranges_overlap      0.096953\n",
      "ATACseq_Jaccard             0.089661\n",
      "DNAseq_overlap               0.45644\n",
      "DNAseq_ranges_overlap         0.1338\n",
      "DNAseq_Jaccard              0.115411\n",
      "genes_overlap               0.649961\n",
      "percentage_open             0.022594\n",
      "percentage_open_adj         0.027035\n",
      "model_name                  test run\n",
      "Name: Length, dtype: object\n"
     ]
    }
   ],
   "source": [
    "print(result)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "aa3c6a9a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "count                        13302.0\n",
      "mean                      128.670501\n",
      "std                       113.512934\n",
      "min                              5.0\n",
      "25%                             80.0\n",
      "50%                             95.0\n",
      "75%                            140.0\n",
      "max                           1985.0\n",
      "mode                            95.0\n",
      "ATACseq_overlap             0.791289\n",
      "ATACseq_ranges_overlap      0.034008\n",
      "ATACseq_Jaccard             0.033706\n",
      "DNAseq_overlap              0.699477\n",
      "DNAseq_ranges_overlap       0.049429\n",
      "DNAseq_Jaccard              0.048401\n",
      "genes_overlap               0.596375\n",
      "percentage_open             0.093725\n",
      "percentage_open_adj         0.112148\n",
      "model_name                   k-means\n",
      "Name: Length, dtype: object\n"
     ]
    }
   ],
   "source": [
    "print(evaluation)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "92562b46",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "count                        14519.0\n",
      "mean                      124.223431\n",
      "std                       114.045787\n",
      "min                              5.0\n",
      "25%                             70.0\n",
      "50%                             90.0\n",
      "75%                            135.0\n",
      "max                           1980.0\n",
      "mode                            90.0\n",
      "ATACseq_overlap             0.793681\n",
      "ATACseq_ranges_overlap      0.032371\n",
      "ATACseq_Jaccard             0.032101\n",
      "DNAseq_overlap              0.707587\n",
      "DNAseq_ranges_overlap       0.047451\n",
      "DNAseq_Jaccard              0.046538\n",
      "genes_overlap                0.59436\n",
      "percentage_open             0.098764\n",
      "percentage_open_adj         0.118178\n",
      "model_name                   k-means\n",
      "Name: Length, dtype: object\n"
     ]
    }
   ],
   "source": [
    "print(evaluation)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "4328129e",
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "File \u001b[0;32m/opt/anaconda3/lib/python3.12/site-packages/pyreadr/librdata.pyx:229\u001b[0m, in \u001b[0;36mpyreadr.librdata.Parser._Parser__handle_column\u001b[0;34m()\u001b[0m\n",
      "File \u001b[0;32m/opt/anaconda3/lib/python3.12/enum.py:726\u001b[0m, in \u001b[0;36mEnumType.__call__\u001b[0;34m(cls, value, names, module, qualname, type, start, boundary, *values)\u001b[0m\n\u001b[1;32m    721\u001b[0m \u001b[38;5;250m    \u001b[39m\u001b[38;5;124;03m\"\"\"\u001b[39;00m\n\u001b[1;32m    722\u001b[0m \u001b[38;5;124;03m    classes/types should always be True.\u001b[39;00m\n\u001b[1;32m    723\u001b[0m \u001b[38;5;124;03m    \"\"\"\u001b[39;00m\n\u001b[1;32m    724\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;01mTrue\u001b[39;00m\n\u001b[0;32m--> 726\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21m__call__\u001b[39m(\u001b[38;5;28mcls\u001b[39m, value, names\u001b[38;5;241m=\u001b[39m_not_given, \u001b[38;5;241m*\u001b[39mvalues, module\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m, qualname\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m, \u001b[38;5;28mtype\u001b[39m\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m, start\u001b[38;5;241m=\u001b[39m\u001b[38;5;241m1\u001b[39m, boundary\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m):\n\u001b[1;32m    727\u001b[0m \u001b[38;5;250m    \u001b[39m\u001b[38;5;124;03m\"\"\"\u001b[39;00m\n\u001b[1;32m    728\u001b[0m \u001b[38;5;124;03m    Either returns an existing member, or creates a new enum class.\u001b[39;00m\n\u001b[1;32m    729\u001b[0m \n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m    751\u001b[0m \u001b[38;5;124;03m    `type`, if set, will be mixed in as the first base class.\u001b[39;00m\n\u001b[1;32m    752\u001b[0m \u001b[38;5;124;03m    \"\"\"\u001b[39;00m\n\u001b[1;32m    753\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mcls\u001b[39m\u001b[38;5;241m.\u001b[39m_member_map_:\n\u001b[1;32m    754\u001b[0m         \u001b[38;5;66;03m# simple value lookup if members exist\u001b[39;00m\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Exception ignored in: 'pyreadr.librdata._handle_column'\n",
      "Traceback (most recent call last):\n",
      "  File \"pyreadr/librdata.pyx\", line 229, in pyreadr.librdata.Parser._Parser__handle_column\n",
      "  File \"/opt/anaconda3/lib/python3.12/enum.py\", line 726, in __call__\n",
      "    def __call__(cls, value, names=_not_given, *values, module=None, qualname=None, type=None, start=1, boundary=None):\n",
      "\n",
      "KeyboardInterrupt: \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0\n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0\n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0\n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0\n",
      "4       18  3000040  3000040  18_3000040    0.0    0.0\n",
      "BINS  IntervalIndex([(3000016, 3000016], (3000016, 3000116], (3000116, 3000216],\n",
      "               (3000216, 3000316], (3000316, 3000416], (3000416, 3000516],\n",
      "               (3000516, 3000616], (3000616, 3000716], (3000716, 3000816],\n",
      "               (3000816, 3000916],\n",
      "               ...\n",
      "               (9716616, 9716716], (9716716, 9716816], (9716816, 9716916],\n",
      "               (9716916, 9717016], (9717016, 9717116], (9717116, 9717216],\n",
      "               (9717216, 9717316], (9717316, 9717416], (9717416, 9717516],\n",
      "               (9717516, 9717593]],\n",
      "              dtype='interval[int64, right]', length=67177)\n",
      "HEAD with BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1        bin_offset_0  \\\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0  (3000016, 3000116]   \n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0  (3000016, 3000116]   \n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0  (3000016, 3000116]   \n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0  (3000016, 3000116]   \n",
      "4       18  3000040  3000040  18_3000040    0.0    0.0  (3000016, 3000116]   \n",
      "\n",
      "   target_count  \n",
      "0             1  \n",
      "1             1  \n",
      "2             1  \n",
      "3             1  \n",
      "4             1  \n",
      "TAIL with BINS\n",
      "       seqnames    start      end       CG_ID  TT_S0  TT_S1  \\\n",
      "699995       18  9717555  9717555  18_9717555    0.0    0.0   \n",
      "699996       19  9717571  9717571  19_9717571    0.0    0.0   \n",
      "699997       19  9717581  9717581  19_9717581    0.0    0.0   \n",
      "699998       18  9717582  9717582  18_9717582    0.0    0.0   \n",
      "699999       18  9717593  9717593  18_9717593    0.0    0.0   \n",
      "\n",
      "              bin_offset_0  target_count  \n",
      "699995  (9717516, 9717593]             1  \n",
      "699996  (9717516, 9717593]             1  \n",
      "699997  (9717516, 9717593]             1  \n",
      "699998  (9717516, 9717593]             1  \n",
      "699999  (9717516, 9717593]             1  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:137: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "ename": "KeyError",
     "evalue": "\"['TT_S2'] not found in axis\"",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[7], line 4\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[38;5;28;01mimport\u001b[39;00m \u001b[38;5;21;01mos\u001b[39;00m\n\u001b[1;32m      2\u001b[0m os\u001b[38;5;241m.\u001b[39menviron[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mCOLUMNS\u001b[39m\u001b[38;5;124m'\u001b[39m]\u001b[38;5;241m=\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m300\u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[0;32m----> 4\u001b[0m result \u001b[38;5;241m=\u001b[39m evaluator\u001b[38;5;241m.\u001b[39mevaluate(merged_ranges \u001b[38;5;241m=\u001b[39m chromatin_algos\u001b[38;5;241m.\u001b[39malgo5d_aggregated(\u001b[38;5;241m1\u001b[39m, n_clusters\u001b[38;5;241m=\u001b[39m\u001b[38;5;241m10\u001b[39m))\n",
      "File \u001b[0;32m~/VU/bio-baigiamasis/scripts/chromatin_algos.py:387\u001b[0m, in \u001b[0;36malgo5d_aggregated\u001b[0;34m(scale, n_clusters, threshold, bin_offset)\u001b[0m\n\u001b[1;32m    386\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21malgo5d_aggregated\u001b[39m(scale \u001b[38;5;241m=\u001b[39m \u001b[38;5;241m1\u001b[39m, n_clusters \u001b[38;5;241m=\u001b[39m \u001b[38;5;241m2\u001b[39m, threshold \u001b[38;5;241m=\u001b[39m \u001b[38;5;241m1.5\u001b[39m, bin_offset \u001b[38;5;241m=\u001b[39m \u001b[38;5;241m0\u001b[39m):\n\u001b[0;32m--> 387\u001b[0m     model_output \u001b[38;5;241m=\u001b[39m algo5d(scale, n_clusters, bin_offset)\n\u001b[1;32m    388\u001b[0m     update_labels(model_output, threshold)\n\u001b[1;32m    389\u001b[0m     model_output\u001b[38;5;241m.\u001b[39mto_csv(\u001b[38;5;124m'\u001b[39m\u001b[38;5;124m../outputs/output_algo5d_aggregated.csv\u001b[39m\u001b[38;5;124m'\u001b[39m, sep \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m'\u001b[39m\u001b[38;5;130;01m\\t\u001b[39;00m\u001b[38;5;124m'\u001b[39m)\n",
      "File \u001b[0;32m~/VU/bio-baigiamasis/scripts/chromatin_algos.py:140\u001b[0m, in \u001b[0;36malgo5d\u001b[0;34m(scale, n_clusters, bin_offset)\u001b[0m\n\u001b[1;32m    137\u001b[0m sumos \u001b[38;5;241m=\u001b[39m df\u001b[38;5;241m.\u001b[39mgroupby(\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mbin_offset_\u001b[39m\u001b[38;5;124m'\u001b[39m \u001b[38;5;241m+\u001b[39m \u001b[38;5;28mstr\u001b[39m(bin_offset))\u001b[38;5;241m.\u001b[39msum()\u001b[38;5;241m.\u001b[39mreset_index()\n\u001b[1;32m    138\u001b[0m sumos[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mstarting_nt\u001b[39m\u001b[38;5;124m'\u001b[39m] \u001b[38;5;241m=\u001b[39m sumos[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mbin_offset_\u001b[39m\u001b[38;5;124m'\u001b[39m \u001b[38;5;241m+\u001b[39m \u001b[38;5;28mstr\u001b[39m(bin_offset)]\u001b[38;5;241m.\u001b[39mapply(\u001b[38;5;28;01mlambda\u001b[39;00m x: x\u001b[38;5;241m.\u001b[39mleft)\n\u001b[0;32m--> 140\u001b[0m sumos\u001b[38;5;241m.\u001b[39mdrop(columns\u001b[38;5;241m=\u001b[39mOTHER_SIGNALS, axis\u001b[38;5;241m=\u001b[39m\u001b[38;5;241m1\u001b[39m, inplace\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m)\n\u001b[1;32m    142\u001b[0m \u001b[38;5;28mprint\u001b[39m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mSUMOS\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[1;32m    143\u001b[0m \u001b[38;5;28mprint\u001b[39m(sumos[:\u001b[38;5;241m50\u001b[39m])\n",
      "File \u001b[0;32m/opt/anaconda3/lib/python3.12/site-packages/pandas/core/frame.py:5581\u001b[0m, in \u001b[0;36mDataFrame.drop\u001b[0;34m(self, labels, axis, index, columns, level, inplace, errors)\u001b[0m\n\u001b[1;32m   5433\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mdrop\u001b[39m(\n\u001b[1;32m   5434\u001b[0m     \u001b[38;5;28mself\u001b[39m,\n\u001b[1;32m   5435\u001b[0m     labels: IndexLabel \u001b[38;5;241m|\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m,\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m   5442\u001b[0m     errors: IgnoreRaise \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mraise\u001b[39m\u001b[38;5;124m\"\u001b[39m,\n\u001b[1;32m   5443\u001b[0m ) \u001b[38;5;241m-\u001b[39m\u001b[38;5;241m>\u001b[39m DataFrame \u001b[38;5;241m|\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[1;32m   5444\u001b[0m \u001b[38;5;250m    \u001b[39m\u001b[38;5;124;03m\"\"\"\u001b[39;00m\n\u001b[1;32m   5445\u001b[0m \u001b[38;5;124;03m    Drop specified labels from rows or columns.\u001b[39;00m\n\u001b[1;32m   5446\u001b[0m \n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m   5579\u001b[0m \u001b[38;5;124;03m            weight  1.0     0.8\u001b[39;00m\n\u001b[1;32m   5580\u001b[0m \u001b[38;5;124;03m    \"\"\"\u001b[39;00m\n\u001b[0;32m-> 5581\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28msuper\u001b[39m()\u001b[38;5;241m.\u001b[39mdrop(\n\u001b[1;32m   5582\u001b[0m         labels\u001b[38;5;241m=\u001b[39mlabels,\n\u001b[1;32m   5583\u001b[0m         axis\u001b[38;5;241m=\u001b[39maxis,\n\u001b[1;32m   5584\u001b[0m         index\u001b[38;5;241m=\u001b[39mindex,\n\u001b[1;32m   5585\u001b[0m         columns\u001b[38;5;241m=\u001b[39mcolumns,\n\u001b[1;32m   5586\u001b[0m         level\u001b[38;5;241m=\u001b[39mlevel,\n\u001b[1;32m   5587\u001b[0m         inplace\u001b[38;5;241m=\u001b[39minplace,\n\u001b[1;32m   5588\u001b[0m         errors\u001b[38;5;241m=\u001b[39merrors,\n\u001b[1;32m   5589\u001b[0m     )\n",
      "File \u001b[0;32m/opt/anaconda3/lib/python3.12/site-packages/pandas/core/generic.py:4788\u001b[0m, in \u001b[0;36mNDFrame.drop\u001b[0;34m(self, labels, axis, index, columns, level, inplace, errors)\u001b[0m\n\u001b[1;32m   4786\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m axis, labels \u001b[38;5;129;01min\u001b[39;00m axes\u001b[38;5;241m.\u001b[39mitems():\n\u001b[1;32m   4787\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m labels \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[0;32m-> 4788\u001b[0m         obj \u001b[38;5;241m=\u001b[39m obj\u001b[38;5;241m.\u001b[39m_drop_axis(labels, axis, level\u001b[38;5;241m=\u001b[39mlevel, errors\u001b[38;5;241m=\u001b[39merrors)\n\u001b[1;32m   4790\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m inplace:\n\u001b[1;32m   4791\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_update_inplace(obj)\n",
      "File \u001b[0;32m/opt/anaconda3/lib/python3.12/site-packages/pandas/core/generic.py:4830\u001b[0m, in \u001b[0;36mNDFrame._drop_axis\u001b[0;34m(self, labels, axis, level, errors, only_slice)\u001b[0m\n\u001b[1;32m   4828\u001b[0m         new_axis \u001b[38;5;241m=\u001b[39m axis\u001b[38;5;241m.\u001b[39mdrop(labels, level\u001b[38;5;241m=\u001b[39mlevel, errors\u001b[38;5;241m=\u001b[39merrors)\n\u001b[1;32m   4829\u001b[0m     \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[0;32m-> 4830\u001b[0m         new_axis \u001b[38;5;241m=\u001b[39m axis\u001b[38;5;241m.\u001b[39mdrop(labels, errors\u001b[38;5;241m=\u001b[39merrors)\n\u001b[1;32m   4831\u001b[0m     indexer \u001b[38;5;241m=\u001b[39m axis\u001b[38;5;241m.\u001b[39mget_indexer(new_axis)\n\u001b[1;32m   4833\u001b[0m \u001b[38;5;66;03m# Case for non-unique axis\u001b[39;00m\n\u001b[1;32m   4834\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n",
      "File \u001b[0;32m/opt/anaconda3/lib/python3.12/site-packages/pandas/core/indexes/base.py:7070\u001b[0m, in \u001b[0;36mIndex.drop\u001b[0;34m(self, labels, errors)\u001b[0m\n\u001b[1;32m   7068\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m mask\u001b[38;5;241m.\u001b[39many():\n\u001b[1;32m   7069\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m errors \u001b[38;5;241m!=\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mignore\u001b[39m\u001b[38;5;124m\"\u001b[39m:\n\u001b[0;32m-> 7070\u001b[0m         \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mKeyError\u001b[39;00m(\u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mlabels[mask]\u001b[38;5;241m.\u001b[39mtolist()\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m not found in axis\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[1;32m   7071\u001b[0m     indexer \u001b[38;5;241m=\u001b[39m indexer[\u001b[38;5;241m~\u001b[39mmask]\n\u001b[1;32m   7072\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mdelete(indexer)\n",
      "\u001b[0;31mKeyError\u001b[0m: \"['TT_S2'] not found in axis\""
     ]
    }
   ],
   "source": [
    "import os\n",
    "os.environ['COLUMNS']=\"300\"\n",
    "\n",
    "result = evaluator.evaluate(merged_ranges = chromatin_algos.algo5d_aggregated(1, n_clusters=10))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "6201148c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0\n",
      "BINS  IntervalIndex([  (3000018, 3000018],   (3000018, 3000118],\n",
      "                 (3000118, 3000218],   (3000218, 3000318],\n",
      "                 (3000318, 3000418],   (3000418, 3000518],\n",
      "                 (3000518, 3000618],   (3000618, 3000718],\n",
      "                 (3000718, 3000818],   (3000818, 3000918],\n",
      "               ...\n",
      "               (18260718, 18260818], (18260818, 18260918],\n",
      "               (18260918, 18261018], (18261018, 18261118],\n",
      "               (18261118, 18261218], (18261218, 18261318],\n",
      "               (18261318, 18261418], (18261418, 18261518],\n",
      "               (18261518, 18261618], (18261618, 18261710]],\n",
      "              dtype='interval[int64, right]', length=152618)\n",
      "HEAD with BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0   \n",
      "\n",
      "         bin_offset_0  target_count  \n",
      "0  (3000018, 3000118]             1  \n",
      "1  (3000018, 3000118]             1  \n",
      "2  (3000018, 3000118]             1  \n",
      "3  (3000018, 3000118]             1  \n",
      "4  (3000018, 3000118]             1  \n",
      "TAIL with BINS\n",
      "       seqnames     start       end        CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  18261611  18261611  18_18261611    0.0    0.0    0.0   \n",
      "699996       18  18261615  18261615  18_18261615    0.0    0.0    0.0   \n",
      "699997       18  18261682  18261682  18_18261682    0.0    0.0    0.0   \n",
      "699998       18  18261702  18261702  18_18261702    4.0    0.0    0.0   \n",
      "699999       18  18261710  18261710  18_18261710    0.0    0.0    0.0   \n",
      "\n",
      "                bin_offset_0  target_count  \n",
      "699995  (18261518, 18261618]             1  \n",
      "699996  (18261518, 18261618]             1  \n",
      "699997  (18261618, 18261710]             1  \n",
      "699998  (18261618, 18261710]             1  \n",
      "699999  (18261618, 18261710]             1  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:137: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "          bin_offset_0  TT_S0  target_count  starting_nt\n",
      "0   (3000018, 3000018]    0.0             0      3000018\n",
      "1   (3000018, 3000118]    7.0            11      3000018\n",
      "2   (3000118, 3000218]    1.0             4      3000118\n",
      "3   (3000218, 3000318]    2.0             4      3000218\n",
      "4   (3000318, 3000418]    0.0             4      3000318\n",
      "5   (3000418, 3000518]    0.0             2      3000418\n",
      "6   (3000518, 3000618]    0.0             2      3000518\n",
      "7   (3000618, 3000718]    0.0             4      3000618\n",
      "8   (3000718, 3000818]    0.0             7      3000718\n",
      "9   (3000818, 3000918]    0.0             3      3000818\n",
      "10  (3000918, 3001018]    0.0             3      3000918\n",
      "11  (3001018, 3001118]    2.0             6      3001018\n",
      "12  (3001118, 3001218]    0.0             2      3001118\n",
      "13  (3001218, 3001318]    0.0             4      3001218\n",
      "14  (3001318, 3001418]    0.0             3      3001318\n",
      "15  (3001418, 3001518]    1.0             4      3001418\n",
      "16  (3001518, 3001618]    3.0            10      3001518\n",
      "17  (3001618, 3001718]    3.0             7      3001618\n",
      "18  (3001718, 3001818]    0.0             6      3001718\n",
      "19  (3001818, 3001918]    0.0             2      3001818\n",
      "20  (3001918, 3002018]    1.0             3      3001918\n",
      "21  (3002018, 3002118]    0.0             2      3002018\n",
      "22  (3002118, 3002218]    1.0             4      3002118\n",
      "23  (3002218, 3002318]   10.0             6      3002218\n",
      "24  (3002318, 3002418]    2.0             1      3002318\n",
      "25  (3002418, 3002518]   19.0             5      3002418\n",
      "26  (3002518, 3002618]    4.0             2      3002518\n",
      "27  (3002618, 3002718]    0.0             3      3002618\n",
      "28  (3002718, 3002818]    0.0             1      3002718\n",
      "29  (3002818, 3002918]    0.0             2      3002818\n",
      "30  (3002918, 3003018]    1.0             6      3002918\n",
      "31  (3003018, 3003118]    2.0             4      3003018\n",
      "32  (3003118, 3003218]    0.0             0      3003118\n",
      "33  (3003218, 3003318]    0.0             1      3003218\n",
      "34  (3003318, 3003418]    1.0             8      3003318\n",
      "35  (3003418, 3003518]    0.0             3      3003418\n",
      "36  (3003518, 3003618]    1.0             2      3003518\n",
      "37  (3003618, 3003718]    3.0             4      3003618\n",
      "38  (3003718, 3003818]    1.0             6      3003718\n",
      "39  (3003818, 3003918]    0.0             3      3003818\n",
      "40  (3003918, 3004018]    8.0             6      3003918\n",
      "41  (3004018, 3004118]    0.0             0      3004018\n",
      "42  (3004118, 3004218]    0.0             1      3004118\n",
      "43  (3004218, 3004318]    5.0             6      3004218\n",
      "44  (3004318, 3004418]    0.0             1      3004318\n",
      "45  (3004418, 3004518]    0.0             1      3004418\n",
      "46  (3004518, 3004618]    1.0             7      3004518\n",
      "47  (3004618, 3004718]    0.0             0      3004618\n",
      "48  (3004718, 3004818]    0.0             1      3004718\n",
      "49  (3004818, 3004918]   13.0             8      3004818\n",
      "(152618, 4)\n",
      "bin_offset_0    category\n",
      "TT_S0            float64\n",
      "target_count       int64\n",
      "starting_nt        int64\n",
      "dtype: object\n",
      "sumos ilgis: 152618\n",
      "ranges ilgis: 152618\n",
      "Model input - NO SCALE\n",
      "        TT_S0  genes  transcripts  exons  CDSs\n",
      "0         0.0      0            0      0     0\n",
      "1         7.0      0            0      0     0\n",
      "2         1.0      0            0      0     0\n",
      "3         2.0      0            0      0     0\n",
      "4         0.0      0            0      0     0\n",
      "...       ...    ...          ...    ...   ...\n",
      "152613   10.0      0            0      0     0\n",
      "152614    3.0      0            0      0     0\n",
      "152615    4.0      0            0      0     0\n",
      "152616    0.0      0            0      0     0\n",
      "152617    4.0      0            0      0     0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "        TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0         0.0             0                   0             0            0\n",
      "1         7.0             0                   0             0            0\n",
      "2         1.0             0                   0             0            0\n",
      "3         2.0             0                   0             0            0\n",
      "4         0.0             0                   0             0            0\n",
      "...       ...           ...                 ...           ...          ...\n",
      "152613   10.0             0                   0             0            0\n",
      "152614    3.0             0                   0             0            0\n",
      "152615    4.0             0                   0             0            0\n",
      "152616    0.0             0                   0             0            0\n",
      "152617    4.0             0                   0             0            0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "0.10474518077815198\n",
      "[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 1 0 1 1 1 0 1 1 1 1 1 1 0 1 1 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 0 0 1 1 1 1 1 1 0 0 1 1 0 1\n",
      " 0 1 0 0 0 0 0 0 0 0 0 0 0 0 1 0 1 1 1 1 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 1\n",
      " 0 0 1 0 0 0 0 0 1 0 0 0 1 0 1 0 0 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 1 1 0 1 1 1 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 1 0 1 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0]\n",
      "threshold density 1.95819\n",
      "cluster signal 219187.0\n",
      "cluster target count 572145\n",
      "!!cluster density 0.3830969422087058\n",
      "cluster signal 237724.0\n",
      "cluster target count 127855\n",
      "!!cluster density 1.8593250166203903\n",
      "\n",
      "Labels mapping\n",
      " {0: 0, 1: 0}\n",
      "DEBUG: 152618 0 \n",
      "\n",
      " +----------------------+-------------+----------------+-------+\n",
      "| bin_offset_0         | TT_S0       | target_count   | +9    |\n",
      "| (category)           | (float64)   | (int64)        | ...   |\n",
      "|----------------------+-------------+----------------+-------|\n",
      "| (3000018, 3000018]   | 0.0         | 0              | ...   |\n",
      "| (3000018, 3000118]   | 7.0         | 11             | ...   |\n",
      "| (3000118, 3000218]   | 1.0         | 4              | ...   |\n",
      "| (3000218, 3000318]   | 2.0         | 4              | ...   |\n",
      "| ...                  | ...         | ...            | ...   |\n",
      "| (18261318, 18261418] | 3.0         | 15             | ...   |\n",
      "| (18261418, 18261518] | 4.0         | 4              | ...   |\n",
      "| (18261518, 18261618] | 0.0         | 3              | ...   |\n",
      "| (18261618, 18261710] | 4.0         | 3              | ...   |\n",
      "+----------------------+-------------+----------------+-------+\n",
      "Unstranded PyRanges object has 152,618 rows and 12 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "9 hidden columns: starting_nt, Chromosome, Start, End, genes, transcripts, ... (+ 3 more.)\n",
      "The model failed to identify open ranges.\n",
      "           0\n",
      "model_name  \n"
     ]
    }
   ],
   "source": [
    "import generate_consensus\n",
    "\n",
    "ranges = generate_consensus.generate_consensus_5x20(chromatin_algos.algo5d_aggregated, [1, 2, 3])\n",
    "result = evaluator.evaluate(merged_ranges = ranges)\n",
    "print(result)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "56123321",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Single Mode Example:\n",
      "Mode: 0    2\n",
      "dtype: int64\n",
      "Mode: <class 'int'>\n",
      "\n",
      "Multiple Modes Example:\n",
      "Modes: 0    2\n",
      "1    3\n",
      "dtype: int64\n",
      "First Mode: <class 'int'>\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Example data with a clear mode\n",
    "data_with_single_mode = pd.Series([1, 2, 2, 3, 4])\n",
    "print(\"Single Mode Example:\")\n",
    "print(\"Mode:\", data_with_single_mode.mode())  # Should work fine\n",
    "print(\"Mode:\", type(data_with_single_mode.mode().iloc[0].item()))  # Should work fine\n",
    "\n",
    "# Example data with multiple modes\n",
    "data_with_multiple_modes = pd.Series([1, 2, 2, 3, 3, 4])\n",
    "print(\"\\nMultiple Modes Example:\")\n",
    "print(\"Modes:\", data_with_multiple_modes.mode())\n",
    "print(\"First Mode:\", type(data_with_multiple_modes.mode().iloc[0].item()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "6b5c5de3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "count                        13118.0\n",
      "mean                      146.683946\n",
      "std                       113.175202\n",
      "min                            100.0\n",
      "25%                            100.0\n",
      "50%                            100.0\n",
      "75%                            200.0\n",
      "max                           2500.0\n",
      "mode                           100.0\n",
      "ATACseq_overlap             0.792635\n",
      "ATACseq_ranges_overlap      0.030302\n",
      "ATACseq_Jaccard             0.030064\n",
      "DNAseq_overlap              0.696856\n",
      "DNAseq_ranges_overlap       0.043802\n",
      "DNAseq_Jaccard              0.042983\n",
      "genes_overlap               0.590862\n",
      "percentage_open             0.105368\n",
      "percentage_open_adj          0.12608\n",
      "model_name                          \n",
      "Name: Length, dtype: object\n"
     ]
    }
   ],
   "source": [
    "result = evaluator.evaluate(merged_ranges = ranges)\n",
    "\n",
    "print(result)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "44720aa9",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "       seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0            19  3000017  3000017  19_3000017    0.0    0.0    0.0\n",
      "1            18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "2            18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "3            18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "4            18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "...         ...      ...      ...         ...    ...    ...    ...\n",
      "699995       18  9717555  9717555  18_9717555    1.0    0.0    0.0\n",
      "699996       19  9717571  9717571  19_9717571    0.0    0.0    0.0\n",
      "699997       19  9717581  9717581  19_9717581    0.0    0.0    0.0\n",
      "699998       18  9717582  9717582  18_9717582    0.0    0.0    0.0\n",
      "699999       18  9717593  9717593  18_9717593    0.0    0.0    0.0\n",
      "\n",
      "[700000 rows x 7 columns]\n"
     ]
    }
   ],
   "source": [
    "import enrichment\n",
    "import pyreadr\n",
    "\n",
    "targets_df = pyreadr.read_r('../inputs/subset1.rds')[None]\n",
    "\n",
    "print(targets_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "bdd299ab",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/enrichment.py:19: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  summed_ranges = targets_df.groupby('ranges_bin')[[SIGNAL_COLUMN, 'target_count']].sum()\n",
      "/Users/lukas/VU/bio-baigiamasis/scripts/enrichment.py:19: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  summed_ranges = targets_df.groupby('ranges_bin')[[SIGNAL_COLUMN, 'target_count']].sum()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dataframe normalized signal: 0.65273\n",
      "                 ranges_bin  norm_signal  pre_signal  post_signal  \\\n",
      "0        [3002218, 3002318)     1.428571    0.333333     2.000000   \n",
      "1        [3002418, 3002518)     3.800000    2.000000     2.000000   \n",
      "2        [3003918, 3004018)     1.333333    0.000000     0.000000   \n",
      "3        [3004818, 3004918)     1.625000    0.000000     0.000000   \n",
      "4        [3005018, 3005318)     5.666667    0.000000     1.500000   \n",
      "...                     ...          ...         ...          ...   \n",
      "13113  [18247918, 18248018)     4.333333    0.500000     0.400000   \n",
      "13114  [18249518, 18249618)     1.285714    1.333333     1.000000   \n",
      "13115  [18250918, 18251018)     1.500000    0.000000     0.142857   \n",
      "13116  [18251518, 18251618)     1.333333    0.200000     0.000000   \n",
      "13117  [18261218, 18261318)     1.111111    0.400000     0.200000   \n",
      "\n",
      "       surround_signal  enrichment_factor1  enrichment_factor2  \n",
      "0             2.333333            0.612245            2.188610  \n",
      "1             4.000000            0.950000            5.821703  \n",
      "2             0.000000            0.000000            2.042703  \n",
      "3             0.000000            0.000000            2.489544  \n",
      "4             1.500000            3.777778            8.681486  \n",
      "...                ...                 ...                 ...  \n",
      "13113         0.900000            4.814815            6.638784  \n",
      "13114         2.333333            0.551020            1.969749  \n",
      "13115         0.142857           10.500000            2.298041  \n",
      "13116         0.200000            6.666667            2.042703  \n",
      "13117         0.600000            1.851852            1.702252  \n",
      "\n",
      "[13118 rows x 7 columns]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/enrichment.py:19: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  summed_ranges = targets_df.groupby('ranges_bin')[[SIGNAL_COLUMN, 'target_count']].sum()\n"
     ]
    }
   ],
   "source": [
    "import enrichment\n",
    "\n",
    "print(enrichment.calculate_enrichment_factors_file(ranges_filepath='../outputs/test_ranges.tsv', targets_filepath='../inputs/subset1.rds'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "d67fb278",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/enrichment.py:19: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  summed_ranges = targets_df.groupby('ranges_bin')[[SIGNAL_COLUMN, 'target_count']].sum()\n",
      "/Users/lukas/VU/bio-baigiamasis/scripts/enrichment.py:19: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  summed_ranges = targets_df.groupby('ranges_bin')[[SIGNAL_COLUMN, 'target_count']].sum()\n",
      "/Users/lukas/VU/bio-baigiamasis/scripts/enrichment.py:19: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  summed_ranges = targets_df.groupby('ranges_bin')[[SIGNAL_COLUMN, 'target_count']].sum()\n",
      "/Users/lukas/VU/bio-baigiamasis/scripts/enrichment.py:19: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  summed_ranges = targets_df.groupby('ranges_bin')[[SIGNAL_COLUMN, 'target_count']].sum()\n"
     ]
    },
    {
     "ename": "KeyError",
     "evalue": "'pre_signal'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "File \u001b[0;32m/opt/anaconda3/lib/python3.12/site-packages/pandas/core/indexes/base.py:3805\u001b[0m, in \u001b[0;36mIndex.get_loc\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m   3804\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m-> 3805\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_engine\u001b[38;5;241m.\u001b[39mget_loc(casted_key)\n\u001b[1;32m   3806\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mKeyError\u001b[39;00m \u001b[38;5;28;01mas\u001b[39;00m err:\n",
      "File \u001b[0;32mindex.pyx:167\u001b[0m, in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[0;34m()\u001b[0m\n",
      "File \u001b[0;32mindex.pyx:191\u001b[0m, in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[0;34m()\u001b[0m\n",
      "File \u001b[0;32mindex.pyx:234\u001b[0m, in \u001b[0;36mpandas._libs.index.IndexEngine._get_loc_duplicates\u001b[0;34m()\u001b[0m\n",
      "File \u001b[0;32mindex.pyx:242\u001b[0m, in \u001b[0;36mpandas._libs.index.IndexEngine._maybe_get_bool_indexer\u001b[0;34m()\u001b[0m\n",
      "File \u001b[0;32mindex.pyx:134\u001b[0m, in \u001b[0;36mpandas._libs.index._unpack_bool_indexer\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;31mKeyError\u001b[0m: 'pre_signal'",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[6], line 2\u001b[0m\n\u001b[1;32m      1\u001b[0m enriched \u001b[38;5;241m=\u001b[39m enrichment\u001b[38;5;241m.\u001b[39mcalculate_norm_signal(ranges, targets_df)\n\u001b[0;32m----> 2\u001b[0m surround \u001b[38;5;241m=\u001b[39m enrichment\u001b[38;5;241m.\u001b[39mcalculate_surround_signal(ranges, targets_df)\n\u001b[1;32m      4\u001b[0m \u001b[38;5;28mprint\u001b[39m(enriched)\n\u001b[1;32m      5\u001b[0m \u001b[38;5;28mprint\u001b[39m(surround)\n",
      "File \u001b[0;32m~/VU/bio-baigiamasis/scripts/enrichment.py:54\u001b[0m, in \u001b[0;36mcalculate_surround_signal\u001b[0;34m(ranges, targets_df)\u001b[0m\n\u001b[1;32m     51\u001b[0m ranges_signal \u001b[38;5;241m=\u001b[39m pd\u001b[38;5;241m.\u001b[39mconcat([ranges_signal\u001b[38;5;241m.\u001b[39mreset_index(drop\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m), pre_signal\u001b[38;5;241m.\u001b[39mreset_index(drop\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m)], axis\u001b[38;5;241m=\u001b[39m\u001b[38;5;241m1\u001b[39m)\n\u001b[1;32m     52\u001b[0m ranges_signal \u001b[38;5;241m=\u001b[39m pd\u001b[38;5;241m.\u001b[39mconcat([ranges_signal\u001b[38;5;241m.\u001b[39mreset_index(drop\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m), post_signal\u001b[38;5;241m.\u001b[39mreset_index(drop\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m)], axis\u001b[38;5;241m=\u001b[39m\u001b[38;5;241m1\u001b[39m)\n\u001b[0;32m---> 54\u001b[0m ranges_signal[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124msurround_signal\u001b[39m\u001b[38;5;124m'\u001b[39m] \u001b[38;5;241m=\u001b[39m ranges_signal[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mpre_signal\u001b[39m\u001b[38;5;124m'\u001b[39m] \u001b[38;5;241m+\u001b[39m ranges_signal[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mpost_signal\u001b[39m\u001b[38;5;124m'\u001b[39m]\n\u001b[1;32m     56\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m ranges_signal\n",
      "File \u001b[0;32m/opt/anaconda3/lib/python3.12/site-packages/pandas/core/frame.py:4102\u001b[0m, in \u001b[0;36mDataFrame.__getitem__\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m   4100\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mcolumns\u001b[38;5;241m.\u001b[39mnlevels \u001b[38;5;241m>\u001b[39m \u001b[38;5;241m1\u001b[39m:\n\u001b[1;32m   4101\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_getitem_multilevel(key)\n\u001b[0;32m-> 4102\u001b[0m indexer \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mcolumns\u001b[38;5;241m.\u001b[39mget_loc(key)\n\u001b[1;32m   4103\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m is_integer(indexer):\n\u001b[1;32m   4104\u001b[0m     indexer \u001b[38;5;241m=\u001b[39m [indexer]\n",
      "File \u001b[0;32m/opt/anaconda3/lib/python3.12/site-packages/pandas/core/indexes/base.py:3812\u001b[0m, in \u001b[0;36mIndex.get_loc\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m   3807\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(casted_key, \u001b[38;5;28mslice\u001b[39m) \u001b[38;5;129;01mor\u001b[39;00m (\n\u001b[1;32m   3808\u001b[0m         \u001b[38;5;28misinstance\u001b[39m(casted_key, abc\u001b[38;5;241m.\u001b[39mIterable)\n\u001b[1;32m   3809\u001b[0m         \u001b[38;5;129;01mand\u001b[39;00m \u001b[38;5;28many\u001b[39m(\u001b[38;5;28misinstance\u001b[39m(x, \u001b[38;5;28mslice\u001b[39m) \u001b[38;5;28;01mfor\u001b[39;00m x \u001b[38;5;129;01min\u001b[39;00m casted_key)\n\u001b[1;32m   3810\u001b[0m     ):\n\u001b[1;32m   3811\u001b[0m         \u001b[38;5;28;01mraise\u001b[39;00m InvalidIndexError(key)\n\u001b[0;32m-> 3812\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mKeyError\u001b[39;00m(key) \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01merr\u001b[39;00m\n\u001b[1;32m   3813\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mTypeError\u001b[39;00m:\n\u001b[1;32m   3814\u001b[0m     \u001b[38;5;66;03m# If we have a listlike key, _check_indexing_error will raise\u001b[39;00m\n\u001b[1;32m   3815\u001b[0m     \u001b[38;5;66;03m#  InvalidIndexError. Otherwise we fall through and re-raise\u001b[39;00m\n\u001b[1;32m   3816\u001b[0m     \u001b[38;5;66;03m#  the TypeError.\u001b[39;00m\n\u001b[1;32m   3817\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_check_indexing_error(key)\n",
      "\u001b[0;31mKeyError\u001b[0m: 'pre_signal'"
     ]
    }
   ],
   "source": [
    "enriched = enrichment.calculate_norm_signal(ranges, targets_df)\n",
    "surround = enrichment.calculate_surround_signal(ranges, targets_df)\n",
    "\n",
    "print(enriched)\n",
    "print(surround)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "49f3ca45",
   "metadata": {},
   "outputs": [],
   "source": [
    "ranges.df.to_csv('../outputs/test_ranges.tsv', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "8de8e4e8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "+--------------+-----------+-----------+-----------+\n",
      "| Chromosome   | Start     | End       | Length    |\n",
      "| (category)   | (int64)   | (int64)   | (int64)   |\n",
      "|--------------+-----------+-----------+-----------|\n",
      "| 18           | 3005016   | 3005116   | 200       |\n",
      "| 18           | 3005516   | 3005616   | 400       |\n",
      "| 18           | 3010016   | 3010116   | 200       |\n",
      "| 18           | 3010616   | 3010716   | 100       |\n",
      "| ...          | ...       | ...       | ...       |\n",
      "| 18           | 9707516   | 9707616   | 100       |\n",
      "| 18           | 9707916   | 9708016   | 200       |\n",
      "| 18           | 9708416   | 9708516   | 300       |\n",
      "| 18           | 9708916   | 9709016   | 100       |\n",
      "+--------------+-----------+-----------+-----------+\n",
      "Unstranded PyRanges object has 4,985 rows and 4 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "+--------------+-----------+-----------+-----------+\n",
      "| Chromosome   | Start     | End       | Length    |\n",
      "| (category)   | (int64)   | (int64)   | (int64)   |\n",
      "|--------------+-----------+-----------+-----------|\n",
      "| 18           | 3005116   | 3005316   | 200       |\n",
      "| 18           | 3005616   | 3006016   | 400       |\n",
      "| 18           | 3010116   | 3010316   | 200       |\n",
      "| 18           | 3010716   | 3010816   | 100       |\n",
      "| ...          | ...       | ...       | ...       |\n",
      "| 18           | 9707616   | 9707716   | 100       |\n",
      "| 18           | 9708016   | 9708216   | 200       |\n",
      "| 18           | 9708516   | 9708816   | 300       |\n",
      "| 18           | 9709016   | 9709116   | 100       |\n",
      "+--------------+-----------+-----------+-----------+\n",
      "Unstranded PyRanges object has 4,985 rows and 4 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n"
     ]
    }
   ],
   "source": [
    "import enrichment\n",
    "\n",
    "print(enrichment.get_pre_regions(ranges))\n",
    "print(ranges)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "fa21a536",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "+--------------+-----------+-----------+-----------+\n",
      "| Chromosome   | Start     | End       | Length    |\n",
      "| (category)   | (int64)   | (int64)   | (int64)   |\n",
      "|--------------+-----------+-----------+-----------|\n",
      "| 18           | 3002416   | 3002516   | 100       |\n",
      "| 18           | 3005116   | 3005316   | 200       |\n",
      "| 18           | 3005416   | 3005516   | 100       |\n",
      "| 18           | 3005616   | 3006016   | 400       |\n",
      "| ...          | ...       | ...       | ...       |\n",
      "| 18           | 9708016   | 9708216   | 200       |\n",
      "| 18           | 9708316   | 9708416   | 100       |\n",
      "| 18           | 9708516   | 9708816   | 300       |\n",
      "| 18           | 9709016   | 9709116   | 100       |\n",
      "+--------------+-----------+-----------+-----------+\n",
      "Unstranded PyRanges object has 6,883 rows and 4 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "os.environ['COLUMNS']=\"300\"\n",
    "print(ranges)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "f475e092",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "24\n",
      "51.0\n"
     ]
    }
   ],
   "source": [
    "test = targets_df[(targets_df[\"start\"] >= 3010116) & (targets_df[\"start\"] <= 3010316)]\n",
    "#[3010116, 3010316]   51.0            24\n",
    "\n",
    "print(test['target_count'].sum())\n",
    "print(test['TT_S0'].sum())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "9435baf8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1615\n",
      "0\n"
     ]
    }
   ],
   "source": [
    "import pyranges as pr\n",
    "\n",
    "ranges.Length = ranges.End - ranges.Start\n",
    "ranges_df = ranges.df\n",
    "ranges_df['']\n",
    "\n",
    "print(len(ranges[ranges.Length < 75]))\n",
    "ranges2 = evaluator.process_ranges(ranges)\n",
    "ranges2.Length = ranges2.End - ranges2.Start\n",
    "print(len(ranges2[ranges2.Length < 75]))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "a929baa0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "count                         8182.0\n",
      "mean                      182.306282\n",
      "std                       197.351081\n",
      "min                              5.0\n",
      "25%                             80.0\n",
      "50%                            115.0\n",
      "75%                            210.0\n",
      "max                           2085.0\n",
      "ATACseq_overlap             0.751796\n",
      "ATACseq_ranges_overlap      0.013469\n",
      "DNAseq_overlap              0.639008\n",
      "DNAseq_ranges_overlap       0.022784\n",
      "genes_overlap                0.54393\n",
      "percentage_open             0.153498\n",
      "percentage_open_adj         0.222048\n",
      "model_name                          \n",
      "Name: Length, dtype: object\n"
     ]
    }
   ],
   "source": [
    "print(evaluator.evaluate(ranges))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "61959364",
   "metadata": {},
   "outputs": [],
   "source": [
    "def calculate_gaps(pyranges_obj):\n",
    "    \"\"\"\n",
    "    Calculate gaps between ranges in a PyRanges object.\n",
    "\n",
    "    Parameters:\n",
    "        pyranges_obj (pr.PyRanges): A PyRanges object with genomic ranges.\n",
    "\n",
    "    Returns:\n",
    "        pr.PyRanges: A PyRanges object containing the gaps.\n",
    "    \"\"\"\n",
    "    # Convert PyRanges to a DataFrame\n",
    "    df = pyranges_obj.df\n",
    "    \n",
    "    # Sort the ranges by Chromosome and Start position\n",
    "    df = df.sort_values([\"Chromosome\", \"Start\"]).reset_index(drop=True)\n",
    "    \n",
    "    # Shift End and Start columns to calculate gaps\n",
    "    prev_end = df.groupby(\"Chromosome\")[\"End\"].shift(0)   # Current End\n",
    "    next_start = df.groupby(\"Chromosome\")[\"Start\"].shift(-1)  # Next Start\n",
    "    \n",
    "    # Identify gaps where End < Next Start\n",
    "    gaps_df = df[(prev_end < next_start)].assign(Start=prev_end, End=next_start)[[\"Chromosome\", \"Start\", \"End\"]]\n",
    "    \n",
    "    # Drop NaN rows caused by shifting and ensure integer type\n",
    "    gaps_df = gaps_df.dropna().astype({\"Start\": \"int\", \"End\": \"int\"})\n",
    "    \n",
    "    # Convert back to PyRanges\n",
    "    return pr.PyRanges(gaps_df)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "cbaf97f0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "               0     1     2     3\n",
      "Chromosome  chr1  chr1  chr1  chr3\n",
      "Start          5    20    80    10\n",
      "End           10    28    95    38\n",
      "Strand         +     +     -     +\n",
      "title          a     b     c     d\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "df=pd.DataFrame(\n",
    "{'Chromosome':['chr1', 'chr1', 'chr1', 'chr3'],\n",
    " 'Start': [5, 20, 80, 10],\n",
    " 'End':   [10, 28, 95, 38],\n",
    " 'Strand':['+', '+', '-', '+'],\n",
    " 'title': ['a', 'b', 'c', 'd']}\n",
    ")\n",
    "\n",
    "print(df.transpose())\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "847f9a77",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "443\n",
      "0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/y9/8yssfmz161s4ghy99nf1f2cw0000gn/T/ipykernel_27239/19855938.py:18: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  prev_end = df.groupby(\"Chromosome\")[\"End\"].shift(0)   # Current End\n",
      "/var/folders/y9/8yssfmz161s4ghy99nf1f2cw0000gn/T/ipykernel_27239/19855938.py:19: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  next_start = df.groupby(\"Chromosome\")[\"Start\"].shift(-1)  # Next Start\n",
      "/var/folders/y9/8yssfmz161s4ghy99nf1f2cw0000gn/T/ipykernel_27239/19855938.py:18: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  prev_end = df.groupby(\"Chromosome\")[\"End\"].shift(0)   # Current End\n",
      "/var/folders/y9/8yssfmz161s4ghy99nf1f2cw0000gn/T/ipykernel_27239/19855938.py:19: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  next_start = df.groupby(\"Chromosome\")[\"Start\"].shift(-1)  # Next Start\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import pyranges as pr\n",
    "\n",
    "ranges_df = ranges.df\n",
    "gaps = calculate_gaps(ranges)\n",
    "gaps.Length = gaps.End - gaps.Start\n",
    "print(len(gaps[gaps.Length <= 50]))\n",
    "\n",
    "ranges2 = evaluator.process_ranges(ranges)\n",
    "gaps2 = calculate_gaps(ranges2)\n",
    "gaps2.Length = gaps2.End - gaps2.Start\n",
    "print(len(gaps2[gaps2.Length <= 50]))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "28e1cdda",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Runnning scale 1\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0\n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "BINS  IntervalIndex([(3000016, 3000016], (3000016, 3000116], (3000116, 3000216],\n",
      "               (3000216, 3000316], (3000316, 3000416], (3000416, 3000516],\n",
      "               (3000516, 3000616], (3000616, 3000716], (3000716, 3000816],\n",
      "               (3000816, 3000916],\n",
      "               ...\n",
      "               (9716616, 9716716], (9716716, 9716816], (9716816, 9716916],\n",
      "               (9716916, 9717016], (9717016, 9717116], (9717116, 9717216],\n",
      "               (9717216, 9717316], (9717316, 9717416], (9717416, 9717516],\n",
      "               (9717516, 9717593]],\n",
      "              dtype='interval[int64, right]', length=67177)\n",
      "HEAD w BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0   \n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "\n",
      "         bin_offset_0  \n",
      "0  (3000016, 3000116]  \n",
      "1  (3000016, 3000116]  \n",
      "2  (3000016, 3000116]  \n",
      "3  (3000016, 3000116]  \n",
      "4  (3000016, 3000116]  \n",
      "TAIL w BINS\n",
      "       seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  9717555  9717555  18_9717555    1.0    0.0    0.0   \n",
      "699996       19  9717571  9717571  19_9717571    0.0    0.0    0.0   \n",
      "699997       19  9717581  9717581  19_9717581    0.0    0.0    0.0   \n",
      "699998       18  9717582  9717582  18_9717582    0.0    0.0    0.0   \n",
      "699999       18  9717593  9717593  18_9717593    0.0    0.0    0.0   \n",
      "\n",
      "              bin_offset_0  \n",
      "699995  (9717516, 9717593]  \n",
      "699996  (9717516, 9717593]  \n",
      "699997  (9717516, 9717593]  \n",
      "699998  (9717516, 9717593]  \n",
      "699999  (9717516, 9717593]  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:151: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "    TT_S0  starting_nt\n",
      "0     0.0      3000016\n",
      "1     7.0      3000016\n",
      "2     1.0      3000116\n",
      "3     2.0      3000216\n",
      "4     0.0      3000316\n",
      "5     0.0      3000416\n",
      "6     0.0      3000516\n",
      "7     0.0      3000616\n",
      "8     0.0      3000716\n",
      "9     0.0      3000816\n",
      "10    0.0      3000916\n",
      "11    2.0      3001016\n",
      "12    0.0      3001116\n",
      "13    0.0      3001216\n",
      "14    0.0      3001316\n",
      "15    1.0      3001416\n",
      "16    3.0      3001516\n",
      "17    3.0      3001616\n",
      "18    0.0      3001716\n",
      "19    0.0      3001816\n",
      "20    1.0      3001916\n",
      "21    0.0      3002016\n",
      "22    1.0      3002116\n",
      "23   10.0      3002216\n",
      "24    2.0      3002316\n",
      "25   19.0      3002416\n",
      "26    4.0      3002516\n",
      "27    0.0      3002616\n",
      "28    0.0      3002716\n",
      "29    0.0      3002816\n",
      "30    1.0      3002916\n",
      "31    2.0      3003016\n",
      "32    0.0      3003116\n",
      "33    0.0      3003216\n",
      "34    1.0      3003316\n",
      "35    0.0      3003416\n",
      "36    1.0      3003516\n",
      "37    1.0      3003616\n",
      "38    3.0      3003716\n",
      "39    0.0      3003816\n",
      "40    8.0      3003916\n",
      "41    0.0      3004016\n",
      "42    0.0      3004116\n",
      "43    5.0      3004216\n",
      "44    0.0      3004316\n",
      "45    0.0      3004416\n",
      "46    1.0      3004516\n",
      "47    0.0      3004616\n",
      "48    0.0      3004716\n",
      "49   13.0      3004816\n",
      "(67177, 2)\n",
      "TT_S0          float64\n",
      "starting_nt      int64\n",
      "dtype: object\n",
      "sumos ilgis: 67177\n",
      "ranges ilgis: 67177\n",
      "Model input - NO SCALE\n",
      "       TT_S0  genes  transcripts  exons  CDSs\n",
      "0        0.0      0            0      0     0\n",
      "1        7.0      0            0      0     0\n",
      "2        1.0      0            0      0     0\n",
      "3        2.0      0            0      0     0\n",
      "4        0.0      0            0      0     0\n",
      "...      ...    ...          ...    ...   ...\n",
      "67172    0.0      0            0      0     0\n",
      "67173    0.0      0            0      0     0\n",
      "67174    0.0      0            0      0     0\n",
      "67175    1.0      0            0      0     0\n",
      "67176    1.0      0            0      0     0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "       TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0        0.0             0                   0             0            0\n",
      "1        7.0             0                   0             0            0\n",
      "2        1.0             0                   0             0            0\n",
      "3        2.0             0                   0             0            0\n",
      "4        0.0             0                   0             0            0\n",
      "...      ...           ...                 ...           ...          ...\n",
      "67172    0.0             0                   0             0            0\n",
      "67173    0.0             0                   0             0            0\n",
      "67174    0.0             0                   0             0            0\n",
      "67175    1.0             0                   0             0            0\n",
      "67176    1.0             0                   0             0            0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "0.17092159518882952\n",
      "[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 1 0 1 1 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 1 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0]\n",
      "Runnning scale 2\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0\n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "BINS  IntervalIndex([(3000016, 3000016], (3000016, 3000116], (3000116, 3000216],\n",
      "               (3000216, 3000316], (3000316, 3000416], (3000416, 3000516],\n",
      "               (3000516, 3000616], (3000616, 3000716], (3000716, 3000816],\n",
      "               (3000816, 3000916],\n",
      "               ...\n",
      "               (9716616, 9716716], (9716716, 9716816], (9716816, 9716916],\n",
      "               (9716916, 9717016], (9717016, 9717116], (9717116, 9717216],\n",
      "               (9717216, 9717316], (9717316, 9717416], (9717416, 9717516],\n",
      "               (9717516, 9717593]],\n",
      "              dtype='interval[int64, right]', length=67177)\n",
      "HEAD w BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0   \n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "\n",
      "         bin_offset_0  \n",
      "0  (3000016, 3000116]  \n",
      "1  (3000016, 3000116]  \n",
      "2  (3000016, 3000116]  \n",
      "3  (3000016, 3000116]  \n",
      "4  (3000016, 3000116]  \n",
      "TAIL w BINS\n",
      "       seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  9717555  9717555  18_9717555    1.0    0.0    0.0   \n",
      "699996       19  9717571  9717571  19_9717571    0.0    0.0    0.0   \n",
      "699997       19  9717581  9717581  19_9717581    0.0    0.0    0.0   \n",
      "699998       18  9717582  9717582  18_9717582    0.0    0.0    0.0   \n",
      "699999       18  9717593  9717593  18_9717593    0.0    0.0    0.0   \n",
      "\n",
      "              bin_offset_0  \n",
      "699995  (9717516, 9717593]  \n",
      "699996  (9717516, 9717593]  \n",
      "699997  (9717516, 9717593]  \n",
      "699998  (9717516, 9717593]  \n",
      "699999  (9717516, 9717593]  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:151: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "    TT_S0  starting_nt\n",
      "0     0.0      3000016\n",
      "1     7.0      3000016\n",
      "2     1.0      3000116\n",
      "3     2.0      3000216\n",
      "4     0.0      3000316\n",
      "5     0.0      3000416\n",
      "6     0.0      3000516\n",
      "7     0.0      3000616\n",
      "8     0.0      3000716\n",
      "9     0.0      3000816\n",
      "10    0.0      3000916\n",
      "11    2.0      3001016\n",
      "12    0.0      3001116\n",
      "13    0.0      3001216\n",
      "14    0.0      3001316\n",
      "15    1.0      3001416\n",
      "16    3.0      3001516\n",
      "17    3.0      3001616\n",
      "18    0.0      3001716\n",
      "19    0.0      3001816\n",
      "20    1.0      3001916\n",
      "21    0.0      3002016\n",
      "22    1.0      3002116\n",
      "23   10.0      3002216\n",
      "24    2.0      3002316\n",
      "25   19.0      3002416\n",
      "26    4.0      3002516\n",
      "27    0.0      3002616\n",
      "28    0.0      3002716\n",
      "29    0.0      3002816\n",
      "30    1.0      3002916\n",
      "31    2.0      3003016\n",
      "32    0.0      3003116\n",
      "33    0.0      3003216\n",
      "34    1.0      3003316\n",
      "35    0.0      3003416\n",
      "36    1.0      3003516\n",
      "37    1.0      3003616\n",
      "38    3.0      3003716\n",
      "39    0.0      3003816\n",
      "40    8.0      3003916\n",
      "41    0.0      3004016\n",
      "42    0.0      3004116\n",
      "43    5.0      3004216\n",
      "44    0.0      3004316\n",
      "45    0.0      3004416\n",
      "46    1.0      3004516\n",
      "47    0.0      3004616\n",
      "48    0.0      3004716\n",
      "49   13.0      3004816\n",
      "(67177, 2)\n",
      "TT_S0          float64\n",
      "starting_nt      int64\n",
      "dtype: object\n",
      "sumos ilgis: 67177\n",
      "ranges ilgis: 67177\n",
      "Model input - NO SCALE\n",
      "       TT_S0  genes  transcripts  exons  CDSs\n",
      "0        0.0      0            0      0     0\n",
      "1        7.0      0            0      0     0\n",
      "2        1.0      0            0      0     0\n",
      "3        2.0      0            0      0     0\n",
      "4        0.0      0            0      0     0\n",
      "...      ...    ...          ...    ...   ...\n",
      "67172    0.0      0            0      0     0\n",
      "67173    0.0      0            0      0     0\n",
      "67174    0.0      0            0      0     0\n",
      "67175    1.0      0            0      0     0\n",
      "67176    1.0      0            0      0     0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "       TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0        0.0             0                   0             0            0\n",
      "1        7.0             0                   0             0            0\n",
      "2        1.0             0                   0             0            0\n",
      "3        2.0             0                   0             0            0\n",
      "4        0.0             0                   0             0            0\n",
      "...      ...           ...                 ...           ...          ...\n",
      "67172    0.0             0                   0             0            0\n",
      "67173    0.0             0                   0             0            0\n",
      "67174    0.0             0                   0             0            0\n",
      "67175    1.0             0                   0             0            0\n",
      "67176    1.0             0                   0             0            0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "0.20496598538190156\n",
      "[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 1 0 1 1 1 1 0 0 1 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 1 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0]\n",
      "Runnning scale 4\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0\n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "BINS  IntervalIndex([(3000016, 3000016], (3000016, 3000116], (3000116, 3000216],\n",
      "               (3000216, 3000316], (3000316, 3000416], (3000416, 3000516],\n",
      "               (3000516, 3000616], (3000616, 3000716], (3000716, 3000816],\n",
      "               (3000816, 3000916],\n",
      "               ...\n",
      "               (9716616, 9716716], (9716716, 9716816], (9716816, 9716916],\n",
      "               (9716916, 9717016], (9717016, 9717116], (9717116, 9717216],\n",
      "               (9717216, 9717316], (9717316, 9717416], (9717416, 9717516],\n",
      "               (9717516, 9717593]],\n",
      "              dtype='interval[int64, right]', length=67177)\n",
      "HEAD w BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0   \n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "\n",
      "         bin_offset_0  \n",
      "0  (3000016, 3000116]  \n",
      "1  (3000016, 3000116]  \n",
      "2  (3000016, 3000116]  \n",
      "3  (3000016, 3000116]  \n",
      "4  (3000016, 3000116]  \n",
      "TAIL w BINS\n",
      "       seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  9717555  9717555  18_9717555    1.0    0.0    0.0   \n",
      "699996       19  9717571  9717571  19_9717571    0.0    0.0    0.0   \n",
      "699997       19  9717581  9717581  19_9717581    0.0    0.0    0.0   \n",
      "699998       18  9717582  9717582  18_9717582    0.0    0.0    0.0   \n",
      "699999       18  9717593  9717593  18_9717593    0.0    0.0    0.0   \n",
      "\n",
      "              bin_offset_0  \n",
      "699995  (9717516, 9717593]  \n",
      "699996  (9717516, 9717593]  \n",
      "699997  (9717516, 9717593]  \n",
      "699998  (9717516, 9717593]  \n",
      "699999  (9717516, 9717593]  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:151: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "    TT_S0  starting_nt\n",
      "0     0.0      3000016\n",
      "1     7.0      3000016\n",
      "2     1.0      3000116\n",
      "3     2.0      3000216\n",
      "4     0.0      3000316\n",
      "5     0.0      3000416\n",
      "6     0.0      3000516\n",
      "7     0.0      3000616\n",
      "8     0.0      3000716\n",
      "9     0.0      3000816\n",
      "10    0.0      3000916\n",
      "11    2.0      3001016\n",
      "12    0.0      3001116\n",
      "13    0.0      3001216\n",
      "14    0.0      3001316\n",
      "15    1.0      3001416\n",
      "16    3.0      3001516\n",
      "17    3.0      3001616\n",
      "18    0.0      3001716\n",
      "19    0.0      3001816\n",
      "20    1.0      3001916\n",
      "21    0.0      3002016\n",
      "22    1.0      3002116\n",
      "23   10.0      3002216\n",
      "24    2.0      3002316\n",
      "25   19.0      3002416\n",
      "26    4.0      3002516\n",
      "27    0.0      3002616\n",
      "28    0.0      3002716\n",
      "29    0.0      3002816\n",
      "30    1.0      3002916\n",
      "31    2.0      3003016\n",
      "32    0.0      3003116\n",
      "33    0.0      3003216\n",
      "34    1.0      3003316\n",
      "35    0.0      3003416\n",
      "36    1.0      3003516\n",
      "37    1.0      3003616\n",
      "38    3.0      3003716\n",
      "39    0.0      3003816\n",
      "40    8.0      3003916\n",
      "41    0.0      3004016\n",
      "42    0.0      3004116\n",
      "43    5.0      3004216\n",
      "44    0.0      3004316\n",
      "45    0.0      3004416\n",
      "46    1.0      3004516\n",
      "47    0.0      3004616\n",
      "48    0.0      3004716\n",
      "49   13.0      3004816\n",
      "(67177, 2)\n",
      "TT_S0          float64\n",
      "starting_nt      int64\n",
      "dtype: object\n",
      "sumos ilgis: 67177\n",
      "ranges ilgis: 67177\n",
      "Model input - NO SCALE\n",
      "       TT_S0  genes  transcripts  exons  CDSs\n",
      "0        0.0      0            0      0     0\n",
      "1        7.0      0            0      0     0\n",
      "2        1.0      0            0      0     0\n",
      "3        2.0      0            0      0     0\n",
      "4        0.0      0            0      0     0\n",
      "...      ...    ...          ...    ...   ...\n",
      "67172    0.0      0            0      0     0\n",
      "67173    0.0      0            0      0     0\n",
      "67174    0.0      0            0      0     0\n",
      "67175    1.0      0            0      0     0\n",
      "67176    1.0      0            0      0     0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "       TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0        0.0             0                   0             0            0\n",
      "1        7.0             0                   0             0            0\n",
      "2        1.0             0                   0             0            0\n",
      "3        2.0             0                   0             0            0\n",
      "4        0.0             0                   0             0            0\n",
      "...      ...           ...                 ...           ...          ...\n",
      "67172    0.0             0                   0             0            0\n",
      "67173    0.0             0                   0             0            0\n",
      "67174    0.0             0                   0             0            0\n",
      "67175    1.0             0                   0             0            0\n",
      "67176    1.0             0                   0             0            0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "0.20496598538190156\n",
      "[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 1 0 1 1 1 1 0 0 1 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 1 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0]\n",
      "Runnning scale 8\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0\n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "BINS  IntervalIndex([(3000016, 3000016], (3000016, 3000116], (3000116, 3000216],\n",
      "               (3000216, 3000316], (3000316, 3000416], (3000416, 3000516],\n",
      "               (3000516, 3000616], (3000616, 3000716], (3000716, 3000816],\n",
      "               (3000816, 3000916],\n",
      "               ...\n",
      "               (9716616, 9716716], (9716716, 9716816], (9716816, 9716916],\n",
      "               (9716916, 9717016], (9717016, 9717116], (9717116, 9717216],\n",
      "               (9717216, 9717316], (9717316, 9717416], (9717416, 9717516],\n",
      "               (9717516, 9717593]],\n",
      "              dtype='interval[int64, right]', length=67177)\n",
      "HEAD w BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0   \n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "\n",
      "         bin_offset_0  \n",
      "0  (3000016, 3000116]  \n",
      "1  (3000016, 3000116]  \n",
      "2  (3000016, 3000116]  \n",
      "3  (3000016, 3000116]  \n",
      "4  (3000016, 3000116]  \n",
      "TAIL w BINS\n",
      "       seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  9717555  9717555  18_9717555    1.0    0.0    0.0   \n",
      "699996       19  9717571  9717571  19_9717571    0.0    0.0    0.0   \n",
      "699997       19  9717581  9717581  19_9717581    0.0    0.0    0.0   \n",
      "699998       18  9717582  9717582  18_9717582    0.0    0.0    0.0   \n",
      "699999       18  9717593  9717593  18_9717593    0.0    0.0    0.0   \n",
      "\n",
      "              bin_offset_0  \n",
      "699995  (9717516, 9717593]  \n",
      "699996  (9717516, 9717593]  \n",
      "699997  (9717516, 9717593]  \n",
      "699998  (9717516, 9717593]  \n",
      "699999  (9717516, 9717593]  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:151: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "    TT_S0  starting_nt\n",
      "0     0.0      3000016\n",
      "1     7.0      3000016\n",
      "2     1.0      3000116\n",
      "3     2.0      3000216\n",
      "4     0.0      3000316\n",
      "5     0.0      3000416\n",
      "6     0.0      3000516\n",
      "7     0.0      3000616\n",
      "8     0.0      3000716\n",
      "9     0.0      3000816\n",
      "10    0.0      3000916\n",
      "11    2.0      3001016\n",
      "12    0.0      3001116\n",
      "13    0.0      3001216\n",
      "14    0.0      3001316\n",
      "15    1.0      3001416\n",
      "16    3.0      3001516\n",
      "17    3.0      3001616\n",
      "18    0.0      3001716\n",
      "19    0.0      3001816\n",
      "20    1.0      3001916\n",
      "21    0.0      3002016\n",
      "22    1.0      3002116\n",
      "23   10.0      3002216\n",
      "24    2.0      3002316\n",
      "25   19.0      3002416\n",
      "26    4.0      3002516\n",
      "27    0.0      3002616\n",
      "28    0.0      3002716\n",
      "29    0.0      3002816\n",
      "30    1.0      3002916\n",
      "31    2.0      3003016\n",
      "32    0.0      3003116\n",
      "33    0.0      3003216\n",
      "34    1.0      3003316\n",
      "35    0.0      3003416\n",
      "36    1.0      3003516\n",
      "37    1.0      3003616\n",
      "38    3.0      3003716\n",
      "39    0.0      3003816\n",
      "40    8.0      3003916\n",
      "41    0.0      3004016\n",
      "42    0.0      3004116\n",
      "43    5.0      3004216\n",
      "44    0.0      3004316\n",
      "45    0.0      3004416\n",
      "46    1.0      3004516\n",
      "47    0.0      3004616\n",
      "48    0.0      3004716\n",
      "49   13.0      3004816\n",
      "(67177, 2)\n",
      "TT_S0          float64\n",
      "starting_nt      int64\n",
      "dtype: object\n",
      "sumos ilgis: 67177\n",
      "ranges ilgis: 67177\n",
      "Model input - NO SCALE\n",
      "       TT_S0  genes  transcripts  exons  CDSs\n",
      "0        0.0      0            0      0     0\n",
      "1        7.0      0            0      0     0\n",
      "2        1.0      0            0      0     0\n",
      "3        2.0      0            0      0     0\n",
      "4        0.0      0            0      0     0\n",
      "...      ...    ...          ...    ...   ...\n",
      "67172    0.0      0            0      0     0\n",
      "67173    0.0      0            0      0     0\n",
      "67174    0.0      0            0      0     0\n",
      "67175    1.0      0            0      0     0\n",
      "67176    1.0      0            0      0     0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "       TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0        0.0             0                   0             0            0\n",
      "1        7.0             0                   0             0            0\n",
      "2        1.0             0                   0             0            0\n",
      "3        2.0             0                   0             0            0\n",
      "4        0.0             0                   0             0            0\n",
      "...      ...           ...                 ...           ...          ...\n",
      "67172    0.0             0                   0             0            0\n",
      "67173    0.0             0                   0             0            0\n",
      "67174    0.0             0                   0             0            0\n",
      "67175    1.0             0                   0             0            0\n",
      "67176    1.0             0                   0             0            0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "0.20496598538190156\n",
      "[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 1 0 1 1 1 1 0 0 1 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 1 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0]\n",
      "Runnning scale 16\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0\n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "BINS  IntervalIndex([(3000016, 3000016], (3000016, 3000116], (3000116, 3000216],\n",
      "               (3000216, 3000316], (3000316, 3000416], (3000416, 3000516],\n",
      "               (3000516, 3000616], (3000616, 3000716], (3000716, 3000816],\n",
      "               (3000816, 3000916],\n",
      "               ...\n",
      "               (9716616, 9716716], (9716716, 9716816], (9716816, 9716916],\n",
      "               (9716916, 9717016], (9717016, 9717116], (9717116, 9717216],\n",
      "               (9717216, 9717316], (9717316, 9717416], (9717416, 9717516],\n",
      "               (9717516, 9717593]],\n",
      "              dtype='interval[int64, right]', length=67177)\n",
      "HEAD w BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0   \n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "\n",
      "         bin_offset_0  \n",
      "0  (3000016, 3000116]  \n",
      "1  (3000016, 3000116]  \n",
      "2  (3000016, 3000116]  \n",
      "3  (3000016, 3000116]  \n",
      "4  (3000016, 3000116]  \n",
      "TAIL w BINS\n",
      "       seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  9717555  9717555  18_9717555    1.0    0.0    0.0   \n",
      "699996       19  9717571  9717571  19_9717571    0.0    0.0    0.0   \n",
      "699997       19  9717581  9717581  19_9717581    0.0    0.0    0.0   \n",
      "699998       18  9717582  9717582  18_9717582    0.0    0.0    0.0   \n",
      "699999       18  9717593  9717593  18_9717593    0.0    0.0    0.0   \n",
      "\n",
      "              bin_offset_0  \n",
      "699995  (9717516, 9717593]  \n",
      "699996  (9717516, 9717593]  \n",
      "699997  (9717516, 9717593]  \n",
      "699998  (9717516, 9717593]  \n",
      "699999  (9717516, 9717593]  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:151: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "    TT_S0  starting_nt\n",
      "0     0.0      3000016\n",
      "1     7.0      3000016\n",
      "2     1.0      3000116\n",
      "3     2.0      3000216\n",
      "4     0.0      3000316\n",
      "5     0.0      3000416\n",
      "6     0.0      3000516\n",
      "7     0.0      3000616\n",
      "8     0.0      3000716\n",
      "9     0.0      3000816\n",
      "10    0.0      3000916\n",
      "11    2.0      3001016\n",
      "12    0.0      3001116\n",
      "13    0.0      3001216\n",
      "14    0.0      3001316\n",
      "15    1.0      3001416\n",
      "16    3.0      3001516\n",
      "17    3.0      3001616\n",
      "18    0.0      3001716\n",
      "19    0.0      3001816\n",
      "20    1.0      3001916\n",
      "21    0.0      3002016\n",
      "22    1.0      3002116\n",
      "23   10.0      3002216\n",
      "24    2.0      3002316\n",
      "25   19.0      3002416\n",
      "26    4.0      3002516\n",
      "27    0.0      3002616\n",
      "28    0.0      3002716\n",
      "29    0.0      3002816\n",
      "30    1.0      3002916\n",
      "31    2.0      3003016\n",
      "32    0.0      3003116\n",
      "33    0.0      3003216\n",
      "34    1.0      3003316\n",
      "35    0.0      3003416\n",
      "36    1.0      3003516\n",
      "37    1.0      3003616\n",
      "38    3.0      3003716\n",
      "39    0.0      3003816\n",
      "40    8.0      3003916\n",
      "41    0.0      3004016\n",
      "42    0.0      3004116\n",
      "43    5.0      3004216\n",
      "44    0.0      3004316\n",
      "45    0.0      3004416\n",
      "46    1.0      3004516\n",
      "47    0.0      3004616\n",
      "48    0.0      3004716\n",
      "49   13.0      3004816\n",
      "(67177, 2)\n",
      "TT_S0          float64\n",
      "starting_nt      int64\n",
      "dtype: object\n",
      "sumos ilgis: 67177\n",
      "ranges ilgis: 67177\n",
      "Model input - NO SCALE\n",
      "       TT_S0  genes  transcripts  exons  CDSs\n",
      "0        0.0      0            0      0     0\n",
      "1        7.0      0            0      0     0\n",
      "2        1.0      0            0      0     0\n",
      "3        2.0      0            0      0     0\n",
      "4        0.0      0            0      0     0\n",
      "...      ...    ...          ...    ...   ...\n",
      "67172    0.0      0            0      0     0\n",
      "67173    0.0      0            0      0     0\n",
      "67174    0.0      0            0      0     0\n",
      "67175    1.0      0            0      0     0\n",
      "67176    1.0      0            0      0     0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "       TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0        0.0             0                   0             0            0\n",
      "1        7.0             0                   0             0            0\n",
      "2        1.0             0                   0             0            0\n",
      "3        2.0             0                   0             0            0\n",
      "4        0.0             0                   0             0            0\n",
      "...      ...           ...                 ...           ...          ...\n",
      "67172    0.0             0                   0             0            0\n",
      "67173    0.0             0                   0             0            0\n",
      "67174    0.0             0                   0             0            0\n",
      "67175    1.0             0                   0             0            0\n",
      "67176    1.0             0                   0             0            0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "0.20496598538190156\n",
      "[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 1 0 1 1 1 1 0 0 1 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 1 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0]\n",
      "Runnning scale 32\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0\n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "BINS  IntervalIndex([(3000016, 3000016], (3000016, 3000116], (3000116, 3000216],\n",
      "               (3000216, 3000316], (3000316, 3000416], (3000416, 3000516],\n",
      "               (3000516, 3000616], (3000616, 3000716], (3000716, 3000816],\n",
      "               (3000816, 3000916],\n",
      "               ...\n",
      "               (9716616, 9716716], (9716716, 9716816], (9716816, 9716916],\n",
      "               (9716916, 9717016], (9717016, 9717116], (9717116, 9717216],\n",
      "               (9717216, 9717316], (9717316, 9717416], (9717416, 9717516],\n",
      "               (9717516, 9717593]],\n",
      "              dtype='interval[int64, right]', length=67177)\n",
      "HEAD w BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0   \n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "\n",
      "         bin_offset_0  \n",
      "0  (3000016, 3000116]  \n",
      "1  (3000016, 3000116]  \n",
      "2  (3000016, 3000116]  \n",
      "3  (3000016, 3000116]  \n",
      "4  (3000016, 3000116]  \n",
      "TAIL w BINS\n",
      "       seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  9717555  9717555  18_9717555    1.0    0.0    0.0   \n",
      "699996       19  9717571  9717571  19_9717571    0.0    0.0    0.0   \n",
      "699997       19  9717581  9717581  19_9717581    0.0    0.0    0.0   \n",
      "699998       18  9717582  9717582  18_9717582    0.0    0.0    0.0   \n",
      "699999       18  9717593  9717593  18_9717593    0.0    0.0    0.0   \n",
      "\n",
      "              bin_offset_0  \n",
      "699995  (9717516, 9717593]  \n",
      "699996  (9717516, 9717593]  \n",
      "699997  (9717516, 9717593]  \n",
      "699998  (9717516, 9717593]  \n",
      "699999  (9717516, 9717593]  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:151: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "    TT_S0  starting_nt\n",
      "0     0.0      3000016\n",
      "1     7.0      3000016\n",
      "2     1.0      3000116\n",
      "3     2.0      3000216\n",
      "4     0.0      3000316\n",
      "5     0.0      3000416\n",
      "6     0.0      3000516\n",
      "7     0.0      3000616\n",
      "8     0.0      3000716\n",
      "9     0.0      3000816\n",
      "10    0.0      3000916\n",
      "11    2.0      3001016\n",
      "12    0.0      3001116\n",
      "13    0.0      3001216\n",
      "14    0.0      3001316\n",
      "15    1.0      3001416\n",
      "16    3.0      3001516\n",
      "17    3.0      3001616\n",
      "18    0.0      3001716\n",
      "19    0.0      3001816\n",
      "20    1.0      3001916\n",
      "21    0.0      3002016\n",
      "22    1.0      3002116\n",
      "23   10.0      3002216\n",
      "24    2.0      3002316\n",
      "25   19.0      3002416\n",
      "26    4.0      3002516\n",
      "27    0.0      3002616\n",
      "28    0.0      3002716\n",
      "29    0.0      3002816\n",
      "30    1.0      3002916\n",
      "31    2.0      3003016\n",
      "32    0.0      3003116\n",
      "33    0.0      3003216\n",
      "34    1.0      3003316\n",
      "35    0.0      3003416\n",
      "36    1.0      3003516\n",
      "37    1.0      3003616\n",
      "38    3.0      3003716\n",
      "39    0.0      3003816\n",
      "40    8.0      3003916\n",
      "41    0.0      3004016\n",
      "42    0.0      3004116\n",
      "43    5.0      3004216\n",
      "44    0.0      3004316\n",
      "45    0.0      3004416\n",
      "46    1.0      3004516\n",
      "47    0.0      3004616\n",
      "48    0.0      3004716\n",
      "49   13.0      3004816\n",
      "(67177, 2)\n",
      "TT_S0          float64\n",
      "starting_nt      int64\n",
      "dtype: object\n",
      "sumos ilgis: 67177\n",
      "ranges ilgis: 67177\n",
      "Model input - NO SCALE\n",
      "       TT_S0  genes  transcripts  exons  CDSs\n",
      "0        0.0      0            0      0     0\n",
      "1        7.0      0            0      0     0\n",
      "2        1.0      0            0      0     0\n",
      "3        2.0      0            0      0     0\n",
      "4        0.0      0            0      0     0\n",
      "...      ...    ...          ...    ...   ...\n",
      "67172    0.0      0            0      0     0\n",
      "67173    0.0      0            0      0     0\n",
      "67174    0.0      0            0      0     0\n",
      "67175    1.0      0            0      0     0\n",
      "67176    1.0      0            0      0     0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "       TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0        0.0             0                   0             0            0\n",
      "1        7.0             0                   0             0            0\n",
      "2        1.0             0                   0             0            0\n",
      "3        2.0             0                   0             0            0\n",
      "4        0.0             0                   0             0            0\n",
      "...      ...           ...                 ...           ...          ...\n",
      "67172    0.0             0                   0             0            0\n",
      "67173    0.0             0                   0             0            0\n",
      "67174    0.0             0                   0             0            0\n",
      "67175    1.0             0                   0             0            0\n",
      "67176    1.0             0                   0             0            0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "0.17092159518882952\n",
      "[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 1 0 1 1 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 1 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0]\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "for i in range(6):\n",
    "    scale = 2 ** i\n",
    "    print(\"Runnning scale\", scale)\n",
    "    evaluation = evaluator.evaluate(dataframe = chromatin_algos.algo5d(scale), model_name=f\"k-means 5 features scale={scale}\")\n",
    "    result = pd.concat([result, evaluation], axis=1)\n",
    "\n",
    "result.to_csv('../outputs/results1.csv', sep = '\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ca1280ed",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                         Length                      Length  \\\n",
      "count                    7301.0                      6456.0   \n",
      "mean                 188.590604                  177.850062   \n",
      "std                  174.998864                  157.257952   \n",
      "min                       100.0                       100.0   \n",
      "25%                       100.0                       100.0   \n",
      "50%                       100.0                       100.0   \n",
      "75%                       200.0                       200.0   \n",
      "max                      2400.0                      2000.0   \n",
      "ATACseq_overlap        0.697949                    0.668276   \n",
      "DNAseq_overlap         0.595329                    0.553813   \n",
      "genes_overlap          0.544833                    0.550809   \n",
      "percentage_open        0.141691                    0.118157   \n",
      "percentage_open_adj    0.204969                    0.170924   \n",
      "model_name                       k-means 5 features scale=1   \n",
      "min_open_signal            17.0                        19.0   \n",
      "max_open_signal           393.0                       393.0   \n",
      "min_closed_signal           0.0                         0.0   \n",
      "max_closed_signal          16.0                        18.0   \n",
      "\n",
      "                                         Length                      Length  \\\n",
      "count                                    7301.0                      7301.0   \n",
      "mean                                 188.590604                  188.590604   \n",
      "std                                  174.998864                  174.998864   \n",
      "min                                       100.0                       100.0   \n",
      "25%                                       100.0                       100.0   \n",
      "50%                                       100.0                       100.0   \n",
      "75%                                       200.0                       200.0   \n",
      "max                                      2400.0                      2400.0   \n",
      "ATACseq_overlap                        0.697949                    0.697949   \n",
      "DNAseq_overlap                         0.595329                    0.595329   \n",
      "genes_overlap                          0.544833                    0.544833   \n",
      "percentage_open                        0.141691                    0.141691   \n",
      "percentage_open_adj                    0.204969                    0.204969   \n",
      "model_name           k-means 5 features scale=2  k-means 5 features scale=4   \n",
      "min_open_signal                            17.0                        17.0   \n",
      "max_open_signal                           393.0                       393.0   \n",
      "min_closed_signal                           0.0                         0.0   \n",
      "max_closed_signal                          16.0                        16.0   \n",
      "\n",
      "                                         Length                       Length  \\\n",
      "count                                    7301.0                       7301.0   \n",
      "mean                                 188.590604                   188.590604   \n",
      "std                                  174.998864                   174.998864   \n",
      "min                                       100.0                        100.0   \n",
      "25%                                       100.0                        100.0   \n",
      "50%                                       100.0                        100.0   \n",
      "75%                                       200.0                        200.0   \n",
      "max                                      2400.0                       2400.0   \n",
      "ATACseq_overlap                        0.697949                     0.697949   \n",
      "DNAseq_overlap                         0.595329                     0.595329   \n",
      "genes_overlap                          0.544833                     0.544833   \n",
      "percentage_open                        0.141691                     0.141691   \n",
      "percentage_open_adj                    0.204969                     0.204969   \n",
      "model_name           k-means 5 features scale=8  k-means 5 features scale=16   \n",
      "min_open_signal                            17.0                         17.0   \n",
      "max_open_signal                           393.0                        393.0   \n",
      "min_closed_signal                           0.0                          0.0   \n",
      "max_closed_signal                          16.0                         16.0   \n",
      "\n",
      "                                          Length  \n",
      "count                                     6456.0  \n",
      "mean                                  177.850062  \n",
      "std                                   157.257952  \n",
      "min                                        100.0  \n",
      "25%                                        100.0  \n",
      "50%                                        100.0  \n",
      "75%                                        200.0  \n",
      "max                                       2000.0  \n",
      "ATACseq_overlap                         0.668276  \n",
      "DNAseq_overlap                          0.553813  \n",
      "genes_overlap                           0.550809  \n",
      "percentage_open                         0.118157  \n",
      "percentage_open_adj                     0.170924  \n",
      "model_name           k-means 5 features scale=32  \n",
      "min_open_signal                             19.0  \n",
      "max_open_signal                            393.0  \n",
      "min_closed_signal                            0.0  \n",
      "max_closed_signal                           18.0  \n"
     ]
    }
   ],
   "source": [
    "print(result)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
