{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "d001e5b1",
   "metadata": {},
   "outputs": [],
   "source": [
    "import chromatin_algos\n",
    "import evaluator"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "4328129e",
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "File \u001b[0;32m/opt/anaconda3/lib/python3.12/site-packages/pyreadr/librdata.pyx:229\u001b[0m, in \u001b[0;36mpyreadr.librdata.Parser._Parser__handle_column\u001b[0;34m()\u001b[0m\n",
      "File \u001b[0;32m/opt/anaconda3/lib/python3.12/enum.py:726\u001b[0m, in \u001b[0;36mEnumType.__call__\u001b[0;34m(cls, value, names, module, qualname, type, start, boundary, *values)\u001b[0m\n\u001b[1;32m    721\u001b[0m \u001b[38;5;250m    \u001b[39m\u001b[38;5;124;03m\"\"\"\u001b[39;00m\n\u001b[1;32m    722\u001b[0m \u001b[38;5;124;03m    classes/types should always be True.\u001b[39;00m\n\u001b[1;32m    723\u001b[0m \u001b[38;5;124;03m    \"\"\"\u001b[39;00m\n\u001b[1;32m    724\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;01mTrue\u001b[39;00m\n\u001b[0;32m--> 726\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21m__call__\u001b[39m(\u001b[38;5;28mcls\u001b[39m, value, names\u001b[38;5;241m=\u001b[39m_not_given, \u001b[38;5;241m*\u001b[39mvalues, module\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m, qualname\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m, \u001b[38;5;28mtype\u001b[39m\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m, start\u001b[38;5;241m=\u001b[39m\u001b[38;5;241m1\u001b[39m, boundary\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m):\n\u001b[1;32m    727\u001b[0m \u001b[38;5;250m    \u001b[39m\u001b[38;5;124;03m\"\"\"\u001b[39;00m\n\u001b[1;32m    728\u001b[0m \u001b[38;5;124;03m    Either returns an existing member, or creates a new enum class.\u001b[39;00m\n\u001b[1;32m    729\u001b[0m \n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m    751\u001b[0m \u001b[38;5;124;03m    `type`, if set, will be mixed in as the first base class.\u001b[39;00m\n\u001b[1;32m    752\u001b[0m \u001b[38;5;124;03m    \"\"\"\u001b[39;00m\n\u001b[1;32m    753\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mcls\u001b[39m\u001b[38;5;241m.\u001b[39m_member_map_:\n\u001b[1;32m    754\u001b[0m         \u001b[38;5;66;03m# simple value lookup if members exist\u001b[39;00m\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Exception ignored in: 'pyreadr.librdata._handle_column'\n",
      "Traceback (most recent call last):\n",
      "  File \"pyreadr/librdata.pyx\", line 229, in pyreadr.librdata.Parser._Parser__handle_column\n",
      "  File \"/opt/anaconda3/lib/python3.12/enum.py\", line 726, in __call__\n",
      "    def __call__(cls, value, names=_not_given, *values, module=None, qualname=None, type=None, start=1, boundary=None):\n",
      "\n",
      "KeyboardInterrupt: \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0\n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0\n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0\n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0\n",
      "4       18  3000040  3000040  18_3000040    0.0    0.0\n",
      "BINS  IntervalIndex([(3000016, 3000016], (3000016, 3000116], (3000116, 3000216],\n",
      "               (3000216, 3000316], (3000316, 3000416], (3000416, 3000516],\n",
      "               (3000516, 3000616], (3000616, 3000716], (3000716, 3000816],\n",
      "               (3000816, 3000916],\n",
      "               ...\n",
      "               (9716616, 9716716], (9716716, 9716816], (9716816, 9716916],\n",
      "               (9716916, 9717016], (9717016, 9717116], (9717116, 9717216],\n",
      "               (9717216, 9717316], (9717316, 9717416], (9717416, 9717516],\n",
      "               (9717516, 9717593]],\n",
      "              dtype='interval[int64, right]', length=67177)\n",
      "HEAD with BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1        bin_offset_0  \\\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0  (3000016, 3000116]   \n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0  (3000016, 3000116]   \n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0  (3000016, 3000116]   \n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0  (3000016, 3000116]   \n",
      "4       18  3000040  3000040  18_3000040    0.0    0.0  (3000016, 3000116]   \n",
      "\n",
      "   target_count  \n",
      "0             1  \n",
      "1             1  \n",
      "2             1  \n",
      "3             1  \n",
      "4             1  \n",
      "TAIL with BINS\n",
      "       seqnames    start      end       CG_ID  TT_S0  TT_S1  \\\n",
      "699995       18  9717555  9717555  18_9717555    0.0    0.0   \n",
      "699996       19  9717571  9717571  19_9717571    0.0    0.0   \n",
      "699997       19  9717581  9717581  19_9717581    0.0    0.0   \n",
      "699998       18  9717582  9717582  18_9717582    0.0    0.0   \n",
      "699999       18  9717593  9717593  18_9717593    0.0    0.0   \n",
      "\n",
      "              bin_offset_0  target_count  \n",
      "699995  (9717516, 9717593]             1  \n",
      "699996  (9717516, 9717593]             1  \n",
      "699997  (9717516, 9717593]             1  \n",
      "699998  (9717516, 9717593]             1  \n",
      "699999  (9717516, 9717593]             1  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:137: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "ename": "KeyError",
     "evalue": "\"['TT_S2'] not found in axis\"",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[7], line 4\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[38;5;28;01mimport\u001b[39;00m \u001b[38;5;21;01mos\u001b[39;00m\n\u001b[1;32m      2\u001b[0m os\u001b[38;5;241m.\u001b[39menviron[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mCOLUMNS\u001b[39m\u001b[38;5;124m'\u001b[39m]\u001b[38;5;241m=\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m300\u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[0;32m----> 4\u001b[0m result \u001b[38;5;241m=\u001b[39m evaluator\u001b[38;5;241m.\u001b[39mevaluate(merged_ranges \u001b[38;5;241m=\u001b[39m chromatin_algos\u001b[38;5;241m.\u001b[39malgo5d_aggregated(\u001b[38;5;241m1\u001b[39m, n_clusters\u001b[38;5;241m=\u001b[39m\u001b[38;5;241m10\u001b[39m))\n",
      "File \u001b[0;32m~/VU/bio-baigiamasis/scripts/chromatin_algos.py:387\u001b[0m, in \u001b[0;36malgo5d_aggregated\u001b[0;34m(scale, n_clusters, threshold, bin_offset)\u001b[0m\n\u001b[1;32m    386\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21malgo5d_aggregated\u001b[39m(scale \u001b[38;5;241m=\u001b[39m \u001b[38;5;241m1\u001b[39m, n_clusters \u001b[38;5;241m=\u001b[39m \u001b[38;5;241m2\u001b[39m, threshold \u001b[38;5;241m=\u001b[39m \u001b[38;5;241m1.5\u001b[39m, bin_offset \u001b[38;5;241m=\u001b[39m \u001b[38;5;241m0\u001b[39m):\n\u001b[0;32m--> 387\u001b[0m     model_output \u001b[38;5;241m=\u001b[39m algo5d(scale, n_clusters, bin_offset)\n\u001b[1;32m    388\u001b[0m     update_labels(model_output, threshold)\n\u001b[1;32m    389\u001b[0m     model_output\u001b[38;5;241m.\u001b[39mto_csv(\u001b[38;5;124m'\u001b[39m\u001b[38;5;124m../outputs/output_algo5d_aggregated.csv\u001b[39m\u001b[38;5;124m'\u001b[39m, sep \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m'\u001b[39m\u001b[38;5;130;01m\\t\u001b[39;00m\u001b[38;5;124m'\u001b[39m)\n",
      "File \u001b[0;32m~/VU/bio-baigiamasis/scripts/chromatin_algos.py:140\u001b[0m, in \u001b[0;36malgo5d\u001b[0;34m(scale, n_clusters, bin_offset)\u001b[0m\n\u001b[1;32m    137\u001b[0m sumos \u001b[38;5;241m=\u001b[39m df\u001b[38;5;241m.\u001b[39mgroupby(\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mbin_offset_\u001b[39m\u001b[38;5;124m'\u001b[39m \u001b[38;5;241m+\u001b[39m \u001b[38;5;28mstr\u001b[39m(bin_offset))\u001b[38;5;241m.\u001b[39msum()\u001b[38;5;241m.\u001b[39mreset_index()\n\u001b[1;32m    138\u001b[0m sumos[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mstarting_nt\u001b[39m\u001b[38;5;124m'\u001b[39m] \u001b[38;5;241m=\u001b[39m sumos[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mbin_offset_\u001b[39m\u001b[38;5;124m'\u001b[39m \u001b[38;5;241m+\u001b[39m \u001b[38;5;28mstr\u001b[39m(bin_offset)]\u001b[38;5;241m.\u001b[39mapply(\u001b[38;5;28;01mlambda\u001b[39;00m x: x\u001b[38;5;241m.\u001b[39mleft)\n\u001b[0;32m--> 140\u001b[0m sumos\u001b[38;5;241m.\u001b[39mdrop(columns\u001b[38;5;241m=\u001b[39mOTHER_SIGNALS, axis\u001b[38;5;241m=\u001b[39m\u001b[38;5;241m1\u001b[39m, inplace\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m)\n\u001b[1;32m    142\u001b[0m \u001b[38;5;28mprint\u001b[39m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mSUMOS\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[1;32m    143\u001b[0m \u001b[38;5;28mprint\u001b[39m(sumos[:\u001b[38;5;241m50\u001b[39m])\n",
      "File \u001b[0;32m/opt/anaconda3/lib/python3.12/site-packages/pandas/core/frame.py:5581\u001b[0m, in \u001b[0;36mDataFrame.drop\u001b[0;34m(self, labels, axis, index, columns, level, inplace, errors)\u001b[0m\n\u001b[1;32m   5433\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mdrop\u001b[39m(\n\u001b[1;32m   5434\u001b[0m     \u001b[38;5;28mself\u001b[39m,\n\u001b[1;32m   5435\u001b[0m     labels: IndexLabel \u001b[38;5;241m|\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m,\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m   5442\u001b[0m     errors: IgnoreRaise \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mraise\u001b[39m\u001b[38;5;124m\"\u001b[39m,\n\u001b[1;32m   5443\u001b[0m ) \u001b[38;5;241m-\u001b[39m\u001b[38;5;241m>\u001b[39m DataFrame \u001b[38;5;241m|\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[1;32m   5444\u001b[0m \u001b[38;5;250m    \u001b[39m\u001b[38;5;124;03m\"\"\"\u001b[39;00m\n\u001b[1;32m   5445\u001b[0m \u001b[38;5;124;03m    Drop specified labels from rows or columns.\u001b[39;00m\n\u001b[1;32m   5446\u001b[0m \n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m   5579\u001b[0m \u001b[38;5;124;03m            weight  1.0     0.8\u001b[39;00m\n\u001b[1;32m   5580\u001b[0m \u001b[38;5;124;03m    \"\"\"\u001b[39;00m\n\u001b[0;32m-> 5581\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28msuper\u001b[39m()\u001b[38;5;241m.\u001b[39mdrop(\n\u001b[1;32m   5582\u001b[0m         labels\u001b[38;5;241m=\u001b[39mlabels,\n\u001b[1;32m   5583\u001b[0m         axis\u001b[38;5;241m=\u001b[39maxis,\n\u001b[1;32m   5584\u001b[0m         index\u001b[38;5;241m=\u001b[39mindex,\n\u001b[1;32m   5585\u001b[0m         columns\u001b[38;5;241m=\u001b[39mcolumns,\n\u001b[1;32m   5586\u001b[0m         level\u001b[38;5;241m=\u001b[39mlevel,\n\u001b[1;32m   5587\u001b[0m         inplace\u001b[38;5;241m=\u001b[39minplace,\n\u001b[1;32m   5588\u001b[0m         errors\u001b[38;5;241m=\u001b[39merrors,\n\u001b[1;32m   5589\u001b[0m     )\n",
      "File \u001b[0;32m/opt/anaconda3/lib/python3.12/site-packages/pandas/core/generic.py:4788\u001b[0m, in \u001b[0;36mNDFrame.drop\u001b[0;34m(self, labels, axis, index, columns, level, inplace, errors)\u001b[0m\n\u001b[1;32m   4786\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m axis, labels \u001b[38;5;129;01min\u001b[39;00m axes\u001b[38;5;241m.\u001b[39mitems():\n\u001b[1;32m   4787\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m labels \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[0;32m-> 4788\u001b[0m         obj \u001b[38;5;241m=\u001b[39m obj\u001b[38;5;241m.\u001b[39m_drop_axis(labels, axis, level\u001b[38;5;241m=\u001b[39mlevel, errors\u001b[38;5;241m=\u001b[39merrors)\n\u001b[1;32m   4790\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m inplace:\n\u001b[1;32m   4791\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_update_inplace(obj)\n",
      "File \u001b[0;32m/opt/anaconda3/lib/python3.12/site-packages/pandas/core/generic.py:4830\u001b[0m, in \u001b[0;36mNDFrame._drop_axis\u001b[0;34m(self, labels, axis, level, errors, only_slice)\u001b[0m\n\u001b[1;32m   4828\u001b[0m         new_axis \u001b[38;5;241m=\u001b[39m axis\u001b[38;5;241m.\u001b[39mdrop(labels, level\u001b[38;5;241m=\u001b[39mlevel, errors\u001b[38;5;241m=\u001b[39merrors)\n\u001b[1;32m   4829\u001b[0m     \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[0;32m-> 4830\u001b[0m         new_axis \u001b[38;5;241m=\u001b[39m axis\u001b[38;5;241m.\u001b[39mdrop(labels, errors\u001b[38;5;241m=\u001b[39merrors)\n\u001b[1;32m   4831\u001b[0m     indexer \u001b[38;5;241m=\u001b[39m axis\u001b[38;5;241m.\u001b[39mget_indexer(new_axis)\n\u001b[1;32m   4833\u001b[0m \u001b[38;5;66;03m# Case for non-unique axis\u001b[39;00m\n\u001b[1;32m   4834\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n",
      "File \u001b[0;32m/opt/anaconda3/lib/python3.12/site-packages/pandas/core/indexes/base.py:7070\u001b[0m, in \u001b[0;36mIndex.drop\u001b[0;34m(self, labels, errors)\u001b[0m\n\u001b[1;32m   7068\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m mask\u001b[38;5;241m.\u001b[39many():\n\u001b[1;32m   7069\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m errors \u001b[38;5;241m!=\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mignore\u001b[39m\u001b[38;5;124m\"\u001b[39m:\n\u001b[0;32m-> 7070\u001b[0m         \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mKeyError\u001b[39;00m(\u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mlabels[mask]\u001b[38;5;241m.\u001b[39mtolist()\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m not found in axis\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[1;32m   7071\u001b[0m     indexer \u001b[38;5;241m=\u001b[39m indexer[\u001b[38;5;241m~\u001b[39mmask]\n\u001b[1;32m   7072\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mdelete(indexer)\n",
      "\u001b[0;31mKeyError\u001b[0m: \"['TT_S2'] not found in axis\""
     ]
    }
   ],
   "source": [
    "import os\n",
    "os.environ['COLUMNS']=\"300\"\n",
    "\n",
    "result = evaluator.evaluate(merged_ranges = chromatin_algos.algo5d_aggregated(1, n_clusters=10))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "6201148c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0\n",
      "BINS  IntervalIndex([  (3000018, 3000018],   (3000018, 3000118],\n",
      "                 (3000118, 3000218],   (3000218, 3000318],\n",
      "                 (3000318, 3000418],   (3000418, 3000518],\n",
      "                 (3000518, 3000618],   (3000618, 3000718],\n",
      "                 (3000718, 3000818],   (3000818, 3000918],\n",
      "               ...\n",
      "               (18260718, 18260818], (18260818, 18260918],\n",
      "               (18260918, 18261018], (18261018, 18261118],\n",
      "               (18261118, 18261218], (18261218, 18261318],\n",
      "               (18261318, 18261418], (18261418, 18261518],\n",
      "               (18261518, 18261618], (18261618, 18261710]],\n",
      "              dtype='interval[int64, right]', length=152618)\n",
      "HEAD with BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0   \n",
      "\n",
      "         bin_offset_0  target_count  \n",
      "0  (3000018, 3000118]             1  \n",
      "1  (3000018, 3000118]             1  \n",
      "2  (3000018, 3000118]             1  \n",
      "3  (3000018, 3000118]             1  \n",
      "4  (3000018, 3000118]             1  \n",
      "TAIL with BINS\n",
      "       seqnames     start       end        CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  18261611  18261611  18_18261611    0.0    0.0    0.0   \n",
      "699996       18  18261615  18261615  18_18261615    0.0    0.0    0.0   \n",
      "699997       18  18261682  18261682  18_18261682    0.0    0.0    0.0   \n",
      "699998       18  18261702  18261702  18_18261702    4.0    0.0    0.0   \n",
      "699999       18  18261710  18261710  18_18261710    0.0    0.0    0.0   \n",
      "\n",
      "                bin_offset_0  target_count  \n",
      "699995  (18261518, 18261618]             1  \n",
      "699996  (18261518, 18261618]             1  \n",
      "699997  (18261618, 18261710]             1  \n",
      "699998  (18261618, 18261710]             1  \n",
      "699999  (18261618, 18261710]             1  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:137: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "          bin_offset_0  TT_S0  target_count  starting_nt\n",
      "0   (3000018, 3000018]    0.0             0      3000018\n",
      "1   (3000018, 3000118]    7.0            11      3000018\n",
      "2   (3000118, 3000218]    1.0             4      3000118\n",
      "3   (3000218, 3000318]    2.0             4      3000218\n",
      "4   (3000318, 3000418]    0.0             4      3000318\n",
      "5   (3000418, 3000518]    0.0             2      3000418\n",
      "6   (3000518, 3000618]    0.0             2      3000518\n",
      "7   (3000618, 3000718]    0.0             4      3000618\n",
      "8   (3000718, 3000818]    0.0             7      3000718\n",
      "9   (3000818, 3000918]    0.0             3      3000818\n",
      "10  (3000918, 3001018]    0.0             3      3000918\n",
      "11  (3001018, 3001118]    2.0             6      3001018\n",
      "12  (3001118, 3001218]    0.0             2      3001118\n",
      "13  (3001218, 3001318]    0.0             4      3001218\n",
      "14  (3001318, 3001418]    0.0             3      3001318\n",
      "15  (3001418, 3001518]    1.0             4      3001418\n",
      "16  (3001518, 3001618]    3.0            10      3001518\n",
      "17  (3001618, 3001718]    3.0             7      3001618\n",
      "18  (3001718, 3001818]    0.0             6      3001718\n",
      "19  (3001818, 3001918]    0.0             2      3001818\n",
      "20  (3001918, 3002018]    1.0             3      3001918\n",
      "21  (3002018, 3002118]    0.0             2      3002018\n",
      "22  (3002118, 3002218]    1.0             4      3002118\n",
      "23  (3002218, 3002318]   10.0             6      3002218\n",
      "24  (3002318, 3002418]    2.0             1      3002318\n",
      "25  (3002418, 3002518]   19.0             5      3002418\n",
      "26  (3002518, 3002618]    4.0             2      3002518\n",
      "27  (3002618, 3002718]    0.0             3      3002618\n",
      "28  (3002718, 3002818]    0.0             1      3002718\n",
      "29  (3002818, 3002918]    0.0             2      3002818\n",
      "30  (3002918, 3003018]    1.0             6      3002918\n",
      "31  (3003018, 3003118]    2.0             4      3003018\n",
      "32  (3003118, 3003218]    0.0             0      3003118\n",
      "33  (3003218, 3003318]    0.0             1      3003218\n",
      "34  (3003318, 3003418]    1.0             8      3003318\n",
      "35  (3003418, 3003518]    0.0             3      3003418\n",
      "36  (3003518, 3003618]    1.0             2      3003518\n",
      "37  (3003618, 3003718]    3.0             4      3003618\n",
      "38  (3003718, 3003818]    1.0             6      3003718\n",
      "39  (3003818, 3003918]    0.0             3      3003818\n",
      "40  (3003918, 3004018]    8.0             6      3003918\n",
      "41  (3004018, 3004118]    0.0             0      3004018\n",
      "42  (3004118, 3004218]    0.0             1      3004118\n",
      "43  (3004218, 3004318]    5.0             6      3004218\n",
      "44  (3004318, 3004418]    0.0             1      3004318\n",
      "45  (3004418, 3004518]    0.0             1      3004418\n",
      "46  (3004518, 3004618]    1.0             7      3004518\n",
      "47  (3004618, 3004718]    0.0             0      3004618\n",
      "48  (3004718, 3004818]    0.0             1      3004718\n",
      "49  (3004818, 3004918]   13.0             8      3004818\n",
      "(152618, 4)\n",
      "bin_offset_0    category\n",
      "TT_S0            float64\n",
      "target_count       int64\n",
      "starting_nt        int64\n",
      "dtype: object\n",
      "sumos ilgis: 152618\n",
      "ranges ilgis: 152618\n",
      "Model input - NO SCALE\n",
      "        TT_S0  genes  transcripts  exons  CDSs\n",
      "0         0.0      0            0      0     0\n",
      "1         7.0      0            0      0     0\n",
      "2         1.0      0            0      0     0\n",
      "3         2.0      0            0      0     0\n",
      "4         0.0      0            0      0     0\n",
      "...       ...    ...          ...    ...   ...\n",
      "152613   10.0      0            0      0     0\n",
      "152614    3.0      0            0      0     0\n",
      "152615    4.0      0            0      0     0\n",
      "152616    0.0      0            0      0     0\n",
      "152617    4.0      0            0      0     0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "        TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0         0.0             0                   0             0            0\n",
      "1         7.0             0                   0             0            0\n",
      "2         1.0             0                   0             0            0\n",
      "3         2.0             0                   0             0            0\n",
      "4         0.0             0                   0             0            0\n",
      "...       ...           ...                 ...           ...          ...\n",
      "152613   10.0             0                   0             0            0\n",
      "152614    3.0             0                   0             0            0\n",
      "152615    4.0             0                   0             0            0\n",
      "152616    0.0             0                   0             0            0\n",
      "152617    4.0             0                   0             0            0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "0.010313331323959165\n",
      "[0 2 0 6 0 0 0 0 0 0 0 6 0 0 0 0 6 6 0 0 0 0 0 3 6 7 6 0 0 0 0 6 0 0 0 0 0\n",
      " 6 0 0 3 0 0 2 0 0 0 0 0 3 0 9 7 9 0 7 3 8 8 4 4 0 7 7 6 0 0 2 0 0 2 6 0 2\n",
      " 0 0 6 0 0 0 6 3 6 0 6 6 6 6 2 0 0 3 7 6 6 0 6 0 6 3 3 3 1 1 7 3 2 3 1 6 3\n",
      " 0 7 6 0 0 6 0 0 0 0 0 2 0 0 7 0 7 7 3 7 0 2 3 6 0 2 3 6 0 0 0 0 0 6 0 0 3\n",
      " 3 0 3 6 0 0 2 0 3 6 6 0 7 0 3 0 0 3 3 6 0 0 0 0 0 0 6 0 0 0 0 0 2 3 6 6 0\n",
      " 0 6 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 3 7 0 0 0 0 6 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 6 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 6 0 0\n",
      " 0 0 0 0 0 0 6 0 0 6 0 6 0 6 6 6 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 2 0 0 0 0 6 6 0 6 0 0 0 0 6 0 2 2 6 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 2 0 6 0 3 0 0 0 0 0 0 0 0 0 3 7 0 3 7 3 2 6 0 6 2 3\n",
      " 6 0 2 0 0 0 3 2 7 6 3 2 0 6 0 0 6 0 2 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 6 0 0 0 2 0 0 0 0 0 6 0 0 0 0 0 0 0 6 3 0 0 0 2 2 6 0 6 2 2 6 3 6 6 6 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 7 2 0 0 0 2 0 0 3 0 6 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 6 0 6 6 0 0 0 0 0 0 0 0 0 0 0 0 0 6 0 0 0 2 0 0 0 0 6 2 0 0 0 0 2\n",
      " 0 0 6 0 0 0 0 0 0 6 3 0 0 6 2 6 0 0 0 0 0 2 0 6 0 0 0 0 0 6 2 0 0 2 6 0 2\n",
      " 0 2 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 6 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 6 6 3 6 6 6 0 6 0 0 0 0 0 0 0 0 6 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 6 0 0 0 0 6 6 0 0 0\n",
      " 0 0 0 0 0 0 0 6 0 0 0 0 0 6 6 0 0 0 0 0 6 0 0 6 0 0 0 6 3 2 3 0 6 0 6 6 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 6 0 0 0 6 0 0 6 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 6 0 0 0 0 6 0 6 2 0 0 0 0 0 6 6 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 6 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 6 0 0 0 0 6 2 0 0 0 0 0 0 0 0 0\n",
      " 2 6 2 2 6 6 0 0 0 0 0 0 0 0 0 6 0 0 0 6 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 6 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 6 0 2 6 0 0 0 0 0 0\n",
      " 0]\n",
      "threshold density 0.979095\n",
      "cluster signal 16270.0\n",
      "cluster target count 300673\n",
      "!!cluster density 0.054111942209643035\n",
      "cluster signal 89679.0\n",
      "cluster target count 92721\n",
      "!!cluster density 0.9671918982754716\n",
      "cluster signal 87190.0\n",
      "cluster target count 157019\n",
      "!!cluster density 0.5552831186034811\n",
      "cluster signal 125621.0\n",
      "cluster target count 89557\n",
      "!!cluster density 1.402693256808513\n",
      "cluster signal 72124.0\n",
      "cluster target count 36924\n",
      "!!cluster density 1.9533095005958185\n",
      "cluster signal 17768.0\n",
      "cluster target count 5725\n",
      "!!cluster density 3.103580786026201\n",
      "cluster signal 2179.0\n",
      "cluster target count 338\n",
      "!!cluster density 6.446745562130178\n",
      "cluster signal 664.0\n",
      "cluster target count 11\n",
      "!!cluster density 60.36363636363637\n",
      "cluster signal 38430.0\n",
      "cluster target count 15240\n",
      "!!cluster density 2.5216535433070866\n",
      "cluster signal 6986.0\n",
      "cluster target count 1792\n",
      "!!cluster density 3.8984375\n",
      "\n",
      "Labels mapping\n",
      " {0: 0, 2: 0, 6: 0, 3: 1, 7: 1, 9: 1, 8: 1, 4: 1, 1: 1, 5: 1}\n",
      "DEBUG: 133376 19242 \n",
      "\n",
      " +----------------------+-------------+----------------+-------+\n",
      "| bin_offset_0         | TT_S0       | target_count   | +9    |\n",
      "| (category)           | (float64)   | (int64)        | ...   |\n",
      "|----------------------+-------------+----------------+-------|\n",
      "| (3000018, 3000018]   | 0.0         | 0              | ...   |\n",
      "| (3000018, 3000118]   | 7.0         | 11             | ...   |\n",
      "| (3000118, 3000218]   | 1.0         | 4              | ...   |\n",
      "| (3000218, 3000318]   | 2.0         | 4              | ...   |\n",
      "| ...                  | ...         | ...            | ...   |\n",
      "| (18261318, 18261418] | 3.0         | 15             | ...   |\n",
      "| (18261418, 18261518] | 4.0         | 4              | ...   |\n",
      "| (18261518, 18261618] | 0.0         | 3              | ...   |\n",
      "| (18261618, 18261710] | 4.0         | 3              | ...   |\n",
      "+----------------------+-------------+----------------+-------+\n",
      "Unstranded PyRanges object has 133,376 rows and 12 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "9 hidden columns: starting_nt, Chromosome, Start, End, genes, transcripts, ... (+ 3 more.)\n",
      "count                        13118.0\n",
      "mean                      146.683946\n",
      "std                       113.175202\n",
      "min                            100.0\n",
      "25%                            100.0\n",
      "50%                            100.0\n",
      "75%                            200.0\n",
      "max                           2500.0\n",
      "mode                           100.0\n",
      "ATACseq_overlap             0.792635\n",
      "ATACseq_ranges_overlap      0.030302\n",
      "ATACseq_Jaccard             0.030064\n",
      "DNAseq_overlap              0.696856\n",
      "DNAseq_ranges_overlap       0.043802\n",
      "DNAseq_Jaccard              0.042983\n",
      "genes_overlap               0.590862\n",
      "percentage_open             0.105368\n",
      "percentage_open_adj          0.12608\n",
      "model_name                          \n",
      "Name: Length, dtype: object\n"
     ]
    }
   ],
   "source": [
    "import generate_consensus\n",
    "\n",
    "ranges = chromatin_algos.algo5d_aggregated(1, 10)\n",
    "result = evaluator.evaluate(merged_ranges = ranges)\n",
    "print(result)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "6b5c5de3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "count                        13118.0\n",
      "mean                      146.683946\n",
      "std                       113.175202\n",
      "min                            100.0\n",
      "25%                            100.0\n",
      "50%                            100.0\n",
      "75%                            200.0\n",
      "max                           2500.0\n",
      "mode                           100.0\n",
      "ATACseq_overlap             0.792635\n",
      "ATACseq_ranges_overlap      0.030302\n",
      "ATACseq_Jaccard             0.030064\n",
      "DNAseq_overlap              0.696856\n",
      "DNAseq_ranges_overlap       0.043802\n",
      "DNAseq_Jaccard              0.042983\n",
      "genes_overlap               0.590862\n",
      "percentage_open             0.105368\n",
      "percentage_open_adj          0.12608\n",
      "model_name                          \n",
      "Name: Length, dtype: object\n"
     ]
    }
   ],
   "source": [
    "result = evaluator.evaluate(merged_ranges = ranges)\n",
    "\n",
    "print(result)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "44720aa9",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "       seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0            19  3000017  3000017  19_3000017    0.0    0.0    0.0\n",
      "1            18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "2            18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "3            18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "4            18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "...         ...      ...      ...         ...    ...    ...    ...\n",
      "699995       18  9717555  9717555  18_9717555    1.0    0.0    0.0\n",
      "699996       19  9717571  9717571  19_9717571    0.0    0.0    0.0\n",
      "699997       19  9717581  9717581  19_9717581    0.0    0.0    0.0\n",
      "699998       18  9717582  9717582  18_9717582    0.0    0.0    0.0\n",
      "699999       18  9717593  9717593  18_9717593    0.0    0.0    0.0\n",
      "\n",
      "[700000 rows x 7 columns]\n"
     ]
    }
   ],
   "source": [
    "import enrichment\n",
    "import pyreadr\n",
    "\n",
    "targets_df = pyreadr.read_r('../inputs/subset1.rds')[None]\n",
    "\n",
    "print(targets_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "bdd299ab",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/enrichment.py:19: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  summed_ranges = targets_df.groupby('ranges_bin')[[SIGNAL_COLUMN, 'target_count']].sum()\n",
      "/Users/lukas/VU/bio-baigiamasis/scripts/enrichment.py:19: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  summed_ranges = targets_df.groupby('ranges_bin')[[SIGNAL_COLUMN, 'target_count']].sum()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dataframe normalized signal: 0.9693985714285714\n",
      "              ranges_bin  norm_signal  pre_signal  post_signal  \\\n",
      "0     [3005116, 3005316)     3.434783    2.000000     0.600000   \n",
      "1     [3005616, 3006016)    21.571429    0.928571     0.000000   \n",
      "2     [3010116, 3010316)     2.125000    0.722222     1.400000   \n",
      "3     [3010716, 3010816)     2.454545    1.250000     0.400000   \n",
      "4     [3113516, 3113616)     1.733333    0.750000     0.578947   \n",
      "...                  ...          ...         ...          ...   \n",
      "5669  [9707616, 9707716)     1.461538    0.653846     0.823529   \n",
      "5670  [9708016, 9708216)     3.619048    0.666667     1.454545   \n",
      "5671  [9708316, 9708416)     3.000000    1.454545     0.727273   \n",
      "5672  [9708516, 9708816)     2.384615    0.727273     1.000000   \n",
      "5673  [9709016, 9709116)     2.333333    0.375000     0.142857   \n",
      "\n",
      "      surround_signal  enrichment_factor1  enrichment_factor2  \n",
      "0            2.600000            1.321070            3.543210  \n",
      "1            0.928571           23.230769           22.252383  \n",
      "2            2.122222            1.001309            2.192081  \n",
      "3            1.650000            1.487603            2.532029  \n",
      "4            1.328947            1.304290            1.788050  \n",
      "...               ...                 ...                 ...  \n",
      "5669         1.477376            0.989280            1.507675  \n",
      "5670         2.121212            1.706122            3.733292  \n",
      "5671         2.181818            1.375000            3.094702  \n",
      "5672         1.727273            1.380567            2.459892  \n",
      "5673         0.517857            4.505747            2.406991  \n",
      "\n",
      "[5674 rows x 7 columns]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/enrichment.py:19: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  summed_ranges = targets_df.groupby('ranges_bin')[[SIGNAL_COLUMN, 'target_count']].sum()\n"
     ]
    }
   ],
   "source": [
    "import enrichment\n",
    "\n",
    "print(enrichment.calculate_enrichment_factors_file(ranges_filepath='../outputs/test_ranges.tsv', targets_filepath='../inputs/subset1.rds'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "d67fb278",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/enrichment.py:19: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  summed_ranges = targets_df.groupby('ranges_bin')[[SIGNAL_COLUMN, 'target_count']].sum()\n",
      "/Users/lukas/VU/bio-baigiamasis/scripts/enrichment.py:19: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  summed_ranges = targets_df.groupby('ranges_bin')[[SIGNAL_COLUMN, 'target_count']].sum()\n",
      "/Users/lukas/VU/bio-baigiamasis/scripts/enrichment.py:19: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  summed_ranges = targets_df.groupby('ranges_bin')[[SIGNAL_COLUMN, 'target_count']].sum()\n",
      "/Users/lukas/VU/bio-baigiamasis/scripts/enrichment.py:19: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  summed_ranges = targets_df.groupby('ranges_bin')[[SIGNAL_COLUMN, 'target_count']].sum()\n"
     ]
    },
    {
     "ename": "KeyError",
     "evalue": "'pre_signal'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "File \u001b[0;32m/opt/anaconda3/lib/python3.12/site-packages/pandas/core/indexes/base.py:3805\u001b[0m, in \u001b[0;36mIndex.get_loc\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m   3804\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m-> 3805\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_engine\u001b[38;5;241m.\u001b[39mget_loc(casted_key)\n\u001b[1;32m   3806\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mKeyError\u001b[39;00m \u001b[38;5;28;01mas\u001b[39;00m err:\n",
      "File \u001b[0;32mindex.pyx:167\u001b[0m, in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[0;34m()\u001b[0m\n",
      "File \u001b[0;32mindex.pyx:191\u001b[0m, in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[0;34m()\u001b[0m\n",
      "File \u001b[0;32mindex.pyx:234\u001b[0m, in \u001b[0;36mpandas._libs.index.IndexEngine._get_loc_duplicates\u001b[0;34m()\u001b[0m\n",
      "File \u001b[0;32mindex.pyx:242\u001b[0m, in \u001b[0;36mpandas._libs.index.IndexEngine._maybe_get_bool_indexer\u001b[0;34m()\u001b[0m\n",
      "File \u001b[0;32mindex.pyx:134\u001b[0m, in \u001b[0;36mpandas._libs.index._unpack_bool_indexer\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;31mKeyError\u001b[0m: 'pre_signal'",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[6], line 2\u001b[0m\n\u001b[1;32m      1\u001b[0m enriched \u001b[38;5;241m=\u001b[39m enrichment\u001b[38;5;241m.\u001b[39mcalculate_norm_signal(ranges, targets_df)\n\u001b[0;32m----> 2\u001b[0m surround \u001b[38;5;241m=\u001b[39m enrichment\u001b[38;5;241m.\u001b[39mcalculate_surround_signal(ranges, targets_df)\n\u001b[1;32m      4\u001b[0m \u001b[38;5;28mprint\u001b[39m(enriched)\n\u001b[1;32m      5\u001b[0m \u001b[38;5;28mprint\u001b[39m(surround)\n",
      "File \u001b[0;32m~/VU/bio-baigiamasis/scripts/enrichment.py:54\u001b[0m, in \u001b[0;36mcalculate_surround_signal\u001b[0;34m(ranges, targets_df)\u001b[0m\n\u001b[1;32m     51\u001b[0m ranges_signal \u001b[38;5;241m=\u001b[39m pd\u001b[38;5;241m.\u001b[39mconcat([ranges_signal\u001b[38;5;241m.\u001b[39mreset_index(drop\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m), pre_signal\u001b[38;5;241m.\u001b[39mreset_index(drop\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m)], axis\u001b[38;5;241m=\u001b[39m\u001b[38;5;241m1\u001b[39m)\n\u001b[1;32m     52\u001b[0m ranges_signal \u001b[38;5;241m=\u001b[39m pd\u001b[38;5;241m.\u001b[39mconcat([ranges_signal\u001b[38;5;241m.\u001b[39mreset_index(drop\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m), post_signal\u001b[38;5;241m.\u001b[39mreset_index(drop\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m)], axis\u001b[38;5;241m=\u001b[39m\u001b[38;5;241m1\u001b[39m)\n\u001b[0;32m---> 54\u001b[0m ranges_signal[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124msurround_signal\u001b[39m\u001b[38;5;124m'\u001b[39m] \u001b[38;5;241m=\u001b[39m ranges_signal[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mpre_signal\u001b[39m\u001b[38;5;124m'\u001b[39m] \u001b[38;5;241m+\u001b[39m ranges_signal[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mpost_signal\u001b[39m\u001b[38;5;124m'\u001b[39m]\n\u001b[1;32m     56\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m ranges_signal\n",
      "File \u001b[0;32m/opt/anaconda3/lib/python3.12/site-packages/pandas/core/frame.py:4102\u001b[0m, in \u001b[0;36mDataFrame.__getitem__\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m   4100\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mcolumns\u001b[38;5;241m.\u001b[39mnlevels \u001b[38;5;241m>\u001b[39m \u001b[38;5;241m1\u001b[39m:\n\u001b[1;32m   4101\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_getitem_multilevel(key)\n\u001b[0;32m-> 4102\u001b[0m indexer \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mcolumns\u001b[38;5;241m.\u001b[39mget_loc(key)\n\u001b[1;32m   4103\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m is_integer(indexer):\n\u001b[1;32m   4104\u001b[0m     indexer \u001b[38;5;241m=\u001b[39m [indexer]\n",
      "File \u001b[0;32m/opt/anaconda3/lib/python3.12/site-packages/pandas/core/indexes/base.py:3812\u001b[0m, in \u001b[0;36mIndex.get_loc\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m   3807\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(casted_key, \u001b[38;5;28mslice\u001b[39m) \u001b[38;5;129;01mor\u001b[39;00m (\n\u001b[1;32m   3808\u001b[0m         \u001b[38;5;28misinstance\u001b[39m(casted_key, abc\u001b[38;5;241m.\u001b[39mIterable)\n\u001b[1;32m   3809\u001b[0m         \u001b[38;5;129;01mand\u001b[39;00m \u001b[38;5;28many\u001b[39m(\u001b[38;5;28misinstance\u001b[39m(x, \u001b[38;5;28mslice\u001b[39m) \u001b[38;5;28;01mfor\u001b[39;00m x \u001b[38;5;129;01min\u001b[39;00m casted_key)\n\u001b[1;32m   3810\u001b[0m     ):\n\u001b[1;32m   3811\u001b[0m         \u001b[38;5;28;01mraise\u001b[39;00m InvalidIndexError(key)\n\u001b[0;32m-> 3812\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mKeyError\u001b[39;00m(key) \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01merr\u001b[39;00m\n\u001b[1;32m   3813\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mTypeError\u001b[39;00m:\n\u001b[1;32m   3814\u001b[0m     \u001b[38;5;66;03m# If we have a listlike key, _check_indexing_error will raise\u001b[39;00m\n\u001b[1;32m   3815\u001b[0m     \u001b[38;5;66;03m#  InvalidIndexError. Otherwise we fall through and re-raise\u001b[39;00m\n\u001b[1;32m   3816\u001b[0m     \u001b[38;5;66;03m#  the TypeError.\u001b[39;00m\n\u001b[1;32m   3817\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_check_indexing_error(key)\n",
      "\u001b[0;31mKeyError\u001b[0m: 'pre_signal'"
     ]
    }
   ],
   "source": [
    "enriched = enrichment.calculate_norm_signal(ranges, targets_df)\n",
    "surround = enrichment.calculate_surround_signal(ranges, targets_df)\n",
    "\n",
    "print(enriched)\n",
    "print(surround)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "49f3ca45",
   "metadata": {},
   "outputs": [],
   "source": [
    "ranges.df.to_csv('../outputs/test_ranges.tsv', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "8de8e4e8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "+--------------+-----------+-----------+-----------+\n",
      "| Chromosome   | Start     | End       | Length    |\n",
      "| (category)   | (int64)   | (int64)   | (int64)   |\n",
      "|--------------+-----------+-----------+-----------|\n",
      "| 18           | 3005016   | 3005116   | 200       |\n",
      "| 18           | 3005516   | 3005616   | 400       |\n",
      "| 18           | 3010016   | 3010116   | 200       |\n",
      "| 18           | 3010616   | 3010716   | 100       |\n",
      "| ...          | ...       | ...       | ...       |\n",
      "| 18           | 9707516   | 9707616   | 100       |\n",
      "| 18           | 9707916   | 9708016   | 200       |\n",
      "| 18           | 9708416   | 9708516   | 300       |\n",
      "| 18           | 9708916   | 9709016   | 100       |\n",
      "+--------------+-----------+-----------+-----------+\n",
      "Unstranded PyRanges object has 4,985 rows and 4 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "+--------------+-----------+-----------+-----------+\n",
      "| Chromosome   | Start     | End       | Length    |\n",
      "| (category)   | (int64)   | (int64)   | (int64)   |\n",
      "|--------------+-----------+-----------+-----------|\n",
      "| 18           | 3005116   | 3005316   | 200       |\n",
      "| 18           | 3005616   | 3006016   | 400       |\n",
      "| 18           | 3010116   | 3010316   | 200       |\n",
      "| 18           | 3010716   | 3010816   | 100       |\n",
      "| ...          | ...       | ...       | ...       |\n",
      "| 18           | 9707616   | 9707716   | 100       |\n",
      "| 18           | 9708016   | 9708216   | 200       |\n",
      "| 18           | 9708516   | 9708816   | 300       |\n",
      "| 18           | 9709016   | 9709116   | 100       |\n",
      "+--------------+-----------+-----------+-----------+\n",
      "Unstranded PyRanges object has 4,985 rows and 4 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n"
     ]
    }
   ],
   "source": [
    "import enrichment\n",
    "\n",
    "print(enrichment.get_pre_regions(ranges))\n",
    "print(ranges)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "fa21a536",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "+--------------+-----------+-----------+-----------+\n",
      "| Chromosome   | Start     | End       | Length    |\n",
      "| (category)   | (int64)   | (int64)   | (int64)   |\n",
      "|--------------+-----------+-----------+-----------|\n",
      "| 18           | 3002416   | 3002516   | 100       |\n",
      "| 18           | 3005116   | 3005316   | 200       |\n",
      "| 18           | 3005416   | 3005516   | 100       |\n",
      "| 18           | 3005616   | 3006016   | 400       |\n",
      "| ...          | ...       | ...       | ...       |\n",
      "| 18           | 9708016   | 9708216   | 200       |\n",
      "| 18           | 9708316   | 9708416   | 100       |\n",
      "| 18           | 9708516   | 9708816   | 300       |\n",
      "| 18           | 9709016   | 9709116   | 100       |\n",
      "+--------------+-----------+-----------+-----------+\n",
      "Unstranded PyRanges object has 6,883 rows and 4 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "os.environ['COLUMNS']=\"300\"\n",
    "print(ranges)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "f475e092",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "24\n",
      "51.0\n"
     ]
    }
   ],
   "source": [
    "test = targets_df[(targets_df[\"start\"] >= 3010116) & (targets_df[\"start\"] <= 3010316)]\n",
    "#[3010116, 3010316]   51.0            24\n",
    "\n",
    "print(test['target_count'].sum())\n",
    "print(test['TT_S0'].sum())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "9435baf8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1615\n",
      "0\n"
     ]
    }
   ],
   "source": [
    "import pyranges as pr\n",
    "\n",
    "ranges.Length = ranges.End - ranges.Start\n",
    "ranges_df = ranges.df\n",
    "ranges_df['']\n",
    "\n",
    "print(len(ranges[ranges.Length < 75]))\n",
    "ranges2 = evaluator.process_ranges(ranges)\n",
    "ranges2.Length = ranges2.End - ranges2.Start\n",
    "print(len(ranges2[ranges2.Length < 75]))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "a929baa0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "count                         8182.0\n",
      "mean                      182.306282\n",
      "std                       197.351081\n",
      "min                              5.0\n",
      "25%                             80.0\n",
      "50%                            115.0\n",
      "75%                            210.0\n",
      "max                           2085.0\n",
      "ATACseq_overlap             0.751796\n",
      "ATACseq_ranges_overlap      0.013469\n",
      "DNAseq_overlap              0.639008\n",
      "DNAseq_ranges_overlap       0.022784\n",
      "genes_overlap                0.54393\n",
      "percentage_open             0.153498\n",
      "percentage_open_adj         0.222048\n",
      "model_name                          \n",
      "Name: Length, dtype: object\n"
     ]
    }
   ],
   "source": [
    "print(evaluator.evaluate(ranges))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "61959364",
   "metadata": {},
   "outputs": [],
   "source": [
    "def calculate_gaps(pyranges_obj):\n",
    "    \"\"\"\n",
    "    Calculate gaps between ranges in a PyRanges object.\n",
    "\n",
    "    Parameters:\n",
    "        pyranges_obj (pr.PyRanges): A PyRanges object with genomic ranges.\n",
    "\n",
    "    Returns:\n",
    "        pr.PyRanges: A PyRanges object containing the gaps.\n",
    "    \"\"\"\n",
    "    # Convert PyRanges to a DataFrame\n",
    "    df = pyranges_obj.df\n",
    "    \n",
    "    # Sort the ranges by Chromosome and Start position\n",
    "    df = df.sort_values([\"Chromosome\", \"Start\"]).reset_index(drop=True)\n",
    "    \n",
    "    # Shift End and Start columns to calculate gaps\n",
    "    prev_end = df.groupby(\"Chromosome\")[\"End\"].shift(0)   # Current End\n",
    "    next_start = df.groupby(\"Chromosome\")[\"Start\"].shift(-1)  # Next Start\n",
    "    \n",
    "    # Identify gaps where End < Next Start\n",
    "    gaps_df = df[(prev_end < next_start)].assign(Start=prev_end, End=next_start)[[\"Chromosome\", \"Start\", \"End\"]]\n",
    "    \n",
    "    # Drop NaN rows caused by shifting and ensure integer type\n",
    "    gaps_df = gaps_df.dropna().astype({\"Start\": \"int\", \"End\": \"int\"})\n",
    "    \n",
    "    # Convert back to PyRanges\n",
    "    return pr.PyRanges(gaps_df)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "cbaf97f0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "+--------------+-----------+-----------+--------------+------------+\n",
      "| Chromosome   |     Start |       End | Strand       | title      |\n",
      "| (category)   |   (int64) |   (int64) | (category)   | (object)   |\n",
      "|--------------+-----------+-----------+--------------+------------|\n",
      "| chr1         |         5 |        10 | +            | a          |\n",
      "| chr1         |        20 |        28 | +            | b          |\n",
      "| chr1         |        80 |        95 | -            | c          |\n",
      "| chr3         |        10 |        38 | +            | d          |\n",
      "+--------------+-----------+-----------+--------------+------------+\n",
      "Stranded PyRanges object has 4 rows and 5 columns from 2 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome and Strand.\n",
      "+--------------+-----------+-----------+--------------+\n",
      "| Chromosome   |     Start |       End | Strand       |\n",
      "| (category)   |   (int64) |   (int64) | (category)   |\n",
      "|--------------+-----------+-----------+--------------|\n",
      "| chr1         |         5 |        28 | +            |\n",
      "| chr1         |        80 |        95 | -            |\n",
      "| chr3         |        10 |        38 | +            |\n",
      "+--------------+-----------+-----------+--------------+\n",
      "Stranded PyRanges object has 3 rows and 4 columns from 2 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome and Strand.\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "df=pd.DataFrame(\n",
    "{'Chromosome':['chr1', 'chr1', 'chr1', 'chr3'],\n",
    " 'Start': [5, 20, 80, 10],\n",
    " 'End':   [10, 28, 95, 38],\n",
    " 'Strand':['+', '+', '-', '+'],\n",
    " 'title': ['a', 'b', 'c', 'd']}\n",
    ")\n",
    "\n",
    "p=pr.PyRanges(df)\n",
    "print(p)\n",
    "p2=p.merge(slack = 10)\n",
    "print(p2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "847f9a77",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "443\n",
      "0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/y9/8yssfmz161s4ghy99nf1f2cw0000gn/T/ipykernel_27239/19855938.py:18: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  prev_end = df.groupby(\"Chromosome\")[\"End\"].shift(0)   # Current End\n",
      "/var/folders/y9/8yssfmz161s4ghy99nf1f2cw0000gn/T/ipykernel_27239/19855938.py:19: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  next_start = df.groupby(\"Chromosome\")[\"Start\"].shift(-1)  # Next Start\n",
      "/var/folders/y9/8yssfmz161s4ghy99nf1f2cw0000gn/T/ipykernel_27239/19855938.py:18: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  prev_end = df.groupby(\"Chromosome\")[\"End\"].shift(0)   # Current End\n",
      "/var/folders/y9/8yssfmz161s4ghy99nf1f2cw0000gn/T/ipykernel_27239/19855938.py:19: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  next_start = df.groupby(\"Chromosome\")[\"Start\"].shift(-1)  # Next Start\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import pyranges as pr\n",
    "\n",
    "ranges_df = ranges.df\n",
    "gaps = calculate_gaps(ranges)\n",
    "gaps.Length = gaps.End - gaps.Start\n",
    "print(len(gaps[gaps.Length <= 50]))\n",
    "\n",
    "ranges2 = evaluator.process_ranges(ranges)\n",
    "gaps2 = calculate_gaps(ranges2)\n",
    "gaps2.Length = gaps2.End - gaps2.Start\n",
    "print(len(gaps2[gaps2.Length <= 50]))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "28e1cdda",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Runnning scale 1\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0\n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "BINS  IntervalIndex([(3000016, 3000016], (3000016, 3000116], (3000116, 3000216],\n",
      "               (3000216, 3000316], (3000316, 3000416], (3000416, 3000516],\n",
      "               (3000516, 3000616], (3000616, 3000716], (3000716, 3000816],\n",
      "               (3000816, 3000916],\n",
      "               ...\n",
      "               (9716616, 9716716], (9716716, 9716816], (9716816, 9716916],\n",
      "               (9716916, 9717016], (9717016, 9717116], (9717116, 9717216],\n",
      "               (9717216, 9717316], (9717316, 9717416], (9717416, 9717516],\n",
      "               (9717516, 9717593]],\n",
      "              dtype='interval[int64, right]', length=67177)\n",
      "HEAD w BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0   \n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "\n",
      "         bin_offset_0  \n",
      "0  (3000016, 3000116]  \n",
      "1  (3000016, 3000116]  \n",
      "2  (3000016, 3000116]  \n",
      "3  (3000016, 3000116]  \n",
      "4  (3000016, 3000116]  \n",
      "TAIL w BINS\n",
      "       seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  9717555  9717555  18_9717555    1.0    0.0    0.0   \n",
      "699996       19  9717571  9717571  19_9717571    0.0    0.0    0.0   \n",
      "699997       19  9717581  9717581  19_9717581    0.0    0.0    0.0   \n",
      "699998       18  9717582  9717582  18_9717582    0.0    0.0    0.0   \n",
      "699999       18  9717593  9717593  18_9717593    0.0    0.0    0.0   \n",
      "\n",
      "              bin_offset_0  \n",
      "699995  (9717516, 9717593]  \n",
      "699996  (9717516, 9717593]  \n",
      "699997  (9717516, 9717593]  \n",
      "699998  (9717516, 9717593]  \n",
      "699999  (9717516, 9717593]  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:151: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "    TT_S0  starting_nt\n",
      "0     0.0      3000016\n",
      "1     7.0      3000016\n",
      "2     1.0      3000116\n",
      "3     2.0      3000216\n",
      "4     0.0      3000316\n",
      "5     0.0      3000416\n",
      "6     0.0      3000516\n",
      "7     0.0      3000616\n",
      "8     0.0      3000716\n",
      "9     0.0      3000816\n",
      "10    0.0      3000916\n",
      "11    2.0      3001016\n",
      "12    0.0      3001116\n",
      "13    0.0      3001216\n",
      "14    0.0      3001316\n",
      "15    1.0      3001416\n",
      "16    3.0      3001516\n",
      "17    3.0      3001616\n",
      "18    0.0      3001716\n",
      "19    0.0      3001816\n",
      "20    1.0      3001916\n",
      "21    0.0      3002016\n",
      "22    1.0      3002116\n",
      "23   10.0      3002216\n",
      "24    2.0      3002316\n",
      "25   19.0      3002416\n",
      "26    4.0      3002516\n",
      "27    0.0      3002616\n",
      "28    0.0      3002716\n",
      "29    0.0      3002816\n",
      "30    1.0      3002916\n",
      "31    2.0      3003016\n",
      "32    0.0      3003116\n",
      "33    0.0      3003216\n",
      "34    1.0      3003316\n",
      "35    0.0      3003416\n",
      "36    1.0      3003516\n",
      "37    1.0      3003616\n",
      "38    3.0      3003716\n",
      "39    0.0      3003816\n",
      "40    8.0      3003916\n",
      "41    0.0      3004016\n",
      "42    0.0      3004116\n",
      "43    5.0      3004216\n",
      "44    0.0      3004316\n",
      "45    0.0      3004416\n",
      "46    1.0      3004516\n",
      "47    0.0      3004616\n",
      "48    0.0      3004716\n",
      "49   13.0      3004816\n",
      "(67177, 2)\n",
      "TT_S0          float64\n",
      "starting_nt      int64\n",
      "dtype: object\n",
      "sumos ilgis: 67177\n",
      "ranges ilgis: 67177\n",
      "Model input - NO SCALE\n",
      "       TT_S0  genes  transcripts  exons  CDSs\n",
      "0        0.0      0            0      0     0\n",
      "1        7.0      0            0      0     0\n",
      "2        1.0      0            0      0     0\n",
      "3        2.0      0            0      0     0\n",
      "4        0.0      0            0      0     0\n",
      "...      ...    ...          ...    ...   ...\n",
      "67172    0.0      0            0      0     0\n",
      "67173    0.0      0            0      0     0\n",
      "67174    0.0      0            0      0     0\n",
      "67175    1.0      0            0      0     0\n",
      "67176    1.0      0            0      0     0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "       TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0        0.0             0                   0             0            0\n",
      "1        7.0             0                   0             0            0\n",
      "2        1.0             0                   0             0            0\n",
      "3        2.0             0                   0             0            0\n",
      "4        0.0             0                   0             0            0\n",
      "...      ...           ...                 ...           ...          ...\n",
      "67172    0.0             0                   0             0            0\n",
      "67173    0.0             0                   0             0            0\n",
      "67174    0.0             0                   0             0            0\n",
      "67175    1.0             0                   0             0            0\n",
      "67176    1.0             0                   0             0            0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "0.17092159518882952\n",
      "[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 1 0 1 1 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 1 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0]\n",
      "Runnning scale 2\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0\n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "BINS  IntervalIndex([(3000016, 3000016], (3000016, 3000116], (3000116, 3000216],\n",
      "               (3000216, 3000316], (3000316, 3000416], (3000416, 3000516],\n",
      "               (3000516, 3000616], (3000616, 3000716], (3000716, 3000816],\n",
      "               (3000816, 3000916],\n",
      "               ...\n",
      "               (9716616, 9716716], (9716716, 9716816], (9716816, 9716916],\n",
      "               (9716916, 9717016], (9717016, 9717116], (9717116, 9717216],\n",
      "               (9717216, 9717316], (9717316, 9717416], (9717416, 9717516],\n",
      "               (9717516, 9717593]],\n",
      "              dtype='interval[int64, right]', length=67177)\n",
      "HEAD w BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0   \n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "\n",
      "         bin_offset_0  \n",
      "0  (3000016, 3000116]  \n",
      "1  (3000016, 3000116]  \n",
      "2  (3000016, 3000116]  \n",
      "3  (3000016, 3000116]  \n",
      "4  (3000016, 3000116]  \n",
      "TAIL w BINS\n",
      "       seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  9717555  9717555  18_9717555    1.0    0.0    0.0   \n",
      "699996       19  9717571  9717571  19_9717571    0.0    0.0    0.0   \n",
      "699997       19  9717581  9717581  19_9717581    0.0    0.0    0.0   \n",
      "699998       18  9717582  9717582  18_9717582    0.0    0.0    0.0   \n",
      "699999       18  9717593  9717593  18_9717593    0.0    0.0    0.0   \n",
      "\n",
      "              bin_offset_0  \n",
      "699995  (9717516, 9717593]  \n",
      "699996  (9717516, 9717593]  \n",
      "699997  (9717516, 9717593]  \n",
      "699998  (9717516, 9717593]  \n",
      "699999  (9717516, 9717593]  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:151: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "    TT_S0  starting_nt\n",
      "0     0.0      3000016\n",
      "1     7.0      3000016\n",
      "2     1.0      3000116\n",
      "3     2.0      3000216\n",
      "4     0.0      3000316\n",
      "5     0.0      3000416\n",
      "6     0.0      3000516\n",
      "7     0.0      3000616\n",
      "8     0.0      3000716\n",
      "9     0.0      3000816\n",
      "10    0.0      3000916\n",
      "11    2.0      3001016\n",
      "12    0.0      3001116\n",
      "13    0.0      3001216\n",
      "14    0.0      3001316\n",
      "15    1.0      3001416\n",
      "16    3.0      3001516\n",
      "17    3.0      3001616\n",
      "18    0.0      3001716\n",
      "19    0.0      3001816\n",
      "20    1.0      3001916\n",
      "21    0.0      3002016\n",
      "22    1.0      3002116\n",
      "23   10.0      3002216\n",
      "24    2.0      3002316\n",
      "25   19.0      3002416\n",
      "26    4.0      3002516\n",
      "27    0.0      3002616\n",
      "28    0.0      3002716\n",
      "29    0.0      3002816\n",
      "30    1.0      3002916\n",
      "31    2.0      3003016\n",
      "32    0.0      3003116\n",
      "33    0.0      3003216\n",
      "34    1.0      3003316\n",
      "35    0.0      3003416\n",
      "36    1.0      3003516\n",
      "37    1.0      3003616\n",
      "38    3.0      3003716\n",
      "39    0.0      3003816\n",
      "40    8.0      3003916\n",
      "41    0.0      3004016\n",
      "42    0.0      3004116\n",
      "43    5.0      3004216\n",
      "44    0.0      3004316\n",
      "45    0.0      3004416\n",
      "46    1.0      3004516\n",
      "47    0.0      3004616\n",
      "48    0.0      3004716\n",
      "49   13.0      3004816\n",
      "(67177, 2)\n",
      "TT_S0          float64\n",
      "starting_nt      int64\n",
      "dtype: object\n",
      "sumos ilgis: 67177\n",
      "ranges ilgis: 67177\n",
      "Model input - NO SCALE\n",
      "       TT_S0  genes  transcripts  exons  CDSs\n",
      "0        0.0      0            0      0     0\n",
      "1        7.0      0            0      0     0\n",
      "2        1.0      0            0      0     0\n",
      "3        2.0      0            0      0     0\n",
      "4        0.0      0            0      0     0\n",
      "...      ...    ...          ...    ...   ...\n",
      "67172    0.0      0            0      0     0\n",
      "67173    0.0      0            0      0     0\n",
      "67174    0.0      0            0      0     0\n",
      "67175    1.0      0            0      0     0\n",
      "67176    1.0      0            0      0     0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "       TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0        0.0             0                   0             0            0\n",
      "1        7.0             0                   0             0            0\n",
      "2        1.0             0                   0             0            0\n",
      "3        2.0             0                   0             0            0\n",
      "4        0.0             0                   0             0            0\n",
      "...      ...           ...                 ...           ...          ...\n",
      "67172    0.0             0                   0             0            0\n",
      "67173    0.0             0                   0             0            0\n",
      "67174    0.0             0                   0             0            0\n",
      "67175    1.0             0                   0             0            0\n",
      "67176    1.0             0                   0             0            0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "0.20496598538190156\n",
      "[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 1 0 1 1 1 1 0 0 1 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 1 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0]\n",
      "Runnning scale 4\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0\n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "BINS  IntervalIndex([(3000016, 3000016], (3000016, 3000116], (3000116, 3000216],\n",
      "               (3000216, 3000316], (3000316, 3000416], (3000416, 3000516],\n",
      "               (3000516, 3000616], (3000616, 3000716], (3000716, 3000816],\n",
      "               (3000816, 3000916],\n",
      "               ...\n",
      "               (9716616, 9716716], (9716716, 9716816], (9716816, 9716916],\n",
      "               (9716916, 9717016], (9717016, 9717116], (9717116, 9717216],\n",
      "               (9717216, 9717316], (9717316, 9717416], (9717416, 9717516],\n",
      "               (9717516, 9717593]],\n",
      "              dtype='interval[int64, right]', length=67177)\n",
      "HEAD w BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0   \n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "\n",
      "         bin_offset_0  \n",
      "0  (3000016, 3000116]  \n",
      "1  (3000016, 3000116]  \n",
      "2  (3000016, 3000116]  \n",
      "3  (3000016, 3000116]  \n",
      "4  (3000016, 3000116]  \n",
      "TAIL w BINS\n",
      "       seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  9717555  9717555  18_9717555    1.0    0.0    0.0   \n",
      "699996       19  9717571  9717571  19_9717571    0.0    0.0    0.0   \n",
      "699997       19  9717581  9717581  19_9717581    0.0    0.0    0.0   \n",
      "699998       18  9717582  9717582  18_9717582    0.0    0.0    0.0   \n",
      "699999       18  9717593  9717593  18_9717593    0.0    0.0    0.0   \n",
      "\n",
      "              bin_offset_0  \n",
      "699995  (9717516, 9717593]  \n",
      "699996  (9717516, 9717593]  \n",
      "699997  (9717516, 9717593]  \n",
      "699998  (9717516, 9717593]  \n",
      "699999  (9717516, 9717593]  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:151: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "    TT_S0  starting_nt\n",
      "0     0.0      3000016\n",
      "1     7.0      3000016\n",
      "2     1.0      3000116\n",
      "3     2.0      3000216\n",
      "4     0.0      3000316\n",
      "5     0.0      3000416\n",
      "6     0.0      3000516\n",
      "7     0.0      3000616\n",
      "8     0.0      3000716\n",
      "9     0.0      3000816\n",
      "10    0.0      3000916\n",
      "11    2.0      3001016\n",
      "12    0.0      3001116\n",
      "13    0.0      3001216\n",
      "14    0.0      3001316\n",
      "15    1.0      3001416\n",
      "16    3.0      3001516\n",
      "17    3.0      3001616\n",
      "18    0.0      3001716\n",
      "19    0.0      3001816\n",
      "20    1.0      3001916\n",
      "21    0.0      3002016\n",
      "22    1.0      3002116\n",
      "23   10.0      3002216\n",
      "24    2.0      3002316\n",
      "25   19.0      3002416\n",
      "26    4.0      3002516\n",
      "27    0.0      3002616\n",
      "28    0.0      3002716\n",
      "29    0.0      3002816\n",
      "30    1.0      3002916\n",
      "31    2.0      3003016\n",
      "32    0.0      3003116\n",
      "33    0.0      3003216\n",
      "34    1.0      3003316\n",
      "35    0.0      3003416\n",
      "36    1.0      3003516\n",
      "37    1.0      3003616\n",
      "38    3.0      3003716\n",
      "39    0.0      3003816\n",
      "40    8.0      3003916\n",
      "41    0.0      3004016\n",
      "42    0.0      3004116\n",
      "43    5.0      3004216\n",
      "44    0.0      3004316\n",
      "45    0.0      3004416\n",
      "46    1.0      3004516\n",
      "47    0.0      3004616\n",
      "48    0.0      3004716\n",
      "49   13.0      3004816\n",
      "(67177, 2)\n",
      "TT_S0          float64\n",
      "starting_nt      int64\n",
      "dtype: object\n",
      "sumos ilgis: 67177\n",
      "ranges ilgis: 67177\n",
      "Model input - NO SCALE\n",
      "       TT_S0  genes  transcripts  exons  CDSs\n",
      "0        0.0      0            0      0     0\n",
      "1        7.0      0            0      0     0\n",
      "2        1.0      0            0      0     0\n",
      "3        2.0      0            0      0     0\n",
      "4        0.0      0            0      0     0\n",
      "...      ...    ...          ...    ...   ...\n",
      "67172    0.0      0            0      0     0\n",
      "67173    0.0      0            0      0     0\n",
      "67174    0.0      0            0      0     0\n",
      "67175    1.0      0            0      0     0\n",
      "67176    1.0      0            0      0     0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "       TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0        0.0             0                   0             0            0\n",
      "1        7.0             0                   0             0            0\n",
      "2        1.0             0                   0             0            0\n",
      "3        2.0             0                   0             0            0\n",
      "4        0.0             0                   0             0            0\n",
      "...      ...           ...                 ...           ...          ...\n",
      "67172    0.0             0                   0             0            0\n",
      "67173    0.0             0                   0             0            0\n",
      "67174    0.0             0                   0             0            0\n",
      "67175    1.0             0                   0             0            0\n",
      "67176    1.0             0                   0             0            0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "0.20496598538190156\n",
      "[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 1 0 1 1 1 1 0 0 1 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 1 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0]\n",
      "Runnning scale 8\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0\n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "BINS  IntervalIndex([(3000016, 3000016], (3000016, 3000116], (3000116, 3000216],\n",
      "               (3000216, 3000316], (3000316, 3000416], (3000416, 3000516],\n",
      "               (3000516, 3000616], (3000616, 3000716], (3000716, 3000816],\n",
      "               (3000816, 3000916],\n",
      "               ...\n",
      "               (9716616, 9716716], (9716716, 9716816], (9716816, 9716916],\n",
      "               (9716916, 9717016], (9717016, 9717116], (9717116, 9717216],\n",
      "               (9717216, 9717316], (9717316, 9717416], (9717416, 9717516],\n",
      "               (9717516, 9717593]],\n",
      "              dtype='interval[int64, right]', length=67177)\n",
      "HEAD w BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0   \n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "\n",
      "         bin_offset_0  \n",
      "0  (3000016, 3000116]  \n",
      "1  (3000016, 3000116]  \n",
      "2  (3000016, 3000116]  \n",
      "3  (3000016, 3000116]  \n",
      "4  (3000016, 3000116]  \n",
      "TAIL w BINS\n",
      "       seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  9717555  9717555  18_9717555    1.0    0.0    0.0   \n",
      "699996       19  9717571  9717571  19_9717571    0.0    0.0    0.0   \n",
      "699997       19  9717581  9717581  19_9717581    0.0    0.0    0.0   \n",
      "699998       18  9717582  9717582  18_9717582    0.0    0.0    0.0   \n",
      "699999       18  9717593  9717593  18_9717593    0.0    0.0    0.0   \n",
      "\n",
      "              bin_offset_0  \n",
      "699995  (9717516, 9717593]  \n",
      "699996  (9717516, 9717593]  \n",
      "699997  (9717516, 9717593]  \n",
      "699998  (9717516, 9717593]  \n",
      "699999  (9717516, 9717593]  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:151: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "    TT_S0  starting_nt\n",
      "0     0.0      3000016\n",
      "1     7.0      3000016\n",
      "2     1.0      3000116\n",
      "3     2.0      3000216\n",
      "4     0.0      3000316\n",
      "5     0.0      3000416\n",
      "6     0.0      3000516\n",
      "7     0.0      3000616\n",
      "8     0.0      3000716\n",
      "9     0.0      3000816\n",
      "10    0.0      3000916\n",
      "11    2.0      3001016\n",
      "12    0.0      3001116\n",
      "13    0.0      3001216\n",
      "14    0.0      3001316\n",
      "15    1.0      3001416\n",
      "16    3.0      3001516\n",
      "17    3.0      3001616\n",
      "18    0.0      3001716\n",
      "19    0.0      3001816\n",
      "20    1.0      3001916\n",
      "21    0.0      3002016\n",
      "22    1.0      3002116\n",
      "23   10.0      3002216\n",
      "24    2.0      3002316\n",
      "25   19.0      3002416\n",
      "26    4.0      3002516\n",
      "27    0.0      3002616\n",
      "28    0.0      3002716\n",
      "29    0.0      3002816\n",
      "30    1.0      3002916\n",
      "31    2.0      3003016\n",
      "32    0.0      3003116\n",
      "33    0.0      3003216\n",
      "34    1.0      3003316\n",
      "35    0.0      3003416\n",
      "36    1.0      3003516\n",
      "37    1.0      3003616\n",
      "38    3.0      3003716\n",
      "39    0.0      3003816\n",
      "40    8.0      3003916\n",
      "41    0.0      3004016\n",
      "42    0.0      3004116\n",
      "43    5.0      3004216\n",
      "44    0.0      3004316\n",
      "45    0.0      3004416\n",
      "46    1.0      3004516\n",
      "47    0.0      3004616\n",
      "48    0.0      3004716\n",
      "49   13.0      3004816\n",
      "(67177, 2)\n",
      "TT_S0          float64\n",
      "starting_nt      int64\n",
      "dtype: object\n",
      "sumos ilgis: 67177\n",
      "ranges ilgis: 67177\n",
      "Model input - NO SCALE\n",
      "       TT_S0  genes  transcripts  exons  CDSs\n",
      "0        0.0      0            0      0     0\n",
      "1        7.0      0            0      0     0\n",
      "2        1.0      0            0      0     0\n",
      "3        2.0      0            0      0     0\n",
      "4        0.0      0            0      0     0\n",
      "...      ...    ...          ...    ...   ...\n",
      "67172    0.0      0            0      0     0\n",
      "67173    0.0      0            0      0     0\n",
      "67174    0.0      0            0      0     0\n",
      "67175    1.0      0            0      0     0\n",
      "67176    1.0      0            0      0     0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "       TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0        0.0             0                   0             0            0\n",
      "1        7.0             0                   0             0            0\n",
      "2        1.0             0                   0             0            0\n",
      "3        2.0             0                   0             0            0\n",
      "4        0.0             0                   0             0            0\n",
      "...      ...           ...                 ...           ...          ...\n",
      "67172    0.0             0                   0             0            0\n",
      "67173    0.0             0                   0             0            0\n",
      "67174    0.0             0                   0             0            0\n",
      "67175    1.0             0                   0             0            0\n",
      "67176    1.0             0                   0             0            0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "0.20496598538190156\n",
      "[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 1 0 1 1 1 1 0 0 1 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 1 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0]\n",
      "Runnning scale 16\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0\n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "BINS  IntervalIndex([(3000016, 3000016], (3000016, 3000116], (3000116, 3000216],\n",
      "               (3000216, 3000316], (3000316, 3000416], (3000416, 3000516],\n",
      "               (3000516, 3000616], (3000616, 3000716], (3000716, 3000816],\n",
      "               (3000816, 3000916],\n",
      "               ...\n",
      "               (9716616, 9716716], (9716716, 9716816], (9716816, 9716916],\n",
      "               (9716916, 9717016], (9717016, 9717116], (9717116, 9717216],\n",
      "               (9717216, 9717316], (9717316, 9717416], (9717416, 9717516],\n",
      "               (9717516, 9717593]],\n",
      "              dtype='interval[int64, right]', length=67177)\n",
      "HEAD w BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0   \n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "\n",
      "         bin_offset_0  \n",
      "0  (3000016, 3000116]  \n",
      "1  (3000016, 3000116]  \n",
      "2  (3000016, 3000116]  \n",
      "3  (3000016, 3000116]  \n",
      "4  (3000016, 3000116]  \n",
      "TAIL w BINS\n",
      "       seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  9717555  9717555  18_9717555    1.0    0.0    0.0   \n",
      "699996       19  9717571  9717571  19_9717571    0.0    0.0    0.0   \n",
      "699997       19  9717581  9717581  19_9717581    0.0    0.0    0.0   \n",
      "699998       18  9717582  9717582  18_9717582    0.0    0.0    0.0   \n",
      "699999       18  9717593  9717593  18_9717593    0.0    0.0    0.0   \n",
      "\n",
      "              bin_offset_0  \n",
      "699995  (9717516, 9717593]  \n",
      "699996  (9717516, 9717593]  \n",
      "699997  (9717516, 9717593]  \n",
      "699998  (9717516, 9717593]  \n",
      "699999  (9717516, 9717593]  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:151: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "    TT_S0  starting_nt\n",
      "0     0.0      3000016\n",
      "1     7.0      3000016\n",
      "2     1.0      3000116\n",
      "3     2.0      3000216\n",
      "4     0.0      3000316\n",
      "5     0.0      3000416\n",
      "6     0.0      3000516\n",
      "7     0.0      3000616\n",
      "8     0.0      3000716\n",
      "9     0.0      3000816\n",
      "10    0.0      3000916\n",
      "11    2.0      3001016\n",
      "12    0.0      3001116\n",
      "13    0.0      3001216\n",
      "14    0.0      3001316\n",
      "15    1.0      3001416\n",
      "16    3.0      3001516\n",
      "17    3.0      3001616\n",
      "18    0.0      3001716\n",
      "19    0.0      3001816\n",
      "20    1.0      3001916\n",
      "21    0.0      3002016\n",
      "22    1.0      3002116\n",
      "23   10.0      3002216\n",
      "24    2.0      3002316\n",
      "25   19.0      3002416\n",
      "26    4.0      3002516\n",
      "27    0.0      3002616\n",
      "28    0.0      3002716\n",
      "29    0.0      3002816\n",
      "30    1.0      3002916\n",
      "31    2.0      3003016\n",
      "32    0.0      3003116\n",
      "33    0.0      3003216\n",
      "34    1.0      3003316\n",
      "35    0.0      3003416\n",
      "36    1.0      3003516\n",
      "37    1.0      3003616\n",
      "38    3.0      3003716\n",
      "39    0.0      3003816\n",
      "40    8.0      3003916\n",
      "41    0.0      3004016\n",
      "42    0.0      3004116\n",
      "43    5.0      3004216\n",
      "44    0.0      3004316\n",
      "45    0.0      3004416\n",
      "46    1.0      3004516\n",
      "47    0.0      3004616\n",
      "48    0.0      3004716\n",
      "49   13.0      3004816\n",
      "(67177, 2)\n",
      "TT_S0          float64\n",
      "starting_nt      int64\n",
      "dtype: object\n",
      "sumos ilgis: 67177\n",
      "ranges ilgis: 67177\n",
      "Model input - NO SCALE\n",
      "       TT_S0  genes  transcripts  exons  CDSs\n",
      "0        0.0      0            0      0     0\n",
      "1        7.0      0            0      0     0\n",
      "2        1.0      0            0      0     0\n",
      "3        2.0      0            0      0     0\n",
      "4        0.0      0            0      0     0\n",
      "...      ...    ...          ...    ...   ...\n",
      "67172    0.0      0            0      0     0\n",
      "67173    0.0      0            0      0     0\n",
      "67174    0.0      0            0      0     0\n",
      "67175    1.0      0            0      0     0\n",
      "67176    1.0      0            0      0     0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "       TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0        0.0             0                   0             0            0\n",
      "1        7.0             0                   0             0            0\n",
      "2        1.0             0                   0             0            0\n",
      "3        2.0             0                   0             0            0\n",
      "4        0.0             0                   0             0            0\n",
      "...      ...           ...                 ...           ...          ...\n",
      "67172    0.0             0                   0             0            0\n",
      "67173    0.0             0                   0             0            0\n",
      "67174    0.0             0                   0             0            0\n",
      "67175    1.0             0                   0             0            0\n",
      "67176    1.0             0                   0             0            0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "0.20496598538190156\n",
      "[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 1 0 1 1 1 1 0 0 1 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 1 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0]\n",
      "Runnning scale 32\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0\n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "BINS  IntervalIndex([(3000016, 3000016], (3000016, 3000116], (3000116, 3000216],\n",
      "               (3000216, 3000316], (3000316, 3000416], (3000416, 3000516],\n",
      "               (3000516, 3000616], (3000616, 3000716], (3000716, 3000816],\n",
      "               (3000816, 3000916],\n",
      "               ...\n",
      "               (9716616, 9716716], (9716716, 9716816], (9716816, 9716916],\n",
      "               (9716916, 9717016], (9717016, 9717116], (9717116, 9717216],\n",
      "               (9717216, 9717316], (9717316, 9717416], (9717416, 9717516],\n",
      "               (9717516, 9717593]],\n",
      "              dtype='interval[int64, right]', length=67177)\n",
      "HEAD w BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0   \n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "\n",
      "         bin_offset_0  \n",
      "0  (3000016, 3000116]  \n",
      "1  (3000016, 3000116]  \n",
      "2  (3000016, 3000116]  \n",
      "3  (3000016, 3000116]  \n",
      "4  (3000016, 3000116]  \n",
      "TAIL w BINS\n",
      "       seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  9717555  9717555  18_9717555    1.0    0.0    0.0   \n",
      "699996       19  9717571  9717571  19_9717571    0.0    0.0    0.0   \n",
      "699997       19  9717581  9717581  19_9717581    0.0    0.0    0.0   \n",
      "699998       18  9717582  9717582  18_9717582    0.0    0.0    0.0   \n",
      "699999       18  9717593  9717593  18_9717593    0.0    0.0    0.0   \n",
      "\n",
      "              bin_offset_0  \n",
      "699995  (9717516, 9717593]  \n",
      "699996  (9717516, 9717593]  \n",
      "699997  (9717516, 9717593]  \n",
      "699998  (9717516, 9717593]  \n",
      "699999  (9717516, 9717593]  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:151: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "    TT_S0  starting_nt\n",
      "0     0.0      3000016\n",
      "1     7.0      3000016\n",
      "2     1.0      3000116\n",
      "3     2.0      3000216\n",
      "4     0.0      3000316\n",
      "5     0.0      3000416\n",
      "6     0.0      3000516\n",
      "7     0.0      3000616\n",
      "8     0.0      3000716\n",
      "9     0.0      3000816\n",
      "10    0.0      3000916\n",
      "11    2.0      3001016\n",
      "12    0.0      3001116\n",
      "13    0.0      3001216\n",
      "14    0.0      3001316\n",
      "15    1.0      3001416\n",
      "16    3.0      3001516\n",
      "17    3.0      3001616\n",
      "18    0.0      3001716\n",
      "19    0.0      3001816\n",
      "20    1.0      3001916\n",
      "21    0.0      3002016\n",
      "22    1.0      3002116\n",
      "23   10.0      3002216\n",
      "24    2.0      3002316\n",
      "25   19.0      3002416\n",
      "26    4.0      3002516\n",
      "27    0.0      3002616\n",
      "28    0.0      3002716\n",
      "29    0.0      3002816\n",
      "30    1.0      3002916\n",
      "31    2.0      3003016\n",
      "32    0.0      3003116\n",
      "33    0.0      3003216\n",
      "34    1.0      3003316\n",
      "35    0.0      3003416\n",
      "36    1.0      3003516\n",
      "37    1.0      3003616\n",
      "38    3.0      3003716\n",
      "39    0.0      3003816\n",
      "40    8.0      3003916\n",
      "41    0.0      3004016\n",
      "42    0.0      3004116\n",
      "43    5.0      3004216\n",
      "44    0.0      3004316\n",
      "45    0.0      3004416\n",
      "46    1.0      3004516\n",
      "47    0.0      3004616\n",
      "48    0.0      3004716\n",
      "49   13.0      3004816\n",
      "(67177, 2)\n",
      "TT_S0          float64\n",
      "starting_nt      int64\n",
      "dtype: object\n",
      "sumos ilgis: 67177\n",
      "ranges ilgis: 67177\n",
      "Model input - NO SCALE\n",
      "       TT_S0  genes  transcripts  exons  CDSs\n",
      "0        0.0      0            0      0     0\n",
      "1        7.0      0            0      0     0\n",
      "2        1.0      0            0      0     0\n",
      "3        2.0      0            0      0     0\n",
      "4        0.0      0            0      0     0\n",
      "...      ...    ...          ...    ...   ...\n",
      "67172    0.0      0            0      0     0\n",
      "67173    0.0      0            0      0     0\n",
      "67174    0.0      0            0      0     0\n",
      "67175    1.0      0            0      0     0\n",
      "67176    1.0      0            0      0     0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "       TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0        0.0             0                   0             0            0\n",
      "1        7.0             0                   0             0            0\n",
      "2        1.0             0                   0             0            0\n",
      "3        2.0             0                   0             0            0\n",
      "4        0.0             0                   0             0            0\n",
      "...      ...           ...                 ...           ...          ...\n",
      "67172    0.0             0                   0             0            0\n",
      "67173    0.0             0                   0             0            0\n",
      "67174    0.0             0                   0             0            0\n",
      "67175    1.0             0                   0             0            0\n",
      "67176    1.0             0                   0             0            0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "0.17092159518882952\n",
      "[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 1 0 1 1 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 1 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0]\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "for i in range(6):\n",
    "    scale = 2 ** i\n",
    "    print(\"Runnning scale\", scale)\n",
    "    evaluation = evaluator.evaluate(dataframe = chromatin_algos.algo5d(scale), model_name=f\"k-means 5 features scale={scale}\")\n",
    "    result = pd.concat([result, evaluation], axis=1)\n",
    "\n",
    "result.to_csv('../outputs/results1.csv', sep = '\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ca1280ed",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                         Length                      Length  \\\n",
      "count                    7301.0                      6456.0   \n",
      "mean                 188.590604                  177.850062   \n",
      "std                  174.998864                  157.257952   \n",
      "min                       100.0                       100.0   \n",
      "25%                       100.0                       100.0   \n",
      "50%                       100.0                       100.0   \n",
      "75%                       200.0                       200.0   \n",
      "max                      2400.0                      2000.0   \n",
      "ATACseq_overlap        0.697949                    0.668276   \n",
      "DNAseq_overlap         0.595329                    0.553813   \n",
      "genes_overlap          0.544833                    0.550809   \n",
      "percentage_open        0.141691                    0.118157   \n",
      "percentage_open_adj    0.204969                    0.170924   \n",
      "model_name                       k-means 5 features scale=1   \n",
      "min_open_signal            17.0                        19.0   \n",
      "max_open_signal           393.0                       393.0   \n",
      "min_closed_signal           0.0                         0.0   \n",
      "max_closed_signal          16.0                        18.0   \n",
      "\n",
      "                                         Length                      Length  \\\n",
      "count                                    7301.0                      7301.0   \n",
      "mean                                 188.590604                  188.590604   \n",
      "std                                  174.998864                  174.998864   \n",
      "min                                       100.0                       100.0   \n",
      "25%                                       100.0                       100.0   \n",
      "50%                                       100.0                       100.0   \n",
      "75%                                       200.0                       200.0   \n",
      "max                                      2400.0                      2400.0   \n",
      "ATACseq_overlap                        0.697949                    0.697949   \n",
      "DNAseq_overlap                         0.595329                    0.595329   \n",
      "genes_overlap                          0.544833                    0.544833   \n",
      "percentage_open                        0.141691                    0.141691   \n",
      "percentage_open_adj                    0.204969                    0.204969   \n",
      "model_name           k-means 5 features scale=2  k-means 5 features scale=4   \n",
      "min_open_signal                            17.0                        17.0   \n",
      "max_open_signal                           393.0                       393.0   \n",
      "min_closed_signal                           0.0                         0.0   \n",
      "max_closed_signal                          16.0                        16.0   \n",
      "\n",
      "                                         Length                       Length  \\\n",
      "count                                    7301.0                       7301.0   \n",
      "mean                                 188.590604                   188.590604   \n",
      "std                                  174.998864                   174.998864   \n",
      "min                                       100.0                        100.0   \n",
      "25%                                       100.0                        100.0   \n",
      "50%                                       100.0                        100.0   \n",
      "75%                                       200.0                        200.0   \n",
      "max                                      2400.0                       2400.0   \n",
      "ATACseq_overlap                        0.697949                     0.697949   \n",
      "DNAseq_overlap                         0.595329                     0.595329   \n",
      "genes_overlap                          0.544833                     0.544833   \n",
      "percentage_open                        0.141691                     0.141691   \n",
      "percentage_open_adj                    0.204969                     0.204969   \n",
      "model_name           k-means 5 features scale=8  k-means 5 features scale=16   \n",
      "min_open_signal                            17.0                         17.0   \n",
      "max_open_signal                           393.0                        393.0   \n",
      "min_closed_signal                           0.0                          0.0   \n",
      "max_closed_signal                          16.0                         16.0   \n",
      "\n",
      "                                          Length  \n",
      "count                                     6456.0  \n",
      "mean                                  177.850062  \n",
      "std                                   157.257952  \n",
      "min                                        100.0  \n",
      "25%                                        100.0  \n",
      "50%                                        100.0  \n",
      "75%                                        200.0  \n",
      "max                                       2000.0  \n",
      "ATACseq_overlap                         0.668276  \n",
      "DNAseq_overlap                          0.553813  \n",
      "genes_overlap                           0.550809  \n",
      "percentage_open                         0.118157  \n",
      "percentage_open_adj                     0.170924  \n",
      "model_name           k-means 5 features scale=32  \n",
      "min_open_signal                             19.0  \n",
      "max_open_signal                            393.0  \n",
      "min_closed_signal                            0.0  \n",
      "max_closed_signal                           18.0  \n"
     ]
    }
   ],
   "source": [
    "print(result)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
