{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "d001e5b1",
   "metadata": {},
   "outputs": [],
   "source": [
    "import chromatin_algos\n",
    "import evaluator\n",
    "import generate_consensus"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "0558c7ce",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0\n",
      "BINS  IntervalIndex([  (3000018, 3000018],   (3000018, 3000118],\n",
      "                 (3000118, 3000218],   (3000218, 3000318],\n",
      "                 (3000318, 3000418],   (3000418, 3000518],\n",
      "                 (3000518, 3000618],   (3000618, 3000718],\n",
      "                 (3000718, 3000818],   (3000818, 3000918],\n",
      "               ...\n",
      "               (18260718, 18260818], (18260818, 18260918],\n",
      "               (18260918, 18261018], (18261018, 18261118],\n",
      "               (18261118, 18261218], (18261218, 18261318],\n",
      "               (18261318, 18261418], (18261418, 18261518],\n",
      "               (18261518, 18261618], (18261618, 18261710]],\n",
      "              dtype='interval[int64, right]', length=152618)\n",
      "HEAD with BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0   \n",
      "\n",
      "         bin_offset_0  target_count  \n",
      "0  (3000018, 3000118]             1  \n",
      "1  (3000018, 3000118]             1  \n",
      "2  (3000018, 3000118]             1  \n",
      "3  (3000018, 3000118]             1  \n",
      "4  (3000018, 3000118]             1  \n",
      "TAIL with BINS\n",
      "       seqnames     start       end        CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  18261611  18261611  18_18261611    0.0    0.0    0.0   \n",
      "699996       18  18261615  18261615  18_18261615    0.0    0.0    0.0   \n",
      "699997       18  18261682  18261682  18_18261682    0.0    0.0    0.0   \n",
      "699998       18  18261702  18261702  18_18261702    4.0    0.0    0.0   \n",
      "699999       18  18261710  18261710  18_18261710    0.0    0.0    0.0   \n",
      "\n",
      "                bin_offset_0  target_count  \n",
      "699995  (18261518, 18261618]             1  \n",
      "699996  (18261518, 18261618]             1  \n",
      "699997  (18261618, 18261710]             1  \n",
      "699998  (18261618, 18261710]             1  \n",
      "699999  (18261618, 18261710]             1  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:137: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "          bin_offset_0  TT_S0  target_count  starting_nt\n",
      "0   (3000018, 3000018]    0.0             0      3000018\n",
      "1   (3000018, 3000118]    7.0            11      3000018\n",
      "2   (3000118, 3000218]    1.0             4      3000118\n",
      "3   (3000218, 3000318]    2.0             4      3000218\n",
      "4   (3000318, 3000418]    0.0             4      3000318\n",
      "5   (3000418, 3000518]    0.0             2      3000418\n",
      "6   (3000518, 3000618]    0.0             2      3000518\n",
      "7   (3000618, 3000718]    0.0             4      3000618\n",
      "8   (3000718, 3000818]    0.0             7      3000718\n",
      "9   (3000818, 3000918]    0.0             3      3000818\n",
      "10  (3000918, 3001018]    0.0             3      3000918\n",
      "11  (3001018, 3001118]    2.0             6      3001018\n",
      "12  (3001118, 3001218]    0.0             2      3001118\n",
      "13  (3001218, 3001318]    0.0             4      3001218\n",
      "14  (3001318, 3001418]    0.0             3      3001318\n",
      "15  (3001418, 3001518]    1.0             4      3001418\n",
      "16  (3001518, 3001618]    3.0            10      3001518\n",
      "17  (3001618, 3001718]    3.0             7      3001618\n",
      "18  (3001718, 3001818]    0.0             6      3001718\n",
      "19  (3001818, 3001918]    0.0             2      3001818\n",
      "20  (3001918, 3002018]    1.0             3      3001918\n",
      "21  (3002018, 3002118]    0.0             2      3002018\n",
      "22  (3002118, 3002218]    1.0             4      3002118\n",
      "23  (3002218, 3002318]   10.0             6      3002218\n",
      "24  (3002318, 3002418]    2.0             1      3002318\n",
      "25  (3002418, 3002518]   19.0             5      3002418\n",
      "26  (3002518, 3002618]    4.0             2      3002518\n",
      "27  (3002618, 3002718]    0.0             3      3002618\n",
      "28  (3002718, 3002818]    0.0             1      3002718\n",
      "29  (3002818, 3002918]    0.0             2      3002818\n",
      "30  (3002918, 3003018]    1.0             6      3002918\n",
      "31  (3003018, 3003118]    2.0             4      3003018\n",
      "32  (3003118, 3003218]    0.0             0      3003118\n",
      "33  (3003218, 3003318]    0.0             1      3003218\n",
      "34  (3003318, 3003418]    1.0             8      3003318\n",
      "35  (3003418, 3003518]    0.0             3      3003418\n",
      "36  (3003518, 3003618]    1.0             2      3003518\n",
      "37  (3003618, 3003718]    3.0             4      3003618\n",
      "38  (3003718, 3003818]    1.0             6      3003718\n",
      "39  (3003818, 3003918]    0.0             3      3003818\n",
      "40  (3003918, 3004018]    8.0             6      3003918\n",
      "41  (3004018, 3004118]    0.0             0      3004018\n",
      "42  (3004118, 3004218]    0.0             1      3004118\n",
      "43  (3004218, 3004318]    5.0             6      3004218\n",
      "44  (3004318, 3004418]    0.0             1      3004318\n",
      "45  (3004418, 3004518]    0.0             1      3004418\n",
      "46  (3004518, 3004618]    1.0             7      3004518\n",
      "47  (3004618, 3004718]    0.0             0      3004618\n",
      "48  (3004718, 3004818]    0.0             1      3004718\n",
      "49  (3004818, 3004918]   13.0             8      3004818\n",
      "(152618, 4)\n",
      "bin_offset_0    category\n",
      "TT_S0            float64\n",
      "target_count       int64\n",
      "starting_nt        int64\n",
      "dtype: object\n",
      "sumos ilgis: 152618\n",
      "ranges ilgis: 152618\n",
      "Model input - NO SCALE\n",
      "        TT_S0  genes  transcripts  exons  CDSs\n",
      "0         0.0      0            0      0     0\n",
      "1         7.0      0            0      0     0\n",
      "2         1.0      0            0      0     0\n",
      "3         2.0      0            0      0     0\n",
      "4         0.0      0            0      0     0\n",
      "...       ...    ...          ...    ...   ...\n",
      "152613   10.0      0            0      0     0\n",
      "152614    3.0      0            0      0     0\n",
      "152615    4.0      0            0      0     0\n",
      "152616    0.0      0            0      0     0\n",
      "152617    4.0      0            0      0     0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "        TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0         0.0             0                   0             0            0\n",
      "1         7.0             0                   0             0            0\n",
      "2         1.0             0                   0             0            0\n",
      "3         2.0             0                   0             0            0\n",
      "4         0.0             0                   0             0            0\n",
      "...       ...           ...                 ...           ...          ...\n",
      "152613   10.0             0                   0             0            0\n",
      "152614    3.0             0                   0             0            0\n",
      "152615    4.0             0                   0             0            0\n",
      "152616    0.0             0                   0             0            0\n",
      "152617    4.0             0                   0             0            0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "0.10474518077815198\n",
      "[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 1 0 1 1 1 0 1 1 1 1 1 1 0 1 1 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 0 0 1 1 1 1 1 1 0 0 1 1 0 1\n",
      " 0 1 0 0 0 0 0 0 0 0 0 0 0 0 1 0 1 1 1 1 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 1\n",
      " 0 0 1 0 0 0 0 0 1 0 0 0 1 0 1 0 0 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 1 1 0 1 1 1 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 1 0 1 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0]\n",
      "threshold density 1.30546\n",
      "cluster signal 219187.0\n",
      "cluster target count 572145\n",
      "!!cluster density 0.3830969422087058\n",
      "cluster signal 237724.0\n",
      "cluster target count 127855\n",
      "!!cluster density 1.8593250166203903\n",
      "\n",
      "Labels mapping\n",
      " {0: 0, 1: 1}\n",
      "DEBUG: 136632 15986 \n",
      "\n",
      " +----------------------+-------------+----------------+-------+\n",
      "| bin_offset_0         | TT_S0       | target_count   | +9    |\n",
      "| (category)           | (float64)   | (int64)        | ...   |\n",
      "|----------------------+-------------+----------------+-------|\n",
      "| (3000018, 3000018]   | 0.0         | 0              | ...   |\n",
      "| (3000018, 3000118]   | 7.0         | 11             | ...   |\n",
      "| (3000118, 3000218]   | 1.0         | 4              | ...   |\n",
      "| (3000218, 3000318]   | 2.0         | 4              | ...   |\n",
      "| ...                  | ...         | ...            | ...   |\n",
      "| (18261318, 18261418] | 3.0         | 15             | ...   |\n",
      "| (18261418, 18261518] | 4.0         | 4              | ...   |\n",
      "| (18261518, 18261618] | 0.0         | 3              | ...   |\n",
      "| (18261618, 18261710] | 4.0         | 3              | ...   |\n",
      "+----------------------+-------------+----------------+-------+\n",
      "Unstranded PyRanges object has 136,632 rows and 12 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "9 hidden columns: starting_nt, Chromosome, Start, End, genes, transcripts, ... (+ 3 more.)\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0\n",
      "BINS  IntervalIndex([  (3000018, 3000023],   (3000023, 3000123],\n",
      "                 (3000123, 3000223],   (3000223, 3000323],\n",
      "                 (3000323, 3000423],   (3000423, 3000523],\n",
      "                 (3000523, 3000623],   (3000623, 3000723],\n",
      "                 (3000723, 3000823],   (3000823, 3000923],\n",
      "               ...\n",
      "               (18260723, 18260823], (18260823, 18260923],\n",
      "               (18260923, 18261023], (18261023, 18261123],\n",
      "               (18261123, 18261223], (18261223, 18261323],\n",
      "               (18261323, 18261423], (18261423, 18261523],\n",
      "               (18261523, 18261623], (18261623, 18261710]],\n",
      "              dtype='interval[int64, right]', length=152618)\n",
      "HEAD with BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0   \n",
      "\n",
      "         bin_offset_5  target_count  \n",
      "0  (3000018, 3000023]             1  \n",
      "1  (3000018, 3000023]             1  \n",
      "2  (3000023, 3000123]             1  \n",
      "3  (3000023, 3000123]             1  \n",
      "4  (3000023, 3000123]             1  \n",
      "TAIL with BINS\n",
      "       seqnames     start       end        CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  18261611  18261611  18_18261611    0.0    0.0    0.0   \n",
      "699996       18  18261615  18261615  18_18261615    0.0    0.0    0.0   \n",
      "699997       18  18261682  18261682  18_18261682    0.0    0.0    0.0   \n",
      "699998       18  18261702  18261702  18_18261702    4.0    0.0    0.0   \n",
      "699999       18  18261710  18261710  18_18261710    0.0    0.0    0.0   \n",
      "\n",
      "                bin_offset_5  target_count  \n",
      "699995  (18261523, 18261623]             1  \n",
      "699996  (18261523, 18261623]             1  \n",
      "699997  (18261623, 18261710]             1  \n",
      "699998  (18261623, 18261710]             1  \n",
      "699999  (18261623, 18261710]             1  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:137: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "          bin_offset_5  TT_S0  target_count starting_nt\n",
      "0   (3000018, 3000023]    0.0             2     3000018\n",
      "1   (3000023, 3000123]    7.0             9     3000023\n",
      "2   (3000123, 3000223]    1.0             5     3000123\n",
      "3   (3000223, 3000323]    2.0             3     3000223\n",
      "4   (3000323, 3000423]    0.0             4     3000323\n",
      "5   (3000423, 3000523]    0.0             2     3000423\n",
      "6   (3000523, 3000623]    0.0             2     3000523\n",
      "7   (3000623, 3000723]    0.0             4     3000623\n",
      "8   (3000723, 3000823]    0.0             7     3000723\n",
      "9   (3000823, 3000923]    0.0             4     3000823\n",
      "10  (3000923, 3001023]    0.0             2     3000923\n",
      "11  (3001023, 3001123]    2.0             6     3001023\n",
      "12  (3001123, 3001223]    0.0             2     3001123\n",
      "13  (3001223, 3001323]    0.0             5     3001223\n",
      "14  (3001323, 3001423]    0.0             2     3001323\n",
      "15  (3001423, 3001523]    1.0             6     3001423\n",
      "16  (3001523, 3001623]    3.0             8     3001523\n",
      "17  (3001623, 3001723]    3.0             8     3001623\n",
      "18  (3001723, 3001823]    0.0             5     3001723\n",
      "19  (3001823, 3001923]    0.0             3     3001823\n",
      "20  (3001923, 3002023]    1.0             2     3001923\n",
      "21  (3002023, 3002123]    0.0             2     3002023\n",
      "22  (3002123, 3002223]    1.0             4     3002123\n",
      "23  (3002223, 3002323]   10.0             6     3002223\n",
      "24  (3002323, 3002423]   12.0             2     3002323\n",
      "25  (3002423, 3002523]    9.0             5     3002423\n",
      "26  (3002523, 3002623]    4.0             1     3002523\n",
      "27  (3002623, 3002723]    0.0             3     3002623\n",
      "28  (3002723, 3002823]    0.0             1     3002723\n",
      "29  (3002823, 3002923]    0.0             2     3002823\n",
      "30  (3002923, 3003023]    1.0             6     3002923\n",
      "31  (3003023, 3003123]    2.0             4     3003023\n",
      "32  (3003123, 3003223]    0.0             0     3003123\n",
      "33  (3003223, 3003323]    0.0             2     3003223\n",
      "34  (3003323, 3003423]    1.0             7     3003323\n",
      "35  (3003423, 3003523]    0.0             3     3003423\n",
      "36  (3003523, 3003623]    1.0             2     3003523\n",
      "37  (3003623, 3003723]    3.0             4     3003623\n",
      "38  (3003723, 3003823]    1.0             6     3003723\n",
      "39  (3003823, 3003923]    1.0             4     3003823\n",
      "40  (3003923, 3004023]    7.0             5     3003923\n",
      "41  (3004023, 3004123]    0.0             0     3004023\n",
      "42  (3004123, 3004223]    0.0             2     3004123\n",
      "43  (3004223, 3004323]    5.0             5     3004223\n",
      "44  (3004323, 3004423]    0.0             1     3004323\n",
      "45  (3004423, 3004523]    0.0             2     3004423\n",
      "46  (3004523, 3004623]    1.0             6     3004523\n",
      "47  (3004623, 3004723]    0.0             0     3004623\n",
      "48  (3004723, 3004823]    0.0             1     3004723\n",
      "49  (3004823, 3004923]   13.0             8     3004823\n",
      "(152618, 4)\n",
      "bin_offset_5    category\n",
      "TT_S0            float64\n",
      "target_count       int64\n",
      "starting_nt     category\n",
      "dtype: object\n",
      "sumos ilgis: 152618\n",
      "ranges ilgis: 152618\n",
      "Model input - NO SCALE\n",
      "        TT_S0  genes  transcripts  exons  CDSs\n",
      "0         0.0      0            0      0     0\n",
      "1         7.0      0            0      0     0\n",
      "2         1.0      0            0      0     0\n",
      "3         2.0      0            0      0     0\n",
      "4         0.0      0            0      0     0\n",
      "...       ...    ...          ...    ...   ...\n",
      "152613   10.0      0            0      0     0\n",
      "152614    3.0      0            0      0     0\n",
      "152615    4.0      0            0      0     0\n",
      "152616    0.0      0            0      0     0\n",
      "152617    4.0      0            0      0     0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "        TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0         0.0             0                   0             0            0\n",
      "1         7.0             0                   0             0            0\n",
      "2         1.0             0                   0             0            0\n",
      "3         2.0             0                   0             0            0\n",
      "4         0.0             0                   0             0            0\n",
      "...       ...           ...                 ...           ...          ...\n",
      "152613   10.0             0                   0             0            0\n",
      "152614    3.0             0                   0             0            0\n",
      "152615    4.0             0                   0             0            0\n",
      "152616    0.0             0                   0             0            0\n",
      "152617    4.0             0                   0             0            0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "0.15202662857592159\n",
      "[0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 1 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 1 0 0 0 0 0 0 0 0 1 0 1 1 1 0 1 1 1 1 1 1 0 1 1 0 0 0 1 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 0 0 1 1 1 1 1 1 1 0 1 1 0 1\n",
      " 0 1 0 0 0 0 0 0 0 0 0 0 0 0 1 0 1 1 1 1 0 1 1 0 0 0 1 0 0 0 0 0 0 0 0 0 1\n",
      " 1 1 0 0 0 0 0 0 1 0 0 0 1 0 1 0 0 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 1 1 0 1 1 1 1 0 0 0 1 1\n",
      " 0 0 0 0 0 0 1 0 1 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 1 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 1 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 1\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0]\n",
      "threshold density 1.30546\n",
      "cluster signal 165721.0\n",
      "cluster target count 525610\n",
      "!!cluster density 0.31529270752078536\n",
      "cluster signal 291190.0\n",
      "cluster target count 174390\n",
      "!!cluster density 1.6697631744939503\n",
      "\n",
      "Labels mapping\n",
      " {0: 0, 1: 1}\n",
      "DEBUG: 129416 23202 \n",
      "\n",
      " +----------------------+-------------+----------------+-------+\n",
      "| bin_offset_5         | TT_S0       | target_count   | +9    |\n",
      "| (category)           | (float64)   | (int64)        | ...   |\n",
      "|----------------------+-------------+----------------+-------|\n",
      "| (3000018, 3000023]   | 0.0         | 2              | ...   |\n",
      "| (3000123, 3000223]   | 1.0         | 5              | ...   |\n",
      "| (3000223, 3000323]   | 2.0         | 3              | ...   |\n",
      "| (3000323, 3000423]   | 0.0         | 4              | ...   |\n",
      "| ...                  | ...         | ...            | ...   |\n",
      "| (18261323, 18261423] | 3.0         | 14             | ...   |\n",
      "| (18261423, 18261523] | 4.0         | 4              | ...   |\n",
      "| (18261523, 18261623] | 0.0         | 3              | ...   |\n",
      "| (18261623, 18261710] | 4.0         | 3              | ...   |\n",
      "+----------------------+-------------+----------------+-------+\n",
      "Unstranded PyRanges object has 129,416 rows and 12 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "9 hidden columns: starting_nt, Chromosome, Start, End, genes, transcripts, ... (+ 3 more.)\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0\n",
      "BINS  IntervalIndex([  (3000018, 3000028],   (3000028, 3000128],\n",
      "                 (3000128, 3000228],   (3000228, 3000328],\n",
      "                 (3000328, 3000428],   (3000428, 3000528],\n",
      "                 (3000528, 3000628],   (3000628, 3000728],\n",
      "                 (3000728, 3000828],   (3000828, 3000928],\n",
      "               ...\n",
      "               (18260728, 18260828], (18260828, 18260928],\n",
      "               (18260928, 18261028], (18261028, 18261128],\n",
      "               (18261128, 18261228], (18261228, 18261328],\n",
      "               (18261328, 18261428], (18261428, 18261528],\n",
      "               (18261528, 18261628], (18261628, 18261710]],\n",
      "              dtype='interval[int64, right]', length=152618)\n",
      "HEAD with BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0   \n",
      "\n",
      "        bin_offset_10  target_count  \n",
      "0  (3000018, 3000028]             1  \n",
      "1  (3000018, 3000028]             1  \n",
      "2  (3000018, 3000028]             1  \n",
      "3  (3000028, 3000128]             1  \n",
      "4  (3000028, 3000128]             1  \n",
      "TAIL with BINS\n",
      "       seqnames     start       end        CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  18261611  18261611  18_18261611    0.0    0.0    0.0   \n",
      "699996       18  18261615  18261615  18_18261615    0.0    0.0    0.0   \n",
      "699997       18  18261682  18261682  18_18261682    0.0    0.0    0.0   \n",
      "699998       18  18261702  18261702  18_18261702    4.0    0.0    0.0   \n",
      "699999       18  18261710  18261710  18_18261710    0.0    0.0    0.0   \n",
      "\n",
      "               bin_offset_10  target_count  \n",
      "699995  (18261528, 18261628]             1  \n",
      "699996  (18261528, 18261628]             1  \n",
      "699997  (18261628, 18261710]             1  \n",
      "699998  (18261628, 18261710]             1  \n",
      "699999  (18261628, 18261710]             1  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:137: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "         bin_offset_10  TT_S0  target_count starting_nt\n",
      "0   (3000018, 3000028]    0.0             3     3000018\n",
      "1   (3000028, 3000128]    7.0             8     3000028\n",
      "2   (3000128, 3000228]    1.0             5     3000128\n",
      "3   (3000228, 3000328]    2.0             3     3000228\n",
      "4   (3000328, 3000428]    0.0             5     3000328\n",
      "5   (3000428, 3000528]    0.0             1     3000428\n",
      "6   (3000528, 3000628]    0.0             2     3000528\n",
      "7   (3000628, 3000728]    0.0             4     3000628\n",
      "8   (3000728, 3000828]    0.0             7     3000728\n",
      "9   (3000828, 3000928]    0.0             4     3000828\n",
      "10  (3000928, 3001028]    0.0             2     3000928\n",
      "11  (3001028, 3001128]    2.0             6     3001028\n",
      "12  (3001128, 3001228]    0.0             2     3001128\n",
      "13  (3001228, 3001328]    0.0             5     3001228\n",
      "14  (3001328, 3001428]    0.0             2     3001328\n",
      "15  (3001428, 3001528]    1.0             6     3001428\n",
      "16  (3001528, 3001628]    3.0             9     3001528\n",
      "17  (3001628, 3001728]    3.0             7     3001628\n",
      "18  (3001728, 3001828]    0.0             5     3001728\n",
      "19  (3001828, 3001928]    0.0             3     3001828\n",
      "20  (3001928, 3002028]    1.0             2     3001928\n",
      "21  (3002028, 3002128]    0.0             2     3002028\n",
      "22  (3002128, 3002228]    1.0             4     3002128\n",
      "23  (3002228, 3002328]   10.0             6     3002228\n",
      "24  (3002328, 3002428]   12.0             2     3002328\n",
      "25  (3002428, 3002528]   13.0             6     3002428\n",
      "26  (3002528, 3002628]    0.0             0     3002528\n",
      "27  (3002628, 3002728]    0.0             3     3002628\n",
      "28  (3002728, 3002828]    0.0             2     3002728\n",
      "29  (3002828, 3002928]    0.0             1     3002828\n",
      "30  (3002928, 3003028]    1.0             6     3002928\n",
      "31  (3003028, 3003128]    2.0             4     3003028\n",
      "32  (3003128, 3003228]    0.0             0     3003128\n",
      "33  (3003228, 3003328]    0.0             3     3003228\n",
      "34  (3003328, 3003428]    1.0             6     3003328\n",
      "35  (3003428, 3003528]    0.0             3     3003428\n",
      "36  (3003528, 3003628]    1.0             2     3003528\n",
      "37  (3003628, 3003728]    3.0             4     3003628\n",
      "38  (3003728, 3003828]    1.0             6     3003728\n",
      "39  (3003828, 3003928]    1.0             4     3003828\n",
      "40  (3003928, 3004028]    7.0             5     3003928\n",
      "41  (3004028, 3004128]    0.0             0     3004028\n",
      "42  (3004128, 3004228]    0.0             3     3004128\n",
      "43  (3004228, 3004328]    5.0             4     3004228\n",
      "44  (3004328, 3004428]    0.0             1     3004328\n",
      "45  (3004428, 3004528]    0.0             3     3004428\n",
      "46  (3004528, 3004628]    1.0             5     3004528\n",
      "47  (3004628, 3004728]    0.0             0     3004628\n",
      "48  (3004728, 3004828]    0.0             1     3004728\n",
      "49  (3004828, 3004928]   13.0             8     3004828\n",
      "(152618, 4)\n",
      "bin_offset_10    category\n",
      "TT_S0             float64\n",
      "target_count        int64\n",
      "starting_nt      category\n",
      "dtype: object\n",
      "sumos ilgis: 152618\n",
      "ranges ilgis: 152618\n",
      "Model input - NO SCALE\n",
      "        TT_S0  genes  transcripts  exons  CDSs\n",
      "0         0.0      0            0      0     0\n",
      "1         7.0      0            0      0     0\n",
      "2         1.0      0            0      0     0\n",
      "3         2.0      0            0      0     0\n",
      "4         0.0      0            0      0     0\n",
      "...       ...    ...          ...    ...   ...\n",
      "152613   10.0      0            0      0     0\n",
      "152614    3.0      0            0      0     0\n",
      "152615    4.0      0            0      0     0\n",
      "152616    0.0      0            0      0     0\n",
      "152617    4.0      0            0      0     0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "        TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0         0.0             0                   0             0            0\n",
      "1         7.0             0                   0             0            0\n",
      "2         1.0             0                   0             0            0\n",
      "3         2.0             0                   0             0            0\n",
      "4         0.0             0                   0             0            0\n",
      "...       ...           ...                 ...           ...          ...\n",
      "152613   10.0             0                   0             0            0\n",
      "152614    3.0             0                   0             0            0\n",
      "152615    4.0             0                   0             0            0\n",
      "152616    0.0             0                   0             0            0\n",
      "152617    4.0             0                   0             0            0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "0.848438585225858\n",
      "[1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 0 0 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 0 1 1 1 1 1 1 1 1 0 1 0 0 0 1 0 0 0 0 0 0 1 0 0 1 1 1 0 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 0 1 1 1 1 1 1 1 1 1 0 0 1 1 1 1 1 1 0 0 0 0 0 0 0 0 0 0 1 0\n",
      " 1 0 1 1 1 1 1 1 1 1 1 1 1 1 0 1 0 0 0 0 1 0 0 1 1 1 0 1 1 1 1 1 1 1 1 1 0\n",
      " 0 0 1 1 1 1 1 1 0 1 1 1 0 1 0 1 1 0 0 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 1 1 1 1 1 0 0 1 0 0 0 1 1 1 1 0 0\n",
      " 1 1 1 1 1 1 0 0 0 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 0 1 1 1 1 1 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 0 1 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 1 1 0\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 0 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1]\n",
      "threshold density 1.30546\n",
      "cluster signal 166125.0\n",
      "cluster target count 526296\n",
      "!!cluster density 0.31564936841625246\n",
      "cluster signal 290786.0\n",
      "cluster target count 173704\n",
      "!!cluster density 1.6740316860866762\n",
      "\n",
      "Labels mapping\n",
      " {1: 0, 0: 1}\n",
      "DEBUG: 129487 23131 \n",
      "\n",
      " +----------------------+-------------+----------------+-------+\n",
      "| bin_offset_10        | TT_S0       | target_count   | +9    |\n",
      "| (category)           | (float64)   | (int64)        | ...   |\n",
      "|----------------------+-------------+----------------+-------|\n",
      "| (3000018, 3000028]   | 0.0         | 3              | ...   |\n",
      "| (3000128, 3000228]   | 1.0         | 5              | ...   |\n",
      "| (3000228, 3000328]   | 2.0         | 3              | ...   |\n",
      "| (3000328, 3000428]   | 0.0         | 5              | ...   |\n",
      "| ...                  | ...         | ...            | ...   |\n",
      "| (18261328, 18261428] | 3.0         | 14             | ...   |\n",
      "| (18261428, 18261528] | 4.0         | 4              | ...   |\n",
      "| (18261528, 18261628] | 0.0         | 3              | ...   |\n",
      "| (18261628, 18261710] | 4.0         | 3              | ...   |\n",
      "+----------------------+-------------+----------------+-------+\n",
      "Unstranded PyRanges object has 129,487 rows and 12 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "9 hidden columns: starting_nt, Chromosome, Start, End, genes, transcripts, ... (+ 3 more.)\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0\n",
      "BINS  IntervalIndex([  (3000018, 3000033],   (3000033, 3000133],\n",
      "                 (3000133, 3000233],   (3000233, 3000333],\n",
      "                 (3000333, 3000433],   (3000433, 3000533],\n",
      "                 (3000533, 3000633],   (3000633, 3000733],\n",
      "                 (3000733, 3000833],   (3000833, 3000933],\n",
      "               ...\n",
      "               (18260733, 18260833], (18260833, 18260933],\n",
      "               (18260933, 18261033], (18261033, 18261133],\n",
      "               (18261133, 18261233], (18261233, 18261333],\n",
      "               (18261333, 18261433], (18261433, 18261533],\n",
      "               (18261533, 18261633], (18261633, 18261710]],\n",
      "              dtype='interval[int64, right]', length=152618)\n",
      "HEAD with BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0   \n",
      "\n",
      "        bin_offset_15  target_count  \n",
      "0  (3000018, 3000033]             1  \n",
      "1  (3000018, 3000033]             1  \n",
      "2  (3000018, 3000033]             1  \n",
      "3  (3000033, 3000133]             1  \n",
      "4  (3000033, 3000133]             1  \n",
      "TAIL with BINS\n",
      "       seqnames     start       end        CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  18261611  18261611  18_18261611    0.0    0.0    0.0   \n",
      "699996       18  18261615  18261615  18_18261615    0.0    0.0    0.0   \n",
      "699997       18  18261682  18261682  18_18261682    0.0    0.0    0.0   \n",
      "699998       18  18261702  18261702  18_18261702    4.0    0.0    0.0   \n",
      "699999       18  18261710  18261710  18_18261710    0.0    0.0    0.0   \n",
      "\n",
      "               bin_offset_15  target_count  \n",
      "699995  (18261533, 18261633]             1  \n",
      "699996  (18261533, 18261633]             1  \n",
      "699997  (18261633, 18261710]             1  \n",
      "699998  (18261633, 18261710]             1  \n",
      "699999  (18261633, 18261710]             1  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:137: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "         bin_offset_15  TT_S0  target_count starting_nt\n",
      "0   (3000018, 3000033]    0.0             3     3000018\n",
      "1   (3000033, 3000133]    8.0             9     3000033\n",
      "2   (3000133, 3000233]    0.0             4     3000133\n",
      "3   (3000233, 3000333]    2.0             3     3000233\n",
      "4   (3000333, 3000433]    0.0             5     3000333\n",
      "5   (3000433, 3000533]    0.0             1     3000433\n",
      "6   (3000533, 3000633]    0.0             2     3000533\n",
      "7   (3000633, 3000733]    0.0             4     3000633\n",
      "8   (3000733, 3000833]    0.0             7     3000733\n",
      "9   (3000833, 3000933]    0.0             4     3000833\n",
      "10  (3000933, 3001033]    0.0             3     3000933\n",
      "11  (3001033, 3001133]    2.0             5     3001033\n",
      "12  (3001133, 3001233]    0.0             2     3001133\n",
      "13  (3001233, 3001333]    0.0             5     3001233\n",
      "14  (3001333, 3001433]    0.0             2     3001333\n",
      "15  (3001433, 3001533]    1.0             7     3001433\n",
      "16  (3001533, 3001633]    3.0             8     3001533\n",
      "17  (3001633, 3001733]    3.0             7     3001633\n",
      "18  (3001733, 3001833]    0.0             6     3001733\n",
      "19  (3001833, 3001933]    0.0             2     3001833\n",
      "20  (3001933, 3002033]    1.0             2     3001933\n",
      "21  (3002033, 3002133]    0.0             3     3002033\n",
      "22  (3002133, 3002233]    4.0             4     3002133\n",
      "23  (3002233, 3002333]    7.0             5     3002233\n",
      "24  (3002333, 3002433]   12.0             2     3002333\n",
      "25  (3002433, 3002533]   13.0             6     3002433\n",
      "26  (3002533, 3002633]    0.0             0     3002533\n",
      "27  (3002633, 3002733]    0.0             3     3002633\n",
      "28  (3002733, 3002833]    0.0             2     3002733\n",
      "29  (3002833, 3002933]    0.0             1     3002833\n",
      "30  (3002933, 3003033]    1.0             6     3002933\n",
      "31  (3003033, 3003133]    2.0             4     3003033\n",
      "32  (3003133, 3003233]    0.0             0     3003133\n",
      "33  (3003233, 3003333]    0.0             4     3003233\n",
      "34  (3003333, 3003433]    1.0             5     3003333\n",
      "35  (3003433, 3003533]    0.0             3     3003433\n",
      "36  (3003533, 3003633]    1.0             2     3003533\n",
      "37  (3003633, 3003733]    3.0             5     3003633\n",
      "38  (3003733, 3003833]    1.0             5     3003733\n",
      "39  (3003833, 3003933]    1.0             4     3003833\n",
      "40  (3003933, 3004033]    7.0             5     3003933\n",
      "41  (3004033, 3004133]    0.0             0     3004033\n",
      "42  (3004133, 3004233]    0.0             3     3004133\n",
      "43  (3004233, 3004333]    5.0             4     3004233\n",
      "44  (3004333, 3004433]    0.0             1     3004333\n",
      "45  (3004433, 3004533]    0.0             3     3004433\n",
      "46  (3004533, 3004633]    1.0             5     3004533\n",
      "47  (3004633, 3004733]    0.0             0     3004633\n",
      "48  (3004733, 3004833]    0.0             1     3004733\n",
      "49  (3004833, 3004933]   13.0             8     3004833\n",
      "(152618, 4)\n",
      "bin_offset_15    category\n",
      "TT_S0             float64\n",
      "target_count        int64\n",
      "starting_nt      category\n",
      "dtype: object\n",
      "sumos ilgis: 152618\n",
      "ranges ilgis: 152618\n",
      "Model input - NO SCALE\n",
      "        TT_S0  genes  transcripts  exons  CDSs\n",
      "0         0.0      0            0      0     0\n",
      "1         8.0      0            0      0     0\n",
      "2         0.0      0            0      0     0\n",
      "3         2.0      0            0      0     0\n",
      "4         0.0      0            0      0     0\n",
      "...       ...    ...          ...    ...   ...\n",
      "152613   13.0      0            0      0     0\n",
      "152614    0.0      0            0      0     0\n",
      "152615    4.0      0            0      0     0\n",
      "152616    0.0      0            0      0     0\n",
      "152617    4.0      0            0      0     0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "        TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0         0.0             0                   0             0            0\n",
      "1         8.0             0                   0             0            0\n",
      "2         0.0             0                   0             0            0\n",
      "3         2.0             0                   0             0            0\n",
      "4         0.0             0                   0             0            0\n",
      "...       ...           ...                 ...           ...          ...\n",
      "152613   13.0             0                   0             0            0\n",
      "152614    0.0             0                   0             0            0\n",
      "152615    4.0             0                   0             0            0\n",
      "152616    0.0             0                   0             0            0\n",
      "152617    4.0             0                   0             0            0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "0.10486312230536372\n",
      "[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 1 0 1 1 1 0 1 1 1 1 1 1 0 1 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 0 0 1 1 1 1 1 0 0 1 0 1 0 1\n",
      " 0 1 0 0 0 0 0 0 0 0 0 0 0 0 1 0 1 1 1 1 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 1\n",
      " 0 0 0 0 0 0 0 0 1 0 0 0 1 0 1 0 0 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 1 1 0 1 1 1 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 1 0 1 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0]\n",
      "threshold density 1.30546\n",
      "cluster signal 219126.0\n",
      "cluster target count 572672\n",
      "!!cluster density 0.3826378799731784\n",
      "cluster signal 237785.0\n",
      "cluster target count 127328\n",
      "!!cluster density 1.8674996858507162\n",
      "\n",
      "Labels mapping\n",
      " {0: 0, 1: 1}\n",
      "DEBUG: 136614 16004 \n",
      "\n",
      " +----------------------+-------------+----------------+-------+\n",
      "| bin_offset_15        | TT_S0       | target_count   | +9    |\n",
      "| (category)           | (float64)   | (int64)        | ...   |\n",
      "|----------------------+-------------+----------------+-------|\n",
      "| (3000018, 3000033]   | 0.0         | 3              | ...   |\n",
      "| (3000033, 3000133]   | 8.0         | 9              | ...   |\n",
      "| (3000133, 3000233]   | 0.0         | 4              | ...   |\n",
      "| (3000233, 3000333]   | 2.0         | 3              | ...   |\n",
      "| ...                  | ...         | ...            | ...   |\n",
      "| (18261333, 18261433] | 0.0         | 12             | ...   |\n",
      "| (18261433, 18261533] | 4.0         | 4              | ...   |\n",
      "| (18261533, 18261633] | 0.0         | 3              | ...   |\n",
      "| (18261633, 18261710] | 4.0         | 3              | ...   |\n",
      "+----------------------+-------------+----------------+-------+\n",
      "Unstranded PyRanges object has 136,614 rows and 12 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "9 hidden columns: starting_nt, Chromosome, Start, End, genes, transcripts, ... (+ 3 more.)\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0\n",
      "BINS  IntervalIndex([  (3000018, 3000038],   (3000038, 3000138],\n",
      "                 (3000138, 3000238],   (3000238, 3000338],\n",
      "                 (3000338, 3000438],   (3000438, 3000538],\n",
      "                 (3000538, 3000638],   (3000638, 3000738],\n",
      "                 (3000738, 3000838],   (3000838, 3000938],\n",
      "               ...\n",
      "               (18260738, 18260838], (18260838, 18260938],\n",
      "               (18260938, 18261038], (18261038, 18261138],\n",
      "               (18261138, 18261238], (18261238, 18261338],\n",
      "               (18261338, 18261438], (18261438, 18261538],\n",
      "               (18261538, 18261638], (18261638, 18261710]],\n",
      "              dtype='interval[int64, right]', length=152618)\n",
      "HEAD with BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0   \n",
      "\n",
      "        bin_offset_20  target_count  \n",
      "0  (3000018, 3000038]             1  \n",
      "1  (3000018, 3000038]             1  \n",
      "2  (3000018, 3000038]             1  \n",
      "3  (3000038, 3000138]             1  \n",
      "4  (3000038, 3000138]             1  \n",
      "TAIL with BINS\n",
      "       seqnames     start       end        CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  18261611  18261611  18_18261611    0.0    0.0    0.0   \n",
      "699996       18  18261615  18261615  18_18261615    0.0    0.0    0.0   \n",
      "699997       18  18261682  18261682  18_18261682    0.0    0.0    0.0   \n",
      "699998       18  18261702  18261702  18_18261702    4.0    0.0    0.0   \n",
      "699999       18  18261710  18261710  18_18261710    0.0    0.0    0.0   \n",
      "\n",
      "               bin_offset_20  target_count  \n",
      "699995  (18261538, 18261638]             1  \n",
      "699996  (18261538, 18261638]             1  \n",
      "699997  (18261638, 18261710]             1  \n",
      "699998  (18261638, 18261710]             1  \n",
      "699999  (18261638, 18261710]             1  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:137: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "         bin_offset_20  TT_S0  target_count starting_nt\n",
      "0   (3000018, 3000038]    0.0             3     3000018\n",
      "1   (3000038, 3000138]    8.0            10     3000038\n",
      "2   (3000138, 3000238]    0.0             3     3000138\n",
      "3   (3000238, 3000338]    2.0             3     3000238\n",
      "4   (3000338, 3000438]    0.0             5     3000338\n",
      "5   (3000438, 3000538]    0.0             1     3000438\n",
      "6   (3000538, 3000638]    0.0             2     3000538\n",
      "7   (3000638, 3000738]    0.0             5     3000638\n",
      "8   (3000738, 3000838]    0.0             6     3000738\n",
      "9   (3000838, 3000938]    0.0             4     3000838\n",
      "10  (3000938, 3001038]    0.0             3     3000938\n",
      "11  (3001038, 3001138]    2.0             5     3001038\n",
      "12  (3001138, 3001238]    0.0             2     3001138\n",
      "13  (3001238, 3001338]    0.0             5     3001238\n",
      "14  (3001338, 3001438]    0.0             2     3001338\n",
      "15  (3001438, 3001538]    1.0             7     3001438\n",
      "16  (3001538, 3001638]    3.0             8     3001538\n",
      "17  (3001638, 3001738]    3.0             7     3001638\n",
      "18  (3001738, 3001838]    0.0             6     3001738\n",
      "19  (3001838, 3001938]    0.0             2     3001838\n",
      "20  (3001938, 3002038]    1.0             2     3001938\n",
      "21  (3002038, 3002138]    0.0             3     3002038\n",
      "22  (3002138, 3002238]    4.0             5     3002138\n",
      "23  (3002238, 3002338]    7.0             4     3002238\n",
      "24  (3002338, 3002438]   12.0             2     3002338\n",
      "25  (3002438, 3002538]   13.0             6     3002438\n",
      "26  (3002538, 3002638]    0.0             0     3002538\n",
      "27  (3002638, 3002738]    0.0             3     3002638\n",
      "28  (3002738, 3002838]    0.0             3     3002738\n",
      "29  (3002838, 3002938]    0.0             0     3002838\n",
      "30  (3002938, 3003038]    1.0             8     3002938\n",
      "31  (3003038, 3003138]    2.0             2     3003038\n",
      "32  (3003138, 3003238]    0.0             0     3003138\n",
      "33  (3003238, 3003338]    0.0             4     3003238\n",
      "34  (3003338, 3003438]    1.0             5     3003338\n",
      "35  (3003438, 3003538]    0.0             3     3003438\n",
      "36  (3003538, 3003638]    1.0             2     3003538\n",
      "37  (3003638, 3003738]    3.0             5     3003638\n",
      "38  (3003738, 3003838]    1.0             5     3003738\n",
      "39  (3003838, 3003938]    1.0             4     3003838\n",
      "40  (3003938, 3004038]    7.0             5     3003938\n",
      "41  (3004038, 3004138]    0.0             0     3004038\n",
      "42  (3004138, 3004238]    1.0             4     3004138\n",
      "43  (3004238, 3004338]    4.0             3     3004238\n",
      "44  (3004338, 3004438]    0.0             1     3004338\n",
      "45  (3004438, 3004538]    0.0             3     3004438\n",
      "46  (3004538, 3004638]    1.0             5     3004538\n",
      "47  (3004638, 3004738]    0.0             0     3004638\n",
      "48  (3004738, 3004838]    1.0             2     3004738\n",
      "49  (3004838, 3004938]   12.0             7     3004838\n",
      "(152618, 4)\n",
      "bin_offset_20    category\n",
      "TT_S0             float64\n",
      "target_count        int64\n",
      "starting_nt      category\n",
      "dtype: object\n",
      "sumos ilgis: 152618\n",
      "ranges ilgis: 152618\n",
      "Model input - NO SCALE\n",
      "        TT_S0  genes  transcripts  exons  CDSs\n",
      "0         0.0      0            0      0     0\n",
      "1         8.0      0            0      0     0\n",
      "2         0.0      0            0      0     0\n",
      "3         2.0      0            0      0     0\n",
      "4         0.0      0            0      0     0\n",
      "...       ...    ...          ...    ...   ...\n",
      "152613   13.0      0            0      0     0\n",
      "152614    1.0      0            0      0     0\n",
      "152615    3.0      0            0      0     0\n",
      "152616    0.0      0            0      0     0\n",
      "152617    4.0      0            0      0     0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "        TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0         0.0             0                   0             0            0\n",
      "1         8.0             0                   0             0            0\n",
      "2         0.0             0                   0             0            0\n",
      "3         2.0             0                   0             0            0\n",
      "4         0.0             0                   0             0            0\n",
      "...       ...           ...                 ...           ...          ...\n",
      "152613   13.0             0                   0             0            0\n",
      "152614    1.0             0                   0             0            0\n",
      "152615    3.0             0                   0             0            0\n",
      "152616    0.0             0                   0             0            0\n",
      "152617    4.0             0                   0             0            0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "0.15105688712995846\n",
      "[0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 1 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 1 0 0 0 0 0 0 0 0 1 0 1 1 1 0 1 1 1 1 1 1 0 1 0 0 0 0 1 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 0 0 1 1 1 1 1 0 0 1 1 1 0 1\n",
      " 0 1 0 0 0 0 0 0 0 0 0 0 0 0 1 0 1 1 1 1 0 1 0 0 0 0 1 0 0 0 0 0 0 0 0 0 1\n",
      " 1 1 0 0 0 0 0 0 1 0 0 0 1 0 1 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 1 1 1 0 1 1 0 0 0 0 1 0\n",
      " 0 0 0 0 0 0 1 1 1 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 1 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 1 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0\n",
      " 0]\n",
      "threshold density 1.30546\n",
      "cluster signal 166589.0\n",
      "cluster target count 526759\n",
      "!!cluster density 0.31625278353098857\n",
      "cluster signal 290322.0\n",
      "cluster target count 173241\n",
      "!!cluster density 1.6758273157047119\n",
      "\n",
      "Labels mapping\n",
      " {0: 0, 1: 1}\n",
      "DEBUG: 129564 23054 \n",
      "\n",
      " +----------------------+-------------+----------------+-------+\n",
      "| bin_offset_20        | TT_S0       | target_count   | +9    |\n",
      "| (category)           | (float64)   | (int64)        | ...   |\n",
      "|----------------------+-------------+----------------+-------|\n",
      "| (3000018, 3000038]   | 0.0         | 3              | ...   |\n",
      "| (3000138, 3000238]   | 0.0         | 3              | ...   |\n",
      "| (3000238, 3000338]   | 2.0         | 3              | ...   |\n",
      "| (3000338, 3000438]   | 0.0         | 5              | ...   |\n",
      "| ...                  | ...         | ...            | ...   |\n",
      "| (18261338, 18261438] | 1.0         | 12             | ...   |\n",
      "| (18261438, 18261538] | 3.0         | 4              | ...   |\n",
      "| (18261538, 18261638] | 0.0         | 2              | ...   |\n",
      "| (18261638, 18261710] | 4.0         | 3              | ...   |\n",
      "+----------------------+-------------+----------------+-------+\n",
      "Unstranded PyRanges object has 129,564 rows and 12 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "9 hidden columns: starting_nt, Chromosome, Start, End, genes, transcripts, ... (+ 3 more.)\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0\n",
      "BINS  IntervalIndex([  (3000018, 3000043],   (3000043, 3000143],\n",
      "                 (3000143, 3000243],   (3000243, 3000343],\n",
      "                 (3000343, 3000443],   (3000443, 3000543],\n",
      "                 (3000543, 3000643],   (3000643, 3000743],\n",
      "                 (3000743, 3000843],   (3000843, 3000943],\n",
      "               ...\n",
      "               (18260743, 18260843], (18260843, 18260943],\n",
      "               (18260943, 18261043], (18261043, 18261143],\n",
      "               (18261143, 18261243], (18261243, 18261343],\n",
      "               (18261343, 18261443], (18261443, 18261543],\n",
      "               (18261543, 18261643], (18261643, 18261710]],\n",
      "              dtype='interval[int64, right]', length=152618)\n",
      "HEAD with BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0   \n",
      "\n",
      "        bin_offset_25  target_count  \n",
      "0  (3000018, 3000043]             1  \n",
      "1  (3000018, 3000043]             1  \n",
      "2  (3000018, 3000043]             1  \n",
      "3  (3000018, 3000043]             1  \n",
      "4  (3000043, 3000143]             1  \n",
      "TAIL with BINS\n",
      "       seqnames     start       end        CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  18261611  18261611  18_18261611    0.0    0.0    0.0   \n",
      "699996       18  18261615  18261615  18_18261615    0.0    0.0    0.0   \n",
      "699997       18  18261682  18261682  18_18261682    0.0    0.0    0.0   \n",
      "699998       18  18261702  18261702  18_18261702    4.0    0.0    0.0   \n",
      "699999       18  18261710  18261710  18_18261710    0.0    0.0    0.0   \n",
      "\n",
      "               bin_offset_25  target_count  \n",
      "699995  (18261543, 18261643]             1  \n",
      "699996  (18261543, 18261643]             1  \n",
      "699997  (18261643, 18261710]             1  \n",
      "699998  (18261643, 18261710]             1  \n",
      "699999  (18261643, 18261710]             1  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:137: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "         bin_offset_25  TT_S0  target_count starting_nt\n",
      "0   (3000018, 3000043]    4.0             4     3000018\n",
      "1   (3000043, 3000143]    4.0             9     3000043\n",
      "2   (3000143, 3000243]    0.0             3     3000143\n",
      "3   (3000243, 3000343]    2.0             3     3000243\n",
      "4   (3000343, 3000443]    0.0             5     3000343\n",
      "5   (3000443, 3000543]    0.0             1     3000443\n",
      "6   (3000543, 3000643]    0.0             2     3000543\n",
      "7   (3000643, 3000743]    0.0             5     3000643\n",
      "8   (3000743, 3000843]    0.0             6     3000743\n",
      "9   (3000843, 3000943]    0.0             4     3000843\n",
      "10  (3000943, 3001043]    2.0             4     3000943\n",
      "11  (3001043, 3001143]    0.0             4     3001043\n",
      "12  (3001143, 3001243]    0.0             2     3001143\n",
      "13  (3001243, 3001343]    0.0             5     3001243\n",
      "14  (3001343, 3001443]    0.0             2     3001343\n",
      "15  (3001443, 3001543]    1.0             7     3001443\n",
      "16  (3001543, 3001643]    3.0             8     3001543\n",
      "17  (3001643, 3001743]    3.0             8     3001643\n",
      "18  (3001743, 3001843]    0.0             5     3001743\n",
      "19  (3001843, 3001943]    0.0             2     3001843\n",
      "20  (3001943, 3002043]    1.0             2     3001943\n",
      "21  (3002043, 3002143]    0.0             3     3002043\n",
      "22  (3002143, 3002243]    4.0             5     3002143\n",
      "23  (3002243, 3002343]    7.0             4     3002243\n",
      "24  (3002343, 3002443]   12.0             3     3002343\n",
      "25  (3002443, 3002543]   13.0             5     3002443\n",
      "26  (3002543, 3002643]    0.0             0     3002543\n",
      "27  (3002643, 3002743]    0.0             3     3002643\n",
      "28  (3002743, 3002843]    0.0             3     3002743\n",
      "29  (3002843, 3002943]    0.0             0     3002843\n",
      "30  (3002943, 3003043]    1.0             8     3002943\n",
      "31  (3003043, 3003143]    2.0             2     3003043\n",
      "32  (3003143, 3003243]    0.0             0     3003143\n",
      "33  (3003243, 3003343]    1.0             5     3003243\n",
      "34  (3003343, 3003443]    0.0             4     3003343\n",
      "35  (3003443, 3003543]    0.0             3     3003443\n",
      "36  (3003543, 3003643]    1.0             2     3003543\n",
      "37  (3003643, 3003743]    3.0             5     3003643\n",
      "38  (3003743, 3003843]    1.0             5     3003743\n",
      "39  (3003843, 3003943]    1.0             5     3003843\n",
      "40  (3003943, 3004043]    7.0             4     3003943\n",
      "41  (3004043, 3004143]    0.0             0     3004043\n",
      "42  (3004143, 3004243]    1.0             4     3004143\n",
      "43  (3004243, 3004343]    4.0             3     3004243\n",
      "44  (3004343, 3004443]    0.0             1     3004343\n",
      "45  (3004443, 3004543]    0.0             3     3004443\n",
      "46  (3004543, 3004643]    1.0             5     3004543\n",
      "47  (3004643, 3004743]    0.0             0     3004643\n",
      "48  (3004743, 3004843]    2.0             3     3004743\n",
      "49  (3004843, 3004943]   11.0             6     3004843\n",
      "(152618, 4)\n",
      "bin_offset_25    category\n",
      "TT_S0             float64\n",
      "target_count        int64\n",
      "starting_nt      category\n",
      "dtype: object\n",
      "sumos ilgis: 152618\n",
      "ranges ilgis: 152618\n",
      "Model input - NO SCALE\n",
      "        TT_S0  genes  transcripts  exons  CDSs\n",
      "0         4.0      0            0      0     0\n",
      "1         4.0      0            0      0     0\n",
      "2         0.0      0            0      0     0\n",
      "3         2.0      0            0      0     0\n",
      "4         0.0      0            0      0     0\n",
      "...       ...    ...          ...    ...   ...\n",
      "152613   13.0      0            0      0     0\n",
      "152614    1.0      0            0      0     0\n",
      "152615    3.0      0            0      0     0\n",
      "152616    0.0      0            0      0     0\n",
      "152617    4.0      0            0      0     0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "        TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0         4.0             0                   0             0            0\n",
      "1         4.0             0                   0             0            0\n",
      "2         0.0             0                   0             0            0\n",
      "3         2.0             0                   0             0            0\n",
      "4         0.0             0                   0             0            0\n",
      "...       ...           ...                 ...           ...          ...\n",
      "152613   13.0             0                   0             0            0\n",
      "152614    1.0             0                   0             0            0\n",
      "152615    3.0             0                   0             0            0\n",
      "152616    0.0             0                   0             0            0\n",
      "152617    4.0             0                   0             0            0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "0.10489588384070031\n",
      "[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 1 0 1 1 1 0 1 1 1 1 1 1 0 1 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 0 0 1 1 1 1 1 0 0 1 0 1 0 1\n",
      " 0 1 0 0 0 0 0 0 0 0 0 0 0 0 1 0 1 1 1 1 0 1 0 0 0 0 1 0 0 0 0 0 0 0 0 0 1\n",
      " 0 0 0 0 0 0 0 1 0 0 0 0 1 0 1 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 1 0 1 0 1 1 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 1 0 1 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 1 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 1 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0]\n",
      "threshold density 1.30546\n",
      "cluster signal 219194.0\n",
      "cluster target count 572548\n",
      "!!cluster density 0.3828395173854419\n",
      "cluster signal 237717.0\n",
      "cluster target count 127452\n",
      "!!cluster density 1.8651492326522927\n",
      "\n",
      "Labels mapping\n",
      " {0: 0, 1: 1}\n",
      "DEBUG: 136609 16009 \n",
      "\n",
      " +----------------------+-------------+----------------+-------+\n",
      "| bin_offset_25        | TT_S0       | target_count   | +9    |\n",
      "| (category)           | (float64)   | (int64)        | ...   |\n",
      "|----------------------+-------------+----------------+-------|\n",
      "| (3000018, 3000043]   | 4.0         | 4              | ...   |\n",
      "| (3000043, 3000143]   | 4.0         | 9              | ...   |\n",
      "| (3000143, 3000243]   | 0.0         | 3              | ...   |\n",
      "| (3000243, 3000343]   | 2.0         | 3              | ...   |\n",
      "| ...                  | ...         | ...            | ...   |\n",
      "| (18261343, 18261443] | 1.0         | 12             | ...   |\n",
      "| (18261443, 18261543] | 3.0         | 4              | ...   |\n",
      "| (18261543, 18261643] | 0.0         | 2              | ...   |\n",
      "| (18261643, 18261710] | 4.0         | 3              | ...   |\n",
      "+----------------------+-------------+----------------+-------+\n",
      "Unstranded PyRanges object has 136,609 rows and 12 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "9 hidden columns: starting_nt, Chromosome, Start, End, genes, transcripts, ... (+ 3 more.)\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0\n",
      "BINS  IntervalIndex([  (3000018, 3000048],   (3000048, 3000148],\n",
      "                 (3000148, 3000248],   (3000248, 3000348],\n",
      "                 (3000348, 3000448],   (3000448, 3000548],\n",
      "                 (3000548, 3000648],   (3000648, 3000748],\n",
      "                 (3000748, 3000848],   (3000848, 3000948],\n",
      "               ...\n",
      "               (18260748, 18260848], (18260848, 18260948],\n",
      "               (18260948, 18261048], (18261048, 18261148],\n",
      "               (18261148, 18261248], (18261248, 18261348],\n",
      "               (18261348, 18261448], (18261448, 18261548],\n",
      "               (18261548, 18261648], (18261648, 18261710]],\n",
      "              dtype='interval[int64, right]', length=152618)\n",
      "HEAD with BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0   \n",
      "\n",
      "        bin_offset_30  target_count  \n",
      "0  (3000018, 3000048]             1  \n",
      "1  (3000018, 3000048]             1  \n",
      "2  (3000018, 3000048]             1  \n",
      "3  (3000018, 3000048]             1  \n",
      "4  (3000048, 3000148]             1  \n",
      "TAIL with BINS\n",
      "       seqnames     start       end        CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  18261611  18261611  18_18261611    0.0    0.0    0.0   \n",
      "699996       18  18261615  18261615  18_18261615    0.0    0.0    0.0   \n",
      "699997       18  18261682  18261682  18_18261682    0.0    0.0    0.0   \n",
      "699998       18  18261702  18261702  18_18261702    4.0    0.0    0.0   \n",
      "699999       18  18261710  18261710  18_18261710    0.0    0.0    0.0   \n",
      "\n",
      "               bin_offset_30  target_count  \n",
      "699995  (18261548, 18261648]             1  \n",
      "699996  (18261548, 18261648]             1  \n",
      "699997  (18261648, 18261710]             1  \n",
      "699998  (18261648, 18261710]             1  \n",
      "699999  (18261648, 18261710]             1  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:137: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "         bin_offset_30  TT_S0  target_count starting_nt\n",
      "0   (3000018, 3000048]    4.0             4     3000018\n",
      "1   (3000048, 3000148]    4.0             9     3000048\n",
      "2   (3000148, 3000248]    0.0             3     3000148\n",
      "3   (3000248, 3000348]    2.0             3     3000248\n",
      "4   (3000348, 3000448]    0.0             5     3000348\n",
      "5   (3000448, 3000548]    0.0             1     3000448\n",
      "6   (3000548, 3000648]    0.0             2     3000548\n",
      "7   (3000648, 3000748]    0.0             5     3000648\n",
      "8   (3000748, 3000848]    0.0             6     3000748\n",
      "9   (3000848, 3000948]    0.0             5     3000848\n",
      "10  (3000948, 3001048]    2.0             3     3000948\n",
      "11  (3001048, 3001148]    0.0             4     3001048\n",
      "12  (3001148, 3001248]    0.0             2     3001148\n",
      "13  (3001248, 3001348]    0.0             5     3001248\n",
      "14  (3001348, 3001448]    0.0             2     3001348\n",
      "15  (3001448, 3001548]    1.0             8     3001448\n",
      "16  (3001548, 3001648]    5.0             8     3001548\n",
      "17  (3001648, 3001748]    1.0             7     3001648\n",
      "18  (3001748, 3001848]    0.0             5     3001748\n",
      "19  (3001848, 3001948]    0.0             2     3001848\n",
      "20  (3001948, 3002048]    1.0             2     3001948\n",
      "21  (3002048, 3002148]    0.0             3     3002048\n",
      "22  (3002148, 3002248]    4.0             5     3002148\n",
      "23  (3002248, 3002348]    7.0             4     3002248\n",
      "24  (3002348, 3002448]   14.0             4     3002348\n",
      "25  (3002448, 3002548]   11.0             4     3002448\n",
      "26  (3002548, 3002648]    0.0             1     3002548\n",
      "27  (3002648, 3002748]    0.0             2     3002648\n",
      "28  (3002748, 3002848]    0.0             3     3002748\n",
      "29  (3002848, 3002948]    0.0             0     3002848\n",
      "30  (3002948, 3003048]    1.0             8     3002948\n",
      "31  (3003048, 3003148]    2.0             2     3003048\n",
      "32  (3003148, 3003248]    0.0             0     3003148\n",
      "33  (3003248, 3003348]    1.0             5     3003248\n",
      "34  (3003348, 3003448]    0.0             4     3003348\n",
      "35  (3003448, 3003548]    0.0             4     3003448\n",
      "36  (3003548, 3003648]    1.0             1     3003548\n",
      "37  (3003648, 3003748]    3.0             5     3003648\n",
      "38  (3003748, 3003848]    1.0             5     3003748\n",
      "39  (3003848, 3003948]    1.0             5     3003848\n",
      "40  (3003948, 3004048]    7.0             4     3003948\n",
      "41  (3004048, 3004148]    0.0             0     3004048\n",
      "42  (3004148, 3004248]    1.0             4     3004148\n",
      "43  (3004248, 3004348]    4.0             3     3004248\n",
      "44  (3004348, 3004448]    0.0             1     3004348\n",
      "45  (3004448, 3004548]    0.0             3     3004448\n",
      "46  (3004548, 3004648]    1.0             5     3004548\n",
      "47  (3004648, 3004748]    0.0             0     3004648\n",
      "48  (3004748, 3004848]    2.0             3     3004748\n",
      "49  (3004848, 3004948]   11.0             6     3004848\n",
      "(152618, 4)\n",
      "bin_offset_30    category\n",
      "TT_S0             float64\n",
      "target_count        int64\n",
      "starting_nt      category\n",
      "dtype: object\n",
      "sumos ilgis: 152618\n",
      "ranges ilgis: 152618\n",
      "Model input - NO SCALE\n",
      "        TT_S0  genes  transcripts  exons  CDSs\n",
      "0         4.0      0            0      0     0\n",
      "1         4.0      0            0      0     0\n",
      "2         0.0      0            0      0     0\n",
      "3         2.0      0            0      0     0\n",
      "4         0.0      0            0      0     0\n",
      "...       ...    ...          ...    ...   ...\n",
      "152613   13.0      0            0      0     0\n",
      "152614    1.0      0            0      0     0\n",
      "152615    3.0      0            0      0     0\n",
      "152616    0.0      0            0      0     0\n",
      "152617    4.0      0            0      0     0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "        TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0         4.0             0                   0             0            0\n",
      "1         4.0             0                   0             0            0\n",
      "2         0.0             0                   0             0            0\n",
      "3         2.0             0                   0             0            0\n",
      "4         0.0             0                   0             0            0\n",
      "...       ...           ...                 ...           ...          ...\n",
      "152613   13.0             0                   0             0            0\n",
      "152614    1.0             0                   0             0            0\n",
      "152615    3.0             0                   0             0            0\n",
      "152616    0.0             0                   0             0            0\n",
      "152617    4.0             0                   0             0            0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "0.8746412611880643\n",
      "[1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 0 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 0 1 0 0 0 1 0 0 0 0 0 0 1 0 1 1 1 1 1 1 1 0 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 0 0 1 1 1 1 1 1 0 0 0 0 0 1 1 0 1 0 1 1\n",
      " 1 0 1 1 1 1 1 1 1 1 1 1 1 1 0 1 0 0 0 0 1 0 1 1 1 1 0 1 1 1 1 1 1 1 1 1 0\n",
      " 0 1 1 1 1 1 1 0 1 1 1 0 0 1 0 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 0 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 1 1 1 1 0 0 0 0 1 0 1 1 1 1 1 0 1\n",
      " 1 1 1 1 1 1 0 0 0 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 1 0 1 1 1 0 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 0 1 1 1 1 1 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 0 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1]\n",
      "threshold density 1.30546\n",
      "cluster signal 194194.0\n",
      "cluster target count 551544\n",
      "!!cluster density 0.35209158290181747\n",
      "cluster signal 262717.0\n",
      "cluster target count 148456\n",
      "!!cluster density 1.7696623915503584\n",
      "\n",
      "Labels mapping\n",
      " {1: 0, 0: 1}\n",
      "DEBUG: 133486 19132 \n",
      "\n",
      " +----------------------+-------------+----------------+-------+\n",
      "| bin_offset_30        | TT_S0       | target_count   | +9    |\n",
      "| (category)           | (float64)   | (int64)        | ...   |\n",
      "|----------------------+-------------+----------------+-------|\n",
      "| (3000018, 3000048]   | 4.0         | 4              | ...   |\n",
      "| (3000048, 3000148]   | 4.0         | 9              | ...   |\n",
      "| (3000148, 3000248]   | 0.0         | 3              | ...   |\n",
      "| (3000248, 3000348]   | 2.0         | 3              | ...   |\n",
      "| ...                  | ...         | ...            | ...   |\n",
      "| (18261348, 18261448] | 1.0         | 10             | ...   |\n",
      "| (18261448, 18261548] | 3.0         | 4              | ...   |\n",
      "| (18261548, 18261648] | 0.0         | 2              | ...   |\n",
      "| (18261648, 18261710] | 4.0         | 3              | ...   |\n",
      "+----------------------+-------------+----------------+-------+\n",
      "Unstranded PyRanges object has 133,486 rows and 12 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "9 hidden columns: starting_nt, Chromosome, Start, End, genes, transcripts, ... (+ 3 more.)\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0\n",
      "BINS  IntervalIndex([  (3000018, 3000053],   (3000053, 3000153],\n",
      "                 (3000153, 3000253],   (3000253, 3000353],\n",
      "                 (3000353, 3000453],   (3000453, 3000553],\n",
      "                 (3000553, 3000653],   (3000653, 3000753],\n",
      "                 (3000753, 3000853],   (3000853, 3000953],\n",
      "               ...\n",
      "               (18260753, 18260853], (18260853, 18260953],\n",
      "               (18260953, 18261053], (18261053, 18261153],\n",
      "               (18261153, 18261253], (18261253, 18261353],\n",
      "               (18261353, 18261453], (18261453, 18261553],\n",
      "               (18261553, 18261653], (18261653, 18261710]],\n",
      "              dtype='interval[int64, right]', length=152618)\n",
      "HEAD with BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0   \n",
      "\n",
      "        bin_offset_35  target_count  \n",
      "0  (3000018, 3000053]             1  \n",
      "1  (3000018, 3000053]             1  \n",
      "2  (3000018, 3000053]             1  \n",
      "3  (3000018, 3000053]             1  \n",
      "4  (3000053, 3000153]             1  \n",
      "TAIL with BINS\n",
      "       seqnames     start       end        CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  18261611  18261611  18_18261611    0.0    0.0    0.0   \n",
      "699996       18  18261615  18261615  18_18261615    0.0    0.0    0.0   \n",
      "699997       18  18261682  18261682  18_18261682    0.0    0.0    0.0   \n",
      "699998       18  18261702  18261702  18_18261702    4.0    0.0    0.0   \n",
      "699999       18  18261710  18261710  18_18261710    0.0    0.0    0.0   \n",
      "\n",
      "               bin_offset_35  target_count  \n",
      "699995  (18261553, 18261653]             1  \n",
      "699996  (18261553, 18261653]             1  \n",
      "699997  (18261653, 18261710]             1  \n",
      "699998  (18261653, 18261710]             1  \n",
      "699999  (18261653, 18261710]             1  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:137: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "         bin_offset_35  TT_S0  target_count starting_nt\n",
      "0   (3000018, 3000053]    4.0             4     3000018\n",
      "1   (3000053, 3000153]    4.0             9     3000053\n",
      "2   (3000153, 3000253]    0.0             3     3000153\n",
      "3   (3000253, 3000353]    2.0             4     3000253\n",
      "4   (3000353, 3000453]    0.0             4     3000353\n",
      "5   (3000453, 3000553]    0.0             1     3000453\n",
      "6   (3000553, 3000653]    0.0             2     3000553\n",
      "7   (3000653, 3000753]    0.0             6     3000653\n",
      "8   (3000753, 3000853]    0.0             5     3000753\n",
      "9   (3000853, 3000953]    0.0             5     3000853\n",
      "10  (3000953, 3001053]    2.0             3     3000953\n",
      "11  (3001053, 3001153]    0.0             4     3001053\n",
      "12  (3001153, 3001253]    0.0             2     3001153\n",
      "13  (3001253, 3001353]    0.0             5     3001253\n",
      "14  (3001353, 3001453]    0.0             3     3001353\n",
      "15  (3001453, 3001553]    1.0             8     3001453\n",
      "16  (3001553, 3001653]    5.0             7     3001553\n",
      "17  (3001653, 3001753]    1.0             7     3001653\n",
      "18  (3001753, 3001853]    0.0             5     3001753\n",
      "19  (3001853, 3001953]    0.0             2     3001853\n",
      "20  (3001953, 3002053]    1.0             2     3001953\n",
      "21  (3002053, 3002153]    0.0             3     3002053\n",
      "22  (3002153, 3002253]    4.0             6     3002153\n",
      "23  (3002253, 3002353]    7.0             3     3002253\n",
      "24  (3002353, 3002453]   14.0             4     3002353\n",
      "25  (3002453, 3002553]   11.0             4     3002453\n",
      "26  (3002553, 3002653]    0.0             1     3002553\n",
      "27  (3002653, 3002753]    0.0             2     3002653\n",
      "28  (3002753, 3002853]    0.0             3     3002753\n",
      "29  (3002853, 3002953]    0.0             1     3002853\n",
      "30  (3002953, 3003053]    1.0             7     3002953\n",
      "31  (3003053, 3003153]    2.0             2     3003053\n",
      "32  (3003153, 3003253]    0.0             0     3003153\n",
      "33  (3003253, 3003353]    1.0             5     3003253\n",
      "34  (3003353, 3003453]    0.0             4     3003353\n",
      "35  (3003453, 3003553]    0.0             4     3003453\n",
      "36  (3003553, 3003653]    1.0             1     3003553\n",
      "37  (3003653, 3003753]    3.0             5     3003653\n",
      "38  (3003753, 3003853]    1.0             5     3003753\n",
      "39  (3003853, 3003953]    1.0             6     3003853\n",
      "40  (3003953, 3004053]    7.0             3     3003953\n",
      "41  (3004053, 3004153]    0.0             0     3004053\n",
      "42  (3004153, 3004253]    1.0             4     3004153\n",
      "43  (3004253, 3004353]    4.0             3     3004253\n",
      "44  (3004353, 3004453]    0.0             1     3004353\n",
      "45  (3004453, 3004553]    0.0             3     3004453\n",
      "46  (3004553, 3004653]    1.0             5     3004553\n",
      "47  (3004653, 3004753]    0.0             0     3004653\n",
      "48  (3004753, 3004853]   10.0             4     3004753\n",
      "49  (3004853, 3004953]    3.0             5     3004853\n",
      "(152618, 4)\n",
      "bin_offset_35    category\n",
      "TT_S0             float64\n",
      "target_count        int64\n",
      "starting_nt      category\n",
      "dtype: object\n",
      "sumos ilgis: 152618\n",
      "ranges ilgis: 152618\n",
      "Model input - NO SCALE\n",
      "        TT_S0  genes  transcripts  exons  CDSs\n",
      "0         4.0      0            0      0     0\n",
      "1         4.0      0            0      0     0\n",
      "2         0.0      0            0      0     0\n",
      "3         2.0      0            0      0     0\n",
      "4         0.0      0            0      0     0\n",
      "...       ...    ...          ...    ...   ...\n",
      "152613   13.0      0            0      0     0\n",
      "152614    1.0      0            0      0     0\n",
      "152615    3.0      0            0      0     0\n",
      "152616    0.0      0            0      0     0\n",
      "152617    4.0      0            0      0     0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "        TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0         4.0             0                   0             0            0\n",
      "1         4.0             0                   0             0            0\n",
      "2         0.0             0                   0             0            0\n",
      "3         2.0             0                   0             0            0\n",
      "4         0.0             0                   0             0            0\n",
      "...       ...           ...                 ...           ...          ...\n",
      "152613   13.0             0                   0             0            0\n",
      "152614    1.0             0                   0             0            0\n",
      "152615    3.0             0                   0             0            0\n",
      "152616    0.0             0                   0             0            0\n",
      "152617    4.0             0                   0             0            0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "0.8482027021714346\n",
      "[1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 0 0 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 0 1 1 1 1 1 1 1 0 1 1 0 0 0 1 0 0 0 0 0 0 0 0 1 1 1 1 0 1 1 0 1 1 1\n",
      " 1 1 1 1 1 1 1 0 1 1 1 1 1 0 1 1 1 0 0 1 1 1 1 1 1 0 0 0 0 0 1 1 0 1 0 1 0\n",
      " 1 0 1 1 1 1 1 1 1 1 1 1 1 1 0 1 0 0 0 0 1 0 1 1 1 1 0 1 1 1 1 1 1 1 1 1 0\n",
      " 0 0 1 1 1 1 1 0 1 1 1 0 0 1 0 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 0 0 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 1 1 1 1 0 0 0 0 1 0 0 1 1 1 1 0 1\n",
      " 1 1 1 1 1 1 0 0 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 1 0 1 1 1 0 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 0 0 1 1 1 1 1 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 0 1 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 0 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 1 1 1\n",
      " 1]\n",
      "threshold density 1.30546\n",
      "cluster signal 165964.0\n",
      "cluster target count 526262\n",
      "!!cluster density 0.31536383018344477\n",
      "cluster signal 290947.0\n",
      "cluster target count 173738\n",
      "!!cluster density 1.674630765865844\n",
      "\n",
      "Labels mapping\n",
      " {1: 0, 0: 1}\n",
      "DEBUG: 129451 23167 \n",
      "\n",
      " +----------------------+-------------+----------------+-------+\n",
      "| bin_offset_35        | TT_S0       | target_count   | +9    |\n",
      "| (category)           | (float64)   | (int64)        | ...   |\n",
      "|----------------------+-------------+----------------+-------|\n",
      "| (3000018, 3000053]   | 4.0         | 4              | ...   |\n",
      "| (3000053, 3000153]   | 4.0         | 9              | ...   |\n",
      "| (3000153, 3000253]   | 0.0         | 3              | ...   |\n",
      "| (3000253, 3000353]   | 2.0         | 4              | ...   |\n",
      "| ...                  | ...         | ...            | ...   |\n",
      "| (18261353, 18261453] | 1.0         | 9              | ...   |\n",
      "| (18261453, 18261553] | 3.0         | 4              | ...   |\n",
      "| (18261553, 18261653] | 0.0         | 2              | ...   |\n",
      "| (18261653, 18261710] | 4.0         | 3              | ...   |\n",
      "+----------------------+-------------+----------------+-------+\n",
      "Unstranded PyRanges object has 129,451 rows and 12 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "9 hidden columns: starting_nt, Chromosome, Start, End, genes, transcripts, ... (+ 3 more.)\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0\n",
      "BINS  IntervalIndex([  (3000018, 3000058],   (3000058, 3000158],\n",
      "                 (3000158, 3000258],   (3000258, 3000358],\n",
      "                 (3000358, 3000458],   (3000458, 3000558],\n",
      "                 (3000558, 3000658],   (3000658, 3000758],\n",
      "                 (3000758, 3000858],   (3000858, 3000958],\n",
      "               ...\n",
      "               (18260758, 18260858], (18260858, 18260958],\n",
      "               (18260958, 18261058], (18261058, 18261158],\n",
      "               (18261158, 18261258], (18261258, 18261358],\n",
      "               (18261358, 18261458], (18261458, 18261558],\n",
      "               (18261558, 18261658], (18261658, 18261710]],\n",
      "              dtype='interval[int64, right]', length=152618)\n",
      "HEAD with BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0   \n",
      "\n",
      "        bin_offset_40  target_count  \n",
      "0  (3000018, 3000058]             1  \n",
      "1  (3000018, 3000058]             1  \n",
      "2  (3000018, 3000058]             1  \n",
      "3  (3000018, 3000058]             1  \n",
      "4  (3000058, 3000158]             1  \n",
      "TAIL with BINS\n",
      "       seqnames     start       end        CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  18261611  18261611  18_18261611    0.0    0.0    0.0   \n",
      "699996       18  18261615  18261615  18_18261615    0.0    0.0    0.0   \n",
      "699997       18  18261682  18261682  18_18261682    0.0    0.0    0.0   \n",
      "699998       18  18261702  18261702  18_18261702    4.0    0.0    0.0   \n",
      "699999       18  18261710  18261710  18_18261710    0.0    0.0    0.0   \n",
      "\n",
      "               bin_offset_40  target_count  \n",
      "699995  (18261558, 18261658]             1  \n",
      "699996  (18261558, 18261658]             1  \n",
      "699997  (18261658, 18261710]             1  \n",
      "699998  (18261658, 18261710]             1  \n",
      "699999  (18261658, 18261710]             1  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:137: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "         bin_offset_40  TT_S0  target_count starting_nt\n",
      "0   (3000018, 3000058]    4.0             4     3000018\n",
      "1   (3000058, 3000158]    4.0             9     3000058\n",
      "2   (3000158, 3000258]    0.0             3     3000158\n",
      "3   (3000258, 3000358]    2.0             5     3000258\n",
      "4   (3000358, 3000458]    0.0             3     3000358\n",
      "5   (3000458, 3000558]    0.0             1     3000458\n",
      "6   (3000558, 3000658]    0.0             2     3000558\n",
      "7   (3000658, 3000758]    0.0             6     3000658\n",
      "8   (3000758, 3000858]    0.0             5     3000758\n",
      "9   (3000858, 3000958]    0.0             5     3000858\n",
      "10  (3000958, 3001058]    2.0             5     3000958\n",
      "11  (3001058, 3001158]    0.0             2     3001058\n",
      "12  (3001158, 3001258]    0.0             2     3001158\n",
      "13  (3001258, 3001358]    0.0             5     3001258\n",
      "14  (3001358, 3001458]    0.0             4     3001358\n",
      "15  (3001458, 3001558]    1.0             7     3001458\n",
      "16  (3001558, 3001658]    5.0             7     3001558\n",
      "17  (3001658, 3001758]    1.0             7     3001658\n",
      "18  (3001758, 3001858]    0.0             5     3001758\n",
      "19  (3001858, 3001958]    0.0             2     3001858\n",
      "20  (3001958, 3002058]    1.0             2     3001958\n",
      "21  (3002058, 3002158]    0.0             3     3002058\n",
      "22  (3002158, 3002258]    4.0             6     3002158\n",
      "23  (3002258, 3002358]    9.0             4     3002258\n",
      "24  (3002358, 3002458]   19.0             4     3002358\n",
      "25  (3002458, 3002558]    4.0             3     3002458\n",
      "26  (3002558, 3002658]    0.0             1     3002558\n",
      "27  (3002658, 3002758]    0.0             3     3002658\n",
      "28  (3002758, 3002858]    0.0             2     3002758\n",
      "29  (3002858, 3002958]    0.0             1     3002858\n",
      "30  (3002958, 3003058]    3.0             8     3002958\n",
      "31  (3003058, 3003158]    0.0             1     3003058\n",
      "32  (3003158, 3003258]    0.0             0     3003158\n",
      "33  (3003258, 3003358]    1.0             6     3003258\n",
      "34  (3003358, 3003458]    0.0             3     3003358\n",
      "35  (3003458, 3003558]    0.0             4     3003458\n",
      "36  (3003558, 3003658]    1.0             1     3003558\n",
      "37  (3003658, 3003758]    4.0             6     3003658\n",
      "38  (3003758, 3003858]    0.0             4     3003758\n",
      "39  (3003858, 3003958]    1.0             6     3003858\n",
      "40  (3003958, 3004058]    7.0             3     3003958\n",
      "41  (3004058, 3004158]    0.0             0     3004058\n",
      "42  (3004158, 3004258]    1.0             4     3004158\n",
      "43  (3004258, 3004358]    4.0             3     3004258\n",
      "44  (3004358, 3004458]    0.0             1     3004358\n",
      "45  (3004458, 3004558]    0.0             5     3004458\n",
      "46  (3004558, 3004658]    1.0             3     3004558\n",
      "47  (3004658, 3004758]    0.0             0     3004658\n",
      "48  (3004758, 3004858]   10.0             4     3004758\n",
      "49  (3004858, 3004958]    3.0             5     3004858\n",
      "(152618, 4)\n",
      "bin_offset_40    category\n",
      "TT_S0             float64\n",
      "target_count        int64\n",
      "starting_nt      category\n",
      "dtype: object\n",
      "sumos ilgis: 152618\n",
      "ranges ilgis: 152618\n",
      "Model input - NO SCALE\n",
      "        TT_S0  genes  transcripts  exons  CDSs\n",
      "0         4.0      0            0      0     0\n",
      "1         4.0      0            0      0     0\n",
      "2         0.0      0            0      0     0\n",
      "3         2.0      0            0      0     0\n",
      "4         0.0      0            0      0     0\n",
      "...       ...    ...          ...    ...   ...\n",
      "152613   13.0      0            0      0     0\n",
      "152614    4.0      0            0      0     0\n",
      "152615    0.0      0            0      0     0\n",
      "152616    0.0      0            0      0     0\n",
      "152617    4.0      0            0      0     0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "        TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0         4.0             0                   0             0            0\n",
      "1         4.0             0                   0             0            0\n",
      "2         0.0             0                   0             0            0\n",
      "3         2.0             0                   0             0            0\n",
      "4         0.0             0                   0             0            0\n",
      "...       ...           ...                 ...           ...          ...\n",
      "152613   13.0             0                   0             0            0\n",
      "152614    4.0             0                   0             0            0\n",
      "152615    0.0             0                   0             0            0\n",
      "152616    0.0             0                   0             0            0\n",
      "152617    4.0             0                   0             0            0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "0.1520659424183255\n",
      "[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 1 0 0 0 0 0 0 0 1 0 0 1 0 1 0 1 1 1 1 1 0 1 1 0 0 0 0 1 0 0 1 0 0 0\n",
      " 0 0 0 0 0 0 0 1 0 0 0 0 0 1 0 0 0 1 1 0 0 0 0 0 0 1 1 1 1 1 0 0 1 1 0 0 1\n",
      " 0 1 0 0 0 0 0 0 0 0 0 0 0 0 1 0 1 1 1 1 0 1 0 0 0 0 1 0 0 0 0 0 0 0 0 0 1\n",
      " 0 1 0 0 0 0 0 1 0 0 0 1 1 0 1 0 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 1 1 1 1 0 1 1 0 0 0 0 1 0\n",
      " 0 0 0 0 0 0 1 1 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 1 0 0 0 1 0 1 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 1 1 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 1 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 1 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0\n",
      " 0]\n",
      "threshold density 1.30546\n",
      "cluster signal 165594.0\n",
      "cluster target count 525957\n",
      "!!cluster density 0.31484322862895636\n",
      "cluster signal 291317.0\n",
      "cluster target count 174043\n",
      "!!cluster density 1.6738219865205726\n",
      "\n",
      "Labels mapping\n",
      " {0: 0, 1: 1}\n",
      "DEBUG: 129410 23208 \n",
      "\n",
      " +----------------------+-------------+----------------+-------+\n",
      "| bin_offset_40        | TT_S0       | target_count   | +9    |\n",
      "| (category)           | (float64)   | (int64)        | ...   |\n",
      "|----------------------+-------------+----------------+-------|\n",
      "| (3000018, 3000058]   | 4.0         | 4              | ...   |\n",
      "| (3000058, 3000158]   | 4.0         | 9              | ...   |\n",
      "| (3000158, 3000258]   | 0.0         | 3              | ...   |\n",
      "| (3000258, 3000358]   | 2.0         | 5              | ...   |\n",
      "| ...                  | ...         | ...            | ...   |\n",
      "| (18261358, 18261458] | 4.0         | 10             | ...   |\n",
      "| (18261458, 18261558] | 0.0         | 2              | ...   |\n",
      "| (18261558, 18261658] | 0.0         | 2              | ...   |\n",
      "| (18261658, 18261710] | 4.0         | 3              | ...   |\n",
      "+----------------------+-------------+----------------+-------+\n",
      "Unstranded PyRanges object has 129,410 rows and 12 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "9 hidden columns: starting_nt, Chromosome, Start, End, genes, transcripts, ... (+ 3 more.)\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0\n",
      "BINS  IntervalIndex([  (3000018, 3000063],   (3000063, 3000163],\n",
      "                 (3000163, 3000263],   (3000263, 3000363],\n",
      "                 (3000363, 3000463],   (3000463, 3000563],\n",
      "                 (3000563, 3000663],   (3000663, 3000763],\n",
      "                 (3000763, 3000863],   (3000863, 3000963],\n",
      "               ...\n",
      "               (18260763, 18260863], (18260863, 18260963],\n",
      "               (18260963, 18261063], (18261063, 18261163],\n",
      "               (18261163, 18261263], (18261263, 18261363],\n",
      "               (18261363, 18261463], (18261463, 18261563],\n",
      "               (18261563, 18261663], (18261663, 18261710]],\n",
      "              dtype='interval[int64, right]', length=152618)\n",
      "HEAD with BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0   \n",
      "\n",
      "        bin_offset_45  target_count  \n",
      "0  (3000018, 3000063]             1  \n",
      "1  (3000018, 3000063]             1  \n",
      "2  (3000018, 3000063]             1  \n",
      "3  (3000018, 3000063]             1  \n",
      "4  (3000063, 3000163]             1  \n",
      "TAIL with BINS\n",
      "       seqnames     start       end        CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  18261611  18261611  18_18261611    0.0    0.0    0.0   \n",
      "699996       18  18261615  18261615  18_18261615    0.0    0.0    0.0   \n",
      "699997       18  18261682  18261682  18_18261682    0.0    0.0    0.0   \n",
      "699998       18  18261702  18261702  18_18261702    4.0    0.0    0.0   \n",
      "699999       18  18261710  18261710  18_18261710    0.0    0.0    0.0   \n",
      "\n",
      "               bin_offset_45  target_count  \n",
      "699995  (18261563, 18261663]             1  \n",
      "699996  (18261563, 18261663]             1  \n",
      "699997  (18261663, 18261710]             1  \n",
      "699998  (18261663, 18261710]             1  \n",
      "699999  (18261663, 18261710]             1  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:137: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "         bin_offset_45  TT_S0  target_count starting_nt\n",
      "0   (3000018, 3000063]    4.0             4     3000018\n",
      "1   (3000063, 3000163]    4.0            10     3000063\n",
      "2   (3000163, 3000263]    0.0             2     3000163\n",
      "3   (3000263, 3000363]    2.0             5     3000263\n",
      "4   (3000363, 3000463]    0.0             3     3000363\n",
      "5   (3000463, 3000563]    0.0             1     3000463\n",
      "6   (3000563, 3000663]    0.0             3     3000563\n",
      "7   (3000663, 3000763]    0.0             5     3000663\n",
      "8   (3000763, 3000863]    0.0             5     3000763\n",
      "9   (3000863, 3000963]    0.0             5     3000863\n",
      "10  (3000963, 3001063]    2.0             6     3000963\n",
      "11  (3001063, 3001163]    0.0             2     3001063\n",
      "12  (3001163, 3001263]    0.0             2     3001163\n",
      "13  (3001263, 3001363]    0.0             5     3001263\n",
      "14  (3001363, 3001463]    0.0             3     3001363\n",
      "15  (3001463, 3001563]    1.0             7     3001463\n",
      "16  (3001563, 3001663]    6.0             8     3001563\n",
      "17  (3001663, 3001763]    0.0             7     3001663\n",
      "18  (3001763, 3001863]    0.0             5     3001763\n",
      "19  (3001863, 3001963]    0.0             1     3001863\n",
      "20  (3001963, 3002063]    1.0             2     3001963\n",
      "21  (3002063, 3002163]    0.0             3     3002063\n",
      "22  (3002163, 3002263]    4.0             6     3002163\n",
      "23  (3002263, 3002363]    9.0             4     3002263\n",
      "24  (3002363, 3002463]   19.0             4     3002363\n",
      "25  (3002463, 3002563]    4.0             3     3002463\n",
      "26  (3002563, 3002663]    0.0             1     3002563\n",
      "27  (3002663, 3002763]    0.0             3     3002663\n",
      "28  (3002763, 3002863]    0.0             2     3002763\n",
      "29  (3002863, 3002963]    0.0             1     3002863\n",
      "30  (3002963, 3003063]    3.0             8     3002963\n",
      "31  (3003063, 3003163]    0.0             1     3003063\n",
      "32  (3003163, 3003263]    0.0             0     3003163\n",
      "33  (3003263, 3003363]    1.0             7     3003263\n",
      "34  (3003363, 3003463]    0.0             3     3003363\n",
      "35  (3003463, 3003563]    0.0             3     3003463\n",
      "36  (3003563, 3003663]    1.0             1     3003563\n",
      "37  (3003663, 3003763]    4.0             8     3003663\n",
      "38  (3003763, 3003863]    0.0             2     3003763\n",
      "39  (3003863, 3003963]    6.0             7     3003863\n",
      "40  (3003963, 3004063]    2.0             2     3003963\n",
      "41  (3004063, 3004163]    0.0             0     3004063\n",
      "42  (3004163, 3004263]    1.0             4     3004163\n",
      "43  (3004263, 3004363]    4.0             3     3004263\n",
      "44  (3004363, 3004463]    0.0             1     3004363\n",
      "45  (3004463, 3004563]    0.0             5     3004463\n",
      "46  (3004563, 3004663]    1.0             3     3004563\n",
      "47  (3004663, 3004763]    0.0             0     3004663\n",
      "48  (3004763, 3004863]   10.0             4     3004763\n",
      "49  (3004863, 3004963]    3.0             5     3004863\n",
      "(152618, 4)\n",
      "bin_offset_45    category\n",
      "TT_S0             float64\n",
      "target_count        int64\n",
      "starting_nt      category\n",
      "dtype: object\n",
      "sumos ilgis: 152618\n",
      "ranges ilgis: 152618\n",
      "Model input - NO SCALE\n",
      "        TT_S0  genes  transcripts  exons  CDSs\n",
      "0         4.0      0            0      0     0\n",
      "1         4.0      0            0      0     0\n",
      "2         0.0      0            0      0     0\n",
      "3         2.0      0            0      0     0\n",
      "4         0.0      0            0      0     0\n",
      "...       ...    ...          ...    ...   ...\n",
      "152613   13.0      0            0      0     0\n",
      "152614    4.0      0            0      0     0\n",
      "152615    0.0      0            0      0     0\n",
      "152616    0.0      0            0      0     0\n",
      "152617    4.0      0            0      0     0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "        TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0         4.0             0                   0             0            0\n",
      "1         4.0             0                   0             0            0\n",
      "2         0.0             0                   0             0            0\n",
      "3         2.0             0                   0             0            0\n",
      "4         0.0             0                   0             0            0\n",
      "...       ...           ...                 ...           ...          ...\n",
      "152613   13.0             0                   0             0            0\n",
      "152614    4.0             0                   0             0            0\n",
      "152615    0.0             0                   0             0            0\n",
      "152616    0.0             0                   0             0            0\n",
      "152617    4.0             0                   0             0            0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "0.15218388394553722\n",
      "[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 1 0 0 1 0 1 0 1 1 1 1 1 0 1 1 0 0 0 0 1 0 0 1 0 0 0\n",
      " 0 0 0 0 0 0 0 1 0 0 0 0 0 1 0 0 0 1 1 0 0 0 0 0 0 0 1 1 1 1 0 0 1 1 0 1 0\n",
      " 0 1 0 0 0 0 0 0 0 0 0 0 0 0 1 0 1 1 1 1 0 1 0 0 0 0 1 0 0 0 0 0 0 0 0 0 1\n",
      " 0 1 0 0 0 0 0 1 0 0 0 1 1 0 1 0 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 1 1 0 1 0 1 1 0 0 0 0 1 0\n",
      " 0 0 0 0 0 0 1 1 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 1 0 0 0 1 0 1 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 1 1 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 1 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 1 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0\n",
      " 0]\n",
      "threshold density 1.30546\n",
      "cluster signal 165593.0\n",
      "cluster target count 525792\n",
      "!!cluster density 0.3149401284157994\n",
      "cluster signal 291318.0\n",
      "cluster target count 174208\n",
      "!!cluster density 1.672242376928729\n",
      "\n",
      "Labels mapping\n",
      " {0: 0, 1: 1}\n",
      "DEBUG: 129392 23226 \n",
      "\n",
      " +----------------------+-------------+----------------+-------+\n",
      "| bin_offset_45        | TT_S0       | target_count   | +9    |\n",
      "| (category)           | (float64)   | (int64)        | ...   |\n",
      "|----------------------+-------------+----------------+-------|\n",
      "| (3000018, 3000063]   | 4.0         | 4              | ...   |\n",
      "| (3000063, 3000163]   | 4.0         | 10             | ...   |\n",
      "| (3000163, 3000263]   | 0.0         | 2              | ...   |\n",
      "| (3000263, 3000363]   | 2.0         | 5              | ...   |\n",
      "| ...                  | ...         | ...            | ...   |\n",
      "| (18261363, 18261463] | 4.0         | 9              | ...   |\n",
      "| (18261463, 18261563] | 0.0         | 2              | ...   |\n",
      "| (18261563, 18261663] | 0.0         | 2              | ...   |\n",
      "| (18261663, 18261710] | 4.0         | 3              | ...   |\n",
      "+----------------------+-------------+----------------+-------+\n",
      "Unstranded PyRanges object has 129,392 rows and 12 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "9 hidden columns: starting_nt, Chromosome, Start, End, genes, transcripts, ... (+ 3 more.)\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0\n",
      "BINS  IntervalIndex([  (3000018, 3000068],   (3000068, 3000168],\n",
      "                 (3000168, 3000268],   (3000268, 3000368],\n",
      "                 (3000368, 3000468],   (3000468, 3000568],\n",
      "                 (3000568, 3000668],   (3000668, 3000768],\n",
      "                 (3000768, 3000868],   (3000868, 3000968],\n",
      "               ...\n",
      "               (18260768, 18260868], (18260868, 18260968],\n",
      "               (18260968, 18261068], (18261068, 18261168],\n",
      "               (18261168, 18261268], (18261268, 18261368],\n",
      "               (18261368, 18261468], (18261468, 18261568],\n",
      "               (18261568, 18261668], (18261668, 18261710]],\n",
      "              dtype='interval[int64, right]', length=152618)\n",
      "HEAD with BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0   \n",
      "\n",
      "        bin_offset_50  target_count  \n",
      "0  (3000018, 3000068]             1  \n",
      "1  (3000018, 3000068]             1  \n",
      "2  (3000018, 3000068]             1  \n",
      "3  (3000018, 3000068]             1  \n",
      "4  (3000018, 3000068]             1  \n",
      "TAIL with BINS\n",
      "       seqnames     start       end        CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  18261611  18261611  18_18261611    0.0    0.0    0.0   \n",
      "699996       18  18261615  18261615  18_18261615    0.0    0.0    0.0   \n",
      "699997       18  18261682  18261682  18_18261682    0.0    0.0    0.0   \n",
      "699998       18  18261702  18261702  18_18261702    4.0    0.0    0.0   \n",
      "699999       18  18261710  18261710  18_18261710    0.0    0.0    0.0   \n",
      "\n",
      "               bin_offset_50  target_count  \n",
      "699995  (18261568, 18261668]             1  \n",
      "699996  (18261568, 18261668]             1  \n",
      "699997  (18261668, 18261710]             1  \n",
      "699998  (18261668, 18261710]             1  \n",
      "699999  (18261668, 18261710]             1  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:137: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "         bin_offset_50  TT_S0  target_count starting_nt\n",
      "0   (3000018, 3000068]    4.0             5     3000018\n",
      "1   (3000068, 3000168]    4.0             9     3000068\n",
      "2   (3000168, 3000268]    0.0             2     3000168\n",
      "3   (3000268, 3000368]    2.0             5     3000268\n",
      "4   (3000368, 3000468]    0.0             3     3000368\n",
      "5   (3000468, 3000568]    0.0             1     3000468\n",
      "6   (3000568, 3000668]    0.0             3     3000568\n",
      "7   (3000668, 3000768]    0.0             5     3000668\n",
      "8   (3000768, 3000868]    0.0             5     3000768\n",
      "9   (3000868, 3000968]    0.0             5     3000868\n",
      "10  (3000968, 3001068]    2.0             6     3000968\n",
      "11  (3001068, 3001168]    0.0             2     3001068\n",
      "12  (3001168, 3001268]    0.0             2     3001168\n",
      "13  (3001268, 3001368]    0.0             5     3001268\n",
      "14  (3001368, 3001468]    0.0             3     3001368\n",
      "15  (3001468, 3001568]    4.0             9     3001468\n",
      "16  (3001568, 3001668]    3.0             6     3001568\n",
      "17  (3001668, 3001768]    0.0             7     3001668\n",
      "18  (3001768, 3001868]    0.0             5     3001768\n",
      "19  (3001868, 3001968]    0.0             1     3001868\n",
      "20  (3001968, 3002068]    1.0             2     3001968\n",
      "21  (3002068, 3002168]    0.0             3     3002068\n",
      "22  (3002168, 3002268]    4.0             6     3002168\n",
      "23  (3002268, 3002368]    9.0             4     3002268\n",
      "24  (3002368, 3002468]   19.0             4     3002368\n",
      "25  (3002468, 3002568]    4.0             3     3002468\n",
      "26  (3002568, 3002668]    0.0             1     3002568\n",
      "27  (3002668, 3002768]    0.0             3     3002668\n",
      "28  (3002768, 3002868]    0.0             2     3002768\n",
      "29  (3002868, 3002968]    0.0             1     3002868\n",
      "30  (3002968, 3003068]    3.0             9     3002968\n",
      "31  (3003068, 3003168]    0.0             0     3003068\n",
      "32  (3003168, 3003268]    0.0             0     3003168\n",
      "33  (3003268, 3003368]    1.0             7     3003268\n",
      "34  (3003368, 3003468]    0.0             3     3003368\n",
      "35  (3003468, 3003568]    0.0             3     3003468\n",
      "36  (3003568, 3003668]    1.0             1     3003568\n",
      "37  (3003668, 3003768]    4.0             8     3003668\n",
      "38  (3003768, 3003868]    0.0             2     3003768\n",
      "39  (3003868, 3003968]    6.0             7     3003868\n",
      "40  (3003968, 3004068]    2.0             2     3003968\n",
      "41  (3004068, 3004168]    0.0             0     3004068\n",
      "42  (3004168, 3004268]    3.0             5     3004168\n",
      "43  (3004268, 3004368]    2.0             3     3004268\n",
      "44  (3004368, 3004468]    0.0             0     3004368\n",
      "45  (3004468, 3004568]    0.0             6     3004468\n",
      "46  (3004568, 3004668]    1.0             2     3004568\n",
      "47  (3004668, 3004768]    0.0             0     3004668\n",
      "48  (3004768, 3004868]   10.0             5     3004768\n",
      "49  (3004868, 3004968]    3.0             4     3004868\n",
      "(152618, 4)\n",
      "bin_offset_50    category\n",
      "TT_S0             float64\n",
      "target_count        int64\n",
      "starting_nt      category\n",
      "dtype: object\n",
      "sumos ilgis: 152618\n",
      "ranges ilgis: 152618\n",
      "Model input - NO SCALE\n",
      "        TT_S0  genes  transcripts  exons  CDSs\n",
      "0         4.0      0            0      0     0\n",
      "1         4.0      0            0      0     0\n",
      "2         0.0      0            0      0     0\n",
      "3         2.0      0            0      0     0\n",
      "4         0.0      0            0      0     0\n",
      "...       ...    ...          ...    ...   ...\n",
      "152613   13.0      0            0      0     0\n",
      "152614    4.0      0            0      0     0\n",
      "152615    0.0      0            0      0     0\n",
      "152616    0.0      0            0      0     0\n",
      "152617    4.0      0            0      0     0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "        TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0         4.0             0                   0             0            0\n",
      "1         4.0             0                   0             0            0\n",
      "2         0.0             0                   0             0            0\n",
      "3         2.0             0                   0             0            0\n",
      "4         0.0             0                   0             0            0\n",
      "...       ...           ...                 ...           ...          ...\n",
      "152613   13.0             0                   0             0            0\n",
      "152614    4.0             0                   0             0            0\n",
      "152615    0.0             0                   0             0            0\n",
      "152616    0.0             0                   0             0            0\n",
      "152617    4.0             0                   0             0            0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "0.847468843779895\n",
      "[1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 0 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 0 1 1 0 0 0 1 0 0 0 0 0 1 0 0 1 1 1 1 0 1 1 0 1 1 1\n",
      " 1 1 1 1 1 1 1 0 1 1 1 1 1 0 1 1 1 0 0 1 1 1 1 1 1 1 0 0 0 0 1 1 0 0 1 0 1\n",
      " 1 0 1 1 1 1 1 1 1 1 1 1 1 1 0 1 0 0 0 0 1 0 1 1 1 1 0 1 1 1 1 1 1 1 1 1 0\n",
      " 1 0 1 1 1 1 1 0 1 1 1 0 1 1 0 1 0 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 0 0 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 1 1 1 1 0 0 1 0 1 0 0 1 1 1 1 0 1\n",
      " 1 1 1 1 1 1 0 0 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 1 0 1 1 1 0 1 0 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 0 0 1 1 1 1 1 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 0 1 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 0 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 1 1 1\n",
      " 1]\n",
      "threshold density 1.30546\n",
      "cluster signal 165267.0\n",
      "cluster target count 525382\n",
      "!!cluster density 0.31456540193611504\n",
      "cluster signal 291644.0\n",
      "cluster target count 174618\n",
      "!!cluster density 1.670182913559885\n",
      "\n",
      "Labels mapping\n",
      " {1: 0, 0: 1}\n",
      "DEBUG: 129339 23279 \n",
      "\n",
      " +----------------------+-------------+----------------+-------+\n",
      "| bin_offset_50        | TT_S0       | target_count   | +9    |\n",
      "| (category)           | (float64)   | (int64)        | ...   |\n",
      "|----------------------+-------------+----------------+-------|\n",
      "| (3000018, 3000068]   | 4.0         | 5              | ...   |\n",
      "| (3000068, 3000168]   | 4.0         | 9              | ...   |\n",
      "| (3000168, 3000268]   | 0.0         | 2              | ...   |\n",
      "| (3000268, 3000368]   | 2.0         | 5              | ...   |\n",
      "| ...                  | ...         | ...            | ...   |\n",
      "| (18261368, 18261468] | 4.0         | 8              | ...   |\n",
      "| (18261468, 18261568] | 0.0         | 2              | ...   |\n",
      "| (18261568, 18261668] | 0.0         | 2              | ...   |\n",
      "| (18261668, 18261710] | 4.0         | 3              | ...   |\n",
      "+----------------------+-------------+----------------+-------+\n",
      "Unstranded PyRanges object has 129,339 rows and 12 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "9 hidden columns: starting_nt, Chromosome, Start, End, genes, transcripts, ... (+ 3 more.)\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0\n",
      "BINS  IntervalIndex([  (3000018, 3000073],   (3000073, 3000173],\n",
      "                 (3000173, 3000273],   (3000273, 3000373],\n",
      "                 (3000373, 3000473],   (3000473, 3000573],\n",
      "                 (3000573, 3000673],   (3000673, 3000773],\n",
      "                 (3000773, 3000873],   (3000873, 3000973],\n",
      "               ...\n",
      "               (18260773, 18260873], (18260873, 18260973],\n",
      "               (18260973, 18261073], (18261073, 18261173],\n",
      "               (18261173, 18261273], (18261273, 18261373],\n",
      "               (18261373, 18261473], (18261473, 18261573],\n",
      "               (18261573, 18261673], (18261673, 18261710]],\n",
      "              dtype='interval[int64, right]', length=152618)\n",
      "HEAD with BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0   \n",
      "\n",
      "        bin_offset_55  target_count  \n",
      "0  (3000018, 3000073]             1  \n",
      "1  (3000018, 3000073]             1  \n",
      "2  (3000018, 3000073]             1  \n",
      "3  (3000018, 3000073]             1  \n",
      "4  (3000018, 3000073]             1  \n",
      "TAIL with BINS\n",
      "       seqnames     start       end        CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  18261611  18261611  18_18261611    0.0    0.0    0.0   \n",
      "699996       18  18261615  18261615  18_18261615    0.0    0.0    0.0   \n",
      "699997       18  18261682  18261682  18_18261682    0.0    0.0    0.0   \n",
      "699998       18  18261702  18261702  18_18261702    4.0    0.0    0.0   \n",
      "699999       18  18261710  18261710  18_18261710    0.0    0.0    0.0   \n",
      "\n",
      "               bin_offset_55  target_count  \n",
      "699995  (18261573, 18261673]             1  \n",
      "699996  (18261573, 18261673]             1  \n",
      "699997  (18261673, 18261710]             1  \n",
      "699998  (18261673, 18261710]             1  \n",
      "699999  (18261673, 18261710]             1  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:137: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "         bin_offset_55  TT_S0  target_count starting_nt\n",
      "0   (3000018, 3000073]    4.0             5     3000018\n",
      "1   (3000073, 3000173]    4.0            10     3000073\n",
      "2   (3000173, 3000273]    0.0             1     3000173\n",
      "3   (3000273, 3000373]    2.0             5     3000273\n",
      "4   (3000373, 3000473]    0.0             3     3000373\n",
      "5   (3000473, 3000573]    0.0             1     3000473\n",
      "6   (3000573, 3000673]    0.0             4     3000573\n",
      "7   (3000673, 3000773]    0.0             4     3000673\n",
      "8   (3000773, 3000873]    0.0             6     3000773\n",
      "9   (3000873, 3000973]    0.0             4     3000873\n",
      "10  (3000973, 3001073]    2.0             6     3000973\n",
      "11  (3001073, 3001173]    0.0             2     3001073\n",
      "12  (3001173, 3001273]    0.0             2     3001173\n",
      "13  (3001273, 3001373]    0.0             5     3001273\n",
      "14  (3001373, 3001473]    0.0             3     3001373\n",
      "15  (3001473, 3001573]    4.0             9     3001473\n",
      "16  (3001573, 3001673]    3.0             6     3001573\n",
      "17  (3001673, 3001773]    0.0             7     3001673\n",
      "18  (3001773, 3001873]    0.0             5     3001773\n",
      "19  (3001873, 3001973]    0.0             1     3001873\n",
      "20  (3001973, 3002073]    1.0             2     3001973\n",
      "21  (3002073, 3002173]    0.0             3     3002073\n",
      "22  (3002173, 3002273]    4.0             7     3002173\n",
      "23  (3002273, 3002373]    9.0             3     3002273\n",
      "24  (3002373, 3002473]   19.0             4     3002373\n",
      "25  (3002473, 3002573]    4.0             3     3002473\n",
      "26  (3002573, 3002673]    0.0             1     3002573\n",
      "27  (3002673, 3002773]    0.0             3     3002673\n",
      "28  (3002773, 3002873]    0.0             2     3002773\n",
      "29  (3002873, 3002973]    1.0             2     3002873\n",
      "30  (3002973, 3003073]    2.0             8     3002973\n",
      "31  (3003073, 3003173]    0.0             0     3003073\n",
      "32  (3003173, 3003273]    0.0             0     3003173\n",
      "33  (3003273, 3003373]    1.0             8     3003273\n",
      "34  (3003373, 3003473]    0.0             2     3003373\n",
      "35  (3003473, 3003573]    0.0             3     3003473\n",
      "36  (3003573, 3003673]    1.0             2     3003573\n",
      "37  (3003673, 3003773]    4.0             8     3003673\n",
      "38  (3003773, 3003873]    0.0             1     3003773\n",
      "39  (3003873, 3003973]    7.0             8     3003873\n",
      "40  (3003973, 3004073]    1.0             1     3003973\n",
      "41  (3004073, 3004173]    0.0             0     3004073\n",
      "42  (3004173, 3004273]    3.0             5     3004173\n",
      "43  (3004273, 3004373]    2.0             3     3004273\n",
      "44  (3004373, 3004473]    0.0             0     3004373\n",
      "45  (3004473, 3004573]    0.0             6     3004473\n",
      "46  (3004573, 3004673]    1.0             2     3004573\n",
      "47  (3004673, 3004773]    0.0             0     3004673\n",
      "48  (3004773, 3004873]   10.0             6     3004773\n",
      "49  (3004873, 3004973]    3.0             3     3004873\n",
      "(152618, 4)\n",
      "bin_offset_55    category\n",
      "TT_S0             float64\n",
      "target_count        int64\n",
      "starting_nt      category\n",
      "dtype: object\n",
      "sumos ilgis: 152618\n",
      "ranges ilgis: 152618\n",
      "Model input - NO SCALE\n",
      "        TT_S0  genes  transcripts  exons  CDSs\n",
      "0         4.0      0            0      0     0\n",
      "1         4.0      0            0      0     0\n",
      "2         0.0      0            0      0     0\n",
      "3         2.0      0            0      0     0\n",
      "4         0.0      0            0      0     0\n",
      "...       ...    ...          ...    ...   ...\n",
      "152613   13.0      0            0      0     0\n",
      "152614    4.0      0            0      0     0\n",
      "152615    0.0      0            0      0     0\n",
      "152616    0.0      0            0      0     0\n",
      "152617    4.0      0            0      0     0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "        TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0         4.0             0                   0             0            0\n",
      "1         4.0             0                   0             0            0\n",
      "2         0.0             0                   0             0            0\n",
      "3         2.0             0                   0             0            0\n",
      "4         0.0             0                   0             0            0\n",
      "...       ...           ...                 ...           ...          ...\n",
      "152613   13.0             0                   0             0            0\n",
      "152614    4.0             0                   0             0            0\n",
      "152615    0.0             0                   0             0            0\n",
      "152616    0.0             0                   0             0            0\n",
      "152617    4.0             0                   0             0            0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "0.1526032315978456\n",
      "[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 1 0 0 0 0 0 0 0 0 1 0 0 1 1 1 0 1 1 1 1 1 0 1 1 0 0 0 0 1 0 0 1 0 0 0\n",
      " 0 0 0 0 0 0 1 0 0 0 0 0 0 1 0 0 0 1 1 0 0 0 0 0 0 0 1 1 1 1 0 0 1 1 0 1 0\n",
      " 1 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 1 1 1 1 0 1 0 0 0 0 1 0 0 0 0 0 0 0 0 0 1\n",
      " 0 1 0 0 0 0 0 1 0 0 0 1 0 0 1 0 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 1 1 0 1 0 1 1 0 0 0 0 1 0\n",
      " 0 0 0 0 0 0 1 1 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 1 0 0 0 1 0 1 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 1 1 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 1 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0]\n",
      "threshold density 1.30546\n",
      "cluster signal 165139.0\n",
      "cluster target count 525308\n",
      "!!cluster density 0.31436604810891894\n",
      "cluster signal 291772.0\n",
      "cluster target count 174692\n",
      "!!cluster density 1.6702081377510132\n",
      "\n",
      "Labels mapping\n",
      " {0: 0, 1: 1}\n",
      "DEBUG: 129328 23290 \n",
      "\n",
      " +----------------------+-------------+----------------+-------+\n",
      "| bin_offset_55        | TT_S0       | target_count   | +9    |\n",
      "| (category)           | (float64)   | (int64)        | ...   |\n",
      "|----------------------+-------------+----------------+-------|\n",
      "| (3000018, 3000073]   | 4.0         | 5              | ...   |\n",
      "| (3000073, 3000173]   | 4.0         | 10             | ...   |\n",
      "| (3000173, 3000273]   | 0.0         | 1              | ...   |\n",
      "| (3000273, 3000373]   | 2.0         | 5              | ...   |\n",
      "| ...                  | ...         | ...            | ...   |\n",
      "| (18261373, 18261473] | 4.0         | 7              | ...   |\n",
      "| (18261473, 18261573] | 0.0         | 2              | ...   |\n",
      "| (18261573, 18261673] | 0.0         | 2              | ...   |\n",
      "| (18261673, 18261710] | 4.0         | 3              | ...   |\n",
      "+----------------------+-------------+----------------+-------+\n",
      "Unstranded PyRanges object has 129,328 rows and 12 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "9 hidden columns: starting_nt, Chromosome, Start, End, genes, transcripts, ... (+ 3 more.)\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0\n",
      "BINS  IntervalIndex([  (3000018, 3000078],   (3000078, 3000178],\n",
      "                 (3000178, 3000278],   (3000278, 3000378],\n",
      "                 (3000378, 3000478],   (3000478, 3000578],\n",
      "                 (3000578, 3000678],   (3000678, 3000778],\n",
      "                 (3000778, 3000878],   (3000878, 3000978],\n",
      "               ...\n",
      "               (18260778, 18260878], (18260878, 18260978],\n",
      "               (18260978, 18261078], (18261078, 18261178],\n",
      "               (18261178, 18261278], (18261278, 18261378],\n",
      "               (18261378, 18261478], (18261478, 18261578],\n",
      "               (18261578, 18261678], (18261678, 18261710]],\n",
      "              dtype='interval[int64, right]', length=152618)\n",
      "HEAD with BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0   \n",
      "\n",
      "        bin_offset_60  target_count  \n",
      "0  (3000018, 3000078]             1  \n",
      "1  (3000018, 3000078]             1  \n",
      "2  (3000018, 3000078]             1  \n",
      "3  (3000018, 3000078]             1  \n",
      "4  (3000018, 3000078]             1  \n",
      "TAIL with BINS\n",
      "       seqnames     start       end        CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  18261611  18261611  18_18261611    0.0    0.0    0.0   \n",
      "699996       18  18261615  18261615  18_18261615    0.0    0.0    0.0   \n",
      "699997       18  18261682  18261682  18_18261682    0.0    0.0    0.0   \n",
      "699998       18  18261702  18261702  18_18261702    4.0    0.0    0.0   \n",
      "699999       18  18261710  18261710  18_18261710    0.0    0.0    0.0   \n",
      "\n",
      "               bin_offset_60  target_count  \n",
      "699995  (18261578, 18261678]             1  \n",
      "699996  (18261578, 18261678]             1  \n",
      "699997  (18261678, 18261710]             1  \n",
      "699998  (18261678, 18261710]             1  \n",
      "699999  (18261678, 18261710]             1  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:137: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "         bin_offset_60  TT_S0  target_count starting_nt\n",
      "0   (3000018, 3000078]    4.0             5     3000018\n",
      "1   (3000078, 3000178]    4.0            10     3000078\n",
      "2   (3000178, 3000278]    0.0             1     3000178\n",
      "3   (3000278, 3000378]    2.0             5     3000278\n",
      "4   (3000378, 3000478]    0.0             3     3000378\n",
      "5   (3000478, 3000578]    0.0             1     3000478\n",
      "6   (3000578, 3000678]    0.0             4     3000578\n",
      "7   (3000678, 3000778]    0.0             4     3000678\n",
      "8   (3000778, 3000878]    0.0             6     3000778\n",
      "9   (3000878, 3000978]    0.0             4     3000878\n",
      "10  (3000978, 3001078]    2.0             6     3000978\n",
      "11  (3001078, 3001178]    0.0             2     3001078\n",
      "12  (3001178, 3001278]    0.0             2     3001178\n",
      "13  (3001278, 3001378]    0.0             5     3001278\n",
      "14  (3001378, 3001478]    0.0             3     3001378\n",
      "15  (3001478, 3001578]    4.0             9     3001478\n",
      "16  (3001578, 3001678]    3.0             6     3001578\n",
      "17  (3001678, 3001778]    0.0             7     3001678\n",
      "18  (3001778, 3001878]    0.0             5     3001778\n",
      "19  (3001878, 3001978]    0.0             1     3001878\n",
      "20  (3001978, 3002078]    1.0             4     3001978\n",
      "21  (3002078, 3002178]    0.0             1     3002078\n",
      "22  (3002178, 3002278]    4.0             7     3002178\n",
      "23  (3002278, 3002378]    9.0             3     3002278\n",
      "24  (3002378, 3002478]   19.0             4     3002378\n",
      "25  (3002478, 3002578]    4.0             3     3002478\n",
      "26  (3002578, 3002678]    0.0             1     3002578\n",
      "27  (3002678, 3002778]    0.0             3     3002678\n",
      "28  (3002778, 3002878]    0.0             2     3002778\n",
      "29  (3002878, 3002978]    1.0             2     3002878\n",
      "30  (3002978, 3003078]    2.0             8     3002978\n",
      "31  (3003078, 3003178]    0.0             0     3003078\n",
      "32  (3003178, 3003278]    0.0             0     3003178\n",
      "33  (3003278, 3003378]    1.0             9     3003278\n",
      "34  (3003378, 3003478]    0.0             1     3003378\n",
      "35  (3003478, 3003578]    0.0             3     3003478\n",
      "36  (3003578, 3003678]    1.0             2     3003578\n",
      "37  (3003678, 3003778]    4.0             8     3003678\n",
      "38  (3003778, 3003878]    0.0             1     3003778\n",
      "39  (3003878, 3003978]    8.0             9     3003878\n",
      "40  (3003978, 3004078]    0.0             0     3003978\n",
      "41  (3004078, 3004178]    0.0             0     3004078\n",
      "42  (3004178, 3004278]    3.0             5     3004178\n",
      "43  (3004278, 3004378]    2.0             3     3004278\n",
      "44  (3004378, 3004478]    0.0             0     3004378\n",
      "45  (3004478, 3004578]    0.0             6     3004478\n",
      "46  (3004578, 3004678]    1.0             2     3004578\n",
      "47  (3004678, 3004778]    0.0             0     3004678\n",
      "48  (3004778, 3004878]   10.0             6     3004778\n",
      "49  (3004878, 3004978]    3.0             3     3004878\n",
      "(152618, 4)\n",
      "bin_offset_60    category\n",
      "TT_S0             float64\n",
      "target_count        int64\n",
      "starting_nt      category\n",
      "dtype: object\n",
      "sumos ilgis: 152618\n",
      "ranges ilgis: 152618\n",
      "Model input - NO SCALE\n",
      "        TT_S0  genes  transcripts  exons  CDSs\n",
      "0         4.0      0            0      0     0\n",
      "1         4.0      0            0      0     0\n",
      "2         0.0      0            0      0     0\n",
      "3         2.0      0            0      0     0\n",
      "4         0.0      0            0      0     0\n",
      "...       ...    ...          ...    ...   ...\n",
      "152613    4.0      0            0      0     0\n",
      "152614    4.0      0            0      0     0\n",
      "152615    0.0      0            0      0     0\n",
      "152616    0.0      0            0      0     0\n",
      "152617    4.0      0            0      0     0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "        TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0         4.0             0                   0             0            0\n",
      "1         4.0             0                   0             0            0\n",
      "2         0.0             0                   0             0            0\n",
      "3         2.0             0                   0             0            0\n",
      "4         0.0             0                   0             0            0\n",
      "...       ...           ...                 ...           ...          ...\n",
      "152613    4.0             0                   0             0            0\n",
      "152614    4.0             0                   0             0            0\n",
      "152615    0.0             0                   0             0            0\n",
      "152616    0.0             0                   0             0            0\n",
      "152617    4.0             0                   0             0            0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "0.10518418535166232\n",
      "[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 1 0 0 1 1 1 0 1 1 1 1 1 0 1 1 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 0 0 0 1 1 1 1 0 0 1 1 0 1 0\n",
      " 1 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 1 1 1 1 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1\n",
      " 0 0 0 0 0 0 0 1 0 0 0 1 0 0 1 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 1 1 0 1 0 1 0 0 0 0 0 1 0\n",
      " 0 0 0 0 0 0 1 1 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 1 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0]\n",
      "threshold density 1.30546\n",
      "cluster signal 218597.0\n",
      "cluster target count 572142\n",
      "!!cluster density 0.38206773842857195\n",
      "cluster signal 238314.0\n",
      "cluster target count 127858\n",
      "!!cluster density 1.8638958844968636\n",
      "\n",
      "Labels mapping\n",
      " {0: 0, 1: 1}\n",
      "DEBUG: 136565 16053 \n",
      "\n",
      " +----------------------+-------------+----------------+-------+\n",
      "| bin_offset_60        | TT_S0       | target_count   | +9    |\n",
      "| (category)           | (float64)   | (int64)        | ...   |\n",
      "|----------------------+-------------+----------------+-------|\n",
      "| (3000018, 3000078]   | 4.0         | 5              | ...   |\n",
      "| (3000078, 3000178]   | 4.0         | 10             | ...   |\n",
      "| (3000178, 3000278]   | 0.0         | 1              | ...   |\n",
      "| (3000278, 3000378]   | 2.0         | 5              | ...   |\n",
      "| ...                  | ...         | ...            | ...   |\n",
      "| (18261378, 18261478] | 4.0         | 6              | ...   |\n",
      "| (18261478, 18261578] | 0.0         | 2              | ...   |\n",
      "| (18261578, 18261678] | 0.0         | 2              | ...   |\n",
      "| (18261678, 18261710] | 4.0         | 3              | ...   |\n",
      "+----------------------+-------------+----------------+-------+\n",
      "Unstranded PyRanges object has 136,565 rows and 12 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "9 hidden columns: starting_nt, Chromosome, Start, End, genes, transcripts, ... (+ 3 more.)\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0\n",
      "BINS  IntervalIndex([  (3000018, 3000083],   (3000083, 3000183],\n",
      "                 (3000183, 3000283],   (3000283, 3000383],\n",
      "                 (3000383, 3000483],   (3000483, 3000583],\n",
      "                 (3000583, 3000683],   (3000683, 3000783],\n",
      "                 (3000783, 3000883],   (3000883, 3000983],\n",
      "               ...\n",
      "               (18260783, 18260883], (18260883, 18260983],\n",
      "               (18260983, 18261083], (18261083, 18261183],\n",
      "               (18261183, 18261283], (18261283, 18261383],\n",
      "               (18261383, 18261483], (18261483, 18261583],\n",
      "               (18261583, 18261683], (18261683, 18261710]],\n",
      "              dtype='interval[int64, right]', length=152618)\n",
      "HEAD with BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0   \n",
      "\n",
      "        bin_offset_65  target_count  \n",
      "0  (3000018, 3000083]             1  \n",
      "1  (3000018, 3000083]             1  \n",
      "2  (3000018, 3000083]             1  \n",
      "3  (3000018, 3000083]             1  \n",
      "4  (3000018, 3000083]             1  \n",
      "TAIL with BINS\n",
      "       seqnames     start       end        CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  18261611  18261611  18_18261611    0.0    0.0    0.0   \n",
      "699996       18  18261615  18261615  18_18261615    0.0    0.0    0.0   \n",
      "699997       18  18261682  18261682  18_18261682    0.0    0.0    0.0   \n",
      "699998       18  18261702  18261702  18_18261702    4.0    0.0    0.0   \n",
      "699999       18  18261710  18261710  18_18261710    0.0    0.0    0.0   \n",
      "\n",
      "               bin_offset_65  target_count  \n",
      "699995  (18261583, 18261683]             1  \n",
      "699996  (18261583, 18261683]             1  \n",
      "699997  (18261583, 18261683]             1  \n",
      "699998  (18261683, 18261710]             1  \n",
      "699999  (18261683, 18261710]             1  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:137: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "         bin_offset_65  TT_S0  target_count starting_nt\n",
      "0   (3000018, 3000083]    4.0             5     3000018\n",
      "1   (3000083, 3000183]    4.0            10     3000083\n",
      "2   (3000183, 3000283]    0.0             1     3000183\n",
      "3   (3000283, 3000383]    2.0             5     3000283\n",
      "4   (3000383, 3000483]    0.0             3     3000383\n",
      "5   (3000483, 3000583]    0.0             1     3000483\n",
      "6   (3000583, 3000683]    0.0             4     3000583\n",
      "7   (3000683, 3000783]    0.0             4     3000683\n",
      "8   (3000783, 3000883]    0.0             6     3000783\n",
      "9   (3000883, 3000983]    0.0             4     3000883\n",
      "10  (3000983, 3001083]    2.0             6     3000983\n",
      "11  (3001083, 3001183]    0.0             2     3001083\n",
      "12  (3001183, 3001283]    0.0             2     3001183\n",
      "13  (3001283, 3001383]    0.0             5     3001283\n",
      "14  (3001383, 3001483]    0.0             3     3001383\n",
      "15  (3001483, 3001583]    4.0             9     3001483\n",
      "16  (3001583, 3001683]    3.0             7     3001583\n",
      "17  (3001683, 3001783]    0.0             7     3001683\n",
      "18  (3001783, 3001883]    0.0             4     3001783\n",
      "19  (3001883, 3001983]    0.0             2     3001883\n",
      "20  (3001983, 3002083]    1.0             3     3001983\n",
      "21  (3002083, 3002183]    0.0             1     3002083\n",
      "22  (3002183, 3002283]    4.0             7     3002183\n",
      "23  (3002283, 3002383]    9.0             3     3002283\n",
      "24  (3002383, 3002483]   19.0             4     3002383\n",
      "25  (3002483, 3002583]    4.0             3     3002483\n",
      "26  (3002583, 3002683]    0.0             1     3002583\n",
      "27  (3002683, 3002783]    0.0             3     3002683\n",
      "28  (3002783, 3002883]    0.0             2     3002783\n",
      "29  (3002883, 3002983]    1.0             3     3002883\n",
      "30  (3002983, 3003083]    2.0             7     3002983\n",
      "31  (3003083, 3003183]    0.0             0     3003083\n",
      "32  (3003183, 3003283]    0.0             0     3003183\n",
      "33  (3003283, 3003383]    1.0             9     3003283\n",
      "34  (3003383, 3003483]    0.0             1     3003383\n",
      "35  (3003483, 3003583]    0.0             3     3003483\n",
      "36  (3003583, 3003683]    2.0             3     3003583\n",
      "37  (3003683, 3003783]    3.0             7     3003683\n",
      "38  (3003783, 3003883]    0.0             1     3003783\n",
      "39  (3003883, 3003983]    8.0             9     3003883\n",
      "40  (3003983, 3004083]    0.0             0     3003983\n",
      "41  (3004083, 3004183]    0.0             0     3004083\n",
      "42  (3004183, 3004283]    3.0             5     3004183\n",
      "43  (3004283, 3004383]    2.0             3     3004283\n",
      "44  (3004383, 3004483]    0.0             0     3004383\n",
      "45  (3004483, 3004583]    0.0             6     3004483\n",
      "46  (3004583, 3004683]    1.0             2     3004583\n",
      "47  (3004683, 3004783]    0.0             0     3004683\n",
      "48  (3004783, 3004883]   10.0             7     3004783\n",
      "49  (3004883, 3004983]    3.0             2     3004883\n",
      "(152618, 4)\n",
      "bin_offset_65    category\n",
      "TT_S0             float64\n",
      "target_count        int64\n",
      "starting_nt      category\n",
      "dtype: object\n",
      "sumos ilgis: 152618\n",
      "ranges ilgis: 152618\n",
      "Model input - NO SCALE\n",
      "        TT_S0  genes  transcripts  exons  CDSs\n",
      "0         4.0      0            0      0     0\n",
      "1         4.0      0            0      0     0\n",
      "2         0.0      0            0      0     0\n",
      "3         2.0      0            0      0     0\n",
      "4         0.0      0            0      0     0\n",
      "...       ...    ...          ...    ...   ...\n",
      "152613    4.0      0            0      0     0\n",
      "152614    4.0      0            0      0     0\n",
      "152615    0.0      0            0      0     0\n",
      "152616    0.0      0            0      0     0\n",
      "152617    4.0      0            0      0     0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "        TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0         4.0             0                   0             0            0\n",
      "1         4.0             0                   0             0            0\n",
      "2         0.0             0                   0             0            0\n",
      "3         2.0             0                   0             0            0\n",
      "4         0.0             0                   0             0            0\n",
      "...       ...           ...                 ...           ...          ...\n",
      "152613    4.0             0                   0             0            0\n",
      "152614    4.0             0                   0             0            0\n",
      "152615    0.0             0                   0             0            0\n",
      "152616    0.0             0                   0             0            0\n",
      "152617    4.0             0                   0             0            0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "0.8475671283859046\n",
      "[1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 0 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 0 1 1 1 1 1 1 1 1 0 1 1 0 0 0 0 0 0 0 0 0 1 0 0 1 1 1 0 1 1 1 0 1 1 1\n",
      " 1 1 1 1 1 1 0 1 1 1 1 1 1 0 1 1 1 0 0 1 1 1 1 1 0 1 0 0 0 0 1 1 0 0 1 0 1\n",
      " 0 0 1 1 1 1 1 1 1 1 1 1 1 1 0 0 0 0 0 0 1 0 1 1 1 1 0 1 1 1 1 1 1 1 1 1 0\n",
      " 1 1 1 1 1 1 1 0 1 1 1 0 1 0 1 1 0 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 1 1 1 1 0 0 1 0 1 0 0 1 1 1 1 0 1\n",
      " 1 1 1 1 1 1 0 0 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 1 0 1 1 1 0 1 0 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 0 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 0 1 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 0 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1]\n",
      "threshold density 1.30546\n",
      "cluster signal 165103.0\n",
      "cluster target count 525349\n",
      "!!cluster density 0.31427298805175224\n",
      "cluster signal 291808.0\n",
      "cluster target count 174651\n",
      "!!cluster density 1.6708063509513258\n",
      "\n",
      "Labels mapping\n",
      " {1: 0, 0: 1}\n",
      "DEBUG: 129354 23264 \n",
      "\n",
      " +----------------------+-------------+----------------+-------+\n",
      "| bin_offset_65        | TT_S0       | target_count   | +9    |\n",
      "| (category)           | (float64)   | (int64)        | ...   |\n",
      "|----------------------+-------------+----------------+-------|\n",
      "| (3000018, 3000083]   | 4.0         | 5              | ...   |\n",
      "| (3000083, 3000183]   | 4.0         | 10             | ...   |\n",
      "| (3000183, 3000283]   | 0.0         | 1              | ...   |\n",
      "| (3000283, 3000383]   | 2.0         | 5              | ...   |\n",
      "| ...                  | ...         | ...            | ...   |\n",
      "| (18261383, 18261483] | 4.0         | 5              | ...   |\n",
      "| (18261483, 18261583] | 0.0         | 2              | ...   |\n",
      "| (18261583, 18261683] | 0.0         | 3              | ...   |\n",
      "| (18261683, 18261710] | 4.0         | 2              | ...   |\n",
      "+----------------------+-------------+----------------+-------+\n",
      "Unstranded PyRanges object has 129,354 rows and 12 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "9 hidden columns: starting_nt, Chromosome, Start, End, genes, transcripts, ... (+ 3 more.)\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0\n",
      "BINS  IntervalIndex([  (3000018, 3000088],   (3000088, 3000188],\n",
      "                 (3000188, 3000288],   (3000288, 3000388],\n",
      "                 (3000388, 3000488],   (3000488, 3000588],\n",
      "                 (3000588, 3000688],   (3000688, 3000788],\n",
      "                 (3000788, 3000888],   (3000888, 3000988],\n",
      "               ...\n",
      "               (18260788, 18260888], (18260888, 18260988],\n",
      "               (18260988, 18261088], (18261088, 18261188],\n",
      "               (18261188, 18261288], (18261288, 18261388],\n",
      "               (18261388, 18261488], (18261488, 18261588],\n",
      "               (18261588, 18261688], (18261688, 18261710]],\n",
      "              dtype='interval[int64, right]', length=152618)\n",
      "HEAD with BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0   \n",
      "\n",
      "        bin_offset_70  target_count  \n",
      "0  (3000018, 3000088]             1  \n",
      "1  (3000018, 3000088]             1  \n",
      "2  (3000018, 3000088]             1  \n",
      "3  (3000018, 3000088]             1  \n",
      "4  (3000018, 3000088]             1  \n",
      "TAIL with BINS\n",
      "       seqnames     start       end        CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  18261611  18261611  18_18261611    0.0    0.0    0.0   \n",
      "699996       18  18261615  18261615  18_18261615    0.0    0.0    0.0   \n",
      "699997       18  18261682  18261682  18_18261682    0.0    0.0    0.0   \n",
      "699998       18  18261702  18261702  18_18261702    4.0    0.0    0.0   \n",
      "699999       18  18261710  18261710  18_18261710    0.0    0.0    0.0   \n",
      "\n",
      "               bin_offset_70  target_count  \n",
      "699995  (18261588, 18261688]             1  \n",
      "699996  (18261588, 18261688]             1  \n",
      "699997  (18261588, 18261688]             1  \n",
      "699998  (18261688, 18261710]             1  \n",
      "699999  (18261688, 18261710]             1  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:137: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "         bin_offset_70  TT_S0  target_count starting_nt\n",
      "0   (3000018, 3000088]    7.0             8     3000018\n",
      "1   (3000088, 3000188]    1.0             7     3000088\n",
      "2   (3000188, 3000288]    0.0             1     3000188\n",
      "3   (3000288, 3000388]    2.0             7     3000288\n",
      "4   (3000388, 3000488]    0.0             1     3000388\n",
      "5   (3000488, 3000588]    0.0             1     3000488\n",
      "6   (3000588, 3000688]    0.0             5     3000588\n",
      "7   (3000688, 3000788]    0.0             5     3000688\n",
      "8   (3000788, 3000888]    0.0             5     3000788\n",
      "9   (3000888, 3000988]    0.0             3     3000888\n",
      "10  (3000988, 3001088]    2.0             6     3000988\n",
      "11  (3001088, 3001188]    0.0             3     3001088\n",
      "12  (3001188, 3001288]    0.0             1     3001188\n",
      "13  (3001288, 3001388]    0.0             5     3001288\n",
      "14  (3001388, 3001488]    0.0             4     3001388\n",
      "15  (3001488, 3001588]    4.0             8     3001488\n",
      "16  (3001588, 3001688]    3.0             7     3001588\n",
      "17  (3001688, 3001788]    0.0             7     3001688\n",
      "18  (3001788, 3001888]    0.0             4     3001788\n",
      "19  (3001888, 3001988]    0.0             2     3001888\n",
      "20  (3001988, 3002088]    1.0             3     3001988\n",
      "21  (3002088, 3002188]    0.0             1     3002088\n",
      "22  (3002188, 3002288]    4.0             7     3002188\n",
      "23  (3002288, 3002388]    9.0             3     3002288\n",
      "24  (3002388, 3002488]   19.0             4     3002388\n",
      "25  (3002488, 3002588]    4.0             3     3002488\n",
      "26  (3002588, 3002688]    0.0             1     3002588\n",
      "27  (3002688, 3002788]    0.0             3     3002688\n",
      "28  (3002788, 3002888]    0.0             2     3002788\n",
      "29  (3002888, 3002988]    1.0             4     3002888\n",
      "30  (3002988, 3003088]    2.0             6     3002988\n",
      "31  (3003088, 3003188]    0.0             0     3003088\n",
      "32  (3003188, 3003288]    0.0             0     3003188\n",
      "33  (3003288, 3003388]    1.0             9     3003288\n",
      "34  (3003388, 3003488]    0.0             1     3003388\n",
      "35  (3003488, 3003588]    1.0             4     3003488\n",
      "36  (3003588, 3003688]    1.0             3     3003588\n",
      "37  (3003688, 3003788]    3.0             6     3003688\n",
      "38  (3003788, 3003888]    0.0             1     3003788\n",
      "39  (3003888, 3003988]    8.0             9     3003888\n",
      "40  (3003988, 3004088]    0.0             0     3003988\n",
      "41  (3004088, 3004188]    0.0             0     3004088\n",
      "42  (3004188, 3004288]    3.0             5     3004188\n",
      "43  (3004288, 3004388]    2.0             3     3004288\n",
      "44  (3004388, 3004488]    0.0             0     3004388\n",
      "45  (3004488, 3004588]    0.0             6     3004488\n",
      "46  (3004588, 3004688]    1.0             2     3004588\n",
      "47  (3004688, 3004788]    0.0             0     3004688\n",
      "48  (3004788, 3004888]   10.0             7     3004788\n",
      "49  (3004888, 3004988]    3.0             2     3004888\n",
      "(152618, 4)\n",
      "bin_offset_70    category\n",
      "TT_S0             float64\n",
      "target_count        int64\n",
      "starting_nt      category\n",
      "dtype: object\n",
      "sumos ilgis: 152618\n",
      "ranges ilgis: 152618\n",
      "Model input - NO SCALE\n",
      "        TT_S0  genes  transcripts  exons  CDSs\n",
      "0         7.0      0            0      0     0\n",
      "1         1.0      0            0      0     0\n",
      "2         0.0      0            0      0     0\n",
      "3         2.0      0            0      0     0\n",
      "4         0.0      0            0      0     0\n",
      "...       ...    ...          ...    ...   ...\n",
      "152613    4.0      0            0      0     0\n",
      "152614    4.0      0            0      0     0\n",
      "152615    0.0      0            0      0     0\n",
      "152616    0.0      0            0      0     0\n",
      "152617    4.0      0            0      0     0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "        TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0         7.0             0                   0             0            0\n",
      "1         1.0             0                   0             0            0\n",
      "2         0.0             0                   0             0            0\n",
      "3         2.0             0                   0             0            0\n",
      "4         0.0             0                   0             0            0\n",
      "...       ...           ...                 ...           ...          ...\n",
      "152613    4.0             0                   0             0            0\n",
      "152614    4.0             0                   0             0            0\n",
      "152615    0.0             0                   0             0            0\n",
      "152616    0.0             0                   0             0            0\n",
      "152617    4.0             0                   0             0            0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "6.5523070673184025e-06\n",
      "[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0]\n",
      "threshold density 1.30546\n",
      "cluster signal 456419.0\n",
      "cluster target count 699993\n",
      "!!cluster density 0.6520336631937748\n",
      "cluster signal 492.0\n",
      "cluster target count 7\n",
      "!!cluster density 70.28571428571429\n",
      "\n",
      "Labels mapping\n",
      " {0: 0, 1: 1}\n",
      "DEBUG: 152617 1 \n",
      "\n",
      " +----------------------+-------------+----------------+-------+\n",
      "| bin_offset_70        | TT_S0       | target_count   | +9    |\n",
      "| (category)           | (float64)   | (int64)        | ...   |\n",
      "|----------------------+-------------+----------------+-------|\n",
      "| (3000018, 3000088]   | 7.0         | 8              | ...   |\n",
      "| (3000088, 3000188]   | 1.0         | 7              | ...   |\n",
      "| (3000188, 3000288]   | 0.0         | 1              | ...   |\n",
      "| (3000288, 3000388]   | 2.0         | 7              | ...   |\n",
      "| ...                  | ...         | ...            | ...   |\n",
      "| (18261388, 18261488] | 4.0         | 6              | ...   |\n",
      "| (18261488, 18261588] | 0.0         | 1              | ...   |\n",
      "| (18261588, 18261688] | 0.0         | 3              | ...   |\n",
      "| (18261688, 18261710] | 4.0         | 2              | ...   |\n",
      "+----------------------+-------------+----------------+-------+\n",
      "Unstranded PyRanges object has 152,617 rows and 12 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "9 hidden columns: starting_nt, Chromosome, Start, End, genes, transcripts, ... (+ 3 more.)\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0\n",
      "BINS  IntervalIndex([  (3000018, 3000093],   (3000093, 3000193],\n",
      "                 (3000193, 3000293],   (3000293, 3000393],\n",
      "                 (3000393, 3000493],   (3000493, 3000593],\n",
      "                 (3000593, 3000693],   (3000693, 3000793],\n",
      "                 (3000793, 3000893],   (3000893, 3000993],\n",
      "               ...\n",
      "               (18260793, 18260893], (18260893, 18260993],\n",
      "               (18260993, 18261093], (18261093, 18261193],\n",
      "               (18261193, 18261293], (18261293, 18261393],\n",
      "               (18261393, 18261493], (18261493, 18261593],\n",
      "               (18261593, 18261693], (18261693, 18261710]],\n",
      "              dtype='interval[int64, right]', length=152618)\n",
      "HEAD with BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0   \n",
      "\n",
      "        bin_offset_75  target_count  \n",
      "0  (3000018, 3000093]             1  \n",
      "1  (3000018, 3000093]             1  \n",
      "2  (3000018, 3000093]             1  \n",
      "3  (3000018, 3000093]             1  \n",
      "4  (3000018, 3000093]             1  \n",
      "TAIL with BINS\n",
      "       seqnames     start       end        CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  18261611  18261611  18_18261611    0.0    0.0    0.0   \n",
      "699996       18  18261615  18261615  18_18261615    0.0    0.0    0.0   \n",
      "699997       18  18261682  18261682  18_18261682    0.0    0.0    0.0   \n",
      "699998       18  18261702  18261702  18_18261702    4.0    0.0    0.0   \n",
      "699999       18  18261710  18261710  18_18261710    0.0    0.0    0.0   \n",
      "\n",
      "               bin_offset_75  target_count  \n",
      "699995  (18261593, 18261693]             1  \n",
      "699996  (18261593, 18261693]             1  \n",
      "699997  (18261593, 18261693]             1  \n",
      "699998  (18261693, 18261710]             1  \n",
      "699999  (18261693, 18261710]             1  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:137: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "         bin_offset_75  TT_S0  target_count starting_nt\n",
      "0   (3000018, 3000093]    7.0             9     3000018\n",
      "1   (3000093, 3000193]    1.0             6     3000093\n",
      "2   (3000193, 3000293]    0.0             1     3000193\n",
      "3   (3000293, 3000393]    2.0             7     3000293\n",
      "4   (3000393, 3000493]    0.0             1     3000393\n",
      "5   (3000493, 3000593]    0.0             1     3000493\n",
      "6   (3000593, 3000693]    0.0             6     3000593\n",
      "7   (3000693, 3000793]    0.0             4     3000693\n",
      "8   (3000793, 3000893]    0.0             6     3000793\n",
      "9   (3000893, 3000993]    0.0             2     3000893\n",
      "10  (3000993, 3001093]    2.0             6     3000993\n",
      "11  (3001093, 3001193]    0.0             3     3001093\n",
      "12  (3001193, 3001293]    0.0             2     3001193\n",
      "13  (3001293, 3001393]    0.0             5     3001293\n",
      "14  (3001393, 3001493]    0.0             3     3001393\n",
      "15  (3001493, 3001593]    4.0             9     3001493\n",
      "16  (3001593, 3001693]    3.0             7     3001593\n",
      "17  (3001693, 3001793]    0.0             7     3001693\n",
      "18  (3001793, 3001893]    0.0             3     3001793\n",
      "19  (3001893, 3001993]    0.0             2     3001893\n",
      "20  (3001993, 3002093]    1.0             3     3001993\n",
      "21  (3002093, 3002193]    0.0             2     3002093\n",
      "22  (3002193, 3002293]    4.0             6     3002193\n",
      "23  (3002293, 3002393]    9.0             3     3002293\n",
      "24  (3002393, 3002493]   19.0             5     3002393\n",
      "25  (3002493, 3002593]    4.0             2     3002493\n",
      "26  (3002593, 3002693]    0.0             2     3002593\n",
      "27  (3002693, 3002793]    0.0             2     3002693\n",
      "28  (3002793, 3002893]    0.0             2     3002793\n",
      "29  (3002893, 3002993]    1.0             5     3002893\n",
      "30  (3002993, 3003093]    2.0             5     3002993\n",
      "31  (3003093, 3003193]    0.0             0     3003093\n",
      "32  (3003193, 3003293]    0.0             0     3003193\n",
      "33  (3003293, 3003393]    1.0             9     3003293\n",
      "34  (3003393, 3003493]    0.0             1     3003393\n",
      "35  (3003493, 3003593]    1.0             4     3003493\n",
      "36  (3003593, 3003693]    1.0             3     3003593\n",
      "37  (3003693, 3003793]    3.0             6     3003693\n",
      "38  (3003793, 3003893]    0.0             2     3003793\n",
      "39  (3003893, 3003993]    8.0             8     3003893\n",
      "40  (3003993, 3004093]    0.0             0     3003993\n",
      "41  (3004093, 3004193]    0.0             1     3004093\n",
      "42  (3004193, 3004293]    3.0             4     3004193\n",
      "43  (3004293, 3004393]    2.0             3     3004293\n",
      "44  (3004393, 3004493]    0.0             1     3004393\n",
      "45  (3004493, 3004593]    0.0             5     3004493\n",
      "46  (3004593, 3004693]    1.0             2     3004593\n",
      "47  (3004693, 3004793]    0.0             0     3004693\n",
      "48  (3004793, 3004893]   10.0             7     3004793\n",
      "49  (3004893, 3004993]    3.0             2     3004893\n",
      "(152618, 4)\n",
      "bin_offset_75    category\n",
      "TT_S0             float64\n",
      "target_count        int64\n",
      "starting_nt      category\n",
      "dtype: object\n",
      "sumos ilgis: 152618\n",
      "ranges ilgis: 152618\n",
      "Model input - NO SCALE\n",
      "        TT_S0  genes  transcripts  exons  CDSs\n",
      "0         7.0      0            0      0     0\n",
      "1         1.0      0            0      0     0\n",
      "2         0.0      0            0      0     0\n",
      "3         2.0      0            0      0     0\n",
      "4         0.0      0            0      0     0\n",
      "...       ...    ...          ...    ...   ...\n",
      "152613    4.0      0            0      0     0\n",
      "152614    4.0      0            0      0     0\n",
      "152615    0.0      0            0      0     0\n",
      "152616    0.0      0            0      0     0\n",
      "152617    4.0      0            0      0     0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "        TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0         7.0             0                   0             0            0\n",
      "1         1.0             0                   0             0            0\n",
      "2         0.0             0                   0             0            0\n",
      "3         2.0             0                   0             0            0\n",
      "4         0.0             0                   0             0            0\n",
      "...       ...           ...                 ...           ...          ...\n",
      "152613    4.0             0                   0             0            0\n",
      "152614    4.0             0                   0             0            0\n",
      "152615    0.0             0                   0             0            0\n",
      "152616    0.0             0                   0             0            0\n",
      "152617    4.0             0                   0             0            0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "0.1528915331088076\n",
      "[1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 1 0 0 0 0 0 0 0 0 1 0 0 1 1 1 1 1 1 1 1 1 0 1 1 0 0 0 1 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 1 0 0 0 0 0 0 1 0 0 0 1 0 0 0 0 0 0 1 0 1 1 1 1 0 0 1 1 0 1 0\n",
      " 1 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 1 1 1 1 0 1 0 0 0 0 1 0 0 0 0 0 0 0 0 0 1\n",
      " 0 0 0 0 0 0 0 1 0 0 0 1 0 1 0 0 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 1 1 0 1 0 1 0 0 0 0 0 1 0\n",
      " 0 0 0 0 0 0 1 1 0 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 1 0 0 0 1 0 1 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 1 1 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0]\n",
      "threshold density 1.30546\n",
      "cluster signal 292143.0\n",
      "cluster target count 175284\n",
      "!!cluster density 1.6666837817484768\n",
      "cluster signal 164768.0\n",
      "cluster target count 524716\n",
      "!!cluster density 0.3140136759694768\n",
      "\n",
      "Labels mapping\n",
      " {1: 1, 0: 0}\n",
      "DEBUG: 129284 23334 \n",
      "\n",
      " +----------------------+-------------+----------------+-------+\n",
      "| bin_offset_75        | TT_S0       | target_count   | +9    |\n",
      "| (category)           | (float64)   | (int64)        | ...   |\n",
      "|----------------------+-------------+----------------+-------|\n",
      "| (3000093, 3000193]   | 1.0         | 6              | ...   |\n",
      "| (3000193, 3000293]   | 0.0         | 1              | ...   |\n",
      "| (3000293, 3000393]   | 2.0         | 7              | ...   |\n",
      "| (3000393, 3000493]   | 0.0         | 1              | ...   |\n",
      "| ...                  | ...         | ...            | ...   |\n",
      "| (18261393, 18261493] | 4.0         | 6              | ...   |\n",
      "| (18261493, 18261593] | 0.0         | 1              | ...   |\n",
      "| (18261593, 18261693] | 0.0         | 3              | ...   |\n",
      "| (18261693, 18261710] | 4.0         | 2              | ...   |\n",
      "+----------------------+-------------+----------------+-------+\n",
      "Unstranded PyRanges object has 129,284 rows and 12 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "9 hidden columns: starting_nt, Chromosome, Start, End, genes, transcripts, ... (+ 3 more.)\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0\n",
      "BINS  IntervalIndex([  (3000018, 3000098],   (3000098, 3000198],\n",
      "                 (3000198, 3000298],   (3000298, 3000398],\n",
      "                 (3000398, 3000498],   (3000498, 3000598],\n",
      "                 (3000598, 3000698],   (3000698, 3000798],\n",
      "                 (3000798, 3000898],   (3000898, 3000998],\n",
      "               ...\n",
      "               (18260798, 18260898], (18260898, 18260998],\n",
      "               (18260998, 18261098], (18261098, 18261198],\n",
      "               (18261198, 18261298], (18261298, 18261398],\n",
      "               (18261398, 18261498], (18261498, 18261598],\n",
      "               (18261598, 18261698], (18261698, 18261710]],\n",
      "              dtype='interval[int64, right]', length=152618)\n",
      "HEAD with BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0   \n",
      "\n",
      "        bin_offset_80  target_count  \n",
      "0  (3000018, 3000098]             1  \n",
      "1  (3000018, 3000098]             1  \n",
      "2  (3000018, 3000098]             1  \n",
      "3  (3000018, 3000098]             1  \n",
      "4  (3000018, 3000098]             1  \n",
      "TAIL with BINS\n",
      "       seqnames     start       end        CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  18261611  18261611  18_18261611    0.0    0.0    0.0   \n",
      "699996       18  18261615  18261615  18_18261615    0.0    0.0    0.0   \n",
      "699997       18  18261682  18261682  18_18261682    0.0    0.0    0.0   \n",
      "699998       18  18261702  18261702  18_18261702    4.0    0.0    0.0   \n",
      "699999       18  18261710  18261710  18_18261710    0.0    0.0    0.0   \n",
      "\n",
      "               bin_offset_80  target_count  \n",
      "699995  (18261598, 18261698]             1  \n",
      "699996  (18261598, 18261698]             1  \n",
      "699997  (18261598, 18261698]             1  \n",
      "699998  (18261698, 18261710]             1  \n",
      "699999  (18261698, 18261710]             1  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:137: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "         bin_offset_80  TT_S0  target_count starting_nt\n",
      "0   (3000018, 3000098]    7.0            10     3000018\n",
      "1   (3000098, 3000198]    1.0             5     3000098\n",
      "2   (3000198, 3000298]    0.0             1     3000198\n",
      "3   (3000298, 3000398]    2.0             7     3000298\n",
      "4   (3000398, 3000498]    0.0             1     3000398\n",
      "5   (3000498, 3000598]    0.0             1     3000498\n",
      "6   (3000598, 3000698]    0.0             6     3000598\n",
      "7   (3000698, 3000798]    0.0             5     3000698\n",
      "8   (3000798, 3000898]    0.0             5     3000798\n",
      "9   (3000898, 3000998]    0.0             2     3000898\n",
      "10  (3000998, 3001098]    2.0             7     3000998\n",
      "11  (3001098, 3001198]    0.0             2     3001098\n",
      "12  (3001198, 3001298]    0.0             2     3001198\n",
      "13  (3001298, 3001398]    0.0             5     3001298\n",
      "14  (3001398, 3001498]    0.0             3     3001398\n",
      "15  (3001498, 3001598]    4.0            10     3001498\n",
      "16  (3001598, 3001698]    3.0             6     3001598\n",
      "17  (3001698, 3001798]    0.0             7     3001698\n",
      "18  (3001798, 3001898]    0.0             3     3001798\n",
      "19  (3001898, 3001998]    0.0             2     3001898\n",
      "20  (3001998, 3002098]    1.0             3     3001998\n",
      "21  (3002098, 3002198]    0.0             2     3002098\n",
      "22  (3002198, 3002298]   11.0             7     3002198\n",
      "23  (3002298, 3002398]    2.0             2     3002298\n",
      "24  (3002398, 3002498]   19.0             5     3002398\n",
      "25  (3002498, 3002598]    4.0             2     3002498\n",
      "26  (3002598, 3002698]    0.0             2     3002598\n",
      "27  (3002698, 3002798]    0.0             2     3002698\n",
      "28  (3002798, 3002898]    0.0             2     3002798\n",
      "29  (3002898, 3002998]    1.0             5     3002898\n",
      "30  (3002998, 3003098]    2.0             5     3002998\n",
      "31  (3003098, 3003198]    0.0             0     3003098\n",
      "32  (3003198, 3003298]    0.0             0     3003198\n",
      "33  (3003298, 3003398]    1.0             9     3003298\n",
      "34  (3003398, 3003498]    0.0             1     3003398\n",
      "35  (3003498, 3003598]    1.0             4     3003498\n",
      "36  (3003598, 3003698]    1.0             3     3003598\n",
      "37  (3003698, 3003798]    3.0             6     3003698\n",
      "38  (3003798, 3003898]    0.0             2     3003798\n",
      "39  (3003898, 3003998]    8.0             8     3003898\n",
      "40  (3003998, 3004098]    0.0             0     3003998\n",
      "41  (3004098, 3004198]    0.0             1     3004098\n",
      "42  (3004198, 3004298]    3.0             5     3004198\n",
      "43  (3004298, 3004398]    2.0             2     3004298\n",
      "44  (3004398, 3004498]    0.0             1     3004398\n",
      "45  (3004498, 3004598]    0.0             6     3004498\n",
      "46  (3004598, 3004698]    1.0             1     3004598\n",
      "47  (3004698, 3004798]    0.0             0     3004698\n",
      "48  (3004798, 3004898]   10.0             7     3004798\n",
      "49  (3004898, 3004998]    3.0             2     3004898\n",
      "(152618, 4)\n",
      "bin_offset_80    category\n",
      "TT_S0             float64\n",
      "target_count        int64\n",
      "starting_nt      category\n",
      "dtype: object\n",
      "sumos ilgis: 152618\n",
      "ranges ilgis: 152618\n",
      "Model input - NO SCALE\n",
      "        TT_S0  genes  transcripts  exons  CDSs\n",
      "0         7.0      0            0      0     0\n",
      "1         1.0      0            0      0     0\n",
      "2         0.0      0            0      0     0\n",
      "3         2.0      0            0      0     0\n",
      "4         0.0      0            0      0     0\n",
      "...       ...    ...          ...    ...   ...\n",
      "152613    3.0      0            0      0     0\n",
      "152614    4.0      0            0      0     0\n",
      "152615    0.0      0            0      0     0\n",
      "152616    0.0      0            0      0     0\n",
      "152617    4.0      0            0      0     0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "        TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0         7.0             0                   0             0            0\n",
      "1         1.0             0                   0             0            0\n",
      "2         0.0             0                   0             0            0\n",
      "3         2.0             0                   0             0            0\n",
      "4         0.0             0                   0             0            0\n",
      "...       ...           ...                 ...           ...          ...\n",
      "152613    3.0             0                   0             0            0\n",
      "152614    4.0             0                   0             0            0\n",
      "152615    0.0             0                   0             0            0\n",
      "152616    0.0             0                   0             0            0\n",
      "152617    4.0             0                   0             0            0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "0.15293739925827884\n",
      "[1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 1 0 0 0 0 0 0 0 0 1 0 0 1 1 1 1 1 1 1 1 1 0 1 1 0 0 0 1 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 1 0 0 0 0 0 0 1 0 0 1 1 0 0 0 0 0 0 1 0 1 1 1 1 0 1 1 1 0 1 0\n",
      " 1 0 0 0 0 0 0 0 0 0 0 0 0 1 0 1 1 1 1 1 0 1 0 0 0 0 1 0 0 0 0 0 0 0 0 0 1\n",
      " 0 0 0 0 0 0 0 1 0 0 0 1 0 1 0 0 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 1 1 0 1 0 1 0 0 0 0 0 1 0\n",
      " 0 0 0 0 0 0 1 1 0 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 1 0 0 0 1 0 1 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 1 1 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 1 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 1 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0\n",
      " 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0]\n",
      "threshold density 1.30546\n",
      "cluster signal 292139.0\n",
      "cluster target count 175344\n",
      "!!cluster density 1.6660906560817592\n",
      "cluster signal 164772.0\n",
      "cluster target count 524656\n",
      "!!cluster density 0.31405721082004207\n",
      "\n",
      "Labels mapping\n",
      " {1: 1, 0: 0}\n",
      "DEBUG: 129277 23341 \n",
      "\n",
      " +----------------------+-------------+----------------+-------+\n",
      "| bin_offset_80        | TT_S0       | target_count   | +9    |\n",
      "| (category)           | (float64)   | (int64)        | ...   |\n",
      "|----------------------+-------------+----------------+-------|\n",
      "| (3000098, 3000198]   | 1.0         | 5              | ...   |\n",
      "| (3000198, 3000298]   | 0.0         | 1              | ...   |\n",
      "| (3000298, 3000398]   | 2.0         | 7              | ...   |\n",
      "| (3000398, 3000498]   | 0.0         | 1              | ...   |\n",
      "| ...                  | ...         | ...            | ...   |\n",
      "| (18261398, 18261498] | 4.0         | 5              | ...   |\n",
      "| (18261498, 18261598] | 0.0         | 1              | ...   |\n",
      "| (18261598, 18261698] | 0.0         | 3              | ...   |\n",
      "| (18261698, 18261710] | 4.0         | 2              | ...   |\n",
      "+----------------------+-------------+----------------+-------+\n",
      "Unstranded PyRanges object has 129,277 rows and 12 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "9 hidden columns: starting_nt, Chromosome, Start, End, genes, transcripts, ... (+ 3 more.)\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0\n",
      "BINS  IntervalIndex([  (3000018, 3000103],   (3000103, 3000203],\n",
      "                 (3000203, 3000303],   (3000303, 3000403],\n",
      "                 (3000403, 3000503],   (3000503, 3000603],\n",
      "                 (3000603, 3000703],   (3000703, 3000803],\n",
      "                 (3000803, 3000903],   (3000903, 3001003],\n",
      "               ...\n",
      "               (18260803, 18260903], (18260903, 18261003],\n",
      "               (18261003, 18261103], (18261103, 18261203],\n",
      "               (18261203, 18261303], (18261303, 18261403],\n",
      "               (18261403, 18261503], (18261503, 18261603],\n",
      "               (18261603, 18261703], (18261703, 18261710]],\n",
      "              dtype='interval[int64, right]', length=152618)\n",
      "HEAD with BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0   \n",
      "\n",
      "        bin_offset_85  target_count  \n",
      "0  (3000018, 3000103]             1  \n",
      "1  (3000018, 3000103]             1  \n",
      "2  (3000018, 3000103]             1  \n",
      "3  (3000018, 3000103]             1  \n",
      "4  (3000018, 3000103]             1  \n",
      "TAIL with BINS\n",
      "       seqnames     start       end        CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  18261611  18261611  18_18261611    0.0    0.0    0.0   \n",
      "699996       18  18261615  18261615  18_18261615    0.0    0.0    0.0   \n",
      "699997       18  18261682  18261682  18_18261682    0.0    0.0    0.0   \n",
      "699998       18  18261702  18261702  18_18261702    4.0    0.0    0.0   \n",
      "699999       18  18261710  18261710  18_18261710    0.0    0.0    0.0   \n",
      "\n",
      "               bin_offset_85  target_count  \n",
      "699995  (18261603, 18261703]             1  \n",
      "699996  (18261603, 18261703]             1  \n",
      "699997  (18261603, 18261703]             1  \n",
      "699998  (18261603, 18261703]             1  \n",
      "699999  (18261703, 18261710]             1  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:137: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "         bin_offset_85  TT_S0  target_count starting_nt\n",
      "0   (3000018, 3000103]    7.0            11     3000018\n",
      "1   (3000103, 3000203]    1.0             4     3000103\n",
      "2   (3000203, 3000303]    0.0             1     3000203\n",
      "3   (3000303, 3000403]    2.0             7     3000303\n",
      "4   (3000403, 3000503]    0.0             1     3000403\n",
      "5   (3000503, 3000603]    0.0             1     3000503\n",
      "6   (3000603, 3000703]    0.0             6     3000603\n",
      "7   (3000703, 3000803]    0.0             5     3000703\n",
      "8   (3000803, 3000903]    0.0             5     3000803\n",
      "9   (3000903, 3001003]    0.0             3     3000903\n",
      "10  (3001003, 3001103]    2.0             6     3001003\n",
      "11  (3001103, 3001203]    0.0             2     3001103\n",
      "12  (3001203, 3001303]    0.0             3     3001203\n",
      "13  (3001303, 3001403]    0.0             4     3001303\n",
      "14  (3001403, 3001503]    1.0             4     3001403\n",
      "15  (3001503, 3001603]    3.0             9     3001503\n",
      "16  (3001603, 3001703]    3.0             6     3001603\n",
      "17  (3001703, 3001803]    0.0             8     3001703\n",
      "18  (3001803, 3001903]    0.0             2     3001803\n",
      "19  (3001903, 3002003]    0.0             2     3001903\n",
      "20  (3002003, 3002103]    1.0             3     3002003\n",
      "21  (3002103, 3002203]    0.0             2     3002103\n",
      "22  (3002203, 3002303]   11.0             7     3002203\n",
      "23  (3002303, 3002403]    2.0             2     3002303\n",
      "24  (3002403, 3002503]   19.0             5     3002403\n",
      "25  (3002503, 3002603]    4.0             2     3002503\n",
      "26  (3002603, 3002703]    0.0             3     3002603\n",
      "27  (3002703, 3002803]    0.0             1     3002703\n",
      "28  (3002803, 3002903]    0.0             2     3002803\n",
      "29  (3002903, 3003003]    1.0             5     3002903\n",
      "30  (3003003, 3003103]    2.0             5     3003003\n",
      "31  (3003103, 3003203]    0.0             0     3003103\n",
      "32  (3003203, 3003303]    0.0             0     3003203\n",
      "33  (3003303, 3003403]    1.0             9     3003303\n",
      "34  (3003403, 3003503]    0.0             2     3003403\n",
      "35  (3003503, 3003603]    1.0             3     3003503\n",
      "36  (3003603, 3003703]    1.0             3     3003603\n",
      "37  (3003703, 3003803]    3.0             6     3003703\n",
      "38  (3003803, 3003903]    0.0             3     3003803\n",
      "39  (3003903, 3004003]    8.0             7     3003903\n",
      "40  (3004003, 3004103]    0.0             0     3004003\n",
      "41  (3004103, 3004203]    0.0             1     3004103\n",
      "42  (3004203, 3004303]    5.0             6     3004203\n",
      "43  (3004303, 3004403]    0.0             1     3004303\n",
      "44  (3004403, 3004503]    0.0             1     3004403\n",
      "45  (3004503, 3004603]    1.0             7     3004503\n",
      "46  (3004603, 3004703]    0.0             0     3004603\n",
      "47  (3004703, 3004803]    0.0             0     3004703\n",
      "48  (3004803, 3004903]   10.0             7     3004803\n",
      "49  (3004903, 3005003]    3.0             2     3004903\n",
      "(152618, 4)\n",
      "bin_offset_85    category\n",
      "TT_S0             float64\n",
      "target_count        int64\n",
      "starting_nt      category\n",
      "dtype: object\n",
      "sumos ilgis: 152618\n",
      "ranges ilgis: 152618\n",
      "Model input - NO SCALE\n",
      "        TT_S0  genes  transcripts  exons  CDSs\n",
      "0         7.0      0            0      0     0\n",
      "1         1.0      0            0      0     0\n",
      "2         0.0      0            0      0     0\n",
      "3         2.0      0            0      0     0\n",
      "4         0.0      0            0      0     0\n",
      "...       ...    ...          ...    ...   ...\n",
      "152613    3.0      0            0      0     0\n",
      "152614    4.0      0            0      0     0\n",
      "152615    0.0      0            0      0     0\n",
      "152616    4.0      0            0      0     0\n",
      "152617    0.0      0            0      0     0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "        TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0         7.0             0                   0             0            0\n",
      "1         1.0             0                   0             0            0\n",
      "2         0.0             0                   0             0            0\n",
      "3         2.0             0                   0             0            0\n",
      "4         0.0             0                   0             0            0\n",
      "...       ...           ...                 ...           ...          ...\n",
      "152613    3.0             0                   0             0            0\n",
      "152614    4.0             0                   0             0            0\n",
      "152615    0.0             0                   0             0            0\n",
      "152616    4.0             0                   0             0            0\n",
      "152617    0.0             0                   0             0            0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "0.8466104915540762\n",
      "[0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 0 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 0 1 1 1 1 1 1 1 1 0 1 1 0 0 0 0 0 0 0 0 0 1 0 0 1 1 1 0 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 0 1 1 1 1 1 1 1 1 1 0 0 1 1 1 1 1 1 0 1 0 0 0 0 1 0 0 0 1 0 1\n",
      " 0 1 1 1 1 1 1 1 1 1 1 1 1 0 1 0 0 0 0 0 1 0 1 1 1 1 0 1 1 1 1 1 1 1 1 1 0\n",
      " 1 0 1 1 1 1 1 0 1 1 1 0 1 0 1 1 0 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 1 1 1 1 1 0 0 1 0 1 0 1 1 1 1 1 0 1\n",
      " 1 1 1 1 1 1 0 0 1 0 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 1 0 1 1 1 0 1 0 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 0 0 1 1 1 1 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 1 0 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 0 1 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1\n",
      " 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 0 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1]\n",
      "threshold density 1.30546\n",
      "cluster signal 292587.0\n",
      "cluster target count 175877\n",
      "!!cluster density 1.6635887580525026\n",
      "cluster signal 164324.0\n",
      "cluster target count 524123\n",
      "!!cluster density 0.31352182598359546\n",
      "\n",
      "Labels mapping\n",
      " {0: 1, 1: 0}\n",
      "DEBUG: 129208 23410 \n",
      "\n",
      " +----------------------+-------------+----------------+-------+\n",
      "| bin_offset_85        | TT_S0       | target_count   | +9    |\n",
      "| (category)           | (float64)   | (int64)        | ...   |\n",
      "|----------------------+-------------+----------------+-------|\n",
      "| (3000103, 3000203]   | 1.0         | 4              | ...   |\n",
      "| (3000203, 3000303]   | 0.0         | 1              | ...   |\n",
      "| (3000303, 3000403]   | 2.0         | 7              | ...   |\n",
      "| (3000403, 3000503]   | 0.0         | 1              | ...   |\n",
      "| ...                  | ...         | ...            | ...   |\n",
      "| (18261403, 18261503] | 4.0         | 4              | ...   |\n",
      "| (18261503, 18261603] | 0.0         | 1              | ...   |\n",
      "| (18261603, 18261703] | 4.0         | 4              | ...   |\n",
      "| (18261703, 18261710] | 0.0         | 1              | ...   |\n",
      "+----------------------+-------------+----------------+-------+\n",
      "Unstranded PyRanges object has 129,208 rows and 12 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "9 hidden columns: starting_nt, Chromosome, Start, End, genes, transcripts, ... (+ 3 more.)\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0\n",
      "BINS  IntervalIndex([  (3000018, 3000108],   (3000108, 3000208],\n",
      "                 (3000208, 3000308],   (3000308, 3000408],\n",
      "                 (3000408, 3000508],   (3000508, 3000608],\n",
      "                 (3000608, 3000708],   (3000708, 3000808],\n",
      "                 (3000808, 3000908],   (3000908, 3001008],\n",
      "               ...\n",
      "               (18260808, 18260908], (18260908, 18261008],\n",
      "               (18261008, 18261108], (18261108, 18261208],\n",
      "               (18261208, 18261308], (18261308, 18261408],\n",
      "               (18261408, 18261508], (18261508, 18261608],\n",
      "               (18261608, 18261708], (18261708, 18261710]],\n",
      "              dtype='interval[int64, right]', length=152618)\n",
      "HEAD with BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0   \n",
      "\n",
      "        bin_offset_90  target_count  \n",
      "0  (3000018, 3000108]             1  \n",
      "1  (3000018, 3000108]             1  \n",
      "2  (3000018, 3000108]             1  \n",
      "3  (3000018, 3000108]             1  \n",
      "4  (3000018, 3000108]             1  \n",
      "TAIL with BINS\n",
      "       seqnames     start       end        CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  18261611  18261611  18_18261611    0.0    0.0    0.0   \n",
      "699996       18  18261615  18261615  18_18261615    0.0    0.0    0.0   \n",
      "699997       18  18261682  18261682  18_18261682    0.0    0.0    0.0   \n",
      "699998       18  18261702  18261702  18_18261702    4.0    0.0    0.0   \n",
      "699999       18  18261710  18261710  18_18261710    0.0    0.0    0.0   \n",
      "\n",
      "               bin_offset_90  target_count  \n",
      "699995  (18261608, 18261708]             1  \n",
      "699996  (18261608, 18261708]             1  \n",
      "699997  (18261608, 18261708]             1  \n",
      "699998  (18261608, 18261708]             1  \n",
      "699999  (18261708, 18261710]             1  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:137: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "         bin_offset_90  TT_S0  target_count starting_nt\n",
      "0   (3000018, 3000108]    7.0            11     3000018\n",
      "1   (3000108, 3000208]    1.0             4     3000108\n",
      "2   (3000208, 3000308]    0.0             1     3000208\n",
      "3   (3000308, 3000408]    2.0             7     3000308\n",
      "4   (3000408, 3000508]    0.0             1     3000408\n",
      "5   (3000508, 3000608]    0.0             2     3000508\n",
      "6   (3000608, 3000708]    0.0             5     3000608\n",
      "7   (3000708, 3000808]    0.0             6     3000708\n",
      "8   (3000808, 3000908]    0.0             4     3000808\n",
      "9   (3000908, 3001008]    0.0             3     3000908\n",
      "10  (3001008, 3001108]    2.0             6     3001008\n",
      "11  (3001108, 3001208]    0.0             2     3001108\n",
      "12  (3001208, 3001308]    0.0             3     3001208\n",
      "13  (3001308, 3001408]    0.0             4     3001308\n",
      "14  (3001408, 3001508]    1.0             4     3001408\n",
      "15  (3001508, 3001608]    3.0             9     3001508\n",
      "16  (3001608, 3001708]    3.0             6     3001608\n",
      "17  (3001708, 3001808]    0.0             8     3001708\n",
      "18  (3001808, 3001908]    0.0             2     3001808\n",
      "19  (3001908, 3002008]    1.0             3     3001908\n",
      "20  (3002008, 3002108]    0.0             2     3002008\n",
      "21  (3002108, 3002208]    0.0             2     3002108\n",
      "22  (3002208, 3002308]   11.0             7     3002208\n",
      "23  (3002308, 3002408]    2.0             2     3002308\n",
      "24  (3002408, 3002508]   19.0             5     3002408\n",
      "25  (3002508, 3002608]    4.0             2     3002508\n",
      "26  (3002608, 3002708]    0.0             3     3002608\n",
      "27  (3002708, 3002808]    0.0             1     3002708\n",
      "28  (3002808, 3002908]    0.0             2     3002808\n",
      "29  (3002908, 3003008]    1.0             5     3002908\n",
      "30  (3003008, 3003108]    2.0             5     3003008\n",
      "31  (3003108, 3003208]    0.0             0     3003108\n",
      "32  (3003208, 3003308]    0.0             1     3003208\n",
      "33  (3003308, 3003408]    1.0             8     3003308\n",
      "34  (3003408, 3003508]    0.0             3     3003408\n",
      "35  (3003508, 3003608]    1.0             2     3003508\n",
      "36  (3003608, 3003708]    1.0             3     3003608\n",
      "37  (3003708, 3003808]    3.0             6     3003708\n",
      "38  (3003808, 3003908]    0.0             3     3003808\n",
      "39  (3003908, 3004008]    8.0             7     3003908\n",
      "40  (3004008, 3004108]    0.0             0     3004008\n",
      "41  (3004108, 3004208]    0.0             1     3004108\n",
      "42  (3004208, 3004308]    5.0             6     3004208\n",
      "43  (3004308, 3004408]    0.0             1     3004308\n",
      "44  (3004408, 3004508]    0.0             1     3004408\n",
      "45  (3004508, 3004608]    1.0             7     3004508\n",
      "46  (3004608, 3004708]    0.0             0     3004608\n",
      "47  (3004708, 3004808]    0.0             1     3004708\n",
      "48  (3004808, 3004908]   10.0             6     3004808\n",
      "49  (3004908, 3005008]    3.0             2     3004908\n",
      "(152618, 4)\n",
      "bin_offset_90    category\n",
      "TT_S0             float64\n",
      "target_count        int64\n",
      "starting_nt      category\n",
      "dtype: object\n",
      "sumos ilgis: 152618\n",
      "ranges ilgis: 152618\n",
      "Model input - NO SCALE\n",
      "        TT_S0  genes  transcripts  exons  CDSs\n",
      "0         7.0      0            0      0     0\n",
      "1         1.0      0            0      0     0\n",
      "2         0.0      0            0      0     0\n",
      "3         2.0      0            0      0     0\n",
      "4         0.0      0            0      0     0\n",
      "...       ...    ...          ...    ...   ...\n",
      "152613    3.0      0            0      0     0\n",
      "152614    4.0      0            0      0     0\n",
      "152615    0.0      0            0      0     0\n",
      "152616    4.0      0            0      0     0\n",
      "152617    0.0      0            0      0     0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "        TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0         7.0             0                   0             0            0\n",
      "1         1.0             0                   0             0            0\n",
      "2         0.0             0                   0             0            0\n",
      "3         2.0             0                   0             0            0\n",
      "4         0.0             0                   0             0            0\n",
      "...       ...           ...                 ...           ...          ...\n",
      "152613    3.0             0                   0             0            0\n",
      "152614    4.0             0                   0             0            0\n",
      "152615    0.0             0                   0             0            0\n",
      "152616    4.0             0                   0             0            0\n",
      "152617    0.0             0                   0             0            0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "0.1261515679670812\n",
      "[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 1 0 0 0 0 0 0 0 0 1 0 0 1 1 0 1 1 1 1 1 1 0 1 1 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 0 0 1 1 1 1 1 1 1 0 1 1 0 1 0\n",
      " 1 0 0 0 0 0 0 0 0 0 0 0 0 1 0 1 1 1 1 1 0 1 0 0 0 0 1 0 0 0 0 0 0 0 0 0 1\n",
      " 0 1 0 0 0 0 0 1 0 0 0 1 0 1 0 0 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 1 1 0 1 0 1 0 0 0 0 0 1 0\n",
      " 0 0 0 0 0 0 1 1 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 1 0 0 0 0 0 1 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 1 1 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0]\n",
      "threshold density 1.30546\n",
      "cluster signal 193427.0\n",
      "cluster target count 550844\n",
      "!!cluster density 0.3511466041202228\n",
      "cluster signal 263484.0\n",
      "cluster target count 149156\n",
      "!!cluster density 1.7664995038751374\n",
      "\n",
      "Labels mapping\n",
      " {0: 0, 1: 1}\n",
      "DEBUG: 133365 19253 \n",
      "\n",
      " +----------------------+-------------+----------------+-------+\n",
      "| bin_offset_90        | TT_S0       | target_count   | +9    |\n",
      "| (category)           | (float64)   | (int64)        | ...   |\n",
      "|----------------------+-------------+----------------+-------|\n",
      "| (3000018, 3000108]   | 7.0         | 11             | ...   |\n",
      "| (3000108, 3000208]   | 1.0         | 4              | ...   |\n",
      "| (3000208, 3000308]   | 0.0         | 1              | ...   |\n",
      "| (3000308, 3000408]   | 2.0         | 7              | ...   |\n",
      "| ...                  | ...         | ...            | ...   |\n",
      "| (18261408, 18261508] | 4.0         | 4              | ...   |\n",
      "| (18261508, 18261608] | 0.0         | 1              | ...   |\n",
      "| (18261608, 18261708] | 4.0         | 4              | ...   |\n",
      "| (18261708, 18261710] | 0.0         | 1              | ...   |\n",
      "+----------------------+-------------+----------------+-------+\n",
      "Unstranded PyRanges object has 133,365 rows and 12 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "9 hidden columns: starting_nt, Chromosome, Start, End, genes, transcripts, ... (+ 3 more.)\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0\n",
      "BINS  IntervalIndex([  (3000018, 3000113],   (3000113, 3000213],\n",
      "                 (3000213, 3000313],   (3000313, 3000413],\n",
      "                 (3000413, 3000513],   (3000513, 3000613],\n",
      "                 (3000613, 3000713],   (3000713, 3000813],\n",
      "                 (3000813, 3000913],   (3000913, 3001013],\n",
      "               ...\n",
      "               (18260713, 18260813], (18260813, 18260913],\n",
      "               (18260913, 18261013], (18261013, 18261113],\n",
      "               (18261113, 18261213], (18261213, 18261313],\n",
      "               (18261313, 18261413], (18261413, 18261513],\n",
      "               (18261513, 18261613], (18261613, 18261710]],\n",
      "              dtype='interval[int64, right]', length=152617)\n",
      "HEAD with BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0   \n",
      "\n",
      "        bin_offset_95  target_count  \n",
      "0  (3000018, 3000113]             1  \n",
      "1  (3000018, 3000113]             1  \n",
      "2  (3000018, 3000113]             1  \n",
      "3  (3000018, 3000113]             1  \n",
      "4  (3000018, 3000113]             1  \n",
      "TAIL with BINS\n",
      "       seqnames     start       end        CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  18261611  18261611  18_18261611    0.0    0.0    0.0   \n",
      "699996       18  18261615  18261615  18_18261615    0.0    0.0    0.0   \n",
      "699997       18  18261682  18261682  18_18261682    0.0    0.0    0.0   \n",
      "699998       18  18261702  18261702  18_18261702    4.0    0.0    0.0   \n",
      "699999       18  18261710  18261710  18_18261710    0.0    0.0    0.0   \n",
      "\n",
      "               bin_offset_95  target_count  \n",
      "699995  (18261513, 18261613]             1  \n",
      "699996  (18261613, 18261710]             1  \n",
      "699997  (18261613, 18261710]             1  \n",
      "699998  (18261613, 18261710]             1  \n",
      "699999  (18261613, 18261710]             1  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:137: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "         bin_offset_95  TT_S0  target_count starting_nt\n",
      "0   (3000018, 3000113]    7.0            11     3000018\n",
      "1   (3000113, 3000213]    1.0             4     3000113\n",
      "2   (3000213, 3000313]    2.0             2     3000213\n",
      "3   (3000313, 3000413]    0.0             6     3000313\n",
      "4   (3000413, 3000513]    0.0             1     3000413\n",
      "5   (3000513, 3000613]    0.0             3     3000513\n",
      "6   (3000613, 3000713]    0.0             4     3000613\n",
      "7   (3000713, 3000813]    0.0             6     3000713\n",
      "8   (3000813, 3000913]    0.0             4     3000813\n",
      "9   (3000913, 3001013]    0.0             3     3000913\n",
      "10  (3001013, 3001113]    2.0             6     3001013\n",
      "11  (3001113, 3001213]    0.0             2     3001113\n",
      "12  (3001213, 3001313]    0.0             3     3001213\n",
      "13  (3001313, 3001413]    0.0             4     3001313\n",
      "14  (3001413, 3001513]    1.0             4     3001413\n",
      "15  (3001513, 3001613]    3.0            10     3001513\n",
      "16  (3001613, 3001713]    3.0             6     3001613\n",
      "17  (3001713, 3001813]    0.0             7     3001713\n",
      "18  (3001813, 3001913]    0.0             2     3001813\n",
      "19  (3001913, 3002013]    1.0             3     3001913\n",
      "20  (3002013, 3002113]    0.0             2     3002013\n",
      "21  (3002113, 3002213]    1.0             3     3002113\n",
      "22  (3002213, 3002313]   10.0             7     3002213\n",
      "23  (3002313, 3002413]    2.0             1     3002313\n",
      "24  (3002413, 3002513]   19.0             5     3002413\n",
      "25  (3002513, 3002613]    4.0             2     3002513\n",
      "26  (3002613, 3002713]    0.0             3     3002613\n",
      "27  (3002713, 3002813]    0.0             1     3002713\n",
      "28  (3002813, 3002913]    0.0             2     3002813\n",
      "29  (3002913, 3003013]    1.0             6     3002913\n",
      "30  (3003013, 3003113]    2.0             4     3003013\n",
      "31  (3003113, 3003213]    0.0             0     3003113\n",
      "32  (3003213, 3003313]    0.0             1     3003213\n",
      "33  (3003313, 3003413]    1.0             8     3003313\n",
      "34  (3003413, 3003513]    0.0             3     3003413\n",
      "35  (3003513, 3003613]    1.0             2     3003513\n",
      "36  (3003613, 3003713]    1.0             3     3003613\n",
      "37  (3003713, 3003813]    3.0             6     3003713\n",
      "38  (3003813, 3003913]    0.0             4     3003813\n",
      "39  (3003913, 3004013]    8.0             6     3003913\n",
      "40  (3004013, 3004113]    0.0             0     3004013\n",
      "41  (3004113, 3004213]    0.0             1     3004113\n",
      "42  (3004213, 3004313]    5.0             6     3004213\n",
      "43  (3004313, 3004413]    0.0             1     3004313\n",
      "44  (3004413, 3004513]    0.0             1     3004413\n",
      "45  (3004513, 3004613]    1.0             7     3004513\n",
      "46  (3004613, 3004713]    0.0             0     3004613\n",
      "47  (3004713, 3004813]    0.0             1     3004713\n",
      "48  (3004813, 3004913]   10.0             7     3004813\n",
      "49  (3004913, 3005013]    3.0             1     3004913\n",
      "(152617, 4)\n",
      "bin_offset_95    category\n",
      "TT_S0             float64\n",
      "target_count        int64\n",
      "starting_nt      category\n",
      "dtype: object\n",
      "sumos ilgis: 152617\n",
      "ranges ilgis: 152617\n",
      "Model input - NO SCALE\n",
      "        TT_S0  genes  transcripts  exons  CDSs\n",
      "0         7.0      0            0      0     0\n",
      "1         1.0      0            0      0     0\n",
      "2         2.0      0            0      0     0\n",
      "3         0.0      0            0      0     0\n",
      "4         0.0      0            0      0     0\n",
      "...       ...    ...          ...    ...   ...\n",
      "152612   10.0      0            0      0     0\n",
      "152613    3.0      0            0      0     0\n",
      "152614    4.0      0            0      0     0\n",
      "152615    0.0      0            0      0     0\n",
      "152616    4.0      0            0      0     0\n",
      "\n",
      "[152617 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "        TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0         7.0             0                   0             0            0\n",
      "1         1.0             0                   0             0            0\n",
      "2         2.0             0                   0             0            0\n",
      "3         0.0             0                   0             0            0\n",
      "4         0.0             0                   0             0            0\n",
      "...       ...           ...                 ...           ...          ...\n",
      "152612   10.0             0                   0             0            0\n",
      "152613    3.0             0                   0             0            0\n",
      "152614    4.0             0                   0             0            0\n",
      "152615    0.0             0                   0             0            0\n",
      "152616    4.0             0                   0             0            0\n",
      "\n",
      "[152617 rows x 5 columns]\n",
      "0.126099975756305\n",
      "[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 1 0 0 0 0 0 0 0 0 1 0 0 1 1 0 1 1 1 1 1 1 0 1 1 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 0 0 1 1 1 1 1 1 1 0 1 1 0 1 0\n",
      " 1 0 0 0 0 0 0 0 0 0 0 0 0 1 0 1 1 1 1 0 0 1 0 0 0 1 0 0 0 0 0 0 0 0 0 0 1\n",
      " 0 1 0 0 0 0 0 1 0 0 0 1 0 1 0 0 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 1 1 0 1 1 1 0 0 0 0 0 1 0\n",
      " 0 0 0 0 0 0 1 1 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 1 0 0 0 0 0 1 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 1 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0]\n",
      "threshold density 1.30546\n",
      "cluster signal 193221.0\n",
      "cluster target count 550863\n",
      "!!cluster density 0.35076053392585815\n",
      "cluster signal 263690.0\n",
      "cluster target count 149137\n",
      "!!cluster density 1.7681058355740025\n",
      "\n",
      "Labels mapping\n",
      " {0: 0, 1: 1}\n",
      "DEBUG: 133372 19245 \n",
      "\n",
      " +----------------------+-------------+----------------+-------+\n",
      "| bin_offset_95        | TT_S0       | target_count   | +9    |\n",
      "| (category)           | (float64)   | (int64)        | ...   |\n",
      "|----------------------+-------------+----------------+-------|\n",
      "| (3000018, 3000113]   | 7.0         | 11             | ...   |\n",
      "| (3000113, 3000213]   | 1.0         | 4              | ...   |\n",
      "| (3000213, 3000313]   | 2.0         | 2              | ...   |\n",
      "| (3000313, 3000413]   | 0.0         | 6              | ...   |\n",
      "| ...                  | ...         | ...            | ...   |\n",
      "| (18261313, 18261413] | 3.0         | 15             | ...   |\n",
      "| (18261413, 18261513] | 4.0         | 4              | ...   |\n",
      "| (18261513, 18261613] | 0.0         | 2              | ...   |\n",
      "| (18261613, 18261710] | 4.0         | 4              | ...   |\n",
      "+----------------------+-------------+----------------+-------+\n",
      "Unstranded PyRanges object has 133,372 rows and 12 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "9 hidden columns: starting_nt, Chromosome, Start, End, genes, transcripts, ... (+ 3 more.)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/anaconda3/lib/python3.12/site-packages/pyranges/methods/intersection.py:95: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '[1 1 1 ... 1 1 1]' has dtype incompatible with int32, please explicitly cast to a compatible dtype first.\n",
      "  sx.loc[vc.index, 0] = vc.values\n",
      "/opt/anaconda3/lib/python3.12/site-packages/pyranges/methods/intersection.py:95: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '[1 1 1 ... 1 1 1]' has dtype incompatible with int32, please explicitly cast to a compatible dtype first.\n",
      "  sx.loc[vc.index, 0] = vc.values\n",
      "/opt/anaconda3/lib/python3.12/site-packages/pyranges/methods/intersection.py:95: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '[1 1 1 ... 1 1 1]' has dtype incompatible with int32, please explicitly cast to a compatible dtype first.\n",
      "  sx.loc[vc.index, 0] = vc.values\n",
      "/opt/anaconda3/lib/python3.12/site-packages/pyranges/methods/intersection.py:95: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '[1 1 1 ... 1 1 1]' has dtype incompatible with int32, please explicitly cast to a compatible dtype first.\n",
      "  sx.loc[vc.index, 0] = vc.values\n",
      "/opt/anaconda3/lib/python3.12/site-packages/pyranges/methods/intersection.py:95: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '[1 1 1 ... 1 1 1]' has dtype incompatible with int32, please explicitly cast to a compatible dtype first.\n",
      "  sx.loc[vc.index, 0] = vc.values\n",
      "/opt/anaconda3/lib/python3.12/site-packages/pyranges/methods/intersection.py:95: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '[1 1 1 ... 1 1 1]' has dtype incompatible with int32, please explicitly cast to a compatible dtype first.\n",
      "  sx.loc[vc.index, 0] = vc.values\n",
      "/opt/anaconda3/lib/python3.12/site-packages/pyranges/methods/intersection.py:95: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '[1 1 1 ... 1 1 1]' has dtype incompatible with int32, please explicitly cast to a compatible dtype first.\n",
      "  sx.loc[vc.index, 0] = vc.values\n",
      "/opt/anaconda3/lib/python3.12/site-packages/pyranges/methods/intersection.py:95: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '[1 1 1 ... 1 1 1]' has dtype incompatible with int32, please explicitly cast to a compatible dtype first.\n",
      "  sx.loc[vc.index, 0] = vc.values\n",
      "/opt/anaconda3/lib/python3.12/site-packages/pyranges/methods/intersection.py:95: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '[1 1 1 ... 1 1 1]' has dtype incompatible with int32, please explicitly cast to a compatible dtype first.\n",
      "  sx.loc[vc.index, 0] = vc.values\n",
      "/opt/anaconda3/lib/python3.12/site-packages/pyranges/methods/intersection.py:95: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '[1 1 1 ... 1 1 1]' has dtype incompatible with int32, please explicitly cast to a compatible dtype first.\n",
      "  sx.loc[vc.index, 0] = vc.values\n",
      "/opt/anaconda3/lib/python3.12/site-packages/pyranges/methods/intersection.py:95: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '[1 1 1 ... 1 1 1]' has dtype incompatible with int32, please explicitly cast to a compatible dtype first.\n",
      "  sx.loc[vc.index, 0] = vc.values\n",
      "/opt/anaconda3/lib/python3.12/site-packages/pyranges/methods/intersection.py:95: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '[1 1 1 ... 1 1 1]' has dtype incompatible with int32, please explicitly cast to a compatible dtype first.\n",
      "  sx.loc[vc.index, 0] = vc.values\n",
      "/opt/anaconda3/lib/python3.12/site-packages/pyranges/methods/intersection.py:95: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '[1 1 1 ... 1 1 1]' has dtype incompatible with int32, please explicitly cast to a compatible dtype first.\n",
      "  sx.loc[vc.index, 0] = vc.values\n",
      "/opt/anaconda3/lib/python3.12/site-packages/pyranges/methods/intersection.py:95: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '[1 1 1 ... 1 1 1]' has dtype incompatible with int32, please explicitly cast to a compatible dtype first.\n",
      "  sx.loc[vc.index, 0] = vc.values\n",
      "/opt/anaconda3/lib/python3.12/site-packages/pyranges/methods/intersection.py:95: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '[1 1 1 1 1 1 1]' has dtype incompatible with int32, please explicitly cast to a compatible dtype first.\n",
      "  sx.loc[vc.index, 0] = vc.values\n",
      "/opt/anaconda3/lib/python3.12/site-packages/pyranges/methods/intersection.py:95: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '[1 1 1 ... 1 1 1]' has dtype incompatible with int32, please explicitly cast to a compatible dtype first.\n",
      "  sx.loc[vc.index, 0] = vc.values\n",
      "/opt/anaconda3/lib/python3.12/site-packages/pyranges/methods/intersection.py:95: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '[1 1 1 ... 1 1 1]' has dtype incompatible with int32, please explicitly cast to a compatible dtype first.\n",
      "  sx.loc[vc.index, 0] = vc.values\n",
      "/opt/anaconda3/lib/python3.12/site-packages/pyranges/methods/intersection.py:95: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '[1 1 1 ... 1 1 1]' has dtype incompatible with int32, please explicitly cast to a compatible dtype first.\n",
      "  sx.loc[vc.index, 0] = vc.values\n",
      "/opt/anaconda3/lib/python3.12/site-packages/pyranges/methods/intersection.py:95: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '[1 1 1 ... 1 1 1]' has dtype incompatible with int32, please explicitly cast to a compatible dtype first.\n",
      "  sx.loc[vc.index, 0] = vc.values\n",
      "/opt/anaconda3/lib/python3.12/site-packages/pyranges/methods/intersection.py:95: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '[1 1 1 ... 1 1 1]' has dtype incompatible with int32, please explicitly cast to a compatible dtype first.\n",
      "  sx.loc[vc.index, 0] = vc.values\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "OVERLAPS +--------------+-----------+-----------+------------+------------+-------+\n",
      "| Chromosome   | Start     | End       | offset_0   | offset_5   | +18   |\n",
      "| (object)     | (int64)   | (int64)   | (int64)    | (int64)    | ...   |\n",
      "|--------------+-----------+-----------+------------+------------+-------|\n",
      "| 18           | 3000018   | 3000023   | 0          | 0          | ...   |\n",
      "| 18           | 3000023   | 3000028   | 0          | 1          | ...   |\n",
      "| 18           | 3000028   | 3000038   | 0          | 1          | ...   |\n",
      "| 18           | 3000038   | 3000093   | 0          | 1          | ...   |\n",
      "| ...          | ...       | ...       | ...        | ...        | ...   |\n",
      "| 18           | 18261353  | 18261358  | 0          | 0          | ...   |\n",
      "| 18           | 18261358  | 18261363  | 0          | 0          | ...   |\n",
      "| 18           | 18261363  | 18261368  | 0          | 0          | ...   |\n",
      "| 18           | 18261368  | 18261373  | 0          | 0          | ...   |\n",
      "+--------------+-----------+-----------+------------+------------+-------+\n",
      "Unstranded PyRanges object has 534,677 rows and 23 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "18 hidden columns: offset_10, offset_15, offset_20, offset_25, offset_30, offset_35, ... (+ 12 more.)\n",
      "OVERLAPS FILTERED +--------------+-----------+-----------+\n",
      "| Chromosome   | Start     | End       |\n",
      "| (category)   | (int64)   | (int64)   |\n",
      "|--------------+-----------+-----------|\n",
      "| 18           | 3002263   | 3002503   |\n",
      "| 18           | 3003953   | 3003983   |\n",
      "| 18           | 3004808   | 3004898   |\n",
      "| 18           | 3005068   | 3005348   |\n",
      "| ...          | ...       | ...       |\n",
      "| 18           | 18247898  | 18247983  |\n",
      "| 18           | 18249543  | 18249633  |\n",
      "| 18           | 18250953  | 18251038  |\n",
      "| 18           | 18261233  | 18261323  |\n",
      "+--------------+-----------+-----------+\n",
      "Unstranded PyRanges object has 14,519 rows and 3 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n"
     ]
    }
   ],
   "source": [
    "result = evaluator.evaluate(merged_ranges = chromatin_algos.algo5d_aggregated(16, 10, 3), model_name=\"test run\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "aa3c6a9a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "count                        13302.0\n",
      "mean                      128.670501\n",
      "std                       113.512934\n",
      "min                              5.0\n",
      "25%                             80.0\n",
      "50%                             95.0\n",
      "75%                            140.0\n",
      "max                           1985.0\n",
      "mode                            95.0\n",
      "ATACseq_overlap             0.791289\n",
      "ATACseq_ranges_overlap      0.034008\n",
      "ATACseq_Jaccard             0.033706\n",
      "DNAseq_overlap              0.699477\n",
      "DNAseq_ranges_overlap       0.049429\n",
      "DNAseq_Jaccard              0.048401\n",
      "genes_overlap               0.596375\n",
      "percentage_open             0.093725\n",
      "percentage_open_adj         0.112148\n",
      "model_name                   k-means\n",
      "Name: Length, dtype: object\n"
     ]
    }
   ],
   "source": [
    "print(evaluation)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "92562b46",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "count                        14519.0\n",
      "mean                      124.223431\n",
      "std                       114.045787\n",
      "min                              5.0\n",
      "25%                             70.0\n",
      "50%                             90.0\n",
      "75%                            135.0\n",
      "max                           1980.0\n",
      "mode                            90.0\n",
      "ATACseq_overlap             0.793681\n",
      "ATACseq_ranges_overlap      0.032371\n",
      "ATACseq_Jaccard             0.032101\n",
      "DNAseq_overlap              0.707587\n",
      "DNAseq_ranges_overlap       0.047451\n",
      "DNAseq_Jaccard              0.046538\n",
      "genes_overlap                0.59436\n",
      "percentage_open             0.098764\n",
      "percentage_open_adj         0.118178\n",
      "model_name                   k-means\n",
      "Name: Length, dtype: object\n"
     ]
    }
   ],
   "source": [
    "print(evaluation)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "4328129e",
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "File \u001b[0;32m/opt/anaconda3/lib/python3.12/site-packages/pyreadr/librdata.pyx:229\u001b[0m, in \u001b[0;36mpyreadr.librdata.Parser._Parser__handle_column\u001b[0;34m()\u001b[0m\n",
      "File \u001b[0;32m/opt/anaconda3/lib/python3.12/enum.py:726\u001b[0m, in \u001b[0;36mEnumType.__call__\u001b[0;34m(cls, value, names, module, qualname, type, start, boundary, *values)\u001b[0m\n\u001b[1;32m    721\u001b[0m \u001b[38;5;250m    \u001b[39m\u001b[38;5;124;03m\"\"\"\u001b[39;00m\n\u001b[1;32m    722\u001b[0m \u001b[38;5;124;03m    classes/types should always be True.\u001b[39;00m\n\u001b[1;32m    723\u001b[0m \u001b[38;5;124;03m    \"\"\"\u001b[39;00m\n\u001b[1;32m    724\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;01mTrue\u001b[39;00m\n\u001b[0;32m--> 726\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21m__call__\u001b[39m(\u001b[38;5;28mcls\u001b[39m, value, names\u001b[38;5;241m=\u001b[39m_not_given, \u001b[38;5;241m*\u001b[39mvalues, module\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m, qualname\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m, \u001b[38;5;28mtype\u001b[39m\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m, start\u001b[38;5;241m=\u001b[39m\u001b[38;5;241m1\u001b[39m, boundary\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m):\n\u001b[1;32m    727\u001b[0m \u001b[38;5;250m    \u001b[39m\u001b[38;5;124;03m\"\"\"\u001b[39;00m\n\u001b[1;32m    728\u001b[0m \u001b[38;5;124;03m    Either returns an existing member, or creates a new enum class.\u001b[39;00m\n\u001b[1;32m    729\u001b[0m \n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m    751\u001b[0m \u001b[38;5;124;03m    `type`, if set, will be mixed in as the first base class.\u001b[39;00m\n\u001b[1;32m    752\u001b[0m \u001b[38;5;124;03m    \"\"\"\u001b[39;00m\n\u001b[1;32m    753\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mcls\u001b[39m\u001b[38;5;241m.\u001b[39m_member_map_:\n\u001b[1;32m    754\u001b[0m         \u001b[38;5;66;03m# simple value lookup if members exist\u001b[39;00m\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Exception ignored in: 'pyreadr.librdata._handle_column'\n",
      "Traceback (most recent call last):\n",
      "  File \"pyreadr/librdata.pyx\", line 229, in pyreadr.librdata.Parser._Parser__handle_column\n",
      "  File \"/opt/anaconda3/lib/python3.12/enum.py\", line 726, in __call__\n",
      "    def __call__(cls, value, names=_not_given, *values, module=None, qualname=None, type=None, start=1, boundary=None):\n",
      "\n",
      "KeyboardInterrupt: \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0\n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0\n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0\n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0\n",
      "4       18  3000040  3000040  18_3000040    0.0    0.0\n",
      "BINS  IntervalIndex([(3000016, 3000016], (3000016, 3000116], (3000116, 3000216],\n",
      "               (3000216, 3000316], (3000316, 3000416], (3000416, 3000516],\n",
      "               (3000516, 3000616], (3000616, 3000716], (3000716, 3000816],\n",
      "               (3000816, 3000916],\n",
      "               ...\n",
      "               (9716616, 9716716], (9716716, 9716816], (9716816, 9716916],\n",
      "               (9716916, 9717016], (9717016, 9717116], (9717116, 9717216],\n",
      "               (9717216, 9717316], (9717316, 9717416], (9717416, 9717516],\n",
      "               (9717516, 9717593]],\n",
      "              dtype='interval[int64, right]', length=67177)\n",
      "HEAD with BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1        bin_offset_0  \\\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0  (3000016, 3000116]   \n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0  (3000016, 3000116]   \n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0  (3000016, 3000116]   \n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0  (3000016, 3000116]   \n",
      "4       18  3000040  3000040  18_3000040    0.0    0.0  (3000016, 3000116]   \n",
      "\n",
      "   target_count  \n",
      "0             1  \n",
      "1             1  \n",
      "2             1  \n",
      "3             1  \n",
      "4             1  \n",
      "TAIL with BINS\n",
      "       seqnames    start      end       CG_ID  TT_S0  TT_S1  \\\n",
      "699995       18  9717555  9717555  18_9717555    0.0    0.0   \n",
      "699996       19  9717571  9717571  19_9717571    0.0    0.0   \n",
      "699997       19  9717581  9717581  19_9717581    0.0    0.0   \n",
      "699998       18  9717582  9717582  18_9717582    0.0    0.0   \n",
      "699999       18  9717593  9717593  18_9717593    0.0    0.0   \n",
      "\n",
      "              bin_offset_0  target_count  \n",
      "699995  (9717516, 9717593]             1  \n",
      "699996  (9717516, 9717593]             1  \n",
      "699997  (9717516, 9717593]             1  \n",
      "699998  (9717516, 9717593]             1  \n",
      "699999  (9717516, 9717593]             1  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:137: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "ename": "KeyError",
     "evalue": "\"['TT_S2'] not found in axis\"",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[7], line 4\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[38;5;28;01mimport\u001b[39;00m \u001b[38;5;21;01mos\u001b[39;00m\n\u001b[1;32m      2\u001b[0m os\u001b[38;5;241m.\u001b[39menviron[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mCOLUMNS\u001b[39m\u001b[38;5;124m'\u001b[39m]\u001b[38;5;241m=\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m300\u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[0;32m----> 4\u001b[0m result \u001b[38;5;241m=\u001b[39m evaluator\u001b[38;5;241m.\u001b[39mevaluate(merged_ranges \u001b[38;5;241m=\u001b[39m chromatin_algos\u001b[38;5;241m.\u001b[39malgo5d_aggregated(\u001b[38;5;241m1\u001b[39m, n_clusters\u001b[38;5;241m=\u001b[39m\u001b[38;5;241m10\u001b[39m))\n",
      "File \u001b[0;32m~/VU/bio-baigiamasis/scripts/chromatin_algos.py:387\u001b[0m, in \u001b[0;36malgo5d_aggregated\u001b[0;34m(scale, n_clusters, threshold, bin_offset)\u001b[0m\n\u001b[1;32m    386\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21malgo5d_aggregated\u001b[39m(scale \u001b[38;5;241m=\u001b[39m \u001b[38;5;241m1\u001b[39m, n_clusters \u001b[38;5;241m=\u001b[39m \u001b[38;5;241m2\u001b[39m, threshold \u001b[38;5;241m=\u001b[39m \u001b[38;5;241m1.5\u001b[39m, bin_offset \u001b[38;5;241m=\u001b[39m \u001b[38;5;241m0\u001b[39m):\n\u001b[0;32m--> 387\u001b[0m     model_output \u001b[38;5;241m=\u001b[39m algo5d(scale, n_clusters, bin_offset)\n\u001b[1;32m    388\u001b[0m     update_labels(model_output, threshold)\n\u001b[1;32m    389\u001b[0m     model_output\u001b[38;5;241m.\u001b[39mto_csv(\u001b[38;5;124m'\u001b[39m\u001b[38;5;124m../outputs/output_algo5d_aggregated.csv\u001b[39m\u001b[38;5;124m'\u001b[39m, sep \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m'\u001b[39m\u001b[38;5;130;01m\\t\u001b[39;00m\u001b[38;5;124m'\u001b[39m)\n",
      "File \u001b[0;32m~/VU/bio-baigiamasis/scripts/chromatin_algos.py:140\u001b[0m, in \u001b[0;36malgo5d\u001b[0;34m(scale, n_clusters, bin_offset)\u001b[0m\n\u001b[1;32m    137\u001b[0m sumos \u001b[38;5;241m=\u001b[39m df\u001b[38;5;241m.\u001b[39mgroupby(\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mbin_offset_\u001b[39m\u001b[38;5;124m'\u001b[39m \u001b[38;5;241m+\u001b[39m \u001b[38;5;28mstr\u001b[39m(bin_offset))\u001b[38;5;241m.\u001b[39msum()\u001b[38;5;241m.\u001b[39mreset_index()\n\u001b[1;32m    138\u001b[0m sumos[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mstarting_nt\u001b[39m\u001b[38;5;124m'\u001b[39m] \u001b[38;5;241m=\u001b[39m sumos[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mbin_offset_\u001b[39m\u001b[38;5;124m'\u001b[39m \u001b[38;5;241m+\u001b[39m \u001b[38;5;28mstr\u001b[39m(bin_offset)]\u001b[38;5;241m.\u001b[39mapply(\u001b[38;5;28;01mlambda\u001b[39;00m x: x\u001b[38;5;241m.\u001b[39mleft)\n\u001b[0;32m--> 140\u001b[0m sumos\u001b[38;5;241m.\u001b[39mdrop(columns\u001b[38;5;241m=\u001b[39mOTHER_SIGNALS, axis\u001b[38;5;241m=\u001b[39m\u001b[38;5;241m1\u001b[39m, inplace\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m)\n\u001b[1;32m    142\u001b[0m \u001b[38;5;28mprint\u001b[39m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mSUMOS\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[1;32m    143\u001b[0m \u001b[38;5;28mprint\u001b[39m(sumos[:\u001b[38;5;241m50\u001b[39m])\n",
      "File \u001b[0;32m/opt/anaconda3/lib/python3.12/site-packages/pandas/core/frame.py:5581\u001b[0m, in \u001b[0;36mDataFrame.drop\u001b[0;34m(self, labels, axis, index, columns, level, inplace, errors)\u001b[0m\n\u001b[1;32m   5433\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mdrop\u001b[39m(\n\u001b[1;32m   5434\u001b[0m     \u001b[38;5;28mself\u001b[39m,\n\u001b[1;32m   5435\u001b[0m     labels: IndexLabel \u001b[38;5;241m|\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m,\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m   5442\u001b[0m     errors: IgnoreRaise \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mraise\u001b[39m\u001b[38;5;124m\"\u001b[39m,\n\u001b[1;32m   5443\u001b[0m ) \u001b[38;5;241m-\u001b[39m\u001b[38;5;241m>\u001b[39m DataFrame \u001b[38;5;241m|\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[1;32m   5444\u001b[0m \u001b[38;5;250m    \u001b[39m\u001b[38;5;124;03m\"\"\"\u001b[39;00m\n\u001b[1;32m   5445\u001b[0m \u001b[38;5;124;03m    Drop specified labels from rows or columns.\u001b[39;00m\n\u001b[1;32m   5446\u001b[0m \n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m   5579\u001b[0m \u001b[38;5;124;03m            weight  1.0     0.8\u001b[39;00m\n\u001b[1;32m   5580\u001b[0m \u001b[38;5;124;03m    \"\"\"\u001b[39;00m\n\u001b[0;32m-> 5581\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28msuper\u001b[39m()\u001b[38;5;241m.\u001b[39mdrop(\n\u001b[1;32m   5582\u001b[0m         labels\u001b[38;5;241m=\u001b[39mlabels,\n\u001b[1;32m   5583\u001b[0m         axis\u001b[38;5;241m=\u001b[39maxis,\n\u001b[1;32m   5584\u001b[0m         index\u001b[38;5;241m=\u001b[39mindex,\n\u001b[1;32m   5585\u001b[0m         columns\u001b[38;5;241m=\u001b[39mcolumns,\n\u001b[1;32m   5586\u001b[0m         level\u001b[38;5;241m=\u001b[39mlevel,\n\u001b[1;32m   5587\u001b[0m         inplace\u001b[38;5;241m=\u001b[39minplace,\n\u001b[1;32m   5588\u001b[0m         errors\u001b[38;5;241m=\u001b[39merrors,\n\u001b[1;32m   5589\u001b[0m     )\n",
      "File \u001b[0;32m/opt/anaconda3/lib/python3.12/site-packages/pandas/core/generic.py:4788\u001b[0m, in \u001b[0;36mNDFrame.drop\u001b[0;34m(self, labels, axis, index, columns, level, inplace, errors)\u001b[0m\n\u001b[1;32m   4786\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m axis, labels \u001b[38;5;129;01min\u001b[39;00m axes\u001b[38;5;241m.\u001b[39mitems():\n\u001b[1;32m   4787\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m labels \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[0;32m-> 4788\u001b[0m         obj \u001b[38;5;241m=\u001b[39m obj\u001b[38;5;241m.\u001b[39m_drop_axis(labels, axis, level\u001b[38;5;241m=\u001b[39mlevel, errors\u001b[38;5;241m=\u001b[39merrors)\n\u001b[1;32m   4790\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m inplace:\n\u001b[1;32m   4791\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_update_inplace(obj)\n",
      "File \u001b[0;32m/opt/anaconda3/lib/python3.12/site-packages/pandas/core/generic.py:4830\u001b[0m, in \u001b[0;36mNDFrame._drop_axis\u001b[0;34m(self, labels, axis, level, errors, only_slice)\u001b[0m\n\u001b[1;32m   4828\u001b[0m         new_axis \u001b[38;5;241m=\u001b[39m axis\u001b[38;5;241m.\u001b[39mdrop(labels, level\u001b[38;5;241m=\u001b[39mlevel, errors\u001b[38;5;241m=\u001b[39merrors)\n\u001b[1;32m   4829\u001b[0m     \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[0;32m-> 4830\u001b[0m         new_axis \u001b[38;5;241m=\u001b[39m axis\u001b[38;5;241m.\u001b[39mdrop(labels, errors\u001b[38;5;241m=\u001b[39merrors)\n\u001b[1;32m   4831\u001b[0m     indexer \u001b[38;5;241m=\u001b[39m axis\u001b[38;5;241m.\u001b[39mget_indexer(new_axis)\n\u001b[1;32m   4833\u001b[0m \u001b[38;5;66;03m# Case for non-unique axis\u001b[39;00m\n\u001b[1;32m   4834\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n",
      "File \u001b[0;32m/opt/anaconda3/lib/python3.12/site-packages/pandas/core/indexes/base.py:7070\u001b[0m, in \u001b[0;36mIndex.drop\u001b[0;34m(self, labels, errors)\u001b[0m\n\u001b[1;32m   7068\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m mask\u001b[38;5;241m.\u001b[39many():\n\u001b[1;32m   7069\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m errors \u001b[38;5;241m!=\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mignore\u001b[39m\u001b[38;5;124m\"\u001b[39m:\n\u001b[0;32m-> 7070\u001b[0m         \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mKeyError\u001b[39;00m(\u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mlabels[mask]\u001b[38;5;241m.\u001b[39mtolist()\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m not found in axis\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[1;32m   7071\u001b[0m     indexer \u001b[38;5;241m=\u001b[39m indexer[\u001b[38;5;241m~\u001b[39mmask]\n\u001b[1;32m   7072\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mdelete(indexer)\n",
      "\u001b[0;31mKeyError\u001b[0m: \"['TT_S2'] not found in axis\""
     ]
    }
   ],
   "source": [
    "import os\n",
    "os.environ['COLUMNS']=\"300\"\n",
    "\n",
    "result = evaluator.evaluate(merged_ranges = chromatin_algos.algo5d_aggregated(1, n_clusters=10))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "6201148c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0\n",
      "BINS  IntervalIndex([  (3000018, 3000018],   (3000018, 3000118],\n",
      "                 (3000118, 3000218],   (3000218, 3000318],\n",
      "                 (3000318, 3000418],   (3000418, 3000518],\n",
      "                 (3000518, 3000618],   (3000618, 3000718],\n",
      "                 (3000718, 3000818],   (3000818, 3000918],\n",
      "               ...\n",
      "               (18260718, 18260818], (18260818, 18260918],\n",
      "               (18260918, 18261018], (18261018, 18261118],\n",
      "               (18261118, 18261218], (18261218, 18261318],\n",
      "               (18261318, 18261418], (18261418, 18261518],\n",
      "               (18261518, 18261618], (18261618, 18261710]],\n",
      "              dtype='interval[int64, right]', length=152618)\n",
      "HEAD with BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "1       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "2       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "3       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "4       18  3000066  3000066  18_3000066    0.0    0.0    0.0   \n",
      "\n",
      "         bin_offset_0  target_count  \n",
      "0  (3000018, 3000118]             1  \n",
      "1  (3000018, 3000118]             1  \n",
      "2  (3000018, 3000118]             1  \n",
      "3  (3000018, 3000118]             1  \n",
      "4  (3000018, 3000118]             1  \n",
      "TAIL with BINS\n",
      "       seqnames     start       end        CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  18261611  18261611  18_18261611    0.0    0.0    0.0   \n",
      "699996       18  18261615  18261615  18_18261615    0.0    0.0    0.0   \n",
      "699997       18  18261682  18261682  18_18261682    0.0    0.0    0.0   \n",
      "699998       18  18261702  18261702  18_18261702    4.0    0.0    0.0   \n",
      "699999       18  18261710  18261710  18_18261710    0.0    0.0    0.0   \n",
      "\n",
      "                bin_offset_0  target_count  \n",
      "699995  (18261518, 18261618]             1  \n",
      "699996  (18261518, 18261618]             1  \n",
      "699997  (18261618, 18261710]             1  \n",
      "699998  (18261618, 18261710]             1  \n",
      "699999  (18261618, 18261710]             1  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:137: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "          bin_offset_0  TT_S0  target_count  starting_nt\n",
      "0   (3000018, 3000018]    0.0             0      3000018\n",
      "1   (3000018, 3000118]    7.0            11      3000018\n",
      "2   (3000118, 3000218]    1.0             4      3000118\n",
      "3   (3000218, 3000318]    2.0             4      3000218\n",
      "4   (3000318, 3000418]    0.0             4      3000318\n",
      "5   (3000418, 3000518]    0.0             2      3000418\n",
      "6   (3000518, 3000618]    0.0             2      3000518\n",
      "7   (3000618, 3000718]    0.0             4      3000618\n",
      "8   (3000718, 3000818]    0.0             7      3000718\n",
      "9   (3000818, 3000918]    0.0             3      3000818\n",
      "10  (3000918, 3001018]    0.0             3      3000918\n",
      "11  (3001018, 3001118]    2.0             6      3001018\n",
      "12  (3001118, 3001218]    0.0             2      3001118\n",
      "13  (3001218, 3001318]    0.0             4      3001218\n",
      "14  (3001318, 3001418]    0.0             3      3001318\n",
      "15  (3001418, 3001518]    1.0             4      3001418\n",
      "16  (3001518, 3001618]    3.0            10      3001518\n",
      "17  (3001618, 3001718]    3.0             7      3001618\n",
      "18  (3001718, 3001818]    0.0             6      3001718\n",
      "19  (3001818, 3001918]    0.0             2      3001818\n",
      "20  (3001918, 3002018]    1.0             3      3001918\n",
      "21  (3002018, 3002118]    0.0             2      3002018\n",
      "22  (3002118, 3002218]    1.0             4      3002118\n",
      "23  (3002218, 3002318]   10.0             6      3002218\n",
      "24  (3002318, 3002418]    2.0             1      3002318\n",
      "25  (3002418, 3002518]   19.0             5      3002418\n",
      "26  (3002518, 3002618]    4.0             2      3002518\n",
      "27  (3002618, 3002718]    0.0             3      3002618\n",
      "28  (3002718, 3002818]    0.0             1      3002718\n",
      "29  (3002818, 3002918]    0.0             2      3002818\n",
      "30  (3002918, 3003018]    1.0             6      3002918\n",
      "31  (3003018, 3003118]    2.0             4      3003018\n",
      "32  (3003118, 3003218]    0.0             0      3003118\n",
      "33  (3003218, 3003318]    0.0             1      3003218\n",
      "34  (3003318, 3003418]    1.0             8      3003318\n",
      "35  (3003418, 3003518]    0.0             3      3003418\n",
      "36  (3003518, 3003618]    1.0             2      3003518\n",
      "37  (3003618, 3003718]    3.0             4      3003618\n",
      "38  (3003718, 3003818]    1.0             6      3003718\n",
      "39  (3003818, 3003918]    0.0             3      3003818\n",
      "40  (3003918, 3004018]    8.0             6      3003918\n",
      "41  (3004018, 3004118]    0.0             0      3004018\n",
      "42  (3004118, 3004218]    0.0             1      3004118\n",
      "43  (3004218, 3004318]    5.0             6      3004218\n",
      "44  (3004318, 3004418]    0.0             1      3004318\n",
      "45  (3004418, 3004518]    0.0             1      3004418\n",
      "46  (3004518, 3004618]    1.0             7      3004518\n",
      "47  (3004618, 3004718]    0.0             0      3004618\n",
      "48  (3004718, 3004818]    0.0             1      3004718\n",
      "49  (3004818, 3004918]   13.0             8      3004818\n",
      "(152618, 4)\n",
      "bin_offset_0    category\n",
      "TT_S0            float64\n",
      "target_count       int64\n",
      "starting_nt        int64\n",
      "dtype: object\n",
      "sumos ilgis: 152618\n",
      "ranges ilgis: 152618\n",
      "Model input - NO SCALE\n",
      "        TT_S0  genes  transcripts  exons  CDSs\n",
      "0         0.0      0            0      0     0\n",
      "1         7.0      0            0      0     0\n",
      "2         1.0      0            0      0     0\n",
      "3         2.0      0            0      0     0\n",
      "4         0.0      0            0      0     0\n",
      "...       ...    ...          ...    ...   ...\n",
      "152613   10.0      0            0      0     0\n",
      "152614    3.0      0            0      0     0\n",
      "152615    4.0      0            0      0     0\n",
      "152616    0.0      0            0      0     0\n",
      "152617    4.0      0            0      0     0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "        TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0         0.0             0                   0             0            0\n",
      "1         7.0             0                   0             0            0\n",
      "2         1.0             0                   0             0            0\n",
      "3         2.0             0                   0             0            0\n",
      "4         0.0             0                   0             0            0\n",
      "...       ...           ...                 ...           ...          ...\n",
      "152613   10.0             0                   0             0            0\n",
      "152614    3.0             0                   0             0            0\n",
      "152615    4.0             0                   0             0            0\n",
      "152616    0.0             0                   0             0            0\n",
      "152617    4.0             0                   0             0            0\n",
      "\n",
      "[152618 rows x 5 columns]\n",
      "0.10474518077815198\n",
      "[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 1 0 1 1 1 0 1 1 1 1 1 1 0 1 1 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 0 0 1 1 1 1 1 1 0 0 1 1 0 1\n",
      " 0 1 0 0 0 0 0 0 0 0 0 0 0 0 1 0 1 1 1 1 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 1\n",
      " 0 0 1 0 0 0 0 0 1 0 0 0 1 0 1 0 0 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 1 1 0 1 1 1 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 1 0 1 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0]\n",
      "threshold density 1.95819\n",
      "cluster signal 219187.0\n",
      "cluster target count 572145\n",
      "!!cluster density 0.3830969422087058\n",
      "cluster signal 237724.0\n",
      "cluster target count 127855\n",
      "!!cluster density 1.8593250166203903\n",
      "\n",
      "Labels mapping\n",
      " {0: 0, 1: 0}\n",
      "DEBUG: 152618 0 \n",
      "\n",
      " +----------------------+-------------+----------------+-------+\n",
      "| bin_offset_0         | TT_S0       | target_count   | +9    |\n",
      "| (category)           | (float64)   | (int64)        | ...   |\n",
      "|----------------------+-------------+----------------+-------|\n",
      "| (3000018, 3000018]   | 0.0         | 0              | ...   |\n",
      "| (3000018, 3000118]   | 7.0         | 11             | ...   |\n",
      "| (3000118, 3000218]   | 1.0         | 4              | ...   |\n",
      "| (3000218, 3000318]   | 2.0         | 4              | ...   |\n",
      "| ...                  | ...         | ...            | ...   |\n",
      "| (18261318, 18261418] | 3.0         | 15             | ...   |\n",
      "| (18261418, 18261518] | 4.0         | 4              | ...   |\n",
      "| (18261518, 18261618] | 0.0         | 3              | ...   |\n",
      "| (18261618, 18261710] | 4.0         | 3              | ...   |\n",
      "+----------------------+-------------+----------------+-------+\n",
      "Unstranded PyRanges object has 152,618 rows and 12 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "9 hidden columns: starting_nt, Chromosome, Start, End, genes, transcripts, ... (+ 3 more.)\n",
      "The model failed to identify open ranges.\n",
      "           0\n",
      "model_name  \n"
     ]
    }
   ],
   "source": [
    "import generate_consensus\n",
    "\n",
    "ranges = generate_consensus.generate_consensus_5x20(chromatin_algos.algo5d_aggregated, [1, 2, 3])\n",
    "result = evaluator.evaluate(merged_ranges = ranges)\n",
    "print(result)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "56123321",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Single Mode Example:\n",
      "Mode: 0    2\n",
      "dtype: int64\n",
      "Mode: <class 'int'>\n",
      "\n",
      "Multiple Modes Example:\n",
      "Modes: 0    2\n",
      "1    3\n",
      "dtype: int64\n",
      "First Mode: <class 'int'>\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Example data with a clear mode\n",
    "data_with_single_mode = pd.Series([1, 2, 2, 3, 4])\n",
    "print(\"Single Mode Example:\")\n",
    "print(\"Mode:\", data_with_single_mode.mode())  # Should work fine\n",
    "print(\"Mode:\", type(data_with_single_mode.mode().iloc[0].item()))  # Should work fine\n",
    "\n",
    "# Example data with multiple modes\n",
    "data_with_multiple_modes = pd.Series([1, 2, 2, 3, 3, 4])\n",
    "print(\"\\nMultiple Modes Example:\")\n",
    "print(\"Modes:\", data_with_multiple_modes.mode())\n",
    "print(\"First Mode:\", type(data_with_multiple_modes.mode().iloc[0].item()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "6b5c5de3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "count                        13118.0\n",
      "mean                      146.683946\n",
      "std                       113.175202\n",
      "min                            100.0\n",
      "25%                            100.0\n",
      "50%                            100.0\n",
      "75%                            200.0\n",
      "max                           2500.0\n",
      "mode                           100.0\n",
      "ATACseq_overlap             0.792635\n",
      "ATACseq_ranges_overlap      0.030302\n",
      "ATACseq_Jaccard             0.030064\n",
      "DNAseq_overlap              0.696856\n",
      "DNAseq_ranges_overlap       0.043802\n",
      "DNAseq_Jaccard              0.042983\n",
      "genes_overlap               0.590862\n",
      "percentage_open             0.105368\n",
      "percentage_open_adj          0.12608\n",
      "model_name                          \n",
      "Name: Length, dtype: object\n"
     ]
    }
   ],
   "source": [
    "result = evaluator.evaluate(merged_ranges = ranges)\n",
    "\n",
    "print(result)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "44720aa9",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "       seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0            19  3000017  3000017  19_3000017    0.0    0.0    0.0\n",
      "1            18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "2            18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "3            18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "4            18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "...         ...      ...      ...         ...    ...    ...    ...\n",
      "699995       18  9717555  9717555  18_9717555    1.0    0.0    0.0\n",
      "699996       19  9717571  9717571  19_9717571    0.0    0.0    0.0\n",
      "699997       19  9717581  9717581  19_9717581    0.0    0.0    0.0\n",
      "699998       18  9717582  9717582  18_9717582    0.0    0.0    0.0\n",
      "699999       18  9717593  9717593  18_9717593    0.0    0.0    0.0\n",
      "\n",
      "[700000 rows x 7 columns]\n"
     ]
    }
   ],
   "source": [
    "import enrichment\n",
    "import pyreadr\n",
    "\n",
    "targets_df = pyreadr.read_r('../inputs/subset1.rds')[None]\n",
    "\n",
    "print(targets_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "bdd299ab",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/enrichment.py:19: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  summed_ranges = targets_df.groupby('ranges_bin')[[SIGNAL_COLUMN, 'target_count']].sum()\n",
      "/Users/lukas/VU/bio-baigiamasis/scripts/enrichment.py:19: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  summed_ranges = targets_df.groupby('ranges_bin')[[SIGNAL_COLUMN, 'target_count']].sum()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dataframe normalized signal: 0.65273\n",
      "                 ranges_bin  norm_signal  pre_signal  post_signal  \\\n",
      "0        [3002218, 3002318)     1.428571    0.333333     2.000000   \n",
      "1        [3002418, 3002518)     3.800000    2.000000     2.000000   \n",
      "2        [3003918, 3004018)     1.333333    0.000000     0.000000   \n",
      "3        [3004818, 3004918)     1.625000    0.000000     0.000000   \n",
      "4        [3005018, 3005318)     5.666667    0.000000     1.500000   \n",
      "...                     ...          ...         ...          ...   \n",
      "13113  [18247918, 18248018)     4.333333    0.500000     0.400000   \n",
      "13114  [18249518, 18249618)     1.285714    1.333333     1.000000   \n",
      "13115  [18250918, 18251018)     1.500000    0.000000     0.142857   \n",
      "13116  [18251518, 18251618)     1.333333    0.200000     0.000000   \n",
      "13117  [18261218, 18261318)     1.111111    0.400000     0.200000   \n",
      "\n",
      "       surround_signal  enrichment_factor1  enrichment_factor2  \n",
      "0             2.333333            0.612245            2.188610  \n",
      "1             4.000000            0.950000            5.821703  \n",
      "2             0.000000            0.000000            2.042703  \n",
      "3             0.000000            0.000000            2.489544  \n",
      "4             1.500000            3.777778            8.681486  \n",
      "...                ...                 ...                 ...  \n",
      "13113         0.900000            4.814815            6.638784  \n",
      "13114         2.333333            0.551020            1.969749  \n",
      "13115         0.142857           10.500000            2.298041  \n",
      "13116         0.200000            6.666667            2.042703  \n",
      "13117         0.600000            1.851852            1.702252  \n",
      "\n",
      "[13118 rows x 7 columns]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/enrichment.py:19: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  summed_ranges = targets_df.groupby('ranges_bin')[[SIGNAL_COLUMN, 'target_count']].sum()\n"
     ]
    }
   ],
   "source": [
    "import enrichment\n",
    "\n",
    "print(enrichment.calculate_enrichment_factors_file(ranges_filepath='../outputs/test_ranges.tsv', targets_filepath='../inputs/subset1.rds'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "d67fb278",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/enrichment.py:19: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  summed_ranges = targets_df.groupby('ranges_bin')[[SIGNAL_COLUMN, 'target_count']].sum()\n",
      "/Users/lukas/VU/bio-baigiamasis/scripts/enrichment.py:19: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  summed_ranges = targets_df.groupby('ranges_bin')[[SIGNAL_COLUMN, 'target_count']].sum()\n",
      "/Users/lukas/VU/bio-baigiamasis/scripts/enrichment.py:19: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  summed_ranges = targets_df.groupby('ranges_bin')[[SIGNAL_COLUMN, 'target_count']].sum()\n",
      "/Users/lukas/VU/bio-baigiamasis/scripts/enrichment.py:19: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  summed_ranges = targets_df.groupby('ranges_bin')[[SIGNAL_COLUMN, 'target_count']].sum()\n"
     ]
    },
    {
     "ename": "KeyError",
     "evalue": "'pre_signal'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "File \u001b[0;32m/opt/anaconda3/lib/python3.12/site-packages/pandas/core/indexes/base.py:3805\u001b[0m, in \u001b[0;36mIndex.get_loc\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m   3804\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m-> 3805\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_engine\u001b[38;5;241m.\u001b[39mget_loc(casted_key)\n\u001b[1;32m   3806\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mKeyError\u001b[39;00m \u001b[38;5;28;01mas\u001b[39;00m err:\n",
      "File \u001b[0;32mindex.pyx:167\u001b[0m, in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[0;34m()\u001b[0m\n",
      "File \u001b[0;32mindex.pyx:191\u001b[0m, in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[0;34m()\u001b[0m\n",
      "File \u001b[0;32mindex.pyx:234\u001b[0m, in \u001b[0;36mpandas._libs.index.IndexEngine._get_loc_duplicates\u001b[0;34m()\u001b[0m\n",
      "File \u001b[0;32mindex.pyx:242\u001b[0m, in \u001b[0;36mpandas._libs.index.IndexEngine._maybe_get_bool_indexer\u001b[0;34m()\u001b[0m\n",
      "File \u001b[0;32mindex.pyx:134\u001b[0m, in \u001b[0;36mpandas._libs.index._unpack_bool_indexer\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;31mKeyError\u001b[0m: 'pre_signal'",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[6], line 2\u001b[0m\n\u001b[1;32m      1\u001b[0m enriched \u001b[38;5;241m=\u001b[39m enrichment\u001b[38;5;241m.\u001b[39mcalculate_norm_signal(ranges, targets_df)\n\u001b[0;32m----> 2\u001b[0m surround \u001b[38;5;241m=\u001b[39m enrichment\u001b[38;5;241m.\u001b[39mcalculate_surround_signal(ranges, targets_df)\n\u001b[1;32m      4\u001b[0m \u001b[38;5;28mprint\u001b[39m(enriched)\n\u001b[1;32m      5\u001b[0m \u001b[38;5;28mprint\u001b[39m(surround)\n",
      "File \u001b[0;32m~/VU/bio-baigiamasis/scripts/enrichment.py:54\u001b[0m, in \u001b[0;36mcalculate_surround_signal\u001b[0;34m(ranges, targets_df)\u001b[0m\n\u001b[1;32m     51\u001b[0m ranges_signal \u001b[38;5;241m=\u001b[39m pd\u001b[38;5;241m.\u001b[39mconcat([ranges_signal\u001b[38;5;241m.\u001b[39mreset_index(drop\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m), pre_signal\u001b[38;5;241m.\u001b[39mreset_index(drop\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m)], axis\u001b[38;5;241m=\u001b[39m\u001b[38;5;241m1\u001b[39m)\n\u001b[1;32m     52\u001b[0m ranges_signal \u001b[38;5;241m=\u001b[39m pd\u001b[38;5;241m.\u001b[39mconcat([ranges_signal\u001b[38;5;241m.\u001b[39mreset_index(drop\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m), post_signal\u001b[38;5;241m.\u001b[39mreset_index(drop\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m)], axis\u001b[38;5;241m=\u001b[39m\u001b[38;5;241m1\u001b[39m)\n\u001b[0;32m---> 54\u001b[0m ranges_signal[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124msurround_signal\u001b[39m\u001b[38;5;124m'\u001b[39m] \u001b[38;5;241m=\u001b[39m ranges_signal[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mpre_signal\u001b[39m\u001b[38;5;124m'\u001b[39m] \u001b[38;5;241m+\u001b[39m ranges_signal[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mpost_signal\u001b[39m\u001b[38;5;124m'\u001b[39m]\n\u001b[1;32m     56\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m ranges_signal\n",
      "File \u001b[0;32m/opt/anaconda3/lib/python3.12/site-packages/pandas/core/frame.py:4102\u001b[0m, in \u001b[0;36mDataFrame.__getitem__\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m   4100\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mcolumns\u001b[38;5;241m.\u001b[39mnlevels \u001b[38;5;241m>\u001b[39m \u001b[38;5;241m1\u001b[39m:\n\u001b[1;32m   4101\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_getitem_multilevel(key)\n\u001b[0;32m-> 4102\u001b[0m indexer \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mcolumns\u001b[38;5;241m.\u001b[39mget_loc(key)\n\u001b[1;32m   4103\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m is_integer(indexer):\n\u001b[1;32m   4104\u001b[0m     indexer \u001b[38;5;241m=\u001b[39m [indexer]\n",
      "File \u001b[0;32m/opt/anaconda3/lib/python3.12/site-packages/pandas/core/indexes/base.py:3812\u001b[0m, in \u001b[0;36mIndex.get_loc\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m   3807\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(casted_key, \u001b[38;5;28mslice\u001b[39m) \u001b[38;5;129;01mor\u001b[39;00m (\n\u001b[1;32m   3808\u001b[0m         \u001b[38;5;28misinstance\u001b[39m(casted_key, abc\u001b[38;5;241m.\u001b[39mIterable)\n\u001b[1;32m   3809\u001b[0m         \u001b[38;5;129;01mand\u001b[39;00m \u001b[38;5;28many\u001b[39m(\u001b[38;5;28misinstance\u001b[39m(x, \u001b[38;5;28mslice\u001b[39m) \u001b[38;5;28;01mfor\u001b[39;00m x \u001b[38;5;129;01min\u001b[39;00m casted_key)\n\u001b[1;32m   3810\u001b[0m     ):\n\u001b[1;32m   3811\u001b[0m         \u001b[38;5;28;01mraise\u001b[39;00m InvalidIndexError(key)\n\u001b[0;32m-> 3812\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mKeyError\u001b[39;00m(key) \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01merr\u001b[39;00m\n\u001b[1;32m   3813\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mTypeError\u001b[39;00m:\n\u001b[1;32m   3814\u001b[0m     \u001b[38;5;66;03m# If we have a listlike key, _check_indexing_error will raise\u001b[39;00m\n\u001b[1;32m   3815\u001b[0m     \u001b[38;5;66;03m#  InvalidIndexError. Otherwise we fall through and re-raise\u001b[39;00m\n\u001b[1;32m   3816\u001b[0m     \u001b[38;5;66;03m#  the TypeError.\u001b[39;00m\n\u001b[1;32m   3817\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_check_indexing_error(key)\n",
      "\u001b[0;31mKeyError\u001b[0m: 'pre_signal'"
     ]
    }
   ],
   "source": [
    "enriched = enrichment.calculate_norm_signal(ranges, targets_df)\n",
    "surround = enrichment.calculate_surround_signal(ranges, targets_df)\n",
    "\n",
    "print(enriched)\n",
    "print(surround)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "49f3ca45",
   "metadata": {},
   "outputs": [],
   "source": [
    "ranges.df.to_csv('../outputs/test_ranges.tsv', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "8de8e4e8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "+--------------+-----------+-----------+-----------+\n",
      "| Chromosome   | Start     | End       | Length    |\n",
      "| (category)   | (int64)   | (int64)   | (int64)   |\n",
      "|--------------+-----------+-----------+-----------|\n",
      "| 18           | 3005016   | 3005116   | 200       |\n",
      "| 18           | 3005516   | 3005616   | 400       |\n",
      "| 18           | 3010016   | 3010116   | 200       |\n",
      "| 18           | 3010616   | 3010716   | 100       |\n",
      "| ...          | ...       | ...       | ...       |\n",
      "| 18           | 9707516   | 9707616   | 100       |\n",
      "| 18           | 9707916   | 9708016   | 200       |\n",
      "| 18           | 9708416   | 9708516   | 300       |\n",
      "| 18           | 9708916   | 9709016   | 100       |\n",
      "+--------------+-----------+-----------+-----------+\n",
      "Unstranded PyRanges object has 4,985 rows and 4 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "+--------------+-----------+-----------+-----------+\n",
      "| Chromosome   | Start     | End       | Length    |\n",
      "| (category)   | (int64)   | (int64)   | (int64)   |\n",
      "|--------------+-----------+-----------+-----------|\n",
      "| 18           | 3005116   | 3005316   | 200       |\n",
      "| 18           | 3005616   | 3006016   | 400       |\n",
      "| 18           | 3010116   | 3010316   | 200       |\n",
      "| 18           | 3010716   | 3010816   | 100       |\n",
      "| ...          | ...       | ...       | ...       |\n",
      "| 18           | 9707616   | 9707716   | 100       |\n",
      "| 18           | 9708016   | 9708216   | 200       |\n",
      "| 18           | 9708516   | 9708816   | 300       |\n",
      "| 18           | 9709016   | 9709116   | 100       |\n",
      "+--------------+-----------+-----------+-----------+\n",
      "Unstranded PyRanges object has 4,985 rows and 4 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n"
     ]
    }
   ],
   "source": [
    "import enrichment\n",
    "\n",
    "print(enrichment.get_pre_regions(ranges))\n",
    "print(ranges)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "fa21a536",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "+--------------+-----------+-----------+-----------+\n",
      "| Chromosome   | Start     | End       | Length    |\n",
      "| (category)   | (int64)   | (int64)   | (int64)   |\n",
      "|--------------+-----------+-----------+-----------|\n",
      "| 18           | 3002416   | 3002516   | 100       |\n",
      "| 18           | 3005116   | 3005316   | 200       |\n",
      "| 18           | 3005416   | 3005516   | 100       |\n",
      "| 18           | 3005616   | 3006016   | 400       |\n",
      "| ...          | ...       | ...       | ...       |\n",
      "| 18           | 9708016   | 9708216   | 200       |\n",
      "| 18           | 9708316   | 9708416   | 100       |\n",
      "| 18           | 9708516   | 9708816   | 300       |\n",
      "| 18           | 9709016   | 9709116   | 100       |\n",
      "+--------------+-----------+-----------+-----------+\n",
      "Unstranded PyRanges object has 6,883 rows and 4 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "os.environ['COLUMNS']=\"300\"\n",
    "print(ranges)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "f475e092",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "24\n",
      "51.0\n"
     ]
    }
   ],
   "source": [
    "test = targets_df[(targets_df[\"start\"] >= 3010116) & (targets_df[\"start\"] <= 3010316)]\n",
    "#[3010116, 3010316]   51.0            24\n",
    "\n",
    "print(test['target_count'].sum())\n",
    "print(test['TT_S0'].sum())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "9435baf8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1615\n",
      "0\n"
     ]
    }
   ],
   "source": [
    "import pyranges as pr\n",
    "\n",
    "ranges.Length = ranges.End - ranges.Start\n",
    "ranges_df = ranges.df\n",
    "ranges_df['']\n",
    "\n",
    "print(len(ranges[ranges.Length < 75]))\n",
    "ranges2 = evaluator.process_ranges(ranges)\n",
    "ranges2.Length = ranges2.End - ranges2.Start\n",
    "print(len(ranges2[ranges2.Length < 75]))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "a929baa0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "count                         8182.0\n",
      "mean                      182.306282\n",
      "std                       197.351081\n",
      "min                              5.0\n",
      "25%                             80.0\n",
      "50%                            115.0\n",
      "75%                            210.0\n",
      "max                           2085.0\n",
      "ATACseq_overlap             0.751796\n",
      "ATACseq_ranges_overlap      0.013469\n",
      "DNAseq_overlap              0.639008\n",
      "DNAseq_ranges_overlap       0.022784\n",
      "genes_overlap                0.54393\n",
      "percentage_open             0.153498\n",
      "percentage_open_adj         0.222048\n",
      "model_name                          \n",
      "Name: Length, dtype: object\n"
     ]
    }
   ],
   "source": [
    "print(evaluator.evaluate(ranges))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "61959364",
   "metadata": {},
   "outputs": [],
   "source": [
    "def calculate_gaps(pyranges_obj):\n",
    "    \"\"\"\n",
    "    Calculate gaps between ranges in a PyRanges object.\n",
    "\n",
    "    Parameters:\n",
    "        pyranges_obj (pr.PyRanges): A PyRanges object with genomic ranges.\n",
    "\n",
    "    Returns:\n",
    "        pr.PyRanges: A PyRanges object containing the gaps.\n",
    "    \"\"\"\n",
    "    # Convert PyRanges to a DataFrame\n",
    "    df = pyranges_obj.df\n",
    "    \n",
    "    # Sort the ranges by Chromosome and Start position\n",
    "    df = df.sort_values([\"Chromosome\", \"Start\"]).reset_index(drop=True)\n",
    "    \n",
    "    # Shift End and Start columns to calculate gaps\n",
    "    prev_end = df.groupby(\"Chromosome\")[\"End\"].shift(0)   # Current End\n",
    "    next_start = df.groupby(\"Chromosome\")[\"Start\"].shift(-1)  # Next Start\n",
    "    \n",
    "    # Identify gaps where End < Next Start\n",
    "    gaps_df = df[(prev_end < next_start)].assign(Start=prev_end, End=next_start)[[\"Chromosome\", \"Start\", \"End\"]]\n",
    "    \n",
    "    # Drop NaN rows caused by shifting and ensure integer type\n",
    "    gaps_df = gaps_df.dropna().astype({\"Start\": \"int\", \"End\": \"int\"})\n",
    "    \n",
    "    # Convert back to PyRanges\n",
    "    return pr.PyRanges(gaps_df)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "cbaf97f0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "               0     1     2     3\n",
      "Chromosome  chr1  chr1  chr1  chr3\n",
      "Start          5    20    80    10\n",
      "End           10    28    95    38\n",
      "Strand         +     +     -     +\n",
      "title          a     b     c     d\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "df=pd.DataFrame(\n",
    "{'Chromosome':['chr1', 'chr1', 'chr1', 'chr3'],\n",
    " 'Start': [5, 20, 80, 10],\n",
    " 'End':   [10, 28, 95, 38],\n",
    " 'Strand':['+', '+', '-', '+'],\n",
    " 'title': ['a', 'b', 'c', 'd']}\n",
    ")\n",
    "\n",
    "print(df.transpose())\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "847f9a77",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "443\n",
      "0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/y9/8yssfmz161s4ghy99nf1f2cw0000gn/T/ipykernel_27239/19855938.py:18: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  prev_end = df.groupby(\"Chromosome\")[\"End\"].shift(0)   # Current End\n",
      "/var/folders/y9/8yssfmz161s4ghy99nf1f2cw0000gn/T/ipykernel_27239/19855938.py:19: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  next_start = df.groupby(\"Chromosome\")[\"Start\"].shift(-1)  # Next Start\n",
      "/var/folders/y9/8yssfmz161s4ghy99nf1f2cw0000gn/T/ipykernel_27239/19855938.py:18: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  prev_end = df.groupby(\"Chromosome\")[\"End\"].shift(0)   # Current End\n",
      "/var/folders/y9/8yssfmz161s4ghy99nf1f2cw0000gn/T/ipykernel_27239/19855938.py:19: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  next_start = df.groupby(\"Chromosome\")[\"Start\"].shift(-1)  # Next Start\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import pyranges as pr\n",
    "\n",
    "ranges_df = ranges.df\n",
    "gaps = calculate_gaps(ranges)\n",
    "gaps.Length = gaps.End - gaps.Start\n",
    "print(len(gaps[gaps.Length <= 50]))\n",
    "\n",
    "ranges2 = evaluator.process_ranges(ranges)\n",
    "gaps2 = calculate_gaps(ranges2)\n",
    "gaps2.Length = gaps2.End - gaps2.Start\n",
    "print(len(gaps2[gaps2.Length <= 50]))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "28e1cdda",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Runnning scale 1\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0\n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "BINS  IntervalIndex([(3000016, 3000016], (3000016, 3000116], (3000116, 3000216],\n",
      "               (3000216, 3000316], (3000316, 3000416], (3000416, 3000516],\n",
      "               (3000516, 3000616], (3000616, 3000716], (3000716, 3000816],\n",
      "               (3000816, 3000916],\n",
      "               ...\n",
      "               (9716616, 9716716], (9716716, 9716816], (9716816, 9716916],\n",
      "               (9716916, 9717016], (9717016, 9717116], (9717116, 9717216],\n",
      "               (9717216, 9717316], (9717316, 9717416], (9717416, 9717516],\n",
      "               (9717516, 9717593]],\n",
      "              dtype='interval[int64, right]', length=67177)\n",
      "HEAD w BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0   \n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "\n",
      "         bin_offset_0  \n",
      "0  (3000016, 3000116]  \n",
      "1  (3000016, 3000116]  \n",
      "2  (3000016, 3000116]  \n",
      "3  (3000016, 3000116]  \n",
      "4  (3000016, 3000116]  \n",
      "TAIL w BINS\n",
      "       seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  9717555  9717555  18_9717555    1.0    0.0    0.0   \n",
      "699996       19  9717571  9717571  19_9717571    0.0    0.0    0.0   \n",
      "699997       19  9717581  9717581  19_9717581    0.0    0.0    0.0   \n",
      "699998       18  9717582  9717582  18_9717582    0.0    0.0    0.0   \n",
      "699999       18  9717593  9717593  18_9717593    0.0    0.0    0.0   \n",
      "\n",
      "              bin_offset_0  \n",
      "699995  (9717516, 9717593]  \n",
      "699996  (9717516, 9717593]  \n",
      "699997  (9717516, 9717593]  \n",
      "699998  (9717516, 9717593]  \n",
      "699999  (9717516, 9717593]  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:151: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "    TT_S0  starting_nt\n",
      "0     0.0      3000016\n",
      "1     7.0      3000016\n",
      "2     1.0      3000116\n",
      "3     2.0      3000216\n",
      "4     0.0      3000316\n",
      "5     0.0      3000416\n",
      "6     0.0      3000516\n",
      "7     0.0      3000616\n",
      "8     0.0      3000716\n",
      "9     0.0      3000816\n",
      "10    0.0      3000916\n",
      "11    2.0      3001016\n",
      "12    0.0      3001116\n",
      "13    0.0      3001216\n",
      "14    0.0      3001316\n",
      "15    1.0      3001416\n",
      "16    3.0      3001516\n",
      "17    3.0      3001616\n",
      "18    0.0      3001716\n",
      "19    0.0      3001816\n",
      "20    1.0      3001916\n",
      "21    0.0      3002016\n",
      "22    1.0      3002116\n",
      "23   10.0      3002216\n",
      "24    2.0      3002316\n",
      "25   19.0      3002416\n",
      "26    4.0      3002516\n",
      "27    0.0      3002616\n",
      "28    0.0      3002716\n",
      "29    0.0      3002816\n",
      "30    1.0      3002916\n",
      "31    2.0      3003016\n",
      "32    0.0      3003116\n",
      "33    0.0      3003216\n",
      "34    1.0      3003316\n",
      "35    0.0      3003416\n",
      "36    1.0      3003516\n",
      "37    1.0      3003616\n",
      "38    3.0      3003716\n",
      "39    0.0      3003816\n",
      "40    8.0      3003916\n",
      "41    0.0      3004016\n",
      "42    0.0      3004116\n",
      "43    5.0      3004216\n",
      "44    0.0      3004316\n",
      "45    0.0      3004416\n",
      "46    1.0      3004516\n",
      "47    0.0      3004616\n",
      "48    0.0      3004716\n",
      "49   13.0      3004816\n",
      "(67177, 2)\n",
      "TT_S0          float64\n",
      "starting_nt      int64\n",
      "dtype: object\n",
      "sumos ilgis: 67177\n",
      "ranges ilgis: 67177\n",
      "Model input - NO SCALE\n",
      "       TT_S0  genes  transcripts  exons  CDSs\n",
      "0        0.0      0            0      0     0\n",
      "1        7.0      0            0      0     0\n",
      "2        1.0      0            0      0     0\n",
      "3        2.0      0            0      0     0\n",
      "4        0.0      0            0      0     0\n",
      "...      ...    ...          ...    ...   ...\n",
      "67172    0.0      0            0      0     0\n",
      "67173    0.0      0            0      0     0\n",
      "67174    0.0      0            0      0     0\n",
      "67175    1.0      0            0      0     0\n",
      "67176    1.0      0            0      0     0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "       TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0        0.0             0                   0             0            0\n",
      "1        7.0             0                   0             0            0\n",
      "2        1.0             0                   0             0            0\n",
      "3        2.0             0                   0             0            0\n",
      "4        0.0             0                   0             0            0\n",
      "...      ...           ...                 ...           ...          ...\n",
      "67172    0.0             0                   0             0            0\n",
      "67173    0.0             0                   0             0            0\n",
      "67174    0.0             0                   0             0            0\n",
      "67175    1.0             0                   0             0            0\n",
      "67176    1.0             0                   0             0            0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "0.17092159518882952\n",
      "[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 1 0 1 1 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 1 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0]\n",
      "Runnning scale 2\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0\n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "BINS  IntervalIndex([(3000016, 3000016], (3000016, 3000116], (3000116, 3000216],\n",
      "               (3000216, 3000316], (3000316, 3000416], (3000416, 3000516],\n",
      "               (3000516, 3000616], (3000616, 3000716], (3000716, 3000816],\n",
      "               (3000816, 3000916],\n",
      "               ...\n",
      "               (9716616, 9716716], (9716716, 9716816], (9716816, 9716916],\n",
      "               (9716916, 9717016], (9717016, 9717116], (9717116, 9717216],\n",
      "               (9717216, 9717316], (9717316, 9717416], (9717416, 9717516],\n",
      "               (9717516, 9717593]],\n",
      "              dtype='interval[int64, right]', length=67177)\n",
      "HEAD w BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0   \n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "\n",
      "         bin_offset_0  \n",
      "0  (3000016, 3000116]  \n",
      "1  (3000016, 3000116]  \n",
      "2  (3000016, 3000116]  \n",
      "3  (3000016, 3000116]  \n",
      "4  (3000016, 3000116]  \n",
      "TAIL w BINS\n",
      "       seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  9717555  9717555  18_9717555    1.0    0.0    0.0   \n",
      "699996       19  9717571  9717571  19_9717571    0.0    0.0    0.0   \n",
      "699997       19  9717581  9717581  19_9717581    0.0    0.0    0.0   \n",
      "699998       18  9717582  9717582  18_9717582    0.0    0.0    0.0   \n",
      "699999       18  9717593  9717593  18_9717593    0.0    0.0    0.0   \n",
      "\n",
      "              bin_offset_0  \n",
      "699995  (9717516, 9717593]  \n",
      "699996  (9717516, 9717593]  \n",
      "699997  (9717516, 9717593]  \n",
      "699998  (9717516, 9717593]  \n",
      "699999  (9717516, 9717593]  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:151: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "    TT_S0  starting_nt\n",
      "0     0.0      3000016\n",
      "1     7.0      3000016\n",
      "2     1.0      3000116\n",
      "3     2.0      3000216\n",
      "4     0.0      3000316\n",
      "5     0.0      3000416\n",
      "6     0.0      3000516\n",
      "7     0.0      3000616\n",
      "8     0.0      3000716\n",
      "9     0.0      3000816\n",
      "10    0.0      3000916\n",
      "11    2.0      3001016\n",
      "12    0.0      3001116\n",
      "13    0.0      3001216\n",
      "14    0.0      3001316\n",
      "15    1.0      3001416\n",
      "16    3.0      3001516\n",
      "17    3.0      3001616\n",
      "18    0.0      3001716\n",
      "19    0.0      3001816\n",
      "20    1.0      3001916\n",
      "21    0.0      3002016\n",
      "22    1.0      3002116\n",
      "23   10.0      3002216\n",
      "24    2.0      3002316\n",
      "25   19.0      3002416\n",
      "26    4.0      3002516\n",
      "27    0.0      3002616\n",
      "28    0.0      3002716\n",
      "29    0.0      3002816\n",
      "30    1.0      3002916\n",
      "31    2.0      3003016\n",
      "32    0.0      3003116\n",
      "33    0.0      3003216\n",
      "34    1.0      3003316\n",
      "35    0.0      3003416\n",
      "36    1.0      3003516\n",
      "37    1.0      3003616\n",
      "38    3.0      3003716\n",
      "39    0.0      3003816\n",
      "40    8.0      3003916\n",
      "41    0.0      3004016\n",
      "42    0.0      3004116\n",
      "43    5.0      3004216\n",
      "44    0.0      3004316\n",
      "45    0.0      3004416\n",
      "46    1.0      3004516\n",
      "47    0.0      3004616\n",
      "48    0.0      3004716\n",
      "49   13.0      3004816\n",
      "(67177, 2)\n",
      "TT_S0          float64\n",
      "starting_nt      int64\n",
      "dtype: object\n",
      "sumos ilgis: 67177\n",
      "ranges ilgis: 67177\n",
      "Model input - NO SCALE\n",
      "       TT_S0  genes  transcripts  exons  CDSs\n",
      "0        0.0      0            0      0     0\n",
      "1        7.0      0            0      0     0\n",
      "2        1.0      0            0      0     0\n",
      "3        2.0      0            0      0     0\n",
      "4        0.0      0            0      0     0\n",
      "...      ...    ...          ...    ...   ...\n",
      "67172    0.0      0            0      0     0\n",
      "67173    0.0      0            0      0     0\n",
      "67174    0.0      0            0      0     0\n",
      "67175    1.0      0            0      0     0\n",
      "67176    1.0      0            0      0     0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "       TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0        0.0             0                   0             0            0\n",
      "1        7.0             0                   0             0            0\n",
      "2        1.0             0                   0             0            0\n",
      "3        2.0             0                   0             0            0\n",
      "4        0.0             0                   0             0            0\n",
      "...      ...           ...                 ...           ...          ...\n",
      "67172    0.0             0                   0             0            0\n",
      "67173    0.0             0                   0             0            0\n",
      "67174    0.0             0                   0             0            0\n",
      "67175    1.0             0                   0             0            0\n",
      "67176    1.0             0                   0             0            0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "0.20496598538190156\n",
      "[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 1 0 1 1 1 1 0 0 1 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 1 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0]\n",
      "Runnning scale 4\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0\n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "BINS  IntervalIndex([(3000016, 3000016], (3000016, 3000116], (3000116, 3000216],\n",
      "               (3000216, 3000316], (3000316, 3000416], (3000416, 3000516],\n",
      "               (3000516, 3000616], (3000616, 3000716], (3000716, 3000816],\n",
      "               (3000816, 3000916],\n",
      "               ...\n",
      "               (9716616, 9716716], (9716716, 9716816], (9716816, 9716916],\n",
      "               (9716916, 9717016], (9717016, 9717116], (9717116, 9717216],\n",
      "               (9717216, 9717316], (9717316, 9717416], (9717416, 9717516],\n",
      "               (9717516, 9717593]],\n",
      "              dtype='interval[int64, right]', length=67177)\n",
      "HEAD w BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0   \n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "\n",
      "         bin_offset_0  \n",
      "0  (3000016, 3000116]  \n",
      "1  (3000016, 3000116]  \n",
      "2  (3000016, 3000116]  \n",
      "3  (3000016, 3000116]  \n",
      "4  (3000016, 3000116]  \n",
      "TAIL w BINS\n",
      "       seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  9717555  9717555  18_9717555    1.0    0.0    0.0   \n",
      "699996       19  9717571  9717571  19_9717571    0.0    0.0    0.0   \n",
      "699997       19  9717581  9717581  19_9717581    0.0    0.0    0.0   \n",
      "699998       18  9717582  9717582  18_9717582    0.0    0.0    0.0   \n",
      "699999       18  9717593  9717593  18_9717593    0.0    0.0    0.0   \n",
      "\n",
      "              bin_offset_0  \n",
      "699995  (9717516, 9717593]  \n",
      "699996  (9717516, 9717593]  \n",
      "699997  (9717516, 9717593]  \n",
      "699998  (9717516, 9717593]  \n",
      "699999  (9717516, 9717593]  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:151: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "    TT_S0  starting_nt\n",
      "0     0.0      3000016\n",
      "1     7.0      3000016\n",
      "2     1.0      3000116\n",
      "3     2.0      3000216\n",
      "4     0.0      3000316\n",
      "5     0.0      3000416\n",
      "6     0.0      3000516\n",
      "7     0.0      3000616\n",
      "8     0.0      3000716\n",
      "9     0.0      3000816\n",
      "10    0.0      3000916\n",
      "11    2.0      3001016\n",
      "12    0.0      3001116\n",
      "13    0.0      3001216\n",
      "14    0.0      3001316\n",
      "15    1.0      3001416\n",
      "16    3.0      3001516\n",
      "17    3.0      3001616\n",
      "18    0.0      3001716\n",
      "19    0.0      3001816\n",
      "20    1.0      3001916\n",
      "21    0.0      3002016\n",
      "22    1.0      3002116\n",
      "23   10.0      3002216\n",
      "24    2.0      3002316\n",
      "25   19.0      3002416\n",
      "26    4.0      3002516\n",
      "27    0.0      3002616\n",
      "28    0.0      3002716\n",
      "29    0.0      3002816\n",
      "30    1.0      3002916\n",
      "31    2.0      3003016\n",
      "32    0.0      3003116\n",
      "33    0.0      3003216\n",
      "34    1.0      3003316\n",
      "35    0.0      3003416\n",
      "36    1.0      3003516\n",
      "37    1.0      3003616\n",
      "38    3.0      3003716\n",
      "39    0.0      3003816\n",
      "40    8.0      3003916\n",
      "41    0.0      3004016\n",
      "42    0.0      3004116\n",
      "43    5.0      3004216\n",
      "44    0.0      3004316\n",
      "45    0.0      3004416\n",
      "46    1.0      3004516\n",
      "47    0.0      3004616\n",
      "48    0.0      3004716\n",
      "49   13.0      3004816\n",
      "(67177, 2)\n",
      "TT_S0          float64\n",
      "starting_nt      int64\n",
      "dtype: object\n",
      "sumos ilgis: 67177\n",
      "ranges ilgis: 67177\n",
      "Model input - NO SCALE\n",
      "       TT_S0  genes  transcripts  exons  CDSs\n",
      "0        0.0      0            0      0     0\n",
      "1        7.0      0            0      0     0\n",
      "2        1.0      0            0      0     0\n",
      "3        2.0      0            0      0     0\n",
      "4        0.0      0            0      0     0\n",
      "...      ...    ...          ...    ...   ...\n",
      "67172    0.0      0            0      0     0\n",
      "67173    0.0      0            0      0     0\n",
      "67174    0.0      0            0      0     0\n",
      "67175    1.0      0            0      0     0\n",
      "67176    1.0      0            0      0     0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "       TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0        0.0             0                   0             0            0\n",
      "1        7.0             0                   0             0            0\n",
      "2        1.0             0                   0             0            0\n",
      "3        2.0             0                   0             0            0\n",
      "4        0.0             0                   0             0            0\n",
      "...      ...           ...                 ...           ...          ...\n",
      "67172    0.0             0                   0             0            0\n",
      "67173    0.0             0                   0             0            0\n",
      "67174    0.0             0                   0             0            0\n",
      "67175    1.0             0                   0             0            0\n",
      "67176    1.0             0                   0             0            0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "0.20496598538190156\n",
      "[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 1 0 1 1 1 1 0 0 1 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 1 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0]\n",
      "Runnning scale 8\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0\n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "BINS  IntervalIndex([(3000016, 3000016], (3000016, 3000116], (3000116, 3000216],\n",
      "               (3000216, 3000316], (3000316, 3000416], (3000416, 3000516],\n",
      "               (3000516, 3000616], (3000616, 3000716], (3000716, 3000816],\n",
      "               (3000816, 3000916],\n",
      "               ...\n",
      "               (9716616, 9716716], (9716716, 9716816], (9716816, 9716916],\n",
      "               (9716916, 9717016], (9717016, 9717116], (9717116, 9717216],\n",
      "               (9717216, 9717316], (9717316, 9717416], (9717416, 9717516],\n",
      "               (9717516, 9717593]],\n",
      "              dtype='interval[int64, right]', length=67177)\n",
      "HEAD w BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0   \n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "\n",
      "         bin_offset_0  \n",
      "0  (3000016, 3000116]  \n",
      "1  (3000016, 3000116]  \n",
      "2  (3000016, 3000116]  \n",
      "3  (3000016, 3000116]  \n",
      "4  (3000016, 3000116]  \n",
      "TAIL w BINS\n",
      "       seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  9717555  9717555  18_9717555    1.0    0.0    0.0   \n",
      "699996       19  9717571  9717571  19_9717571    0.0    0.0    0.0   \n",
      "699997       19  9717581  9717581  19_9717581    0.0    0.0    0.0   \n",
      "699998       18  9717582  9717582  18_9717582    0.0    0.0    0.0   \n",
      "699999       18  9717593  9717593  18_9717593    0.0    0.0    0.0   \n",
      "\n",
      "              bin_offset_0  \n",
      "699995  (9717516, 9717593]  \n",
      "699996  (9717516, 9717593]  \n",
      "699997  (9717516, 9717593]  \n",
      "699998  (9717516, 9717593]  \n",
      "699999  (9717516, 9717593]  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:151: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "    TT_S0  starting_nt\n",
      "0     0.0      3000016\n",
      "1     7.0      3000016\n",
      "2     1.0      3000116\n",
      "3     2.0      3000216\n",
      "4     0.0      3000316\n",
      "5     0.0      3000416\n",
      "6     0.0      3000516\n",
      "7     0.0      3000616\n",
      "8     0.0      3000716\n",
      "9     0.0      3000816\n",
      "10    0.0      3000916\n",
      "11    2.0      3001016\n",
      "12    0.0      3001116\n",
      "13    0.0      3001216\n",
      "14    0.0      3001316\n",
      "15    1.0      3001416\n",
      "16    3.0      3001516\n",
      "17    3.0      3001616\n",
      "18    0.0      3001716\n",
      "19    0.0      3001816\n",
      "20    1.0      3001916\n",
      "21    0.0      3002016\n",
      "22    1.0      3002116\n",
      "23   10.0      3002216\n",
      "24    2.0      3002316\n",
      "25   19.0      3002416\n",
      "26    4.0      3002516\n",
      "27    0.0      3002616\n",
      "28    0.0      3002716\n",
      "29    0.0      3002816\n",
      "30    1.0      3002916\n",
      "31    2.0      3003016\n",
      "32    0.0      3003116\n",
      "33    0.0      3003216\n",
      "34    1.0      3003316\n",
      "35    0.0      3003416\n",
      "36    1.0      3003516\n",
      "37    1.0      3003616\n",
      "38    3.0      3003716\n",
      "39    0.0      3003816\n",
      "40    8.0      3003916\n",
      "41    0.0      3004016\n",
      "42    0.0      3004116\n",
      "43    5.0      3004216\n",
      "44    0.0      3004316\n",
      "45    0.0      3004416\n",
      "46    1.0      3004516\n",
      "47    0.0      3004616\n",
      "48    0.0      3004716\n",
      "49   13.0      3004816\n",
      "(67177, 2)\n",
      "TT_S0          float64\n",
      "starting_nt      int64\n",
      "dtype: object\n",
      "sumos ilgis: 67177\n",
      "ranges ilgis: 67177\n",
      "Model input - NO SCALE\n",
      "       TT_S0  genes  transcripts  exons  CDSs\n",
      "0        0.0      0            0      0     0\n",
      "1        7.0      0            0      0     0\n",
      "2        1.0      0            0      0     0\n",
      "3        2.0      0            0      0     0\n",
      "4        0.0      0            0      0     0\n",
      "...      ...    ...          ...    ...   ...\n",
      "67172    0.0      0            0      0     0\n",
      "67173    0.0      0            0      0     0\n",
      "67174    0.0      0            0      0     0\n",
      "67175    1.0      0            0      0     0\n",
      "67176    1.0      0            0      0     0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "       TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0        0.0             0                   0             0            0\n",
      "1        7.0             0                   0             0            0\n",
      "2        1.0             0                   0             0            0\n",
      "3        2.0             0                   0             0            0\n",
      "4        0.0             0                   0             0            0\n",
      "...      ...           ...                 ...           ...          ...\n",
      "67172    0.0             0                   0             0            0\n",
      "67173    0.0             0                   0             0            0\n",
      "67174    0.0             0                   0             0            0\n",
      "67175    1.0             0                   0             0            0\n",
      "67176    1.0             0                   0             0            0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "0.20496598538190156\n",
      "[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 1 0 1 1 1 1 0 0 1 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 1 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0]\n",
      "Runnning scale 16\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0\n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "BINS  IntervalIndex([(3000016, 3000016], (3000016, 3000116], (3000116, 3000216],\n",
      "               (3000216, 3000316], (3000316, 3000416], (3000416, 3000516],\n",
      "               (3000516, 3000616], (3000616, 3000716], (3000716, 3000816],\n",
      "               (3000816, 3000916],\n",
      "               ...\n",
      "               (9716616, 9716716], (9716716, 9716816], (9716816, 9716916],\n",
      "               (9716916, 9717016], (9717016, 9717116], (9717116, 9717216],\n",
      "               (9717216, 9717316], (9717316, 9717416], (9717416, 9717516],\n",
      "               (9717516, 9717593]],\n",
      "              dtype='interval[int64, right]', length=67177)\n",
      "HEAD w BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0   \n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "\n",
      "         bin_offset_0  \n",
      "0  (3000016, 3000116]  \n",
      "1  (3000016, 3000116]  \n",
      "2  (3000016, 3000116]  \n",
      "3  (3000016, 3000116]  \n",
      "4  (3000016, 3000116]  \n",
      "TAIL w BINS\n",
      "       seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  9717555  9717555  18_9717555    1.0    0.0    0.0   \n",
      "699996       19  9717571  9717571  19_9717571    0.0    0.0    0.0   \n",
      "699997       19  9717581  9717581  19_9717581    0.0    0.0    0.0   \n",
      "699998       18  9717582  9717582  18_9717582    0.0    0.0    0.0   \n",
      "699999       18  9717593  9717593  18_9717593    0.0    0.0    0.0   \n",
      "\n",
      "              bin_offset_0  \n",
      "699995  (9717516, 9717593]  \n",
      "699996  (9717516, 9717593]  \n",
      "699997  (9717516, 9717593]  \n",
      "699998  (9717516, 9717593]  \n",
      "699999  (9717516, 9717593]  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:151: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "    TT_S0  starting_nt\n",
      "0     0.0      3000016\n",
      "1     7.0      3000016\n",
      "2     1.0      3000116\n",
      "3     2.0      3000216\n",
      "4     0.0      3000316\n",
      "5     0.0      3000416\n",
      "6     0.0      3000516\n",
      "7     0.0      3000616\n",
      "8     0.0      3000716\n",
      "9     0.0      3000816\n",
      "10    0.0      3000916\n",
      "11    2.0      3001016\n",
      "12    0.0      3001116\n",
      "13    0.0      3001216\n",
      "14    0.0      3001316\n",
      "15    1.0      3001416\n",
      "16    3.0      3001516\n",
      "17    3.0      3001616\n",
      "18    0.0      3001716\n",
      "19    0.0      3001816\n",
      "20    1.0      3001916\n",
      "21    0.0      3002016\n",
      "22    1.0      3002116\n",
      "23   10.0      3002216\n",
      "24    2.0      3002316\n",
      "25   19.0      3002416\n",
      "26    4.0      3002516\n",
      "27    0.0      3002616\n",
      "28    0.0      3002716\n",
      "29    0.0      3002816\n",
      "30    1.0      3002916\n",
      "31    2.0      3003016\n",
      "32    0.0      3003116\n",
      "33    0.0      3003216\n",
      "34    1.0      3003316\n",
      "35    0.0      3003416\n",
      "36    1.0      3003516\n",
      "37    1.0      3003616\n",
      "38    3.0      3003716\n",
      "39    0.0      3003816\n",
      "40    8.0      3003916\n",
      "41    0.0      3004016\n",
      "42    0.0      3004116\n",
      "43    5.0      3004216\n",
      "44    0.0      3004316\n",
      "45    0.0      3004416\n",
      "46    1.0      3004516\n",
      "47    0.0      3004616\n",
      "48    0.0      3004716\n",
      "49   13.0      3004816\n",
      "(67177, 2)\n",
      "TT_S0          float64\n",
      "starting_nt      int64\n",
      "dtype: object\n",
      "sumos ilgis: 67177\n",
      "ranges ilgis: 67177\n",
      "Model input - NO SCALE\n",
      "       TT_S0  genes  transcripts  exons  CDSs\n",
      "0        0.0      0            0      0     0\n",
      "1        7.0      0            0      0     0\n",
      "2        1.0      0            0      0     0\n",
      "3        2.0      0            0      0     0\n",
      "4        0.0      0            0      0     0\n",
      "...      ...    ...          ...    ...   ...\n",
      "67172    0.0      0            0      0     0\n",
      "67173    0.0      0            0      0     0\n",
      "67174    0.0      0            0      0     0\n",
      "67175    1.0      0            0      0     0\n",
      "67176    1.0      0            0      0     0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "       TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0        0.0             0                   0             0            0\n",
      "1        7.0             0                   0             0            0\n",
      "2        1.0             0                   0             0            0\n",
      "3        2.0             0                   0             0            0\n",
      "4        0.0             0                   0             0            0\n",
      "...      ...           ...                 ...           ...          ...\n",
      "67172    0.0             0                   0             0            0\n",
      "67173    0.0             0                   0             0            0\n",
      "67174    0.0             0                   0             0            0\n",
      "67175    1.0             0                   0             0            0\n",
      "67176    1.0             0                   0             0            0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "0.20496598538190156\n",
      "[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 1 0 1 1 1 1 0 0 1 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 1 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0]\n",
      "Runnning scale 32\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0\n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "BINS  IntervalIndex([(3000016, 3000016], (3000016, 3000116], (3000116, 3000216],\n",
      "               (3000216, 3000316], (3000316, 3000416], (3000416, 3000516],\n",
      "               (3000516, 3000616], (3000616, 3000716], (3000716, 3000816],\n",
      "               (3000816, 3000916],\n",
      "               ...\n",
      "               (9716616, 9716716], (9716716, 9716816], (9716816, 9716916],\n",
      "               (9716916, 9717016], (9717016, 9717116], (9717116, 9717216],\n",
      "               (9717216, 9717316], (9717316, 9717416], (9717416, 9717516],\n",
      "               (9717516, 9717593]],\n",
      "              dtype='interval[int64, right]', length=67177)\n",
      "HEAD w BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0   \n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "\n",
      "         bin_offset_0  \n",
      "0  (3000016, 3000116]  \n",
      "1  (3000016, 3000116]  \n",
      "2  (3000016, 3000116]  \n",
      "3  (3000016, 3000116]  \n",
      "4  (3000016, 3000116]  \n",
      "TAIL w BINS\n",
      "       seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  9717555  9717555  18_9717555    1.0    0.0    0.0   \n",
      "699996       19  9717571  9717571  19_9717571    0.0    0.0    0.0   \n",
      "699997       19  9717581  9717581  19_9717581    0.0    0.0    0.0   \n",
      "699998       18  9717582  9717582  18_9717582    0.0    0.0    0.0   \n",
      "699999       18  9717593  9717593  18_9717593    0.0    0.0    0.0   \n",
      "\n",
      "              bin_offset_0  \n",
      "699995  (9717516, 9717593]  \n",
      "699996  (9717516, 9717593]  \n",
      "699997  (9717516, 9717593]  \n",
      "699998  (9717516, 9717593]  \n",
      "699999  (9717516, 9717593]  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:151: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "    TT_S0  starting_nt\n",
      "0     0.0      3000016\n",
      "1     7.0      3000016\n",
      "2     1.0      3000116\n",
      "3     2.0      3000216\n",
      "4     0.0      3000316\n",
      "5     0.0      3000416\n",
      "6     0.0      3000516\n",
      "7     0.0      3000616\n",
      "8     0.0      3000716\n",
      "9     0.0      3000816\n",
      "10    0.0      3000916\n",
      "11    2.0      3001016\n",
      "12    0.0      3001116\n",
      "13    0.0      3001216\n",
      "14    0.0      3001316\n",
      "15    1.0      3001416\n",
      "16    3.0      3001516\n",
      "17    3.0      3001616\n",
      "18    0.0      3001716\n",
      "19    0.0      3001816\n",
      "20    1.0      3001916\n",
      "21    0.0      3002016\n",
      "22    1.0      3002116\n",
      "23   10.0      3002216\n",
      "24    2.0      3002316\n",
      "25   19.0      3002416\n",
      "26    4.0      3002516\n",
      "27    0.0      3002616\n",
      "28    0.0      3002716\n",
      "29    0.0      3002816\n",
      "30    1.0      3002916\n",
      "31    2.0      3003016\n",
      "32    0.0      3003116\n",
      "33    0.0      3003216\n",
      "34    1.0      3003316\n",
      "35    0.0      3003416\n",
      "36    1.0      3003516\n",
      "37    1.0      3003616\n",
      "38    3.0      3003716\n",
      "39    0.0      3003816\n",
      "40    8.0      3003916\n",
      "41    0.0      3004016\n",
      "42    0.0      3004116\n",
      "43    5.0      3004216\n",
      "44    0.0      3004316\n",
      "45    0.0      3004416\n",
      "46    1.0      3004516\n",
      "47    0.0      3004616\n",
      "48    0.0      3004716\n",
      "49   13.0      3004816\n",
      "(67177, 2)\n",
      "TT_S0          float64\n",
      "starting_nt      int64\n",
      "dtype: object\n",
      "sumos ilgis: 67177\n",
      "ranges ilgis: 67177\n",
      "Model input - NO SCALE\n",
      "       TT_S0  genes  transcripts  exons  CDSs\n",
      "0        0.0      0            0      0     0\n",
      "1        7.0      0            0      0     0\n",
      "2        1.0      0            0      0     0\n",
      "3        2.0      0            0      0     0\n",
      "4        0.0      0            0      0     0\n",
      "...      ...    ...          ...    ...   ...\n",
      "67172    0.0      0            0      0     0\n",
      "67173    0.0      0            0      0     0\n",
      "67174    0.0      0            0      0     0\n",
      "67175    1.0      0            0      0     0\n",
      "67176    1.0      0            0      0     0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "       TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0        0.0             0                   0             0            0\n",
      "1        7.0             0                   0             0            0\n",
      "2        1.0             0                   0             0            0\n",
      "3        2.0             0                   0             0            0\n",
      "4        0.0             0                   0             0            0\n",
      "...      ...           ...                 ...           ...          ...\n",
      "67172    0.0             0                   0             0            0\n",
      "67173    0.0             0                   0             0            0\n",
      "67174    0.0             0                   0             0            0\n",
      "67175    1.0             0                   0             0            0\n",
      "67176    1.0             0                   0             0            0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "0.17092159518882952\n",
      "[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 1 0 1 1 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 1 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0]\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "for i in range(6):\n",
    "    scale = 2 ** i\n",
    "    print(\"Runnning scale\", scale)\n",
    "    evaluation = evaluator.evaluate(dataframe = chromatin_algos.algo5d(scale), model_name=f\"k-means 5 features scale={scale}\")\n",
    "    result = pd.concat([result, evaluation], axis=1)\n",
    "\n",
    "result.to_csv('../outputs/results1.csv', sep = '\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ca1280ed",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                         Length                      Length  \\\n",
      "count                    7301.0                      6456.0   \n",
      "mean                 188.590604                  177.850062   \n",
      "std                  174.998864                  157.257952   \n",
      "min                       100.0                       100.0   \n",
      "25%                       100.0                       100.0   \n",
      "50%                       100.0                       100.0   \n",
      "75%                       200.0                       200.0   \n",
      "max                      2400.0                      2000.0   \n",
      "ATACseq_overlap        0.697949                    0.668276   \n",
      "DNAseq_overlap         0.595329                    0.553813   \n",
      "genes_overlap          0.544833                    0.550809   \n",
      "percentage_open        0.141691                    0.118157   \n",
      "percentage_open_adj    0.204969                    0.170924   \n",
      "model_name                       k-means 5 features scale=1   \n",
      "min_open_signal            17.0                        19.0   \n",
      "max_open_signal           393.0                       393.0   \n",
      "min_closed_signal           0.0                         0.0   \n",
      "max_closed_signal          16.0                        18.0   \n",
      "\n",
      "                                         Length                      Length  \\\n",
      "count                                    7301.0                      7301.0   \n",
      "mean                                 188.590604                  188.590604   \n",
      "std                                  174.998864                  174.998864   \n",
      "min                                       100.0                       100.0   \n",
      "25%                                       100.0                       100.0   \n",
      "50%                                       100.0                       100.0   \n",
      "75%                                       200.0                       200.0   \n",
      "max                                      2400.0                      2400.0   \n",
      "ATACseq_overlap                        0.697949                    0.697949   \n",
      "DNAseq_overlap                         0.595329                    0.595329   \n",
      "genes_overlap                          0.544833                    0.544833   \n",
      "percentage_open                        0.141691                    0.141691   \n",
      "percentage_open_adj                    0.204969                    0.204969   \n",
      "model_name           k-means 5 features scale=2  k-means 5 features scale=4   \n",
      "min_open_signal                            17.0                        17.0   \n",
      "max_open_signal                           393.0                       393.0   \n",
      "min_closed_signal                           0.0                         0.0   \n",
      "max_closed_signal                          16.0                        16.0   \n",
      "\n",
      "                                         Length                       Length  \\\n",
      "count                                    7301.0                       7301.0   \n",
      "mean                                 188.590604                   188.590604   \n",
      "std                                  174.998864                   174.998864   \n",
      "min                                       100.0                        100.0   \n",
      "25%                                       100.0                        100.0   \n",
      "50%                                       100.0                        100.0   \n",
      "75%                                       200.0                        200.0   \n",
      "max                                      2400.0                       2400.0   \n",
      "ATACseq_overlap                        0.697949                     0.697949   \n",
      "DNAseq_overlap                         0.595329                     0.595329   \n",
      "genes_overlap                          0.544833                     0.544833   \n",
      "percentage_open                        0.141691                     0.141691   \n",
      "percentage_open_adj                    0.204969                     0.204969   \n",
      "model_name           k-means 5 features scale=8  k-means 5 features scale=16   \n",
      "min_open_signal                            17.0                         17.0   \n",
      "max_open_signal                           393.0                        393.0   \n",
      "min_closed_signal                           0.0                          0.0   \n",
      "max_closed_signal                          16.0                         16.0   \n",
      "\n",
      "                                          Length  \n",
      "count                                     6456.0  \n",
      "mean                                  177.850062  \n",
      "std                                   157.257952  \n",
      "min                                        100.0  \n",
      "25%                                        100.0  \n",
      "50%                                        100.0  \n",
      "75%                                        200.0  \n",
      "max                                       2000.0  \n",
      "ATACseq_overlap                         0.668276  \n",
      "DNAseq_overlap                          0.553813  \n",
      "genes_overlap                           0.550809  \n",
      "percentage_open                         0.118157  \n",
      "percentage_open_adj                     0.170924  \n",
      "model_name           k-means 5 features scale=32  \n",
      "min_open_signal                             19.0  \n",
      "max_open_signal                            393.0  \n",
      "min_closed_signal                            0.0  \n",
      "max_closed_signal                           18.0  \n"
     ]
    }
   ],
   "source": [
    "print(result)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
