{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "d001e5b1",
   "metadata": {},
   "outputs": [],
   "source": [
    "import chromatin_algos\n",
    "import evaluator\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "4328129e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "HEAD\n",
      "  seqnames     start       end        CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       18  10000018  10000018  18_10000018    0.0    0.0    0.0\n",
      "1       18  10000117  10000117  18_10000117    0.0    0.0    0.0\n",
      "2       18  10000155  10000155  18_10000155    0.0    0.0    1.0\n",
      "3       18  10000222  10000222  18_10000222    0.0    4.0    0.0\n",
      "4       18  10000281  10000281  18_10000281    0.0    0.0    0.0\n",
      "BINS  IntervalIndex([(10000017, 10000017], (10000017, 10000117],\n",
      "               (10000117, 10000217], (10000217, 10000317],\n",
      "               (10000317, 10000417], (10000417, 10000517],\n",
      "               (10000517, 10000617], (10000617, 10000717],\n",
      "               (10000717, 10000817], (10000817, 10000917],\n",
      "               ...\n",
      "               (25556017, 25556117], (25556117, 25556217],\n",
      "               (25556217, 25556317], (25556317, 25556417],\n",
      "               (25556417, 25556517], (25556517, 25556617],\n",
      "               (25556617, 25556717], (25556717, 25556817],\n",
      "               (25556817, 25556917], (25556917, 25556940]],\n",
      "              dtype='interval[int64, right]', length=155571)\n",
      "HEAD w BINS\n",
      "  seqnames     start       end        CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       18  10000018  10000018  18_10000018    0.0    0.0    0.0   \n",
      "1       18  10000117  10000117  18_10000117    0.0    0.0    0.0   \n",
      "2       18  10000155  10000155  18_10000155    0.0    0.0    1.0   \n",
      "3       18  10000222  10000222  18_10000222    0.0    4.0    0.0   \n",
      "4       18  10000281  10000281  18_10000281    0.0    0.0    0.0   \n",
      "\n",
      "           bin_offset_0  \n",
      "0  (10000017, 10000117]  \n",
      "1  (10000017, 10000117]  \n",
      "2  (10000117, 10000217]  \n",
      "3  (10000217, 10000317]  \n",
      "4  (10000217, 10000317]  \n",
      "TAIL w BINS\n",
      "       seqnames     start       end        CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  25556879  25556879  18_25556879    1.0    9.0    3.0   \n",
      "699996       18  25556894  25556894  18_25556894    1.0   14.0    1.0   \n",
      "699997       18  25556919  25556919  18_25556919    0.0    0.0    0.0   \n",
      "699998       18  25556927  25556927  18_25556927    0.0    3.0    2.0   \n",
      "699999       18  25556940  25556940  18_25556940    7.0    5.0    7.0   \n",
      "\n",
      "                bin_offset_0  \n",
      "699995  (25556817, 25556917]  \n",
      "699996  (25556817, 25556917]  \n",
      "699997  (25556917, 25556940]  \n",
      "699998  (25556917, 25556940]  \n",
      "699999  (25556917, 25556940]  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/chromatin_algos.py:150: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "    TT_S0  starting_nt\n",
      "0     0.0     10000017\n",
      "1     0.0     10000017\n",
      "2     0.0     10000117\n",
      "3     0.0     10000217\n",
      "4    14.0     10000317\n",
      "5     3.0     10000417\n",
      "6     3.0     10000517\n",
      "7     0.0     10000617\n",
      "8    16.0     10000717\n",
      "9     2.0     10000817\n",
      "10    0.0     10000917\n",
      "11    0.0     10001017\n",
      "12    1.0     10001117\n",
      "13    1.0     10001217\n",
      "14    0.0     10001317\n",
      "15    4.0     10001417\n",
      "16    3.0     10001517\n",
      "17    0.0     10001617\n",
      "18    0.0     10001717\n",
      "19    1.0     10001817\n",
      "20    0.0     10001917\n",
      "21    6.0     10002017\n",
      "22   21.0     10002117\n",
      "23   14.0     10002217\n",
      "24    1.0     10002317\n",
      "25    0.0     10002417\n",
      "26    0.0     10002517\n",
      "27    2.0     10002617\n",
      "28   12.0     10002717\n",
      "29    9.0     10002817\n",
      "30    0.0     10002917\n",
      "31    1.0     10003017\n",
      "32    0.0     10003117\n",
      "33    7.0     10003217\n",
      "34    0.0     10003317\n",
      "35    4.0     10003417\n",
      "36    1.0     10003517\n",
      "37    0.0     10003617\n",
      "38    0.0     10003717\n",
      "39    1.0     10003817\n",
      "40    4.0     10003917\n",
      "41    1.0     10004017\n",
      "42    4.0     10004117\n",
      "43    1.0     10004217\n",
      "44    2.0     10004317\n",
      "45    7.0     10004417\n",
      "46   29.0     10004517\n",
      "47    1.0     10004617\n",
      "48   12.0     10004717\n",
      "49    8.0     10004817\n",
      "(155571, 2)\n",
      "TT_S0          float64\n",
      "starting_nt      int64\n",
      "dtype: object\n",
      "sumos ilgis: 155571\n",
      "ranges ilgis: 155571\n",
      "Model input - NO SCALE\n",
      "        TT_S0  genes  transcripts  exons  CDSs\n",
      "0         0.0      0            0      0     0\n",
      "1         0.0      0            0      0     0\n",
      "2         0.0      0            0      0     0\n",
      "3         0.0      0            0      0     0\n",
      "4        14.0      0            0      0     0\n",
      "...       ...    ...          ...    ...   ...\n",
      "155566   17.0      0            0      0     0\n",
      "155567    5.0      0            0      0     0\n",
      "155568   12.0      0            0      0     0\n",
      "155569    8.0      0            0      0     0\n",
      "155570    7.0      0            0      0     0\n",
      "\n",
      "[155571 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "        TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0         0.0             0                   0             0            0\n",
      "1         0.0             0                   0             0            0\n",
      "2         0.0             0                   0             0            0\n",
      "3         0.0             0                   0             0            0\n",
      "4        14.0             0                   0             0            0\n",
      "...       ...           ...                 ...           ...          ...\n",
      "155566   17.0             0                   0             0            0\n",
      "155567    5.0             0                   0             0            0\n",
      "155568   12.0             0                   0             0            0\n",
      "155569    8.0             0                   0             0            0\n",
      "155570    7.0             0                   0             0            0\n",
      "\n",
      "[155571 rows x 5 columns]\n",
      "0.10966696877952832\n",
      "[0 0 0 0 1 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 1 1 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0\n",
      " 0 0 1 0 0 0 0 0 1 0 0 0 1 0 0 0 0 0 0 0 0 1 0 0 1 0 1 0 0 0 0 1 0 1 0 0 0\n",
      " 0 0 1 0 0 1 1 0 0 0 0 0 0 0 1 0 0 0 0 0 0 1 0 0 0 0 0 0 1 1 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 0 0 0 0 0 1 0\n",
      " 1 0 1 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 0 0 0 0 0 0 0 1 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 1 0 0 1 0 0 0 0 0 1\n",
      " 0 0 1 0 1 0 0 1 1 1 0 0 0 0 0 0 0 0 0 1 0 0 0 1 1 0 0 0 0 0 0 0 0 1 0 1 1\n",
      " 1 1 1 1 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 1 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1\n",
      " 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0\n",
      " 0 0 1 1 0 0 1 1 1 1 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 1 1 1 0 0\n",
      " 0 0 0 0 0 0 0 1 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 1 0 1 1 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 1\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0]\n"
     ]
    }
   ],
   "source": [
    "result = evaluator.evaluate(dataframe = chromatin_algos.algo5d(1))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "94dc4e42",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "count                  11670.000000\n",
      "mean                     146.195373\n",
      "std                      109.606325\n",
      "min                      100.000000\n",
      "25%                      100.000000\n",
      "50%                      100.000000\n",
      "75%                      200.000000\n",
      "max                     1600.000000\n",
      "ATACseq_overlap            0.564254\n",
      "DNAseq_overlap             0.486887\n",
      "genes_overlap              0.650512\n",
      "percentage_open            0.066757\n",
      "percentage_open_adj        0.109668\n",
      "Name: Length, dtype: float64\n"
     ]
    }
   ],
   "source": [
    "print(result)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "28e1cdda",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Runnning scale 1\n",
      "HEAD\n",
      "  seqnames     start       end        CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       18  10000018  10000018  18_10000018    0.0    0.0    0.0\n",
      "1       18  10000117  10000117  18_10000117    0.0    0.0    0.0\n",
      "2       18  10000155  10000155  18_10000155    0.0    0.0    1.0\n",
      "3       18  10000222  10000222  18_10000222    0.0    4.0    0.0\n",
      "4       18  10000281  10000281  18_10000281    0.0    0.0    0.0\n",
      "BINS  IntervalIndex([(10000017, 10000017], (10000017, 10000117],\n",
      "               (10000117, 10000217], (10000217, 10000317],\n",
      "               (10000317, 10000417], (10000417, 10000517],\n",
      "               (10000517, 10000617], (10000617, 10000717],\n",
      "               (10000717, 10000817], (10000817, 10000917],\n",
      "               ...\n",
      "               (25556017, 25556117], (25556117, 25556217],\n",
      "               (25556217, 25556317], (25556317, 25556417],\n",
      "               (25556417, 25556517], (25556517, 25556617],\n",
      "               (25556617, 25556717], (25556717, 25556817],\n",
      "               (25556817, 25556917], (25556917, 25556940]],\n",
      "              dtype='interval[int64, right]', length=155571)\n",
      "HEAD w BINS\n",
      "  seqnames     start       end        CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       18  10000018  10000018  18_10000018    0.0    0.0    0.0   \n",
      "1       18  10000117  10000117  18_10000117    0.0    0.0    0.0   \n",
      "2       18  10000155  10000155  18_10000155    0.0    0.0    1.0   \n",
      "3       18  10000222  10000222  18_10000222    0.0    4.0    0.0   \n",
      "4       18  10000281  10000281  18_10000281    0.0    0.0    0.0   \n",
      "\n",
      "           bin_offset_0  \n",
      "0  (10000017, 10000117]  \n",
      "1  (10000017, 10000117]  \n",
      "2  (10000117, 10000217]  \n",
      "3  (10000217, 10000317]  \n",
      "4  (10000217, 10000317]  \n",
      "TAIL w BINS\n",
      "       seqnames     start       end        CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  25556879  25556879  18_25556879    1.0    9.0    3.0   \n",
      "699996       18  25556894  25556894  18_25556894    1.0   14.0    1.0   \n",
      "699997       18  25556919  25556919  18_25556919    0.0    0.0    0.0   \n",
      "699998       18  25556927  25556927  18_25556927    0.0    3.0    2.0   \n",
      "699999       18  25556940  25556940  18_25556940    7.0    5.0    7.0   \n",
      "\n",
      "                bin_offset_0  \n",
      "699995  (25556817, 25556917]  \n",
      "699996  (25556817, 25556917]  \n",
      "699997  (25556917, 25556940]  \n",
      "699998  (25556917, 25556940]  \n",
      "699999  (25556917, 25556940]  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/chromatin_algos.py:150: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "    TT_S0  starting_nt\n",
      "0     0.0     10000017\n",
      "1     0.0     10000017\n",
      "2     0.0     10000117\n",
      "3     0.0     10000217\n",
      "4    14.0     10000317\n",
      "5     3.0     10000417\n",
      "6     3.0     10000517\n",
      "7     0.0     10000617\n",
      "8    16.0     10000717\n",
      "9     2.0     10000817\n",
      "10    0.0     10000917\n",
      "11    0.0     10001017\n",
      "12    1.0     10001117\n",
      "13    1.0     10001217\n",
      "14    0.0     10001317\n",
      "15    4.0     10001417\n",
      "16    3.0     10001517\n",
      "17    0.0     10001617\n",
      "18    0.0     10001717\n",
      "19    1.0     10001817\n",
      "20    0.0     10001917\n",
      "21    6.0     10002017\n",
      "22   21.0     10002117\n",
      "23   14.0     10002217\n",
      "24    1.0     10002317\n",
      "25    0.0     10002417\n",
      "26    0.0     10002517\n",
      "27    2.0     10002617\n",
      "28   12.0     10002717\n",
      "29    9.0     10002817\n",
      "30    0.0     10002917\n",
      "31    1.0     10003017\n",
      "32    0.0     10003117\n",
      "33    7.0     10003217\n",
      "34    0.0     10003317\n",
      "35    4.0     10003417\n",
      "36    1.0     10003517\n",
      "37    0.0     10003617\n",
      "38    0.0     10003717\n",
      "39    1.0     10003817\n",
      "40    4.0     10003917\n",
      "41    1.0     10004017\n",
      "42    4.0     10004117\n",
      "43    1.0     10004217\n",
      "44    2.0     10004317\n",
      "45    7.0     10004417\n",
      "46   29.0     10004517\n",
      "47    1.0     10004617\n",
      "48   12.0     10004717\n",
      "49    8.0     10004817\n",
      "(155571, 2)\n",
      "TT_S0          float64\n",
      "starting_nt      int64\n",
      "dtype: object\n",
      "sumos ilgis: 155571\n",
      "ranges ilgis: 155571\n",
      "Model input - NO SCALE\n",
      "        TT_S0  genes  transcripts  exons  CDSs\n",
      "0         0.0      0            0      0     0\n",
      "1         0.0      0            0      0     0\n",
      "2         0.0      0            0      0     0\n",
      "3         0.0      0            0      0     0\n",
      "4        14.0      0            0      0     0\n",
      "...       ...    ...          ...    ...   ...\n",
      "155566   17.0      0            0      0     0\n",
      "155567    5.0      0            0      0     0\n",
      "155568   12.0      0            0      0     0\n",
      "155569    8.0      0            0      0     0\n",
      "155570    7.0      0            0      0     0\n",
      "\n",
      "[155571 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "        TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0         0.0             0                   0             0            0\n",
      "1         0.0             0                   0             0            0\n",
      "2         0.0             0                   0             0            0\n",
      "3         0.0             0                   0             0            0\n",
      "4        14.0             0                   0             0            0\n",
      "...       ...           ...                 ...           ...          ...\n",
      "155566   17.0             0                   0             0            0\n",
      "155567    5.0             0                   0             0            0\n",
      "155568   12.0             0                   0             0            0\n",
      "155569    8.0             0                   0             0            0\n",
      "155570    7.0             0                   0             0            0\n",
      "\n",
      "[155571 rows x 5 columns]\n",
      "0.13064131489802083\n",
      "[0 0 0 0 1 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 1 1 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 1 0 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 0\n",
      " 0 0 1 0 0 0 0 0 1 0 0 0 1 0 0 0 0 0 0 0 0 1 0 0 1 0 1 0 0 0 0 1 0 1 0 0 0\n",
      " 0 0 1 0 0 1 1 0 0 0 0 0 0 0 1 0 0 0 0 0 0 1 0 0 0 0 0 0 1 1 0 0 0 0 0 0 0\n",
      " 0 0 0 0 1 0 0 0 1 1 0 0 0 0 0 0 0 1 0 0 0 0 0 0 1 1 0 0 0 0 0 0 0 0 0 1 0\n",
      " 1 0 1 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 0 1 0 0 0 0 0 1 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 1 0 0 1 0 0 0 0 0 1\n",
      " 0 0 1 0 1 0 0 1 1 1 0 0 0 0 0 0 0 0 0 1 0 0 0 1 1 0 0 0 0 0 0 0 0 1 0 1 1\n",
      " 1 1 1 1 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 1 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1\n",
      " 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 1 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0\n",
      " 0 0 1 1 0 0 1 1 1 1 1 1 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 1 1 1 0 0\n",
      " 0 0 0 0 0 0 0 1 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 1 1 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 1 1 1 1 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 1 0 0 0 0 0 0 0 0 1\n",
      " 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0]\n",
      "Runnning scale 2\n",
      "HEAD\n",
      "  seqnames     start       end        CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       18  10000018  10000018  18_10000018    0.0    0.0    0.0\n",
      "1       18  10000117  10000117  18_10000117    0.0    0.0    0.0\n",
      "2       18  10000155  10000155  18_10000155    0.0    0.0    1.0\n",
      "3       18  10000222  10000222  18_10000222    0.0    4.0    0.0\n",
      "4       18  10000281  10000281  18_10000281    0.0    0.0    0.0\n",
      "BINS  IntervalIndex([(10000017, 10000017], (10000017, 10000117],\n",
      "               (10000117, 10000217], (10000217, 10000317],\n",
      "               (10000317, 10000417], (10000417, 10000517],\n",
      "               (10000517, 10000617], (10000617, 10000717],\n",
      "               (10000717, 10000817], (10000817, 10000917],\n",
      "               ...\n",
      "               (25556017, 25556117], (25556117, 25556217],\n",
      "               (25556217, 25556317], (25556317, 25556417],\n",
      "               (25556417, 25556517], (25556517, 25556617],\n",
      "               (25556617, 25556717], (25556717, 25556817],\n",
      "               (25556817, 25556917], (25556917, 25556940]],\n",
      "              dtype='interval[int64, right]', length=155571)\n",
      "HEAD w BINS\n",
      "  seqnames     start       end        CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       18  10000018  10000018  18_10000018    0.0    0.0    0.0   \n",
      "1       18  10000117  10000117  18_10000117    0.0    0.0    0.0   \n",
      "2       18  10000155  10000155  18_10000155    0.0    0.0    1.0   \n",
      "3       18  10000222  10000222  18_10000222    0.0    4.0    0.0   \n",
      "4       18  10000281  10000281  18_10000281    0.0    0.0    0.0   \n",
      "\n",
      "           bin_offset_0  \n",
      "0  (10000017, 10000117]  \n",
      "1  (10000017, 10000117]  \n",
      "2  (10000117, 10000217]  \n",
      "3  (10000217, 10000317]  \n",
      "4  (10000217, 10000317]  \n",
      "TAIL w BINS\n",
      "       seqnames     start       end        CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  25556879  25556879  18_25556879    1.0    9.0    3.0   \n",
      "699996       18  25556894  25556894  18_25556894    1.0   14.0    1.0   \n",
      "699997       18  25556919  25556919  18_25556919    0.0    0.0    0.0   \n",
      "699998       18  25556927  25556927  18_25556927    0.0    3.0    2.0   \n",
      "699999       18  25556940  25556940  18_25556940    7.0    5.0    7.0   \n",
      "\n",
      "                bin_offset_0  \n",
      "699995  (25556817, 25556917]  \n",
      "699996  (25556817, 25556917]  \n",
      "699997  (25556917, 25556940]  \n",
      "699998  (25556917, 25556940]  \n",
      "699999  (25556917, 25556940]  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/chromatin_algos.py:150: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "    TT_S0  starting_nt\n",
      "0     0.0     10000017\n",
      "1     0.0     10000017\n",
      "2     0.0     10000117\n",
      "3     0.0     10000217\n",
      "4    14.0     10000317\n",
      "5     3.0     10000417\n",
      "6     3.0     10000517\n",
      "7     0.0     10000617\n",
      "8    16.0     10000717\n",
      "9     2.0     10000817\n",
      "10    0.0     10000917\n",
      "11    0.0     10001017\n",
      "12    1.0     10001117\n",
      "13    1.0     10001217\n",
      "14    0.0     10001317\n",
      "15    4.0     10001417\n",
      "16    3.0     10001517\n",
      "17    0.0     10001617\n",
      "18    0.0     10001717\n",
      "19    1.0     10001817\n",
      "20    0.0     10001917\n",
      "21    6.0     10002017\n",
      "22   21.0     10002117\n",
      "23   14.0     10002217\n",
      "24    1.0     10002317\n",
      "25    0.0     10002417\n",
      "26    0.0     10002517\n",
      "27    2.0     10002617\n",
      "28   12.0     10002717\n",
      "29    9.0     10002817\n",
      "30    0.0     10002917\n",
      "31    1.0     10003017\n",
      "32    0.0     10003117\n",
      "33    7.0     10003217\n",
      "34    0.0     10003317\n",
      "35    4.0     10003417\n",
      "36    1.0     10003517\n",
      "37    0.0     10003617\n",
      "38    0.0     10003717\n",
      "39    1.0     10003817\n",
      "40    4.0     10003917\n",
      "41    1.0     10004017\n",
      "42    4.0     10004117\n",
      "43    1.0     10004217\n",
      "44    2.0     10004317\n",
      "45    7.0     10004417\n",
      "46   29.0     10004517\n",
      "47    1.0     10004617\n",
      "48   12.0     10004717\n",
      "49    8.0     10004817\n",
      "(155571, 2)\n",
      "TT_S0          float64\n",
      "starting_nt      int64\n",
      "dtype: object\n",
      "sumos ilgis: 155571\n",
      "ranges ilgis: 155571\n",
      "Model input - NO SCALE\n",
      "        TT_S0  genes  transcripts  exons  CDSs\n",
      "0         0.0      0            0      0     0\n",
      "1         0.0      0            0      0     0\n",
      "2         0.0      0            0      0     0\n",
      "3         0.0      0            0      0     0\n",
      "4        14.0      0            0      0     0\n",
      "...       ...    ...          ...    ...   ...\n",
      "155566   17.0      0            0      0     0\n",
      "155567    5.0      0            0      0     0\n",
      "155568   12.0      0            0      0     0\n",
      "155569    8.0      0            0      0     0\n",
      "155570    7.0      0            0      0     0\n",
      "\n",
      "[155571 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "        TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0         0.0             0                   0             0            0\n",
      "1         0.0             0                   0             0            0\n",
      "2         0.0             0                   0             0            0\n",
      "3         0.0             0                   0             0            0\n",
      "4        14.0             0                   0             0            0\n",
      "...       ...           ...                 ...           ...          ...\n",
      "155566   17.0             0                   0             0            0\n",
      "155567    5.0             0                   0             0            0\n",
      "155568   12.0             0                   0             0            0\n",
      "155569    8.0             0                   0             0            0\n",
      "155570    7.0             0                   0             0            0\n",
      "\n",
      "[155571 rows x 5 columns]\n",
      "0.8693586851019791\n",
      "[1 1 1 1 0 1 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 0 0 1 1 1 1 0 0 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 0 1 0 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 0 1 1 1 1 1\n",
      " 1 1 0 1 1 1 1 1 0 1 1 1 0 1 1 1 1 1 1 1 1 0 1 1 0 1 0 1 1 1 1 0 1 0 1 1 1\n",
      " 1 1 0 1 1 0 0 1 1 1 1 1 1 1 0 1 1 1 1 1 1 0 1 1 1 1 1 1 0 0 1 1 1 1 1 1 1\n",
      " 1 1 1 1 0 1 1 1 0 0 1 1 1 1 1 1 1 0 1 1 1 1 1 1 0 0 1 1 1 1 1 1 1 1 1 0 1\n",
      " 0 1 0 0 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 0 1 1 1 1 1 0 1 1 1 1 1 0 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 0 1 1 0 1 1 1 1 1 0\n",
      " 1 1 0 1 0 1 1 0 0 0 1 1 1 1 1 1 1 1 1 0 1 1 1 0 0 1 1 1 1 1 1 1 1 0 1 0 0\n",
      " 0 0 0 0 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 1 0 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0\n",
      " 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 0 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1\n",
      " 1 1 0 0 1 1 0 0 0 0 0 0 1 1 1 1 1 1 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 0 0 0 0 1 1\n",
      " 1 1 1 1 1 1 1 0 1 1 1 1 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 1 1 1 1 0 0 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 0 0 0 0 0 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 0 0 1 1 1 1 1 1 1 1 0\n",
      " 1 1 1 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1]\n",
      "Runnning scale 4\n",
      "HEAD\n",
      "  seqnames     start       end        CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       18  10000018  10000018  18_10000018    0.0    0.0    0.0\n",
      "1       18  10000117  10000117  18_10000117    0.0    0.0    0.0\n",
      "2       18  10000155  10000155  18_10000155    0.0    0.0    1.0\n",
      "3       18  10000222  10000222  18_10000222    0.0    4.0    0.0\n",
      "4       18  10000281  10000281  18_10000281    0.0    0.0    0.0\n",
      "BINS  IntervalIndex([(10000017, 10000017], (10000017, 10000117],\n",
      "               (10000117, 10000217], (10000217, 10000317],\n",
      "               (10000317, 10000417], (10000417, 10000517],\n",
      "               (10000517, 10000617], (10000617, 10000717],\n",
      "               (10000717, 10000817], (10000817, 10000917],\n",
      "               ...\n",
      "               (25556017, 25556117], (25556117, 25556217],\n",
      "               (25556217, 25556317], (25556317, 25556417],\n",
      "               (25556417, 25556517], (25556517, 25556617],\n",
      "               (25556617, 25556717], (25556717, 25556817],\n",
      "               (25556817, 25556917], (25556917, 25556940]],\n",
      "              dtype='interval[int64, right]', length=155571)\n",
      "HEAD w BINS\n",
      "  seqnames     start       end        CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       18  10000018  10000018  18_10000018    0.0    0.0    0.0   \n",
      "1       18  10000117  10000117  18_10000117    0.0    0.0    0.0   \n",
      "2       18  10000155  10000155  18_10000155    0.0    0.0    1.0   \n",
      "3       18  10000222  10000222  18_10000222    0.0    4.0    0.0   \n",
      "4       18  10000281  10000281  18_10000281    0.0    0.0    0.0   \n",
      "\n",
      "           bin_offset_0  \n",
      "0  (10000017, 10000117]  \n",
      "1  (10000017, 10000117]  \n",
      "2  (10000117, 10000217]  \n",
      "3  (10000217, 10000317]  \n",
      "4  (10000217, 10000317]  \n",
      "TAIL w BINS\n",
      "       seqnames     start       end        CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  25556879  25556879  18_25556879    1.0    9.0    3.0   \n",
      "699996       18  25556894  25556894  18_25556894    1.0   14.0    1.0   \n",
      "699997       18  25556919  25556919  18_25556919    0.0    0.0    0.0   \n",
      "699998       18  25556927  25556927  18_25556927    0.0    3.0    2.0   \n",
      "699999       18  25556940  25556940  18_25556940    7.0    5.0    7.0   \n",
      "\n",
      "                bin_offset_0  \n",
      "699995  (25556817, 25556917]  \n",
      "699996  (25556817, 25556917]  \n",
      "699997  (25556917, 25556940]  \n",
      "699998  (25556917, 25556940]  \n",
      "699999  (25556917, 25556940]  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/chromatin_algos.py:150: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "    TT_S0  starting_nt\n",
      "0     0.0     10000017\n",
      "1     0.0     10000017\n",
      "2     0.0     10000117\n",
      "3     0.0     10000217\n",
      "4    14.0     10000317\n",
      "5     3.0     10000417\n",
      "6     3.0     10000517\n",
      "7     0.0     10000617\n",
      "8    16.0     10000717\n",
      "9     2.0     10000817\n",
      "10    0.0     10000917\n",
      "11    0.0     10001017\n",
      "12    1.0     10001117\n",
      "13    1.0     10001217\n",
      "14    0.0     10001317\n",
      "15    4.0     10001417\n",
      "16    3.0     10001517\n",
      "17    0.0     10001617\n",
      "18    0.0     10001717\n",
      "19    1.0     10001817\n",
      "20    0.0     10001917\n",
      "21    6.0     10002017\n",
      "22   21.0     10002117\n",
      "23   14.0     10002217\n",
      "24    1.0     10002317\n",
      "25    0.0     10002417\n",
      "26    0.0     10002517\n",
      "27    2.0     10002617\n",
      "28   12.0     10002717\n",
      "29    9.0     10002817\n",
      "30    0.0     10002917\n",
      "31    1.0     10003017\n",
      "32    0.0     10003117\n",
      "33    7.0     10003217\n",
      "34    0.0     10003317\n",
      "35    4.0     10003417\n",
      "36    1.0     10003517\n",
      "37    0.0     10003617\n",
      "38    0.0     10003717\n",
      "39    1.0     10003817\n",
      "40    4.0     10003917\n",
      "41    1.0     10004017\n",
      "42    4.0     10004117\n",
      "43    1.0     10004217\n",
      "44    2.0     10004317\n",
      "45    7.0     10004417\n",
      "46   29.0     10004517\n",
      "47    1.0     10004617\n",
      "48   12.0     10004717\n",
      "49    8.0     10004817\n",
      "(155571, 2)\n",
      "TT_S0          float64\n",
      "starting_nt      int64\n",
      "dtype: object\n",
      "sumos ilgis: 155571\n",
      "ranges ilgis: 155571\n",
      "Model input - NO SCALE\n",
      "        TT_S0  genes  transcripts  exons  CDSs\n",
      "0         0.0      0            0      0     0\n",
      "1         0.0      0            0      0     0\n",
      "2         0.0      0            0      0     0\n",
      "3         0.0      0            0      0     0\n",
      "4        14.0      0            0      0     0\n",
      "...       ...    ...          ...    ...   ...\n",
      "155566   17.0      0            0      0     0\n",
      "155567    5.0      0            0      0     0\n",
      "155568   12.0      0            0      0     0\n",
      "155569    8.0      0            0      0     0\n",
      "155570    7.0      0            0      0     0\n",
      "\n",
      "[155571 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "        TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0         0.0             0                   0             0            0\n",
      "1         0.0             0                   0             0            0\n",
      "2         0.0             0                   0             0            0\n",
      "3         0.0             0                   0             0            0\n",
      "4        14.0             0                   0             0            0\n",
      "...       ...           ...                 ...           ...          ...\n",
      "155566   17.0             0                   0             0            0\n",
      "155567    5.0             0                   0             0            0\n",
      "155568   12.0             0                   0             0            0\n",
      "155569    8.0             0                   0             0            0\n",
      "155570    7.0             0                   0             0            0\n",
      "\n",
      "[155571 rows x 5 columns]\n",
      "0.13064131489802083\n",
      "[0 0 0 0 1 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 1 1 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 1 0 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 0\n",
      " 0 0 1 0 0 0 0 0 1 0 0 0 1 0 0 0 0 0 0 0 0 1 0 0 1 0 1 0 0 0 0 1 0 1 0 0 0\n",
      " 0 0 1 0 0 1 1 0 0 0 0 0 0 0 1 0 0 0 0 0 0 1 0 0 0 0 0 0 1 1 0 0 0 0 0 0 0\n",
      " 0 0 0 0 1 0 0 0 1 1 0 0 0 0 0 0 0 1 0 0 0 0 0 0 1 1 0 0 0 0 0 0 0 0 0 1 0\n",
      " 1 0 1 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 0 1 0 0 0 0 0 1 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 1 0 0 1 0 0 0 0 0 1\n",
      " 0 0 1 0 1 0 0 1 1 1 0 0 0 0 0 0 0 0 0 1 0 0 0 1 1 0 0 0 0 0 0 0 0 1 0 1 1\n",
      " 1 1 1 1 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 1 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1\n",
      " 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 1 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0\n",
      " 0 0 1 1 0 0 1 1 1 1 1 1 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 1 1 1 0 0\n",
      " 0 0 0 0 0 0 0 1 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 1 1 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 1 1 1 1 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 1 0 0 0 0 0 0 0 0 1\n",
      " 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0]\n",
      "Runnning scale 8\n",
      "HEAD\n",
      "  seqnames     start       end        CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       18  10000018  10000018  18_10000018    0.0    0.0    0.0\n",
      "1       18  10000117  10000117  18_10000117    0.0    0.0    0.0\n",
      "2       18  10000155  10000155  18_10000155    0.0    0.0    1.0\n",
      "3       18  10000222  10000222  18_10000222    0.0    4.0    0.0\n",
      "4       18  10000281  10000281  18_10000281    0.0    0.0    0.0\n",
      "BINS  IntervalIndex([(10000017, 10000017], (10000017, 10000117],\n",
      "               (10000117, 10000217], (10000217, 10000317],\n",
      "               (10000317, 10000417], (10000417, 10000517],\n",
      "               (10000517, 10000617], (10000617, 10000717],\n",
      "               (10000717, 10000817], (10000817, 10000917],\n",
      "               ...\n",
      "               (25556017, 25556117], (25556117, 25556217],\n",
      "               (25556217, 25556317], (25556317, 25556417],\n",
      "               (25556417, 25556517], (25556517, 25556617],\n",
      "               (25556617, 25556717], (25556717, 25556817],\n",
      "               (25556817, 25556917], (25556917, 25556940]],\n",
      "              dtype='interval[int64, right]', length=155571)\n",
      "HEAD w BINS\n",
      "  seqnames     start       end        CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       18  10000018  10000018  18_10000018    0.0    0.0    0.0   \n",
      "1       18  10000117  10000117  18_10000117    0.0    0.0    0.0   \n",
      "2       18  10000155  10000155  18_10000155    0.0    0.0    1.0   \n",
      "3       18  10000222  10000222  18_10000222    0.0    4.0    0.0   \n",
      "4       18  10000281  10000281  18_10000281    0.0    0.0    0.0   \n",
      "\n",
      "           bin_offset_0  \n",
      "0  (10000017, 10000117]  \n",
      "1  (10000017, 10000117]  \n",
      "2  (10000117, 10000217]  \n",
      "3  (10000217, 10000317]  \n",
      "4  (10000217, 10000317]  \n",
      "TAIL w BINS\n",
      "       seqnames     start       end        CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  25556879  25556879  18_25556879    1.0    9.0    3.0   \n",
      "699996       18  25556894  25556894  18_25556894    1.0   14.0    1.0   \n",
      "699997       18  25556919  25556919  18_25556919    0.0    0.0    0.0   \n",
      "699998       18  25556927  25556927  18_25556927    0.0    3.0    2.0   \n",
      "699999       18  25556940  25556940  18_25556940    7.0    5.0    7.0   \n",
      "\n",
      "                bin_offset_0  \n",
      "699995  (25556817, 25556917]  \n",
      "699996  (25556817, 25556917]  \n",
      "699997  (25556917, 25556940]  \n",
      "699998  (25556917, 25556940]  \n",
      "699999  (25556917, 25556940]  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/chromatin_algos.py:150: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "    TT_S0  starting_nt\n",
      "0     0.0     10000017\n",
      "1     0.0     10000017\n",
      "2     0.0     10000117\n",
      "3     0.0     10000217\n",
      "4    14.0     10000317\n",
      "5     3.0     10000417\n",
      "6     3.0     10000517\n",
      "7     0.0     10000617\n",
      "8    16.0     10000717\n",
      "9     2.0     10000817\n",
      "10    0.0     10000917\n",
      "11    0.0     10001017\n",
      "12    1.0     10001117\n",
      "13    1.0     10001217\n",
      "14    0.0     10001317\n",
      "15    4.0     10001417\n",
      "16    3.0     10001517\n",
      "17    0.0     10001617\n",
      "18    0.0     10001717\n",
      "19    1.0     10001817\n",
      "20    0.0     10001917\n",
      "21    6.0     10002017\n",
      "22   21.0     10002117\n",
      "23   14.0     10002217\n",
      "24    1.0     10002317\n",
      "25    0.0     10002417\n",
      "26    0.0     10002517\n",
      "27    2.0     10002617\n",
      "28   12.0     10002717\n",
      "29    9.0     10002817\n",
      "30    0.0     10002917\n",
      "31    1.0     10003017\n",
      "32    0.0     10003117\n",
      "33    7.0     10003217\n",
      "34    0.0     10003317\n",
      "35    4.0     10003417\n",
      "36    1.0     10003517\n",
      "37    0.0     10003617\n",
      "38    0.0     10003717\n",
      "39    1.0     10003817\n",
      "40    4.0     10003917\n",
      "41    1.0     10004017\n",
      "42    4.0     10004117\n",
      "43    1.0     10004217\n",
      "44    2.0     10004317\n",
      "45    7.0     10004417\n",
      "46   29.0     10004517\n",
      "47    1.0     10004617\n",
      "48   12.0     10004717\n",
      "49    8.0     10004817\n",
      "(155571, 2)\n",
      "TT_S0          float64\n",
      "starting_nt      int64\n",
      "dtype: object\n",
      "sumos ilgis: 155571\n",
      "ranges ilgis: 155571\n",
      "Model input - NO SCALE\n",
      "        TT_S0  genes  transcripts  exons  CDSs\n",
      "0         0.0      0            0      0     0\n",
      "1         0.0      0            0      0     0\n",
      "2         0.0      0            0      0     0\n",
      "3         0.0      0            0      0     0\n",
      "4        14.0      0            0      0     0\n",
      "...       ...    ...          ...    ...   ...\n",
      "155566   17.0      0            0      0     0\n",
      "155567    5.0      0            0      0     0\n",
      "155568   12.0      0            0      0     0\n",
      "155569    8.0      0            0      0     0\n",
      "155570    7.0      0            0      0     0\n",
      "\n",
      "[155571 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "        TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0         0.0             0                   0             0            0\n",
      "1         0.0             0                   0             0            0\n",
      "2         0.0             0                   0             0            0\n",
      "3         0.0             0                   0             0            0\n",
      "4        14.0             0                   0             0            0\n",
      "...       ...           ...                 ...           ...          ...\n",
      "155566   17.0             0                   0             0            0\n",
      "155567    5.0             0                   0             0            0\n",
      "155568   12.0             0                   0             0            0\n",
      "155569    8.0             0                   0             0            0\n",
      "155570    7.0             0                   0             0            0\n",
      "\n",
      "[155571 rows x 5 columns]\n",
      "0.10966696877952832\n",
      "[0 0 0 0 1 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 1 1 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0\n",
      " 0 0 1 0 0 0 0 0 1 0 0 0 1 0 0 0 0 0 0 0 0 1 0 0 1 0 1 0 0 0 0 1 0 1 0 0 0\n",
      " 0 0 1 0 0 1 1 0 0 0 0 0 0 0 1 0 0 0 0 0 0 1 0 0 0 0 0 0 1 1 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 0 0 0 0 0 1 0\n",
      " 1 0 1 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 0 0 0 0 0 0 0 1 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 1 0 0 1 0 0 0 0 0 1\n",
      " 0 0 1 0 1 0 0 1 1 1 0 0 0 0 0 0 0 0 0 1 0 0 0 1 1 0 0 0 0 0 0 0 0 1 0 1 1\n",
      " 1 1 1 1 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 1 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1\n",
      " 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0\n",
      " 0 0 1 1 0 0 1 1 1 1 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 1 1 1 0 0\n",
      " 0 0 0 0 0 0 0 1 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 1 0 1 1 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 1\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0]\n",
      "Runnning scale 16\n",
      "HEAD\n",
      "  seqnames     start       end        CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       18  10000018  10000018  18_10000018    0.0    0.0    0.0\n",
      "1       18  10000117  10000117  18_10000117    0.0    0.0    0.0\n",
      "2       18  10000155  10000155  18_10000155    0.0    0.0    1.0\n",
      "3       18  10000222  10000222  18_10000222    0.0    4.0    0.0\n",
      "4       18  10000281  10000281  18_10000281    0.0    0.0    0.0\n",
      "BINS  IntervalIndex([(10000017, 10000017], (10000017, 10000117],\n",
      "               (10000117, 10000217], (10000217, 10000317],\n",
      "               (10000317, 10000417], (10000417, 10000517],\n",
      "               (10000517, 10000617], (10000617, 10000717],\n",
      "               (10000717, 10000817], (10000817, 10000917],\n",
      "               ...\n",
      "               (25556017, 25556117], (25556117, 25556217],\n",
      "               (25556217, 25556317], (25556317, 25556417],\n",
      "               (25556417, 25556517], (25556517, 25556617],\n",
      "               (25556617, 25556717], (25556717, 25556817],\n",
      "               (25556817, 25556917], (25556917, 25556940]],\n",
      "              dtype='interval[int64, right]', length=155571)\n",
      "HEAD w BINS\n",
      "  seqnames     start       end        CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       18  10000018  10000018  18_10000018    0.0    0.0    0.0   \n",
      "1       18  10000117  10000117  18_10000117    0.0    0.0    0.0   \n",
      "2       18  10000155  10000155  18_10000155    0.0    0.0    1.0   \n",
      "3       18  10000222  10000222  18_10000222    0.0    4.0    0.0   \n",
      "4       18  10000281  10000281  18_10000281    0.0    0.0    0.0   \n",
      "\n",
      "           bin_offset_0  \n",
      "0  (10000017, 10000117]  \n",
      "1  (10000017, 10000117]  \n",
      "2  (10000117, 10000217]  \n",
      "3  (10000217, 10000317]  \n",
      "4  (10000217, 10000317]  \n",
      "TAIL w BINS\n",
      "       seqnames     start       end        CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  25556879  25556879  18_25556879    1.0    9.0    3.0   \n",
      "699996       18  25556894  25556894  18_25556894    1.0   14.0    1.0   \n",
      "699997       18  25556919  25556919  18_25556919    0.0    0.0    0.0   \n",
      "699998       18  25556927  25556927  18_25556927    0.0    3.0    2.0   \n",
      "699999       18  25556940  25556940  18_25556940    7.0    5.0    7.0   \n",
      "\n",
      "                bin_offset_0  \n",
      "699995  (25556817, 25556917]  \n",
      "699996  (25556817, 25556917]  \n",
      "699997  (25556917, 25556940]  \n",
      "699998  (25556917, 25556940]  \n",
      "699999  (25556917, 25556940]  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/chromatin_algos.py:150: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "    TT_S0  starting_nt\n",
      "0     0.0     10000017\n",
      "1     0.0     10000017\n",
      "2     0.0     10000117\n",
      "3     0.0     10000217\n",
      "4    14.0     10000317\n",
      "5     3.0     10000417\n",
      "6     3.0     10000517\n",
      "7     0.0     10000617\n",
      "8    16.0     10000717\n",
      "9     2.0     10000817\n",
      "10    0.0     10000917\n",
      "11    0.0     10001017\n",
      "12    1.0     10001117\n",
      "13    1.0     10001217\n",
      "14    0.0     10001317\n",
      "15    4.0     10001417\n",
      "16    3.0     10001517\n",
      "17    0.0     10001617\n",
      "18    0.0     10001717\n",
      "19    1.0     10001817\n",
      "20    0.0     10001917\n",
      "21    6.0     10002017\n",
      "22   21.0     10002117\n",
      "23   14.0     10002217\n",
      "24    1.0     10002317\n",
      "25    0.0     10002417\n",
      "26    0.0     10002517\n",
      "27    2.0     10002617\n",
      "28   12.0     10002717\n",
      "29    9.0     10002817\n",
      "30    0.0     10002917\n",
      "31    1.0     10003017\n",
      "32    0.0     10003117\n",
      "33    7.0     10003217\n",
      "34    0.0     10003317\n",
      "35    4.0     10003417\n",
      "36    1.0     10003517\n",
      "37    0.0     10003617\n",
      "38    0.0     10003717\n",
      "39    1.0     10003817\n",
      "40    4.0     10003917\n",
      "41    1.0     10004017\n",
      "42    4.0     10004117\n",
      "43    1.0     10004217\n",
      "44    2.0     10004317\n",
      "45    7.0     10004417\n",
      "46   29.0     10004517\n",
      "47    1.0     10004617\n",
      "48   12.0     10004717\n",
      "49    8.0     10004817\n",
      "(155571, 2)\n",
      "TT_S0          float64\n",
      "starting_nt      int64\n",
      "dtype: object\n",
      "sumos ilgis: 155571\n",
      "ranges ilgis: 155571\n",
      "Model input - NO SCALE\n",
      "        TT_S0  genes  transcripts  exons  CDSs\n",
      "0         0.0      0            0      0     0\n",
      "1         0.0      0            0      0     0\n",
      "2         0.0      0            0      0     0\n",
      "3         0.0      0            0      0     0\n",
      "4        14.0      0            0      0     0\n",
      "...       ...    ...          ...    ...   ...\n",
      "155566   17.0      0            0      0     0\n",
      "155567    5.0      0            0      0     0\n",
      "155568   12.0      0            0      0     0\n",
      "155569    8.0      0            0      0     0\n",
      "155570    7.0      0            0      0     0\n",
      "\n",
      "[155571 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "        TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0         0.0             0                   0             0            0\n",
      "1         0.0             0                   0             0            0\n",
      "2         0.0             0                   0             0            0\n",
      "3         0.0             0                   0             0            0\n",
      "4        14.0             0                   0             0            0\n",
      "...       ...           ...                 ...           ...          ...\n",
      "155566   17.0             0                   0             0            0\n",
      "155567    5.0             0                   0             0            0\n",
      "155568   12.0             0                   0             0            0\n",
      "155569    8.0             0                   0             0            0\n",
      "155570    7.0             0                   0             0            0\n",
      "\n",
      "[155571 rows x 5 columns]\n",
      "0.8693586851019791\n",
      "[1 1 1 1 0 1 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 0 0 1 1 1 1 0 0 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 0 1 0 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 0 1 1 1 1 1\n",
      " 1 1 0 1 1 1 1 1 0 1 1 1 0 1 1 1 1 1 1 1 1 0 1 1 0 1 0 1 1 1 1 0 1 0 1 1 1\n",
      " 1 1 0 1 1 0 0 1 1 1 1 1 1 1 0 1 1 1 1 1 1 0 1 1 1 1 1 1 0 0 1 1 1 1 1 1 1\n",
      " 1 1 1 1 0 1 1 1 0 0 1 1 1 1 1 1 1 0 1 1 1 1 1 1 0 0 1 1 1 1 1 1 1 1 1 0 1\n",
      " 0 1 0 0 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 0 1 1 1 1 1 0 1 1 1 1 1 0 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 0 1 1 0 1 1 1 1 1 0\n",
      " 1 1 0 1 0 1 1 0 0 0 1 1 1 1 1 1 1 1 1 0 1 1 1 0 0 1 1 1 1 1 1 1 1 0 1 0 0\n",
      " 0 0 0 0 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 1 0 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0\n",
      " 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 0 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1\n",
      " 1 1 0 0 1 1 0 0 0 0 0 0 1 1 1 1 1 1 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 0 0 0 0 1 1\n",
      " 1 1 1 1 1 1 1 0 1 1 1 1 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 1 1 1 1 0 0 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 0 0 0 0 0 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 0 0 1 1 1 1 1 1 1 1 0\n",
      " 1 1 1 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1]\n",
      "Runnning scale 32\n",
      "HEAD\n",
      "  seqnames     start       end        CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       18  10000018  10000018  18_10000018    0.0    0.0    0.0\n",
      "1       18  10000117  10000117  18_10000117    0.0    0.0    0.0\n",
      "2       18  10000155  10000155  18_10000155    0.0    0.0    1.0\n",
      "3       18  10000222  10000222  18_10000222    0.0    4.0    0.0\n",
      "4       18  10000281  10000281  18_10000281    0.0    0.0    0.0\n",
      "BINS  IntervalIndex([(10000017, 10000017], (10000017, 10000117],\n",
      "               (10000117, 10000217], (10000217, 10000317],\n",
      "               (10000317, 10000417], (10000417, 10000517],\n",
      "               (10000517, 10000617], (10000617, 10000717],\n",
      "               (10000717, 10000817], (10000817, 10000917],\n",
      "               ...\n",
      "               (25556017, 25556117], (25556117, 25556217],\n",
      "               (25556217, 25556317], (25556317, 25556417],\n",
      "               (25556417, 25556517], (25556517, 25556617],\n",
      "               (25556617, 25556717], (25556717, 25556817],\n",
      "               (25556817, 25556917], (25556917, 25556940]],\n",
      "              dtype='interval[int64, right]', length=155571)\n",
      "HEAD w BINS\n",
      "  seqnames     start       end        CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       18  10000018  10000018  18_10000018    0.0    0.0    0.0   \n",
      "1       18  10000117  10000117  18_10000117    0.0    0.0    0.0   \n",
      "2       18  10000155  10000155  18_10000155    0.0    0.0    1.0   \n",
      "3       18  10000222  10000222  18_10000222    0.0    4.0    0.0   \n",
      "4       18  10000281  10000281  18_10000281    0.0    0.0    0.0   \n",
      "\n",
      "           bin_offset_0  \n",
      "0  (10000017, 10000117]  \n",
      "1  (10000017, 10000117]  \n",
      "2  (10000117, 10000217]  \n",
      "3  (10000217, 10000317]  \n",
      "4  (10000217, 10000317]  \n",
      "TAIL w BINS\n",
      "       seqnames     start       end        CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  25556879  25556879  18_25556879    1.0    9.0    3.0   \n",
      "699996       18  25556894  25556894  18_25556894    1.0   14.0    1.0   \n",
      "699997       18  25556919  25556919  18_25556919    0.0    0.0    0.0   \n",
      "699998       18  25556927  25556927  18_25556927    0.0    3.0    2.0   \n",
      "699999       18  25556940  25556940  18_25556940    7.0    5.0    7.0   \n",
      "\n",
      "                bin_offset_0  \n",
      "699995  (25556817, 25556917]  \n",
      "699996  (25556817, 25556917]  \n",
      "699997  (25556917, 25556940]  \n",
      "699998  (25556917, 25556940]  \n",
      "699999  (25556917, 25556940]  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/chromatin_algos.py:150: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "    TT_S0  starting_nt\n",
      "0     0.0     10000017\n",
      "1     0.0     10000017\n",
      "2     0.0     10000117\n",
      "3     0.0     10000217\n",
      "4    14.0     10000317\n",
      "5     3.0     10000417\n",
      "6     3.0     10000517\n",
      "7     0.0     10000617\n",
      "8    16.0     10000717\n",
      "9     2.0     10000817\n",
      "10    0.0     10000917\n",
      "11    0.0     10001017\n",
      "12    1.0     10001117\n",
      "13    1.0     10001217\n",
      "14    0.0     10001317\n",
      "15    4.0     10001417\n",
      "16    3.0     10001517\n",
      "17    0.0     10001617\n",
      "18    0.0     10001717\n",
      "19    1.0     10001817\n",
      "20    0.0     10001917\n",
      "21    6.0     10002017\n",
      "22   21.0     10002117\n",
      "23   14.0     10002217\n",
      "24    1.0     10002317\n",
      "25    0.0     10002417\n",
      "26    0.0     10002517\n",
      "27    2.0     10002617\n",
      "28   12.0     10002717\n",
      "29    9.0     10002817\n",
      "30    0.0     10002917\n",
      "31    1.0     10003017\n",
      "32    0.0     10003117\n",
      "33    7.0     10003217\n",
      "34    0.0     10003317\n",
      "35    4.0     10003417\n",
      "36    1.0     10003517\n",
      "37    0.0     10003617\n",
      "38    0.0     10003717\n",
      "39    1.0     10003817\n",
      "40    4.0     10003917\n",
      "41    1.0     10004017\n",
      "42    4.0     10004117\n",
      "43    1.0     10004217\n",
      "44    2.0     10004317\n",
      "45    7.0     10004417\n",
      "46   29.0     10004517\n",
      "47    1.0     10004617\n",
      "48   12.0     10004717\n",
      "49    8.0     10004817\n",
      "(155571, 2)\n",
      "TT_S0          float64\n",
      "starting_nt      int64\n",
      "dtype: object\n",
      "sumos ilgis: 155571\n",
      "ranges ilgis: 155571\n",
      "Model input - NO SCALE\n",
      "        TT_S0  genes  transcripts  exons  CDSs\n",
      "0         0.0      0            0      0     0\n",
      "1         0.0      0            0      0     0\n",
      "2         0.0      0            0      0     0\n",
      "3         0.0      0            0      0     0\n",
      "4        14.0      0            0      0     0\n",
      "...       ...    ...          ...    ...   ...\n",
      "155566   17.0      0            0      0     0\n",
      "155567    5.0      0            0      0     0\n",
      "155568   12.0      0            0      0     0\n",
      "155569    8.0      0            0      0     0\n",
      "155570    7.0      0            0      0     0\n",
      "\n",
      "[155571 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "        TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0         0.0             0                   0             0            0\n",
      "1         0.0             0                   0             0            0\n",
      "2         0.0             0                   0             0            0\n",
      "3         0.0             0                   0             0            0\n",
      "4        14.0             0                   0             0            0\n",
      "...       ...           ...                 ...           ...          ...\n",
      "155566   17.0             0                   0             0            0\n",
      "155567    5.0             0                   0             0            0\n",
      "155568   12.0             0                   0             0            0\n",
      "155569    8.0             0                   0             0            0\n",
      "155570    7.0             0                   0             0            0\n",
      "\n",
      "[155571 rows x 5 columns]\n",
      "0.8693586851019791\n",
      "[1 1 1 1 0 1 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 0 0 1 1 1 1 0 0 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 0 1 0 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 0 1 1 1 1 1\n",
      " 1 1 0 1 1 1 1 1 0 1 1 1 0 1 1 1 1 1 1 1 1 0 1 1 0 1 0 1 1 1 1 0 1 0 1 1 1\n",
      " 1 1 0 1 1 0 0 1 1 1 1 1 1 1 0 1 1 1 1 1 1 0 1 1 1 1 1 1 0 0 1 1 1 1 1 1 1\n",
      " 1 1 1 1 0 1 1 1 0 0 1 1 1 1 1 1 1 0 1 1 1 1 1 1 0 0 1 1 1 1 1 1 1 1 1 0 1\n",
      " 0 1 0 0 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 0 1 1 1 1 1 0 1 1 1 1 1 0 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 0 1 1 0 1 1 1 1 1 0\n",
      " 1 1 0 1 0 1 1 0 0 0 1 1 1 1 1 1 1 1 1 0 1 1 1 0 0 1 1 1 1 1 1 1 1 0 1 0 0\n",
      " 0 0 0 0 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 1 0 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0\n",
      " 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 0 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1\n",
      " 1 1 0 0 1 1 0 0 0 0 0 0 1 1 1 1 1 1 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 0 0 0 0 1 1\n",
      " 1 1 1 1 1 1 1 0 1 1 1 1 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 1 1 1 1 0 0 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1 1 1 0 0 0 0 0 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 0 0 1 1 1 1 1 1 1 1 0\n",
      " 1 1 1 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1\n",
      " 1]\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "for i in range(6):\n",
    "    scale = 2 ** i\n",
    "    print(\"Runnning scale\", scale)\n",
    "    evaluation = evaluator.evaluate(dataframe = chromatin_algos.algo5d(scale))\n",
    "    result = pd.concat([result, evaluation], axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "ca1280ed",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                           Length        Length        Length        Length  \\\n",
      "count                11670.000000  13484.000000  13484.000000  13484.000000   \n",
      "mean                   146.195373    150.726787    150.726787    150.726787   \n",
      "std                    109.606325    116.833191    116.833191    116.833191   \n",
      "min                    100.000000    100.000000    100.000000    100.000000   \n",
      "25%                    100.000000    100.000000    100.000000    100.000000   \n",
      "50%                    100.000000    100.000000    100.000000    100.000000   \n",
      "75%                    200.000000    200.000000    200.000000    200.000000   \n",
      "max                   1600.000000   1800.000000   1800.000000   1800.000000   \n",
      "ATACseq_overlap          0.564254      0.583832      0.583832      0.583832   \n",
      "DNAseq_overlap           0.486887      0.510847      0.510847      0.510847   \n",
      "genes_overlap            0.650512      0.648027      0.648027      0.648027   \n",
      "percentage_open          0.066757      0.079524      0.079524      0.079524   \n",
      "percentage_open_adj      0.109668      0.130643      0.130643      0.130643   \n",
      "\n",
      "                           Length        Length        Length  \n",
      "count                11670.000000  13484.000000  13484.000000  \n",
      "mean                   146.195373    150.726787    150.726787  \n",
      "std                    109.606325    116.833191    116.833191  \n",
      "min                    100.000000    100.000000    100.000000  \n",
      "25%                    100.000000    100.000000    100.000000  \n",
      "50%                    100.000000    100.000000    100.000000  \n",
      "75%                    200.000000    200.000000    200.000000  \n",
      "max                   1600.000000   1800.000000   1800.000000  \n",
      "ATACseq_overlap          0.564254      0.583832      0.583832  \n",
      "DNAseq_overlap           0.486887      0.510847      0.510847  \n",
      "genes_overlap            0.650512      0.648027      0.648027  \n",
      "percentage_open          0.066757      0.079524      0.079524  \n",
      "percentage_open_adj      0.109668      0.130643      0.130643  \n"
     ]
    }
   ],
   "source": [
    "print(result)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
