{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "d001e5b1",
   "metadata": {},
   "outputs": [],
   "source": [
    "import chromatin_algos\n",
    "import evaluator"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "4328129e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0\n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "BINS  IntervalIndex([(3000016, 3000016], (3000016, 3000116], (3000116, 3000216],\n",
      "               (3000216, 3000316], (3000316, 3000416], (3000416, 3000516],\n",
      "               (3000516, 3000616], (3000616, 3000716], (3000716, 3000816],\n",
      "               (3000816, 3000916],\n",
      "               ...\n",
      "               (9716616, 9716716], (9716716, 9716816], (9716816, 9716916],\n",
      "               (9716916, 9717016], (9717016, 9717116], (9717116, 9717216],\n",
      "               (9717216, 9717316], (9717316, 9717416], (9717416, 9717516],\n",
      "               (9717516, 9717593]],\n",
      "              dtype='interval[int64, right]', length=67177)\n",
      "HEAD with BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0   \n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "\n",
      "         bin_offset_0  \n",
      "0  (3000016, 3000116]  \n",
      "1  (3000016, 3000116]  \n",
      "2  (3000016, 3000116]  \n",
      "3  (3000016, 3000116]  \n",
      "4  (3000016, 3000116]  \n",
      "TAIL with BINS\n",
      "       seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  9717555  9717555  18_9717555    1.0    0.0    0.0   \n",
      "699996       19  9717571  9717571  19_9717571    0.0    0.0    0.0   \n",
      "699997       19  9717581  9717581  19_9717581    0.0    0.0    0.0   \n",
      "699998       18  9717582  9717582  18_9717582    0.0    0.0    0.0   \n",
      "699999       18  9717593  9717593  18_9717593    0.0    0.0    0.0   \n",
      "\n",
      "              bin_offset_0  \n",
      "699995  (9717516, 9717593]  \n",
      "699996  (9717516, 9717593]  \n",
      "699997  (9717516, 9717593]  \n",
      "699998  (9717516, 9717593]  \n",
      "699999  (9717516, 9717593]  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:136: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "          bin_offset_0  TT_S0  starting_nt\n",
      "0   (3000016, 3000016]    0.0      3000016\n",
      "1   (3000016, 3000116]    7.0      3000016\n",
      "2   (3000116, 3000216]    1.0      3000116\n",
      "3   (3000216, 3000316]    2.0      3000216\n",
      "4   (3000316, 3000416]    0.0      3000316\n",
      "5   (3000416, 3000516]    0.0      3000416\n",
      "6   (3000516, 3000616]    0.0      3000516\n",
      "7   (3000616, 3000716]    0.0      3000616\n",
      "8   (3000716, 3000816]    0.0      3000716\n",
      "9   (3000816, 3000916]    0.0      3000816\n",
      "10  (3000916, 3001016]    0.0      3000916\n",
      "11  (3001016, 3001116]    2.0      3001016\n",
      "12  (3001116, 3001216]    0.0      3001116\n",
      "13  (3001216, 3001316]    0.0      3001216\n",
      "14  (3001316, 3001416]    0.0      3001316\n",
      "15  (3001416, 3001516]    1.0      3001416\n",
      "16  (3001516, 3001616]    3.0      3001516\n",
      "17  (3001616, 3001716]    3.0      3001616\n",
      "18  (3001716, 3001816]    0.0      3001716\n",
      "19  (3001816, 3001916]    0.0      3001816\n",
      "20  (3001916, 3002016]    1.0      3001916\n",
      "21  (3002016, 3002116]    0.0      3002016\n",
      "22  (3002116, 3002216]    1.0      3002116\n",
      "23  (3002216, 3002316]   10.0      3002216\n",
      "24  (3002316, 3002416]    2.0      3002316\n",
      "25  (3002416, 3002516]   19.0      3002416\n",
      "26  (3002516, 3002616]    4.0      3002516\n",
      "27  (3002616, 3002716]    0.0      3002616\n",
      "28  (3002716, 3002816]    0.0      3002716\n",
      "29  (3002816, 3002916]    0.0      3002816\n",
      "30  (3002916, 3003016]    1.0      3002916\n",
      "31  (3003016, 3003116]    2.0      3003016\n",
      "32  (3003116, 3003216]    0.0      3003116\n",
      "33  (3003216, 3003316]    0.0      3003216\n",
      "34  (3003316, 3003416]    1.0      3003316\n",
      "35  (3003416, 3003516]    0.0      3003416\n",
      "36  (3003516, 3003616]    1.0      3003516\n",
      "37  (3003616, 3003716]    1.0      3003616\n",
      "38  (3003716, 3003816]    3.0      3003716\n",
      "39  (3003816, 3003916]    0.0      3003816\n",
      "40  (3003916, 3004016]    8.0      3003916\n",
      "41  (3004016, 3004116]    0.0      3004016\n",
      "42  (3004116, 3004216]    0.0      3004116\n",
      "43  (3004216, 3004316]    5.0      3004216\n",
      "44  (3004316, 3004416]    0.0      3004316\n",
      "45  (3004416, 3004516]    0.0      3004416\n",
      "46  (3004516, 3004616]    1.0      3004516\n",
      "47  (3004616, 3004716]    0.0      3004616\n",
      "48  (3004716, 3004816]    0.0      3004716\n",
      "49  (3004816, 3004916]   13.0      3004816\n",
      "(67177, 3)\n",
      "bin_offset_0    category\n",
      "TT_S0            float64\n",
      "starting_nt        int64\n",
      "dtype: object\n",
      "sumos ilgis: 67177\n",
      "ranges ilgis: 67177\n",
      "Model input - NO SCALE\n",
      "       TT_S0  genes  transcripts  exons  CDSs\n",
      "0        0.0      0            0      0     0\n",
      "1        7.0      0            0      0     0\n",
      "2        1.0      0            0      0     0\n",
      "3        2.0      0            0      0     0\n",
      "4        0.0      0            0      0     0\n",
      "...      ...    ...          ...    ...   ...\n",
      "67172    0.0      0            0      0     0\n",
      "67173    0.0      0            0      0     0\n",
      "67174    0.0      0            0      0     0\n",
      "67175    1.0      0            0      0     0\n",
      "67176    1.0      0            0      0     0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "       TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0        0.0             0                   0             0            0\n",
      "1        7.0             0                   0             0            0\n",
      "2        1.0             0                   0             0            0\n",
      "3        2.0             0                   0             0            0\n",
      "4        0.0             0                   0             0            0\n",
      "...      ...           ...                 ...           ...          ...\n",
      "67172    0.0             0                   0             0            0\n",
      "67173    0.0             0                   0             0            0\n",
      "67174    0.0             0                   0             0            0\n",
      "67175    1.0             0                   0             0            0\n",
      "67176    1.0             0                   0             0            0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "0.052309570239814224\n",
      "[0 2 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 2 0 7 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 2 0 0 2 0 0 0 0 0 6 0 2 9 9 2 7 6 3 3 8 8 0 6 6 0 0 0 2 0 0 2 0 0 2\n",
      " 0 0 0 0 0 0 0 2 0 0 0 0 0 0 2 0 0 6 6 0 0 0 0 0 0 2 6 6 1 7 6 2 2 2 1 0 6\n",
      " 0 6 0 0 0 0 0 0 0 0 0 2 0 0 6 0 6 6 6 6 0 2 2 0 0 2 2 0 0 0 0 0 0 0 0 0 0\n",
      " 6 0 2 0 0 0 2 0 6 0 0 0 6 0 6 0 0 2 6 0 0 0 0 0 0 0 0 0 0 0 0 0 0 2 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 6 7 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 2 0 0 0 0 0 0 0 0 0 0 0 0 0 0 2 2 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 2 0 0 0 6 0 0 0 0 0 0 0 0 0 6 6 0 6 2 6 2 0 0 0 2 2\n",
      " 0 0 2 0 0 0 6 2 6 0 2 2 0 0 0 0 0 0 2 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 2 0 0 0 0 0 0 0 0 0 0 0 0 0 0 2 0 0 0 2 2 0 0 0 2 2 0 2 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 6 2 0 0 0 2 0 0 2 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 2 0 0 0 0 0 2 0 0 0 0 2\n",
      " 0 0 0 0 0 0 0 0 0 0 2 0 0 0 2 0 0 0 0 0 0 2 0 0 0 0 0 0 0 0 2 0 0 2 0 0 2\n",
      " 0 2 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 2 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 6 0 0 2 0 0 0 0 2 2 2 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 2 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 2 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 2 0 0 0 0 2 0 0 0 2\n",
      " 2 0 2 2 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 6 0 0 0 0 0 2 0 0 0 0 0 0 0\n",
      " 0]\n",
      "threshold density 0.15152264201500534\n",
      "cluster signal 40053.0\n",
      "cluster length 2751100\n",
      "!!cluster density 0.014558903711242776\n",
      "cluster signal 117505.0\n",
      "cluster length 1623600\n",
      "!!cluster density 0.07237312145848732\n",
      "cluster signal 119719.0\n",
      "cluster length 580500\n",
      "!!cluster density 0.20623428079242032\n",
      "cluster signal 148606.0\n",
      "cluster length 1088200\n",
      "!!cluster density 0.1365612938798015\n",
      "cluster signal 76216.0\n",
      "cluster length 197600\n",
      "!!cluster density 0.3857085020242915\n",
      "cluster signal 9968.0\n",
      "cluster length 10200\n",
      "!!cluster density 0.9772549019607844\n",
      "cluster signal 678.0\n",
      "cluster length 200\n",
      "!!cluster density 3.39\n",
      "cluster signal 99791.0\n",
      "cluster length 351400\n",
      "!!cluster density 0.28398121798520204\n",
      "cluster signal 42357.0\n",
      "cluster length 81200\n",
      "!!cluster density 0.5216379310344827\n",
      "cluster signal 23686.0\n",
      "cluster length 33700\n",
      "!!cluster density 0.7028486646884273\n",
      "\n",
      "Labels mapping\n",
      " {0: 0, 2: 0, 7: 1, 6: 0, 9: 1, 3: 1, 8: 1, 1: 1, 4: 1, 5: 1}\n",
      "DEBUG: 54629 12548 \n",
      "\n",
      " +--------------------+-------------+---------------+--------------+-----------+-----------+-----------+---------------+-----------+-----------+-----------+\n",
      "| bin_offset_0       | TT_S0       | starting_nt   | Chromosome   | Start     | End       | genes     | transcripts   | exons     | CDSs      | labels    |\n",
      "| (category)         | (float64)   | (int64)       | (category)   | (int64)   | (int64)   | (int64)   | (int64)       | (int64)   | (int64)   | (int64)   |\n",
      "|--------------------+-------------+---------------+--------------+-----------+-----------+-----------+---------------+-----------+-----------+-----------|\n",
      "| (3000016, 3000016] | 0.0         | 3000016       | 18           | 3000016   | 3000016   | 0         | 0             | 0         | 0         | 0         |\n",
      "| (3000016, 3000116] | 7.0         | 3000016       | 18           | 3000016   | 3000116   | 0         | 0             | 0         | 0         | 0         |\n",
      "| (3000116, 3000216] | 1.0         | 3000116       | 18           | 3000116   | 3000216   | 0         | 0             | 0         | 0         | 0         |\n",
      "| (3000216, 3000316] | 2.0         | 3000216       | 18           | 3000216   | 3000316   | 0         | 0             | 0         | 0         | 0         |\n",
      "| ...                | ...         | ...           | ...          | ...       | ...       | ...       | ...           | ...       | ...       | ...       |\n",
      "| (9717216, 9717316] | 0.0         | 9717216       | 18           | 9717216   | 9717316   | 0         | 0             | 0         | 0         | 0         |\n",
      "| (9717316, 9717416] | 0.0         | 9717316       | 18           | 9717316   | 9717416   | 0         | 0             | 0         | 0         | 0         |\n",
      "| (9717416, 9717516] | 1.0         | 9717416       | 18           | 9717416   | 9717516   | 0         | 0             | 0         | 0         | 0         |\n",
      "| (9717516, 9717593] | 1.0         | 9717516       | 18           | 9717516   | 9717593   | 0         | 0             | 0         | 0         | 0         |\n",
      "+--------------------+-------------+---------------+--------------+-----------+-----------+-----------+---------------+-----------+-----------+-----------+\n",
      "Unstranded PyRanges object has 54,629 rows and 11 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "os.environ['COLUMNS']=\"300\"\n",
    "\n",
    "result = evaluator.evaluate(merged_ranges = chromatin_algos.algo5d_aggregated(1, n_clusters=10))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "6201148c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0\n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "BINS  IntervalIndex([(3000016, 3000016], (3000016, 3000116], (3000116, 3000216],\n",
      "               (3000216, 3000316], (3000316, 3000416], (3000416, 3000516],\n",
      "               (3000516, 3000616], (3000616, 3000716], (3000716, 3000816],\n",
      "               (3000816, 3000916],\n",
      "               ...\n",
      "               (9716616, 9716716], (9716716, 9716816], (9716816, 9716916],\n",
      "               (9716916, 9717016], (9717016, 9717116], (9717116, 9717216],\n",
      "               (9717216, 9717316], (9717316, 9717416], (9717416, 9717516],\n",
      "               (9717516, 9717593]],\n",
      "              dtype='interval[int64, right]', length=67177)\n",
      "HEAD with BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0   \n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "\n",
      "         bin_offset_0  \n",
      "0  (3000016, 3000116]  \n",
      "1  (3000016, 3000116]  \n",
      "2  (3000016, 3000116]  \n",
      "3  (3000016, 3000116]  \n",
      "4  (3000016, 3000116]  \n",
      "TAIL with BINS\n",
      "       seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  9717555  9717555  18_9717555    1.0    0.0    0.0   \n",
      "699996       19  9717571  9717571  19_9717571    0.0    0.0    0.0   \n",
      "699997       19  9717581  9717581  19_9717581    0.0    0.0    0.0   \n",
      "699998       18  9717582  9717582  18_9717582    0.0    0.0    0.0   \n",
      "699999       18  9717593  9717593  18_9717593    0.0    0.0    0.0   \n",
      "\n",
      "              bin_offset_0  \n",
      "699995  (9717516, 9717593]  \n",
      "699996  (9717516, 9717593]  \n",
      "699997  (9717516, 9717593]  \n",
      "699998  (9717516, 9717593]  \n",
      "699999  (9717516, 9717593]  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:136: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "          bin_offset_0  TT_S0  starting_nt\n",
      "0   (3000016, 3000016]    0.0      3000016\n",
      "1   (3000016, 3000116]    7.0      3000016\n",
      "2   (3000116, 3000216]    1.0      3000116\n",
      "3   (3000216, 3000316]    2.0      3000216\n",
      "4   (3000316, 3000416]    0.0      3000316\n",
      "5   (3000416, 3000516]    0.0      3000416\n",
      "6   (3000516, 3000616]    0.0      3000516\n",
      "7   (3000616, 3000716]    0.0      3000616\n",
      "8   (3000716, 3000816]    0.0      3000716\n",
      "9   (3000816, 3000916]    0.0      3000816\n",
      "10  (3000916, 3001016]    0.0      3000916\n",
      "11  (3001016, 3001116]    2.0      3001016\n",
      "12  (3001116, 3001216]    0.0      3001116\n",
      "13  (3001216, 3001316]    0.0      3001216\n",
      "14  (3001316, 3001416]    0.0      3001316\n",
      "15  (3001416, 3001516]    1.0      3001416\n",
      "16  (3001516, 3001616]    3.0      3001516\n",
      "17  (3001616, 3001716]    3.0      3001616\n",
      "18  (3001716, 3001816]    0.0      3001716\n",
      "19  (3001816, 3001916]    0.0      3001816\n",
      "20  (3001916, 3002016]    1.0      3001916\n",
      "21  (3002016, 3002116]    0.0      3002016\n",
      "22  (3002116, 3002216]    1.0      3002116\n",
      "23  (3002216, 3002316]   10.0      3002216\n",
      "24  (3002316, 3002416]    2.0      3002316\n",
      "25  (3002416, 3002516]   19.0      3002416\n",
      "26  (3002516, 3002616]    4.0      3002516\n",
      "27  (3002616, 3002716]    0.0      3002616\n",
      "28  (3002716, 3002816]    0.0      3002716\n",
      "29  (3002816, 3002916]    0.0      3002816\n",
      "30  (3002916, 3003016]    1.0      3002916\n",
      "31  (3003016, 3003116]    2.0      3003016\n",
      "32  (3003116, 3003216]    0.0      3003116\n",
      "33  (3003216, 3003316]    0.0      3003216\n",
      "34  (3003316, 3003416]    1.0      3003316\n",
      "35  (3003416, 3003516]    0.0      3003416\n",
      "36  (3003516, 3003616]    1.0      3003516\n",
      "37  (3003616, 3003716]    1.0      3003616\n",
      "38  (3003716, 3003816]    3.0      3003716\n",
      "39  (3003816, 3003916]    0.0      3003816\n",
      "40  (3003916, 3004016]    8.0      3003916\n",
      "41  (3004016, 3004116]    0.0      3004016\n",
      "42  (3004116, 3004216]    0.0      3004116\n",
      "43  (3004216, 3004316]    5.0      3004216\n",
      "44  (3004316, 3004416]    0.0      3004316\n",
      "45  (3004416, 3004516]    0.0      3004416\n",
      "46  (3004516, 3004616]    1.0      3004516\n",
      "47  (3004616, 3004716]    0.0      3004616\n",
      "48  (3004716, 3004816]    0.0      3004716\n",
      "49  (3004816, 3004916]   13.0      3004816\n",
      "(67177, 3)\n",
      "bin_offset_0    category\n",
      "TT_S0            float64\n",
      "starting_nt        int64\n",
      "dtype: object\n",
      "sumos ilgis: 67177\n",
      "ranges ilgis: 67177\n",
      "Model input - NO SCALE\n",
      "       TT_S0  genes  transcripts  exons  CDSs\n",
      "0        0.0      0            0      0     0\n",
      "1        7.0      0            0      0     0\n",
      "2        1.0      0            0      0     0\n",
      "3        2.0      0            0      0     0\n",
      "4        0.0      0            0      0     0\n",
      "...      ...    ...          ...    ...   ...\n",
      "67172    0.0      0            0      0     0\n",
      "67173    0.0      0            0      0     0\n",
      "67174    0.0      0            0      0     0\n",
      "67175    1.0      0            0      0     0\n",
      "67176    1.0      0            0      0     0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "       TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0        0.0             0                   0             0            0\n",
      "1        7.0             0                   0             0            0\n",
      "2        1.0             0                   0             0            0\n",
      "3        2.0             0                   0             0            0\n",
      "4        0.0             0                   0             0            0\n",
      "...      ...           ...                 ...           ...          ...\n",
      "67172    0.0             0                   0             0            0\n",
      "67173    0.0             0                   0             0            0\n",
      "67174    0.0             0                   0             0            0\n",
      "67175    1.0             0                   0             0            0\n",
      "67176    1.0             0                   0             0            0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "0.031885913333432575\n",
      "[0 6 0 0 0 0 0 0 0 0 0 0 0 0 0 0 6 6 0 0 0 0 0 2 0 7 6 0 0 0 0 0 0 0 0 0 0\n",
      " 0 6 0 6 0 0 6 0 0 0 0 0 2 0 6 8 1 6 7 2 9 9 4 4 0 7 7 6 0 0 6 0 0 6 6 0 6\n",
      " 0 0 0 0 0 0 0 2 0 0 6 6 0 6 6 0 0 2 2 0 0 0 6 0 6 2 2 2 5 5 2 6 6 2 5 6 2\n",
      " 0 2 6 0 0 0 0 0 0 0 0 6 0 0 2 0 7 7 2 7 0 6 6 6 0 6 2 6 0 0 0 0 0 0 0 0 6\n",
      " 2 0 2 6 0 0 6 0 2 6 0 0 7 0 2 0 0 2 2 6 0 0 0 0 0 0 0 0 0 0 0 0 6 2 0 6 0\n",
      " 0 6 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 2 7 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 6 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 6 0 0 0 0 0 6 0 0 0 0 0 0 6 0 6 6 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 6 0 0 0 2 0 0 0 0 0 0 0 0 0 2 7 0 2 2 2 6 6 0 0 6 6\n",
      " 6 0 6 0 0 0 2 6 2 6 2 6 0 6 0 0 0 0 6 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 6 0 0 0 6 0 0 0 0 0 6 0 0 0 0 0 0 0 6 2 0 0 0 6 6 6 0 6 6 6 6 6 0 0 6 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 2 6 0 0 0 6 0 0 6 0 6 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 6 0 0 6 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 6 0 0 0 0 0 6 0 0 0 0 6\n",
      " 0 0 0 6 0 0 0 0 0 6 2 0 0 6 6 0 0 0 0 0 0 6 0 0 0 0 0 0 0 6 6 0 0 6 6 0 6\n",
      " 0 6 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 6 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 6 0 6 6 0 6 0 0 0 0 0 0 0 0 0 0 6 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 6 0 0 0 0 0 6 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 6 0 0 0 0 0 2 0 0 6 0 0 0 0 2 6 2 0 6 0 0 6 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 6 6 6 0 0 6 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 6 0 0 0 0 6 0 6 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 6 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 6 6 6 0 0 0 0 6 6 0 0 6\n",
      " 6 0 6 6 0 6 0 0 0 0 0 0 0 6 6 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 6 6 0\n",
      " 0 0 0 0 0 0 6 0 6 0 0 0 0 0 0 0 0 0 0 0 0 0 0 2 6 0 0 6 0 6 6 0 0 0 0 0 0\n",
      " 0]\n",
      "threshold density 0.15152264201500534\n",
      "cluster signal 13441.0\n",
      "cluster length 1985500\n",
      "!!cluster density 0.006769579451019895\n",
      "cluster signal 101917.0\n",
      "cluster length 1944100\n",
      "!!cluster density 0.0524237436345867\n",
      "cluster signal 149985.0\n",
      "cluster length 1285800\n",
      "!!cluster density 0.11664722351843211\n",
      "cluster signal 130512.0\n",
      "cluster length 700100\n",
      "!!cluster density 0.18641908298814455\n",
      "cluster signal 42683.0\n",
      "cluster length 84200\n",
      "!!cluster density 0.5069239904988123\n",
      "cluster signal 80542.0\n",
      "cluster length 214200\n",
      "!!cluster density 0.3760130718954248\n",
      "cluster signal 11221.0\n",
      "cluster length 11700\n",
      "!!cluster density 0.959059829059829\n",
      "cluster signal 678.0\n",
      "cluster length 200\n",
      "!!cluster density 3.39\n",
      "cluster signal 121173.0\n",
      "cluster length 453100\n",
      "!!cluster density 0.26743103067755464\n",
      "cluster signal 26427.0\n",
      "cluster length 38800\n",
      "!!cluster density 0.6811082474226804\n",
      "\n",
      "Labels mapping\n",
      " {0: 0, 6: 0, 2: 0, 7: 1, 8: 1, 1: 1, 9: 1, 4: 1, 5: 1, 3: 1}\n",
      "DEBUG: 52154 15023 \n",
      "\n",
      " +--------------------+-------------+---------------+--------------+-------+\n",
      "| bin_offset_0       | TT_S0       | starting_nt   | Chromosome   | +7    |\n",
      "| (category)         | (float64)   | (int64)       | (category)   | ...   |\n",
      "|--------------------+-------------+---------------+--------------+-------|\n",
      "| (3000016, 3000016] | 0.0         | 3000016       | 18           | ...   |\n",
      "| (3000016, 3000116] | 7.0         | 3000016       | 18           | ...   |\n",
      "| (3000116, 3000216] | 1.0         | 3000116       | 18           | ...   |\n",
      "| (3000216, 3000316] | 2.0         | 3000216       | 18           | ...   |\n",
      "| ...                | ...         | ...           | ...          | ...   |\n",
      "| (9717216, 9717316] | 0.0         | 9717216       | 18           | ...   |\n",
      "| (9717316, 9717416] | 0.0         | 9717316       | 18           | ...   |\n",
      "| (9717416, 9717516] | 1.0         | 9717416       | 18           | ...   |\n",
      "| (9717516, 9717593] | 1.0         | 9717516       | 18           | ...   |\n",
      "+--------------------+-------------+---------------+--------------+-------+\n",
      "Unstranded PyRanges object has 52,154 rows and 11 columns from 1 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome.\n",
      "7 hidden columns: Start, End, genes, transcripts, exons, CDSs, labels\n",
      "count                         7724.0\n",
      "mean                       194.49767\n",
      "std                       183.945899\n",
      "min                            100.0\n",
      "25%                            100.0\n",
      "50%                            100.0\n",
      "75%                            200.0\n",
      "max                           2400.0\n",
      "mode                           100.0\n",
      "ATACseq_overlap             0.719428\n",
      "ATACseq_ranges_overlap      0.012798\n",
      "ATACseq_Jaccard             0.012734\n",
      "DNAseq_overlap              0.605445\n",
      "DNAseq_ranges_overlap       0.021434\n",
      "DNAseq_Jaccard              0.021139\n",
      "genes_overlap               0.541727\n",
      "percentage_open             0.154596\n",
      "percentage_open_adj         0.223637\n",
      "model_name                          \n",
      "Name: Length, dtype: object\n"
     ]
    }
   ],
   "source": [
    "import generate_consensus\n",
    "\n",
    "ranges = chromatin_algos.algo5d_aggregated(1, 10)\n",
    "result = evaluator.evaluate(merged_ranges = ranges)\n",
    "print(result)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "6b5c5de3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "count                         7724.0\n",
      "mean                       194.49767\n",
      "std                       183.945899\n",
      "min                            100.0\n",
      "25%                            100.0\n",
      "50%                            100.0\n",
      "75%                            200.0\n",
      "max                           2400.0\n",
      "mode                           100.0\n",
      "ATACseq_overlap             0.719428\n",
      "ATACseq_ranges_overlap      0.012798\n",
      "ATACseq_Jaccard             0.012734\n",
      "DNAseq_overlap              0.605445\n",
      "DNAseq_ranges_overlap       0.021434\n",
      "DNAseq_Jaccard              0.021139\n",
      "genes_overlap               0.541727\n",
      "percentage_open             0.154596\n",
      "percentage_open_adj         0.223637\n",
      "model_name                          \n",
      "Name: Length, dtype: object\n"
     ]
    }
   ],
   "source": [
    "result = evaluator.evaluate(merged_ranges = ranges)\n",
    "\n",
    "print(result)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "d67fb278",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.series.Series'>\n"
     ]
    }
   ],
   "source": [
    "print(type(result['mode']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "9435baf8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1615\n",
      "0\n"
     ]
    }
   ],
   "source": [
    "import pyranges as pr\n",
    "\n",
    "ranges.Length = ranges.End - ranges.Start\n",
    "ranges_df = ranges.df\n",
    "ranges_df['']\n",
    "\n",
    "print(len(ranges[ranges.Length < 75]))\n",
    "ranges2 = evaluator.process_ranges(ranges)\n",
    "ranges2.Length = ranges2.End - ranges2.Start\n",
    "print(len(ranges2[ranges2.Length < 75]))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "a929baa0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "count                         8182.0\n",
      "mean                      182.306282\n",
      "std                       197.351081\n",
      "min                              5.0\n",
      "25%                             80.0\n",
      "50%                            115.0\n",
      "75%                            210.0\n",
      "max                           2085.0\n",
      "ATACseq_overlap             0.751796\n",
      "ATACseq_ranges_overlap      0.013469\n",
      "DNAseq_overlap              0.639008\n",
      "DNAseq_ranges_overlap       0.022784\n",
      "genes_overlap                0.54393\n",
      "percentage_open             0.153498\n",
      "percentage_open_adj         0.222048\n",
      "model_name                          \n",
      "Name: Length, dtype: object\n"
     ]
    }
   ],
   "source": [
    "print(evaluator.evaluate(ranges))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "61959364",
   "metadata": {},
   "outputs": [],
   "source": [
    "def calculate_gaps(pyranges_obj):\n",
    "    \"\"\"\n",
    "    Calculate gaps between ranges in a PyRanges object.\n",
    "\n",
    "    Parameters:\n",
    "        pyranges_obj (pr.PyRanges): A PyRanges object with genomic ranges.\n",
    "\n",
    "    Returns:\n",
    "        pr.PyRanges: A PyRanges object containing the gaps.\n",
    "    \"\"\"\n",
    "    # Convert PyRanges to a DataFrame\n",
    "    df = pyranges_obj.df\n",
    "    \n",
    "    # Sort the ranges by Chromosome and Start position\n",
    "    df = df.sort_values([\"Chromosome\", \"Start\"]).reset_index(drop=True)\n",
    "    \n",
    "    # Shift End and Start columns to calculate gaps\n",
    "    prev_end = df.groupby(\"Chromosome\")[\"End\"].shift(0)   # Current End\n",
    "    next_start = df.groupby(\"Chromosome\")[\"Start\"].shift(-1)  # Next Start\n",
    "    \n",
    "    # Identify gaps where End < Next Start\n",
    "    gaps_df = df[(prev_end < next_start)].assign(Start=prev_end, End=next_start)[[\"Chromosome\", \"Start\", \"End\"]]\n",
    "    \n",
    "    # Drop NaN rows caused by shifting and ensure integer type\n",
    "    gaps_df = gaps_df.dropna().astype({\"Start\": \"int\", \"End\": \"int\"})\n",
    "    \n",
    "    # Convert back to PyRanges\n",
    "    return pr.PyRanges(gaps_df)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "cbaf97f0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "+--------------+-----------+-----------+--------------+------------+\n",
      "| Chromosome   |     Start |       End | Strand       | title      |\n",
      "| (category)   |   (int64) |   (int64) | (category)   | (object)   |\n",
      "|--------------+-----------+-----------+--------------+------------|\n",
      "| chr1         |         5 |        10 | +            | a          |\n",
      "| chr1         |        20 |        28 | +            | b          |\n",
      "| chr1         |        80 |        95 | -            | c          |\n",
      "| chr3         |        10 |        38 | +            | d          |\n",
      "+--------------+-----------+-----------+--------------+------------+\n",
      "Stranded PyRanges object has 4 rows and 5 columns from 2 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome and Strand.\n",
      "+--------------+-----------+-----------+--------------+\n",
      "| Chromosome   |     Start |       End | Strand       |\n",
      "| (category)   |   (int64) |   (int64) | (category)   |\n",
      "|--------------+-----------+-----------+--------------|\n",
      "| chr1         |         5 |        28 | +            |\n",
      "| chr1         |        80 |        95 | -            |\n",
      "| chr3         |        10 |        38 | +            |\n",
      "+--------------+-----------+-----------+--------------+\n",
      "Stranded PyRanges object has 3 rows and 4 columns from 2 chromosomes.\n",
      "For printing, the PyRanges was sorted on Chromosome and Strand.\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "df=pd.DataFrame(\n",
    "{'Chromosome':['chr1', 'chr1', 'chr1', 'chr3'],\n",
    " 'Start': [5, 20, 80, 10],\n",
    " 'End':   [10, 28, 95, 38],\n",
    " 'Strand':['+', '+', '-', '+'],\n",
    " 'title': ['a', 'b', 'c', 'd']}\n",
    ")\n",
    "\n",
    "p=pr.PyRanges(df)\n",
    "print(p)\n",
    "p2=p.merge(slack = 10)\n",
    "print(p2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "847f9a77",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "443\n",
      "0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/y9/8yssfmz161s4ghy99nf1f2cw0000gn/T/ipykernel_27239/19855938.py:18: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  prev_end = df.groupby(\"Chromosome\")[\"End\"].shift(0)   # Current End\n",
      "/var/folders/y9/8yssfmz161s4ghy99nf1f2cw0000gn/T/ipykernel_27239/19855938.py:19: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  next_start = df.groupby(\"Chromosome\")[\"Start\"].shift(-1)  # Next Start\n",
      "/var/folders/y9/8yssfmz161s4ghy99nf1f2cw0000gn/T/ipykernel_27239/19855938.py:18: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  prev_end = df.groupby(\"Chromosome\")[\"End\"].shift(0)   # Current End\n",
      "/var/folders/y9/8yssfmz161s4ghy99nf1f2cw0000gn/T/ipykernel_27239/19855938.py:19: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  next_start = df.groupby(\"Chromosome\")[\"Start\"].shift(-1)  # Next Start\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import pyranges as pr\n",
    "\n",
    "ranges_df = ranges.df\n",
    "gaps = calculate_gaps(ranges)\n",
    "gaps.Length = gaps.End - gaps.Start\n",
    "print(len(gaps[gaps.Length <= 50]))\n",
    "\n",
    "ranges2 = evaluator.process_ranges(ranges)\n",
    "gaps2 = calculate_gaps(ranges2)\n",
    "gaps2.Length = gaps2.End - gaps2.Start\n",
    "print(len(gaps2[gaps2.Length <= 50]))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "28e1cdda",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Runnning scale 1\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0\n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "BINS  IntervalIndex([(3000016, 3000016], (3000016, 3000116], (3000116, 3000216],\n",
      "               (3000216, 3000316], (3000316, 3000416], (3000416, 3000516],\n",
      "               (3000516, 3000616], (3000616, 3000716], (3000716, 3000816],\n",
      "               (3000816, 3000916],\n",
      "               ...\n",
      "               (9716616, 9716716], (9716716, 9716816], (9716816, 9716916],\n",
      "               (9716916, 9717016], (9717016, 9717116], (9717116, 9717216],\n",
      "               (9717216, 9717316], (9717316, 9717416], (9717416, 9717516],\n",
      "               (9717516, 9717593]],\n",
      "              dtype='interval[int64, right]', length=67177)\n",
      "HEAD w BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0   \n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "\n",
      "         bin_offset_0  \n",
      "0  (3000016, 3000116]  \n",
      "1  (3000016, 3000116]  \n",
      "2  (3000016, 3000116]  \n",
      "3  (3000016, 3000116]  \n",
      "4  (3000016, 3000116]  \n",
      "TAIL w BINS\n",
      "       seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  9717555  9717555  18_9717555    1.0    0.0    0.0   \n",
      "699996       19  9717571  9717571  19_9717571    0.0    0.0    0.0   \n",
      "699997       19  9717581  9717581  19_9717581    0.0    0.0    0.0   \n",
      "699998       18  9717582  9717582  18_9717582    0.0    0.0    0.0   \n",
      "699999       18  9717593  9717593  18_9717593    0.0    0.0    0.0   \n",
      "\n",
      "              bin_offset_0  \n",
      "699995  (9717516, 9717593]  \n",
      "699996  (9717516, 9717593]  \n",
      "699997  (9717516, 9717593]  \n",
      "699998  (9717516, 9717593]  \n",
      "699999  (9717516, 9717593]  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:151: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "    TT_S0  starting_nt\n",
      "0     0.0      3000016\n",
      "1     7.0      3000016\n",
      "2     1.0      3000116\n",
      "3     2.0      3000216\n",
      "4     0.0      3000316\n",
      "5     0.0      3000416\n",
      "6     0.0      3000516\n",
      "7     0.0      3000616\n",
      "8     0.0      3000716\n",
      "9     0.0      3000816\n",
      "10    0.0      3000916\n",
      "11    2.0      3001016\n",
      "12    0.0      3001116\n",
      "13    0.0      3001216\n",
      "14    0.0      3001316\n",
      "15    1.0      3001416\n",
      "16    3.0      3001516\n",
      "17    3.0      3001616\n",
      "18    0.0      3001716\n",
      "19    0.0      3001816\n",
      "20    1.0      3001916\n",
      "21    0.0      3002016\n",
      "22    1.0      3002116\n",
      "23   10.0      3002216\n",
      "24    2.0      3002316\n",
      "25   19.0      3002416\n",
      "26    4.0      3002516\n",
      "27    0.0      3002616\n",
      "28    0.0      3002716\n",
      "29    0.0      3002816\n",
      "30    1.0      3002916\n",
      "31    2.0      3003016\n",
      "32    0.0      3003116\n",
      "33    0.0      3003216\n",
      "34    1.0      3003316\n",
      "35    0.0      3003416\n",
      "36    1.0      3003516\n",
      "37    1.0      3003616\n",
      "38    3.0      3003716\n",
      "39    0.0      3003816\n",
      "40    8.0      3003916\n",
      "41    0.0      3004016\n",
      "42    0.0      3004116\n",
      "43    5.0      3004216\n",
      "44    0.0      3004316\n",
      "45    0.0      3004416\n",
      "46    1.0      3004516\n",
      "47    0.0      3004616\n",
      "48    0.0      3004716\n",
      "49   13.0      3004816\n",
      "(67177, 2)\n",
      "TT_S0          float64\n",
      "starting_nt      int64\n",
      "dtype: object\n",
      "sumos ilgis: 67177\n",
      "ranges ilgis: 67177\n",
      "Model input - NO SCALE\n",
      "       TT_S0  genes  transcripts  exons  CDSs\n",
      "0        0.0      0            0      0     0\n",
      "1        7.0      0            0      0     0\n",
      "2        1.0      0            0      0     0\n",
      "3        2.0      0            0      0     0\n",
      "4        0.0      0            0      0     0\n",
      "...      ...    ...          ...    ...   ...\n",
      "67172    0.0      0            0      0     0\n",
      "67173    0.0      0            0      0     0\n",
      "67174    0.0      0            0      0     0\n",
      "67175    1.0      0            0      0     0\n",
      "67176    1.0      0            0      0     0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "       TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0        0.0             0                   0             0            0\n",
      "1        7.0             0                   0             0            0\n",
      "2        1.0             0                   0             0            0\n",
      "3        2.0             0                   0             0            0\n",
      "4        0.0             0                   0             0            0\n",
      "...      ...           ...                 ...           ...          ...\n",
      "67172    0.0             0                   0             0            0\n",
      "67173    0.0             0                   0             0            0\n",
      "67174    0.0             0                   0             0            0\n",
      "67175    1.0             0                   0             0            0\n",
      "67176    1.0             0                   0             0            0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "0.17092159518882952\n",
      "[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 1 0 1 1 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 1 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0]\n",
      "Runnning scale 2\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0\n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "BINS  IntervalIndex([(3000016, 3000016], (3000016, 3000116], (3000116, 3000216],\n",
      "               (3000216, 3000316], (3000316, 3000416], (3000416, 3000516],\n",
      "               (3000516, 3000616], (3000616, 3000716], (3000716, 3000816],\n",
      "               (3000816, 3000916],\n",
      "               ...\n",
      "               (9716616, 9716716], (9716716, 9716816], (9716816, 9716916],\n",
      "               (9716916, 9717016], (9717016, 9717116], (9717116, 9717216],\n",
      "               (9717216, 9717316], (9717316, 9717416], (9717416, 9717516],\n",
      "               (9717516, 9717593]],\n",
      "              dtype='interval[int64, right]', length=67177)\n",
      "HEAD w BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0   \n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "\n",
      "         bin_offset_0  \n",
      "0  (3000016, 3000116]  \n",
      "1  (3000016, 3000116]  \n",
      "2  (3000016, 3000116]  \n",
      "3  (3000016, 3000116]  \n",
      "4  (3000016, 3000116]  \n",
      "TAIL w BINS\n",
      "       seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  9717555  9717555  18_9717555    1.0    0.0    0.0   \n",
      "699996       19  9717571  9717571  19_9717571    0.0    0.0    0.0   \n",
      "699997       19  9717581  9717581  19_9717581    0.0    0.0    0.0   \n",
      "699998       18  9717582  9717582  18_9717582    0.0    0.0    0.0   \n",
      "699999       18  9717593  9717593  18_9717593    0.0    0.0    0.0   \n",
      "\n",
      "              bin_offset_0  \n",
      "699995  (9717516, 9717593]  \n",
      "699996  (9717516, 9717593]  \n",
      "699997  (9717516, 9717593]  \n",
      "699998  (9717516, 9717593]  \n",
      "699999  (9717516, 9717593]  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:151: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "    TT_S0  starting_nt\n",
      "0     0.0      3000016\n",
      "1     7.0      3000016\n",
      "2     1.0      3000116\n",
      "3     2.0      3000216\n",
      "4     0.0      3000316\n",
      "5     0.0      3000416\n",
      "6     0.0      3000516\n",
      "7     0.0      3000616\n",
      "8     0.0      3000716\n",
      "9     0.0      3000816\n",
      "10    0.0      3000916\n",
      "11    2.0      3001016\n",
      "12    0.0      3001116\n",
      "13    0.0      3001216\n",
      "14    0.0      3001316\n",
      "15    1.0      3001416\n",
      "16    3.0      3001516\n",
      "17    3.0      3001616\n",
      "18    0.0      3001716\n",
      "19    0.0      3001816\n",
      "20    1.0      3001916\n",
      "21    0.0      3002016\n",
      "22    1.0      3002116\n",
      "23   10.0      3002216\n",
      "24    2.0      3002316\n",
      "25   19.0      3002416\n",
      "26    4.0      3002516\n",
      "27    0.0      3002616\n",
      "28    0.0      3002716\n",
      "29    0.0      3002816\n",
      "30    1.0      3002916\n",
      "31    2.0      3003016\n",
      "32    0.0      3003116\n",
      "33    0.0      3003216\n",
      "34    1.0      3003316\n",
      "35    0.0      3003416\n",
      "36    1.0      3003516\n",
      "37    1.0      3003616\n",
      "38    3.0      3003716\n",
      "39    0.0      3003816\n",
      "40    8.0      3003916\n",
      "41    0.0      3004016\n",
      "42    0.0      3004116\n",
      "43    5.0      3004216\n",
      "44    0.0      3004316\n",
      "45    0.0      3004416\n",
      "46    1.0      3004516\n",
      "47    0.0      3004616\n",
      "48    0.0      3004716\n",
      "49   13.0      3004816\n",
      "(67177, 2)\n",
      "TT_S0          float64\n",
      "starting_nt      int64\n",
      "dtype: object\n",
      "sumos ilgis: 67177\n",
      "ranges ilgis: 67177\n",
      "Model input - NO SCALE\n",
      "       TT_S0  genes  transcripts  exons  CDSs\n",
      "0        0.0      0            0      0     0\n",
      "1        7.0      0            0      0     0\n",
      "2        1.0      0            0      0     0\n",
      "3        2.0      0            0      0     0\n",
      "4        0.0      0            0      0     0\n",
      "...      ...    ...          ...    ...   ...\n",
      "67172    0.0      0            0      0     0\n",
      "67173    0.0      0            0      0     0\n",
      "67174    0.0      0            0      0     0\n",
      "67175    1.0      0            0      0     0\n",
      "67176    1.0      0            0      0     0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "       TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0        0.0             0                   0             0            0\n",
      "1        7.0             0                   0             0            0\n",
      "2        1.0             0                   0             0            0\n",
      "3        2.0             0                   0             0            0\n",
      "4        0.0             0                   0             0            0\n",
      "...      ...           ...                 ...           ...          ...\n",
      "67172    0.0             0                   0             0            0\n",
      "67173    0.0             0                   0             0            0\n",
      "67174    0.0             0                   0             0            0\n",
      "67175    1.0             0                   0             0            0\n",
      "67176    1.0             0                   0             0            0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "0.20496598538190156\n",
      "[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 1 0 1 1 1 1 0 0 1 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 1 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0]\n",
      "Runnning scale 4\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0\n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "BINS  IntervalIndex([(3000016, 3000016], (3000016, 3000116], (3000116, 3000216],\n",
      "               (3000216, 3000316], (3000316, 3000416], (3000416, 3000516],\n",
      "               (3000516, 3000616], (3000616, 3000716], (3000716, 3000816],\n",
      "               (3000816, 3000916],\n",
      "               ...\n",
      "               (9716616, 9716716], (9716716, 9716816], (9716816, 9716916],\n",
      "               (9716916, 9717016], (9717016, 9717116], (9717116, 9717216],\n",
      "               (9717216, 9717316], (9717316, 9717416], (9717416, 9717516],\n",
      "               (9717516, 9717593]],\n",
      "              dtype='interval[int64, right]', length=67177)\n",
      "HEAD w BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0   \n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "\n",
      "         bin_offset_0  \n",
      "0  (3000016, 3000116]  \n",
      "1  (3000016, 3000116]  \n",
      "2  (3000016, 3000116]  \n",
      "3  (3000016, 3000116]  \n",
      "4  (3000016, 3000116]  \n",
      "TAIL w BINS\n",
      "       seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  9717555  9717555  18_9717555    1.0    0.0    0.0   \n",
      "699996       19  9717571  9717571  19_9717571    0.0    0.0    0.0   \n",
      "699997       19  9717581  9717581  19_9717581    0.0    0.0    0.0   \n",
      "699998       18  9717582  9717582  18_9717582    0.0    0.0    0.0   \n",
      "699999       18  9717593  9717593  18_9717593    0.0    0.0    0.0   \n",
      "\n",
      "              bin_offset_0  \n",
      "699995  (9717516, 9717593]  \n",
      "699996  (9717516, 9717593]  \n",
      "699997  (9717516, 9717593]  \n",
      "699998  (9717516, 9717593]  \n",
      "699999  (9717516, 9717593]  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:151: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "    TT_S0  starting_nt\n",
      "0     0.0      3000016\n",
      "1     7.0      3000016\n",
      "2     1.0      3000116\n",
      "3     2.0      3000216\n",
      "4     0.0      3000316\n",
      "5     0.0      3000416\n",
      "6     0.0      3000516\n",
      "7     0.0      3000616\n",
      "8     0.0      3000716\n",
      "9     0.0      3000816\n",
      "10    0.0      3000916\n",
      "11    2.0      3001016\n",
      "12    0.0      3001116\n",
      "13    0.0      3001216\n",
      "14    0.0      3001316\n",
      "15    1.0      3001416\n",
      "16    3.0      3001516\n",
      "17    3.0      3001616\n",
      "18    0.0      3001716\n",
      "19    0.0      3001816\n",
      "20    1.0      3001916\n",
      "21    0.0      3002016\n",
      "22    1.0      3002116\n",
      "23   10.0      3002216\n",
      "24    2.0      3002316\n",
      "25   19.0      3002416\n",
      "26    4.0      3002516\n",
      "27    0.0      3002616\n",
      "28    0.0      3002716\n",
      "29    0.0      3002816\n",
      "30    1.0      3002916\n",
      "31    2.0      3003016\n",
      "32    0.0      3003116\n",
      "33    0.0      3003216\n",
      "34    1.0      3003316\n",
      "35    0.0      3003416\n",
      "36    1.0      3003516\n",
      "37    1.0      3003616\n",
      "38    3.0      3003716\n",
      "39    0.0      3003816\n",
      "40    8.0      3003916\n",
      "41    0.0      3004016\n",
      "42    0.0      3004116\n",
      "43    5.0      3004216\n",
      "44    0.0      3004316\n",
      "45    0.0      3004416\n",
      "46    1.0      3004516\n",
      "47    0.0      3004616\n",
      "48    0.0      3004716\n",
      "49   13.0      3004816\n",
      "(67177, 2)\n",
      "TT_S0          float64\n",
      "starting_nt      int64\n",
      "dtype: object\n",
      "sumos ilgis: 67177\n",
      "ranges ilgis: 67177\n",
      "Model input - NO SCALE\n",
      "       TT_S0  genes  transcripts  exons  CDSs\n",
      "0        0.0      0            0      0     0\n",
      "1        7.0      0            0      0     0\n",
      "2        1.0      0            0      0     0\n",
      "3        2.0      0            0      0     0\n",
      "4        0.0      0            0      0     0\n",
      "...      ...    ...          ...    ...   ...\n",
      "67172    0.0      0            0      0     0\n",
      "67173    0.0      0            0      0     0\n",
      "67174    0.0      0            0      0     0\n",
      "67175    1.0      0            0      0     0\n",
      "67176    1.0      0            0      0     0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "       TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0        0.0             0                   0             0            0\n",
      "1        7.0             0                   0             0            0\n",
      "2        1.0             0                   0             0            0\n",
      "3        2.0             0                   0             0            0\n",
      "4        0.0             0                   0             0            0\n",
      "...      ...           ...                 ...           ...          ...\n",
      "67172    0.0             0                   0             0            0\n",
      "67173    0.0             0                   0             0            0\n",
      "67174    0.0             0                   0             0            0\n",
      "67175    1.0             0                   0             0            0\n",
      "67176    1.0             0                   0             0            0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "0.20496598538190156\n",
      "[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 1 0 1 1 1 1 0 0 1 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 1 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0]\n",
      "Runnning scale 8\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0\n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "BINS  IntervalIndex([(3000016, 3000016], (3000016, 3000116], (3000116, 3000216],\n",
      "               (3000216, 3000316], (3000316, 3000416], (3000416, 3000516],\n",
      "               (3000516, 3000616], (3000616, 3000716], (3000716, 3000816],\n",
      "               (3000816, 3000916],\n",
      "               ...\n",
      "               (9716616, 9716716], (9716716, 9716816], (9716816, 9716916],\n",
      "               (9716916, 9717016], (9717016, 9717116], (9717116, 9717216],\n",
      "               (9717216, 9717316], (9717316, 9717416], (9717416, 9717516],\n",
      "               (9717516, 9717593]],\n",
      "              dtype='interval[int64, right]', length=67177)\n",
      "HEAD w BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0   \n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "\n",
      "         bin_offset_0  \n",
      "0  (3000016, 3000116]  \n",
      "1  (3000016, 3000116]  \n",
      "2  (3000016, 3000116]  \n",
      "3  (3000016, 3000116]  \n",
      "4  (3000016, 3000116]  \n",
      "TAIL w BINS\n",
      "       seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  9717555  9717555  18_9717555    1.0    0.0    0.0   \n",
      "699996       19  9717571  9717571  19_9717571    0.0    0.0    0.0   \n",
      "699997       19  9717581  9717581  19_9717581    0.0    0.0    0.0   \n",
      "699998       18  9717582  9717582  18_9717582    0.0    0.0    0.0   \n",
      "699999       18  9717593  9717593  18_9717593    0.0    0.0    0.0   \n",
      "\n",
      "              bin_offset_0  \n",
      "699995  (9717516, 9717593]  \n",
      "699996  (9717516, 9717593]  \n",
      "699997  (9717516, 9717593]  \n",
      "699998  (9717516, 9717593]  \n",
      "699999  (9717516, 9717593]  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:151: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "    TT_S0  starting_nt\n",
      "0     0.0      3000016\n",
      "1     7.0      3000016\n",
      "2     1.0      3000116\n",
      "3     2.0      3000216\n",
      "4     0.0      3000316\n",
      "5     0.0      3000416\n",
      "6     0.0      3000516\n",
      "7     0.0      3000616\n",
      "8     0.0      3000716\n",
      "9     0.0      3000816\n",
      "10    0.0      3000916\n",
      "11    2.0      3001016\n",
      "12    0.0      3001116\n",
      "13    0.0      3001216\n",
      "14    0.0      3001316\n",
      "15    1.0      3001416\n",
      "16    3.0      3001516\n",
      "17    3.0      3001616\n",
      "18    0.0      3001716\n",
      "19    0.0      3001816\n",
      "20    1.0      3001916\n",
      "21    0.0      3002016\n",
      "22    1.0      3002116\n",
      "23   10.0      3002216\n",
      "24    2.0      3002316\n",
      "25   19.0      3002416\n",
      "26    4.0      3002516\n",
      "27    0.0      3002616\n",
      "28    0.0      3002716\n",
      "29    0.0      3002816\n",
      "30    1.0      3002916\n",
      "31    2.0      3003016\n",
      "32    0.0      3003116\n",
      "33    0.0      3003216\n",
      "34    1.0      3003316\n",
      "35    0.0      3003416\n",
      "36    1.0      3003516\n",
      "37    1.0      3003616\n",
      "38    3.0      3003716\n",
      "39    0.0      3003816\n",
      "40    8.0      3003916\n",
      "41    0.0      3004016\n",
      "42    0.0      3004116\n",
      "43    5.0      3004216\n",
      "44    0.0      3004316\n",
      "45    0.0      3004416\n",
      "46    1.0      3004516\n",
      "47    0.0      3004616\n",
      "48    0.0      3004716\n",
      "49   13.0      3004816\n",
      "(67177, 2)\n",
      "TT_S0          float64\n",
      "starting_nt      int64\n",
      "dtype: object\n",
      "sumos ilgis: 67177\n",
      "ranges ilgis: 67177\n",
      "Model input - NO SCALE\n",
      "       TT_S0  genes  transcripts  exons  CDSs\n",
      "0        0.0      0            0      0     0\n",
      "1        7.0      0            0      0     0\n",
      "2        1.0      0            0      0     0\n",
      "3        2.0      0            0      0     0\n",
      "4        0.0      0            0      0     0\n",
      "...      ...    ...          ...    ...   ...\n",
      "67172    0.0      0            0      0     0\n",
      "67173    0.0      0            0      0     0\n",
      "67174    0.0      0            0      0     0\n",
      "67175    1.0      0            0      0     0\n",
      "67176    1.0      0            0      0     0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "       TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0        0.0             0                   0             0            0\n",
      "1        7.0             0                   0             0            0\n",
      "2        1.0             0                   0             0            0\n",
      "3        2.0             0                   0             0            0\n",
      "4        0.0             0                   0             0            0\n",
      "...      ...           ...                 ...           ...          ...\n",
      "67172    0.0             0                   0             0            0\n",
      "67173    0.0             0                   0             0            0\n",
      "67174    0.0             0                   0             0            0\n",
      "67175    1.0             0                   0             0            0\n",
      "67176    1.0             0                   0             0            0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "0.20496598538190156\n",
      "[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 1 0 1 1 1 1 0 0 1 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 1 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0]\n",
      "Runnning scale 16\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0\n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "BINS  IntervalIndex([(3000016, 3000016], (3000016, 3000116], (3000116, 3000216],\n",
      "               (3000216, 3000316], (3000316, 3000416], (3000416, 3000516],\n",
      "               (3000516, 3000616], (3000616, 3000716], (3000716, 3000816],\n",
      "               (3000816, 3000916],\n",
      "               ...\n",
      "               (9716616, 9716716], (9716716, 9716816], (9716816, 9716916],\n",
      "               (9716916, 9717016], (9717016, 9717116], (9717116, 9717216],\n",
      "               (9717216, 9717316], (9717316, 9717416], (9717416, 9717516],\n",
      "               (9717516, 9717593]],\n",
      "              dtype='interval[int64, right]', length=67177)\n",
      "HEAD w BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0   \n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "\n",
      "         bin_offset_0  \n",
      "0  (3000016, 3000116]  \n",
      "1  (3000016, 3000116]  \n",
      "2  (3000016, 3000116]  \n",
      "3  (3000016, 3000116]  \n",
      "4  (3000016, 3000116]  \n",
      "TAIL w BINS\n",
      "       seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  9717555  9717555  18_9717555    1.0    0.0    0.0   \n",
      "699996       19  9717571  9717571  19_9717571    0.0    0.0    0.0   \n",
      "699997       19  9717581  9717581  19_9717581    0.0    0.0    0.0   \n",
      "699998       18  9717582  9717582  18_9717582    0.0    0.0    0.0   \n",
      "699999       18  9717593  9717593  18_9717593    0.0    0.0    0.0   \n",
      "\n",
      "              bin_offset_0  \n",
      "699995  (9717516, 9717593]  \n",
      "699996  (9717516, 9717593]  \n",
      "699997  (9717516, 9717593]  \n",
      "699998  (9717516, 9717593]  \n",
      "699999  (9717516, 9717593]  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:151: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "    TT_S0  starting_nt\n",
      "0     0.0      3000016\n",
      "1     7.0      3000016\n",
      "2     1.0      3000116\n",
      "3     2.0      3000216\n",
      "4     0.0      3000316\n",
      "5     0.0      3000416\n",
      "6     0.0      3000516\n",
      "7     0.0      3000616\n",
      "8     0.0      3000716\n",
      "9     0.0      3000816\n",
      "10    0.0      3000916\n",
      "11    2.0      3001016\n",
      "12    0.0      3001116\n",
      "13    0.0      3001216\n",
      "14    0.0      3001316\n",
      "15    1.0      3001416\n",
      "16    3.0      3001516\n",
      "17    3.0      3001616\n",
      "18    0.0      3001716\n",
      "19    0.0      3001816\n",
      "20    1.0      3001916\n",
      "21    0.0      3002016\n",
      "22    1.0      3002116\n",
      "23   10.0      3002216\n",
      "24    2.0      3002316\n",
      "25   19.0      3002416\n",
      "26    4.0      3002516\n",
      "27    0.0      3002616\n",
      "28    0.0      3002716\n",
      "29    0.0      3002816\n",
      "30    1.0      3002916\n",
      "31    2.0      3003016\n",
      "32    0.0      3003116\n",
      "33    0.0      3003216\n",
      "34    1.0      3003316\n",
      "35    0.0      3003416\n",
      "36    1.0      3003516\n",
      "37    1.0      3003616\n",
      "38    3.0      3003716\n",
      "39    0.0      3003816\n",
      "40    8.0      3003916\n",
      "41    0.0      3004016\n",
      "42    0.0      3004116\n",
      "43    5.0      3004216\n",
      "44    0.0      3004316\n",
      "45    0.0      3004416\n",
      "46    1.0      3004516\n",
      "47    0.0      3004616\n",
      "48    0.0      3004716\n",
      "49   13.0      3004816\n",
      "(67177, 2)\n",
      "TT_S0          float64\n",
      "starting_nt      int64\n",
      "dtype: object\n",
      "sumos ilgis: 67177\n",
      "ranges ilgis: 67177\n",
      "Model input - NO SCALE\n",
      "       TT_S0  genes  transcripts  exons  CDSs\n",
      "0        0.0      0            0      0     0\n",
      "1        7.0      0            0      0     0\n",
      "2        1.0      0            0      0     0\n",
      "3        2.0      0            0      0     0\n",
      "4        0.0      0            0      0     0\n",
      "...      ...    ...          ...    ...   ...\n",
      "67172    0.0      0            0      0     0\n",
      "67173    0.0      0            0      0     0\n",
      "67174    0.0      0            0      0     0\n",
      "67175    1.0      0            0      0     0\n",
      "67176    1.0      0            0      0     0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "       TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0        0.0             0                   0             0            0\n",
      "1        7.0             0                   0             0            0\n",
      "2        1.0             0                   0             0            0\n",
      "3        2.0             0                   0             0            0\n",
      "4        0.0             0                   0             0            0\n",
      "...      ...           ...                 ...           ...          ...\n",
      "67172    0.0             0                   0             0            0\n",
      "67173    0.0             0                   0             0            0\n",
      "67174    0.0             0                   0             0            0\n",
      "67175    1.0             0                   0             0            0\n",
      "67176    1.0             0                   0             0            0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "0.20496598538190156\n",
      "[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 1 0 1 1 1 1 0 0 1 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 1 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0]\n",
      "Runnning scale 32\n",
      "HEAD\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0\n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0\n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0\n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0\n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0\n",
      "BINS  IntervalIndex([(3000016, 3000016], (3000016, 3000116], (3000116, 3000216],\n",
      "               (3000216, 3000316], (3000316, 3000416], (3000416, 3000516],\n",
      "               (3000516, 3000616], (3000616, 3000716], (3000716, 3000816],\n",
      "               (3000816, 3000916],\n",
      "               ...\n",
      "               (9716616, 9716716], (9716716, 9716816], (9716816, 9716916],\n",
      "               (9716916, 9717016], (9717016, 9717116], (9717116, 9717216],\n",
      "               (9717216, 9717316], (9717316, 9717416], (9717416, 9717516],\n",
      "               (9717516, 9717593]],\n",
      "              dtype='interval[int64, right]', length=67177)\n",
      "HEAD w BINS\n",
      "  seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "0       19  3000017  3000017  19_3000017    0.0    0.0    0.0   \n",
      "1       18  3000019  3000019  18_3000019    0.0    0.0    0.0   \n",
      "2       18  3000023  3000023  18_3000023    0.0    0.0    0.0   \n",
      "3       18  3000025  3000025  18_3000025    0.0    0.0    0.0   \n",
      "4       18  3000040  3000040  18_3000040    4.0    0.0    0.0   \n",
      "\n",
      "         bin_offset_0  \n",
      "0  (3000016, 3000116]  \n",
      "1  (3000016, 3000116]  \n",
      "2  (3000016, 3000116]  \n",
      "3  (3000016, 3000116]  \n",
      "4  (3000016, 3000116]  \n",
      "TAIL w BINS\n",
      "       seqnames    start      end       CG_ID  TT_S0  TT_S1  TT_S2  \\\n",
      "699995       18  9717555  9717555  18_9717555    1.0    0.0    0.0   \n",
      "699996       19  9717571  9717571  19_9717571    0.0    0.0    0.0   \n",
      "699997       19  9717581  9717581  19_9717581    0.0    0.0    0.0   \n",
      "699998       18  9717582  9717582  18_9717582    0.0    0.0    0.0   \n",
      "699999       18  9717593  9717593  18_9717593    0.0    0.0    0.0   \n",
      "\n",
      "              bin_offset_0  \n",
      "699995  (9717516, 9717593]  \n",
      "699996  (9717516, 9717593]  \n",
      "699997  (9717516, 9717593]  \n",
      "699998  (9717516, 9717593]  \n",
      "699999  (9717516, 9717593]  \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lukas/VU/bio-baigiamasis/scripts/chromatin_algos.py:151: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  sumos = df.groupby('bin_offset_' + str(bin_offset)).sum().reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMOS\n",
      "    TT_S0  starting_nt\n",
      "0     0.0      3000016\n",
      "1     7.0      3000016\n",
      "2     1.0      3000116\n",
      "3     2.0      3000216\n",
      "4     0.0      3000316\n",
      "5     0.0      3000416\n",
      "6     0.0      3000516\n",
      "7     0.0      3000616\n",
      "8     0.0      3000716\n",
      "9     0.0      3000816\n",
      "10    0.0      3000916\n",
      "11    2.0      3001016\n",
      "12    0.0      3001116\n",
      "13    0.0      3001216\n",
      "14    0.0      3001316\n",
      "15    1.0      3001416\n",
      "16    3.0      3001516\n",
      "17    3.0      3001616\n",
      "18    0.0      3001716\n",
      "19    0.0      3001816\n",
      "20    1.0      3001916\n",
      "21    0.0      3002016\n",
      "22    1.0      3002116\n",
      "23   10.0      3002216\n",
      "24    2.0      3002316\n",
      "25   19.0      3002416\n",
      "26    4.0      3002516\n",
      "27    0.0      3002616\n",
      "28    0.0      3002716\n",
      "29    0.0      3002816\n",
      "30    1.0      3002916\n",
      "31    2.0      3003016\n",
      "32    0.0      3003116\n",
      "33    0.0      3003216\n",
      "34    1.0      3003316\n",
      "35    0.0      3003416\n",
      "36    1.0      3003516\n",
      "37    1.0      3003616\n",
      "38    3.0      3003716\n",
      "39    0.0      3003816\n",
      "40    8.0      3003916\n",
      "41    0.0      3004016\n",
      "42    0.0      3004116\n",
      "43    5.0      3004216\n",
      "44    0.0      3004316\n",
      "45    0.0      3004416\n",
      "46    1.0      3004516\n",
      "47    0.0      3004616\n",
      "48    0.0      3004716\n",
      "49   13.0      3004816\n",
      "(67177, 2)\n",
      "TT_S0          float64\n",
      "starting_nt      int64\n",
      "dtype: object\n",
      "sumos ilgis: 67177\n",
      "ranges ilgis: 67177\n",
      "Model input - NO SCALE\n",
      "       TT_S0  genes  transcripts  exons  CDSs\n",
      "0        0.0      0            0      0     0\n",
      "1        7.0      0            0      0     0\n",
      "2        1.0      0            0      0     0\n",
      "3        2.0      0            0      0     0\n",
      "4        0.0      0            0      0     0\n",
      "...      ...    ...          ...    ...   ...\n",
      "67172    0.0      0            0      0     0\n",
      "67173    0.0      0            0      0     0\n",
      "67174    0.0      0            0      0     0\n",
      "67175    1.0      0            0      0     0\n",
      "67176    1.0      0            0      0     0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "Model input - SCALED\n",
      "       TT_S0  genes_scaled  transcripts_scaled  exons_scaled  CDSs_scaled\n",
      "0        0.0             0                   0             0            0\n",
      "1        7.0             0                   0             0            0\n",
      "2        1.0             0                   0             0            0\n",
      "3        2.0             0                   0             0            0\n",
      "4        0.0             0                   0             0            0\n",
      "...      ...           ...                 ...           ...          ...\n",
      "67172    0.0             0                   0             0            0\n",
      "67173    0.0             0                   0             0            0\n",
      "67174    0.0             0                   0             0            0\n",
      "67175    1.0             0                   0             0            0\n",
      "67176    1.0             0                   0             0            0\n",
      "\n",
      "[67177 rows x 5 columns]\n",
      "0.17092159518882952\n",
      "[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 1 0 1 1 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 1 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0]\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "for i in range(6):\n",
    "    scale = 2 ** i\n",
    "    print(\"Runnning scale\", scale)\n",
    "    evaluation = evaluator.evaluate(dataframe = chromatin_algos.algo5d(scale), model_name=f\"k-means 5 features scale={scale}\")\n",
    "    result = pd.concat([result, evaluation], axis=1)\n",
    "\n",
    "result.to_csv('../outputs/results1.csv', sep = '\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ca1280ed",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                         Length                      Length  \\\n",
      "count                    7301.0                      6456.0   \n",
      "mean                 188.590604                  177.850062   \n",
      "std                  174.998864                  157.257952   \n",
      "min                       100.0                       100.0   \n",
      "25%                       100.0                       100.0   \n",
      "50%                       100.0                       100.0   \n",
      "75%                       200.0                       200.0   \n",
      "max                      2400.0                      2000.0   \n",
      "ATACseq_overlap        0.697949                    0.668276   \n",
      "DNAseq_overlap         0.595329                    0.553813   \n",
      "genes_overlap          0.544833                    0.550809   \n",
      "percentage_open        0.141691                    0.118157   \n",
      "percentage_open_adj    0.204969                    0.170924   \n",
      "model_name                       k-means 5 features scale=1   \n",
      "min_open_signal            17.0                        19.0   \n",
      "max_open_signal           393.0                       393.0   \n",
      "min_closed_signal           0.0                         0.0   \n",
      "max_closed_signal          16.0                        18.0   \n",
      "\n",
      "                                         Length                      Length  \\\n",
      "count                                    7301.0                      7301.0   \n",
      "mean                                 188.590604                  188.590604   \n",
      "std                                  174.998864                  174.998864   \n",
      "min                                       100.0                       100.0   \n",
      "25%                                       100.0                       100.0   \n",
      "50%                                       100.0                       100.0   \n",
      "75%                                       200.0                       200.0   \n",
      "max                                      2400.0                      2400.0   \n",
      "ATACseq_overlap                        0.697949                    0.697949   \n",
      "DNAseq_overlap                         0.595329                    0.595329   \n",
      "genes_overlap                          0.544833                    0.544833   \n",
      "percentage_open                        0.141691                    0.141691   \n",
      "percentage_open_adj                    0.204969                    0.204969   \n",
      "model_name           k-means 5 features scale=2  k-means 5 features scale=4   \n",
      "min_open_signal                            17.0                        17.0   \n",
      "max_open_signal                           393.0                       393.0   \n",
      "min_closed_signal                           0.0                         0.0   \n",
      "max_closed_signal                          16.0                        16.0   \n",
      "\n",
      "                                         Length                       Length  \\\n",
      "count                                    7301.0                       7301.0   \n",
      "mean                                 188.590604                   188.590604   \n",
      "std                                  174.998864                   174.998864   \n",
      "min                                       100.0                        100.0   \n",
      "25%                                       100.0                        100.0   \n",
      "50%                                       100.0                        100.0   \n",
      "75%                                       200.0                        200.0   \n",
      "max                                      2400.0                       2400.0   \n",
      "ATACseq_overlap                        0.697949                     0.697949   \n",
      "DNAseq_overlap                         0.595329                     0.595329   \n",
      "genes_overlap                          0.544833                     0.544833   \n",
      "percentage_open                        0.141691                     0.141691   \n",
      "percentage_open_adj                    0.204969                     0.204969   \n",
      "model_name           k-means 5 features scale=8  k-means 5 features scale=16   \n",
      "min_open_signal                            17.0                         17.0   \n",
      "max_open_signal                           393.0                        393.0   \n",
      "min_closed_signal                           0.0                          0.0   \n",
      "max_closed_signal                          16.0                         16.0   \n",
      "\n",
      "                                          Length  \n",
      "count                                     6456.0  \n",
      "mean                                  177.850062  \n",
      "std                                   157.257952  \n",
      "min                                        100.0  \n",
      "25%                                        100.0  \n",
      "50%                                        100.0  \n",
      "75%                                        200.0  \n",
      "max                                       2000.0  \n",
      "ATACseq_overlap                         0.668276  \n",
      "DNAseq_overlap                          0.553813  \n",
      "genes_overlap                           0.550809  \n",
      "percentage_open                         0.118157  \n",
      "percentage_open_adj                     0.170924  \n",
      "model_name           k-means 5 features scale=32  \n",
      "min_open_signal                             19.0  \n",
      "max_open_signal                            393.0  \n",
      "min_closed_signal                            0.0  \n",
      "max_closed_signal                           18.0  \n"
     ]
    }
   ],
   "source": [
    "print(result)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
